<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Castro-Giner" />

<meta name="date" content="2025-02-07" />

<title>p26532_o26674 10x RNA-seq analysis of BR16 tumor and mets</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.30/datatables.js"></script>
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.13.4/jszip.min.js"></script>
<link href="site_libs/dt-ext-buttons-1.13.4/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.13.4/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.print.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">met-paths</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheAcetoLab/sznurkowska-met-paths">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">p26532_o26674 10x RNA-seq analysis of BR16
tumor and mets</h1>
<h3 class="subtitle">Analysis of cancer cells in primary tumor, lung and
liver metastases</h3>
<h4 class="author">Francesc Castro-Giner</h4>
<h4 class="date">February 07, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-07
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>sznurkowska-met-paths/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250206code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250206)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250206code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250206)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsznurkowskametpathstree8427f4a878c7175f5894265c9157cc804f1000cdtargetblank8427f4aa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/tree/8427f4a878c7175f5894265c9157cc804f1000cd" target="_blank">8427f4a</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsznurkowskametpathstree8427f4a878c7175f5894265c9157cc804f1000cdtargetblank8427f4aa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/tree/8427f4a878c7175f5894265c9157cc804f1000cd" target="_blank">8427f4a</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/patients/
    Ignored:    data/resources/
    Ignored:    data/rnaseq/
    Ignored:    data/seer/
    Ignored:    output/p26532_o26674/
    Ignored:    output/p26532_o28268/
    Ignored:    output/p26532_o28268_o34980/
    Ignored:    output/p26532_o34980/

Untracked files:
    Untracked:  analysis/mvt1-tumor-10x_rnaseq-qc.Rmd
    Untracked:  analysis/templates/
    Untracked:  code/R-functions/
    Untracked:  configuration/
    Untracked:  workflowr_update.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   README.md
    Modified:   sznurkowska-met-paths.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/br16-10x_rnaseq-cancer_cells-main.Rmd</code>) and HTML
(<code>docs/br16-10x_rnaseq-cancer_cells-main.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/8427f4a878c7175f5894265c9157cc804f1000cd/analysis/br16-10x_rnaseq-cancer_cells-main.Rmd" target="_blank">8427f4a</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
<td>
Add Br16 analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries-additional-functions-and-data"
class="section level2" number="1">
<h2><span class="header-section-number">1</span> Load libraries,
additional functions and data</h2>
<p>Setup environment</p>
<pre class="r"><code>knitr::opts_chunk$set(results=&#39;asis&#39;, echo=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.align = &#39;center&#39;, fig.width = 3.5, fig.asp = 0.618, dpi = 600, dev = c(&quot;png&quot;, &quot;pdf&quot;), fig.showtext = FALSE)

options(stringsAsFactors = FALSE)

use_seed &lt;- 1100101
set.seed(use_seed)

if(!dir.exists(params$output_dir))
  dir.create(params$output_dir, recursive = TRUE, showWarnings = FALSE)</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(tidyverse)
library(showtext)
library(foreach)
library(DT)
library(knitr)
library(kableExtra)
library(cowplot)
library(colorblindr)
library(ggbeeswarm)
library(arsenal)
library(RColorBrewer)
library(ggpubr)
library(pheatmap)
library(patchwork)
library(openxlsx)
library(magrittr)

library(scater)
library(DropletUtils)
library(Seurat)
library(scran)
library(BiocSingular)
library(batchelor)
library(bluster)
library(scDblFinder)
library(speckle)
library(miloR)
library(ComplexHeatmap)
library(clusterProfiler)
library(GSVA)
library(circlize)
library(corrplot)
library(factoextra)
library(SingleR)</code></pre>
<p>Load ggplot theme</p>
<pre class="r"><code>source(&quot;./configuration/rmarkdown/ggplot_theme.R&quot;)
source(&quot;./configuration/rmarkdown/color_palettes.R&quot;)</code></pre>
<p>Load custom functions</p>
<pre class="r"><code>source(&#39;./code/R-functions/gse_omnibus.r&#39;)
source(&#39;./code/R-functions/gse_report.r&#39;)
clean_msigdb_names &lt;- function(x) x %&gt;% gsub(&#39;REACTOME_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;WP_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;BIOCARTA_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;KEGG_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;PID_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;GOBP_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;_&#39;, &#39; &#39;, .)
source(&#39;./code/R-functions/pairwise_terms_jc.r&#39;)</code></pre>
<p>Clean files generated in previous runs</p>
<pre class="r"><code>rmd_file &lt;- current_input()
if(!is.null(rmd_file)) {
  figures_dir &lt;- file.path(&#39;./docs/figure&#39;,rmd_file)
  if(dir.exists(figures_dir)) {
    unlink(file.path(figures_dir, &quot;*&quot;))
  }
}</code></pre>
</div>
<div id="data-processing" class="section level2" number="2">
<h2><span class="header-section-number">2</span> Data processing</h2>
<div id="batch-integration" class="section level3" number="2.1">
<h3><span class="header-section-number">2.1</span> Batch
integration</h3>
<p>Here we prepare the data for batch correction (next section). First,
we combine cells from all SCE objects into a single SCE and return
pre-computed per-gene variance at batch level. Then we adjust for batch
level library size. Returned object will contain log-normalized
expression values after adjusting the size factors for systematic
differences in coverage between</p>
<pre class="r"><code># File paths to SCE files
sce_files &lt;- system(paste(&quot;find&quot;, params$cellranger_count_dir, &quot;-maxdepth 2 -type f -regex &#39;.*/sce_normalized[.]rds$&#39;&quot;), intern = TRUE) %&gt;% sort

# Define files to include. Only liver, primary_tumor and lung. We also exclude 4_Br16_GFP_luc_Tum_B due to bad quality
sce_files &lt;- grep(&quot;1_Br16_GFP_luc_Tum_C_D|2_Br16_GFP_luc_Liver_B_C_D|3_Br16_GFP_luc_Lungs_B_C_D&quot;, sce_files, value = TRUE)

# Combine all SCE files into sce_comb and generate an combine gene  with pre-computed gene variance 
if(exists(&#39;sce_comb&#39;)) 
  rm(sce_comb)
sce_file &lt;-  sce_files[1]
precomputed_geneVar &lt;- foreach(sce_file = sce_files) %do% {
  use_sce &lt;- readRDS(sce_file)

  use_sce$cluster_id_batch &lt;- as.character(use_sce$cluster_id_batch)
  if(ncol(use_sce) == 0) {
    NULL
  } else {
    # Combine sce
    if(!exists(&#39;sce_comb&#39;)) {
      sce_comb &lt;- use_sce
    } else {
      new_counts &lt;- cbind(counts(sce_comb), counts(use_sce))
      new_logcounts &lt;- cbind(logcounts(sce_comb), logcounts(use_sce))
      new_colData &lt;- rbind(colData(sce_comb), colData(use_sce))
      sce_comb &lt;- SingleCellExperiment(
        assays = list(counts = new_counts, logcounts = new_logcounts),
        colData = new_colData,
        rowData = rowData(sce_comb)
      )
    }
    metadata(use_sce)$dec
  }
}
metadata(sce_comb)$precomputed_geneVar &lt;- compact(precomputed_geneVar)
rm(use_sce)

# Modify coldata and rownames
sce_comb$batch &lt;-  gsub(&quot;_.*&quot;, &quot;&quot;, sce_comb$Sample)
sce_comb$cluster_id_batch &lt;-  paste(sce_comb$batch, sce_comb$cluster_id_batch, sep = &#39;_&#39;) %&gt;% as.factor()
colData(sce_comb) &lt;- colData(sce_comb) %&gt;% data.frame %&gt;% 
  mutate(
    site = ifelse(grepl(&quot;_Tum_&quot;, Sample), &quot;primary_tumor&quot;, Sample),
    site = ifelse(grepl(&quot;_Liver_&quot;, site),&quot;liver&quot;, site),
    site = ifelse(grepl(&quot;_Lungs_&quot;, site), &quot;lung&quot;, site)
  ) %&gt;% 
  DataFrame
rownames(sce_comb) &lt;- make.names(rowData(sce_comb)$gene_name, unique = TRUE)

# Rescale each batch to adjust for differences in sequencing depth between batches. The multiBatchNorm() function recomputes log-normalized expression values after adjusting the size factors for systematic differences in coverage between SingleCellExperiment objects. (Size factors only remove biases between cells within a single batch.) This improves the quality of the correction by removing one aspect of the technical differences between batches.
sce_comb &lt;- multiBatchNorm(sce_comb, batch = sce_comb$Sample, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))

# Identifying a set of HVGs using stats from all batches.
combined_dec &lt;- combineVar(metadata(sce_comb)$precomputed_geneVar)
# combined_hvg &lt;- getTopHVGs(combined_dec, n=5000, row.names = FALSE)
chosen_hvgs &lt;- combined_dec$bio &gt; 0 &amp; !rowData(sce_comb)$is.mito &amp; !rowData(sce_comb)$is.ribo

# Dimensionality reduction
sce_comb &lt;- runPCA(sce_comb, subset_row=chosen_hvgs, BSPARAM=BiocSingular::RandomParam(), BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
sce_comb$cluster_id  &lt;- clusterCells(sce_comb, use.dimred=&quot;PCA&quot;, BLUSPARAM=SNNGraphParam(k=25, type=&quot;rank&quot;, cluster.fun=&quot;louvain&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)))
tab &lt;- table(Cluster=sce_comb$cluster_id, Batch=sce_comb$Sample)
sce_comb &lt;- runTSNE(sce_comb, dimred=&quot;PCA&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
sce_comb &lt;- runUMAP(sce_comb, dimred=&quot;PCA&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))

saveRDS(sce_comb, file.path(params$output_dir, &#39;sce_integrated.rds&#39;))</code></pre>
</div>
<div id="additional-doublet-detection" class="section level3"
number="2.2">
<h3><span class="header-section-number">2.2</span> Additional doublet
detection</h3>
<p><a href="https://github.com/plger/scDblFinder">scDblFinder</a> : We
suggest using the cluster-based approach when the datasets are
segregated into clear clusters, and the random one for the rest
(e.g. developmental trajectories).</p>
<pre class="r"><code>sce_comb &lt;- scDblFinder(
  sce_comb, 
  samples =sce_comb$Sample, 
  BPPARAM=BiocParallel::MulticoreParam(params$ncores/2, RNGseed=use_seed))

saveRDS(sce_comb, file.path(params$output_dir, &#39;sce_integrated.rds&#39;))</code></pre>
</div>
<div id="batch-correction-using-mnn" class="section level3"
number="2.3">
<h3><span class="header-section-number">2.3</span> Batch correction
using MNN</h3>
<p>Here, we actually perform the correction itself. We also perform the
clustering.</p>
<pre class="r"><code>if(!exists(&#39;sce_comb&#39;))
  sce_comb &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated.rds&#39;))

# Identifying a set of HVGs using stats from all batches.
combined_dec &lt;- combineVar(metadata(sce_comb)$precomputed_geneVar)
# combined_hvg &lt;- getTopHVGs(combined_dec, n=2000, row.names = FALSE)
chosen_hvgs &lt;- combined_dec$bio &gt; 0 &amp; !rowData(sce_comb)$is.mito &amp; !rowData(sce_comb)$is.ribo

# MNN correction
sce_mnn &lt;- fastMNN(sce_comb,
                   batch = sce_comb$Sample, 
                   auto.merge = TRUE, 
                      subset.row=chosen_hvgs, 
                      BSPARAM=RandomParam(deferred=TRUE), 
                      BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed),
                      d=50, k=20)
# colData(sce_mnn) &lt;- colData(sce_comb)
rowData(sce_mnn) &lt;-  rowData(sce_comb)[chosen_hvgs,]

# Dimensionality reduction
sce_mnn &lt;- runTSNE(sce_mnn, dimred=&quot;corrected&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
sce_mnn &lt;- runUMAP(sce_mnn, dimred=&quot;corrected&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))

# Clustering using corrected values
sce_mnn$cluster_mnn  &lt;- clusterCells(sce_mnn, use.dimred=&quot;corrected&quot;, BLUSPARAM=SNNGraphParam(k=25, type=&quot;rank&quot;, cluster.fun=&quot;louvain&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)))

# Add corrected info to sce_comb
sce_comb$cluster_mnn &lt;- sce_mnn$cluster_mnn
reducedDim(sce_comb, &#39;corrected&#39;) &lt;- reducedDim(sce_mnn, &#39;corrected&#39;)
reducedDim(sce_comb, &#39;TSNE_corrected&#39;) &lt;- reducedDim(sce_mnn, &#39;TSNE&#39;)
reducedDim(sce_comb, &#39;UMAP_corrected&#39;) &lt;- reducedDim(sce_mnn, &#39;UMAP&#39;)
# plotUMAP(sce_mnn, color_by = &#39;cluster_mnn&#39;)

# MNN-related diagnostic examining the variance in the differences in expression between MNN pairs.
use_sce &lt;- sce_comb
rownames(use_sce) &lt;- make.names(rowData(use_sce)$gene_name, unique = TRUE)
metadata(sce_mnn)$mnnDeltaVariance &lt;- mnnDeltaVariance(
  use_sce,
  pairs=metadata(sce_mnn)$merge.info$pairs, 
  BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)
  )
rm(use_sce)

# Save SCE objects
saveRDS(sce_mnn, file.path(params$output_dir, &#39;sce_integrated_corrected-keep_doublets.rds&#39;))
saveRDS(sce_comb, file.path(params$output_dir, &#39;sce_integrated.rds&#39;))</code></pre>
</div>
<div id="remove-clusters" class="section level3" number="2.4">
<h3><span class="header-section-number">2.4</span> Remove clusters</h3>
<p>Here we remove - Clusters with a high proportion of doublets as
detected in the <a href="#additional-doublet-detection">Additional
doublet detection</a> - Clusters with high proportion of mitochondrial
reads AND low library size</p>
<pre class="r"><code>if(!exists(&#39;sce_comb&#39;))
  sce_comb &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated.rds&#39;))

# keep a version with doublets for QC visualization
saveRDS(sce_comb, file.path(params$output_dir, &#39;sce_integrated-keep_doublets.rds&#39;))

# plotReducedDim(sce_comb, dimred = &#39;UMAP_corrected&#39;, color_by = &#39;scDblFinder.score&#39;)
# plotReducedDim(sce_comb, dimred = &#39;UMAP_corrected&#39;, color_by = &#39;scDblFinder.class&#39;)

# Clusters to remove by doublet detection
cl_to_remove_dbl &lt;- table(sce_comb$cluster_mnn, sce_comb$scDblFinder.class) %&gt;% 
  prop.table(1) %&gt;% data.frame %&gt;% set_names(c(&#39;Cluster&#39;, &#39;Class&#39;, &#39;Proportion&#39;)) %&gt;% 
  filter(Class == &#39;doublet&#39; &amp; Proportion &gt; 0.25) %&gt;% 
  pull(Cluster) %&gt;% as.character

# Clusters to remove by mito pct and library size
cl_to_remove_q &lt;- colData(sce_comb) %&gt;% data.frame %&gt;% 
  group_by(cluster_mnn) %&gt;% 
  summarise(
    m_mito_pct = median(subsets_Mito_percent),
    m_sum = median(sum)
    ) %&gt;% 
  filter(m_mito_pct &gt; 15 &amp; m_sum &lt; 3000) %&gt;% 
  pull(cluster_mnn) %&gt;% as.character

# Save SCE without doublets clusters
cl_to_remove &lt;- c(cl_to_remove_dbl, cl_to_remove_q)
sce_comb &lt;- sce_comb[,!sce_comb$cluster_mnn %in% cl_to_remove]
sce_comb$cluster_mnn_doublets &lt;- droplevels(sce_comb$cluster_mnn)
saveRDS(sce_comb, file.path(params$output_dir, &#39;sce_integrated.rds&#39;))</code></pre>
</div>
<div id="batch-correction-after-doublet-removal-using-mnn"
class="section level3" number="2.5">
<h3><span class="header-section-number">2.5</span> Batch correction
after doublet removal using MNN</h3>
<p>Here, we actually perform the correction itself. We also perform the
clustering.</p>
<p>There were no clusters to remove based on doublets proportions. Since
HTO didn’t work and we couldn’t remove cells based on HTO, I remove
cells annotated as doublets. A total of 1’457 were removed</p>
<pre class="r"><code>if(!exists(&#39;sce_comb&#39;))
  sce_comb &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated.rds&#39;))

# There were no clusters to remove based on doublets proportions. Since HTO didn&#39;t work and we couldn&#39;t remove cells based on HTO, I remove cells annotated as doublets
sce_comb &lt;- sce_comb[,sce_comb$scDblFinder.class == &#39;singlet&#39;]

# Identifying a set of HVGs using stats from all batches.
combined_dec &lt;- combineVar(metadata(sce_comb)$precomputed_geneVar)
# combined_hvg &lt;- getTopHVGs(combined_dec, n=2000, row.names = FALSE)
chosen_hvgs &lt;- combined_dec$bio &gt; 0 &amp; !rowData(sce_comb)$is.mito &amp; !rowData(sce_comb)$is.ribo

# MNN correction
sce_mnn &lt;- fastMNN(sce_comb,
                   batch = sce_comb$Sample, 
                   auto.merge = TRUE, 
                      subset.row=chosen_hvgs, 
                      BSPARAM=RandomParam(deferred=TRUE), 
                      BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed),
                      d=50, k=20)
# colData(sce_mnn) &lt;- colData(sce_comb)
rowData(sce_mnn) &lt;-  rowData(sce_comb)[chosen_hvgs,]

# Dimensionality reduction
sce_mnn &lt;- runTSNE(sce_mnn, dimred=&quot;corrected&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
sce_mnn &lt;- runUMAP(sce_mnn, dimred=&quot;corrected&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))

# Clustering using corrected values
sce_mnn$cluster_mnn  &lt;- clusterCells(sce_mnn, use.dimred=&quot;corrected&quot;, BLUSPARAM=SNNGraphParam(k=25, type=&quot;rank&quot;, cluster.fun=&quot;louvain&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)))

# Add corrected info to sce_comb
sce_comb$cluster_mnn &lt;- sce_mnn$cluster_mnn
reducedDim(sce_comb, &#39;corrected&#39;) &lt;- reducedDim(sce_mnn, &#39;corrected&#39;)
reducedDim(sce_comb, &#39;TSNE_corrected&#39;) &lt;- reducedDim(sce_mnn, &#39;TSNE&#39;)
reducedDim(sce_comb, &#39;UMAP_corrected&#39;) &lt;- reducedDim(sce_mnn, &#39;UMAP&#39;)
# plotUMAP(sce_mnn, color_by = &#39;cluster_mnn&#39;)

# MNN-related diagnostic examining the variance in the differences in expression between MNN pairs.
use_sce &lt;- sce_comb
rownames(use_sce) &lt;- make.names(rowData(use_sce)$gene_name, unique = TRUE)
metadata(sce_mnn)$mnnDeltaVariance &lt;- mnnDeltaVariance(
  use_sce,
  pairs=metadata(sce_mnn)$merge.info$pairs, 
  BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)
  )
rm(use_sce)

# Save SCE objects
saveRDS(sce_mnn, file.path(params$output_dir, &#39;sce_integrated_corrected.rds&#39;))
saveRDS(sce_comb, file.path(params$output_dir, &#39;sce_integrated.rds&#39;))</code></pre>
</div>
<div id="markers-detection" class="section level3" number="2.6">
<h3><span class="header-section-number">2.6</span> Markers
detection</h3>
<p>We identify the genes that drive separation between clusters and cell
types. These marker genes allow us to assign biological meaning to each
cluster based on their functional annotation. We added batch as a
blocking factor to the model. The block argument works for all effect
sizes shown above and is robust to differences in the log-fold changes
or variance between batches. However, it assumes that each pair of
clusters is present in at least one batch. In scenarios where cells from
two clusters never co-occur in the same batch, the associated pairwise
comparison will be impossible and is ignored during calculation of
summary statistics.</p>
<pre class="r"><code>marker_info &lt;- list()

if(!exists(&#39;sce_comb&#39;))
  sce_comb &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated-tme.rds&#39;))

# Cluster markers
marker_info$cluster_markers &lt;- scoreMarkers(
  sce_comb, 
  groups = sce_comb$cluster_mnn, 
  block = sce_comb$batch,
  lfc=1, 
  full.stats=TRUE,
  row.data=rowData(sce_comb)[,c(&#39;gene_name&#39;, &#39;is.mito&#39;, &#39;is.ribo&#39;),drop=FALSE], 
  BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)
)


# Site markers
marker_info$site_markers &lt;- scoreMarkers(
  sce_comb, 
  groups = sce_comb$site, 
  lfc=1, 
  full.stats=TRUE,
  row.data=rowData(sce_comb)[,c(&#39;gene_name&#39;, &#39;is.mito&#39;, &#39;is.ribo&#39;),drop=FALSE],
  BPPARAM=BiocParallel::MulticoreParam(1, RNGseed=use_seed)
)

saveRDS(marker_info, file.path(params$output_dir, &#39;scoreMarkers.rds&#39;))</code></pre>
</div>
<div id="functional-enrichment" class="section level3" number="2.7">
<h3><span class="header-section-number">2.7</span> Functional
enrichment</h3>
<p>Run functional enrichment based on results from markers detection</p>
<pre class="r"><code># Load markers
score_markers &lt;- readRDS(file.path(params$output_dir, &#39;scoreMarkers.rds&#39;))
marker_info &lt;- score_markers$cluster_markers

# Load MSigDB files
gmt_files_symbols &lt;- list(
  msigdb.c2.cp = &#39;./data/resources/MSigDB/v2023.1.Hs/c2.cp.v2023.1.Hs.symbols.gmt&#39;,
  msigdb.c2.cp.reactome = &#39;./data/resources/MSigDB/v2023.1.Hs/c2.cp.reactome.v2023.1.Hs.symbols.gmt&#39;
)

# Load mouse annotation for Go 
library(org.Hs.eg.db)

# Run ORA using mean.AUC
i &lt;- marker_info[[1]]
gse_res&lt;- foreach(i=marker_info) %do% {
  x &lt;- i %&gt;% data.frame %&gt;% rownames_to_column(&#39;features&#39;) %&gt;% 
    mutate(
      p = 0
    )
  gse_omnibus(
    feature_names = x$features,
    p = x$p,
    fc = x$mean.AUC,
    fc_thrs = 0.25,
    return_sets = c(&#39;abs&#39;),
    
    gmt_files = gmt_files_symbols, 

    annot_db = &#39;org.Hs.eg.db&#39;,
    organism = &#39;hsapiens&#39;,
    kegg_organism = &#39;hsa&#39;,
    reactome_organism = &#39;human&#39;,
    go_ontologies = c(&#39;BP&#39;),
    
    run_all_ora = FALSE,
    run_all_gsea = FALSE,
    run_GSEA = FALSE,
    run_enricher = TRUE,
    run_enrichGO = TRUE,

    # args_gse = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1),
    args_enricher = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1, qvalueCutoff = 1),
    
    go_simplify_p = 0.1,
    go_simplify_cutoff = 0.4
    )
} 
names(gse_res) &lt;- names(marker_info)
saveRDS(gse_res, file.path(params$output_dir, &#39;scoreMarkers_cluster_gse_auc.rds&#39;))


# # Load MSigDB files
# gmt_files_symbols &lt;- list(
#   msigdb.c2.cp.reactome = &#39;./data/resources/MSigDB/v2023.1.Hs/c2.cp.reactome.v2023.1.Hs.symbols.gmt&#39;
# )
# gse_cl_auc &lt;- readRDS( file.path(params$output_dir, &#39;scoreMarkers_cluster_gse_auc.rds&#39;))
# i &lt;- marker_info[[1]]
# gse_res&lt;- foreach(i=marker_info) %do% {
#   x &lt;- i %&gt;% data.frame %&gt;% rownames_to_column(&#39;features&#39;) %&gt;% 
#     mutate(
#       p = 0
#     )
#   gse_omnibus(
#     feature_names = x$features,
#     p = x$p,
#     fc = x$mean.AUC,
#     fc_thrs = 0.25,
#     return_sets = c(&#39;abs&#39;),
#     
#     gmt_files = gmt_files_symbols, 
# 
#     annot_db = &#39;org.Hs.eg.db&#39;,
#     organism = &#39;hsapiens&#39;,
#     kegg_organism = &#39;hsa&#39;,
#     reactome_organism = &#39;human&#39;,
#     
#     run_all_ora = FALSE,
#     run_all_gsea = FALSE,
#     run_GSEA = FALSE,
#     run_enricher = TRUE,
#     run_enrichGO = FALSE,
# 
#     args_gse = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1)
#     )
# } 
# names(gse_res) &lt;- names(marker_info)
# 
# for(i in names(gse_res)) {
#   gse_cl_auc[[i]]$enricher$abs$msigdb.c2.cp.reactome &lt;- gse_res[[i]]$enricher$abs$msigdb.c2.cp.reactome 
# }
# 
# saveRDS(gse_cl_auc, file.path(params$output_dir, &#39;scoreMarkers_cluster_gse_auc.rds&#39;))</code></pre>
<p>Run functional enrichment based on results from markers detection</p>
<pre class="r"><code># Load markers
score_markers &lt;- readRDS(file.path(params$output_dir, &#39;scoreMarkers.rds&#39;))
marker_info &lt;- score_markers$site_markers

# Load MSigDB files
gmt_files_symbols &lt;- list(
  msigdb.m2.cp = &#39;./data/resources/MSigDB/v2023.1.Hs/c2.cp.v2023.1.Hs.symbols.gmt&#39;,
  msigdb.c2.cp.reactome = &#39;./data/resources/MSigDB/v2023.1.Hs/c2.cp.reactome.v2023.1.Hs.symbols.gmt&#39;
)

# Load mouse annotation for Go 
library(org.Hs.eg.db)

# Run ORA using mean.AUC
i &lt;- marker_info[[1]]
gse_res&lt;- foreach(i=marker_info) %do% {
  x &lt;- i %&gt;% data.frame %&gt;% rownames_to_column(&#39;features&#39;) %&gt;% 
    mutate(
      p = 0
    )
  gse_omnibus(
    feature_names = x$features,
    p = x$p,
    fc = x$mean.AUC,
    fc_thrs = 0.25,
    return_sets = c(&#39;abs&#39;),
    
    gmt_files = gmt_files_symbols, 

    annot_db = &#39;org.Hs.eg.db&#39;,
    organism = &#39;hsapiens&#39;,
    kegg_organism = &#39;hsa&#39;,
    reactome_organism = &#39;human&#39;,
    go_ontologies = c(&#39;BP&#39;),
    
    run_all_ora = FALSE,
    run_all_gsea = FALSE,
    run_GSEA = FALSE,
    run_enricher = TRUE,
    run_enrichGO = TRUE,

    # args_gse = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1)
    args_enricher = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1, qvalueCutoff = 1),
    
    go_simplify_p = 0.1,
    go_simplify_cutoff = 0.4
    )
} 
names(gse_res) &lt;- names(marker_info)
saveRDS(gse_res, file.path(params$output_dir, &#39;scoreMarkers_site_gse_auc.rds&#39;))
# 
# # Run GSEA using Cohen&#39;s logFC
# i &lt;- marker_info[[1]]
# gse_res&lt;- foreach(i=marker_info) %do% {
#   x &lt;- i %&gt;% data.frame %&gt;% rownames_to_column(&#39;features&#39;) %&gt;% 
#     mutate(
#       p = 0,
#       mean.logFC.cohen = ifelse(mean.logFC.cohen &lt;= -5, -5, mean.logFC.cohen),
#       mean.logFC.cohen = ifelse(mean.logFC.cohen &gt;= 5, 5, mean.logFC.cohen)
#     )
#   
#   gse_omnibus(
#     feature_names = x$features,
#     p = x$p,
#     fc = x$mean.logFC.cohen,
#     
#     gmt_files = gmt_files_symbols, 
# 
#     annot_db = &#39;org.Hs.eg.db&#39;,
#     organism = &#39;hsapiens&#39;,
#     kegg_organism = &#39;hsa&#39;,
#     reactome_organism = &#39;human&#39;,
#     
#     run_all_ora = FALSE,
#     run_all_gsea = FALSE,
#     run_GSEA = TRUE,
#     run_gseGO = TRUE,
# 
#     args_gse = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1)
#     )
# }
# names(gse_res) &lt;- names(marker_info)
# saveRDS(gse_res, file.path(params$output_dir, &#39;scoreMarkers_site_gse_cohen.rds&#39;))


# # Load MSigDB files
# gmt_files_symbols &lt;- list(
#   msigdb.c2.cp.reactome = &#39;./data/resources/MSigDB/v2023.1.Hs/c2.cp.reactome.v2023.1.Hs.symbols.gmt&#39;
# )
# gse_site_auc &lt;- readRDS( file.path(params$output_dir, &#39;scoreMarkers_site_gse_auc.rds&#39;))
# i &lt;- marker_info[[1]]
# gse_res&lt;- foreach(i=marker_info) %do% {
#   x &lt;- i %&gt;% data.frame %&gt;% rownames_to_column(&#39;features&#39;) %&gt;% 
#     mutate(
#       p = 0
#     )
#   gse_omnibus(
#     feature_names = x$features,
#     p = x$p,
#     fc = x$mean.AUC,
#     fc_thrs = 0.25,
#     return_sets = c(&#39;abs&#39;),
#     
#     gmt_files = gmt_files_symbols, 
# 
#     annot_db = &#39;org.Hs.eg.db&#39;,
#     organism = &#39;hsapiens&#39;,
#     kegg_organism = &#39;hsa&#39;,
#     reactome_organism = &#39;human&#39;,
#     
#     run_all_ora = FALSE,
#     run_all_gsea = FALSE,
#     run_GSEA = FALSE,
#     run_enricher = TRUE,
#     run_enrichGO = FALSE,
# 
#     args_gse = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1)
#     )
# } 
# names(gse_res) &lt;- names(marker_info)
# 
# for(i in names(gse_res)) {
#   gse_cl_auc[[i]]$enricher$abs$msigdb.c2.cp.reactome &lt;- gse_res[[i]]$enricher$abs$msigdb.c2.cp.reactome 
# }
# 
# saveRDS(gse_cl_auc, file.path(params$output_dir, &#39;scoreMarkers_site_gse_auc.rds&#39;))
# 
# 
# # Load MSigDB files
# gmt_files_symbols &lt;- list(
#   msigdb.c2.cp.reactome = &#39;./data/resources/MSigDB/v2023.1.Hs/c2.cp.reactome.v2023.1.Hs.symbols.gmt&#39;
# )
# gse_site_auc &lt;- readRDS( file.path(params$output_dir, &#39;scoreMarkers_site_gse_cohen.rds&#39;))
# i &lt;- marker_info[[1]]
# gse_res&lt;- foreach(i=marker_info) %do% {
#  x &lt;- i %&gt;% data.frame %&gt;% rownames_to_column(&#39;features&#39;) %&gt;% 
#     mutate(
#       p = 0,
#       mean.logFC.cohen = ifelse(mean.logFC.cohen &lt;= -5, -5, mean.logFC.cohen),
#       mean.logFC.cohen = ifelse(mean.logFC.cohen &gt;= 5, 5, mean.logFC.cohen)
#     )
#   gse_omnibus(
#     feature_names = x$features,
#     p = x$p,
#     fc = x$mean.AUC,
#     fc_thrs = 0.25,
#     return_sets = c(&#39;abs&#39;),
#     
#     gmt_files = gmt_files_symbols, 
# 
#     annot_db = &#39;org.Hs.eg.db&#39;,
#     organism = &#39;hsapiens&#39;,
#     kegg_organism = &#39;hsa&#39;,
#     reactome_organism = &#39;human&#39;,
#     
#     run_all_ora = FALSE,
#     run_all_gsea = FALSE,
#     run_GSEA = FALSE,
#     run_enricher = TRUE,
#     run_enrichGO = FALSE,
# 
#     args_gse = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1)
#     )
# } 
# names(gse_res) &lt;- names(marker_info)
# 
# for(i in names(gse_res)) {
#   gse_cl_auc[[i]]$enricher$abs$msigdb.c2.cp.reactome &lt;- gse_res[[i]]$enricher$abs$msigdb.c2.cp.reactome 
# }
# 
# saveRDS(gse_cl_auc, file.path(params$output_dir, &#39;scoreMarkers_site_gse_cohen.rds&#39;))</code></pre>
</div>
<div id="load-data" class="section level3" number="2.8">
<h3><span class="header-section-number">2.8</span> Load data</h3>
<pre class="r"><code>sce_comb &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated.rds&#39;))
sce_mnn &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated_corrected.rds&#39;))
score_markers &lt;- readRDS(file.path(params$output_dir, &#39;scoreMarkers.rds&#39;))
gse_cl_auc &lt;- readRDS( file.path(params$output_dir, &#39;scoreMarkers_cluster_gse_auc.rds&#39;))
# gse_cl_cohen &lt;- readRDS( file.path(params$output_dir, &#39;scoreMarkers_cluster_gse_cohen.rds&#39;))
gse_site_auc &lt;- readRDS( file.path(params$output_dir, &#39;scoreMarkers_site_gse_auc.rds&#39;))
# gse_site_cohen &lt;- readRDS( file.path(params$output_dir, &#39;scoreMarkers_site_gse_cohen.rds&#39;))

# Cluster MNN levels
cl_levels_sort &lt;- sce_comb$cluster_mnn %&gt;% as.character %&gt;% as.numeric %&gt;% unique %&gt;% sort %&gt;% as.character()
cl_levels_rev &lt;- sce_comb$cluster_mnn %&gt;% as.character %&gt;% as.numeric %&gt;% unique %&gt;% sort(decreasing = TRUE) %&gt;% as.character()
sce_comb$cluster_mnn %&lt;&gt;% as.character %&gt;% factor(levels = cl_levels_sort)
sce_comb$cluster_mnn_rev &lt;- sce_comb$cluster_mnn %&gt;% as.character %&gt;% factor(levels = cl_levels_rev)

# Coordinate colData between sce_comb and sce_mnn
colData(sce_mnn) &lt;- colData(sce_comb)
assay(sce_mnn, &#39;logcounts&#39;) &lt;- logcounts(sce_comb[rownames(sce_mnn),])
reducedDim(sce_comb, &#39;UMAP_corrected&#39;) &lt;- reducedDim(sce_mnn, &#39;UMAP&#39;)</code></pre>
</div>
</div>
<div id="data-description" class="section level2" number="3">
<h2><span class="header-section-number">3</span> Data description</h2>
<div id="site-distribution-per-batch" class="section level3"
number="3.1">
<h3><span class="header-section-number">3.1</span> Site distribution per
batch</h3>
<pre class="r"><code>table(sce_comb$Sample, sce_comb$site) %&gt;% 
  kbl(caption = &#39;Number of cells per batch and site.&#39;) %&gt;%
  kable_paper(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;), full_width = F)</code></pre>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Number of cells per batch and site.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
liver
</th>
<th style="text-align:right;">
lung
</th>
<th style="text-align:right;">
primary_tumor
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
1_Br16_GFP_luc_Tum_C_D
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
7328
</td>
</tr>
<tr>
<td style="text-align:left;">
2_Br16_GFP_luc_Liver_B_C_D
</td>
<td style="text-align:right;">
5690
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
3_Br16_GFP_luc_Lungs_B_C_D
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3491
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>sites_selected &lt;- sce_comb$site %&gt;% table %&gt;% data.frame %&gt;% dplyr::rename(site = &quot;.&quot;) %&gt;% filter(Freq &gt;= 500) %&gt;% collect %&gt;% .[[&#39;site&#39;]] %&gt;% as.character</code></pre>
</div>
</div>
<div id="batch-correction-diagnostics" class="section level2"
number="4">
<h2><span class="header-section-number">4</span> Batch correction
diagnostics</h2>
<p>Distribution of batches across clusters after batch correction.</p>
<pre class="r"><code>tab.mnn &lt;- table(Cluster=sce_mnn$cluster_mnn %&gt;% as.character %&gt;% as.numeric, Batch=sce_mnn$Sample) %&gt;% t()
tab.mnn.p &lt;- 100*prop.table(tab.mnn, margin = 1)
tab_res &lt;- foreach(i = 1:ncol(tab.mnn), .combine = cbind) %do% {
  paste0(tab.mnn[,i], &quot; (&quot;, round(tab.mnn.p[,i],2),&quot;%)&quot;)
}
dimnames(tab_res) &lt;- dimnames(tab.mnn)
tab_res %&gt;%
  kbl(caption = &#39;Ditribution of batches across clusters after batch correction.&#39;) %&gt;%
  kable_paper(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;), full_width = F)</code></pre>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Ditribution of batches across clusters after batch correction.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
1
</th>
<th style="text-align:left;">
2
</th>
<th style="text-align:left;">
3
</th>
<th style="text-align:left;">
4
</th>
<th style="text-align:left;">
5
</th>
<th style="text-align:left;">
6
</th>
<th style="text-align:left;">
7
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
1_Br16_GFP_luc_Tum_C_D
</td>
<td style="text-align:left;">
470 (6.41%)
</td>
<td style="text-align:left;">
969 (13.22%)
</td>
<td style="text-align:left;">
1787 (24.39%)
</td>
<td style="text-align:left;">
1573 (21.47%)
</td>
<td style="text-align:left;">
1901 (25.94%)
</td>
<td style="text-align:left;">
393 (5.36%)
</td>
<td style="text-align:left;">
235 (3.21%)
</td>
</tr>
<tr>
<td style="text-align:left;">
2_Br16_GFP_luc_Liver_B_C_D
</td>
<td style="text-align:left;">
916 (16.1%)
</td>
<td style="text-align:left;">
1734 (30.47%)
</td>
<td style="text-align:left;">
370 (6.5%)
</td>
<td style="text-align:left;">
1170 (20.56%)
</td>
<td style="text-align:left;">
256 (4.5%)
</td>
<td style="text-align:left;">
732 (12.86%)
</td>
<td style="text-align:left;">
512 (9%)
</td>
</tr>
<tr>
<td style="text-align:left;">
3_Br16_GFP_luc_Lungs_B_C_D
</td>
<td style="text-align:left;">
439 (12.58%)
</td>
<td style="text-align:left;">
823 (23.57%)
</td>
<td style="text-align:left;">
118 (3.38%)
</td>
<td style="text-align:left;">
866 (24.81%)
</td>
<td style="text-align:left;">
55 (1.58%)
</td>
<td style="text-align:left;">
749 (21.46%)
</td>
<td style="text-align:left;">
441 (12.63%)
</td>
</tr>
</tbody>
</table>
<p>MNN-specific diagnostics : one useful diagnostic is the proportion of
variance within each batch that is lost during MNN correction.
Specifically, this refers to the within-batch variance that is removed
during orthogonalization with respect to the average correction vector
at each merge step. Large proportions of lost variance (&gt;10%) suggest
that correction is removing genuine biological heterogeneity. This would
occur due to violations of the assumption of orthogonality between the
batch effect and the biological subspace (Haghverdi et al. 2018).</p>
<pre class="r"><code>tab_res &lt;- metadata(sce_mnn)$merge.info$lost.var
rownames(tab_res) &lt;- paste(&#39;Merge step&#39;, 1:nrow(tab_res))
tab_res %&gt;% t() %&gt;% 
  kbl(caption = &#39;Matrix of the variance lost in each batch (column) at each merge step (row) in the MNN step.&#39;) %&gt;%
  kable_paper(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;), full_width = F)</code></pre>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Matrix of the variance lost in each batch (column) at each merge step
(row) in the MNN step.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Merge step 1
</th>
<th style="text-align:right;">
Merge step 2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
1_Br16_GFP_luc_Tum_C_D
</td>
<td style="text-align:right;">
0.0112967
</td>
<td style="text-align:right;">
0.0114023
</td>
</tr>
<tr>
<td style="text-align:left;">
2_Br16_GFP_luc_Liver_B_C_D
</td>
<td style="text-align:right;">
0.0100463
</td>
<td style="text-align:right;">
0.0141552
</td>
</tr>
<tr>
<td style="text-align:left;">
3_Br16_GFP_luc_Lungs_B_C_D
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0258027
</td>
</tr>
</tbody>
</table>
<p>Another MNN-related diagnostic involves examining the variance in the
differences in expression between MNN pairs. A small variance indicates
that the correction had little effect - either there was no batch
effect, or any batch effect was simply a constant shift across all
cells. On the other hand, a large variance indicates that the correction
was highly non-linear, most likely involving subpopulation-specific
batch effects.</p>
<p>Such genes with large variances are particularly interesting as they
exhibit complex differences between batches that may reflect real
biology (whether this is biological or technical in nature is open for
interpretation.).</p>
<pre class="r"><code>vars &lt;- metadata(sce_mnn)$mnnDeltaVariance

vars[order(vars$adjusted, decreasing=TRUE),] %&gt;% 
  data.frame %&gt;% 
  dplyr::select(mean:adjusted) %&gt;% 
  head(10) %&gt;% 
  kbl(caption = &#39;Top-30 genes showing the largest variance between MNN pairs.&#39;) %&gt;%
  kable_paper(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;), full_width = F)</code></pre>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Top-30 genes showing the largest variance between MNN pairs.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
total
</th>
<th style="text-align:right;">
trend
</th>
<th style="text-align:right;">
adjusted
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
IFI27
</td>
<td style="text-align:right;">
4.898938
</td>
<td style="text-align:right;">
3.878988
</td>
<td style="text-align:right;">
0.4938617
</td>
<td style="text-align:right;">
3.385126
</td>
</tr>
<tr>
<td style="text-align:left;">
CRYAB
</td>
<td style="text-align:right;">
1.238740
</td>
<td style="text-align:right;">
4.223730
</td>
<td style="text-align:right;">
0.9054646
</td>
<td style="text-align:right;">
3.318265
</td>
</tr>
<tr>
<td style="text-align:left;">
FOS
</td>
<td style="text-align:right;">
3.347760
</td>
<td style="text-align:right;">
3.033850
</td>
<td style="text-align:right;">
0.5537037
</td>
<td style="text-align:right;">
2.480146
</td>
</tr>
<tr>
<td style="text-align:left;">
JUN
</td>
<td style="text-align:right;">
2.795236
</td>
<td style="text-align:right;">
3.011592
</td>
<td style="text-align:right;">
0.6723468
</td>
<td style="text-align:right;">
2.339245
</td>
</tr>
<tr>
<td style="text-align:left;">
SAT1
</td>
<td style="text-align:right;">
3.401990
</td>
<td style="text-align:right;">
2.813830
</td>
<td style="text-align:right;">
0.5582525
</td>
<td style="text-align:right;">
2.255578
</td>
</tr>
<tr>
<td style="text-align:left;">
H1.2
</td>
<td style="text-align:right;">
2.784211
</td>
<td style="text-align:right;">
2.802340
</td>
<td style="text-align:right;">
0.6799372
</td>
<td style="text-align:right;">
2.122403
</td>
</tr>
<tr>
<td style="text-align:left;">
DUSP1
</td>
<td style="text-align:right;">
2.742412
</td>
<td style="text-align:right;">
2.474777
</td>
<td style="text-align:right;">
0.6880190
</td>
<td style="text-align:right;">
1.786758
</td>
</tr>
<tr>
<td style="text-align:left;">
S100P
</td>
<td style="text-align:right;">
3.187387
</td>
<td style="text-align:right;">
2.331367
</td>
<td style="text-align:right;">
0.5839318
</td>
<td style="text-align:right;">
1.747435
</td>
</tr>
<tr>
<td style="text-align:left;">
ID3
</td>
<td style="text-align:right;">
1.834769
</td>
<td style="text-align:right;">
2.625579
</td>
<td style="text-align:right;">
0.8944616
</td>
<td style="text-align:right;">
1.731118
</td>
</tr>
<tr>
<td style="text-align:left;">
HES1
</td>
<td style="text-align:right;">
2.206465
</td>
<td style="text-align:right;">
2.487651
</td>
<td style="text-align:right;">
0.8265268
</td>
<td style="text-align:right;">
1.661124
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>top &lt;- rownames(vars)[order(vars$adjusted, decreasing=TRUE)[1:4]]
plotExpression(sce_comb, x = &quot;batch&quot;, features = top, point_size = 0.8, color_by = &quot;site&quot;) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  scale_color_manual(values = site_palette) +
  labs(caption = &#39;Distribution of the expression of the genes with the largest variance of MNN pair differences in each batch.&#39; )</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/batch-correction-mnn-DeltaVariance-1.png" width="4320" style="display: block; margin: auto;" /></p>
<pre class="r"><code>rm(vars)</code></pre>
<p>Finally, we can check the effect of batch correction on
clustering</p>
<pre class="r"><code>gridExtra::grid.arrange(
  
    plotUMAP(sce_comb, colour_by=&quot;batch&quot;, point_alpha = 0.1, point_size = 0.3) +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
      ggtitle (&#39;Raw data: batch&#39;),
    plotUMAP(sce_mnn, colour_by=&quot;batch&quot;, point_alpha = 0.1, point_size = 0.3) + 
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
      ggtitle (&#39;Corrected: batch&#39;),
    
    plotUMAP(sce_comb, colour_by=&quot;site&quot;, point_alpha = 0.1, point_size = 0.3) +
      scale_color_manual(values = site_palette) +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
      ggtitle (&#39;Raw data: site&#39;),
    plotUMAP(sce_mnn, colour_by=&quot;site&quot;, point_alpha = 0.1, point_size = 0.3) + 
      scale_color_manual(values = site_palette) +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
      ggtitle (&#39;Corrected: site&#39;),
    
    plotUMAP(sce_comb, colour_by=&quot;cluster_id&quot;, point_alpha = 0.1, point_size = 0.3) +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
      ggtitle (&#39;Raw data: cluster id from corrected&#39;),
    plotUMAP(sce_mnn, colour_by=&quot;cluster_mnn&quot;, point_alpha = 0.1, point_size = 0.3) + 
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
      ggtitle (&#39;Corrected:cluster id&#39;),
    
    ncol = 2
)</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/batch-correction-mnn-umap-1.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
<div id="clustering" class="section level2" number="5">
<h2><span class="header-section-number">5</span> Clustering</h2>
<div id="qc-parameters" class="section level3" number="5.1">
<h3><span class="header-section-number">5.1</span> QC parameters</h3>
<pre class="r"><code>gridExtra::grid.arrange(
    plotUMAP(sce_mnn, colour_by=&quot;site&quot;, point_alpha = 0.1, point_size = 0.3) + 
      scale_color_manual(values = site_palette) +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
      labs(color = &#39;&#39;) +
      ggtitle (&#39;Site&#39;),
    plotUMAP(sce_mnn, colour_by=&quot;cluster_mnn&quot;, text_by=&quot;cluster_mnn&quot;, point_alpha = 0.1, point_size = 0.3) +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
      labs(color = &#39;&#39;) +
      ggtitle (&#39;Cluster&#39;),
    plotUMAP(sce_mnn, colour_by=&quot;sum&quot;, point_alpha = 0.1, point_size = 0.3) + 
      scale_colour_continuous_diverging() +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
       labs(color = &#39;&#39;) +
      ggtitle (&#39;Library size (proxy for RNA content)&#39;),
    plotUMAP(sce_mnn, colour_by=&quot;detected&quot;, point_alpha = 0.1, point_size = 0.3) + 
      scale_colour_continuous_diverging() +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
       labs(color = &#39;&#39;) +
      ggtitle (&#39;UMI counts&#39;),
    plotUMAP(sce_mnn, colour_by=&quot;cell_cycle_phase&quot;, point_alpha = 0.1, point_size = 0.3) + 
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
      ggtitle (&#39;Cell cycle&#39;),
    plotUMAP(sce_mnn, colour_by=&quot;subsets_Mito_percent&quot;, point_alpha = 0.1, point_size = 0.3) + 
      scale_colour_continuous_diverging() +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
       labs(color = &#39;&#39;) +
      ggtitle (&#39;Mito pcte&#39;),
    plotUMAP(sce_mnn, colour_by=&quot;subsets_Ribo_percent&quot;, point_alpha = 0.1, point_size = 0.3) + 
      scale_colour_continuous_diverging() +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
      labs(color = &#39;&#39;) +
      ggtitle (&#39;Ribo pct&#39;),
    plotUMAP(sce_mnn, colour_by=&quot;scDblFinder.class&quot;, point_alpha = 0.1, point_size = 0.3) +
      scale_color_manual (values = c(alpha(&#39;grey90&#39;, 0.3), &#39;firebrick&#39;)) +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
      labs(color = &#39;&#39;) +
      ggtitle (&#39;Classification of doublets with scDblFinder&#39;),
    plotUMAP(sce_mnn, colour_by=&quot;scDblFinder.score&quot;, point_alpha = 0.1, point_size = 0.3) + 
      scale_colour_continuous_diverging() +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
       labs(color = &#39;&#39;) +
      ggtitle (&#39;Doublet score with scDblFinder&#39;),
    
    ncol = 2
)</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/clustering-umap-qc-1.png" width="4320" style="display: block; margin: auto;" /></p>
<pre class="r"><code>gridExtra::grid.arrange(
  plotColData(sce_mnn, x = &quot;cluster_mnn&quot;, y = &quot;detected&quot;, colour_by=&quot;cluster_mnn&quot;) +
    labs(title = &#39;UMI counts&#39;),
  plotColData(sce_mnn, x = &quot;cluster_mnn&quot;, y = &quot;subsets_Mito_percent&quot;, colour_by=&quot;cluster_mnn&quot;) +
    labs(title = &#39;Mito pcte&#39;),
  plotColData(sce_mnn, x = &quot;cluster_mnn&quot;, y = &quot;subsets_Ribo_percent&quot;, colour_by=&quot;cluster_mnn&quot;) +
    labs(title = &#39;Ribo pct&#39;),
  plotColData(sce_mnn, x = &quot;cluster_mnn&quot;, y = &quot;scDblFinder.score&quot;, colour_by=&quot;cluster_mnn&quot;) +
    labs(title = &#39;Doublet score with scDblFinder&#39;),
  
    ncol = 2
)</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/clustering-plotcoldata-qc-1.png" width="6120" style="display: block; margin: auto;" /></p>
</div>
<div id="cluster-identification" class="section level3" number="5.2">
<h3><span class="header-section-number">5.2</span> Cluster
identification</h3>
<pre class="r"><code>plotUMAP(sce_mnn, colour_by=&quot;cluster_mnn&quot;, text_by=&quot;cluster_mnn&quot;, point_alpha = 0.1, point_size = 0.3, text_size = 2) +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
      ggtitle (&#39;Cluster&#39;)</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/clustering-umap-id-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="differential-abundance-of-clusters" class="section level2"
number="6">
<h2><span class="header-section-number">6</span> Differential abundance
of clusters</h2>
<p>Differential abundance of clusters by site</p>
<div id="statistical-analysis" class="section level3" number="6.1">
<h3><span class="header-section-number">6.1</span> Statistical
analysis</h3>
<p>Differential abundance using <code>speckle::propeller</code>
package</p>
<pre class="r"><code>ct_label &lt;- &#39;cluster_mnn&#39;
use_data &lt;- colData(sce_comb) %&gt;% data.frame</code></pre>
</div>
<div id="barplots-by-site" class="section level3" number="6.2">
<h3><span class="header-section-number">6.2</span> Barplots by site</h3>
<pre class="r"><code># Create data
use_data &lt;- colData(sce_comb) %&gt;% 
  data.frame %&gt;% 
  dplyr::rename(&#39;celltype_label&#39; = ct_label) %&gt;% 
  group_by(site, celltype_label) %&gt;% 
  summarise(freq = n()) %&gt;% 
  dplyr::rename(cell_type = celltype_label) %&gt;% 
  na.omit()

# Remove low abundant cell types
cell_type_freq &lt;- use_data %&gt;% group_by(cell_type) %&gt;% summarise(totalfreq = sum(freq))
site_filtered &lt;- use_data %&gt;% group_by(site) %&gt;% summarise(freq = sum(freq)) %&gt;% filter(freq &gt;= 0)
use_data &lt;- use_data %&gt;% 
  filter(site %in% site_filtered$site) %&gt;% 
  left_join(cell_type_freq) %&gt;% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data &lt;- use_data %&gt;% 
  arrange(site, cell_type) %&gt;% 
  group_by(site) %&gt;%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion &lt; 0.03, &#39;&#39;, cell_type_label)
    
  )</code></pre>
<pre class="r"><code># Generate palette
get_palette &lt;- colorRampPalette(palette_OkabeIto)
use_palette &lt;- get_palette(arranged_data$cell_type %&gt;% nlevels)

# Final plot
arranged_data %&gt;% 
  ggplot(aes(x = site, y = proportion, fill = cell_type)) +
  geom_col(alpha = 0.85, color = &#39;black&#39;, size = 0.2) +
  scale_fill_manual(values = use_palette) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  theme(axis.ticks.x = element_blank()) +
  geom_text(aes(y = label_y, label = cell_type_label), colour = &quot;white&quot;, size = geom_text_size) +
  labs(
    fill = &#39;Cluster&#39;, 
    x = NULL,
    y = &#39;Proportion of cells&#39;
  )</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/da-barplot-site-stacked-cluster_mnn-1.png" width="4320" style="display: block; margin: auto;" /></p>
<pre class="r"><code>arranged_data %&gt;% 
  ggplot(aes(x = site, y = proportion, fill = site, group = cell_type)) +
  geom_col(alpha = 0.85, color = &#39;black&#39;, size = 0.2) +
  scale_fill_OkabeIto() +
  scale_y_continuous(expand = c(0.01, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  facet_wrap(vars(cell_type), ncol = 3, scales = &#39;free_y&#39;)+
  labs(
    fill = &#39;Site&#39;, 
    x = NULL,
    y = &#39;Proportion of cells&#39;
  )</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/da-barplot-site-cluster_mnn-facet-1.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="cluster-marker-genes" class="section level2" number="7">
<h2><span class="header-section-number">7</span> Cluster marker
genes</h2>
<pre class="r"><code>marker_info &lt;- score_markers$cluster_markers

# Table of results
rmd_file &lt;- current_input()
if(is.null(rmd_file))
  rmd_file &lt;- &#39;tmp&#39;
file_xlsx &lt;- file.path(&#39;./docs/file&#39;,rmd_file, &#39;cluster-markers-scoremarkers-tables.xlsx&#39;)
dir.create(dirname(file_xlsx), recursive = TRUE, showWarnings = FALSE)</code></pre>
<pre class="r"><code>wb &lt;- createWorkbook()

for(use_cluster in names(marker_info)){
  if(nchar(use_cluster) &gt; 31)
    use_cluster &lt;- paste0(substr(use_cluster, 1, 30), &quot;.&quot;)
  addWorksheet(wb, paste(&quot;Cluster&quot;, use_cluster))
  writeData(wb, paste(&quot;Cluster&quot;, use_cluster), marker_info[[use_cluster]])
}
saveWorkbook(wb, file_xlsx, TRUE)</code></pre>
<p>The tables of results can be downloaded using the following <a
href="./file/br16-10x_rnaseq-cancer_cells-main.Rmd/cluster-markers-scoremarkers-tables.xlsx">link</a>
.</p>
<div id="top-10-markers-plot" class="section level3" number="7.1">
<h3><span class="header-section-number">7.1</span> Top-10 markers
plot</h3>
<p>Here we show the selection of top markers upregulated in each
cluster.</p>
<pre class="r"><code>ntop &lt;- 10

# Dataframe of top markers
top_mk &lt;- foreach(i = names(marker_info), .combine = rbind) %do% {
  x &lt;- marker_info[[i]]
  ordered &lt;- x[order(x$mean.AUC, decreasing=TRUE),]
  data.frame(cluster = i, gene = rownames(ordered)[1:ntop])
}
top_mk$cluster &lt;- factor(top_mk$cluster, levels = names(marker_info) %&gt;% str_sort(numeric = TRUE))

# Matrix of top markers
top_mk_mat &lt;- top_mk %&gt;% 
  mutate(value = &#39;yes&#39;) %&gt;%
  pivot_wider(names_from = cluster, values_fill = &#39;no&#39;) %&gt;% 
  column_to_rownames(&#39;gene&#39;)

# Cluster matrix
x &lt;- top_mk_mat
x[x==&#39;no&#39;] &lt;- 0
x[x==&#39;yes&#39;] &lt;- 0
row_hclust &lt;- hclust(dist(x))
row_hclust_labels &lt;- row_hclust$labels[row_hclust$order]
col_hclust &lt;- hclust(dist(t(x)))
col_hclust_labels &lt;- col_hclust$labels[col_hclust$order]
top_mk_mat &lt;- top_mk_mat[row_hclust_labels, col_hclust_labels]

# # Prepare annotation for each cluster
# col_annot &lt;- colData(sce_comb) %&gt;% data.frame %&gt;% 
#   dplyr::select(cluster_mnn, celltype_immgen_cluster_pruned.labels) %&gt;% 
#   unique %&gt;% 
#   dplyr::rename(`Cell type` = celltype_immgen_cluster_pruned.labels)
# rownames(col_annot) &lt;- NULL
# col_annot %&lt;&gt;% column_to_rownames(&#39;cluster_mnn&#39;)
# col_annot &lt;- col_annot[colnames(top_mk_mat),, drop=FALSE]
# col_annot$`Cell type` &lt;- ifelse(is.na(col_annot$`Cell type`), &#39;Pruned&#39;, col_annot$`Cell type`)
# cell_type_levels &lt;- col_annot$`Cell type` %&gt;% unique
# 
# column_ha &lt;- HeatmapAnnotation(
#   `Cell type` = col_annot$`Cell type`,
#   col = list(
#     `Cell type` = brewer.pal(cell_type_levels %&gt;% length, &#39;Paired&#39;) %&gt;% set_names(cell_type_levels)
#     )
#   )

# Heatmap
ht_colors &lt;- c(yes = &#39;#E66B57&#39;, no = &#39;grey90&#39;)
Heatmap(
  top_mk_mat,
  name = &#39;top-marker&#39;,
  col = ht_colors, 
  # top_annotation = column_ha, 
  rect_gp = gpar(col= &quot;white&quot;)
)</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/marker-genes-heatmap-1.png" width="6120" style="display: block; margin: auto;" /></p>
</div>
<div id="grouped-expression-heatmaps-by-cluster-id"
class="section level3" number="7.2">
<h3><span class="header-section-number">7.2</span> Grouped expression
heatmaps by cluster ID</h3>
<p>Here we show the expression heatmap for the top-10 markers
upregulated in each cluster.</p>
<pre class="r"><code>ntop &lt;- 10

top_mk&lt;- foreach(x = marker_info) %do% {
  ordered &lt;- x[order(x$mean.AUC, decreasing=TRUE),]
  rownames(ordered)[1:ntop]
}

# # Prepare annotation for each cluster
# col_annot &lt;- colData(sce_comb) %&gt;% data.frame %&gt;% dplyr::select(cluster_mnn, celltype_immgen_cluster_pruned.labels) %&gt;% unique %&gt;% dplyr::rename(`Cell type` = celltype_immgen_cluster_pruned.labels)
# rownames(col_annot) &lt;- NULL
# col_annot %&lt;&gt;% column_to_rownames(&#39;cluster_mnn&#39;)
# col_annot$`Cell type` &lt;- ifelse(is.na(col_annot$`Cell type`), &#39;Pruned&#39;, col_annot$`Cell type`)
# cell_type_levels &lt;- col_annot$`Cell type` %&gt;% unique
# ann_colors &lt;- list(
#     `Cell type` = brewer.pal(cell_type_levels %&gt;% length, &#39;Paired&#39;) %&gt;% set_names(cell_type_levels)
# )

# Final plot
plotGroupedHeatmap(sce_comb,
                   features=unlist(top_mk) %&gt;% unique, 
                   group=&quot;cluster_mnn&quot;,
                   # annotation_col = col_annot,
                   # annotation_colors = ann_colors,
                   center=TRUE, 
                   zlim=c(-3, 3),
                   main = paste(&quot;Top&quot;, ntop,&quot;pairwise markers&quot;))</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/marker-genes-exprs-heatmap-1.png" width="6120" style="display: block; margin: auto;" /></p>
<p>Here we show the expression heatmap for the top-10 markers
upregulated in each cluster, but grouping samples by cluster and
site.</p>
<pre class="r"><code>use_sce &lt;- sce_comb
use_sce$group &lt;- paste(sce_comb$cluster_mnn, sce_comb$site, sep = &#39;-&#39;)

# # Prepare annotation for each cluster
# col_annot &lt;- colData(use_sce) %&gt;% data.frame %&gt;% dplyr::select(group, celltype_immgen_cluster_pruned.labels) %&gt;% unique %&gt;% dplyr::rename(`Cell type` = celltype_immgen_cluster_pruned.labels)
# rownames(col_annot) &lt;- NULL
# col_annot %&lt;&gt;% column_to_rownames(&#39;group&#39;)
# col_annot$`Cell type` &lt;- ifelse(is.na(col_annot$`Cell type`), &#39;Pruned&#39;, col_annot$`Cell type`)
# cell_type_levels &lt;- col_annot$`Cell type` %&gt;% unique
# ann_colors &lt;- list(
#     `Cell type` = brewer.pal(cell_type_levels %&gt;% length, &#39;Paired&#39;) %&gt;% set_names(cell_type_levels)
# )

# Final plot
plotGroupedHeatmap(use_sce,
                   features=unlist(top_mk) %&gt;% unique, 
                   group=&quot;group&quot;,
                   # annotation_col = col_annot,
                   # annotation_colors = ann_colors,
                   center=TRUE, 
                   zlim=c(-3, 3),
                   main = paste(&quot;Top&quot;, ntop,&quot;pairwise markers&quot;))</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/marker-genes-exprs-group_by_site_cluster-heatmap-1.png" width="6120" style="display: block; margin: auto;" /></p>
<pre class="r"><code>rm(use_sce)</code></pre>
</div>
<div id="umap-for-top-1-markers-for-each-cluster" class="section level3"
number="7.3">
<h3><span class="header-section-number">7.3</span> UMAP for top-1
markers for each cluster</h3>
<p>Here we remove mitochondrial and ribosomal genes</p>
<pre class="r"><code>marker_info_sorted_levels &lt;- names(marker_info) %&gt;% str_sort(numeric = TRUE)
top1_markers_umap &lt;- foreach(i = marker_info_sorted_levels) %do% {
  x &lt;- marker_info[[i]]
  ordered &lt;- x[order(x$mean.AUC, decreasing=TRUE),]
  top_mk &lt;- rownames(ordered)[1]
  plotReducedDim(sce_comb, &#39;UMAP_corrected&#39;, colour_by=top_mk, point_alpha = 0.5, point_size = 0.4) + 
      scale_colour_continuous_diverging() +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
      ggtitle (paste(&#39;Cluster&#39;,i, &#39;  &#39;, top_mk)) +
      labs(
        title = paste(&#39;Cluster&#39;,i, &#39;  &#39;, top_mk),
        x = &#39;UMAP 1&#39;,
        y = &#39;UMAP 2&#39;
      )
}

plot_grid(plotlist = top1_markers_umap)</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/marker-genes-umap-1.png" width="6120" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="cluster-markers-enrichment" class="section level2" number="8">
<h2><span class="header-section-number">8</span> Cluster markers
enrichment</h2>
<div id="ora---c2.cp.reactome-and-gobp" class="section level3"
number="8.1">
<h3><span class="header-section-number">8.1</span> ORA - C2.CP.REACTOME
and <a href="GO:BP" class="uri">GO:BP</a></h3>
<p>Over-enrichment analysis using genes with a mean AUC ≥ 0.25 in the
score markers results.</p>
<pre class="r"><code>gse_list &lt;- gse_cl_auc
test_type &lt;- &#39;enricher&#39;
gset_collection &lt;- &#39;msigdb.c2.cp.reactome&#39;
direction &lt;- &#39;abs&#39;
p_thrs &lt;- 0.01
topn &lt;- 20

# Load all the results
gse_df &lt;- foreach(i = names(gse_list), .combine = rbind) %do% {
  i_1 &lt;- gse_list[[i]][[test_type]][[direction]][[gset_collection]]@result %&gt;% 
    mutate(group = i)
  i_2 &lt;- gse_list[[i]][[&#39;enrichGO&#39;]][[direction]][[&#39;BP&#39;]]@result %&gt;% 
    filter(simplify) %&gt;% 
    dplyr::select(-simplify) %&gt;% 
    mutate(group = i)
  rbind(i_1, i_2)
} %&gt;% 
  mutate(
    group_n = as.numeric(group),
    group = fct_reorder(group, group_n),
    Description = clean_msigdb_names(Description) %&gt;% tolower() %&gt;% str_to_title,
    Description = ifelse(grepl(&quot;^GO:&quot;, ID), paste0(&quot;GO:&quot;, Description), Description)
  ) %&gt;% 
  dplyr::select(-group_n)

# Select topn by group in selected gset
top_by_group &lt;- foreach(i = names(gse_list), .combine = c) %do% {
  gse_df %&gt;% 
    filter(group == i) %&gt;% 
    filter(p.adjust &lt; p_thrs) %&gt;% 
    arrange(p.adjust) %&gt;% pull(Description) %&gt;%
    head(topn)
}

# Data frame to plot
res_top &lt;- gse_df %&gt;% 
  filter(Description %in% top_by_group) %&gt;% 
  mutate(log.p.adjust = -log10(p.adjust))

# Rows and column orders by hclust
use_mat &lt;- res_top %&gt;% 
  dplyr::select(Description, log.p.adjust, group) %&gt;% 
  pivot_wider(names_from = group, values_from = log.p.adjust) %&gt;% 
  column_to_rownames(&#39;Description&#39;)
use_mat[is.na(use_mat)] &lt;- 0

row_hclust &lt;- hclust(dist(use_mat))
row_hclust_labels &lt;- row_hclust$labels[row_hclust$order]

col_hclust &lt;- hclust(dist(t(use_mat)))
col_hclust_labels &lt;- col_hclust$labels[col_hclust$order]

use_mat &lt;- use_mat[row_hclust_labels,col_hclust_labels]

# GSET clusters by kmeans using silhouette
nb_clust &lt;- fviz_nbclust(use_mat, kmeans, method = &quot;silhouette&quot;, k.max = 20)
silhouette_best_k &lt;- nb_clust$data %&gt;% slice_max(y) %&gt;% pull(clusters) %&gt;% as.character %&gt;% as.numeric
# silhouette_best_k &lt;- 5
kmeans_mat &lt;- kmeans(use_mat, silhouette_best_k, nstart = 25)
kmeans_mat_df &lt;- kmeans_mat$cluster %&gt;%
  data.frame %&gt;%
  set_names(&#39;kcluster&#39;) %&gt;%
  rownames_to_column(&#39;Description&#39;) %&gt;%
  mutate(kcluster = as.character(kcluster))


# Order factors by hclust
res_top %&lt;&gt;% 
  left_join(kmeans_mat_df) %&gt;% 
  mutate(
    Description = factor(Description, levels = row_hclust_labels),
    group = fct_relevel(group, col_hclust_labels)
  )

# Calculate jaccard similarity matrix
gset_collection &lt;- foreach(i = levels(res_top$Description)) %do% {
  res_top %&gt;% filter(Description == i) %&gt;% pull(geneID) %&gt;% 
    paste(collapse = &#39;/&#39;) %&gt;% 
    strsplit(&#39;/&#39;) %&gt;% unlist %&gt;% unique
}
names(gset_collection) &lt;- levels(res_top$Description)

use_terms &lt;- res_top$Description %&gt;% unique
jc_matrix &lt;- pairwise_terms_jc(gset_collection[use_terms])</code></pre>
<div id="results-table" class="section level4" number="8.1.1">
<h4><span class="header-section-number">8.1.1</span> Results table</h4>
<pre class="r"><code>gse_df %&gt;% 
  filter(p.adjust &lt; p_thrs) %&gt;% 
  arrange(desc(p.adjust)) %&gt;% 
  dplyr::select(group, ID:p.adjust, Count:GeneProp, geneID)  %&gt;%
  dplyr::rename(cluster = group) %&gt;% 
  datatable(., 
            rownames = FALSE, 
            filter = &#39;top&#39;, 
            caption = &#39;Results from over enrichment analysis&#39;,
            extensions = &#39;Buttons&#39;, 
            options = list(
              dom = &#39;Blfrtip&#39;,
              buttons = c(&#39;csv&#39;, &#39;excel&#39;)
            ))  %&gt;% 
  formatRound(c(&#39;GeneProp&#39;), digits = 2) %&gt;%
  formatSignif(c(&#39;pvalue&#39;, &#39;p.adjust&#39;), digits = 3)</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-49e08474188736d8542c" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-49e08474188736d8542c">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;1&quot;,&quot;2&quot;,&quot;3&quot;,&quot;4&quot;,&quot;5&quot;,&quot;6&quot;,&quot;7&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"0.001792609202373\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"0.009992903703432\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"3\" data-max=\"113\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"10\" data-max=\"499\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.036939313984168\" data-max=\"0.9375\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Results from over enrichment analysis<\/caption>","data":[["6","4","4","4","4","1","7","4","1","7","4","7","6","1","1","6","6","6","6","6","6","4","6","1","1","7","6","6","5","1","6","6","6","6","4","7","7","6","7","5","7","4","4","3","4","4","7","4","3","6","7","4","4","4","4","4","7","2","2","2","4","4","6","2","6","6","7","7","4","4","1","4","4","4","4","4","4","4","2","2","2","2","7","4","1","1","4","4","4","6","4","1","7","5","1","3","7","7","4","5","1","3","4","4","7","1","5","2","5","4","4","4","6","6","3","7","7","3","3","1","1","1","4","7","5","4","7","7","7","4","7","3","6","6","4","6","6","6","7","6","1","7","4","7","1","4","1","5","5","6","6","7","7","7","7","6","7","6","4","6","1","1","7","7","5","6","6","5","6","1","2","6","6","1","3","4","4","4","4","4","1","1","2","7","2","7","6","1","1","7","7","1","7","3","1","1","6","7","1","1","1","1","1","1","2","1","1","1","5","6","7","7","4","7","2","6","6","6","6","1","6","1","4","1","7","6","2","5","6","7","7","6","4","6","6","1","7","6","3","7","7","7","7","6","6","7","7","7","3","1","7","7","1","4","1","5","3","3","3","3","1","4","4","6","1","4","3","7","6","4","1","6","2","7","1","1","1","2","1","6","1","7","4","6","1","1","1","6","6","6","1","1","6","7","1","7","1","6","5","1","5","7","1","1","1","1","1","1","6","6","1","5","7","6","6","1","4","4","4","6","6","5","7","1","3","1","1","1","7","5","1","1","1","1","1","6","7","7","7","1","2","6","7","1","6","1","7","6","7","7","1","7","3","3","3","2","2","7","7","1","7","7","6","6","6","5","6","1","7","3","1","6","6","4","6","6","1","7","6","6","6","7","7","6","6","6","1","7","3","3","3","3","3","1","6","7","1","7","6","7","7","7","7","7","6","6","6","6","1","7","7","1","7","6","6","6","7","7","7","7","7","4","7","6","6","6","6","7","7","7","6","7","6","6","1","7","1","7","6","1","1","6","2","7","6","6","6","7","7","7","1","6","6","1","7","7","5","5","7","7","1","7","7","7","6","7","1","1","1","1","7","7","7","6","5","7","7","7","7","7","7","7","7","7","7","7","7","7","7","7","7","6","7","7","7","6","6","7","6","7","6","7","6","7","7","7","7","7","7","7","7","7","7","7","7","7","6","7","7","7","6","7","6","7","7","7","7","6","7","6","7","6","7","6"],["REACTOME_MITOCHONDRIAL_FATTY_ACID_BETA_OXIDATION","REACTOME_HSF1_ACTIVATION","REACTOME_COPI_INDEPENDENT_GOLGI_TO_ER_RETROGRADE_TRAFFIC","REACTOME_SENESCENCE_ASSOCIATED_SECRETORY_PHENOTYPE_SASP","REACTOME_RHO_GTPASES_ACTIVATE_FORMINS","REACTOME_DOWNREGULATION_OF_TGF_BETA_RECEPTOR_SIGNALING","REACTOME_GOLGI_TO_ER_RETROGRADE_TRANSPORT","REACTOME_AMYLOID_FIBER_FORMATION","REACTOME_APC_C_MEDIATED_DEGRADATION_OF_CELL_CYCLE_PROTEINS","REACTOME_ORGANELLE_BIOGENESIS_AND_MAINTENANCE","REACTOME_MEIOTIC_SYNAPSIS","REACTOME_TRNA_PROCESSING_IN_THE_NUCLEUS","REACTOME_SRP_DEPENDENT_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE","REACTOME_HATS_ACETYLATE_HISTONES","REACTOME_REGULATION_OF_MRNA_STABILITY_BY_PROTEINS_THAT_BIND_AU_RICH_ELEMENTS","REACTOME_COPI_MEDIATED_ANTEROGRADE_TRANSPORT","REACTOME_ATTENUATION_PHASE","REACTOME_SEPARATION_OF_SISTER_CHROMATIDS","REACTOME_TRANS_GOLGI_NETWORK_VESICLE_BUDDING","REACTOME_DETOXIFICATION_OF_REACTIVE_OXYGEN_SPECIES","REACTOME_RHO_GTPASES_ACTIVATE_WASPS_AND_WAVES","REACTOME_ACTIVATION_OF_AMPK_DOWNSTREAM_OF_NMDARS","REACTOME_DNA_REPLICATION_PRE_INITIATION","REACTOME_DEACTIVATION_OF_THE_BETA_CATENIN_TRANSACTIVATING_COMPLEX","REACTOME_INTERLEUKIN_4_AND_INTERLEUKIN_13_SIGNALING","REACTOME_PRE_NOTCH_EXPRESSION_AND_PROCESSING","REACTOME_SARS_COV_INFECTIONS","REACTOME_ER_TO_GOLGI_ANTEROGRADE_TRANSPORT","REACTOME_GOLGI_ASSOCIATED_VESICLE_BIOGENESIS","REACTOME_DEFECTIVE_INTRINSIC_PATHWAY_FOR_APOPTOSIS","REACTOME_MHC_CLASS_II_ANTIGEN_PRESENTATION","REACTOME_GLUTATHIONE_CONJUGATION","REACTOME_LYSOSOME_VESICLE_BIOGENESIS","REACTOME_RESPONSE_TO_ELEVATED_PLATELET_CYTOSOLIC_CA2","REACTOME_CELLULAR_RESPONSE_TO_HEAT_STRESS","REACTOME_G1_S_DNA_DAMAGE_CHECKPOINTS","REACTOME_TRANSPORT_OF_THE_SLBP_DEPENDANT_MATURE_MRNA","REACTOME_SMOOTH_MUSCLE_CONTRACTION","REACTOME_HOST_INTERACTIONS_OF_HIV_FACTORS","REACTOME_REGULATION_OF_CHOLESTEROL_BIOSYNTHESIS_BY_SREBP_SREBF","GO:0007623","REACTOME_AURKA_ACTIVATION_BY_TPX2","REACTOME_ERCC6_CSB_AND_EHMT2_G9A_POSITIVELY_REGULATE_RRNA_EXPRESSION","REACTOME_GLUCONEOGENESIS","REACTOME_DEPOSITION_OF_NEW_CENPA_CONTAINING_NUCLEOSOMES_AT_THE_CENTROMERE","REACTOME_DNA_DAMAGE_TELOMERE_STRESS_INDUCED_SENESCENCE","REACTOME_NONHOMOLOGOUS_END_JOINING_NHEJ","GO:0140694","GO:0034109","REACTOME_LATE_ENDOSOMAL_MICROAUTOPHAGY","REACTOME_POSITIVE_EPIGENETIC_REGULATION_OF_RRNA_EXPRESSION","REACTOME_CONDENSATION_OF_PROPHASE_CHROMOSOMES","REACTOME_PRC2_METHYLATES_HISTONES_AND_DNA","REACTOME_FORMATION_OF_TUBULIN_FOLDING_INTERMEDIATES_BY_CCT_TRIC","REACTOME_RECRUITMENT_OF_NUMA_TO_MITOTIC_CENTROSOMES","REACTOME_APOPTOTIC_CLEAVAGE_OF_CELL_ADHESION_PROTEINS","REACTOME_INTRINSIC_PATHWAY_FOR_APOPTOSIS","REACTOME_METABOLISM_OF_STEROIDS","REACTOME_PRE_NOTCH_EXPRESSION_AND_PROCESSING","REACTOME_NEUTROPHIL_DEGRANULATION","REACTOME_AGGREPHAGY","REACTOME_RUNX1_REGULATES_GENES_INVOLVED_IN_MEGAKARYOCYTE_DIFFERENTIATION_AND_PLATELET_FUNCTION","GO:2001242","GO:0046034","REACTOME_DNA_REPLICATION","REACTOME_RHO_GTPASES_ACTIVATE_ROCKS","REACTOME_ASSEMBLY_OF_THE_ORC_COMPLEX_AT_THE_ORIGIN_OF_REPLICATION","REACTOME_NUCLEAR_IMPORT_OF_REV_PROTEIN","REACTOME_OXIDATIVE_STRESS_INDUCED_SENESCENCE","REACTOME_ASSEMBLY_OF_THE_ORC_COMPLEX_AT_THE_ORIGIN_OF_REPLICATION","REACTOME_HSF1_DEPENDENT_TRANSACTIVATION","REACTOME_ACTIVATION_OF_GENE_EXPRESSION_BY_SREBF_SREBP","REACTOME_SIRT1_NEGATIVELY_REGULATES_RRNA_EXPRESSION","REACTOME_ACTIVATION_OF_ANTERIOR_HOX_GENES_IN_HINDBRAIN_DEVELOPMENT_DURING_EARLY_EMBRYOGENESIS","REACTOME_ESTROGEN_DEPENDENT_GENE_EXPRESSION","REACTOME_ACTIVATED_PKN1_STIMULATES_TRANSCRIPTION_OF_AR_ANDROGEN_RECEPTOR_REGULATED_GENES_KLK2_AND_KLK3","REACTOME_INHIBITION_OF_DNA_RECOMBINATION_AT_TELOMERE","REACTOME_POST_CHAPERONIN_TUBULIN_FOLDING_PATHWAY","REACTOME_ESTROGEN_DEPENDENT_GENE_EXPRESSION","REACTOME_NGF_STIMULATED_TRANSCRIPTION","REACTOME_SIGNALING_BY_NUCLEAR_RECEPTORS","REACTOME_SIGNALING_BY_NOTCH","REACTOME_COHESIN_LOADING_ONTO_CHROMATIN","REACTOME_DNA_METHYLATION","REACTOME_PROTEIN_UBIQUITINATION","REACTOME_INTERFERON_SIGNALING","REACTOME_BASE_EXCISION_REPAIR_AP_SITE_FORMATION","GO:0032728","GO:0043618","REACTOME_INTERFERON_ALPHA_BETA_SIGNALING","REACTOME_TRANSPORT_OF_CONNEXONS_TO_THE_PLASMA_MEMBRANE","REACTOME_RESOLUTION_OF_SISTER_CHROMATID_COHESION","REACTOME_SIGNALING_BY_NOTCH","GO:0019083","GO:0071479","REACTOME_RESPONSE_TO_ELEVATED_PLATELET_CYTOSOLIC_CA2","REACTOME_ESR_MEDIATED_SIGNALING","REACTOME_METABOLISM_OF_NUCLEOTIDES","REACTOME_SELECTIVE_AUTOPHAGY","REACTOME_TRANS_GOLGI_NETWORK_VESICLE_BUDDING","REACTOME_CHROMATIN_MODIFYING_ENZYMES","GO:0046718","REACTOME_SEMA4D_INDUCED_CELL_MIGRATION_AND_GROWTH_CONE_COLLAPSE","REACTOME_ANTIVIRAL_MECHANISM_BY_IFN_STIMULATED_GENES","REACTOME_TP53_REGULATES_TRANSCRIPTION_OF_DNA_REPAIR_GENES","GO:0043542","REACTOME_CYTOSOLIC_TRNA_AMINOACYLATION","REACTOME_INTERLEUKIN_4_AND_INTERLEUKIN_13_SIGNALING","REACTOME_ER_TO_GOLGI_ANTEROGRADE_TRANSPORT","REACTOME_RHO_GTPASES_ACTIVATE_CIT","REACTOME_RHO_GTPASES_ACTIVATE_ROCKS","REACTOME_PRE_NOTCH_EXPRESSION_AND_PROCESSING","REACTOME_DEFECTIVE_INTRINSIC_PATHWAY_FOR_APOPTOSIS","REACTOME_MTORC1_MEDIATED_SIGNALLING","GO:0010634","REACTOME_DNA_METHYLATION","REACTOME_ERCC6_CSB_AND_EHMT2_G9A_POSITIVELY_REGULATE_RRNA_EXPRESSION","REACTOME_GLYCOLYSIS","REACTOME_TRNA_AMINOACYLATION","REACTOME_PROCESSING_OF_DNA_DOUBLE_STRAND_BREAK_ENDS","REACTOME_INITIATION_OF_NUCLEAR_ENVELOPE_NE_REFORMATION","REACTOME_SIGNALING_BY_FLT3_FUSION_PROTEINS","REACTOME_RECOGNITION_AND_ASSOCIATION_OF_DNA_GLYCOSYLASE_WITH_SITE_CONTAINING_AN_AFFECTED_PURINE","REACTOME_PKMTS_METHYLATE_HISTONE_LYSINES","GO:0031647","REACTOME_DISEASES_OF_PROGRAMMED_CELL_DEATH","REACTOME_DEPOLYMERISATION_OF_THE_NUCLEAR_LAMINA","REACTOME_REGULATION_OF_HSF1_MEDIATED_HEAT_SHOCK_RESPONSE","REACTOME_ACTIVATION_OF_ANTERIOR_HOX_GENES_IN_HINDBRAIN_DEVELOPMENT_DURING_EARLY_EMBRYOGENESIS","REACTOME_RHO_GTPASES_ACTIVATE_IQGAPS","REACTOME_FANCONI_ANEMIA_PATHWAY","REACTOME_ATF4_ACTIVATES_GENES_IN_RESPONSE_TO_ENDOPLASMIC_RETICULUM_STRESS","REACTOME_INTERLEUKIN_1_FAMILY_SIGNALING","REACTOME_SIGNALING_BY_ROBO_RECEPTORS","REACTOME_RECYCLING_PATHWAY_OF_L1","REACTOME_METABOLISM_OF_AMINO_ACIDS_AND_DERIVATIVES","REACTOME_INTERCONVERSION_OF_NUCLEOTIDE_DI_AND_TRIPHOSPHATES","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_TP53","REACTOME_THE_ROLE_OF_GTSE1_IN_G2_M_PROGRESSION_AFTER_G2_CHECKPOINT","REACTOME_AGGREPHAGY","GO:0071364","REACTOME_INTERACTIONS_OF_VPR_WITH_HOST_CELLULAR_PROTEINS","REACTOME_RESPONSE_OF_EIF2AK1_HRI_TO_HEME_DEFICIENCY","REACTOME_COPI_DEPENDENT_GOLGI_TO_ER_RETROGRADE_TRAFFIC","REACTOME_SIGNALING_BY_FGFR","REACTOME_ATTENUATION_PHASE","REACTOME_UB_SPECIFIC_PROCESSING_PROTEASES","REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA","REACTOME_MRNA_SPLICING","REACTOME_BETA_CATENIN_INDEPENDENT_WNT_SIGNALING","REACTOME_ASPARAGINE_N_LINKED_GLYCOSYLATION","REACTOME_NUCLEAR_PORE_COMPLEX_NPC_DISASSEMBLY","REACTOME_DUAL_INCISION_IN_TC_NER","REACTOME_INHIBITION_OF_REPLICATION_INITIATION_OF_DAMAGED_DNA_BY_RB1_E2F1","REACTOME_MITOTIC_TELOPHASE_CYTOKINESIS","REACTOME_NEDDYLATION","REACTOME_POSTMITOTIC_NUCLEAR_PORE_COMPLEX_NPC_REFORMATION","REACTOME_IRON_UPTAKE_AND_TRANSPORT","REACTOME_SMOOTH_MUSCLE_CONTRACTION","REACTOME_SIGNALING_BY_NOTCH","REACTOME_ATTENUATION_PHASE","REACTOME_FLT3_SIGNALING_IN_DISEASE","REACTOME_ESTROGEN_DEPENDENT_GENE_EXPRESSION","GO:0098727","REACTOME_ATF4_ACTIVATES_GENES_IN_RESPONSE_TO_ENDOPLASMIC_RETICULUM_STRESS","REACTOME_ABC_FAMILY_PROTEINS_MEDIATED_TRANSPORT","GO:0051881","REACTOME_ASPARAGINE_N_LINKED_GLYCOSYLATION","REACTOME_HSP90_CHAPERONE_CYCLE_FOR_STEROID_HORMONE_RECEPTORS_SHR_IN_THE_PRESENCE_OF_LIGAND","REACTOME_NUCLEAR_EVENTS_KINASE_AND_TRANSCRIPTION_FACTOR_ACTIVATION","GO:0052547","REACTOME_S_PHASE","REACTOME_COOPERATION_OF_PREFOLDIN_AND_TRIC_CCT_IN_ACTIN_AND_TUBULIN_FOLDING","REACTOME_DNA_REPAIR","GO:0042692","REACTOME_SEMA4D_IN_SEMAPHORIN_SIGNALING","REACTOME_HCMV_INFECTION","REACTOME_HSF1_DEPENDENT_TRANSACTIVATION","REACTOME_FORMATION_OF_THE_CORNIFIED_ENVELOPE","REACTOME_KERATINIZATION","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX1","REACTOME_MITOTIC_PROMETAPHASE","REACTOME_REGULATION_OF_CHOLESTEROL_BIOSYNTHESIS_BY_SREBP_SREBF","REACTOME_EXPORT_OF_VIRAL_RIBONUCLEOPROTEINS_FROM_NUCLEUS","GO:0007623","REACTOME_DNA_DAMAGE_TELOMERE_STRESS_INDUCED_SENESCENCE","REACTOME_MITOTIC_G1_PHASE_AND_G1_S_TRANSITION","REACTOME_DNA_DOUBLE_STRAND_BREAK_RESPONSE","REACTOME_MITOTIC_G2_G2_M_PHASES","REACTOME_ANTIVIRAL_MECHANISM_BY_IFN_STIMULATED_GENES","REACTOME_NS1_MEDIATED_EFFECTS_ON_HOST_PATHWAYS","GO:0051098","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_SMALL_RNAS","REACTOME_PERK_REGULATES_GENE_EXPRESSION","GO:0006913","GO:0051169","REACTOME_RUNX1_REGULATES_TRANSCRIPTION_OF_GENES_INVOLVED_IN_DIFFERENTIATION_OF_HSCS","GO:0019079","REACTOME_SIGNALING_BY_FGFR2","REACTOME_G2_M_CHECKPOINTS","REACTOME_G2_M_DNA_DAMAGE_CHECKPOINT","REACTOME_RUNX1_REGULATES_GENES_INVOLVED_IN_MEGAKARYOCYTE_DIFFERENTIATION_AND_PLATELET_FUNCTION","REACTOME_HDACS_DEACETYLATE_HISTONES","REACTOME_SEPARATION_OF_SISTER_CHROMATIDS","REACTOME_INTERFERON_ALPHA_BETA_SIGNALING","REACTOME_HCMV_LATE_EVENTS","REACTOME_HCMV_EARLY_EVENTS","REACTOME_GENE_SILENCING_BY_RNA","REACTOME_SIGNALING_BY_ALK_IN_CANCER","GO:0016126","REACTOME_SUMOYLATION_OF_DNA_REPLICATION_PROTEINS","REACTOME_FACTORS_INVOLVED_IN_MEGAKARYOCYTE_DEVELOPMENT_AND_PLATELET_PRODUCTION","REACTOME_HCMV_EARLY_EVENTS","REACTOME_INTERACTIONS_OF_REV_WITH_HOST_CELLULAR_PROTEINS","GO:0071364","REACTOME_EPH_EPHRIN_SIGNALING","REACTOME_ORGANELLE_BIOGENESIS_AND_MAINTENANCE","REACTOME_C_TYPE_LECTIN_RECEPTORS_CLRS","REACTOME_TCR_SIGNALING","GO:0030099","REACTOME_SELECTIVE_AUTOPHAGY","REACTOME_NEGATIVE_EPIGENETIC_REGULATION_OF_RRNA_EXPRESSION","REACTOME_INTERFERON_SIGNALING","REACTOME_HSF1_ACTIVATION","REACTOME_CILIUM_ASSEMBLY","REACTOME_CHOLESTEROL_BIOSYNTHESIS","REACTOME_ACTIVATION_OF_GENE_EXPRESSION_BY_SREBF_SREBP","GO:2000737","GO:0007035","REACTOME_TP53_REGULATES_TRANSCRIPTION_OF_GENES_INVOLVED_IN_G1_CELL_CYCLE_ARREST","REACTOME_HCMV_INFECTION","REACTOME_FC_EPSILON_RECEPTOR_FCERI_SIGNALING","GO:0043065","REACTOME_INTERLEUKIN_1_SIGNALING","REACTOME_SIGNALING_BY_THE_B_CELL_RECEPTOR_BCR","REACTOME_EPIGENETIC_REGULATION_OF_GENE_EXPRESSION","REACTOME_CONDENSATION_OF_PROPHASE_CHROMOSOMES","REACTOME_REGULATION_OF_EXPRESSION_OF_SLITS_AND_ROBOS","GO:0018401","REACTOME_TRANSCRIPTION_COUPLED_NUCLEOTIDE_EXCISION_REPAIR_TC_NER","REACTOME_TRANSLESION_SYNTHESIS_BY_POLH","REACTOME_ONCOGENE_INDUCED_SENESCENCE","REACTOME_EPIGENETIC_REGULATION_OF_GENE_EXPRESSION","REACTOME_SIGNALING_BY_HEDGEHOG","REACTOME_SYNTHESIS_OF_DNA","REACTOME_CHK1_CHK2_CDS1_MEDIATED_INACTIVATION_OF_CYCLIN_B_CDK1_COMPLEX","REACTOME_RMTS_METHYLATE_HISTONE_ARGININES","GO:0032781","REACTOME_CYTOSOLIC_TRNA_AMINOACYLATION","REACTOME_FGFR2_ALTERNATIVE_SPLICING","REACTOME_MEIOTIC_SYNAPSIS","REACTOME_TRANSCRIPTIONAL_REGULATION_OF_GRANULOPOIESIS","REACTOME_SIGNALING_BY_WNT","REACTOME_RHO_GTPASE_EFFECTORS","REACTOME_HCMV_INFECTION","REACTOME_PERK_REGULATES_GENE_EXPRESSION","GO:0006457","GO:0006735","GO:0061621","GO:0061718","REACTOME_DNA_REPLICATION","GO:0042026","GO:0016126","REACTOME_UCH_PROTEINASES","REACTOME_NGF_STIMULATED_TRANSCRIPTION","REACTOME_RHO_GTPASES_ACTIVATE_PKNS","GO:0050821","REACTOME_DISEASES_OF_MITOTIC_CELL_CYCLE","REACTOME_PROTEIN_LOCALIZATION","REACTOME_INTERFERON_ALPHA_BETA_SIGNALING","REACTOME_BASE_EXCISION_REPAIR","GO:0016032","GO:0043068","REACTOME_TRANSLESION_SYNTHESIS_BY_POLK","REACTOME_RNA_POLYMERASE_I_TRANSCRIPTION","REACTOME_B_WICH_COMPLEX_POSITIVELY_REGULATES_RRNA_EXPRESSION","REACTOME_RNA_POLYMERASE_I_PROMOTER_ESCAPE","GO:0016126","REACTOME_TRANSCRIPTIONAL_REGULATION_OF_GRANULOPOIESIS","REACTOME_APC_C_MEDIATED_DEGRADATION_OF_CELL_CYCLE_PROTEINS","REACTOME_PRE_NOTCH_EXPRESSION_AND_PROCESSING","REACTOME_NUCLEAR_ENVELOPE_BREAKDOWN","GO:0034109","REACTOME_AUTOPHAGY","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_SMALL_RNAS","REACTOME_ASSEMBLY_OF_THE_ORC_COMPLEX_AT_THE_ORIGIN_OF_REPLICATION","REACTOME_POSITIVE_EPIGENETIC_REGULATION_OF_RRNA_EXPRESSION","REACTOME_MITOTIC_G2_G2_M_PHASES","REACTOME_GLUCONEOGENESIS","REACTOME_ABC_TRANSPORTER_DISORDERS","REACTOME_SIRT1_NEGATIVELY_REGULATES_RRNA_EXPRESSION","REACTOME_MEIOTIC_RECOMBINATION","REACTOME_HEDGEHOG_ON_STATE","REACTOME_ESTABLISHMENT_OF_SISTER_CHROMATID_COHESION","REACTOME_ACTIVATED_PKN1_STIMULATES_TRANSCRIPTION_OF_AR_ANDROGEN_RECEPTOR_REGULATED_GENES_KLK2_AND_KLK3","REACTOME_NUCLEOTIDE_EXCISION_REPAIR","REACTOME_OXIDATIVE_STRESS_INDUCED_SENESCENCE","REACTOME_CYTOPROTECTION_BY_HMOX1","REACTOME_ATF6_ATF6_ALPHA_ACTIVATES_CHAPERONES","REACTOME_REPRODUCTION","REACTOME_IRE1ALPHA_ACTIVATES_CHAPERONES","REACTOME_APC_C_MEDIATED_DEGRADATION_OF_CELL_CYCLE_PROTEINS","REACTOME_DNA_METHYLATION","REACTOME_RHO_GTPASE_EFFECTORS","REACTOME_BASE_EXCISION_REPAIR_AP_SITE_FORMATION","REACTOME_MITOTIC_PROPHASE","REACTOME_MITOTIC_METAPHASE_AND_ANAPHASE","REACTOME_CELLULAR_SENESCENCE","REACTOME_TNFR2_NON_CANONICAL_NF_KB_PATHWAY","REACTOME_SWITCHING_OF_ORIGINS_TO_A_POST_REPLICATIVE_STATE","REACTOME_SIGNALING_BY_NOTCH","REACTOME_CHOLESTEROL_BIOSYNTHESIS","REACTOME_SUMOYLATION_OF_DNA_DAMAGE_RESPONSE_AND_REPAIR_PROTEINS","REACTOME_CLEC7A_DECTIN_1_SIGNALING","REACTOME_MAPK6_MAPK4_SIGNALING","REACTOME_DNA_REPLICATION_PRE_INITIATION","REACTOME_CHOLESTEROL_BIOSYNTHESIS","REACTOME_NEUTROPHIL_DEGRANULATION","REACTOME_HSP90_CHAPERONE_CYCLE_FOR_STEROID_HORMONE_RECEPTORS_SHR_IN_THE_PRESENCE_OF_LIGAND","REACTOME_APOPTOSIS","REACTOME_APC_C_CDH1_MEDIATED_DEGRADATION_OF_CDC20_AND_OTHER_APC_C_CDH1_TARGETED_PROTEINS_IN_LATE_MITOSIS_EARLY_G1","REACTOME_RESPONSE_OF_EIF2AK1_HRI_TO_HEME_DEFICIENCY","REACTOME_CYCLIN_A_CDK2_ASSOCIATED_EVENTS_AT_S_PHASE_ENTRY","GO:0050678","REACTOME_ATF6_ATF6_ALPHA_ACTIVATES_CHAPERONES","REACTOME_ERCC6_CSB_AND_EHMT2_G9A_POSITIVELY_REGULATE_RRNA_EXPRESSION","REACTOME_RHO_GTPASES_ACTIVATE_PKNS","REACTOME_DNA_DAMAGE_TELOMERE_STRESS_INDUCED_SENESCENCE","REACTOME_HCMV_EARLY_EVENTS","REACTOME_ATF6_ATF6_ALPHA_ACTIVATES_CHAPERONE_GENES","REACTOME_RECOGNITION_AND_ASSOCIATION_OF_DNA_GLYCOSYLASE_WITH_SITE_CONTAINING_AN_AFFECTED_PURINE","REACTOME_PRC2_METHYLATES_HISTONES_AND_DNA","REACTOME_RUNX1_REGULATES_TRANSCRIPTION_OF_GENES_INVOLVED_IN_DIFFERENTIATION_OF_HSCS","REACTOME_FORMATION_OF_THE_BETA_CATENIN_TCF_TRANSACTIVATING_COMPLEX","REACTOME_AMYLOID_FIBER_FORMATION","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX3","REACTOME_CYCLIN_D_ASSOCIATED_EVENTS_IN_G1","REACTOME_PROCESSIVE_SYNTHESIS_ON_THE_C_STRAND_OF_THE_TELOMERE","REACTOME_TRANSCRIPTION_OF_E2F_TARGETS_UNDER_NEGATIVE_CONTROL_BY_DREAM_COMPLEX","REACTOME_CELL_CYCLE_CHECKPOINTS","REACTOME_CHOLESTEROL_BIOSYNTHESIS","REACTOME_PROGRAMMED_CELL_DEATH","REACTOME_ORC1_REMOVAL_FROM_CHROMATIN","REACTOME_ACTIVATION_OF_ANTERIOR_HOX_GENES_IN_HINDBRAIN_DEVELOPMENT_DURING_EARLY_EMBRYOGENESIS","REACTOME_DEGRADATION_OF_BETA_CATENIN_BY_THE_DESTRUCTION_COMPLEX","REACTOME_TCF_DEPENDENT_SIGNALING_IN_RESPONSE_TO_WNT","REACTOME_KINESINS","REACTOME_FCERI_MEDIATED_NF_KB_ACTIVATION","REACTOME_TP53_REGULATES_TRANSCRIPTION_OF_GENES_INVOLVED_IN_G2_CELL_CYCLE_ARREST","REACTOME_SENESCENCE_ASSOCIATED_SECRETORY_PHENOTYPE_SASP","REACTOME_MEIOSIS","REACTOME_PRC2_METHYLATES_HISTONES_AND_DNA","REACTOME_RESPONSE_OF_EIF2AK1_HRI_TO_HEME_DEFICIENCY","REACTOME_NEUTROPHIL_DEGRANULATION","REACTOME_ATF6_ATF6_ALPHA_ACTIVATES_CHAPERONE_GENES","GO:0048525","GO:0016032","REACTOME_RNA_POLYMERASE_II_TRANSCRIPTION_TERMINATION","REACTOME_ABERRANT_REGULATION_OF_MITOTIC_G1_S_TRANSITION_IN_CANCER_DUE_TO_RB1_DEFECTS","REACTOME_TELOMERE_MAINTENANCE","REACTOME_HDR_THROUGH_MMEJ_ALT_NHEJ","REACTOME_E2F_MEDIATED_REGULATION_OF_DNA_REPLICATION","REACTOME_PTEN_REGULATION","REACTOME_HEDGEHOG_LIGAND_BIOGENESIS","REACTOME_MITOCHONDRIAL_PROTEIN_IMPORT","GO:0006888","REACTOME_ASYMMETRIC_LOCALIZATION_OF_PCP_PROTEINS","GO:0016032","REACTOME_TRANSCRIPTION_OF_E2F_TARGETS_UNDER_NEGATIVE_CONTROL_BY_P107_RBL1_AND_P130_RBL2_IN_COMPLEX_WITH_HDAC1","REACTOME_UNFOLDED_PROTEIN_RESPONSE_UPR","REACTOME_CONDENSATION_OF_PROPHASE_CHROMOSOMES","REACTOME_CROSS_PRESENTATION_OF_SOLUBLE_EXOGENOUS_ANTIGENS_ENDOSOMES","REACTOME_CYCLIN_A_CDK2_ASSOCIATED_EVENTS_AT_S_PHASE_ENTRY","GO:0006979","REACTOME_PCP_CE_PATHWAY","REACTOME_REGULATION_OF_PTEN_STABILITY_AND_ACTIVITY","REACTOME_DISEASES_OF_PROGRAMMED_CELL_DEATH","REACTOME_POLYMERASE_SWITCHING_ON_THE_C_STRAND_OF_THE_TELOMERE","REACTOME_G1_S_DNA_DAMAGE_CHECKPOINTS","REACTOME_REGULATION_OF_RAS_BY_GAPS","REACTOME_DEFECTIVE_CFTR_CAUSES_CYSTIC_FIBROSIS","REACTOME_ANCHORING_OF_THE_BASAL_BODY_TO_THE_PLASMA_MEMBRANE","REACTOME_CYCLIN_A_B1_B2_ASSOCIATED_EVENTS_DURING_G2_M_TRANSITION","GO:1990748","REACTOME_HEDGEHOG_OFF_STATE","REACTOME_DOWNSTREAM_SIGNALING_EVENTS_OF_B_CELL_RECEPTOR_BCR","REACTOME_INHIBITION_OF_DNA_RECOMBINATION_AT_TELOMERE","REACTOME_INITIATION_OF_NUCLEAR_ENVELOPE_NE_REFORMATION","GO:0140467","GO:0006986","GO:0090083","GO:0036294","GO:0035966","REACTOME_MEIOTIC_SYNAPSIS","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX2","REACTOME_GLOBAL_GENOME_NUCLEOTIDE_EXCISION_REPAIR_GG_NER","REACTOME_SENESCENCE_ASSOCIATED_SECRETORY_PHENOTYPE_SASP","REACTOME_SWITCHING_OF_ORIGINS_TO_A_POST_REPLICATIVE_STATE","REACTOME_ORC1_REMOVAL_FROM_CHROMATIN","REACTOME_MEIOTIC_RECOMBINATION","GO:0050657","GO:0050658","REACTOME_RESOLUTION_OF_D_LOOP_STRUCTURES","REACTOME_DUAL_INCISION_IN_GG_NER","REACTOME_MITOCHONDRIAL_BIOGENESIS","REACTOME_DECTIN_1_MEDIATED_NONCANONICAL_NF_KB_SIGNALING","REACTOME_DEGRADATION_OF_AXIN","REACTOME_CELLULAR_RESPONSE_TO_HYPOXIA","REACTOME_CHROMOSOME_MAINTENANCE","REACTOME_RESOLUTION_OF_D_LOOP_STRUCTURES_THROUGH_SYNTHESIS_DEPENDENT_STRAND_ANNEALING_SDSA","REACTOME_SUMOYLATION","GO:0006338","REACTOME_TRANSLESION_SYNTHESIS_BY_Y_FAMILY_DNA_POLYMERASES_BYPASSES_LESIONS_ON_DNA_TEMPLATE","REACTOME_METABOLISM_OF_POLYAMINES","REACTOME_REGULATION_OF_MRNA_STABILITY_BY_PROTEINS_THAT_BIND_AU_RICH_ELEMENTS","REACTOME_REGULATION_OF_RUNX2_EXPRESSION_AND_ACTIVITY","REACTOME_NUCLEAR_ENVELOPE_NE_REASSEMBLY","REACTOME_TRANSPORT_OF_MATURE_TRANSCRIPT_TO_CYTOPLASM","REACTOME_POLO_LIKE_KINASE_MEDIATED_EVENTS","REACTOME_REPRODUCTION","REACTOME_OXIDATIVE_STRESS_INDUCED_SENESCENCE","GO:0048525","REACTOME_DISEASES_OF_PROGRAMMED_CELL_DEATH","REACTOME_ANTIGEN_PROCESSING_CROSS_PRESENTATION","REACTOME_DEGRADATION_OF_DVL","REACTOME_STABILIZATION_OF_P53","REACTOME_HIV_INFECTION","REACTOME_MITOTIC_PROPHASE","REACTOME_MISMATCH_REPAIR","REACTOME_PROCESSIVE_SYNTHESIS_ON_THE_LAGGING_STRAND","REACTOME_SIGNALING_BY_NOTCH4","REACTOME_TP53_REGULATES_TRANSCRIPTION_OF_CELL_CYCLE_GENES","REACTOME_REGULATION_OF_RUNX3_EXPRESSION_AND_ACTIVITY","REACTOME_SCF_BETA_TRCP_MEDIATED_DEGRADATION_OF_EMI1","GO:0048511","REACTOME_CONDENSATION_OF_PROMETAPHASE_CHROMOSOMES","REACTOME_SIGNALING_BY_NUCLEAR_RECEPTORS","REACTOME_REGULATION_OF_PLK1_ACTIVITY_AT_G2_M_TRANSITION","REACTOME_DEGRADATION_OF_GLI1_BY_THE_PROTEASOME","REACTOME_DEPOSITION_OF_NEW_CENPA_CONTAINING_NUCLEOSOMES_AT_THE_CENTROMERE","REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA","REACTOME_THE_ROLE_OF_GTSE1_IN_G2_M_PROGRESSION_AFTER_G2_CHECKPOINT","GO:0050678","REACTOME_POLYMERASE_SWITCHING","REACTOME_KEAP1_NFE2L2_PATHWAY","REACTOME_HOST_INTERACTIONS_OF_HIV_FACTORS","REACTOME_TP53_REGULATES_METABOLIC_GENES","REACTOME_TELOMERE_MAINTENANCE","REACTOME_TERMINATION_OF_TRANSLESION_DNA_SYNTHESIS","REACTOME_RECRUITMENT_OF_MITOTIC_CENTROSOME_PROTEINS_AND_COMPLEXES","REACTOME_M_PHASE","REACTOME_SCF_SKP2_MEDIATED_DEGRADATION_OF_P27_P21","REACTOME_NEGATIVE_REGULATION_OF_NOTCH4_SIGNALING","REACTOME_MRNA_SPLICING","REACTOME_DNA_DAMAGE_BYPASS","REACTOME_UNWINDING_OF_DNA","GO:0034976","GO:2000112","REACTOME_CELLULAR_SENESCENCE","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_E2F6","REACTOME_ESR_MEDIATED_SIGNALING","REACTOME_MEIOSIS","REACTOME_EXTENSION_OF_TELOMERES","REACTOME_AURKA_ACTIVATION_BY_TPX2","REACTOME_AUF1_HNRNP_D0_BINDS_AND_DESTABILIZES_MRNA","REACTOME_RECRUITMENT_OF_NUMA_TO_MITOTIC_CENTROSOMES","REACTOME_ESTROGEN_DEPENDENT_GENE_EXPRESSION","GO:0000377","GO:0000398","GO:0000375","REACTOME_RECOGNITION_OF_DNA_DAMAGE_BY_PCNA_CONTAINING_REPLICATION_COMPLEX","REACTOME_GAP_FILLING_DNA_REPAIR_SYNTHESIS_AND_LIGATION_IN_GG_NER","REACTOME_TELOMERE_C_STRAND_LAGGING_STRAND_SYNTHESIS","REACTOME_NUCLEAR_EVENTS_MEDIATED_BY_NFE2L2","REACTOME_UNFOLDED_PROTEIN_RESPONSE_UPR","REACTOME_G0_AND_EARLY_G1","REACTOME_HDR_THROUGH_SINGLE_STRAND_ANNEALING_SSA","GO:0010212","REACTOME_PCNA_DEPENDENT_LONG_PATCH_BASE_EXCISION_REPAIR","REACTOME_DEPOSITION_OF_NEW_CENPA_CONTAINING_NUCLEOSOMES_AT_THE_CENTROMERE","REACTOME_G1_S_SPECIFIC_TRANSCRIPTION","REACTOME_LAGGING_STRAND_SYNTHESIS","REACTOME_PROCESSING_OF_DNA_DOUBLE_STRAND_BREAK_ENDS","REACTOME_BASE_EXCISION_REPAIR","REACTOME_REGULATION_OF_TP53_ACTIVITY","REACTOME_DNA_REPLICATION_PRE_INITIATION","REACTOME_MRNA_SPLICING","REACTOME_RESOLUTION_OF_AP_SITES_VIA_THE_MULTIPLE_NUCLEOTIDE_PATCH_REPLACEMENT_PATHWAY","REACTOME_G2_M_DNA_DAMAGE_CHECKPOINT","REACTOME_REGULATION_OF_TP53_ACTIVITY_THROUGH_PHOSPHORYLATION","REACTOME_DISEASES_OF_DNA_REPAIR","REACTOME_TRANSLATION","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_TP53","REACTOME_HOMOLOGOUS_DNA_PAIRING_AND_STRAND_EXCHANGE","REACTOME_RESOLUTION_OF_ABASIC_SITES_AP_SITES","REACTOME_CELLULAR_RESPONSE_TO_CHEMICAL_STRESS","REACTOME_CRISTAE_FORMATION","REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA","REACTOME_MITOCHONDRIAL_TRANSLATION","REACTOME_RHO_GTPASE_EFFECTORS","REACTOME_FORMATION_OF_ATP_BY_CHEMIOSMOTIC_COUPLING","REACTOME_ACTIVATION_OF_ATR_IN_RESPONSE_TO_REPLICATION_STRESS","REACTOME_NEUTROPHIL_DEGRANULATION","REACTOME_MITOTIC_G2_G2_M_PHASES","REACTOME_HDR_THROUGH_HOMOLOGOUS_RECOMBINATION_HRR","REACTOME_ACTIVATION_OF_THE_PRE_REPLICATIVE_COMPLEX","REACTOME_RHO_GTPASES_ACTIVATE_FORMINS","GO:0140694","REACTOME_SYNTHESIS_OF_DNA","REACTOME_MITOTIC_SPINDLE_CHECKPOINT","REACTOME_CHROMOSOME_MAINTENANCE","REACTOME_SEPARATION_OF_SISTER_CHROMATIDS","REACTOME_DNA_REPLICATION","REACTOME_HOMOLOGY_DIRECTED_REPAIR","REACTOME_DNA_STRAND_ELONGATION","REACTOME_DNA_DOUBLE_STRAND_BREAK_REPAIR","GO:1902600","REACTOME_G2_M_CHECKPOINTS","REACTOME_DNA_REPAIR","REACTOME_RESOLUTION_OF_SISTER_CHROMATID_COHESION","REACTOME_COMPLEX_I_BIOGENESIS","REACTOME_MITOTIC_G1_PHASE_AND_G1_S_TRANSITION","GO:0033108","REACTOME_MITOTIC_METAPHASE_AND_ANAPHASE","REACTOME_S_PHASE","REACTOME_MITOTIC_PROMETAPHASE","REACTOME_M_PHASE","REACTOME_RESPIRATORY_ELECTRON_TRANSPORT","REACTOME_CELL_CYCLE_CHECKPOINTS","REACTOME_THE_CITRIC_ACID_TCA_CYCLE_AND_RESPIRATORY_ELECTRON_TRANSPORT","GO:0000280","GO:0006119","GO:0007059","REACTOME_RESPIRATORY_ELECTRON_TRANSPORT_ATP_SYNTHESIS_BY_CHEMIOSMOTIC_COUPLING_AND_HEAT_PRODUCTION_BY_UNCOUPLING_PROTEINS"],["Mitochondrial Fatty Acid Beta Oxidation","Hsf1 Activation","Copi Independent Golgi To Er Retrograde Traffic","Senescence Associated Secretory Phenotype Sasp","Rho Gtpases Activate Formins","Downregulation Of Tgf Beta Receptor Signaling","Golgi To Er Retrograde Transport","Amyloid Fiber Formation","Apc C Mediated Degradation Of Cell Cycle Proteins","Organelle Biogenesis And Maintenance","Meiotic Synapsis","Trna Processing In The Nucleus","Srp Dependent Cotranslational Protein Targeting To Membrane","Hats Acetylate Histones","Regulation Of Mrna Stability By Proteins That Bind Au Rich Elements","Copi Mediated Anterograde Transport","Attenuation Phase","Separation Of Sister Chromatids","Trans Golgi Network Vesicle Budding","Detoxification Of Reactive Oxygen Species","Rho Gtpases Activate Wasps And Waves","Activation Of Ampk Downstream Of Nmdars","Dna Replication Pre Initiation","Deactivation Of The Beta Catenin Transactivating Complex","Interleukin 4 And Interleukin 13 Signaling","Pre Notch Expression And Processing","Sars Cov Infections","Er To Golgi Anterograde Transport","Golgi Associated Vesicle Biogenesis","Defective Intrinsic Pathway For Apoptosis","Mhc Class Ii Antigen Presentation","Glutathione Conjugation","Lysosome Vesicle Biogenesis","Response To Elevated Platelet Cytosolic Ca2","Cellular Response To Heat Stress","G1 S Dna Damage Checkpoints","Transport Of The Slbp Dependant Mature Mrna","Smooth Muscle Contraction","Host Interactions Of Hiv Factors","Regulation Of Cholesterol Biosynthesis By Srebp Srebf","GO:Circadian Rhythm","Aurka Activation By Tpx2","Ercc6 Csb And Ehmt2 G9a Positively Regulate Rrna Expression","Gluconeogenesis","Deposition Of New Cenpa Containing Nucleosomes At The Centromere","Dna Damage Telomere Stress Induced Senescence","Nonhomologous End Joining Nhej","GO:Non-Membrane-Bounded Organelle Assembly","GO:Homotypic Cell-Cell Adhesion","Late Endosomal Microautophagy","Positive Epigenetic Regulation Of Rrna Expression","Condensation Of Prophase Chromosomes","Prc2 Methylates Histones And Dna","Formation Of Tubulin Folding Intermediates By Cct Tric","Recruitment Of Numa To Mitotic Centrosomes","Apoptotic Cleavage Of Cell Adhesion Proteins","Intrinsic Pathway For Apoptosis","Metabolism Of Steroids","Pre Notch Expression And Processing","Neutrophil Degranulation","Aggrephagy","Runx1 Regulates Genes Involved In Megakaryocyte Differentiation And Platelet Function","GO:Regulation Of Intrinsic Apoptotic Signaling Pathway","GO:Atp Metabolic Process","Dna Replication","Rho Gtpases Activate Rocks","Assembly Of The Orc Complex At The Origin Of Replication","Nuclear Import Of Rev Protein","Oxidative Stress Induced Senescence","Assembly Of The Orc Complex At The Origin Of Replication","Hsf1 Dependent Transactivation","Activation Of Gene Expression By Srebf Srebp","Sirt1 Negatively Regulates Rrna Expression","Activation Of Anterior Hox Genes In Hindbrain Development During Early Embryogenesis","Estrogen Dependent Gene Expression","Activated Pkn1 Stimulates Transcription Of Ar Androgen Receptor Regulated Genes Klk2 And Klk3","Inhibition Of Dna Recombination At Telomere","Post Chaperonin Tubulin Folding Pathway","Estrogen Dependent Gene Expression","Ngf Stimulated Transcription","Signaling By Nuclear Receptors","Signaling By Notch","Cohesin Loading Onto Chromatin","Dna Methylation","Protein Ubiquitination","Interferon Signaling","Base Excision Repair Ap Site Formation","GO:Positive Regulation Of Interferon-Beta Production","GO:Regulation Of Transcription From Rna Polymerase Ii Promoter In Response To Stress","Interferon Alpha Beta Signaling","Transport Of Connexons To The Plasma Membrane","Resolution Of Sister Chromatid Cohesion","Signaling By Notch","GO:Viral Transcription","GO:Cellular Response To Ionizing Radiation","Response To Elevated Platelet Cytosolic Ca2","Esr Mediated Signaling","Metabolism Of Nucleotides","Selective Autophagy","Trans Golgi Network Vesicle Budding","Chromatin Modifying Enzymes","GO:Viral Entry Into Host Cell","Sema4d Induced Cell Migration And Growth Cone Collapse","Antiviral Mechanism By Ifn Stimulated Genes","Tp53 Regulates Transcription Of Dna Repair Genes","GO:Endothelial Cell Migration","Cytosolic Trna Aminoacylation","Interleukin 4 And Interleukin 13 Signaling","Er To Golgi Anterograde Transport","Rho Gtpases Activate Cit","Rho Gtpases Activate Rocks","Pre Notch Expression And Processing","Defective Intrinsic Pathway For Apoptosis","Mtorc1 Mediated Signalling","GO:Positive Regulation Of Epithelial Cell Migration","Dna Methylation","Ercc6 Csb And Ehmt2 G9a Positively Regulate Rrna Expression","Glycolysis","Trna Aminoacylation","Processing Of Dna Double Strand Break Ends","Initiation Of Nuclear Envelope Ne Reformation","Signaling By Flt3 Fusion Proteins","Recognition And Association Of Dna Glycosylase With Site Containing An Affected Purine","Pkmts Methylate Histone Lysines","GO:Regulation Of Protein Stability","Diseases Of Programmed Cell Death","Depolymerisation Of The Nuclear Lamina","Regulation Of Hsf1 Mediated Heat Shock Response","Activation Of Anterior Hox Genes In Hindbrain Development During Early Embryogenesis","Rho Gtpases Activate Iqgaps","Fanconi Anemia Pathway","Atf4 Activates Genes In Response To Endoplasmic Reticulum Stress","Interleukin 1 Family Signaling","Signaling By Robo Receptors","Recycling Pathway Of L1","Metabolism Of Amino Acids And Derivatives","Interconversion Of Nucleotide Di And Triphosphates","Transcriptional Regulation By Tp53","The Role Of Gtse1 In G2 M Progression After G2 Checkpoint","Aggrephagy","GO:Cellular Response To Epidermal Growth Factor Stimulus","Interactions Of Vpr With Host Cellular Proteins","Response Of Eif2ak1 Hri To Heme Deficiency","Copi Dependent Golgi To Er Retrograde Traffic","Signaling By Fgfr","Attenuation Phase","Ub Specific Processing Proteases","Processing Of Capped Intron Containing Pre Mrna","Mrna Splicing","Beta Catenin Independent Wnt Signaling","Asparagine N Linked Glycosylation","Nuclear Pore Complex Npc Disassembly","Dual Incision In Tc Ner","Inhibition Of Replication Initiation Of Damaged Dna By Rb1 E2f1","Mitotic Telophase Cytokinesis","Neddylation","Postmitotic Nuclear Pore Complex Npc Reformation","Iron Uptake And Transport","Smooth Muscle Contraction","Signaling By Notch","Attenuation Phase","Flt3 Signaling In Disease","Estrogen Dependent Gene Expression","GO:Maintenance Of Cell Number","Atf4 Activates Genes In Response To Endoplasmic Reticulum Stress","Abc Family Proteins Mediated Transport","GO:Regulation Of Mitochondrial Membrane Potential","Asparagine N Linked Glycosylation","Hsp90 Chaperone Cycle For Steroid Hormone Receptors Shr In The Presence Of Ligand","Nuclear Events Kinase And Transcription Factor Activation","GO:Regulation Of Peptidase Activity","S Phase","Cooperation Of Prefoldin And Tric Cct In Actin And Tubulin Folding","Dna Repair","GO:Muscle Cell Differentiation","Sema4d In Semaphorin Signaling","Hcmv Infection","Hsf1 Dependent Transactivation","Formation Of The Cornified Envelope","Keratinization","Transcriptional Regulation By Runx1","Mitotic Prometaphase","Regulation Of Cholesterol Biosynthesis By Srebp Srebf","Export Of Viral Ribonucleoproteins From Nucleus","GO:Circadian Rhythm","Dna Damage Telomere Stress Induced Senescence","Mitotic G1 Phase And G1 S Transition","Dna Double Strand Break Response","Mitotic G2 G2 M Phases","Antiviral Mechanism By Ifn Stimulated Genes","Ns1 Mediated Effects On Host Pathways","GO:Regulation Of Binding","Transcriptional Regulation By Small Rnas","Perk Regulates Gene Expression","GO:Nucleocytoplasmic Transport","GO:Nuclear Transport","Runx1 Regulates Transcription Of Genes Involved In Differentiation Of Hscs","GO:Viral Genome Replication","Signaling By Fgfr2","G2 M Checkpoints","G2 M Dna Damage Checkpoint","Runx1 Regulates Genes Involved In Megakaryocyte Differentiation And Platelet Function","Hdacs Deacetylate Histones","Separation Of Sister Chromatids","Interferon Alpha Beta Signaling","Hcmv Late Events","Hcmv Early Events","Gene Silencing By Rna","Signaling By Alk In Cancer","GO:Sterol Biosynthetic Process","Sumoylation Of Dna Replication Proteins","Factors Involved In Megakaryocyte Development And Platelet Production","Hcmv Early Events","Interactions Of Rev With Host Cellular Proteins","GO:Cellular Response To Epidermal Growth Factor Stimulus","Eph Ephrin Signaling","Organelle Biogenesis And Maintenance","C Type Lectin Receptors Clrs","Tcr Signaling","GO:Myeloid Cell Differentiation","Selective Autophagy","Negative Epigenetic Regulation Of Rrna Expression","Interferon Signaling","Hsf1 Activation","Cilium Assembly","Cholesterol Biosynthesis","Activation Of Gene Expression By Srebf Srebp","GO:Negative Regulation Of Stem Cell Differentiation","GO:Vacuolar Acidification","Tp53 Regulates Transcription Of Genes Involved In G1 Cell Cycle Arrest","Hcmv Infection","Fc Epsilon Receptor Fceri Signaling","GO:Positive Regulation Of Apoptotic Process","Interleukin 1 Signaling","Signaling By The B Cell Receptor Bcr","Epigenetic Regulation Of Gene Expression","Condensation Of Prophase Chromosomes","Regulation Of Expression Of Slits And Robos","GO:Peptidyl-Proline Hydroxylation To 4-Hydroxy-L-Proline","Transcription Coupled Nucleotide Excision Repair Tc Ner","Translesion Synthesis By Polh","Oncogene Induced Senescence","Epigenetic Regulation Of Gene Expression","Signaling By Hedgehog","Synthesis Of Dna","Chk1 Chk2 Cds1 Mediated Inactivation Of Cyclin B Cdk1 Complex","Rmts Methylate Histone Arginines","GO:Positive Regulation Of Atp-Dependent Activity","Cytosolic Trna Aminoacylation","Fgfr2 Alternative Splicing","Meiotic Synapsis","Transcriptional Regulation Of Granulopoiesis","Signaling By Wnt","Rho Gtpase Effectors","Hcmv Infection","Perk Regulates Gene Expression","GO:Protein Folding","GO:Nadh Regeneration","GO:Canonical Glycolysis","GO:Glucose Catabolic Process To Pyruvate","Dna Replication","GO:Protein Refolding","GO:Sterol Biosynthetic Process","Uch Proteinases","Ngf Stimulated Transcription","Rho Gtpases Activate Pkns","GO:Protein Stabilization","Diseases Of Mitotic Cell Cycle","Protein Localization","Interferon Alpha Beta Signaling","Base Excision Repair","GO:Viral Process","GO:Positive Regulation Of Programmed Cell Death","Translesion Synthesis By Polk","Rna Polymerase I Transcription","B Wich Complex Positively Regulates Rrna Expression","Rna Polymerase I Promoter Escape","GO:Sterol Biosynthetic Process","Transcriptional Regulation Of Granulopoiesis","Apc C Mediated Degradation Of Cell Cycle Proteins","Pre Notch Expression And Processing","Nuclear Envelope Breakdown","GO:Homotypic Cell-Cell Adhesion","Autophagy","Transcriptional Regulation By Small Rnas","Assembly Of The Orc Complex At The Origin Of Replication","Positive Epigenetic Regulation Of Rrna Expression","Mitotic G2 G2 M Phases","Gluconeogenesis","Abc Transporter Disorders","Sirt1 Negatively Regulates Rrna Expression","Meiotic Recombination","Hedgehog On State","Establishment Of Sister Chromatid Cohesion","Activated Pkn1 Stimulates Transcription Of Ar Androgen Receptor Regulated Genes Klk2 And Klk3","Nucleotide Excision Repair","Oxidative Stress Induced Senescence","Cytoprotection By Hmox1","Atf6 Atf6 Alpha Activates Chaperones","Reproduction","Ire1alpha Activates Chaperones","Apc C Mediated Degradation Of Cell Cycle Proteins","Dna Methylation","Rho Gtpase Effectors","Base Excision Repair Ap Site Formation","Mitotic Prophase","Mitotic Metaphase And Anaphase","Cellular Senescence","Tnfr2 Non Canonical Nf Kb Pathway","Switching Of Origins To A Post Replicative State","Signaling By Notch","Cholesterol Biosynthesis","Sumoylation Of Dna Damage Response And Repair Proteins","Clec7a Dectin 1 Signaling","Mapk6 Mapk4 Signaling","Dna Replication Pre Initiation","Cholesterol Biosynthesis","Neutrophil Degranulation","Hsp90 Chaperone Cycle For Steroid Hormone Receptors Shr In The Presence Of Ligand","Apoptosis","Apc C Cdh1 Mediated Degradation Of Cdc20 And Other Apc C Cdh1 Targeted Proteins In Late Mitosis Early G1","Response Of Eif2ak1 Hri To Heme Deficiency","Cyclin A Cdk2 Associated Events At S Phase Entry","GO:Regulation Of Epithelial Cell Proliferation","Atf6 Atf6 Alpha Activates Chaperones","Ercc6 Csb And Ehmt2 G9a Positively Regulate Rrna Expression","Rho Gtpases Activate Pkns","Dna Damage Telomere Stress Induced Senescence","Hcmv Early Events","Atf6 Atf6 Alpha Activates Chaperone Genes","Recognition And Association Of Dna Glycosylase With Site Containing An Affected Purine","Prc2 Methylates Histones And Dna","Runx1 Regulates Transcription Of Genes Involved In Differentiation Of Hscs","Formation Of The Beta Catenin Tcf Transactivating Complex","Amyloid Fiber Formation","Transcriptional Regulation By Runx3","Cyclin D Associated Events In G1","Processive Synthesis On The C Strand Of The Telomere","Transcription Of E2f Targets Under Negative Control By Dream Complex","Cell Cycle Checkpoints","Cholesterol Biosynthesis","Programmed Cell Death","Orc1 Removal From Chromatin","Activation Of Anterior Hox Genes In Hindbrain Development During Early Embryogenesis","Degradation Of Beta Catenin By The Destruction Complex","Tcf Dependent Signaling In Response To Wnt","Kinesins","Fceri Mediated Nf Kb Activation","Tp53 Regulates Transcription Of Genes Involved In G2 Cell Cycle Arrest","Senescence Associated Secretory Phenotype Sasp","Meiosis","Prc2 Methylates Histones And Dna","Response Of Eif2ak1 Hri To Heme Deficiency","Neutrophil Degranulation","Atf6 Atf6 Alpha Activates Chaperone Genes","GO:Negative Regulation Of Viral Process","GO:Viral Process","Rna Polymerase Ii Transcription Termination","Aberrant Regulation Of Mitotic G1 S Transition In Cancer Due To Rb1 Defects","Telomere Maintenance","Hdr Through Mmej Alt Nhej","E2f Mediated Regulation Of Dna Replication","Pten Regulation","Hedgehog Ligand Biogenesis","Mitochondrial Protein Import","GO:Endoplasmic Reticulum To Golgi Vesicle-Mediated Transport","Asymmetric Localization Of Pcp Proteins","GO:Viral Process","Transcription Of E2f Targets Under Negative Control By P107 Rbl1 And P130 Rbl2 In Complex With Hdac1","Unfolded Protein Response Upr","Condensation Of Prophase Chromosomes","Cross Presentation Of Soluble Exogenous Antigens Endosomes","Cyclin A Cdk2 Associated Events At S Phase Entry","GO:Response To Oxidative Stress","Pcp Ce Pathway","Regulation Of Pten Stability And Activity","Diseases Of Programmed Cell Death","Polymerase Switching On The C Strand Of The Telomere","G1 S Dna Damage Checkpoints","Regulation Of Ras By Gaps","Defective Cftr Causes Cystic Fibrosis","Anchoring Of The Basal Body To The Plasma Membrane","Cyclin A B1 B2 Associated Events During G2 M Transition","GO:Cellular Detoxification","Hedgehog Off State","Downstream Signaling Events Of B Cell Receptor Bcr","Inhibition Of Dna Recombination At Telomere","Initiation Of Nuclear Envelope Ne Reformation","GO:Integrated Stress Response Signaling","GO:Response To Unfolded Protein","GO:Regulation Of Inclusion Body Assembly","GO:Cellular Response To Decreased Oxygen Levels","GO:Response To Topologically Incorrect Protein","Meiotic Synapsis","Transcriptional Regulation By Runx2","Global Genome Nucleotide Excision Repair Gg Ner","Senescence Associated Secretory Phenotype Sasp","Switching Of Origins To A Post Replicative State","Orc1 Removal From Chromatin","Meiotic Recombination","GO:Nucleic Acid Transport","GO:Rna Transport","Resolution Of D Loop Structures","Dual Incision In Gg Ner","Mitochondrial Biogenesis","Dectin 1 Mediated Noncanonical Nf Kb Signaling","Degradation Of Axin","Cellular Response To Hypoxia","Chromosome Maintenance","Resolution Of D Loop Structures Through Synthesis Dependent Strand Annealing Sdsa","Sumoylation","GO:Chromatin Remodeling","Translesion Synthesis By Y Family Dna Polymerases Bypasses Lesions On Dna Template","Metabolism Of Polyamines","Regulation Of Mrna Stability By Proteins That Bind Au Rich Elements","Regulation Of Runx2 Expression And Activity","Nuclear Envelope Ne Reassembly","Transport Of Mature Transcript To Cytoplasm","Polo Like Kinase Mediated Events","Reproduction","Oxidative Stress Induced Senescence","GO:Negative Regulation Of Viral Process","Diseases Of Programmed Cell Death","Antigen Processing Cross Presentation","Degradation Of Dvl","Stabilization Of P53","Hiv Infection","Mitotic Prophase","Mismatch Repair","Processive Synthesis On The Lagging Strand","Signaling By Notch4","Tp53 Regulates Transcription Of Cell Cycle Genes","Regulation Of Runx3 Expression And Activity","Scf Beta Trcp Mediated Degradation Of Emi1","GO:Rhythmic Process","Condensation Of Prometaphase Chromosomes","Signaling By Nuclear Receptors","Regulation Of Plk1 Activity At G2 M Transition","Degradation Of Gli1 By The Proteasome","Deposition Of New Cenpa Containing Nucleosomes At The Centromere","Processing Of Capped Intron Containing Pre Mrna","The Role Of Gtse1 In G2 M Progression After G2 Checkpoint","GO:Regulation Of Epithelial Cell Proliferation","Polymerase Switching","Keap1 Nfe2l2 Pathway","Host Interactions Of Hiv Factors","Tp53 Regulates Metabolic Genes","Telomere Maintenance","Termination Of Translesion Dna Synthesis","Recruitment Of Mitotic Centrosome Proteins And Complexes","M Phase","Scf Skp2 Mediated Degradation Of P27 P21","Negative Regulation Of Notch4 Signaling","Mrna Splicing","Dna Damage Bypass","Unwinding Of Dna","GO:Response To Endoplasmic Reticulum Stress","GO:Regulation Of Cellular Macromolecule Biosynthetic Process","Cellular Senescence","Transcriptional Regulation By E2f6","Esr Mediated Signaling","Meiosis","Extension Of Telomeres","Aurka Activation By Tpx2","Auf1 Hnrnp D0 Binds And Destabilizes Mrna","Recruitment Of Numa To Mitotic Centrosomes","Estrogen Dependent Gene Expression","GO:Rna Splicing, Via Transesterification Reactions With Bulged Adenosine As Nucleophile","GO:Mrna Splicing, Via Spliceosome","GO:Rna Splicing, Via Transesterification Reactions","Recognition Of Dna Damage By Pcna Containing Replication Complex","Gap Filling Dna Repair Synthesis And Ligation In Gg Ner","Telomere C Strand Lagging Strand Synthesis","Nuclear Events Mediated By Nfe2l2","Unfolded Protein Response Upr","G0 And Early G1","Hdr Through Single Strand Annealing Ssa","GO:Response To Ionizing Radiation","Pcna Dependent Long Patch Base Excision Repair","Deposition Of New Cenpa Containing Nucleosomes At The Centromere","G1 S Specific Transcription","Lagging Strand Synthesis","Processing Of Dna Double Strand Break Ends","Base Excision Repair","Regulation Of Tp53 Activity","Dna Replication Pre Initiation","Mrna Splicing","Resolution Of Ap Sites Via The Multiple Nucleotide Patch Replacement Pathway","G2 M Dna Damage Checkpoint","Regulation Of Tp53 Activity Through Phosphorylation","Diseases Of Dna Repair","Translation","Transcriptional Regulation By Tp53","Homologous Dna Pairing And Strand Exchange","Resolution Of Abasic Sites Ap Sites","Cellular Response To Chemical Stress","Cristae Formation","Processing Of Capped Intron Containing Pre Mrna","Mitochondrial Translation","Rho Gtpase Effectors","Formation Of Atp By Chemiosmotic Coupling","Activation Of Atr In Response To Replication Stress","Neutrophil Degranulation","Mitotic G2 G2 M Phases","Hdr Through Homologous Recombination Hrr","Activation Of The Pre Replicative Complex","Rho Gtpases Activate Formins","GO:Non-Membrane-Bounded Organelle Assembly","Synthesis Of Dna","Mitotic Spindle Checkpoint","Chromosome Maintenance","Separation Of Sister Chromatids","Dna Replication","Homology Directed Repair","Dna Strand Elongation","Dna Double Strand Break Repair","GO:Proton Transmembrane Transport","G2 M Checkpoints","Dna Repair","Resolution Of Sister Chromatid Cohesion","Complex I Biogenesis","Mitotic G1 Phase And G1 S Transition","GO:Mitochondrial Respiratory Chain Complex Assembly","Mitotic Metaphase And Anaphase","S Phase","Mitotic Prometaphase","M Phase","Respiratory Electron Transport","Cell Cycle Checkpoints","The Citric Acid Tca Cycle And Respiratory Electron Transport","GO:Nuclear Division","GO:Oxidative Phosphorylation","GO:Chromosome Segregation","Respiratory Electron Transport Atp Synthesis By Chemiosmotic Coupling And Heat Production By Uncoupling Proteins"],["8/647","4/159","5/159","7/159","8/159","4/194","15/516","7/159","7/194","26/516","6/159","9/516","16/647","9/194","7/194","15/647","7/647","23/647","12/647","8/647","8/647","4/159","20/647","5/194","8/194","13/516","41/647","20/647","10/477","4/194","16/647","8/647","8/647","18/647","7/159","10/516","7/516","9/647","15/516","10/477","19/675","6/159","6/159","6/257","6/159","6/159","10/516","14/200","9/357","8/647","13/516","6/159","6/159","4/159","7/159","3/159","9/516","11/193","9/193","21/193","5/159","7/159","20/828","11/243","23/647","6/647","10/516","7/516","8/159","6/159","5/194","5/159","6/159","8/159","9/159","6/159","6/159","4/159","11/193","6/193","16/193","14/193","4/516","6/159","7/194","11/194","6/159","5/200","5/200","12/647","4/159","9/194","23/516","10/722","7/250","12/257","22/516","13/516","7/159","12/477","14/194","12/357","4/159","7/159","10/516","12/250","7/477","10/193","19/477","4/159","4/159","8/159","7/647","7/647","12/357","10/516","11/516","9/257","7/257","8/194","4/194","4/194","6/159","11/516","30/722","8/159","5/516","12/516","15/516","5/159","8/516","6/257","21/647","27/647","6/159","40/647","8/647","39/647","12/516","10/647","6/250","8/516","4/159","14/516","8/194","5/159","13/194","29/477","24/477","21/647","35/647","8/516","11/516","5/516","5/516","30/647","7/516","12/647","6/159","29/647","5/194","5/194","18/516","19/675","8/477","17/647","13/828","31/477","12/647","7/194","19/243","23/647","9/647","17/194","22/357","5/159","11/159","6/159","10/159","10/159","14/194","13/194","8/193","8/516","13/243","12/516","22/647","8/194","13/194","13/516","9/516","18/250","15/516","7/257","17/250","17/250","20/647","16/675","8/194","12/194","9/194","9/194","9/194","13/194","9/193","10/194","11/194","11/194","12/477","13/828","10/516","21/516","11/159","9/516","7/243","17/647","36/647","22/647","19/647","20/250","16/647","10/194","13/159","6/194","24/516","9/647","8/193","9/722","10/828","6/516","21/516","22/647","20/200","20/647","20/647","12/194","13/516","26/647","5/357","14/516","7/516","9/516","20/516","24/647","21/647","6/516","14/516","10/675","7/257","6/194","14/516","15/516","19/194","18/159","13/194","10/477","18/357","6/357","6/357","6/357","14/194","6/200","8/200","19/647","7/194","10/159","18/357","10/516","26/647","9/159","10/194","44/828","24/243","7/516","11/194","10/194","10/194","9/243","10/194","18/647","11/194","12/516","10/200","25/647","11/194","9/194","11/194","30/647","11/647","17/647","9/194","10/194","18/647","6/516","9/194","18/516","12/194","15/647","7/477","13/194","13/477","16/516","9/194","20/194","9/194","13/194","17/194","15/194","20/647","19/647","17/194","10/477","15/516","20/647","19/647","14/194","7/159","24/159","9/159","28/647","17/647","8/477","16/516","21/250","6/257","10/194","11/194","10/194","21/516","7/477","9/194","10/194","13/194","11/194","12/194","20/647","12/516","8/516","8/516","20/194","8/193","32/647","15/516","13/194","19/647","18/194","14/516","19/647","8/516","19/516","13/194","15/516","7/257","33/257","6/257","12/243","24/243","15/516","8/516","13/194","7/516","9/516","26/647","17/647","17/647","23/722","17/647","24/250","8/516","15/257","11/194","15/647","20/647","22/200","21/647","18/647","13/194","10/516","18/647","18/647","17/647","19/516","10/516","22/828","24/647","20/647","11/194","9/516","11/357","18/357","8/357","18/357","19/357","12/194","25/647","18/516","14/194","19/516","19/647","18/516","24/675","24/675","12/516","13/516","22/647","18/647","17/647","20/647","16/194","11/516","29/516","24/250","13/516","18/647","22/647","20/647","18/516","19/516","9/516","25/516","23/516","13/200","21/516","24/647","18/647","18/647","39/647","25/516","9/516","9/516","22/647","15/516","18/647","18/647","23/250","8/516","24/194","20/516","19/647","13/194","24/194","22/647","26/243","9/516","26/647","29/647","23/647","23/516","13/516","20/516","30/194","20/647","19/647","22/194","16/516","9/516","40/722","57/722","32/516","14/516","23/194","25/516","17/516","20/516","20/647","23/516","20/194","26/250","26/250","26/250","14/516","13/516","15/516","25/647","25/477","14/516","16/516","28/675","13/516","22/516","15/516","13/516","26/516","25/516","34/516","34/516","40/516","15/516","27/516","27/516","21/516","57/647","56/516","20/516","19/516","47/647","19/647","51/516","33/647","55/516","15/647","20/516","81/647","44/516","27/516","20/516","38/516","59/675","36/516","35/516","39/516","47/516","47/516","41/516","22/516","46/516","44/828","46/516","68/516","44/516","33/647","49/516","40/828","61/516","52/516","66/516","95/516","56/647","86/516","78/647","113/675","70/828","104/675","72/647"],["37/10811","31/10811","53/10811","108/10811","140/10811","26/10811","134/10811","106/10811","88/10811","292/10811","76/10811","58/10811","113/10811","140/10811","87/10811","102/10811","28/10811","191/10811","72/10811","36/10811","36/10811","29/10811","156/10811","42/10811","111/10811","105/10811","413/10811","155/10811","56/10811","24/10811","111/10811","35/10811","35/10811","132/10811","101/10811","68/10811","36/10811","43/10811","130/10811","55/10811","208/17124","72/10811","72/10811","34/10811","71/10811","71/10811","67/10811","379/17124","95/17124","34/10811","102/10811","69/10811","69/10811","26/10811","95/10811","11/10811","55/10811","154/10811","105/10811","475/10811","44/10811","93/10811","162/17124","203/17124","184/10811","19/10811","65/10811","34/10811","120/10811","65/10811","38/10811","42/10811","64/10811","118/10811","146/10811","63/10811","63/10811","23/10811","146/10811","39/10811","292/10811","232/10811","10/10811","61/10811","78/10811","184/10811","60/10811","39/17124","39/17124","66/10811","21/10811","126/10811","232/10811","51/17124","74/17124","132/10811","217/10811","97/10811","82/10811","72/10811","272/10811","150/17124","20/10811","81/10811","62/10811","216/17124","24/10811","111/10811","155/10811","19/10811","19/10811","105/10811","24/10811","24/10811","148/17124","61/10811","72/10811","72/10811","42/10811","96/10811","19/10811","19/10811","53/10811","71/10811","316/17124","100/10811","15/10811","82/10811","118/10811","32/10811","39/10811","27/10811","152/10811","218/10811","49/10811","374/10811","29/10811","360/10811","79/10811","44/10811","43/17124","37/10811","15/10811","100/10811","87/10811","28/10811","219/10811","283/10811","212/10811","146/10811","305/10811","36/10811","65/10811","13/10811","13/10811","245/10811","27/10811","58/10811","43/10811","232/10811","28/10811","28/10811","146/10811","180/17124","27/10811","103/10811","68/17124","305/10811","57/10811","61/10811","433/17124","163/10811","33/10811","330/10811","373/17124","24/10811","160/10811","38/10811","129/10811","129/10811","234/10811","203/10811","55/10811","33/10811","208/17124","71/10811","149/10811","77/10811","200/10811","81/10811","41/10811","366/17124","103/10811","32/10811","326/17124","326/17124","126/10811","127/17124","73/10811","166/10811","93/10811","93/10811","92/10811","191/10811","66/10811","114/10811","136/10811","135/10811","55/10811","60/17124","46/10811","168/10811","136/10811","37/10811","43/17124","92/10811","292/10811","140/10811","110/10811","399/17124","82/10811","105/10811","184/10811","31/10811","201/10811","27/10811","42/10811","27/17124","33/17124","14/10811","160/10811","134/10811","486/17124","114/10811","114/10811","144/10811","69/10811","172/10811","10/17124","78/10811","19/10811","33/10811","144/10811","150/10811","121/10811","13/10811","77/10811","44/17124","24/10811","27/10811","76/10811","86/10811","326/10811","320/10811","160/10811","32/10811","210/17124","16/17124","16/17124","16/17124","184/10811","26/17124","60/17124","100/10811","39/10811","90/10811","207/17124","38/10811","164/10811","66/10811","88/10811","411/17124","499/17124","17/10811","107/10811","87/10811","87/10811","60/17124","86/10811","88/10811","105/10811","53/10811","95/17124","151/10811","103/10811","65/10811","102/10811","200/10811","34/10811","78/10811","64/10811","82/10811","86/10811","11/10811","63/10811","110/10811","120/10811","61/10811","12/10811","140/10811","50/10811","88/10811","61/10811","320/10811","60/10811","138/10811","236/10811","185/10811","101/10811","92/10811","232/10811","27/10811","77/10811","100/10811","91/10811","156/10811","27/10811","475/10811","57/10811","173/10811","74/10811","15/10811","85/10811","358/17124","12/10811","72/10811","90/10811","71/10811","136/10811","10/10811","53/10811","69/10811","126/10811","87/10811","106/10811","96/10811","47/10811","19/10811","19/10811","290/10811","27/10811","205/10811","71/10811","118/10811","85/10811","229/10811","61/10811","84/10811","18/10811","108/10811","114/10811","69/10811","15/10811","475/10811","10/10811","89/17124","411/17124","67/10811","17/10811","108/10811","12/10811","22/10811","139/10811","65/10811","65/10811","129/17124","64/10811","411/17124","16/10811","92/10811","69/10811","49/10811","85/10811","413/17124","92/10811","69/10811","100/10811","26/10811","68/10811","68/10811","61/10811","97/10811","25/10811","109/17124","113/10811","81/10811","63/10811","19/10811","40/17124","138/17124","17/17124","142/17124","160/17124","76/10811","119/10811","84/10811","108/10811","92/10811","71/10811","82/10811","160/17124","160/17124","35/10811","41/10811","91/10811","62/10811","55/10811","75/10811","135/10811","27/10811","187/10811","353/17124","39/10811","59/10811","87/10811","72/10811","76/10811","84/10811","16/10811","140/10811","120/10811","89/17124","100/10811","99/10811","57/10811","57/10811","230/10811","138/10811","15/10811","15/10811","82/10811","49/10811","55/10811","55/10811","300/17124","11/10811","292/10811","87/10811","60/10811","71/10811","283/10811","79/10811","358/17124","14/10811","107/10811","130/10811","84/10811","108/10811","32/10811","81/10811","412/10811","60/10811","54/10811","212/10811","48/10811","12/10811","258/17124","481/17124","185/10811","34/10811","217/10811","114/10811","51/10811","72/10811","55/10811","95/10811","146/10811","296/17124","296/17124","300/17124","30/10811","25/10811","34/10811","79/10811","92/10811","27/10811","37/10811","141/17124","21/10811","71/10811","29/10811","20/10811","96/10811","88/10811","160/10811","156/10811","212/10811","25/10811","93/10811","92/10811","51/10811","293/10811","360/10811","43/10811","38/10811","195/10811","29/10811","283/10811","94/10811","320/10811","16/10811","37/10811","475/10811","200/10811","68/10811","33/10811","140/10811","379/17124","121/10811","111/10811","135/10811","191/10811","184/10811","137/10811","32/10811","167/10811","147/17124","166/10811","330/10811","126/10811","51/10811","149/10811","92/17124","236/10811","163/10811","203/10811","412/10811","93/10811","290/10811","166/10811","478/17124","128/17124","371/17124","115/10811"],[0.001292941487943968,0.001040164584969624,0.001049320376914707,0.001049696739606582,0.001062488308921978,0.001102897953270784,0.001792609202372484,0.0009401198907788209,0.0010014348633927,0.001627198769118423,0.000861495651091529,0.001578370751449568,0.001087538188352792,0.0009293114866733079,0.0009357271991951803,0.001039171968254323,0.00104259751422563,0.001044441948662134,0.001068052870300506,0.00106845488105775,0.00106845488105775,0.0008032876749660111,0.001002276930066247,0.0008767781765342871,0.0008488428564614339,0.001429145743978535,0.0009352502298920896,0.0009241542418297504,0.0001454967909676402,0.0008061777024787461,0.000894159464974265,0.00087621804511883,0.00087621804511883,0.0008576725039535442,0.0007053021476979625,0.001347509354870079,0.001351724251458417,0.0008367242019952717,0.00131992763237636,0.0001243348273774552,0.0005946614574675962,0.0006460725834261302,0.0006460725834261302,0.000130855034515526,0.0005994147692771123,0.0005994147692771123,0.001199362823526247,0.0001521835340887372,0.0001633293362984716,0.0007127342416994495,0.001090299574551224,0.0005139694572310942,0.0005139694572310942,0.0005232621815531096,0.0004876274072146131,0.0004723158202232914,0.001069776802615286,9.623924992017211e-05,0.0001049251612480678,0.0001134032228964695,0.0004415995336286758,0.0004284179979075214,0.0001103457267580905,0.0001574239032612589,0.0006214428550267651,0.0006216287955856351,0.0009428018959568858,0.000946117021495824,0.0003827948221409288,0.0003716497533075159,0.0005482305244148669,0.0003541786920613457,0.0003414407912839593,0.0003416030169453871,0.0002978275566011252,0.0003131897069774494,0.0003131897069774494,0.0003207564962260757,5.933855069500811e-05,5.982530435211421e-05,6.732350107420273e-05,6.967865521121693e-05,0.0008555756429234919,0.0002621981903148826,0.0004843832965109624,0.0004761261882955177,0.0002392820082845315,8.618681296695792e-05,8.618681296695792e-05,0.000475202943347242,0.0002218202082338586,0.0004331176453284931,0.0007131518382482515,4.440312059012811e-05,0.0001017303402235281,7.012909973903178e-05,0.0006848652728643748,0.0006749735036033865,0.0001967245373788689,6.509481344502072e-05,0.000387634252405358,7.339801882472157e-05,0.0001816389625054643,0.0001821508972123498,0.0006436090557070791,8.21074695773689e-05,5.600433677581465e-05,2.811935359510134e-05,5.074494681462352e-05,0.0001469871714884623,0.0001469871714884623,0.000152935524646775,0.0003757500544590597,0.0003757500544590597,6.443776496315707e-05,0.0005633589404397256,0.0005644553082765314,4.851350767301444e-05,5.236130357023959e-05,0.0003202876898662267,0.0003154437792240609,0.0003154437792240609,0.0001194191883420173,0.000499239271144678,3.014994162613699e-05,0.0001085961389164165,0.0004894709382141848,0.000481029338377495,0.0004772407714571913,9.438367980533933e-05,0.0004230035440663137,3.311383359741427e-05,0.000274329158340998,0.0002521033054646494,7.634985929595435e-05,0.0002421553750041042,0.0002197643810937253,0.0002219256719421532,0.0003385604863604513,0.0002132018251795024,3.535480789098535e-05,0.0002887223023502199,5.419652789025161e-05,0.000268262882440896,0.0001619020624770752,4.83022919281534e-05,0.0001575217873207941,2.01955959546154e-05,2.078974756780061e-05,0.0001550833511825442,0.0001529441340280327,0.0002360174521557847,0.000225981661098587,0.0002271583595471193,0.0002271583595471193,0.0001430898115279918,0.000208740493981189,0.0001335322130118866,3.579658259426839e-05,0.0001279940859956807,0.0001241364047579591,0.0001241364047579591,0.0001985671153743587,9.266854181778683e-05,1.431053060987814e-05,0.0001195635669107982,1.90808678328263e-05,1.233037792025277e-05,0.0001118906915859917,0.0001033685345436621,1.424797496608717e-05,0.0001031583417546916,9.793106024939386e-05,9.106438053556257e-05,1.264531326245314e-05,2.190633142695797e-05,2.321026611680865e-05,1.72237647613346e-05,1.95617778133566e-05,1.95617778133566e-05,8.081484746724388e-05,7.32813148934043e-05,5.297495627178856e-06,0.0001228734644824962,8.884897029784235e-06,0.0001197425836058216,7.321351898416729e-05,6.792028508905549e-05,6.290817326238475e-05,0.0001090698813340627,0.0001067721135798624,7.632798956302973e-06,0.0001039872286485692,8.002453438047674e-06,6.385123941377421e-06,6.385123941377421e-06,5.484729687540477e-05,3.963311848931297e-05,4.612501312869092e-05,4.327799048634847e-05,4.202819222811053e-05,4.202819222811053e-05,3.855056614459915e-05,3.904612607392876e-05,2.350974528259745e-06,3.705193222984162e-05,3.23346679412878e-05,3.018569804069997e-05,3.67795671778481e-06,4.494773019190337e-06,4.843936451998202e-05,4.87517994211678e-05,4.932556495171383e-06,4.496237020095524e-05,2.216046709009544e-06,2.7159643677047e-05,2.739784431377182e-05,2.767865475231776e-05,2.518199927864918e-05,1.760118323420568e-06,2.305170022426193e-05,1.805778151857521e-05,3.03856654053042e-06,1.565651130735669e-05,3.078827586044008e-05,1.645545060993351e-05,6.262530399244625e-07,9.475920516374242e-07,2.218689752059987e-06,2.483827726232176e-05,2.33346922133645e-05,1.365155314751665e-05,1.118738202184312e-06,1.219955411005361e-05,1.219955411005361e-05,1.039460439101747e-05,1.896974530755269e-05,1.083719868644137e-05,8.856377897781503e-07,1.629454767806047e-05,1.65303113504517e-05,1.65357847987724e-05,1.572821249369513e-05,8.821780060965757e-06,8.840924120688665e-06,1.478815943909527e-05,1.397020559076402e-05,6.238473863590148e-06,9.683237241033984e-07,6.682136443727284e-06,1.194350145177231e-05,1.172191524699903e-05,6.118403561510231e-06,1.04014801367044e-06,5.840127780847621e-06,6.834569282400988e-07,4.42758303717668e-07,5.285012999882865e-07,5.285012999882865e-07,5.285012999882865e-07,5.475406703637399e-06,4.462255597294691e-07,4.577989728686147e-07,6.063639375087997e-06,5.085216402312971e-06,7.432413491102783e-07,3.576647419327445e-07,7.928035275074712e-06,4.466908288176881e-06,4.640541817755865e-07,3.678641071857427e-06,6.709623338319903e-07,1.953316007289113e-07,6.936624292840491e-06,3.233252806004788e-06,3.311663696654133e-06,3.311663696654133e-06,1.584471317237203e-07,2.97690579681875e-06,3.566285471421976e-06,2.68220577173937e-06,5.462745525522624e-06,1.609590254825008e-07,3.061671056693016e-06,2.215281649565174e-06,2.152001943366814e-06,2.009807398861494e-06,2.713011798485688e-06,2.611145665350155e-06,2.65324217261394e-06,1.882917683165817e-06,1.913878614600179e-06,2.518281198632775e-06,4.322612649530159e-06,1.643344924565883e-06,4.100574908072754e-06,1.539256595668735e-06,2.060056186535052e-06,2.035885696931307e-07,1.307938526437654e-06,1.619583586314186e-07,3.427023766976393e-06,1.241931041938039e-06,1.172180742799947e-06,1.075196074211686e-06,1.109769706354773e-06,1.125325362680136e-06,1.126212677405056e-06,1.76778733650876e-06,1.646229798275394e-06,8.867925552077258e-07,1.091011599358981e-07,2.823332826309666e-06,1.50022856627888e-06,1.382145944444357e-06,7.669644778250608e-07,9.044105692243513e-08,1.060837052002595e-07,1.2626924806362e-07,1.261217110403958e-06,1.215663866602547e-06,6.634090537024315e-08,2.126930228835518e-06,7.144329178823307e-08,1.3956776227732e-07,5.604076987684455e-07,5.660181599013116e-07,4.901375781190899e-07,1.734843294326779e-06,3.33312216548871e-08,3.59184703413702e-07,3.723518548461988e-07,3.87567008549652e-07,3.992699680862322e-07,4.000100271722517e-07,7.587728021037288e-07,1.39045867287715e-06,1.209621804356468e-06,1.209621804356468e-06,2.456175956425284e-07,1.481640845096304e-08,5.114080943460108e-07,9.541167980172802e-07,1.790159800908604e-07,4.547172368007492e-07,1.458725914830586e-07,7.569643203741784e-07,3.736093519050149e-07,7.307032402735339e-07,7.105652592449255e-07,1.187319583488934e-07,6.459143927402633e-07,2.163056382835224e-08,2.660441619483894e-08,3.30197917525317e-08,4.595540662094457e-09,4.948100790020579e-09,4.304051578084385e-07,4.235750131703181e-07,6.198862474467561e-08,3.483430614963731e-07,3.416951829159265e-07,1.680383215074821e-07,1.642774104650967e-07,1.642774104650967e-07,4.109904435590856e-09,1.283473059860405e-07,8.6745022586113e-09,2.339888204382142e-07,3.720057556207805e-09,3.444292372445549e-08,9.015433367275721e-08,9.18661727935077e-08,3.242626176487361e-09,7.70713811835604e-08,7.422326602773897e-08,2.426521058685539e-08,1.496378557756513e-07,5.80496294267521e-08,5.80496294267521e-08,5.915735549307798e-08,1.227798671879025e-07,9.616156114492065e-08,9.693217353559681e-09,4.053088439878209e-08,3.853362810046014e-08,1.277238242668609e-08,7.219021465949663e-08,3.733558056534487e-10,6.003146464174033e-10,6.811688752970136e-10,9.630542323986332e-10,1.00956770042041e-09,9.019221732295468e-09,2.664959022702474e-08,6.212274617478466e-08,7.159944577980224e-09,5.026063795996382e-08,2.045597785048031e-08,4.17850119345944e-08,1.755858971661559e-08,1.755858971661559e-08,3.754755211440166e-08,2.955320523478531e-08,1.225041827538948e-08,1.183070866502832e-08,1.055104352852764e-08,9.264234252656111e-09,2.213066902216981e-09,1.712275475347999e-08,1.676021131931674e-08,4.245175703550899e-10,1.487717551637006e-08,4.909067130817759e-09,4.974019208726965e-09,4.26708016510833e-09,1.168727813395086e-08,1.036969351791332e-08,1.017178735170217e-08,9.131646798788854e-09,8.901044837159874e-09,3.287068808050314e-11,7.219862303278396e-09,2.502672320664279e-09,2.634539962959132e-09,2.634539962959132e-09,2.652764843721746e-09,6.72836482361246e-09,4.645978020056119e-09,4.645978020056119e-09,1.477497363347484e-09,4.350436987158521e-09,1.369757681864259e-09,1.369757681864259e-09,9.364150931721804e-11,3.711884023666566e-09,4.231307697935309e-10,3.131480738125318e-09,8.999368228751026e-10,3.12499264773209e-10,2.222742784630583e-10,6.771155195085742e-10,9.22990623419286e-12,1.940198625808846e-09,5.10830331031971e-10,4.586224268328905e-10,4.008129339503198e-10,1.029757559552485e-09,8.661136677487834e-10,8.090813328288118e-10,4.641894328785155e-11,1.132729057829241e-10,1.135823601374623e-10,2.64358776537335e-11,3.146289467297208e-10,2.32404399952759e-10,8.579857308399849e-13,1.55750986242263e-12,1.736979630461984e-10,1.542087915770925e-10,5.822849942215227e-12,9.797194236047265e-11,8.709258842514231e-11,8.040855077926861e-11,1.812943827490233e-11,6.611509038777495e-11,1.274097720394613e-12,2.273434051653408e-13,2.273434051653408e-13,3.109717288414311e-13,1.923117622011721e-11,1.765688241460176e-11,9.968424532614112e-12,1.968237091022373e-12,9.884367906987263e-14,3.029476669105223e-12,2.930383356145841e-12,1.15011450288525e-12,8.243878095224921e-13,7.712413935737457e-13,5.201388465111207e-13,3.282355731409583e-13,2.160674526395684e-13,1.906575822750235e-13,1.019548796363172e-13,4.640403043142597e-14,4.301788144815215e-14,2.618510512887952e-14,1.117806739732908e-14,8.296150110027079e-15,3.623135262436061e-15,9.791473386406057e-16,2.156237433648657e-15,9.05411281861748e-16,8.657258324097557e-16,6.156529440709947e-17,5.122531354112896e-17,8.393900398589521e-17,1.536964510773715e-17,4.318595904163902e-17,5.86663848608656e-18,1.960903919304204e-17,2.464729300599203e-18,5.01399121586355e-18,1.194349213929119e-18,8.451467454558128e-19,4.78102014578257e-19,8.371848132962647e-20,1.468801654660476e-19,5.731569768736786e-20,1.383640299224273e-20,2.38318314289045e-21,4.328252344408406e-22,3.111699330911217e-22,2.293472767077326e-22,3.904161302639318e-23,2.471898141580911e-23,2.956119974305457e-23,1.129224919730815e-25,6.222744015285779e-27,2.012879111022817e-28,1.651488504422824e-28,1.647809244586752e-28,9.174626848658177e-29,1.782940785526552e-29,7.163253172824026e-38,2.526031021950235e-40,4.910589428723304e-45,8.303154074685191e-46,1.578614244084535e-51,9.720990140517975e-57,4.716818303364344e-58,5.687180302417896e-60,1.412707656422797e-59],[0.009992903703431349,0.0098979174041679,0.0098979174041679,0.0098979174041679,0.0098979174041679,0.00974226525389193,0.009652511089697993,0.009418937018934979,0.008945401307833665,0.008818367522964358,0.008797195975569267,0.008609295007906734,0.008477220750237149,0.008453444583638276,0.008453444583638276,0.008400265961419549,0.008400265961419549,0.008400265961419549,0.008400265961419549,0.008400265961419549,0.008400265961419549,0.00836364226288141,0.008309786911094699,0.008105100585404165,0.007939177304551058,0.007846290359097837,0.007825212932675098,0.007803969153229003,0.007779798999387347,0.007629896112745276,0.007621247028565697,0.007538781671211066,0.007538781671211066,0.007521128111592619,0.007490308808552361,0.007470055073849147,0.007470055073849147,0.00740866477883192,0.007391594741307616,0.007063772380381675,0.007001550719561979,0.007001317179577044,0.007001317179577044,0.006946221415532504,0.006772111542258439,0.006772111542258439,0.006761508535315756,0.006675989564564649,0.00650084785412979,0.006372682631665667,0.006188186774479919,0.006174493742326694,0.006174493742326694,0.006174493742326694,0.006165003648356179,0.006117065866794335,0.006113010300658775,0.006095423230685234,0.006095423230685234,0.006095423230685234,0.005862233808920671,0.005833075817663944,0.005828715554631188,0.005683534021744262,0.005613123381921774,0.005613123381921774,0.005443413000386933,0.005443413000386933,0.005349053962021927,0.005333676189359215,0.005251123697708665,0.005224135707904849,0.005182605771371444,0.005182605771371444,0.005161263621092309,0.005161263621092309,0.005161263621092309,0.005161263621092309,0.00499363695680388,0.00499363695680388,0.00499363695680388,0.00499363695680388,0.004990857917053703,0.004800939277834574,0.004696155130807502,0.004673090366604155,0.004537812371395936,0.004416696152219019,0.004416696152219019,0.004377627114471563,0.004362464095265887,0.0043041066004519,0.004189143665234484,0.004124959284209452,0.004072036870708603,0.004061112412160294,0.00405131569863433,0.004021118744871238,0.004017720359545361,0.003944745694768255,0.003900876337496958,0.003895192082350849,0.00386888505679031,0.00386888505679031,0.003861654334242475,0.003687165899324653,0.003636281580658251,0.003627396613768073,0.003548242819576368,0.003530815808149458,0.003530815808149458,0.003530815808149458,0.003496776017006759,0.003496776017006759,0.003467840844003143,0.003411096826994866,0.003411096826994866,0.003335415037424262,0.003335415037424262,0.003264470685175002,0.003256854603676993,0.003256854603676993,0.00317057945048056,0.00306102910774839,0.003049834095159458,0.003034976303400903,0.003023202853675847,0.002993071438793302,0.002991658567343587,0.00278431855425751,0.002671601330945139,0.002636689000194111,0.002606127004239481,0.002420191732460634,0.002384810310950104,0.002349422361741947,0.002176303363561761,0.002176303363561761,0.002154475822293781,0.002136704006194573,0.002065261024450745,0.001851349114306754,0.001798647269357725,0.001733390932695021,0.001716161862256997,0.00170990113425663,0.001692294877297721,0.001574823378260896,0.001574823378260896,0.001571511291983114,0.001567247755433324,0.001536857827991156,0.001490726734527971,0.001490726734527971,0.001490726734527971,0.001482930774017369,0.00140273611955359,0.001399785957090122,0.001357713239825466,0.001357332632884428,0.001351896462775034,0.001351896462775034,0.001345132071890817,0.001318720195335307,0.001300827232437923,0.001282846741442917,0.001252887406149806,0.00124536816994553,0.001214813222933625,0.00115743640791847,0.001157408901464288,0.001133498887714202,0.001089184474968868,0.001034231178939603,0.0009663548395166694,0.0009480501006173379,0.0009480501006173379,0.0009443003653538506,0.0009443003653538506,0.0009443003653538506,0.0009311275903834622,0.0008567447844155356,0.0008542211698825906,0.0008391358550024133,0.0008278488346563617,0.0008244571330236893,0.0008243299915254392,0.0008059198006835688,0.0007577575415696344,0.0007571793414926667,0.0007474047950590365,0.0007370808289424465,0.0007340274963428413,0.0007282232628623384,0.0006337235511817091,0.0006337235511817091,0.0006252591843796144,0.0005985175284907842,0.0005641443913432197,0.0005375937880726099,0.0005303557590690138,0.0005303557590690138,0.0005088798398159567,0.0005088798398159567,0.0005054595235758452,0.0004992590868258321,0.0004432079485055828,0.0004210110516202891,0.0004179078320582991,0.0003492438635910892,0.0003470467077439063,0.0003470467077439063,0.0003273984373670005,0.0003255895773172621,0.0003207949215962216,0.0003195307991659974,0.0003195307991659974,0.0003195307991659974,0.0003021839913437901,0.0002848130639949242,0.0002803086747270251,0.0002563560054869159,0.0002304969761459504,0.0002263077543517922,0.000224888275850171,0.0002028023102197211,0.0002019666053756392,0.000187540826915372,0.0001847059218589939,0.0001830188850907919,0.0001734614288427095,0.0001705509105552765,0.0001556111565990654,0.0001545276853940124,0.0001545276853940124,0.0001530316757566461,0.0001422730898066452,0.0001411932171719218,0.000125334147953419,0.0001251356687474668,0.0001251356687474668,0.0001251356687474668,0.0001223305416176288,0.0001168539535951893,0.0001168539535951893,0.0001160939619517759,0.0001107072895871866,0.0001106466343125266,0.0001028037020423108,0.0001002320466559093,9.554801161417852e-05,9.467700776422291e-05,9.354097752693525e-05,9.205309920983393e-05,9.103728599556584e-05,8.875176396717855e-05,8.780015075022795e-05,8.780015075022795e-05,8.780015075022795e-05,8.780015075022795e-05,8.705896658783465e-05,8.357692498432647e-05,8.357692498432647e-05,8.253789716537691e-05,8.25050416293635e-05,7.893223127551156e-05,7.592427660694536e-05,6.465582166080348e-05,6.172455089117145e-05,6.160319263070911e-05,6.092749275263863e-05,6.015435385239883e-05,5.890875525316083e-05,5.712514123515698e-05,5.601643912425609e-05,5.601643912425609e-05,5.601643912425609e-05,5.498798100837109e-05,5.378727519252058e-05,5.003772846056681e-05,4.958961833797207e-05,4.543273506375252e-05,4.475607065374592e-05,4.362881255787548e-05,4.193211693819794e-05,4.17278425604053e-05,3.994492205237219e-05,3.927407555903091e-05,3.902833647457924e-05,3.902833647457924e-05,3.901367945146518e-05,3.901367945146518e-05,3.827787421921818e-05,3.630994625605334e-05,3.5309708514321e-05,3.479275679576882e-05,3.399191648768457e-05,3.184358037491471e-05,3.08436683085093e-05,2.970888938622672e-05,2.94440295991919e-05,2.937448943122622e-05,2.903926995119827e-05,2.82388997129078e-05,2.797934620428186e-05,2.797934620428186e-05,2.797934620428186e-05,2.797934620428186e-05,2.779693191199981e-05,2.633967677240631e-05,2.517857433536222e-05,2.479323859543284e-05,2.444948014536205e-05,2.443229379368462e-05,2.291849275151369e-05,2.258284295818235e-05,2.234965690726074e-05,2.234965690726074e-05,2.234965690726074e-05,2.130055564237796e-05,2.091859332719855e-05,2.010129432718367e-05,1.861063950231079e-05,1.82333486828112e-05,1.778093291413057e-05,1.730709373544395e-05,1.730709373544395e-05,1.623580727519485e-05,1.533966702352099e-05,1.514904024214619e-05,1.382643354791044e-05,1.382643354791044e-05,1.382643354791044e-05,1.382643354791044e-05,1.382643354791044e-05,1.330770760612694e-05,1.242537537464687e-05,1.092561629741326e-05,1.092561629741326e-05,1.084811047421167e-05,9.556583450871158e-06,9.145180040069839e-06,8.80723198169797e-06,8.371629657190237e-06,8.294042399245665e-06,7.248044389314475e-06,7.065000323492332e-06,6.953708753823951e-06,6.896524964379421e-06,6.782668383701561e-06,6.292793792491352e-06,6.236414826457715e-06,5.258401836590673e-06,5.258401836590673e-06,5.258401836590673e-06,4.476794189771119e-06,4.476794189771119e-06,4.203957355338237e-06,4.185917777212556e-06,3.520068333715508e-06,3.483430614963731e-06,3.458119923486485e-06,3.19272810864216e-06,3.187680815833366e-06,3.187680815833366e-06,3.118047498468263e-06,2.601172067983754e-06,2.582833047501514e-06,2.396958648391463e-06,2.369676663304372e-06,2.106317258534009e-06,1.904135217901796e-06,1.904135217901796e-06,1.894342212303917e-06,1.673549991414454e-06,1.651015088226779e-06,1.607570201379169e-06,1.551799985821569e-06,1.348787705242178e-06,1.348787705242178e-06,1.348787705242178e-06,1.289188605472976e-06,1.022477359009283e-06,1.004217317828783e-06,9.990315289645748e-07,9.761852452116568e-07,9.230949117468581e-07,7.774330809484253e-07,7.715116366612774e-07,7.715116366612774e-07,7.715116366612774e-07,7.715116366612774e-07,7.715116366612774e-07,7.170281277174897e-07,6.944121796299018e-07,6.777026855431053e-07,6.324617710549197e-07,5.55512314294337e-07,5.48701523518766e-07,4.679921336674572e-07,4.462451337734985e-07,4.462451337734985e-07,4.262154564337485e-07,3.400642794139679e-07,3.385570141562183e-07,3.371751969533071e-07,3.104048934844261e-07,2.816327212807458e-07,2.199235234078125e-07,1.997654721239332e-07,1.98289824059522e-07,1.89600159859842e-07,1.785261061964408e-07,1.564243282192756e-07,1.564243282192756e-07,1.441324855769925e-07,1.422799077176626e-07,1.280962140448116e-07,1.275268862004451e-07,1.162209592573127e-07,1.150288871263738e-07,9.601527988314966e-08,9.476069273052894e-08,9.305082836439358e-08,9.305082836439358e-08,9.305082836439358e-08,9.305082836439358e-08,8.971153098149947e-08,6.294550865882485e-08,6.294550865882485e-08,6.124898160785933e-08,6.090611782021929e-08,5.948661932667637e-08,5.948661932667637e-08,5.576351879840334e-08,5.284716237084603e-08,4.805556599797959e-08,4.535247965560806e-08,4.319696749800492e-08,4.14061525824502e-08,3.534161027562628e-08,3.430718632176776e-08,3.340303066154396e-08,2.859240080139352e-08,2.740454481771515e-08,2.614147832947476e-08,2.436942638417944e-08,1.544636339328728e-08,1.322791783470869e-08,1.258570962178152e-08,9.225764978460495e-09,7.39907946038326e-09,7.39907946038326e-09,7.005507578239376e-09,4.986571985905008e-09,3.7542249223138e-09,3.544892446873905e-09,3.544892446873905e-09,2.860907626643267e-09,2.590707698495153e-09,2.314582852030553e-09,1.679519011893817e-09,1.52412029743999e-09,1.437088992650758e-09,1.377837308892577e-09,1.207319041863716e-09,1.012907687713717e-09,3.703673290501445e-10,3.703673290501445e-10,3.703673290501445e-10,3.589819561088546e-10,3.370859370060336e-10,1.94732014125485e-10,1.631847479102185e-10,8.984890427451421e-11,6.058953338210447e-11,6.003712241859771e-11,4.522337026439363e-11,1.731214399997233e-11,1.66113530923576e-11,1.149780608077214e-11,7.451834633470403e-12,5.041573894923263e-12,4.575781974600564e-12,2.51888526160313e-12,1.181193501890843e-12,1.129219388013994e-12,7.095318809115742e-13,3.129858871252142e-13,2.403022790490602e-13,1.086940578730818e-13,8.929823728402325e-14,6.708294238018042e-14,2.925174910630263e-14,2.908838796896779e-14,6.238616499919413e-15,5.839685743688702e-15,2.937865139506332e-15,2.002445191179469e-15,1.577226330216381e-15,8.917290498851572e-16,7.487087691888778e-16,4.495666244292946e-16,2.00559648634542e-16,5.0162666985023e-17,3.736438243067804e-17,2.231142734698532e-17,7.42422617408262e-18,7.257608175969412e-18,3.009074128586812e-18,7.748385675655928e-19,1.42990988573427e-19,2.796716899463893e-20,2.178189531637852e-20,1.751379203949958e-20,3.279495494217027e-21,2.954869009243643e-21,2.75904530935176e-21,1.185686165717356e-23,7.467292818342934e-25,4.589364373132023e-26,2.312083906191954e-26,2.194881913789553e-26,1.541337310574574e-26,3.744175649605759e-27,2.005710888390727e-35,1.060933029219099e-37,1.492819186331885e-42,6.974649422735561e-43,7.198480953025481e-49,2.025854345283946e-53,2.198980693028457e-54,2.370416750047779e-56,1.288389382657591e-56],[8,4,5,7,8,4,15,7,7,26,6,9,16,9,7,15,7,23,12,8,8,4,20,5,8,13,41,20,10,4,16,8,8,18,7,10,7,9,15,10,19,6,6,6,6,6,10,14,9,8,13,6,6,4,7,3,9,11,9,21,5,7,20,11,23,6,10,7,8,6,5,5,6,8,9,6,6,4,11,6,16,14,4,6,7,11,6,5,5,12,4,9,23,10,7,12,22,13,7,12,14,12,4,7,10,12,7,10,19,4,4,8,7,7,12,10,11,9,7,8,4,4,6,11,30,8,5,12,15,5,8,6,21,27,6,40,8,39,12,10,6,8,4,14,8,5,13,29,24,21,35,8,11,5,5,30,7,12,6,29,5,5,18,19,8,17,13,31,12,7,19,23,9,17,22,5,11,6,10,10,14,13,8,8,13,12,22,8,13,13,9,18,15,7,17,17,20,16,8,12,9,9,9,13,9,10,11,11,12,13,10,21,11,9,7,17,36,22,19,20,16,10,13,6,24,9,8,9,10,6,21,22,20,20,20,12,13,26,5,14,7,9,20,24,21,6,14,10,7,6,14,15,19,18,13,10,18,6,6,6,14,6,8,19,7,10,18,10,26,9,10,44,24,7,11,10,10,9,10,18,11,12,10,25,11,9,11,30,11,17,9,10,18,6,9,18,12,15,7,13,13,16,9,20,9,13,17,15,20,19,17,10,15,20,19,14,7,24,9,28,17,8,16,21,6,10,11,10,21,7,9,10,13,11,12,20,12,8,8,20,8,32,15,13,19,18,14,19,8,19,13,15,7,33,6,12,24,15,8,13,7,9,26,17,17,23,17,24,8,15,11,15,20,22,21,18,13,10,18,18,17,19,10,22,24,20,11,9,11,18,8,18,19,12,25,18,14,19,19,18,24,24,12,13,22,18,17,20,16,11,29,24,13,18,22,20,18,19,9,25,23,13,21,24,18,18,39,25,9,9,22,15,18,18,23,8,24,20,19,13,24,22,26,9,26,29,23,23,13,20,30,20,19,22,16,9,40,57,32,14,23,25,17,20,20,23,20,26,26,26,14,13,15,25,25,14,16,28,13,22,15,13,26,25,34,34,40,15,27,27,21,57,56,20,19,47,19,51,33,55,15,20,81,44,27,20,38,59,36,35,39,47,47,41,22,46,44,46,68,44,33,49,40,61,52,66,95,56,86,78,113,70,104,72],[37,31,53,108,140,26,134,106,88,292,76,58,113,140,87,102,28,191,72,36,36,29,156,42,111,105,413,155,56,24,111,35,35,132,101,68,36,43,130,55,208,72,72,34,71,71,67,379,95,34,102,69,69,26,95,11,55,154,105,475,44,93,162,203,184,19,65,34,120,65,38,42,64,118,146,63,63,23,146,39,292,232,10,61,78,184,60,39,39,66,21,126,232,51,74,132,217,97,82,72,272,150,20,81,62,216,24,111,155,19,19,105,24,24,148,61,72,72,42,96,19,19,53,71,316,100,15,82,118,32,39,27,152,218,49,374,29,360,79,44,43,37,15,100,87,28,219,283,212,146,305,36,65,13,13,245,27,58,43,232,28,28,146,180,27,103,68,305,57,61,433,163,33,330,373,24,160,38,129,129,234,203,55,33,208,71,149,77,200,81,41,366,103,32,326,326,126,127,73,166,93,93,92,191,66,114,136,135,55,60,46,168,136,37,43,92,292,140,110,399,82,105,184,31,201,27,42,27,33,14,160,134,486,114,114,144,69,172,10,78,19,33,144,150,121,13,77,44,24,27,76,86,326,320,160,32,210,16,16,16,184,26,60,100,39,90,207,38,164,66,88,411,499,17,107,87,87,60,86,88,105,53,95,151,103,65,102,200,34,78,64,82,86,11,63,110,120,61,12,140,50,88,61,320,60,138,236,185,101,92,232,27,77,100,91,156,27,475,57,173,74,15,85,358,12,72,90,71,136,10,53,69,126,87,106,96,47,19,19,290,27,205,71,118,85,229,61,84,18,108,114,69,15,475,10,89,411,67,17,108,12,22,139,65,65,129,64,411,16,92,69,49,85,413,92,69,100,26,68,68,61,97,25,109,113,81,63,19,40,138,17,142,160,76,119,84,108,92,71,82,160,160,35,41,91,62,55,75,135,27,187,353,39,59,87,72,76,84,16,140,120,89,100,99,57,57,230,138,15,15,82,49,55,55,300,11,292,87,60,71,283,79,358,14,107,130,84,108,32,81,412,60,54,212,48,12,258,481,185,34,217,114,51,72,55,95,146,296,296,300,30,25,34,79,92,27,37,141,21,71,29,20,96,88,160,156,212,25,93,92,51,293,360,43,38,195,29,283,94,320,16,37,475,200,68,33,140,379,121,111,135,191,184,137,32,167,147,166,330,126,51,149,92,236,163,203,412,93,290,166,478,128,371,115],[0.2162162162162162,0.1290322580645161,0.09433962264150944,0.06481481481481481,0.05714285714285714,0.1538461538461539,0.1119402985074627,0.0660377358490566,0.07954545454545454,0.08904109589041095,0.07894736842105263,0.1551724137931035,0.1415929203539823,0.06428571428571428,0.08045977011494253,0.1470588235294118,0.25,0.1204188481675393,0.1666666666666667,0.2222222222222222,0.2222222222222222,0.1379310344827586,0.1282051282051282,0.119047619047619,0.07207207207207207,0.1238095238095238,0.09927360774818401,0.1290322580645161,0.1785714285714286,0.1666666666666667,0.1441441441441441,0.2285714285714286,0.2285714285714286,0.1363636363636364,0.06930693069306931,0.1470588235294118,0.1944444444444444,0.2093023255813954,0.1153846153846154,0.1818181818181818,0.09134615384615384,0.08333333333333333,0.08333333333333333,0.1764705882352941,0.08450704225352113,0.08450704225352113,0.1492537313432836,0.03693931398416887,0.09473684210526316,0.2352941176470588,0.1274509803921569,0.08695652173913043,0.08695652173913043,0.1538461538461539,0.07368421052631578,0.2727272727272727,0.1636363636363636,0.07142857142857142,0.08571428571428572,0.04421052631578947,0.1136363636363636,0.07526881720430108,0.1234567901234568,0.0541871921182266,0.125,0.3157894736842105,0.1538461538461539,0.2058823529411765,0.06666666666666667,0.09230769230769231,0.131578947368421,0.119047619047619,0.09375,0.06779661016949153,0.06164383561643835,0.09523809523809523,0.09523809523809523,0.1739130434782609,0.07534246575342465,0.1538461538461539,0.0547945205479452,0.0603448275862069,0.4,0.09836065573770492,0.08974358974358974,0.05978260869565218,0.1,0.1282051282051282,0.1282051282051282,0.1818181818181818,0.1904761904761905,0.07142857142857142,0.09913793103448276,0.196078431372549,0.0945945945945946,0.09090909090909091,0.1013824884792627,0.134020618556701,0.08536585365853659,0.1666666666666667,0.05147058823529412,0.08,0.2,0.08641975308641975,0.1612903225806452,0.05555555555555555,0.2916666666666667,0.09009009009009009,0.1225806451612903,0.2105263157894737,0.2105263157894737,0.0761904761904762,0.2916666666666667,0.2916666666666667,0.08108108108108109,0.1639344262295082,0.1527777777777778,0.125,0.1666666666666667,0.08333333333333333,0.2105263157894737,0.2105263157894737,0.1132075471698113,0.1549295774647887,0.0949367088607595,0.08,0.3333333333333333,0.1463414634146341,0.1271186440677966,0.15625,0.2051282051282051,0.2222222222222222,0.1381578947368421,0.1238532110091743,0.1224489795918367,0.106951871657754,0.2758620689655172,0.1083333333333333,0.1518987341772152,0.2272727272727273,0.1395348837209302,0.2162162162162162,0.2666666666666667,0.14,0.09195402298850575,0.1785714285714286,0.0593607305936073,0.1024734982332156,0.1132075471698113,0.1438356164383562,0.1147540983606557,0.2222222222222222,0.1692307692307692,0.3846153846153846,0.3846153846153846,0.1224489795918367,0.2592592592592592,0.2068965517241379,0.1395348837209302,0.125,0.1785714285714286,0.1785714285714286,0.1232876712328767,0.1055555555555556,0.2962962962962963,0.1650485436893204,0.1911764705882353,0.1016393442622951,0.2105263157894737,0.1147540983606557,0.04387990762124711,0.1411042944785276,0.2727272727272727,0.05151515151515151,0.05898123324396783,0.2083333333333333,0.06875000000000001,0.1578947368421053,0.07751937984496124,0.07751937984496124,0.05982905982905983,0.06403940886699508,0.1454545454545454,0.2424242424242424,0.0625,0.1690140845070423,0.1476510067114094,0.1038961038961039,0.065,0.1604938271604938,0.2195121951219512,0.04918032786885246,0.145631067961165,0.21875,0.05214723926380368,0.05214723926380368,0.1587301587301587,0.1259842519685039,0.1095890410958904,0.07228915662650602,0.09677419354838709,0.09677419354838709,0.09782608695652174,0.06806282722513089,0.1363636363636364,0.08771929824561403,0.08088235294117647,0.08148148148148149,0.2181818181818182,0.2166666666666667,0.2173913043478261,0.125,0.08088235294117647,0.2432432432432433,0.1627906976744186,0.1847826086956522,0.1232876712328767,0.1571428571428571,0.1727272727272727,0.05012531328320802,0.1951219512195122,0.09523809523809523,0.07065217391304347,0.1935483870967742,0.1194029850746269,0.3333333333333333,0.1904761904761905,0.3333333333333333,0.303030303030303,0.4285714285714285,0.13125,0.1641791044776119,0.0411522633744856,0.1754385964912281,0.1754385964912281,0.08333333333333333,0.1884057971014493,0.1511627906976744,0.5,0.1794871794871795,0.3684210526315789,0.2727272727272727,0.1388888888888889,0.16,0.1735537190082645,0.4615384615384616,0.1818181818181818,0.2272727272727273,0.2916666666666667,0.2222222222222222,0.1842105263157895,0.1744186046511628,0.05828220858895705,0.05625,0.08125,0.3125,0.08571428571428572,0.375,0.375,0.375,0.07608695652173914,0.2307692307692308,0.1333333333333333,0.19,0.1794871794871795,0.1111111111111111,0.08695652173913043,0.2631578947368421,0.1585365853658537,0.1363636363636364,0.1136363636363636,0.1070559610705596,0.04809619238476954,0.4117647058823529,0.102803738317757,0.1149425287356322,0.1149425287356322,0.15,0.1162790697674419,0.2045454545454546,0.1047619047619048,0.2264150943396226,0.1052631578947368,0.1655629139072848,0.1067961165048544,0.1384615384615385,0.107843137254902,0.15,0.3235294117647059,0.217948717948718,0.140625,0.1219512195121951,0.2093023255813954,0.5454545454545454,0.1428571428571428,0.1636363636363636,0.1,0.2459016393442623,0.5833333333333334,0.09285714285714286,0.26,0.1818181818181818,0.1475409836065574,0.0625,0.15,0.09420289855072464,0.07203389830508475,0.08108108108108109,0.198019801980198,0.2065217391304348,0.07327586206896551,0.3703703703703703,0.1948051948051948,0.2,0.2087912087912088,0.08974358974358974,0.2592592592592592,0.05052631578947368,0.1578947368421053,0.161849710982659,0.2297297297297297,0.5333333333333333,0.1882352941176471,0.05865921787709497,0.5,0.1388888888888889,0.1222222222222222,0.1408450704225352,0.1544117647058824,0.7,0.169811320754717,0.1449275362318841,0.1031746031746032,0.1264367816091954,0.1132075471698113,0.2083333333333333,0.2553191489361702,0.4210526315789473,0.4210526315789473,0.06896551724137931,0.2962962962962963,0.1560975609756098,0.2112676056338028,0.1101694915254237,0.2235294117647059,0.07860262008733625,0.2295081967213115,0.2261904761904762,0.4444444444444444,0.1759259259259259,0.1140350877192982,0.2173913043478261,0.4666666666666667,0.06947368421052631,0.6,0.1348314606741573,0.0583941605839416,0.2238805970149254,0.4705882352941176,0.1203703703703704,0.5833333333333334,0.4090909090909091,0.1870503597122302,0.2615384615384616,0.2615384615384616,0.1782945736434109,0.265625,0.0583941605839416,0.5,0.1630434782608696,0.1594202898550725,0.3061224489795918,0.2352941176470588,0.05326876513317191,0.2282608695652174,0.2608695652173913,0.13,0.3846153846153846,0.2647058823529412,0.2647058823529412,0.2786885245901639,0.1958762886597938,0.4,0.2018348623853211,0.2123893805309734,0.2469135802469136,0.1746031746031746,0.4736842105263158,0.275,0.1304347826086956,0.4705882352941176,0.1267605633802817,0.11875,0.1578947368421053,0.2100840336134454,0.2142857142857143,0.1296296296296296,0.2065217391304348,0.2676056338028169,0.2195121951219512,0.15,0.15,0.3428571428571429,0.3170731707317073,0.2417582417582418,0.2903225806451613,0.3090909090909091,0.2666666666666667,0.1185185185185185,0.4074074074074074,0.1550802139037433,0.0679886685552408,0.3333333333333333,0.3050847457627119,0.2528735632183908,0.2777777777777778,0.2368421052631579,0.2261904761904762,0.5625,0.1785714285714286,0.1916666666666667,0.1460674157303371,0.21,0.2424242424242424,0.3157894736842105,0.3157894736842105,0.1695652173913043,0.1811594202898551,0.6,0.6,0.2682926829268293,0.3061224489795918,0.3272727272727273,0.3272727272727273,0.07666666666666666,0.7272727272727273,0.0821917808219178,0.2298850574712644,0.3166666666666667,0.1830985915492958,0.08480565371024736,0.2784810126582278,0.07262569832402235,0.6428571428571429,0.2429906542056075,0.2230769230769231,0.2738095238095238,0.212962962962963,0.40625,0.2469135802469136,0.07281553398058252,0.3333333333333333,0.3518518518518519,0.1037735849056604,0.3333333333333333,0.75,0.1550387596899225,0.1185031185031185,0.172972972972973,0.4117647058823529,0.1059907834101382,0.2192982456140351,0.3333333333333333,0.2777777777777778,0.3636363636363636,0.2421052631578947,0.136986301369863,0.08783783783783784,0.08783783783783784,0.08666666666666667,0.4666666666666667,0.52,0.4411764705882353,0.3164556962025317,0.2717391304347826,0.5185185185185185,0.4324324324324325,0.198581560283688,0.6190476190476191,0.3098591549295774,0.5172413793103449,0.65,0.2708333333333333,0.2840909090909091,0.2125,0.217948717948718,0.1886792452830189,0.6,0.2903225806451613,0.2934782608695652,0.4117647058823529,0.1945392491467577,0.1555555555555556,0.4651162790697674,0.5,0.241025641025641,0.6551724137931034,0.1802120141342756,0.351063829787234,0.171875,0.9375,0.5405405405405406,0.1705263157894737,0.22,0.3970588235294117,0.6060606060606061,0.2714285714285714,0.1556728232189974,0.2975206611570248,0.3153153153153153,0.2888888888888889,0.2460732984293194,0.2554347826086957,0.2992700729927008,0.6875,0.2754491017964072,0.2993197278911565,0.2771084337349398,0.2060606060606061,0.3492063492063492,0.6470588235294118,0.3288590604026846,0.4347826086956522,0.2584745762711864,0.3190184049079755,0.3251231527093596,0.2305825242718447,0.6021505376344086,0.296551724137931,0.4698795180722892,0.2364016736401674,0.546875,0.2803234501347709,0.6260869565217392],["HADHB/DBI/ACOT13/DECR1/ECHS1/ECI1/NDUFAB1/ACADVL","HSPA1A/HSPA1B/HSPB1/HSPH1","TUBA4A/TUBB4B/TUBA1A/TUBA1C/DYNC1H1","JUN/H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","RHOC/RHOB/TUBA4A/TUBB4B/TUBA1A/TUBA1C/CLIP1/DYNC1H1","XPO1/PPP1CC/UBB/PPP1R15A","KIF15/CENPE/KIF2A/KIFC1/TUBB4B/KIF20B/KIF11/KIF21A/TUBA1B/TUBA1C/RACGAP1/KIF23/KIF22/PAFAH1B3/KIF4A","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/GSN","CCNB1/PTTG1/PSME2/PSMA4/UBB/UBE2S/AURKA","NEK2/IMMT/CEP97/DYNLT2B/TUBB/PCM1/ASAP1/CEP78/CDK5RAP2/DYNC2I2/TUBB4B/CDK1/CKAP5/CEP57/TUBA1B/TUBA1C/RAB3IP/CENPJ/HAUS2/CEP152/IDH2/CCP110/TUBG1/HAUS1/HAUS5/MAPRE1","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","TPR/NUP155/POP7/NUP107/RAN/NUP58/RPP25/NUP62/TSEN34","DDOST/SSR2/RPSA/RPL29/SPCS1/RPS12/SEC61G/SEC61B/RPS24/RPS13/RPS26/SRP14/RPS27L/SEC11C/RPN2/SSR4","H2BC21/H2AC25/RUVBL1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","XPO1/HNRNPD/PSME2/ZFP36L1/PSMA4/UBB/ZFP36","CAPZB/TUBA4A/ARF4/ARF5/DCTN3/TUBB4B/CD59/TUBA1B/TUBA1A/TUBA1C/COPZ1/DYNLL1/TMED10/TMED3/COPE","HSPA1A/HSPA1B/HSPB1/HSPA8/FKBP4/HSBP1/UBB","PSMB2/TUBA4A/PSMB8/PSMB1/SEM1/PSMB7/TUBB4B/PSMA1/PSMC3/TUBA1B/TUBA1A/TUBA1C/DYNLL1/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","SNAPIN/VAMP8/CLTB/AP1S1/FTH1/HSPA8/BLOC1S1/RAB5C/AP1M2/FTL/CHMP2A/CTSZ","PRDX1/PRDX6/ATOX1/CYCS/TXN/PRDX5/CYBA/PRDX2","ARPC4/BRK1/ACTB/ARPC1A/ARPC1B/ARPC3/MAPK3/ACTG1","TUBA4A/TUBB4B/TUBA1A/TUBA1C","PSMB2/POLE4/H2BC4/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/H2AJ/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","XPO1/SOX4/TLE1/UBB/SOX9","CDKN1A/VEGFA/CEBPD/GATA3/HSP90B1/FOS/BIRC5/JUNB","JUN/H2AC20/TFDP2/H2AZ1/H3C2/H4C3/H3C7/H2BC9/H2AZ2/AGO2/H2AX/TFDP1/E2F1","ISG15/DDOST/SFN/ATP1B1/SDC1/STAT1/RPSA/PARP9/ATP1B3/AP2M1/TUBB/RPS12/PPIA/MAN1B1/RPS24/RPS13/FKBP4/RPS26/UBE2N/UBC/DAD1/CHMP4A/MTA1/B2M/RPS27L/TUFM/PYCARD/UBB/HMG20B/CRB3/BST2/FXYD3/AP2S1/PRMT1/CHMP2A/FKBP1A/RPN2/IFNGR2/COMT/SNRPD3/RBX1","CAPZB/TUBA4A/ARF4/LMAN2/ARF5/DCTN3/TUBB4B/CD59/TUBA1B/TUBA1A/TUBA1C/COPZ1/DYNLL1/TMED10/TMED3/TRAPPC2L/TRAPPC1/COPE/TRAPPC6A/CTSZ","ACBD3/GOLGB1/TFRC/CLINT1/AP1S1/FTH1/BLOC1S6/VAMP2/FTL/YIPF6","JUN/LMNA/PRDX2/CDC25B","CAPZB/TUBA4A/AP2M1/AP1S1/DCTN3/TUBB4B/CTSD/CTSF/TUBA1B/TUBA1A/TUBA1C/DYNLL1/CTSH/AP1M2/AP2S1/CTSA","AKR1A1/MGST3/MGST2/GGCT/GSTK1/GSTO1/MGST1/ESD","VAMP8/CLTB/AP1S1/HSPA8/BLOC1S1/AP1M2/CHMP2A/CTSZ","TAGLN2/QSOX1/CYB5R1/SCCPDH/TUBA4A/PPIA/CLU/CFL1/CD9/CD63/VTI1B/ALDOA/PFN1/LGALS3BP/CFD/A1BG/TMSB4X/MAGED2","HSPA1A/HSPA1B/HSPB1/BAG1/CRYAB/HSPA8/HSPH1","PSMD1/CCNA2/PSMB9/CDKN1A/CCNE2/CDKN2A/PSMC3/CHEK1/CDK2/MDM2","TPR/SLBP/NUP155/NUP107/NUP58/ALYREF/NUP62","ANXA6/ANXA1/MYL6B/MYL6/ANXA2/TPM1/MYL12A/MYL12B/MYL9","TPR/XPO1/PSMD1/NUP155/PSMB9/HMGA1/PSIP1/PSMC3/NUP107/RAN/NUP58/KPNB1/NUP62/RANBP1/SLC25A5","FDPS/HMGCS1/HMGCR/SAR1B/INSIG1/FDFT1/SQLE/IDI1/SCD/DHCR7","ID3/SFPQ/PARP1/HNRNPU/ID2/ROCK2/HNRNPD/EZH2/PRKDC/KLF10/CDK1/PPP1CA/TIMELESS/TOP2A/TYMS/ID1/AHCY/DYRK1A/NONO","TUBA4A/TUBB/TUBB4B/TUBA1A/DYNC1H1/OFD1","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","PGAM1/GAPDH/TPI1/ENO2/GPI/PGK1","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","RIF1/BARD1/RNF168/NSD2/H4C3/H2BC9/PRKDC/DCLRE1C/H2AX/BRCA1","CSRP1/TUBB/HSPA1A/HSPA1B/SQLE/ASAP1/GSN/DYNC1H1/TPM1/KRT19/RNF213/CIRBP/MYL9/FLNA","CSRP1/DSP/HSPB1/MYL12A/DSC2/DSG2/MYL9/CXADR/FLNA","PARK7/VPS28/HSPA8/UBC/CHMP4A/UBB/PLIN3/CHMP2A","RBBP4/H2AC20/H2AZ1/DEK/H3C2/H4C3/H3C7/H2BC9/CBX3/H2AZ2/BAZ1B/H2AX/RBBP7","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","TUBA4A/TUBB4B/TUBA1A/TUBA1C","TUBA4A/TUBB/TUBB4B/TUBA1A/TUBA1C/DYNC1H1/OFD1","PKP1/DSP/DSG2","SFN/TFDP2/GSDMD/CDKN2A/TFDP1/UACA/BAX/E2F1/YWHAH","DHCR24/FDPS/CYP27A1/MSMO1/HMGCS1/FDFT1/SQLE/IDI1/SCD/TM7SF2/FASN","JUN/ELF3/PRKCI/H2BC4/H2AC6/H2BC5/B4GALT1/CCND1/ATP2A2","S100A9/PKP1/TMEM63A/IDH1/S100P/DSP/HSPA1A/HSPA1B/B4GALT1/HSPA8/CTSH/PYCARD/CYBA/ACLY/CFD/BST2/GPI/CST3/SLPI/DDX3X/HUWE1","TUBA4A/TUBB4B/TUBA1A/TUBA1C/DYNC1H1","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/MYL9","PARK7/ENO1/S100A9/MUC1/HSPA1A/PPIA/HSPB1/FIS1/CLU/GRINA/CREB3/CD44/NDUFS3/BAD/PYCARD/UBB/NDUFA13/MIF/XBP1/BCAP31","ATP1B1/ATP5MC3/NDUFB3/ATP6V1A/HSPA1A/HSPA1B/FBP1/HSPA8/STAT3/ATP5MC1/GPI","PSMB2/POLE4/SKP1/H2BC4/PSMB8/PSMB1/POLD2/SEM1/PSMB7/PSMA1/PSMC3/H2AJ/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","RHOC/RHOB/CFL1/MYL6/MYL12B/MYL9","H2AC20/H2AZ1/H3C2/H4C3/H3C7/H2BC9/H2AZ2/H2AX/ORC6/KPNB1","TPR/NUP155/NUP107/RAN/NUP58/KPNB1/NUP62","JUN/H2BC21/MDM4/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","HSPA1B/SERPINH1/HSPH1/UBB/DNAJB1","FDPS/HMGCS1/SQLE/IDI1/DHCR7","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","JUN/H2BC21/PBX1/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","JUN/H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/CTSD/CCND1","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","TUBA4A/TUBB4B/TUBA1A/TUBA1C","JUN/H2BC4/H2AC6/H2BC5/GATA3/POLR2L/CCND1/FOXA1/FOS/FOSB/ZNF217","ID3/ID2/FOS/JUNB/FOSB/ID1","JUN/CRABP2/APOD/H2BC4/H2AC6/H2BC5/GATA3/SCD/POLR2L/CPT1A/CCND1/FOXA1/FOS/FASN/FOSB/ZNF217","JUN/PBX1/ELF3/STAT1/PRKCI/HES1/H2BC4/H2AC6/H2BC5/B4GALT1/MDK/CCND1/ATP2A2/PSME2","RAD21/SMC3/PDS5B/SMC1A","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","H2BC4/H2BC5/H2BC7/USP7/UBB/UBE2S/USP9X","ISG15/IFI6/IFITM3/EIF4G2/IFI27/B2M/MT2A/UBB/KPNA2/EIF4A3/MX1","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","ISG15/IFIH1/FLOT1/TRIM56/OAS1","JUN/ATF3/EPAS1/DDIT3/ATF4","ISG15/IFI6/STAT1/PSMB8/IFITM2/IFITM3/OAS1/IFI27/ISG20/IFI35/BST2/MX1","TUBA4A/TUBB4B/TUBA1A/TUBA1C","CENPF/XPO1/CCNB1/RAD21/TUBA1B/PPP1CC/TAOK1/SKA2/BIRC5","JUN/H2AC20/STAT1/PSMD1/TFDP2/HES1/TACC3/H2AZ1/NEURL1B/H3C2/H4C3/H3C7/H2BC9/PSMB9/H2AZ2/AGO2/TLE1/MDK/PSMC3/H2AX/TFDP1/DLGAP5/E2F1","JUN/TRIM11/CCNT2/REST/CHD1/TRIM27/CCNT1/TRIM13/HEXIM1/ZFP36","NUCKS1/RHOB/ECT2/CDKN1A/NET1/GATA3/HRAS","QSOX1/TUBA4A/VEGFA/CLU/HSPA5/PSAP/ACTN1/CFD/A1BG/TMSB4X/LAMP2/FLNA","JUN/H2AC20/XPO1/GNB4/H2AZ1/H3C2/H4C3/H3C7/H2BC9/H2AZ2/RAD21/AGO2/GATA3/SMC3/HRAS/H2AX/FOS/KPNA2/SPHK1/FOSB/ZNF217/SMC1A","RRM2/DTYMK/UMPS/GMPS/PAICS/DCK/NUDT1/RRM1/PNP/DUT/TK1/TYMS/HPRT1","EPAS1/TUBA4A/TUBB4B/HSPA8/TUBA1A/TUBA1C/DYNC1H1","ACBD3/GOLGB1/TFRC/CLINT1/AP1S1/FTH1/BLOC1S6/VAMP2/HGS/FTL/APP/YIPF6","H2BC21/KDM5B/H2AC25/RUVBL1/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/KDM5A/CHD4","NECTIN4/CD55/HSPA1A/HSPA1B/SNX3/CTSB/SLC20A2/CD81/SLC3A2/BSG/CXADR/LGALS1","RHOC/RHOB/MYL9/MYH9","ISG15/EIF2AK2/STAT1/FLNB/OAS1/MX1/FLNA","JUN/MSH2/FANCD2/NELFE/SSRP1/CHEK1/SUPT16H/FOS/FANCI/BRCA1","RAB13/GLUL/ROCK2/RHOB/BMPR2/S100P/VEGFA/GATA3/NR4A1/HMGB1/NR2F2/ID1","SARS1/EPRS1/TARS1/GARS1/IARS1/MARS1/AARS1","MUC1/STAT1/CDKN1A/VEGFA/GATA3/CCND1/HSPA8/FOS/STAT3/JUNB","CD55/MIA3/MCFD2/SEC13/ARF4/GOLGB1/COPG1/SEC31A/SAR1B/GOLGA2/COG3/MIA2/TMED3/DYNC1LI2/GOSR1/GOSR2/TRAPPC6A/TBC1D20/ARFGAP3","RHOC/RHOB/MYL9/MYH9","RHOC/RHOB/MYL9/MYH9","JUN/H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/CCND1","PRDX1/JUN/CDKN2A/CAPN1/C1QBP/PRDX2/CAPNS1","LAMTOR5/LAMTOR2/LAMTOR4/EIF4EBP1/RRAGA/LAMTOR1/FKBP1A","JUN/RHOB/VEGFA/HSPB1/ADAM9/ANXA1/PLCG2/SOX9/CALR/GPI/RRAS/TMSB4X","H2AC20/H2AZ1/H3C2/H4C3/H3C7/H2BC9/H2AZ2/H2AX/UHRF1/DNMT1","RBBP4/H2AC20/H2AZ1/H3C2/H4C3/H3C7/H2BC9/CBX3/H2AZ2/H2AX/RBBP7","HK2/PFKP/PGAM1/GAPDH/TPI1/ENO2/PKM/GPI/PGK1","SARS1/EPRS1/DARS1/TARS1/LARS1/GARS1/IARS1","H2BC21/H4C3/H2BC4/H2BC5/H2BC7/H4C8/RPA3/UBB","LMNA/LBR/CCNB1/TMPO","SPTBN1/CDKN1A/HRAS/ZMYM2","H2BC21/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","RBBP4/SMYD2/NSD2/H3C2/H4C3/H3C7/EZH2/NSD3/EED/SUZ12/RBBP7","EFNA1/DPM3/TMEM183A/MDM4/AAK1/MARCHF7/VHL/SENP2/UBE2B/CAMLG/FBXW11/MYLIP/SOX4/HSPA1B/CLU/DERL1/RNF139/RAB21/COG3/SAV1/SEL1L/BAG5/CHP1/ZNF207/NR1D1/USP36/CALR/NOP53/STK4/MORC3","JUN/H2BC21/CAPN2/H2BC4/H2AC6/H2BC5/H2BC7/H4C8","CCNB1/LMNB1/CDK1/TMPO/EMD","RPA2/TPR/NUP155/HSPA1A/HSPA1B/RPA3/HSPA14/HSPA8/NUP107/NUP58/DNAJB1/NUP62","RBBP4/JUN/H2AC20/H2AZ1/H3C2/H4C3/H3C7/H2BC9/H2AZ2/EZH2/EED/H2AX/CTCF/SUZ12/RBBP7","TUBA4A/TUBB4B/TUBA1A/TUBA1C/CLIP1","RPA2/USP1/UBE2T/FANCL/FANCD2/RPA3/FANCI/CENPX","ATF3/DDIT3/HERPUD1/CEBPG/CEBPB/ATF4","PSMB2/SKP1/PSMB8/PSMB1/SEM1/GSDMD/PSMB7/PSMA1/PSMC3/UBE2N/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","PSMB2/RPSA/RPL29/PSMB8/RPS12/PSMB1/SEM1/PSMB7/RPS24/PSMA1/RPS13/PSMC3/RPS26/UBC/PSMB5/PSME1/PSME2/PSMC1/RPS27L/ELOB/PSMB6/PFN1/UBB/PSMB3/PSMC5/PSMC4/RBX1","TUBA4A/TUBB4B/SHTN1/TUBA1A/TUBA1C/AP2B1","SERINC2/PSMB2/TSTD1/RPSA/RPL29/PSMB8/RPS12/PSMB1/SEM1/RIDA/GRHPR/PSMB7/ASS1/PCBD1/RPS24/OAT/ECHS1/PSMA1/RPS13/PSMC3/RPS26/PSMB5/PSME1/PSME2/PSMC1/CKB/SQOR/RPS27L/NMRAL1/NDUFAB1/NQO1/PSMB6/PSMB3/PSMC5/GAMT/PSMC4/ETHE1/SMS/SAT1/HSD17B10","AK2/GUK1/TXN/DUT/NME4/NME3/DCTPP1/NME1","SFN/PRDX1/JUN/LAMTOR5/LAMTOR2/COX20/COX5B/COX7C/PRELID1/CSNK2B/GTF2H5/NDUFA4/CYCS/LAMTOR4/RRAGA/CDKN2A/TXN/SURF1/POLR2L/TAF10/POLR2G/COX8A/PRDX5/LAMTOR1/COX14/COX6A1/UBC/FOS/COX5A/ELOB/TMEM219/UBB/POLR2E/PRDX2/COX6B1/POLR2I/PRMT1/COX7B/G6PD","PSMD1/CCNB1/PSMB9/CDKN1A/TUBB4B/CDK1/PSMC3/TUBA1B/TUBA1C/CCNB2/MAPRE1/GTSE1","PARK7/TUBA4A/TUBB4B/TUBA1B/TUBA1A/TUBA1C/UBE2N/DYNLL1/UBC/UBB","NCL/ZFP36L1/FOS/SOX9/ZFP36/ID1","TPR/NUP155/HMGA1/PSIP1/NUP107/NUP58/NUP62/SLC25A5","ATF3/DDIT3/TRIB3/ATF4","KIF15/CENPE/KIF2A/KIFC1/TUBB4B/KIF20B/KIF11/KIF21A/TUBA1B/TUBA1C/RACGAP1/KIF23/KIF22/KIF4A","HNRNPH1/HRAS/HNRNPA1/POLR2A/UBB/PTBP1/HNRNPM/POLR2F","HSPA1A/HSPA1B/HSPB1/HSPA8/HSPH1","H2BC21/H2AC25/RUVBL1/H2BC4/H2AC6/H2BC5/H2BC7/GATA3/PSME2/PSMA4/USP7/UBB/USP9X","SYF2/SRSF4/SRSF11/PRPF3/CHTOP/SDE2/SRSF7/SEC13/RBM5/TRA2B/IK/RBM22/SLU7/PRPF4B/C9orf78/PCF11/NUP58/HNRNPC/PDCD7/SNRPA1/POLR2A/CWC25/CASC3/LUC7L3/HNRNPL/NXT1/RBM39/SRSF6/NKAP","SYF2/SRSF4/SRSF11/PRPF3/SDE2/SRSF7/RBM5/TRA2B/IK/RBM22/SLU7/PRPF4B/C9orf78/HNRNPC/PDCD7/SNRPA1/POLR2A/CWC25/CASC3/LUC7L3/HNRNPL/RBM39/SRSF6/NKAP","PSMB2/AP2M1/CLTB/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/PFN1/UBB/PSMB3/PSMC5/PSMC4/AP2S1","CAPZB/DDOST/DPM3/NAGK/TUBA4A/ARF4/ALG3/LMAN2/NEU1/ARF5/GFUS/DCTN3/NANS/MAN1B1/TUBB4B/CD59/TUBA1B/TUBA1A/TUBA1C/COPZ1/DYNLL1/UBC/DAD1/TMED10/PSMC1/TMED3/TRAPPC2L/TRAPPC1/UBB/CALR/COPE/TRAPPC6A/RPN2/CTSA/CTSZ","TPR/NUP155/CCNB1/CDK1/NUP107/NUP58/CCNB2/NUP62","RPA2/POLE4/RFC4/RFC1/RPA3/POLD2/RFC2/POLE3/POLD3/RFC3/PCNA","TFDP2/PRIM1/TFDP1/E2F1/POLA1","RAD21/SMC3/PDS5B/KIF23/SMC1A","PSMB2/COMMD1/RNF7/SKP1/PSMB8/PSMB1/SEM1/COPS6/COMMD5/PSMB7/FBXW5/COMMD3/PSMA1/PSMC3/UBC/COMMD6/PSMB5/PSME1/PSME2/NEDD8/PSMC1/COMMD4/ELOB/PSMB6/UBB/PSMB3/PSMC5/PSMC4/COMMD7/RBX1","AHCTF1/NUP155/NUP107/RAN/NUP58/KPNB1/NUP62","ATP6V0B/SKP1/ATP6V0E1/ATP6V1F/ATP6V0E2/FTH1/UBC/NEDD8/ATP6V0D1/UBB/FTL/ATP6AP1","ANXA6/ANXA1/TPM1/MYL12A/TPM4/MYL9","PSMB2/JUN/NCSTN/ELF3/STAT1/HES1/SKP1/H2BC4/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/MDK/PSMC3/CCND1/H2AJ/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSENEN/PSMC4/RBX1","HSPA1B/SERPINH1/HSPH1/UBB/DNAJB1","SPTBN1/CDKN1A/HRAS/ZMYM2/UBB","JUN/H2AC20/H2AZ1/H3C2/H4C3/H3C7/H2BC9/H2AZ2/RAD21/AGO2/GATA3/SMC3/H2AX/FOS/KPNA2/FOSB/ZNF217/SMC1A","ARID1A/RBBP4/ASPM/RIF1/FANCD2/ACTL6A/HES1/ING2/SOX4/SRRT/EZH2/PCM1/KLF10/PHF19/SMC3/BCL7C/BMP7/RBBP7/SMC1A","ATF6/ATF3/DDIT3/DIS3/HERPUD1/CEBPG/CEBPB/ATF4","PSMB2/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","PARK7/IFI6/ATP5IF1/PPA2/PRELID1/CLIC1/HEBP2/STOML2/BAD/UCP2/NDUFC2/UBB/TSPO","DPM3/CD55/MIA3/MCFD2/GFPT1/SEC13/ARF4/GOLGB1/COPG1/SEC31A/MARCHF6/SAR1B/MGAT1/NEU1/DERL1/RNF139/GOLGA2/MLEC/COG3/TRIM13/MIA2/SEL1L/TMED3/PMM2/DYNC1LI2/GOSR1/GOSR2/CALR/TRAPPC6A/TBC1D20/ARFGAP3","CAPZB/TUBA4A/HSPA1A/HSPA1B/DCTN3/TUBB4B/HSPA8/FKBP4/TUBA1B/TUBA1A/TUBA1C/DYNLL1","ID3/ID2/CHD4/FOS/JUNB/FOSB/ID1","SFN/IFI6/DHCR24/S100A9/BIRC6/VEGFA/CLDN3/CLDN4/NR4A1/PSME2/CTSH/PYCARD/STAT3/BST2/GPI/CST3/SLPI/DDX3X/BEX3","PSMB2/POLE4/SKP1/PSMB8/PSMB1/POLD2/SEM1/PSMB7/PSMA1/PSMC3/CCND1/CDK4/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","CCT7/TUBA4A/ACTB/TUBB4B/TUBA1B/TUBA1A/TUBA1C/PFDN5/PFDN4","ISG15/H2BC21/RUVBL1/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/RPA3/H2AZ2/USP7/POLR2A/UBB/KPNA2/POLR2F","CSRP1/ID2/HES1/PDLIM5/VEGFA/IFRD1/NEBL/CD81/ADM/KLF5/ACTN1/RORA/TPM1/MEF2A/KRT19/SOX9/CALR/GDF15/TNNT1/MYL9/CXADR/XBP1","RHOC/RHOB/RRAS/MYL9/MYH9","H2BC21/TUBA4A/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/TUBB4B/TUBA1A/TUBA1C/DYNC1H1","HSPA1A/HSPA1B/HSPB1/CRYAB/HSPA8/HSPH1","PKP1/DSP/KRT80/KRT7/KRT8/KRT18/KRT19/JUP/DSC2/DSG2","PKP1/DSP/KRT80/KRT7/KRT8/KRT18/KRT19/JUP/DSC2/DSG2","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/CTSV/GATA3/PSME2/PSMA4/UBB","CENPF/XPO1/MZT2B/MZT2A/SMC4/CCNB1/RAD21/DCTN3/TUBA1B/PPP1CC/TAOK1/SKA2/BIRC5","FDPS/HMGCS1/FDFT1/SQLE/IDI1/SCD/TM7SF2/FASN","TPR/XPO1/NUP155/HSPA1A/NUP107/RAN/NUP58/NUP62","ID3/SFPQ/ID2/BHLHE40/HNRNPD/CLDN4/NAMPT/PRKDC/ASS1/MYCBP2/ID1/USP9X/HUWE1","H2AC20/H2AZ1/CCNA2/LMNB1/H4C3/H2BC9/HMGA1/CDKN1A/H2AZ2/CCNE2/H2AX/CDK2","PSMB2/POLE4/SKP1/PSMB8/PSMB1/SEM1/CDKN2A/PSMB7/PSMA1/PSMC3/CCND1/CDK4/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","H2BC21/H4C3/H2BC4/H2BC5/H2BC7/H4C8/UBB/KPNA2","CENPF/XPO1/MZT2B/MZT2A/CCNB1/CDKN1A/DCTN3/TUBA1B/PSME2/PSMA4/UBB/CDC25B/AURKA","ISG15/TPR/EIF2AK2/STAT1/NUP155/NUP107/OAS3/NUP58/KPNB1/KPNA2/PIN1/NUP62/MX1","ISG15/TPR/EIF2AK2/NUP155/NUP107/NUP58/KPNB1/KPNA2/NUP62","STMN1/JUN/DHX9/NUCKS1/HES1/CDKN1A/GATA3/RSF1/HMGB1/B2M/EIF4A3/CALM3/TRIM28/ID1/SLPI/ADNP/AURKA/USP9X","H2AC20/TPR/H2AZ1/NUP155/H3C2/H4C3/H3C7/H2BC9/H2AZ2/AGO2/H2AX/NUP107/RAN/NUP58/NUP62","ATF3/HSPA5/DDIT3/HERPUD1/CEBPG/CEBPB/ATF4","SFN/CHTOP/LMNA/DHX9/XPO1/ECT2/APOD/SRSF3/CDKN1A/HNRNPA2B1/HNRNPA1/PPP1CC/IFI27/KPNA2/EIF4A3/DDX39A/TRIM28","SFN/CHTOP/LMNA/DHX9/XPO1/ECT2/APOD/SRSF3/CDKN1A/HNRNPA2B1/HNRNPA1/PPP1CC/IFI27/KPNA2/EIF4A3/DDX39A/TRIM28","PSMB2/H2BC4/PSMB8/PSMB1/SEM1/PSMB7/GATA3/PSMA1/PSMC3/H2AJ/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","ISG15/NUCKS1/EIF2AK2/PLSCR1/DEK/SRPK1/FAM111A/HSPA8/OAS3/IFI27/TOP2A/ILF3/RAD23A/TRIM28/SLPI/MX1","HNRNPH1/HRAS/HNRNPA1/POLR2A/UBB/PTBP1/HNRNPM/POLR2F","SFN/H2BC21/CCNB1/H4C3/H2BC4/H2BC5/H2BC7/H4C8/RPA3/PSME2/PSMA4/UBB","SFN/H2BC21/CCNB1/H4C3/H2BC4/H2BC5/H2BC7/H4C8/RPA3","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2","H2BC21/H2AC25/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/CHD4","CENPF/XPO1/PTTG1/RAD21/TUBA1B/PPP1CC/PSME2/PSMA4/UBB/TAOK1/SKA2/BIRC5/UBE2S","ISG15/IFI6/STAT1/IFITM2/IFITM3/OAS1/IFI27/BST2/MX1","H2BC21/H2AC25/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/CEBPD/SNF8","H2BC21/H2AC25/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/TUBA1B/CBX1/TRIM28","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/POLR2A/POLR2F","SHC1/BIRC6/GCC2/PIK3CA/SEC31A/SQSTM1/KIF5B/STAT3/GRB2/JUNB/TPM4/CEBPB","PMVK/FDPS/CYB5R1/MSMO1/FDFT1/IDI1/TM7SF2/DHCR7/LPCAT3/ERG28/CYB5R3/EBP/G6PD","TPR/NUP155/NUP107/NUP58/AURKB/TOP2A/BIRC5/NUP62/PCNA/AURKA","KIF15/CENPE/KIF2A/H3C2/H3C7/KIFC1/TUBB4B/GATA3/KIF20B/KIF11/WEE1/KIF21A/TUBA1B/TUBA1C/RACGAP1/CBX5/CDK2/KIF23/KIF22/RAD51C/KIF4A","H2BC21/TUBA4A/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/TUBB4B/TUBA1A/TUBA1C/DYNC1H1","TPR/XPO1/NUP155/NUP107/RAN/NUP58/KPNB1/NUP62/RANBP1","ZFP36L1/FOS/ERBB2/SOX9/BAIAP2/ZFP36/ID1","NCSTN/ARPC4/AP2M1/CLTB/ACTB/ARPC1A/ARPC1B/SDCBP/CFL1/MYL6/ARPC3/ACTG1/MYL12A/MYL12B/PSENEN/AP2S1/MYL9","MICOS10/ATP5PB/MTX1/ATP5MC3/TUBA4A/ARF4/CHCHD6/ATP5ME/TUBB/CYCS/ATP5MF/SSBP1/DCTN3/DYNC2I2/TUBB4B/ATP5F1C/ATP5MG/TUBA1B/TUBA1A/TUBA1C/ATP5MC2/ATP5F1B/DYNLL1/IFT43/RAB11A/IDH2/ATP5MC1/ATP5PD/ATP5F1A/ATP5F1D/MICOS13/DYNLRB1/ATP5F1E/ATP5PF/ATP5PO/CETN2","PSMB2/MUC1/MUC20/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBE2N/UBC/PSMB5/PSME1/PSME2/PSMC1/PYCARD/PSMB6/UBB/PSMB3/PSMC5/PSMC4","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBE2N/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","ISG15/ATP5IF1/JUN/LBR/ID2/TFRC/H4C3/H4C8/HSPA1B/VEGFA/KLF10/PTBP3/GATA3/ACIN1/ZFP36L1/FOS/B2M/JUNB/ZFP36/HMGB3","PARK7/TUBA4A/CSNK2B/TOMM7/TUBB4B/HSPA8/TUBA1B/TUBA1A/TUBA1C/UBE2N/DYNLL1/UBC/UBB/PLIN3/MAP1LC3A/TOMM22","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/POLR2F","ISG15/IFI6/EIF2AK2/STAT1/FLNB/IFITM3/CD44/OAS1/IFI27/ISG20/BST2/MX1/FLNA","HSPA1B/RPA3/SERPINH1/HSPH1/UBB/DNAJB1","NEK2/CEP97/DYNLT2B/TUBB/PCM1/ASAP1/CEP78/CDK5RAP2/DYNC2I2/TUBB4B/CDK1/CKAP5/CEP57/TUBA1B/TUBA1C/RAB3IP/CENPJ/HAUS2/CEP152/CCP110/TUBG1/HAUS1/HAUS5/MAPRE1","PMVK/FDPS/MSMO1/ACAT2/FDFT1/IDI1/TM7SF2/DHCR7/EBP","FDPS/HMGCS1/FDFT1/SQLE/IDI1/SCD/TM7SF2/FASN","NFE2L2/GSK3B/REST/HSPA9/CDK13/N4BP2L2/CDK12/STAT3/ZFP36","PPT1/ATP6V0B/SNAPIN/TMEM9/ATP6V0E1/ATP6V1F/ATP6V0E2/ATP6V0D1/GRN/ATP6AP1","CCNA2/CDKN1A/CCNE2/CDK2/E2F7/E2F1","RBBP4/H2AC20/TPR/NUP155/H3C2/H4C3/H3C7/H2BC9/H2AC16/EZH2/TUBB4B/EED/TUBA1B/TUBA1C/NUP107/NUP58/SUZ12/CBX1/NUP62/TRIM28/RBBP7","PSMB2/JUN/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBE2N/UBC/PSMB5/PSME1/PSME2/FOS/PSMC1/MAPK3/PSMB6/UBB/PSMB3/PSMC5/PSMC4","ID3/JUN/MCL1/S100A9/PHLDA3/ATF3/CAPN2/RHOB/INHBB/SPDEF/CLU/ANXA1/GSN/CTSD/PHLDA2/DDIT3/PAWR/TNFRSF12A/DAPK3/ATF4","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBE2N/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","PSMB2/SKP1/PSMB8/PSMB1/PPIA/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/FKBP1A","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/PHF19/CHD4/POLR2F","H2AC20/SMC4/H2AZ1/CCNB1/H3C2/H4C3/H3C7/H2BC9/H2AZ2/SMC2/CDK1/H2AX/NCAPD3","PSMB2/RPSA/RPL29/PSMB8/RPS12/PSMB1/SEM1/PSMB7/RPS24/PSMA1/RPS13/PSMC3/RPS26/UBC/PSMB5/PSME1/PSME2/PSMC1/RPS27L/ELOB/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","P4HA2/P4HA1/EGLN3/ERO1A/PRDX4","RPA2/POLE4/RFC4/RFC1/RPA3/POLD2/RFC2/COPS6/POLE3/POLD3/RFC3/XRCC1/LIG1/PCNA","RPA2/RFC4/RFC1/RPA3/RFC2/RFC3/PCNA","CDKN2C/TFDP2/CDKN2A/CDK4/MDM2/TFDP1/CDKN2D/ID1/E2F1","RBBP4/H2AC20/H2AZ1/DEK/H3C2/H4C3/H3C7/H2BC9/CBX3/H2AZ2/BAZ1B/EZH2/PHF19/EED/H2AX/SUZ12/UBTF/UHRF1/DNMT1/RBBP7","PSMB2/TUBA4A/SKP1/PSMB8/PSMB1/SEM1/PSMB7/TUBB4B/PSMA1/PSMC3/TUBA1B/TUBA1A/TUBA1C/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","PSMB2/POLE4/SKP1/PSMB8/PSMB1/POLD2/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","SFN/CCNB1/CDK1/WEE1/CHEK1/YWHAH","ARID1A/H2AC20/ACTL6A/H2AZ1/H3C2/H4C3/H3C7/H2AC16/H2AZ2/H2AX/SMARCC2/CDK4/COPRS/RBBP7","ATP1B1/DHX9/HNRNPU/MSH2/MSH6/ATP1B3/PLSCR1/DNAJC9/UHRF1/DNAJB1","SARS1/EPRS1/DARS1/TARS1/LARS1/GARS1/IARS1","HNRNPH1/HNRNPA1/POLR2A/PTBP1/HNRNPM/POLR2F","H2AC20/H2AZ1/LMNB1/H4C3/H2BC9/H2AZ2/RAD21/SMC3/H2AX/SYNE2/BRCA1/SYCP2/SMC1B/SMC1A","H2AC20/TFDP2/H2AZ1/DEK/H3C2/H4C3/H3C7/H2BC9/CDKN1A/H2AZ2/H2AX/CDK2/CDK4/TFDP1/E2F1","H2BC21/XPO1/CUL3/RUVBL1/GNB4/H2AZ1/SOX4/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/TLE1/PSME2/PSMA4/UBB/SOX9","RHOC/H2BC21/S100A9/RHOB/TUBA4A/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/TUBB4B/TUBA1A/TUBA1C/CLIP1/DYNC1H1/MYL9/MYH9/FLNA","H2BC21/H2AC25/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/CEBPD/TUBA1B/CBX1/SNF8/TRIM28","ATF6/ATF3/HSPA5/DDIT3/DIS3/HERPUD1/CEBPG/EIF2S2/CEBPB/ATF4","QSOX1/DNAJB2/NKTR/HSPA9/HSPA1A/HSPA1B/HSPB1/DNAJB6/CLU/HSPA5/HSP90B1/ERO1A/BAG5/LMAN1/CALR/DNAJB1/PFDN4/PRDX4","HK2/PFKP/TPI1/ENO2/PKM/PGK1","HK2/PFKP/TPI1/ENO2/PKM/PGK1","HK2/PFKP/TPI1/ENO2/PKM/PGK1","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/RPA3/H2AZ2/PSME2/PSMA4/UBB/UBE2S","DNAJB2/HSPA1A/HSPA1B/HSPB1/CRYAB/HSPA8","DHCR24/FDPS/MSMO1/HMGCS1/SQLE/IDI1/DHCR7/G6PD","PSMB2/PSMB8/PSMB1/ACTB/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/NEDD8/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","ID3/ID2/CHD4/FOS/JUNB/FOSB/ID1","RHOC/H2BC21/RHOB/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/MYL9/MYH9","EFNA1/ATP1B1/MDM4/TAF9/SOX4/HSPA1A/HSPA1B/CLU/CTNND1/GAPDH/SAV1/SEL1L/BAG5/CALR/NOP53/PFDN4/LAMP2/FLNA","TFDP2/CDKN1A/CCND3/CCNE2/CDK2/CDK4/TFDP1/UBE2S/E2F1/UBE2C","SCP2/MTX1/COA6/CHCHD5/COX17/TOMM7/SEC61G/FIS1/GSTK1/CYC1/SEC61B/NDUFB8/TIMM10/COA4/TIMM8B/ATP5F1B/UBC/UBB/ATP5MC1/ATP5F1A/CYB5A/TIMM13/ECH1/CHCHD10/TOMM22/TIMM17B","ISG15/IFI6/STAT1/IFITM3/OAS1/IFI27/ISG20/BST2/MX1","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/RPA3/H2AZ2","ISG15/JUN/LAMTOR5/VAMP8/STAT1/RPSA/SPCS1/PARP9/PLSCR1/HSPA1A/HSPA1B/SNX3/DYNLT1/PPIA/LY6E/VPS28/GSN/STOM/MAN1B1/IFITM2/IFITM3/PSMC3/CFL1/BANF1/HSPA8/OAS1/LAMP1/CHMP4A/IFI27/ISG20/SNF8/PHB1/BSG/EIF3G/BST2/ZFP36/NECTIN2/SLC1A5/PRMT1/CHMP2A/SLPI/MX1/LGALS1/MCTS1","ID3/SFPQ/JUN/S100A9/ATF3/RHOB/PRKCI/SKIL/SPDEF/CDKN1A/PRKDC/B4GALT1/NET1/PHLDA2/NR4A1/ATP2A2/FOXA1/CTSH/PYCARD/NQO1/GADD45B/BMP7/LGALS1/DDX3X","RPA2/RFC4/RFC1/RPA3/RFC2/RFC3/PCNA","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/CHD4/POLR2F","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/POLR2F","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/POLR2F","DHCR24/FDPS/MSMO1/HMGCS1/FDFT1/SQLE/IDI1/TM7SF2/ACLY","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/CDKN1A/H2AZ2","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","JUN/H2BC21/ELF3/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2","TPR/NUP155/CCNB1/LMNB1/CDK1/NUP107/TMPO/NUP58/VRK1/CCNB2/NUP62/EMD","CSRP1/DSP/HSPB1/JUP/MYL12A/DSC2/DSG2/MYL9/MYH9/FLNA","PARK7/LAMTOR5/LAMTOR2/TUBA4A/CSNK2B/TOMM7/LAMTOR4/VPS28/RRAGA/TUBB4B/LAMTOR1/HSPA8/TUBA1B/TUBA1A/TUBA1C/UBE2N/DYNLL1/UBC/CHMP4A/GABARAPL2/UBB/PLIN3/CHMP2A/MAP1LC3A/TOMM22","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/POLR2A/POLR2F","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/CHD4/POLR2F","PSMB2/MZT2B/MZT2A/TUBA4A/SKP1/TUBB/PSMB8/PSMB1/SEM1/DCTN3/PSMB7/TUBB4B/PSMA1/PSMC3/TUBA1B/TUBA1A/TUBA1C/DYNLL1/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1/CETN2","ENO1/MDH1/MDH2/FBP1/PGAM1/GAPDH/TPI1/ALDOA/SLC25A11/G6PC3/PGK1","PSMB2/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/RPA3/H2AZ2","PSMB2/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","ESCO2/RAD21/SMC3/CDCA5/PDS5B/SMC1A","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2","RPA2/PARP1/POLE4/ACTL6A/RFC4/RFC1/RPA3/POLD2/RFC2/COPS6/POLE3/POLD3/RFC3/PARP2/RAD23A/XRCC1/LIG1/PCNA","JUN/H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/FOS/UBB","COX20/COX5B/COX7C/NDUFA4/CYCS/BLVRA/SURF1/COX8A/COX14/COX6A1/COX5A/COX6B1/BLVRB/HM13/COX7B","ATF6/HSPA5/DDIT3/HSP90B1/CALR/XBP1/ATF4","H2BC21/LMNA/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/RPA3/H2AZ2/RAD21/SYNE2","SHC1/GFPT1/SRPRB/SERP1/SEC31A/DNAJB9/HSPA5/YIF1A/SRPRA/DNAJC3/GOSR2/ERN1/XBP1","NEK2/PSMD1/MAD2L1/CCNA2/CCNB1/PTTG1/PSMB9/FBXO5/CDK1/BUB3/PSMC3/CDK2/AURKB/UBE2S/UBE2C/AURKA","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2","SFN/H2BC21/S100A9/CENPF/ROCK2/RHOB/XPO1/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/TUBA1B/PPP1CC/TAOK1/SKA2/BIRC5","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2","H2BC21/LMNA/SMC4/H2AZ1/CCNB1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/TMPO","LMNA/CENPF/LBR/XPO1/CCNB1/PTTG1/RAD21/TUBA1B/TMPO/PPP1CC/PSME2/PSMA4/UBB/TAOK1/SKA2/BIRC5/UBE2S","JUN/H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/CDKN1A/H2AZ2/FOS/UBB/UBE2S/ID1","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/TNFRSF1A/UBC/PSMB5/PSME1/PSME2/PSMC1/TNFRSF12A/PSMB6/UBB/PSMB3/PSMC5/PSMC4","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","JUN/H2BC21/ELF3/HES1/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/TLE1/MDK/PSME2/PSMA4/UBB","FDPS/HSD17B7/MSMO1/HMGCS1/HMGCR/FDFT1/SQLE/IDI1/DHCR7/EBP","TPR/PARP1/SMC6/RNF168/NUP155/RAD21/NSMCE2/CDKN2A/SMC3/NUP107/NUP58/BLM/BRCA1/NUP62/SMC1A","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBE2N/UBC/PSMB5/PSME1/PSME2/PSMC1/PYCARD/PSMB6/UBB/PSMB3/PSMC5/PSMC4","PSMB2/JUN/PSMB8/PSMB1/HSPB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/RPA3/H2AZ2/PSME2/PSMA4/UBB/UBE2S","DHCR24/FDPS/MSMO1/HMGCS1/SQLE/IDI1/DHCR7","S100A9/PKP1/DEGS1/TMEM63A/CD47/S100P/DSP/TUBB/HSPA1A/HSPA1B/GSN/STOM/TUBB4B/CTSD/CD59/CD44/HSPA8/MGST1/LGALS3/DYNC1H1/COTL1/JUP/BST2/SLPI","TUBA4A/HSPA1A/HSPA1B/TUBB4B/STIP1/HSPA8/TUBA1A/TUBA1C/DYNC1H1","SFN/PSMB2/PSMB8/PSMB1/CYCS/DBNL/SEM1/GSDMD/CDKN2A/GSN/PSMB7/PSMA1/PSMC3/BAD/DYNLL1/UBC/PSMB5/PSME1/PSME2/PSMC1/MAPK3/PSMB6/C1QBP/UBB/PSMB3/PSMC5/PSMC4/BCAP31","PSMB2/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","ATF3/DDIT3/CEBPG/PPP1R15A/TRIB3/EIF2S2/CEBPB/ATF4","RBBP4/CKS1B/PSMD1/TFDP2/LIN54/CCNA2/PSMB9/CDKN1A/CCNE2/WEE1/PSMC3/CDK2/CDK4/TFDP1/CDC25B/E2F1","SFN/ATP5IF1/JUN/GLUL/KDM5B/BMPR2/HES1/VEGFA/LAMB1/MTSS1/GATA3/HRAS/MDK/NR4A1/HMGB1/ZFP36L1/B2M/NR2F2/SOX9/ZFP36/ID1","HSPA5/DDIT3/HSP90B1/CALR/XBP1/ATF4","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/CHD4","SFN/H2BC21/RHOB/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/CDKN1A/H2AZ2","RBBP4/H2AC20/TPR/NUP155/H3C2/H4C3/H3C7/H2BC9/H2AC16/EZH2/TUBB4B/EED/TUBA1B/TUBA1C/NUP107/NUP58/SUZ12/CBX1/NUP62/TRIM28/RBBP7","ATF6/HSPA5/DDIT3/HSP90B1/CALR/XBP1/ATF4","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/PHF19","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/GATA3/PSME2/PSMA4/UBB","H2BC21/RUVBL1/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/TLE1","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/B2M/UBB/CST3/USP9X","PSMB2/HES1/PSMB8/PSMB1/SEM1/CDKN2A/PSMB7/PSMA1/PSMC3/CCND1/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","CDKN2C/CKS1B/TFDP2/CDKN1A/CCND3/CCNE2/CDKN2A/CDK2/CDK4/TFDP1/CDKN2D/E2F1","RPA2/RPA3/POLD2/FEN1/POLD3/BLM/LIG1/PCNA","RBBP4/TFDP2/LIN54/TFDP1/CDC6/TOP2A/PCNA/E2F1","SFN/H2BC21/CENPF/XPO1/CCNB1/H4C3/H2BC4/H2BC5/H2BC7/H4C8/CDKN1A/RPA3/PPP1CC/PSME2/PSMA4/UBB/TAOK1/SKA2/BIRC5/UBE2S","DHCR24/FDPS/MSMO1/HMGCS1/FDFT1/SQLE/IDI1/TM7SF2","SFN/PSMB2/FLOT1/PSMB8/PSMB1/CYCS/DBNL/SEM1/SDCBP/GSDMD/CDKN2A/GSN/PSMB7/PSMA1/PSMC3/BAD/DYNLL1/UBC/PSMB5/PSME1/PSME2/CHMP4A/PSMC1/MAPK3/PSMB6/C1QBP/UBB/PSMB3/PSMC5/PSMC4/CHMP2A/BCAP31","MCM6/PSMD1/MCM2/CCNA2/PSMB9/MCM3/MCM7/MCM4/PSMC3/CDK2/ORC6/CDT1/CDC6/MCM8/MCM5","JUN/H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/POLR2A/NCOA3/POLR2F","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","H2BC21/XPO1/CUL3/RUVBL1/H2AZ1/SOX4/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/TLE1/PSME2/PSMA4/UBB/SOX9","KIF15/CENPE/KIF2A/KIFC1/TUBB4B/KIF20B/KIF11/KIF21A/TUBA1B/TUBA1C/RACGAP1/KIF23/KIF22/KIF4A","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBE2N/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","SFN/TFDP2/CCNB1/CDK1/TFDP1/BAX/PCNA/AURKA","CDKN2C/JUN/H2AC20/H2AZ1/CCNA2/H3C2/H4C3/H3C7/H2BC9/CDKN1A/H2AZ2/CDKN2A/H2AX/CDK2/CDK4/FOS/CDKN2D/UBE2S/UBE2C","H2BC21/LMNA/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/RPA3/H2AZ2/RAD21/SYNE2","RBBP4/H2AC20/H2AZ1/H3C2/H4C3/H3C7/H2BC9/H2AZ2/EZH2/PHF19/EED/H2AX/SUZ12/DNMT1/RBBP7","ATF3/DDIT3/CEBPG/PPP1R15A/TRIB3/CEBPB/ATF4","CREG1/QSOX1/CD55/TMEM63A/S100P/LPCAT1/CYSTM1/DSP/HSPA1A/HSPA1B/NEU1/BRI3/CTSB/PSAP/PGAM1/CD59/CD44/FTH1/MGST1/LGALS3/NPC2/PKM/COTL1/CFD/GPI/FTL/A1BG/PDXK/CSTB/MIF/PRDX4/DDX3X/LAMP2","HSPA5/DDIT3/HSP90B1/CALR/XBP1/ATF4","ISG15/EIF2AK2/STAT1/PLSCR1/LY6E/IFITM2/IFITM3/OAS1/BST2/ZFP36/SLPI/MX1","ISG15/JUN/NFIA/EIF2AK2/STAT1/PARP9/PLSCR1/ST6GAL1/TFRC/HSPA1A/HSPA1B/WWP1/LY6E/IFITM2/IFITM3/HSPA8/OAS1/IFI27/BST2/ZFP36/SLPI/MX1/LGALS1/DDX3X","SRRM1/SRSF4/SRSF7/SNRPG/SLBP/THOC3/SRSF3/MAGOHB/RNPS1/SRSF2/ALYREF/DDX39A/SNRPB/UPF3B/THOC2","TFDP2/CDKN1A/CCND3/CCNE2/CDK2/CDK4/TFDP1/E2F1","H2BC21/RUVBL1/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/RPA3/H2AZ2/POLR2A/POLR2F","PARP1/POLQ/FEN1/BRCA2/PARP2/RBBP8/XRCC1","TFDP2/CCNB1/CDK1/PRIM1/TFDP1/ORC6/MCM8/E2F1/POLA1","PSMB2/JUN/LAMTOR5/LAMTOR2/CSNK2B/PSMB8/PSMB1/SEM1/LAMTOR4/RRAGA/PSMB7/PSMA1/PSMC3/LAMTOR1/UBC/PSMB5/PSME1/PSME2/PSMC1/MTA1/MAPK3/PSMB6/UBB/PSMB3/PSMC5/PSMC4","PSMB2/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","MTX1/COA6/CHCHD5/COX17/TOMM7/CYC1/NDUFB8/TIMM10/COA4/TIMM8B/ATP5F1B/ATP5MC1/ATP5F1A/TIMM13/CHCHD10/TOMM22/TIMM17B","MIA3/NRBP1/SEC13/ARF4/COPG1/SEC31A/SAR1B/TMED4/INSIG1/RNF139/GOLGA2/YIF1A/GOLT1B/COG3/MIA2/TRIP11/TMED3/WHAMM/GOSR1/TRAPPC6A/TRAPPC2B/TBC1D20/YIPF6","PSMB2/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","ISG15/JUN/NFIA/DHX9/NUCKS1/TFRC/HSPA1B/ST3GAL1/EIF3A/IFITM3/RSF1/HMGB1/IFI27/TOP2A/HEXIM1/SNF8/KPNA2/PTBP1/ZFP36/TRIM28/SLPI/MX1/LGALS1/DDX3X","RBBP4/TFDP2/LIN54/CCNA2/CDK1/TFDP1/E2F1/MYBL2","ATF3/GFPT1/CREBRF/SSR1/HSPA5/YIF1A/DDIT3/HSP90B1/HERPUD1/GOSR2/CALR/CEBPG/CEBPB/XBP1/ATF4","H2BC21/SMC4/H2AZ1/CCNB1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2","PSMB2/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/PSMB5/PSME1/PSME2/PSMC1/PSMB6/PSMB3/PSMC5/PSMC4","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/CCND1/CDK4/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","DHCR24/JUN/MCL1/CAPN2/SDC1/RHOB/EPAS1/PNPT1/STAT1/HSPA1A/HSPA1B/HSPB1/ANXA1/MSRB2/UCP2/CRYAB/MGST1/PAWR/TPM1/NFE2L1/ATF4/G6PD","PSMB2/AP2M1/CLTB/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/PFN1/UBB/PSMB3/PSMC5/PSMC4/AP2S1","PSMB2/CSNK2B/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","JUN/H2BC21/LMNA/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/PRDX2/CDC25B","RFC4/RFC1/POLD2/RFC2/DSCC1/POLD3/PRIM1/RFC3/PCNA/POLA1","PSMB2/PSMB8/PSMB1/SEM1/CDKN2A/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","PSMB2/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","PSMB2/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","NEK2/CEP97/TUBB/PCM1/CEP78/CDK5RAP2/TUBB4B/CDK1/CKAP5/CEP57/RAB3IP/CENPJ/HAUS2/CEP152/CCP110/TUBG1/HAUS1/HAUS5/MAPRE1","XPO1/CCNA2/CCNB1/CDK1/WEE1/FOXM1/CDK2/CCNB2/PKMYT1/CDC25B","PARK7/PRDX1/AKR1A1/S100A9/MGST3/PRDX6/ADH5/MGST2/GSTK1/TXN/PRXL2A/GSTO1/PRDX5/MGST1/ESD/NQO1/TXNDC17/GPX4/PRDX2/SELENOW/BMP7/PRDX4","PSMB2/TUBA4A/SKP1/PSMB8/PSMB1/SEM1/PSMB7/TUBB4B/PSMA1/PSMC3/TUBA1B/TUBA1A/TUBA1C/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","PSMB2/SKP1/PSMB8/PSMB1/PPIA/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/FKBP1A","H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/POLR2A/POLR2F","LBR/CCNB1/LMNB1/CDK1/TMPO/VRK1/CCNB2/KPNB1/EMD","JUN/NCK1/CEBPD/HSPA5/DDIT3/FOS/JUNB/CEBPG/PPP1R15A/CEBPB/ATF4","ATF3/DNAJB2/NCK1/HSPA9/CREBRF/STC2/HSPA1A/HSPA1B/HSPB1/HSPA5/CCND1/DDIT3/ERO1A/HERPUD1/DNAJB1/PPP1R15A/XBP1/ATF4","DNAJB2/HSPA1A/HSPA1B/DNAJB6/CLU/BAG5/CDC34/DNAJB1","MDM4/FAM162A/STC2/VEGFA/CITED2/HILPDA/BNIP3L/NDRG1/BNIP3/EGLN3/ERO1A/ZFP36L1/FOS/RORA/JUND/NOP53/ATF4/PGK1","ATF3/DNAJB2/NCK1/HSPA9/CREBRF/STC2/HSPA1A/HSPA1B/HSPB1/CLU/HSPA5/CCND1/DDIT3/ERO1A/HERPUD1/DNAJB1/PPP1R15A/XBP1/ATF4","H2BC21/LMNA/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/RAD21/SYNE2","PSMB2/STAT1/HES1/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/CCND1/CDK4/UBC/PSMB5/PSME1/PSME2/LGALS3/PSMC1/MAPK3/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","RPA2/PARP1/POLE4/ACTL6A/RFC4/RFC1/RPA3/POLD2/RFC2/COPS6/POLE3/POLD3/RFC3/PARP2/RAD23A/XRCC1/LIG1/PCNA","JUN/H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/CDKN1A/H2AZ2/FOS/UBB/UBE2S","MCM6/PSMD1/MCM2/CCNA2/GMNN/PSMB9/MCM3/MCM7/MCM4/CCNE2/PSMC3/CDK2/ORC6/CDT1/CDC6/UBE2S/MCM8/UBE2C/MCM5","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","RPA2/H2AC20/H2AZ1/H3C2/H4C3/H3C7/H2BC9/RPA3/H2AZ2/H2AX/CDK2/CDK4/BRCA2/BLM/PSMC3IP/BRCA1/RAD51C/RBBP8","DHX9/TPR/AHCTF1/SRSF7/XPO1/SLBP/NUP155/THOC3/SRSF3/HNRNPA2B1/CKAP5/MAGOHB/HNRNPA1/NUP107/RAN/NUP58/KPNB1/ALYREF/KHSRP/DDX39A/NUP62/NXT1/UPF3B/THOC2","DHX9/TPR/AHCTF1/SRSF7/XPO1/SLBP/NUP155/THOC3/SRSF3/HNRNPA2B1/CKAP5/MAGOHB/HNRNPA1/NUP107/RAN/NUP58/KPNB1/ALYREF/KHSRP/DDX39A/NUP62/NXT1/UPF3B/THOC2","EXO1/GEN1/BARD1/RMI1/RAD51AP1/BRCA2/BLM/RMI2/BRCA1/RAD51C/BRIP1/RBBP8","RPA2/PARP1/POLE4/RFC4/RFC1/RPA3/POLD2/RFC2/POLE3/POLD3/RFC3/PARP2/PCNA","MICOS10/ATP5PB/MTX1/ATP5MC3/CHCHD6/ATP5ME/CYCS/ATP5MF/SSBP1/ATP5F1C/ATP5MG/ATP5MC2/ATP5F1B/IDH2/ATP5MC1/ATP5PD/ATP5F1A/ATP5F1D/MICOS13/ATP5F1E/ATP5PF/ATP5PO","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","PSMB2/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","PSMB2/HIGD1A/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/ELOB/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","H2BC21/RUVBL1/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/CENPW/RPA3/H2AZ2/RSF1/POLR2A/MIS18A/POLR2F","EXO1/BARD1/RMI1/RAD51AP1/BRCA2/BLM/RMI2/BRCA1/RAD51C/BRIP1/RBBP8","TPR/PARP1/SMC6/RNF168/ING2/NUP155/H4C3/RAD21/NSMCE2/CDKN2A/SMC3/CBX5/NUP107/MDM2/NUP58/BLM/AURKB/SUZ12/TOP2A/BRCA1/BIRC5/DNMT1/UBA2/SAE1/NUP62/TRIM28/PCNA/AURKA/SMC1A","H2BC21/ANP32E/LMNA/KDM5B/H2AC25/RUVBL1/H4C3/H2BC4/H2BC5/H2BC7/H4C8/CENPW/GATA3/RSF1/KDM5A/CHD4/HMGB1/BAZ1A/FOXA1/BPTF/SOX9/TRIM28/TOP1/MIS18A","ISG15/RPA2/POLE4/RFC4/RFC1/RPA3/POLD2/RFC2/POLE3/POLD3/RFC3/PCLAF/PCNA","PSMB2/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/PSMB5/PSME1/PSME2/PSMC1/NQO1/PSMB6/PSMB3/PSMC5/PSMC4/SMS/SAT1","PSMB2/HSPA1A/PSMB8/PSMB1/HSPB1/SEM1/EXOSC4/PSMB7/PSMA1/PSMC3/HSPA8/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/ZFP36/PSMC4","PSMB2/STAT1/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","LBR/AHCTF1/NUP155/CCNB1/LMNB1/TUBB4B/CDK1/TUBA1B/TUBA1C/NUP107/TMPO/RAN/NUP58/VRK1/CCNB2/KPNB1/NUP62/EMD","SRRM1/SRSF4/TPR/SRSF7/SLBP/NUP155/THOC3/SRSF3/MAGOHB/NUP107/NUP58/RNPS1/SRSF2/ALYREF/DDX39A/NUP62/NXT1/UPF3B/THOC2","RBBP4/CENPF/LIN54/CCNB1/WEE1/FOXM1/CCNB2/PKMYT1/MYBL2","RPA2/H2AC20/H2AZ1/LMNB1/H3C2/H4C3/H3C7/H2BC9/RPA3/H2AZ2/RAD21/SMC3/H2AX/CDK2/CDK4/BRCA2/SYNE2/BLM/PSMC3IP/BRCA1/RAD51C/RBBP8/SYCP2/SMC1B/SMC1A","RBBP4/CDKN2C/JUN/H2AC20/TFDP2/H2AZ1/H3C2/H4C3/H3C7/H2BC9/H2AZ2/EZH2/CDKN2A/EED/H2AX/CDK4/MDM2/TFDP1/FOS/SUZ12/CDKN2D/E2F1/RBBP7","ISG15/EIF2AK2/IFIH1/STAT1/PLSCR1/GSN/IFITM3/OAS1/ISG20/BST2/ZFP36/SLPI/MX1","RBBP4/JUN/H2AC20/H2AZ1/LMNB1/H3C2/H4C3/H3C7/H2BC9/H2AZ2/EZH2/CDKN2A/EED/H2AX/PRIM1/SUZ12/DNMT1/PRDX2/CDC25B/RBBP7/POLA1","PSMB2/S100A9/VAMP8/PSMB8/PSMB1/SEC61G/SEM1/SEC61B/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/B2M/CYBA/PSMB6/UBB/PSMB3/PSMC5/CALR/PSMC4","PSMB2/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","PSMB2/PSMB8/PSMB1/SEM1/CDKN2A/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","PSMB2/AP2M1/SKP1/PSMB8/HMGA1/GTF2H5/PSMB1/PPIA/SEM1/AP1S1/VPS28/PSMB7/POLR2L/TAF10/PSMA1/PSMC3/POLR2G/BANF1/UBC/PSMB5/PSME1/PSME2/CHMP4A/PSMC1/B2M/GTF2A2/ELOB/PSMB6/UBB/PSMB3/PSMC5/POLR2E/AP1M2/POLR2I/PSMC4/AP2S1/CHMP2A/RBX1/SLC25A5","H2AC20/TPR/SMC4/H2AZ1/NUP155/CCNB1/LMNB1/H3C2/H4C3/H3C7/H2BC9/H2AZ2/SMC2/MASTL/CDK1/NUMA1/H2AX/NCAPD3/NUP107/TMPO/NUP58/VRK1/CCNB2/NUP62/EMD","RPA2/EXO1/MSH2/MSH6/RPA3/POLD2/POLD3/LIG1/PCNA","RPA2/RPA3/POLD2/FEN1/POLD3/PRIM1/LIG1/PCNA/POLA1","PSMB2/NCSTN/HES1/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSENEN/PSMC4/RBX1","SFN/TFDP2/CCNA2/CCNB1/CDKN1A/CCNE2/CDK1/CDK2/E2F7/CENPJ/TFDP1/BAX/PCNA/E2F1/AURKA","PSMB2/PSMB8/PSMB1/SEM1/CDKN2A/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","ID3/SFPQ/DHX9/KDM5B/HNRNPU/ID2/ROCK2/BHLHE40/HNRNPD/NAMPT/KLF10/SETX/MDK/KDM5A/PPP1CC/USP7/TOP2A/ATF5/ID1/TOP1/ADNP/USP9X/HUWE1","NCAPH/SMC4/NCAPG/CCNB1/SMC2/CDK1/NCAPD2/CCNB2","JUN/H2BC21/XPO1/GNB4/APOD/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/RAD21/GATA3/HRAS/FOXA1/FOS/POLR2A/KPNA2/FOSB/NCOA3/ZNF217/POLR2F","NEK2/CCNB1/TUBB/PCM1/CEP78/CDK5RAP2/TUBB4B/CDK1/CKAP5/CEP57/CENPJ/HAUS2/CEP152/CCNB2/CCP110/TUBG1/HAUS1/HAUS5/MAPRE1/AURKA","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","H2BC21/RUVBL1/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/CENPW/H2AZ2/RSF1/MIS18A","CHTOP/DHX9/HNRNPU/HNRNPA3/LSM3/HNRNPD/HNRNPH1/SRSF3/HNRNPA2B1/LSM5/CCAR1/HNRNPA1/SNRPF/ACIN1/PNN/POLR2A/EIF4A3/PTBP1/HNRNPM/DDX39A/RBM42/SNRPA/SNRPB/POLR2F","PSMB2/TUBA4A/PSMB8/PSMB1/SEM1/PSMB7/TUBB4B/PSMA1/PSMC3/TUBA1B/TUBA1A/TUBA1C/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","SFN/TACSTD2/JUN/KDM5B/STAT1/BMPR2/HES1/VEGFA/PRKDC/MTSS1/EPPK1/NFIB/B4GALT1/GATA3/MDK/CCND1/NR4A1/ZFP36L1/SYNJ2BP/NR2F2/CYBA/ERBB2/STAT3/SOX9/ZFP36/ID1","RFC4/RFC1/POLD2/RFC2/POLD3/PRIM1/RFC3/PCNA/POLA1","PSMB2/PRDX1/TKT/SKP1/CSNK2B/PSMB8/PSMB1/SEM1/CDKN2A/PSMB7/TALDO1/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/NQO1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1/G6PD","PSMB2/AP2M1/SKP1/PSMB8/HMGA1/PSMB1/PPIA/SEM1/AP1S1/PSMB7/PSMA1/PSMC3/BANF1/UBC/PSMB5/PSME1/PSME2/PSMC1/B2M/ELOB/PSMB6/UBB/PSMB3/PSMC5/AP1M2/PSMC4/AP2S1/RBX1/SLC25A5","SFN/PRDX1/LAMTOR5/LAMTOR2/COX20/COX5B/COX7C/NDUFA4/CYCS/LAMTOR4/RRAGA/TXN/SURF1/COX8A/PRDX5/LAMTOR1/COX14/COX6A1/COX5A/PRDX2/COX6B1/COX7B/G6PD","RPA2/H2AC20/RFC4/RFC1/H2AZ1/CCNA2/H4C3/H2BC9/RPA3/POLD2/H2AZ2/RFC2/DSCC1/FEN1/POLD3/H2AX/CDK2/PRIM1/RFC3/BLM/LIG1/PCNA/POLA1","ISG15/RPA2/POLE4/RFC4/RFC1/RPA3/POLD2/RFC2/POLE3/POLD3/RFC3/PCLAF/PCNA","NEK2/MZT2B/MZT2A/TUBB/PCM1/CEP78/CDK5RAP2/TUBB4B/CDK1/CKAP5/CEP57/CENPJ/MZT1/HAUS2/CEP152/CCP110/TUBG1/HAUS1/HAUS5/MAPRE1","H2BC21/LMNA/CENPF/LBR/XPO1/MZT2B/MZT2A/SMC4/H2AZ1/CCNB1/PTTG1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/RAD21/DCTN3/TUBA1B/TMPO/PPP1CC/PSME2/PSMA4/UBB/TAOK1/SKA2/BIRC5/UBE2S","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/CCND1/CDK4/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","PSMB2/SKP1/PSMB8/PSMB1/SEM1/PSMB7/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1","DHX9/HNRNPU/HNRNPA3/LSM3/HNRNPD/HNRNPH1/SRSF3/HNRNPA2B1/LSM5/CCAR1/HNRNPA1/SNRPF/ACIN1/PNN/POLR2A/EIF4A3/PTBP1/HNRNPM/RBM42/SNRPA/SNRPB/POLR2F","ISG15/RPA2/USP1/DTL/POLE4/RAD18/RFC4/RFC1/RPA3/POLD2/RFC2/POLE3/POLD3/RFC3/PCLAF/PCNA","MCM6/MCM2/MCM3/MCM7/MCM4/GINS2/MCM8/GINS1/MCM5","JUN/ATF6/NIBAN1/PPP1R15B/ATF3/GORASP2/NFE2L2/DNAJB2/SELENOK/GSK3B/NCK1/SERP1/DNAJB14/MARCHF6/CERT1/DNAJB9/CLU/DERL1/RNF139/HSPA5/CREBZF/DDIT3/HSP90B1/UFM1/TRIM13/DNAJC3/SEL1L/SELENOS/HERPUD1/ERN1/NPLOC4/CALR/PPP1R15A/NR1H2/TRIB3/SERINC3/CEBPB/XBP1/ATF4/DDX3X","UHMK1/RC3H1/NIBAN1/PPP1R15B/EPRS1/EIF1B/GSK3B/CNBP/NCK1/XRN1/SERP1/EIF4A2/PURA/CNOT8/CPEB4/SOX4/FOXO3/MTPN/SLC25A37/EIF4EBP1/RB1CC1/RNF139/AGO2/TUT7/KLF4/GOLGA2/TNKS1BP1/ATG101/EIF4B/PAWR/ATXN2/DIS3/DNAJC3/ZFP36L1/EIF5/CHP1/SECISBP2L/SELENOS/AARS1/GRB7/CASC3/EIF1/TOB1/MKNK2/DAPK3/EEF2/CALR/HNRNPL/SAMD4B/ZFP36/PPP1R15A/PPP1R3D/YTHDF1/APP/ATF4/TNRC6B/DDX3X","RBBP4/CDKN2C/JUN/H2AC20/TFDP2/H2AZ1/CCNA2/LMNB1/H3C2/H4C3/H3C7/H2BC9/HMGA1/CDKN1A/H2AZ2/EZH2/CCNE2/CDKN2A/EED/H2AX/CDK2/CDK4/MDM2/TFDP1/FOS/SUZ12/CDKN2D/UBE2S/ID1/E2F1/UBE2C/RBBP7","RBBP4/RRM2/TFDP2/CBX3/EZH2/EED/CHEK1/CBX5/TFDP1/SUZ12/BRCA1/RBBP8/E2F1/RBBP7","JUN/H2BC21/XPO1/GNB4/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/RAD21/GATA3/HRAS/FOXA1/FOS/POLR2A/KPNA2/FOSB/NCOA3/ZNF217/POLR2F","RPA2/H2AC20/H2AZ1/LMNB1/H3C2/H4C3/H3C7/H2BC9/RPA3/H2AZ2/RAD21/SMC3/H2AX/CDK2/CDK4/BRCA2/SYNE2/BLM/PSMC3IP/BRCA1/RAD51C/RBBP8/SYCP2/SMC1B/SMC1A","RPA2/RFC4/RFC1/CCNA2/RPA3/POLD2/RFC2/DSCC1/FEN1/POLD3/CDK2/PRIM1/RFC3/BLM/LIG1/PCNA/POLA1","NEK2/HMMR/TUBB/PCM1/CEP78/CDK5RAP2/TUBB4B/CDK1/CKAP5/CEP57/CENPJ/HAUS2/CEP152/CCP110/TUBG1/HAUS1/HAUS5/TPX2/MAPRE1/AURKA","PSMB2/HSPA1A/PSMB8/PSMB1/HSPB1/SEM1/PSMB7/PSMA1/PSMC3/HSPA8/UBC/PSMB5/PSME1/PSME2/PSMC1/PSMB6/UBB/PSMB3/PSMC5/PSMC4","NEK2/MZT2B/MZT2A/TUBB/PCM1/CEP78/CDK5RAP2/TUBB4B/CDK1/CKAP5/NUMA1/CEP57/TUBA1B/TUBA1C/CENPJ/MZT1/HAUS2/CEP152/CCP110/TUBG1/HAUS1/HAUS5/MAPRE1","JUN/H2BC21/H2AZ1/H4C3/H2BC4/H2AC6/H2BC5/H2BC7/H4C8/H2AZ2/RAD21/GATA3/FOXA1/FOS/POLR2A/KPNA2/FOSB/NCOA3/ZNF217/POLR2F","SFPQ/DHX9/HNRNPU/HNRNPA3/NCL/LSM3/HNRNPH1/SRSF3/HNRNPA2B1/LSM5/SETX/HNRNPH3/CELF1/HNRNPA1/SNRPF/PNN/EIF4A3/PTBP1/DAZAP1/HNRNPM/DDX39A/RBM42/SNRPA/SNRPB/RBM38/RBM3","SFPQ/DHX9/HNRNPU/HNRNPA3/NCL/LSM3/HNRNPH1/SRSF3/HNRNPA2B1/LSM5/SETX/HNRNPH3/CELF1/HNRNPA1/SNRPF/PNN/EIF4A3/PTBP1/DAZAP1/HNRNPM/DDX39A/RBM42/SNRPA/SNRPB/RBM38/RBM3","SFPQ/DHX9/HNRNPU/HNRNPA3/NCL/LSM3/HNRNPH1/SRSF3/HNRNPA2B1/LSM5/SETX/HNRNPH3/CELF1/HNRNPA1/SNRPF/PNN/EIF4A3/PTBP1/DAZAP1/HNRNPM/DDX39A/RBM42/SNRPA/SNRPB/RBM38/RBM3","RPA2/USP1/DTL/POLE4/RAD18/RFC4/RFC1/RPA3/POLD2/RFC2/POLE3/POLD3/RFC3/PCNA","RPA2/POLE4/RFC4/RFC1/RPA3/POLD2/RFC2/POLE3/POLD3/RFC3/XRCC1/LIG1/PCNA","RPA2/RFC4/RFC1/RPA3/POLD2/RFC2/DSCC1/FEN1/POLD3/PRIM1/RFC3/BLM/LIG1/PCNA/POLA1","PSMB2/PRDX1/TKT/SKP1/PSMB8/PSMB1/SEM1/CDKN2A/PSMB7/TALDO1/PSMA1/PSMC3/UBC/PSMB5/PSME1/PSME2/PSMC1/NQO1/PSMB6/UBB/PSMB3/PSMC5/PSMC4/RBX1/G6PD","SHC1/ATF6/ATF3/GFPT1/SRPRB/SERP1/SEC31A/CREBRF/DNAJB9/HSPA5/YIF1A/SRPRA/DDIT3/HSP90B1/DIS3/DNAJC3/HERPUD1/GOSR2/ERN1/CALR/CEBPG/EIF2S2/CEBPB/XBP1/ATF4","RBBP4/TFDP2/LIN54/CCNA2/CCNE2/CDK1/CDK2/TFDP1/CDC6/TOP2A/PCNA/E2F1/MYBL2/DYRK1A","RPA2/EXO1/BARD1/TOPBP1/RFC4/RAD1/RPA3/RFC2/RMI1/RHNO1/RFC3/BLM/RMI2/BRCA1/BRIP1/RBBP8","NUCKS1/PARP1/MSH2/MAP3K20/BARD1/FANCD2/TOPBP1/ECT2/RNF168/RAD1/CDKN1A/FIGNL1/PRKDC/RAD54B/NET1/GATA3/DCLRE1C/HRAS/H2AX/RHNO1/RAD51AP1/NABP2/MDM2/BRCA2/BLM/CYBA/BRCA1/BAX","RPA2/POLE4/RFC4/RFC1/RPA3/POLD2/RFC2/POLE3/FEN1/POLD3/RFC3/LIG1/PCNA","RBBP4/ITGB3BP/H2AC20/CENPL/CENPA/H2AZ1/CENPU/CENPK/CENPH/H4C3/H2BC9/CENPQ/CENPW/H2AZ2/H2AX/MIS18BP1/KNL1/CENPN/CENPX/MIS18A/CENPM/RBBP7","RBBP4/RRM2/TFDP2/LIN54/DHFR/FBXO5/CDK1/TFDP1/CDT1/CDC6/TK1/TYMS/PCNA/E2F1/POLA1","RPA2/RFC4/RFC1/RPA3/POLD2/RFC2/FEN1/POLD3/PRIM1/RFC3/LIG1/PCNA/POLA1","RPA2/CLSPN/EXO1/BARD1/TOPBP1/RFC4/RNF168/NSD2/CCNA2/RAD1/H4C3/H2BC9/RPA3/RFC2/RMI1/H2AX/CHEK1/RHNO1/CDK2/TIMELESS/RFC3/BLM/RMI2/BRCA1/BRIP1/RBBP8","RPA2/H2AC20/PARP1/POLE4/MBD4/RFC4/RFC1/H2AZ1/H4C3/H2BC9/RPA3/POLD2/H2AZ2/RFC2/POLE3/FEN1/POLD3/H2AX/UNG/RFC3/PARP2/XRCC1/LIG1/PNKP/PCNA","RPA2/RBBP4/SMYD2/EXO1/BARD1/TOPBP1/RFC4/CCNA2/ING2/RAD1/RPA3/RFC2/CDKN2A/RMI1/CDK1/SSRP1/CHEK1/RHNO1/CDK2/MDM2/RFC3/SUPT16H/BLM/RMI2/BRD7/AURKB/TAF15/BRCA1/BRIP1/RBBP8/PIN1/TPX2/AURKA/RBBP7","RPA2/H2AC20/POLE4/MCM6/PSMD1/MCM2/H2AZ1/GMNN/H3C2/H4C3/H3C7/H2BC9/PSMB9/MCM3/RPA3/H2AZ2/DBF4/MCM7/MCM4/POLE3/MCM10/PSMC3/H2AX/CDK2/PRIM1/ORC6/CDT1/CDC6/KPNB1/UBE2S/MCM8/UBE2C/MCM5/POLA1","HNRNPR/SRSF10/SRRM1/SRSF4/SNRNP40/DHX9/HNRNPU/SRSF7/SNRPG/SNRNP200/HNRNPA3/LSM3/TRA2B/HNRNPD/TCERG1/HNRNPH1/LSM2/SRSF3/HNRNPA2B1/LSM5/SRRT/LSM8/PRPF4/RBM17/HNRNPF/HSPA8/MAGOHB/HNRNPA1/RNPS1/FUS/SRSF2/ALYREF/SNRPD1/PTBP1/LSM4/SNRPA/SNRPB/UPF3B/RBMX/FAM50A","RPA2/PARP1/POLE4/RFC4/RFC1/RPA3/POLD2/RFC2/POLE3/FEN1/POLD3/RFC3/PARP2/LIG1/PCNA","SFN/RPA2/EXO1/BARD1/TOPBP1/RFC4/RNF168/NSD2/RAD1/CCNB1/H4C3/H2BC9/RPA3/RFC2/RMI1/CDK1/WEE1/H2AX/CHEK1/RHNO1/RFC3/BLM/RMI2/BRCA1/BRIP1/RBBP8/YWHAH","RPA2/EXO1/BARD1/TOPBP1/RFC4/CCNA2/RAD1/RPA3/RFC2/RMI1/SSRP1/CHEK1/RHNO1/CDK2/MDM2/RFC3/SUPT16H/BLM/RMI2/AURKB/TAF15/BRCA1/BRIP1/RBBP8/PIN1/TPX2/AURKA","RPA2/EXO1/MSH2/MSH6/BARD1/TOPBP1/RFC4/RAD1/RPA3/RFC2/RMI1/RHNO1/RAD51AP1/BRCA2/RFC3/BLM/RMI2/BRCA1/RAD51C/BRIP1/RBBP8","MRPL20/DDOST/EIF3I/MRPS15/MRPS21/SSR2/MRPL24/MRPS14/MRPL33/RPSA/RPL29/SPCS1/MRPL47/PPA2/MRPS36/MRPL22/MRPS18A/RPS12/SEC61G/MRPS33/EIF4EBP1/MRPL15/MRPL13/EEF1D/SEC61B/MRPL41/PPA1/CHCHD1/RPS24/MRPL43/RPS13/TRMT112/MRPL11/MRPL21/RPS26/MRPL57/MRPL52/SRP14/RPS27L/MRPL28/MRPS34/TUFM/MRPL27/MRPS23/MRPL58/MRPS7/MRPL12/SEC11C/MRPL54/EIF3G/GADD45GIP1/EIF3K/MRPS12/RPN2/MRPS6/MRPL40/SSR4","SFN/RPA2/RBBP4/JUN/SMYD2/EXO1/MSH2/BARD1/FANCD2/TOPBP1/TFDP2/RFC4/CCNA2/ING2/RAD1/CCNB1/NELFE/CDKN1A/RPA3/RFC2/RHEB/CCNE2/AGO2/CDKN2A/RMI1/CDK1/SSRP1/CHEK1/RHNO1/CDK2/MDM2/E2F7/CENPJ/RFC3/TFDP1/SUPT16H/FOS/FANCI/BLM/RMI2/BRD7/AURKB/TAF15/BRCA1/BRIP1/BIRC5/RBBP8/PIN1/PRDX2/BAX/PCNA/TPX2/E2F1/AURKA/YWHAH/RBBP7","RPA2/EXO1/BARD1/TOPBP1/RFC4/RAD1/RPA3/RFC2/RMI1/CHEK1/RHNO1/RAD51AP1/BRCA2/RFC3/BLM/RMI2/BRCA1/RAD51C/BRIP1/RBBP8","RPA2/PARP1/POLE4/MBD4/RFC4/RFC1/RPA3/POLD2/RFC2/POLE3/FEN1/POLD3/UNG/RFC3/PARP2/XRCC1/LIG1/PNKP/PCNA","PSMB2/PRDX1/PRDX6/COX20/COX5B/TKT/COX7C/SKP1/ATOX1/CSNK2B/PSMB8/PSMB1/NDUFA4/CYCS/BLVRA/SEM1/CDKN2A/TXN/PSMB7/SURF1/TALDO1/PSMA1/PSMC3/COX8A/PRDX5/COX14/COX6A1/UBC/PSMB5/PSME1/PSME2/PSMC1/COX5A/NQO1/CYBA/PSMB6/UBB/PSMB3/PSMC5/PRDX2/COX6B1/PSMC4/BLVRB/HM13/RBX1/COX7B/G6PD","MICOS10/ATP5PB/MTX1/ATP5MC3/CHCHD6/ATP5ME/ATP5MF/ATP5F1C/ATP5MG/ATP5MC2/ATP5F1B/ATP5MC1/ATP5PD/ATP5F1A/ATP5F1D/MICOS13/ATP5F1E/ATP5PF/ATP5PO","HNRNPR/SRSF10/SRRM1/SRSF4/SNRNP40/DHX9/TPR/HNRNPU/SRSF7/SNRPG/SNRNP200/HNRNPA3/LSM3/TRA2B/SLBP/HNRNPD/NUP155/TCERG1/THOC3/HNRNPH1/LSM2/SRSF3/WTAP/HNRNPA2B1/LSM5/SRRT/LSM8/PRPF4/RBM17/HNRNPF/HSPA8/MAGOHB/HNRNPA1/NUP107/NUP58/RNPS1/FUS/SRSF2/ALYREF/SNRPD1/PTBP1/DDX39A/LSM4/SNRPA/NUP62/SNRPB/NXT1/UPF3B/THOC2/RBMX/FAM50A","MRPL20/MRPS15/MRPS21/MRPL24/MRPS14/MRPL33/MRPL47/MRPS36/MRPL22/MRPS18A/MRPS33/MRPL15/MRPL13/MRPL41/CHCHD1/MRPL43/MRPL11/MRPL21/MRPL57/MRPL52/MRPL28/MRPS34/TUFM/MRPL27/MRPS23/MRPL58/MRPS7/MRPL12/MRPL54/GADD45GIP1/MRPS12/MRPS6/MRPL40","SFN/ITGB3BP/H2AC20/S100A9/PMF1/IQGAP3/NUF2/CENPL/KIF14/CENPF/AHCTF1/ROCK2/CENPA/XPO1/SPC25/SGO2/SGO1/H2AZ1/CENPE/MAD2L1/CENPU/KIF2A/CENPK/CENPH/DIAPH1/SPDL1/H3C2/H4C3/H3C7/H2BC9/CENPQ/H2AZ2/TUBB4B/ZWINT/BUB3/CKAP5/H2AX/TUBA1B/TUBA1C/NUP107/DIAPH3/KNL1/ZWILCH/PRC1/CENPN/CYBA/AURKB/SKA2/BIRC5/NDC80/PIN1/MAPRE1/DSN1/YWHAH/CENPM","ATP5PB/ATP5MC3/ATP5ME/ATP5MF/ATP5F1C/ATP5MG/ATP5MC2/ATP5F1B/ATP5MC1/ATP5PD/ATP5F1A/ATP5F1D/ATP5F1E/ATP5PF/ATP5PO","RPA2/CLSPN/MCM6/MCM2/RFC4/RAD1/MCM3/RPA3/RFC2/DBF4/MCM7/MCM4/MCM10/CHEK1/CDK2/RFC3/ORC6/CDC6/MCM8/MCM5","DDOST/S100A11/S100A9/LAMTOR2/NCSTN/PRDX6/QSOX1/DEGS1/VAMP8/NIT2/CD47/S100P/CYSTM1/TUBB/CSNK2B/HSPA1A/HSPA1B/NEU1/HEBP2/FUCA2/DYNLT1/RNASET2/PSMB1/DBNL/PPIA/ASAH1/SDCBP/GSDMD/NAPRT/ERP44/GSN/STOM/PSMB7/PTGES2/AGPAT2/DPP7/TUBB4B/COMMD3/PGAM1/CTSD/CD59/CD44/PSMC3/FTH1/TMEM179B/CAPN1/LAMTOR1/NDUFC2/HSPA8/MGST1/CD63/DYNLL1/LAMP1/LGALS3/NPC2/SRP14/B2M/ANXA2/PKM/CTSH/ALDOA/PYCARD/COTL1/CYBA/APRT/TRAPPC1/RAB5C/GRN/CFD/BST2/FTL/A1BG/CST3/SLPI/CTSA/CTSZ/PDXK/CSTB/MIF/CYB5R3/PRDX4","RBBP4/NEK2/CENPF/XPO1/MZT2B/MZT2A/PSMD1/LIN54/CCNA2/CCNB1/HMMR/TUBB/PSMB9/CDKN1A/PCM1/CEP78/CDK5RAP2/TUBB4B/CDK1/WEE1/CKAP5/PSMC3/CEP57/FOXM1/TUBA1B/TUBA1C/CDK2/CENPJ/MZT1/HAUS2/CEP152/CCNB2/PKMYT1/CCP110/TUBG1/HAUS1/HAUS5/CDC25B/TPX2/MAPRE1/E2F1/MYBL2/AURKA/GTSE1","RPA2/EXO1/GEN1/POLE4/BARD1/TOPBP1/RFC4/RFC1/RAD1/RPA3/POLD2/RFC2/RMI1/POLE3/POLD3/CHEK1/RHNO1/RAD51AP1/BRCA2/RFC3/BLM/RMI2/BRCA1/RAD51C/BRIP1/RBBP8/PCNA","RPA2/POLE4/MCM6/MCM2/GMNN/MCM3/RPA3/DBF4/MCM7/MCM4/POLE3/MCM10/CDK2/PRIM1/ORC6/CDT1/CDC6/MCM8/MCM5/POLA1","ITGB3BP/PMF1/NUF2/CENPL/CENPF/AHCTF1/CENPA/XPO1/SPC25/SGO2/SGO1/CENPE/MAD2L1/CENPU/KIF2A/CENPK/CENPH/DIAPH1/SPDL1/CENPQ/TUBB4B/ZWINT/BUB3/CKAP5/TUBA1B/TUBA1C/NUP107/DIAPH3/KNL1/ZWILCH/CENPN/AURKB/SKA2/BIRC5/NDC80/MAPRE1/DSN1/CENPM","GPSM2/TPR/ASPM/NEK2/CENPF/HNRNPU/CENPA/LSM3/KIF15/CEP97/CENPE/KIF2A/CENPK/CENPH/TUBB/HSPA1A/HSPA1B/KIFC1/CENPW/FBXO5/ANLN/PRKDC/ASAP1/CDK5RAP2/KIF11/SMC3/SAC3D1/NUMA1/CEP295/RACGAP1/CDK2/CENPJ/MZT1/DLGAP5/HAUS2/CEP152/CCNB2/KIF23/MRPS11/PRC1/CCP110/CENPN/AURKB/SPAG5/BRCA1/KPNB1/BIRC5/CENPX/NDC80/HAUS1/LSM4/HAUS5/NUP62/TPX2/MAPRE1/MYBL2/AURKA/SMC1A/KIF4A","RPA2/POLE4/MCM6/PSMD1/MCM2/RFC4/RFC1/CCNA2/GMNN/PSMB9/MCM3/RPA3/POLD2/RFC2/MCM7/MCM4/CCNE2/POLE3/PSMC3/FEN1/POLD3/CDK2/PRIM1/RFC3/ORC6/GINS2/CDT1/CDC6/LIG1/UBE2S/PCNA/MCM8/GINS1/UBE2C/MCM5/POLA1","ITGB3BP/PMF1/NUF2/CENPL/CENPF/AHCTF1/CENPA/XPO1/SPC25/SGO2/SGO1/CENPE/MAD2L1/CENPU/KIF2A/CENPK/CENPH/SPDL1/CENPQ/ZWINT/BUB3/CKAP5/NUP107/KNL1/ZWILCH/CENPN/AURKB/SKA2/BIRC5/NDC80/UBE2S/MAPRE1/DSN1/UBE2C/CENPM","RPA2/RBBP4/ITGB3BP/H2AC20/CENPL/CENPA/RFC4/RFC1/H2AZ1/CCNA2/CENPU/CENPK/CENPH/H4C3/H2BC9/CENPQ/CENPW/RPA3/POLD2/H2AZ2/RFC2/DSCC1/FEN1/POLD3/H2AX/CDK2/PRIM1/RFC3/MIS18BP1/KNL1/BLM/CENPN/CENPX/LIG1/PCNA/MIS18A/CENPM/RBBP7/POLA1","ITGB3BP/PMF1/NUF2/CENPL/CENPF/AHCTF1/CENPA/XPO1/SPC25/SGO2/PSMD1/SGO1/CENPE/MAD2L1/CENPU/KIF2A/CENPK/CENPH/PTTG1/SPDL1/PSMB9/CENPQ/RAD21/TUBB4B/ZWINT/SMC3/BUB3/CKAP5/PSMC3/CDCA5/TUBA1B/TUBA1C/NUP107/PDS5B/KNL1/ZWILCH/CENPN/AURKB/SKA2/BIRC5/NDC80/UBE2S/MAPRE1/DSN1/UBE2C/CENPM/SMC1A","RPA2/H2AC20/POLE4/MCM6/PSMD1/MCM2/RFC4/RFC1/H2AZ1/CCNA2/GMNN/H3C2/H4C3/H3C7/H2BC9/PSMB9/MCM3/RPA3/POLD2/H2AZ2/RFC2/DBF4/MCM7/MCM4/CCNE2/POLE3/MCM10/PSMC3/FEN1/POLD3/H2AX/CDK2/PRIM1/RFC3/ORC6/GINS2/CDT1/CDC6/KPNB1/LIG1/UBE2S/PCNA/MCM8/GINS1/UBE2C/MCM5/POLA1","RPA2/CLSPN/PARP1/EXO1/GEN1/POLE4/BARD1/POLQ/TOPBP1/RFC4/RNF168/NSD2/RFC1/CCNA2/RAD1/H4C3/H2BC9/RPA3/POLD2/RFC2/RMI1/POLE3/FEN1/POLD3/H2AX/CHEK1/RHNO1/RAD51AP1/CDK2/TIMELESS/BRCA2/RFC3/PARP2/BLM/RMI2/BRCA1/RAD51C/BRIP1/RBBP8/XRCC1/PCNA","RPA2/MCM6/MCM2/RFC4/RFC1/MCM3/RPA3/POLD2/RFC2/MCM7/MCM4/FEN1/POLD3/PRIM1/RFC3/GINS2/LIG1/PCNA/MCM8/GINS1/MCM5/POLA1","RPA2/CLSPN/PARP1/EXO1/GEN1/POLE4/RIF1/BARD1/POLQ/TOPBP1/RFC4/RNF168/NSD2/RFC1/CCNA2/RAD1/H4C3/H2BC9/RPA3/POLD2/BAZ1B/RFC2/PRKDC/RMI1/POLE3/DCLRE1C/FEN1/POLD3/H2AX/CHEK1/RHNO1/RAD51AP1/CDK2/TIMELESS/BRCA2/RFC3/PARP2/BLM/RMI2/BRCA1/RAD51C/BRIP1/KPNA2/RBBP8/XRCC1/PCNA","PARK7/ATP6V0B/UQCRH/ATP5PB/ATP1B1/COX5B/ATP5MC3/COX17/ATP5ME/ATP6V0E1/NDUFA4/ATP5MF/ATP6V1F/ATP6V0E2/CYC1/MFSD3/SURF1/ATP5F1C/COX8A/UCP2/ATP5MG/PHB2/ATP5MC2/ATP5F1B/COX6A1/COX5A/ANTKMT/ATP6V0D1/ATP5MC1/ATP5PD/ATP5F1A/CYB5A/ATP5F1D/NDUFS7/UQCRFS1/COX6B1/ATP5F1E/ATP5PF/ATP5PO/UQCR10/TMSB4X/COX7B/SLC25A5/ATP6AP1","SFN/RPA2/CLSPN/EXO1/MCM6/BARD1/PSMD1/MCM2/TOPBP1/RFC4/RNF168/NSD2/RAD1/CCNB1/H4C3/H2BC9/PSMB9/MCM3/RPA3/RFC2/DBF4/MCM7/MCM4/RMI1/MCM10/CDK1/WEE1/PSMC3/H2AX/CHEK1/RHNO1/CDK2/RFC3/CCNB2/BLM/PKMYT1/RMI2/ORC6/CDC6/BRCA1/BRIP1/RBBP8/MCM8/YWHAH/MCM5/GTSE1","ISG15/RPA2/CLSPN/USP1/H2AC20/UBE2T/DTL/PARP1/EXO1/GEN1/MSH2/MSH6/FANCL/POLE4/RIF1/BARD1/RAD18/FANCD2/POLQ/MBD4/TOPBP1/ACTL6A/RFC4/RNF168/NSD2/RFC1/H2AZ1/CCNA2/RAD1/H4C3/H2BC9/RPA3/POLD2/H2AZ2/BAZ1B/RFC2/COPS6/PRKDC/RMI1/POLE3/DCLRE1C/FEN1/POLD3/H2AX/CHEK1/RHNO1/RAD51AP1/CDK2/TIMELESS/UNG/BRCA2/RFC3/PARP2/PCLAF/FANCI/BLM/RMI2/BRCA1/RAD51C/BRIP1/KPNA2/CENPX/RBBP8/RAD23A/XRCC1/LIG1/PNKP/PCNA","ITGB3BP/PMF1/NUF2/CENPL/CENPF/AHCTF1/CENPA/XPO1/SPC25/SGO2/SGO1/CENPE/MAD2L1/CENPU/KIF2A/CENPK/CCNB1/CENPH/SPDL1/CENPQ/RAD21/TUBB4B/ZWINT/CDK1/SMC3/BUB3/CKAP5/CDCA5/TUBA1B/TUBA1C/NUP107/PDS5B/KNL1/CCNB2/ZWILCH/CENPN/AURKB/SKA2/BIRC5/NDC80/MAPRE1/DSN1/CENPM/SMC1A","NDUFS5/NDUFS2/NDUFB3/NDUFAF3/TIMMDC1/NDUFB5/NDUFC1/NDUFS6/NDUFS4/NDUFA2/NDUFB2/NDUFB9/NDUFA8/NDUFB8/NDUFS3/NDUFV1/NDUFS8/NDUFC2/NDUFA9/NDUFA12/NDUFB1/NDUFB10/NDUFAB1/NDUFV2/NDUFS7/NDUFA11/NDUFB7/NDUFA13/NDUFA3/NDUFV3/NDUFA6/NDUFB11/NDUFA1","RPA2/RBBP4/CDKN2C/CKS1B/RRM2/POLE4/MCM6/PSMD1/MCM2/TFDP2/LIN54/CCNA2/CCNB1/DHFR/GMNN/PSMB9/CDKN1A/CCND3/MCM3/FBXO5/RPA3/DBF4/MCM7/MCM4/CCNE2/CDKN2A/POLE3/MCM10/CDK1/WEE1/PSMC3/CDK2/PRIM1/CDK4/TFDP1/ORC6/CDT1/CDC6/TOP2A/TK1/TYMS/CDKN2D/PCNA/MCM8/E2F1/MYBL2/DYRK1A/MCM5/POLA1","NDUFS5/NDUFS2/COX20/NDUFB3/NDUFAF3/COX17/NDUFB5/NDUFC1/NDUFS4/NDUFA2/UQCC2/STMP1/NDUFB2/NDUFB9/NDUFA8/SURF1/NDUFB8/NDUFS3/UQCC3/NDUFS8/COA4/NDUFC2/NDUFA9/COX14/NDUFA12/NDUFB1/NDUFB10/NDUFAB1/COA3/NDUFAF8/NDUFS7/NDUFA11/PET100/NDUFB7/NDUFA13/UQCRFS1/NDUFA3/NDUFA6/NDUFB11/NDUFA1","ITGB3BP/PMF1/NUF2/CENPL/CENPF/LBR/AHCTF1/CENPA/XPO1/SPC25/SGO2/PSMD1/SGO1/CENPE/MAD2L1/CENPU/NUP155/KIF2A/CENPK/CCNB1/CENPH/LMNB1/PTTG1/SPDL1/PSMB9/CENPQ/FBXO5/RAD21/TUBB4B/ZWINT/CDK1/SMC3/BUB3/CKAP5/PSMC3/CDCA5/TUBA1B/TUBA1C/NUP107/TMPO/RAN/NUP58/PDS5B/VRK1/KNL1/CCNB2/ZWILCH/CENPN/AURKB/KPNB1/SKA2/BIRC5/NDC80/NUP62/UBE2S/MAPRE1/DSN1/UBE2C/CENPM/SMC1A/EMD","RPA2/RBBP4/CKS1B/POLE4/MCM6/PSMD1/MCM2/TFDP2/RFC4/RFC1/LIN54/CCNA2/GMNN/PSMB9/CDKN1A/MCM3/RPA3/POLD2/RFC2/MCM7/ESCO2/MCM4/CCNE2/RAD21/POLE3/SMC3/WEE1/PSMC3/FEN1/CDCA5/POLD3/CDK2/PRIM1/CDK4/PDS5B/RFC3/TFDP1/ORC6/GINS2/CDT1/CDC6/LIG1/UBE2S/CDC25B/PCNA/MCM8/GINS1/E2F1/UBE2C/MCM5/POLA1/SMC1A","ITGB3BP/PMF1/NUF2/CENPL/NEK2/CENPF/AHCTF1/CENPA/XPO1/NCAPH/MZT2B/MZT2A/SPC25/SGO2/SGO1/SMC4/NCAPG/CENPE/MAD2L1/CENPU/KIF2A/CENPK/CCNB1/CENPH/SPDL1/TUBB/CENPQ/PCM1/RAD21/CEP78/SMC2/CDK5RAP2/TUBB4B/ZWINT/CDK1/SMC3/BUB3/CKAP5/CDCA5/NUMA1/CEP57/NCAPD2/TUBA1B/TUBA1C/NUP107/CENPJ/PDS5B/MZT1/KNL1/HAUS2/CEP152/CCNB2/ZWILCH/CCP110/CENPN/AURKB/TUBG1/SKA2/BIRC5/NDC80/HAUS1/HAUS5/MAPRE1/DSN1/CENPM/SMC1A","ITGB3BP/H2AC20/PMF1/NUF2/CENPL/TPR/NEK2/CENPF/LBR/AHCTF1/CENPA/XPO1/NCAPH/MZT2B/MZT2A/SPC25/SGO2/PSMD1/SGO1/SMC4/NCAPG/H2AZ1/CENPE/MAD2L1/CENPU/NUP155/KIF2A/CENPK/CCNB1/CENPH/LMNB1/PTTG1/SPDL1/H3C2/H4C3/H3C7/H2BC9/TUBB/PSMB9/CENPQ/FBXO5/H2AZ2/PCM1/RAD21/CEP78/SMC2/CDK5RAP2/TUBB4B/MASTL/ZWINT/CDK1/SMC3/BUB3/CKAP5/PSMC3/CDCA5/NUMA1/CEP57/H2AX/NCAPD3/NCAPD2/TUBA1B/TUBA1C/NUP107/TMPO/RAN/CENPJ/NUP58/PDS5B/MZT1/VRK1/KNL1/HAUS2/CEP152/CCNB2/ZWILCH/KIF23/CCP110/CENPN/AURKB/TUBG1/KPNB1/SKA2/BIRC5/NDC80/HAUS1/HAUS5/NUP62/UBE2S/MAPRE1/DSN1/UBE2C/CENPM/SMC1A/EMD","NDUFS5/UQCRH/NDUFS2/SDHC/COX20/COX5B/NDUFB3/NDUFAF3/TIMMDC1/NDUFB5/NDUFC1/NDUFS6/NDUFS4/COX7C/UQCRQ/NDUFA2/NDUFA4/CYCS/NDUFB2/UQCRB/NDUFB9/CYC1/NDUFA8/SURF1/NDUFB8/NDUFS3/COX8A/NDUFV1/NDUFS8/NDUFC2/NDUFA9/COX14/NDUFA12/COX6A1/NDUFB1/COX5A/ETFA/NDUFB10/UQCRC2/NDUFAB1/NDUFV2/NDUFS7/UQCR11/NDUFA11/NDUFB7/NDUFA13/UQCRFS1/COX6B1/ETFB/NDUFA3/NDUFV3/UQCR10/NDUFA6/NDUFB11/COX7B/NDUFA1","SFN/RPA2/CLSPN/ITGB3BP/PMF1/NUF2/CENPL/CENPF/EXO1/AHCTF1/CENPA/XPO1/MCM6/SPC25/SGO2/BARD1/PSMD1/SGO1/MCM2/TOPBP1/RFC4/RNF168/NSD2/CENPE/MAD2L1/CCNA2/CENPU/RAD1/KIF2A/CENPK/CCNB1/CENPH/SPDL1/H4C3/H2BC9/PSMB9/CDKN1A/CENPQ/MCM3/RPA3/RFC2/DBF4/MCM7/MCM4/CCNE2/CDKN2A/RMI1/MCM10/ZWINT/CDK1/BUB3/WEE1/CKAP5/PSMC3/H2AX/CHEK1/RHNO1/CDK2/NUP107/MDM2/RFC3/KNL1/CCNB2/ZWILCH/BLM/PKMYT1/RMI2/ORC6/CENPN/AURKB/CDC6/BRCA1/SKA2/BRIP1/BIRC5/NDC80/RBBP8/UBE2S/MCM8/MAPRE1/DSN1/UBE2C/YWHAH/MCM5/CENPM/GTSE1","NDUFS5/UQCRH/ATP5PB/NDUFS2/SDHC/MPC2/COX20/COX5B/ATP5MC3/NDUFB3/NDUFAF3/TIMMDC1/NDUFB5/ATP5ME/NDUFC1/NDUFS6/NDUFS4/COX7C/UQCRQ/NDUFA2/NDUFA4/CYCS/MDH2/ATP5MF/NDUFB2/UQCRB/NDUFB9/CYC1/NDUFA8/SURF1/ATP5F1C/NDUFB8/LDHA/NDUFS3/COX8A/NDUFV1/NDUFS8/UCP2/NDUFC2/ATP5MG/NDUFA9/COX14/ATP5MC2/ATP5F1B/NDUFA12/COX6A1/NDUFB1/COX5A/ETFA/IDH2/NDUFB10/UQCRC2/NDUFAB1/ATP5MC1/ATP5PD/SLC16A3/NDUFV2/ATP5F1A/BSG/ATP5F1D/NDUFS7/UQCR11/NDUFA11/NDUFB7/NDUFA13/UQCRFS1/COX6B1/ETFB/NDUFA3/ATP5F1E/ATP5PF/ATP5PO/NDUFV3/UQCR10/NDUFA6/NDUFB11/COX7B/NDUFA1","ARID1A/NUF2/TPR/ASPM/KIF14/NEK2/CENPF/HNRNPU/GEN1/NCAPH/SPC25/SGO2/FANCD2/SGO1/KIF15/CEP97/SMC4/ACTL6A/TACC3/NCAPG/CENPE/MAD2L1/ING2/TRIP13/RAD1/KIF2A/CENPK/CCNB1/SLF1/MACROH2A1/PTTG1/SPDL1/DUSP1/HSPA1A/HSPA1B/KIFC1/PHIP/FBXO5/ANLN/FIGNL1/BAZ1B/CDCA2/ESCO2/RAD54B/CCNE2/RAD21/DSCC1/NSMCE2/RMI1/CKS2/SMC2/CDK5RAP2/MASTL/ZWINT/KIF20B/KIF11/SMC3/BUB3/MKI67/CDCA5/NUMA1/CHEK1/NCAPD3/RAD51AP1/NCAPD2/EPS8/RACGAP1/SMARCC2/RAN/BRCA2/PDS5B/MZT1/CHAMP1/DLGAP5/NUSAP1/CCNB2/ZWILCH/KIF23/PRC1/PKMYT1/KIF22/BCL7C/BRD7/CDT1/AURKB/SPAG5/CDC6/TOP2A/PSMC3IP/TUBG1/KPNB1/RAD51C/BRIP1/SPHK1/BIRC5/CENPX/NDC80/PSMG2/PIN1/NUP62/UBE2S/CDC25B/TPX2/MAPRE1/DSN1/MYBL2/UBE2C/AURKA/BMP7/SYCP2/RANBP1/SMC1A/KIF4A","PARK7/NDUFS5/UQCRH/ATP5PB/NDUFS2/SDHC/COA6/COX5B/NDUFB3/NDUFB5/ATP5ME/NDUFC1/NDUFS6/NDUFS4/COX7C/UQCRQ/NDUFA2/UQCC2/COX7A2/NDUFA4/CYCS/ATP5MF/NDUFB2/UQCRB/NDUFB9/CYC1/STOML2/NDUFA8/ATP5F1C/NDUFB8/NDUFS3/UQCC3/COX8A/NDUFV1/NDUFS8/NDUFC2/ATP5MG/NDUFA9/ATP5F1B/NDUFA12/COX6A1/DNAJC15/NDUFB1/COX5A/ANTKMT/NDUFB10/UQCRC2/NDUFAB1/ATP5PD/NDUFV2/ATP5F1A/ATP5F1D/NDUFS7/UQCR11/NDUFA11/NDUFB7/NDUFA13/UQCRFS1/COX6B1/NDUFA3/ATP5F1E/ATP5PF/ATP5PO/NDUFV3/CHCHD10/UQCR10/NDUFA6/NDUFB11/COX7B/NDUFA1","ARID1A/SFPQ/PMF1/NUF2/TPR/KIF14/NEK2/CENPF/HNRNPU/SMC6/GEN1/NCAPH/SPC25/SGO2/RAD18/FANCD2/SGO1/PLSCR1/SMC4/ECT2/ACTL6A/TACC3/NCAPG/CENPE/MAD2L1/TRIP13/CENPK/CCNB1/CENPH/SLF1/MACROH2A1/PTTG1/SPDL1/DUSP1/KIFC1/SRPK1/CENPQ/CENPW/FBXO5/BAZ1B/CDCA2/ESCO2/CCNE2/RAD21/DSCC1/NSMCE2/RMI1/SMC2/CDK5RAP2/ZWINT/KIF11/SMC3/BUB3/MKI67/FEN1/CDCA5/NUMA1/NCAPD3/NCAPD2/DDX11/RACGAP1/SMARCC2/RAN/SKA3/PDS5B/CHAMP1/DLGAP5/KNL1/NUSAP1/ZWILCH/KIF23/PRC1/RMI2/KIF22/BCL7C/BRD7/CTCF/CENPN/CDT1/AURKB/SPAG5/CDC6/TOP2A/PSMC3IP/TUBG1/BRCA1/KPNB1/RAD51C/SKA2/BRIP1/BIRC5/CENPX/NDC80/PSMG2/NUP62/MAPRE1/DSN1/UBE2C/SYCP2/MIS18A/SMC1B/SMC1A/KIF4A/SLC25A5","NDUFS5/UQCRH/ATP5PB/NDUFS2/SDHC/COX20/COX5B/ATP5MC3/NDUFB3/NDUFAF3/TIMMDC1/NDUFB5/ATP5ME/NDUFC1/NDUFS6/NDUFS4/COX7C/UQCRQ/NDUFA2/NDUFA4/CYCS/ATP5MF/NDUFB2/UQCRB/NDUFB9/CYC1/NDUFA8/SURF1/ATP5F1C/NDUFB8/NDUFS3/COX8A/NDUFV1/NDUFS8/UCP2/NDUFC2/ATP5MG/NDUFA9/COX14/ATP5MC2/ATP5F1B/NDUFA12/COX6A1/NDUFB1/COX5A/ETFA/NDUFB10/UQCRC2/NDUFAB1/ATP5MC1/ATP5PD/NDUFV2/ATP5F1A/ATP5F1D/NDUFS7/UQCR11/NDUFA11/NDUFB7/NDUFA13/UQCRFS1/COX6B1/ETFB/NDUFA3/ATP5F1E/ATP5PF/ATP5PO/NDUFV3/UQCR10/NDUFA6/NDUFB11/COX7B/NDUFA1"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>cluster<\/th>\n      <th>ID<\/th>\n      <th>Description<\/th>\n      <th>GeneRatio<\/th>\n      <th>BgRatio<\/th>\n      <th>pvalue<\/th>\n      <th>p.adjust<\/th>\n      <th>Count<\/th>\n      <th>setSize<\/th>\n      <th>GeneProp<\/th>\n      <th>geneID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":9,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[5,6,7,8,9]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render"],"jsHooks":[]}</script>
</div>
<div id="number-of-gene-sets" class="section level4" number="8.1.2">
<h4><span class="header-section-number">8.1.2</span> Number of gene
sets</h4>
<pre class="r"><code>gse_df %&gt;% 
  filter(p.adjust &lt; p_thrs) %&gt;% 
  group_by(group) %&gt;% 
  summarise(n = n()) %&gt;% 
  ggplot(aes(x=group, y=n)) +
  geom_bar(stat=&quot;identity&quot;, color = &#39;black&#39;)+
  geom_text(aes(label=n), vjust=1.4, color=&quot;white&quot;, size=geom_text_size)+
  labs(
    x = &#39;Clusters&#39;,
    y = &#39;Number of gene sets&#39;,
    title = &#39;Number of significant gene sets by cluster&#39;,
    caption = paste(&quot;Adjusted P value &lt;&quot;, p_thrs)
    
  )</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/cluster-marker-gse-enricher-c2reactome_gobp-ngset-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-set-similarity-matrix" class="section level4"
number="8.1.3">
<h4><span class="header-section-number">8.1.3</span> Gene-set similarity
matrix</h4>
<p>Heatmap showing the pair-wise similarity matrix of top 20 gene sets
with an adjusted P value &lt; 0.01 in each gene group. Heatmap colors
represent the Jaccard similarity coefficient.</p>
<pre class="r"><code>col_fun &lt;- colorRamp2(seq(0, 1, length.out = 4), brewer.pal(4, &quot;GnBu&quot;))
n_split &lt;- 10

use_mat &lt;- jc_matrix
colnames(use_mat) &lt;- colnames(use_mat) %&gt;% clean_msigdb_names %&gt;% tolower() %&gt;% str_to_title %&gt;% ep_str_wrap(70)
rownames(use_mat) &lt;- rownames(use_mat) %&gt;% clean_msigdb_names %&gt;% tolower() %&gt;% str_to_title %&gt;% ep_str_wrap(70)

ha_top &lt;- HeatmapAnnotation(
  foo = anno_block(
    # labels = c(&quot;Translation&quot;, &quot;Cell division&quot;),
    labels = 1:n_split,
    labels_gp = gpar(col = &quot;black&quot;, fontsize = 8),
    gp = gpar(lwd = 0, lty = 0))
  )

ht &lt;- Heatmap(
  use_mat,
  name = &#39;Jaccard index&#39;,
  column_split = n_split,
  row_split = n_split,
  column_title = NULL,
  row_title = NULL,
  col = col_fun,
  show_column_dend = FALSE,
  show_column_names = FALSE,
  border = TRUE,
  top_annotation = ha_top,
  heatmap_legend_param = list(title_gp = gpar(fontsize = 8),labels_gp = gpar(fontsize = 8)),
  width = unit(7, &quot;cm&quot;))

ht = draw(ht)</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/cluster-marker-gse-enricher-c2reactome_gobp-jcmatrix-1.png" width="4320" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;\n\n&quot;)</code></pre>
<pre class="r"><code># Extract term order according to hclust from similarity heatmap
row_order &lt;- row_order(ht) %&gt;% set_names(1:n_split)
term_similarity_clusters &lt;- foreach(i=names(row_order), .combine = rbind) %do% {
  data.frame(
    jc_sim_cl = i, 
    row_order = row_order[[i]], 
    Description =  rownames(jc_matrix)[row_order[[i]]]
  )
}

# Add similarity order to data for dot plot
res_top %&lt;&gt;% left_join(term_similarity_clusters)</code></pre>
</div>
<div id="dotplots" class="section level4" number="8.1.4">
<h4><span class="header-section-number">8.1.4</span> Dotplots</h4>
<p>The plots below show the top 20 gene sets with an adjusted P value
&lt; 0.01 in each gene group</p>
<pre class="r"><code>res_top %&gt;%
  # Combine similarity clusters
  mutate(
    # jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;5&#39;, &#39;10&#39;), &#39;5-10&#39;, jc_sim_cl),
    # jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;2&#39;, &#39;4&#39;), &#39;2-4&#39;, jc_sim_cl),
    # jc_sim_cl = factor(
    #   jc_sim_cl,
    #   levels = c(&#39;1&#39;, &#39;3&#39;, &#39;5-10&#39;, &#39;6&#39;, &#39;8&#39;, &#39;9&#39;, &#39;2-4&#39;, &#39;7&#39;)
    #   ),

    p.adjust = ifelse(p.adjust &gt; 0.05, 1, p.adjust),
    p.adjust = -log10( p.adjust),
    p.adjust = ifelse(p.adjust &gt;= 10, 10, p.adjust),
  ) %&gt;% 
  ggplot(aes(group, Description, color = GeneProp, size = p.adjust)) +
  geom_point() +
  scale_color_distiller(palette = &quot;Purples&quot;, na.value = &quot;grey70&quot;, direction = 1) +
  scale_y_discrete(labels = default_labeller(70)) +
  scale_x_discrete(position = &quot;top&quot;) +
  labs(y = &#39;&#39;, x = &#39;&#39;) +
  # facet_grid(rows = vars(cluster), scale = &#39;free&#39;, space = &#39;free&#39;) +
  # facet_grid(rows = vars(kcluster), scale = &#39;free&#39;, space = &#39;free&#39;) +
  facet_grid(rows = vars(jc_sim_cl), scale = &#39;free&#39;, space = &#39;free&#39;) +
  theme(
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, size = 1),
    # axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    axis.line.x = element_blank(),
    panel.grid.major.y = element_line(size = 0.2, colour=&quot;grey60&quot;, linetype = &#39;longdash&#39;),
    panel.grid.major.x = element_line(size = 0.2, colour=&quot;grey60&quot;, linetype = &#39;longdash&#39;),
    strip.background = element_blank()#,
    # strip.text.y = element_blank()
    )</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/cluster-marker-gse-enricher-c2reactome_gobp-dotplot-1.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="site-marker-genes" class="section level2" number="9">
<h2><span class="header-section-number">9</span> Site marker genes</h2>
<pre class="r"><code>marker_info &lt;- score_markers$site_markers

# Table of results
rmd_file &lt;- current_input()
if(is.null(rmd_file))
  rmd_file &lt;- &#39;tmp&#39;
file_xlsx &lt;- file.path(&#39;./docs/file&#39;,rmd_file, &#39;site-markers-scoremarkers-tables.xlsx&#39;)
dir.create(dirname(file_xlsx), recursive = TRUE, showWarnings = FALSE)</code></pre>
<pre class="r"><code>wb &lt;- createWorkbook()

for(use_site in names(marker_info)){
  if(nchar(use_site) &gt; 31)
    use_site &lt;- paste0(substr(use_site, 1, 30), &quot;.&quot;)
  addWorksheet(wb, use_site)
  writeData(wb, use_site, marker_info[[use_site]])
}
saveWorkbook(wb, file_xlsx, TRUE)</code></pre>
<p>The tables of results can be downloaded using the following <a
href="./file/br16-10x_rnaseq-cancer_cells-main.Rmd/site-markers-scoremarkers-tables.xlsx">link</a>
.</p>
<div id="top-10-markers-plot-1" class="section level3" number="9.1">
<h3><span class="header-section-number">9.1</span> Top-10 markers
plot</h3>
<p>Here we show the selection of top markers upregulated in each
site</p>
<pre class="r"><code>ntop &lt;- 10

# Dataframe of top markers
top_mk &lt;- foreach(i = names(marker_info), .combine = rbind) %do% {
  x &lt;- marker_info[[i]]
  ordered &lt;- x[order(x$mean.AUC, decreasing=TRUE),]
  data.frame(group = i, gene = rownames(ordered)[1:ntop])
}

# Matrix of top markers
top_mk_mat &lt;- top_mk %&gt;% 
  mutate(value = &#39;yes&#39;) %&gt;%
  pivot_wider(names_from = group, values_fill = &#39;no&#39;) %&gt;% 
  column_to_rownames(&#39;gene&#39;)

# Cluster matrix
x &lt;- top_mk_mat
x[x==&#39;no&#39;] &lt;- 0
x[x==&#39;yes&#39;] &lt;- 0
row_hclust &lt;- hclust(dist(x))
row_hclust_labels &lt;- row_hclust$labels[row_hclust$order]
col_hclust &lt;- hclust(dist(t(x)))
col_hclust_labels &lt;- col_hclust$labels[col_hclust$order]
top_mk_mat &lt;- top_mk_mat[row_hclust_labels, col_hclust_labels]

# Heatmap
ht_colors &lt;- c(yes = &#39;#E66B57&#39;, no = &#39;grey90&#39;)
Heatmap(
  top_mk_mat,
  name = &#39;top-marker&#39;,
  col = ht_colors, 
  # top_annotation = column_ha, 
  rect_gp = gpar(col= &quot;white&quot;)
)</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/site-marker-genes-heatmap-1.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
<div id="grouped-expression-heatmaps-by-site" class="section level3"
number="9.2">
<h3><span class="header-section-number">9.2</span> Grouped expression
heatmaps by site</h3>
<p>Here we show the expression heatmap for the top-10 markers
upregulated in each site</p>
<pre class="r"><code>ntop &lt;- 10

top_mk&lt;- foreach(x = marker_info) %do% {
  ordered &lt;- x[order(x$mean.AUC, decreasing=TRUE),]
  rownames(ordered)[1:ntop]
}

# Final plot
plotGroupedHeatmap(sce_comb,
                   features=unlist(top_mk) %&gt;% unique, 
                   group=&quot;site&quot;,
                   # annotation_col = col_annot,
                   # annotation_colors = ann_colors,
                   center=TRUE, 
                   zlim=c(-2, 2),
                   main = paste(&quot;Top&quot;, ntop,&quot;pairwise markers&quot;))</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/site-marker-genes-exprs-heatmap-1.png" width="4320" style="display: block; margin: auto;" /></p>
<p>Here we show the expression heatmap for the top-10 markers
upregulated in each site, but grouping samples by cluster and site.</p>
<pre class="r"><code>use_sce &lt;- sce_comb
use_sce$group &lt;- paste(sce_comb$cluster_mnn, sce_comb$site, sep = &#39;-&#39;)

# Final plot
plotGroupedHeatmap(use_sce,
                   features=unlist(top_mk) %&gt;% unique, 
                   group=&quot;group&quot;,
                   # annotation_col = col_annot,
                   # annotation_colors = ann_colors,
                   center=TRUE, 
                   zlim=c(-3, 3),
                   main = paste(&quot;Top&quot;, ntop,&quot;pairwise markers&quot;))</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/site-marker-genes-exprs-group_by_site_cluster-heatmap-1.png" width="6120" style="display: block; margin: auto;" /></p>
<pre class="r"><code>rm(use_sce)</code></pre>
</div>
<div id="umap-for-top-1-markers-for-each-cluster-1"
class="section level3" number="9.3">
<h3><span class="header-section-number">9.3</span> UMAP for top-1
markers for each cluster</h3>
<p>Here we remove mitochondrial and ribosomal genes</p>
<pre class="r"><code>marker_info_sorted_levels &lt;- names(marker_info) %&gt;% str_sort(numeric = TRUE)
top1_markers_umap &lt;- foreach(i = marker_info_sorted_levels) %do% {
  x &lt;- marker_info[[i]]
  ordered &lt;- x[order(x$mean.AUC, decreasing=TRUE),]
  top_mk &lt;- rownames(ordered)[1]
  plotReducedDim(sce_comb, &#39;UMAP_corrected&#39;, colour_by=top_mk, point_alpha = 0.7, point_size = 0.4) + 
      scale_colour_continuous_diverging() +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
      ggtitle (paste(i, &#39;  &#39;, top_mk)) +
      labs(
        title = paste(i, &#39;  &#39;, top_mk),
        x = &#39;UMAP 1&#39;,
        y = &#39;UMAP 2&#39;
      )
}

plot_grid(plotlist = top1_markers_umap)</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/site-marker-genes-umap-1.png" width="6120" style="display: block; margin: auto;" /></p>
</div>
<div id="expression-distribution" class="section level3" number="9.4">
<h3><span class="header-section-number">9.4</span> Expression
distribution</h3>
<p>Distribution of expression values for Top-1 genes by site</p>
<pre class="r"><code>top1_markers &lt;- foreach(i = marker_info_sorted_levels, .combine = c) %do% {
  x &lt;- marker_info[[i]]
  ordered &lt;- x[order(x$mean.AUC, decreasing=TRUE),]
  rownames(ordered)[1]
}

plotExpression(
    sce_comb, 
    x = &#39;site&#39;, 
    features =  top1_markers %&gt;% unique, 
    colour_by = &#39;site&#39;,
    show_median = TRUE, show_violin = TRUE,
    point_alpha = 0.08, point_size = 0.5
  )  +
  labs(
    x = &#39;&#39;
  ) +
  guides(color = &#39;none&#39;)</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/site-marker-genes-plotExpression-1.png" width="3120" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="site-markers-enrichment" class="section level2" number="10">
<h2><span class="header-section-number">10</span> Site markers
enrichment</h2>
<pre class="r"><code>gse_list &lt;- gse_site_auc</code></pre>
<!-- Write GSE results to xlsx file -->
<pre class="r"><code>gse_collections &lt;- list(
  enricher = &#39;msigdb.m2.cp&#39;,
  enrichGO = c(&quot;BP&quot;, &quot;MF&quot;)
)

# File name summary
rmd_file &lt;- current_input()
if(is.null(rmd_file))
  rmd_file &lt;- &#39;tmp&#39;

file_xlsx_list &lt;- list()
for(sr in names(gse_collections)) {
  file_xlsx_list[[sr]] &lt;- list()
  for(gc in gse_collections[[sr]]) {
    # Generate file
    file_xlsx &lt;- file.path(&#39;./docs/file&#39;,
                             rmd_file,
                             paste0(&#39;gse_&#39;, tolower(sr), &#39;_&#39;, tolower(gc), &#39;.xlsx&#39;)
                             )
    file_xlsx_list[[sr]][[gc]] &lt;- file_xlsx
    dir.create(dirname(file_xlsx), recursive = TRUE, showWarnings = FALSE)
    # Generate workbook
    wb &lt;- createWorkbook()
    for(i in names(gse_list)) {
      addWorksheet(wb, i)
      if(!is.null(gse_list[[i]][[sr]]$abs[[gc]])) {
        res &lt;- gse_list[[i]][[sr]]$abs[[gc]]@result
        writeData(wb, i, res)
      }
      }
      saveWorkbook(wb, file_xlsx, TRUE)
  }
}</code></pre>
<p>The tables of results can be downloaded using the following
links:</p>
<!-- Write links to download GSE results -->
<ul>
<li><p><a
href="./file/br16-10x_rnaseq-cancer_cells-main.Rmd/gse_enricher_msigdb.m2.cp.xlsx"><strong>Over
Representation analysis: msigdb.m2.cp</strong></a></p></li>
<li><p><a
href="./file/br16-10x_rnaseq-cancer_cells-main.Rmd/gse_enrichgo_bp.xlsx"><strong>Over
Representation analysis: GO:BP</strong></a></p></li>
<li><p><a
href="./file/br16-10x_rnaseq-cancer_cells-main.Rmd/gse_enrichgo_mf.xlsx"><strong>Over
Representation analysis: GO:MF</strong></a></p></li>
</ul>
<div id="ora---c2.cp.reactome-and-gobp-1" class="section level3"
number="10.1">
<h3><span class="header-section-number">10.1</span> ORA - C2.CP.REACTOME
and <a href="GO:BP" class="uri">GO:BP</a></h3>
<p>Over-enrichment analysis using genes with a mean AUC ≥ 0.25 in the
score markers results.</p>
<pre class="r"><code>gse_list &lt;- gse_site_auc
test_type &lt;- &#39;enricher&#39;
gset_collection &lt;- &#39;msigdb.c2.cp.reactome&#39;
direction &lt;- &#39;abs&#39;
p_thrs &lt;- 0.01
topn &lt;- 20

# Load all the results
gse_df &lt;- foreach(i = names(gse_list), .combine = rbind) %do% {
  i_1 &lt;- gse_list[[i]][[test_type]][[direction]][[gset_collection]]@result %&gt;% 
    mutate(group = i)
  i_2 &lt;- gse_list[[i]][[&#39;enrichGO&#39;]][[direction]][[&#39;BP&#39;]]@result %&gt;% 
    filter(simplify) %&gt;% 
    dplyr::select(-simplify) %&gt;% 
    mutate(group = i)
  rbind(i_1, i_2)
} %&gt;% 
  mutate(
    group = factor(group, c(&#39;primary_tumor&#39;, &#39;liver&#39;, &#39;lung&#39;)),
    Description = clean_msigdb_names(Description) %&gt;% tolower() %&gt;% str_to_title,
    Description = ifelse(grepl(&quot;^GO:&quot;, ID), paste0(&quot;GO:&quot;, Description), Description)
  ) 

# Select topn by group in selected gset
top_by_group &lt;- foreach(i = names(gse_list), .combine = c) %do% {
  gse_df %&gt;%
    filter(group == i) %&gt;%
    filter(p.adjust &lt; p_thrs) %&gt;%
    arrange(p.adjust) %&gt;% pull(Description) %&gt;%
    head(topn)
}

# Data frame to plot
res_top &lt;- gse_df %&gt;%
  filter(Description %in% top_by_group) %&gt;%
  mutate(log.p.adjust = -log10(p.adjust))

# Rows and column orders by hclust
use_mat &lt;- res_top %&gt;% 
  dplyr::select(Description, log.p.adjust, group) %&gt;% 
  pivot_wider(names_from = group, values_from = log.p.adjust) %&gt;% 
  column_to_rownames(&#39;Description&#39;)
use_mat[is.na(use_mat)] &lt;- 0

row_hclust &lt;- hclust(dist(use_mat))
row_hclust_labels &lt;- row_hclust$labels[row_hclust$order]

col_hclust &lt;- hclust(dist(t(use_mat)))
col_hclust_labels &lt;- col_hclust$labels[col_hclust$order]

use_mat &lt;- use_mat[row_hclust_labels,col_hclust_labels]

# GSET clusters by kmeans using silhouette
use_kmax &lt;- ifelse(nrow(use_mat) &lt;= 20, nrow(use_mat)-2, 20)
nb_clust &lt;- fviz_nbclust(use_mat, kmeans, method = &quot;silhouette&quot;, k.max = use_kmax)
silhouette_best_k &lt;- nb_clust$data %&gt;% slice_max(y) %&gt;% pull(clusters) %&gt;% as.character %&gt;% as.numeric
# silhouette_best_k &lt;- 5
kmeans_mat &lt;- kmeans(use_mat, silhouette_best_k, nstart = 25)
kmeans_mat_df &lt;- kmeans_mat$cluster %&gt;%
  data.frame %&gt;%
  set_names(&#39;kcluster&#39;) %&gt;%
  rownames_to_column(&#39;Description&#39;) %&gt;%
  mutate(kcluster = as.character(kcluster))


# Order factors by hclust
res_top %&lt;&gt;% 
  left_join(kmeans_mat_df) %&gt;% 
  mutate(
    Description = factor(Description, levels = row_hclust_labels),
    group = fct_relevel(group, col_hclust_labels)
  )

# Calculate jaccard similarity matrix
gset_collection &lt;- foreach(i = levels(res_top$Description)) %do% {
  res_top %&gt;% filter(Description == i) %&gt;% 
    # filter(p.adjust &lt; p_thrs) %&gt;%
    pull(geneID) %&gt;% 
    paste(collapse = &#39;/&#39;) %&gt;% 
    strsplit(&#39;/&#39;) %&gt;% unlist %&gt;% unique
}
names(gset_collection) &lt;- levels(res_top$Description)

use_terms &lt;- res_top$Description %&gt;% unique
jc_matrix &lt;- pairwise_terms_jc(gset_collection[use_terms])</code></pre>
<div id="results-table-1" class="section level4" number="10.1.1">
<h4><span class="header-section-number">10.1.1</span> Results table</h4>
<pre class="r"><code>gse_df %&gt;% 
  filter(p.adjust &lt; p_thrs) %&gt;% 
  arrange(desc(p.adjust)) %&gt;% 
  dplyr::select(group, ID:p.adjust, Count:GeneProp, geneID)  %&gt;%
  dplyr::rename(site = group) %&gt;% 
  datatable(., 
            rownames = FALSE, 
            filter = &#39;top&#39;, 
            caption = &#39;Results from over enrichment analysis&#39;,
            extensions = &#39;Buttons&#39;, 
            options = list(
              dom = &#39;Blfrtip&#39;,
              buttons = c(&#39;csv&#39;, &#39;excel&#39;)
            ))  %&gt;% 
  formatRound(c(&#39;GeneProp&#39;), digits = 2) %&gt;%
  formatSignif(c(&#39;pvalue&#39;, &#39;p.adjust&#39;), digits = 3)</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-91a99e85c6bc68af90f4" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-91a99e85c6bc68af90f4">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;primary_tumor&quot;,&quot;liver&quot;,&quot;lung&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"1e-15\" data-max=\"0.000288798294904\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"1.064e-12\" data-max=\"0.009925020369626\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"4\" data-max=\"28\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"10\" data-max=\"499\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.041162227602905\" data-max=\"0.533333333333334\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Results from over enrichment analysis<\/caption>","data":[["liver","lung","lung","lung","lung","lung","lung","lung","liver","liver","liver","lung","lung","lung","liver","lung","lung","lung","lung","primary_tumor","primary_tumor","liver","liver","liver","lung","primary_tumor","liver","lung","primary_tumor","primary_tumor","liver","primary_tumor","liver","liver","liver","primary_tumor","primary_tumor","primary_tumor","liver","primary_tumor","liver","liver","liver","liver","liver","liver","liver","liver","liver","liver","liver","lung","primary_tumor","primary_tumor","liver","liver","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor"],["REACTOME_TRANSCRIPTIONAL_ACTIVITY_OF_SMAD2_SMAD3_SMAD4_HETEROTRIMER","REACTOME_SENESCENCE_ASSOCIATED_SECRETORY_PHENOTYPE_SASP","REACTOME_BASE_EXCISION_REPAIR","REACTOME_HSF1_DEPENDENT_TRANSACTIVATION","REACTOME_RESOLUTION_OF_ABASIC_SITES_AP_SITES","REACTOME_TRANSLOCATION_OF_SLC2A4_GLUT4_TO_THE_PLASMA_MEMBRANE","REACTOME_GAP_FILLING_DNA_REPAIR_SYNTHESIS_AND_LIGATION_IN_GG_NER","REACTOME_RESOLUTION_OF_AP_SITES_VIA_THE_MULTIPLE_NUCLEOTIDE_PATCH_REPLACEMENT_PATHWAY","REACTOME_ESTROGEN_DEPENDENT_GENE_EXPRESSION","REACTOME_DOWNREGULATION_OF_SMAD2_3_SMAD4_TRANSCRIPTIONAL_ACTIVITY","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX2","REACTOME_CYCLIN_D_ASSOCIATED_EVENTS_IN_G1","REACTOME_PCNA_DEPENDENT_LONG_PATCH_BASE_EXCISION_REPAIR","REACTOME_NEUTROPHIL_DEGRANULATION","GO:0010631","REACTOME_ATTENUATION_PHASE","REACTOME_HSP90_CHAPERONE_CYCLE_FOR_STEROID_HORMONE_RECEPTORS_SHR_IN_THE_PRESENCE_OF_LIGAND","REACTOME_CELLULAR_SENESCENCE","REACTOME_HSF1_ACTIVATION","REACTOME_INTERFERON_ALPHA_BETA_SIGNALING","REACTOME_RUNX1_REGULATES_GENES_INVOLVED_IN_MEGAKARYOCYTE_DIFFERENTIATION_AND_PLATELET_FUNCTION","REACTOME_NEUTROPHIL_DEGRANULATION","REACTOME_CHOLESTEROL_BIOSYNTHESIS","REACTOME_ACTIVATION_OF_GENE_EXPRESSION_BY_SREBF_SREBP","GO:0000082","REACTOME_ESTROGEN_DEPENDENT_GENE_EXPRESSION","REACTOME_NGF_STIMULATED_TRANSCRIPTION","GO:0006979","REACTOME_SENESCENCE_ASSOCIATED_SECRETORY_PHENOTYPE_SASP","REACTOME_RRNA_PROCESSING","GO:0006979","GO:0034101","REACTOME_SIGNALING_BY_NOTCH","REACTOME_INTERFERON_SIGNALING","REACTOME_SENESCENCE_ASSOCIATED_SECRETORY_PHENOTYPE_SASP","REACTOME_TRANSLATION","REACTOME_SIGNALING_BY_ROBO_RECEPTORS","GO:0034975","REACTOME_INTERFERON_ALPHA_BETA_SIGNALING","GO:0019058","GO:0019362","GO:0046496","GO:0035455","GO:0007015","GO:0052547","GO:0050678","GO:0030856","GO:0072080","GO:0061326","GO:0072160","GO:0050673","GO:0043068","REACTOME_REGULATION_OF_EXPRESSION_OF_SLITS_AND_ROBOS","REACTOME_ATF4_ACTIVATES_GENES_IN_RESPONSE_TO_ENDOPLASMIC_RETICULUM_STRESS","GO:0050792","GO:0048525","GO:0002181","GO:0036003","GO:0090083","REACTOME_SRP_DEPENDENT_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE","REACTOME_ATF6_ATF6_ALPHA_ACTIVATES_CHAPERONES","REACTOME_ATF6_ATF6_ALPHA_ACTIVATES_CHAPERONE_GENES","REACTOME_EUKARYOTIC_TRANSLATION_INITIATION","REACTOME_SELENOAMINO_ACID_METABOLISM","REACTOME_EUKARYOTIC_TRANSLATION_ELONGATION","REACTOME_PERK_REGULATES_GENE_EXPRESSION","REACTOME_NONSENSE_MEDIATED_DECAY_NMD","GO:0034976","REACTOME_INFLUENZA_INFECTION","REACTOME_UNFOLDED_PROTEIN_RESPONSE_UPR","REACTOME_RESPONSE_OF_EIF2AK1_HRI_TO_HEME_DEFICIENCY","REACTOME_CELLULAR_RESPONSE_TO_STARVATION","REACTOME_RESPONSE_OF_EIF2AK4_GCN2_TO_AMINO_ACID_DEFICIENCY"],["Transcriptional Activity Of Smad2 Smad3 Smad4 Heterotrimer","Senescence Associated Secretory Phenotype Sasp","Base Excision Repair","Hsf1 Dependent Transactivation","Resolution Of Abasic Sites Ap Sites","Translocation Of Slc2a4 Glut4 To The Plasma Membrane","Gap Filling Dna Repair Synthesis And Ligation In Gg Ner","Resolution Of Ap Sites Via The Multiple Nucleotide Patch Replacement Pathway","Estrogen Dependent Gene Expression","Downregulation Of Smad2 3 Smad4 Transcriptional Activity","Transcriptional Regulation By Runx2","Cyclin D Associated Events In G1","Pcna Dependent Long Patch Base Excision Repair","Neutrophil Degranulation","GO:Epithelial Cell Migration","Attenuation Phase","Hsp90 Chaperone Cycle For Steroid Hormone Receptors Shr In The Presence Of Ligand","Cellular Senescence","Hsf1 Activation","Interferon Alpha Beta Signaling","Runx1 Regulates Genes Involved In Megakaryocyte Differentiation And Platelet Function","Neutrophil Degranulation","Cholesterol Biosynthesis","Activation Of Gene Expression By Srebf Srebp","GO:G1/S Transition Of Mitotic Cell Cycle","Estrogen Dependent Gene Expression","Ngf Stimulated Transcription","GO:Response To Oxidative Stress","Senescence Associated Secretory Phenotype Sasp","Rrna Processing","GO:Response To Oxidative Stress","GO:Erythrocyte Homeostasis","Signaling By Notch","Interferon Signaling","Senescence Associated Secretory Phenotype Sasp","Translation","Signaling By Robo Receptors","GO:Protein Folding In Endoplasmic Reticulum","Interferon Alpha Beta Signaling","GO:Viral Life Cycle","GO:Pyridine Nucleotide Metabolic Process","GO:Nicotinamide Nucleotide Metabolic Process","GO:Response To Interferon-Alpha","GO:Actin Filament Organization","GO:Regulation Of Peptidase Activity","GO:Regulation Of Epithelial Cell Proliferation","GO:Regulation Of Epithelial Cell Differentiation","GO:Nephron Tubule Development","GO:Renal Tubule Development","GO:Nephron Tubule Epithelial Cell Differentiation","GO:Epithelial Cell Proliferation","GO:Positive Regulation Of Programmed Cell Death","Regulation Of Expression Of Slits And Robos","Atf4 Activates Genes In Response To Endoplasmic Reticulum Stress","GO:Regulation Of Viral Process","GO:Negative Regulation Of Viral Process","GO:Cytoplasmic Translation","GO:Positive Regulation Of Transcription From Rna Polymerase Ii Promoter In Response To Stress","GO:Regulation Of Inclusion Body Assembly","Srp Dependent Cotranslational Protein Targeting To Membrane","Atf6 Atf6 Alpha Activates Chaperones","Atf6 Atf6 Alpha Activates Chaperone Genes","Eukaryotic Translation Initiation","Selenoamino Acid Metabolism","Eukaryotic Translation Elongation","Perk Regulates Gene Expression","Nonsense Mediated Decay Nmd","GO:Response To Endoplasmic Reticulum Stress","Influenza Infection","Unfolded Protein Response Upr","Response Of Eif2ak1 Hri To Heme Deficiency","Cellular Response To Starvation","Response Of Eif2ak4 Gcn2 To Amino Acid Deficiency"],["6/174","10/230","9/230","6/230","6/230","8/230","5/230","5/230","10/174","5/174","9/174","7/230","5/230","24/230","13/221","6/230","8/230","14/230","6/230","6/149","7/149","20/174","5/174","6/174","15/295","9/149","6/174","22/295","8/149","11/149","17/221","9/199","15/174","13/174","10/174","14/149","12/149","4/199","9/174","15/199","8/221","8/221","5/221","20/221","20/221","19/221","12/221","10/221","10/221","5/221","20/221","28/295","12/149","6/149","14/221","11/221","13/199","6/199","6/199","11/149","5/149","5/149","12/149","12/149","11/149","8/149","13/149","19/199","15/149","13/149","8/149","19/149","18/149"],["51/10811","108/10811","88/10811","38/10811","38/10811","72/10811","25/10811","25/10811","146/10811","31/10811","119/10811","47/10811","21/10811","475/10811","299/17124","28/10811","57/10811","185/10811","31/10811","66/10811","93/10811","475/10811","27/10811","42/10811","224/17124","146/10811","39/10811","413/17124","108/10811","203/10811","413/17124","134/17124","232/10811","184/10811","108/10811","293/10811","218/10811","11/17124","66/10811","312/17124","73/17124","73/17124","20/17124","444/17124","433/17124","358/17124","146/17124","96/17124","99/17124","14/17124","424/17124","499/17124","172/10811","27/10811","158/17124","89/17124","159/17124","19/17124","17/17124","113/10811","12/10811","10/10811","120/10811","118/10811","94/10811","32/10811","116/10811","258/17124","157/10811","92/10811","15/10811","157/10811","102/10811"],[0.0001575400058670638,9.777159906762573e-05,0.0001013223551275299,0.0001356056055693597,0.0001356056055693597,0.0001376977945623612,0.0001565475415683218,0.0001565475415683218,0.0001209686173503885,0.0001233972460349184,0.0001250294816452645,5.485548216582948e-05,6.427163884824767e-05,7.33344243616077e-05,0.000142330732015944,2.21061771002944e-05,2.499837320144827e-05,3.990390884761753e-05,4.095669009672834e-05,0.0002849951352822482,0.0002887982949034313,7.67438342570148e-05,6.176471108160063e-05,5.172301521343852e-05,8.253143817736493e-06,0.0001831031598634567,3.348053340428034e-05,3.044665731260808e-06,0.0001186678909564388,0.000112309625758966,2.721114598720067e-05,2.692729522290702e-05,4.85337412388404e-06,8.213686845501182e-06,8.949445084099759e-06,5.150503803121125e-05,4.674628453300652e-05,5.478800826020742e-06,9.92078038428936e-07,3.844627616333021e-06,4.43361279523962e-06,4.43361279523962e-06,4.53014045022575e-06,1.332897125177292e-06,9.043082715203696e-07,2.105925847370457e-07,4.268889918819094e-07,4.481613890120853e-07,5.984841381383383e-07,6.231516048364198e-07,6.51799951288555e-07,4.394222918635581e-08,4.299456436073431e-06,1.444381236341139e-06,1.752984434703433e-08,1.949696880939258e-08,4.346256578102645e-08,5.464268072372024e-08,2.541267083452429e-08,4.103266156852472e-07,3.406905884159341e-07,1.108465073180849e-07,8.853302720555258e-08,7.33542834413042e-08,6.094879017998053e-08,8.571900387154497e-09,6.223405217228318e-09,1.790675038969411e-10,3.766700082212158e-09,3.350990670015246e-10,6.393673663141313e-12,3.847694119199133e-13,1.917469357150771e-15],[0.009925020369625021,0.009113303312727307,0.009113303312727307,0.009113303312727307,0.009113303312727307,0.009113303312727307,0.009113303312727307,0.009113303312727307,0.008592935283983633,0.008592935283983633,0.008592935283983633,0.00853822226495861,0.00853822226495861,0.00853822226495861,0.008373389336056634,0.008344925607208398,0.008344925607208398,0.008344925607208398,0.008344925607208398,0.007285593348700199,0.007285593348700199,0.007252292337287899,0.006670588796812867,0.006517099916893253,0.005195354033265123,0.005081112686210923,0.005062256650727187,0.003833234155657357,0.003466351551622292,0.003462880127568119,0.002777937874163928,0.002720554394021039,0.001691445120894854,0.001691445120894854,0.001691445120894854,0.001681488006313073,0.001621511744738664,0.0009225691835371595,0.0007500109970522756,0.0007283166440690867,0.0006836572827275468,0.0006836572827275468,0.0006836572827275468,0.0003558835324223369,0.0002615711675372669,0.0002262397630922574,0.0002262397630922574,0.0002262397630922574,0.0002262397630922574,0.0002262397630922574,0.0002262397630922574,0.0001659697996368659,0.000159079888134717,5.725939901209516e-05,3.383698936870082e-05,3.383698936870082e-05,2.760366087893268e-05,2.760366087893268e-05,2.567526843314771e-05,1.751779013117786e-05,1.575693971423695e-05,5.592710141957917e-06,4.913583009908168e-06,4.523514145547092e-06,4.228322318736149e-06,6.796292449815352e-07,5.756649825936194e-07,5.427536043116285e-07,4.181037091255495e-07,4.649499554646153e-08,1.182829627681143e-09,1.06773511807776e-10,1.064195493218678e-12],[6,10,9,6,6,8,5,5,10,5,9,7,5,24,13,6,8,14,6,6,7,20,5,6,15,9,6,22,8,11,17,9,15,13,10,14,12,4,9,15,8,8,5,20,20,19,12,10,10,5,20,28,12,6,14,11,13,6,6,11,5,5,12,12,11,8,13,19,15,13,8,19,18],[51,108,88,38,38,72,25,25,146,31,119,47,21,475,299,28,57,185,31,66,93,475,27,42,224,146,39,413,108,203,413,134,232,184,108,293,218,11,66,312,73,73,20,444,433,358,146,96,99,14,424,499,172,27,158,89,159,19,17,113,12,10,120,118,94,32,116,258,157,92,15,157,102],[0.1176470588235294,0.09259259259259259,0.1022727272727273,0.1578947368421053,0.1578947368421053,0.1111111111111111,0.2,0.2,0.0684931506849315,0.1612903225806452,0.07563025210084033,0.148936170212766,0.2380952380952381,0.05052631578947368,0.04347826086956522,0.2142857142857143,0.1403508771929824,0.07567567567567568,0.1935483870967742,0.09090909090909091,0.07526881720430108,0.04210526315789474,0.1851851851851852,0.1428571428571428,0.06696428571428571,0.06164383561643835,0.1538461538461539,0.05326876513317191,0.07407407407407407,0.0541871921182266,0.04116222760290557,0.06716417910447761,0.06465517241379311,0.07065217391304347,0.09259259259259259,0.04778156996587031,0.05504587155963303,0.3636363636363636,0.1363636363636364,0.04807692307692308,0.1095890410958904,0.1095890410958904,0.25,0.04504504504504504,0.04618937644341801,0.05307262569832402,0.0821917808219178,0.1041666666666667,0.101010101010101,0.3571428571428572,0.04716981132075472,0.0561122244488978,0.06976744186046512,0.2222222222222222,0.08860759493670886,0.1235955056179775,0.08176100628930817,0.3157894736842105,0.3529411764705883,0.09734513274336283,0.4166666666666667,0.5,0.1,0.1016949152542373,0.1170212765957447,0.25,0.1120689655172414,0.07364341085271318,0.09554140127388536,0.1413043478260869,0.5333333333333333,0.1210191082802548,0.1764705882352941],["PARP1/STAT1/WWTR1/SKIL/UBB/JUNB","JUN/H2AZ1/H4C3/CDKN1A/H2AZ2/CDKN2A/CDK4/FOS/UBB/UBE2S","POLE4/H2AZ1/H4C3/RPA3/POLD2/H2AZ2/UNG/APEX1/PCNA","HSPA1A/HSPA1B/HSPA8/HSBP1/UBB/DNAJB1","POLE4/RPA3/POLD2/UNG/APEX1/PCNA","SFN/RAB13/TUBB4B/TUBA1B/TUBA1A/TUBA1C/ACTG1/YWHAH","POLE4/RPA3/POLD2/UBB/PCNA","POLE4/RPA3/POLD2/APEX1/PCNA","JUN/H2BC4/H2AC6/H2BC5/H2BC7/H2AZ2/GATA3/FOS/FOSB/ZNF217","PARP1/STAT1/WWTR1/SKIL/UBB","STAT1/WWTR1/HES1/PSMB8/CDKN1A/PSME2/LGALS3/UBB/SOX9","CKS1B/CDKN1A/CCNE2/CDKN2A/CCND1/CDK4/UBB","POLE4/RPA3/POLD2/APEX1/PCNA","S100A9/QSOX1/TMEM63A/CD47/S100P/TUBB/HSPA1A/HSPA1B/SDCBP/GSN/TUBB4B/CD59/CD44/HSPA8/MGST1/CTSH/PYCARD/CYBA/TRAPPC1/CST3/SLPI/CTSA/PRDX4/DDX3X","EDN2/TACSTD2/JUN/GLUL/RHOB/S100P/VEGFA/GATA3/NR4A1/CTSH/NR2F2/SOX9/ID1","HSPA1A/HSPA1B/HSPA8/HSBP1/UBB/DNAJB1","HSPA1A/HSPA1B/TUBB4B/HSPA8/TUBA1B/TUBA1A/TUBA1C/DNAJB1","JUN/H2AZ1/H4C3/HMGA1/CDKN1A/H2AZ2/CCNE2/CDKN2A/CDK4/FOS/UBB/UBE2S/ID1/RBBP7","HSPA1A/HSPA1B/RPA3/HSBP1/UBB/DNAJB1","ISG15/IFI6/STAT1/IFITM3/IFI27/MX1","H2BC21/H2BC4/H2AC6/H2BC5/KMT2E/KMT2C/MYL9","S100A9/S100P/DSP/TUBB/CTSB/GSDMD/NAPRT/B4GALT1/STOM/LCN2/AGPAT2/TUBB4B/LGALS3/CTSH/PYCARD/CYBA/CFD/BST2/CST3/SLPI","FDPS/HMGCS1/SQLE/IDI1/TM7SF2","FDPS/HMGCS1/SQLE/IDI1/TM7SF2/FASN","NASP/MUC1/ID2/PLK2/CDKN1A/EIF4EBP1/PRKDC/CCNE2/CDKN2A/ANXA1/CDK2AP2/CCND1/CDK4/APEX1/DDX3X","JUN/H2BC21/H2BC4/H2AC6/H2BC5/AGO2/FOS/JUND/FOSB","ID3/ID2/FOS/JUNB/FOSB/ID1","SFPQ/DHCR24/JUN/PYCR2/RHOB/EPAS1/APOD/ATOX1/HSPA1A/HSPA1B/PON2/ANXA1/UCP2/MGST1/APEX1/FOS/NQO1/HNRNPM/PCNA/MAP1LC3A/RCAN1/PRDX4","JUN/H2BC21/H2BC4/H2AC6/H2BC5/CDKN1A/FOS/CEBPB","RPS27/RPL37A/RPL22L1/RPL34/RPL37/RPS29/RPS17/RPL38/RPL36/RPS21/RPL39","SFPQ/JUN/PARP1/SDC1/RHOB/NR4A2/STAT1/APOD/NAPRT/NET1/MSRB2/CRYAB/FOS/TPM1/NQO1/PRDX2/RCAN1","ISG15/STAT1/HSPA1A/HSPA1B/VEGFA/KMT2E/ZFP36L1/RPS17/CEBPG","JUN/PBX1/ELF3/STAT1/HES1/H2BC4/H2AC6/H2BC5/H2BC7/PSMB8/H2AZ2/B4GALT1/MDK/PSME2/UBB","ISG15/IFI6/EIF2AK2/STAT1/PSMB8/IFITM2/IFITM3/UBE2L6/IFI27/UBB/EIF4A3/BST2/MX1","JUN/H2BC4/H2AC6/H2BC5/H2BC7/CDKN1A/H2AZ2/FOS/UBB/UBE2S","RPS27/RPL37A/RPL22L1/RPL34/RPL37/GARS1/IARS1/RPS29/RPS17/RPL38/RPL36/EIF2S2/RPS21/RPL39","RPS27/RPL37A/RPL22L1/RPL34/RPL37/UPF2/RPS29/RPS17/RPL38/RPL36/RPS21/RPL39","HSPA5/HSP90B1/DNAJC3/CALR","ISG15/IFI6/STAT1/PSMB8/IFITM2/IFITM3/IFI27/BST2/MX1","ISG15/NECTIN4/CD55/PLSCR1/HSPA1A/HSPA1B/ITGB1/IFITM3/CD81/SLC3A2/IFI27/SLPI/MX1/LGALS1/DDX3X","TKT/PARP9/NAPRT/TALDO1/IDH2/QPRT/NQO1/DCXR","TKT/PARP9/NAPRT/TALDO1/IDH2/QPRT/NQO1/DCXR","EIF2AK2/GATA3/IFITM2/IFITM3/BST2","STMN1/MARCKSL1/TACSTD2/RHOB/SPTBN1/CAPG/RND3/PIK3R1/ARPC1B/MTSS1/AIF1L/MSRB2/EPS8/ACTN1/TPM1/PYCARD/ACTG1/TPM4/MYADM/ID1","IFI6/S100A9/RARRES1/PSMB8/VEGFA/CLDN3/CLDN4/CTSB/CST6/CRYAB/GAPDH/NR4A1/PEBP1/PSME2/CTSH/ARL6IP1/PYCARD/BST2/CST3/SLPI","TACSTD2/JUN/GLUL/STAT1/HES1/VEGFA/LAMB1/MTSS1/B4GALT1/EGFL7/GATA3/MDK/NR4A1/ZFP36L1/NR2F2/CYBA/SOX9/ZFP36/ID1","STAT1/WWTR1/HES1/VEGFA/CD24/GATA3/ZFP36L1/MESP1/SOX9/ZFP36/SULT2B1/ID1","TACSTD2/PBX1/STAT1/WWTR1/HES1/VEGFA/CD24/MTSS1/GATA3/SOX9","TACSTD2/PBX1/STAT1/WWTR1/HES1/VEGFA/CD24/MTSS1/GATA3/SOX9","STAT1/WWTR1/CD24/MTSS1/GATA3","TACSTD2/JUN/GLUL/ID2/STAT1/HES1/VEGFA/LAMB1/MTSS1/B4GALT1/EGFL7/GATA3/MDK/NR4A1/ZFP36L1/NR2F2/CYBA/SOX9/ZFP36/ID1","ID3/SFPQ/JUN/S100A9/ATF3/RHOB/SPDEF/CDKN1A/CLU/PRKDC/CDKN2A/ANXA1/GSN/PHLDA2/MTCH2/GAL/TNFRSF1A/NR4A1/CTSH/TNFRSF12A/PYCARD/NQO1/C1QBP/UBB/GADD45B/BAX/LGALS1/DDX3X","RPS27/RPL37A/RPL22L1/RPL34/RPL37/UPF2/RPS29/RPS17/RPL38/RPL36/RPS21/RPL39","ATF3/DDIT3/HERPUD1/CEBPG/CEBPB/ATF4","ISG15/EIF2AK2/STAT1/PLSCR1/LY6E/STOM/IFITM2/IFITM3/HACD3/BST2/ZFP36/SLPI/MX1/LGALS1","ISG15/EIF2AK2/STAT1/PLSCR1/LY6E/IFITM2/IFITM3/BST2/ZFP36/SLPI/MX1","RPS27/RPL37A/RPL22L1/RPL34/RPL37/CPEB4/RPS29/RPS17/RPL38/RPL36/EIF2S2/RPS21/RPL39","ATF3/VEGFA/HSPA5/DDIT3/CEBPB/ATF4","DNAJB2/HSPA1A/HSPA1B/CLU/BAG5/DNAJB1","RPS27/RPL37A/RPL22L1/RPL34/RPL37/RPS29/RPS17/RPL38/RPL36/RPS21/RPL39","HSPA5/DDIT3/HSP90B1/CALR/ATF4","HSPA5/DDIT3/HSP90B1/CALR/ATF4","RPS27/RPL37A/RPL22L1/RPL34/RPL37/RPS29/RPS17/RPL38/RPL36/EIF2S2/RPS21/RPL39","RPS27/RPL37A/RPL22L1/RPL34/RPL37/IARS1/RPS29/RPS17/RPL38/RPL36/RPS21/RPL39","RPS27/RPL37A/RPL22L1/RPL34/RPL37/RPS29/RPS17/RPL38/RPL36/RPS21/RPL39","ATF3/HSPA5/DDIT3/HERPUD1/CEBPG/EIF2S2/CEBPB/ATF4","RPS27/RPL37A/RPL22L1/RPL34/RPL37/UPF2/RPS29/RPS17/SMG1/RPL38/RPL36/RPS21/RPL39","JUN/NIBAN1/ATF3/DNAJB2/MARCHF6/HSPA1A/CLU/HSPA5/DDIT3/HSP90B1/UFM1/DNAJC3/HERPUD1/CALR/PPP1R15A/TRIB3/CEBPB/ATF4/DDX3X","ISG15/RPS27/RPL37A/RPL22L1/RPL34/RPL37/HSPA1A/DNAJC3/RPS29/RPS17/RPL38/RPL36/CALR/RPS21/RPL39","ATF3/GFPT1/CREBRF/HSPA5/DDIT3/HSP90B1/DNAJC3/HERPUD1/CALR/CEBPG/EIF2S2/CEBPB/ATF4","ATF3/DDIT3/CEBPG/PPP1R15A/TRIB3/EIF2S2/CEBPB/ATF4","RPS27/ATF3/RPL37A/RPL22L1/RPL34/RPL37/FNIP1/DDIT3/RPS29/RPS17/RPL38/RPL36/CEBPG/TRIB3/EIF2S2/CEBPB/RPS21/ATF4/RPL39","RPS27/ATF3/RPL37A/RPL22L1/RPL34/RPL37/DDIT3/RPS29/RPS17/RPL38/RPL36/CEBPG/TRIB3/EIF2S2/CEBPB/RPS21/ATF4/RPL39"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>site<\/th>\n      <th>ID<\/th>\n      <th>Description<\/th>\n      <th>GeneRatio<\/th>\n      <th>BgRatio<\/th>\n      <th>pvalue<\/th>\n      <th>p.adjust<\/th>\n      <th>Count<\/th>\n      <th>setSize<\/th>\n      <th>GeneProp<\/th>\n      <th>geneID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":9,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[5,6,7,8,9]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render"],"jsHooks":[]}</script>
</div>
<div id="number-of-gene-sets-1" class="section level4" number="10.1.2">
<h4><span class="header-section-number">10.1.2</span> Number of gene
sets</h4>
<pre class="r"><code>gse_df %&gt;% 
  filter(p.adjust &lt; p_thrs) %&gt;% 
  group_by(group) %&gt;% 
  summarise(n = n()) %&gt;% 
  ggplot(aes(x=group, y=n)) +
  geom_bar(stat=&quot;identity&quot;, color = &#39;black&#39;)+
  geom_text(aes(label=n), vjust=1.4, color=&quot;white&quot;, size=geom_text_size)+
  labs(
    x = &#39;Site&#39;,
    y = &#39;Number of gene sets&#39;,
    title = &#39;Number of significant gene sets by site&#39;,
    caption = paste(&quot;Adjusted P value &lt;&quot;, p_thrs)
    
  )</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/site-marker-gse-enricher-c2reactome_gobp-ngset-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-set-similarity-matrix-1" class="section level4"
number="10.1.3">
<h4><span class="header-section-number">10.1.3</span> Gene-set
similarity matrix</h4>
<p>Heatmap showing the pair-wise similarity matrix of top 20 gene sets
with an adjusted P value &lt; 0.01 in each gene group. Heatmap colors
represent the Jaccard similarity coefficient.</p>
<pre class="r"><code>col_fun &lt;- colorRamp2(seq(0, 1, length.out = 4), brewer.pal(4, &quot;GnBu&quot;))
n_split &lt;- 16

use_mat &lt;- jc_matrix
colnames(use_mat) &lt;- colnames(use_mat) %&gt;% clean_msigdb_names %&gt;% tolower() %&gt;% str_to_title %&gt;% ep_str_wrap(70)
rownames(use_mat) &lt;- rownames(use_mat) %&gt;% clean_msigdb_names %&gt;% tolower() %&gt;% str_to_title %&gt;% ep_str_wrap(70)

ha_top &lt;- HeatmapAnnotation(
  foo = anno_block(
    # labels = c(&quot;Translation&quot;, &quot;Cell division&quot;),
    labels = 1:n_split,
    labels_gp = gpar(col = &quot;black&quot;, fontsize = 8),
    gp = gpar(lwd = 0, lty = 0))
  )

ht &lt;- Heatmap(
  use_mat,
  name = &#39;Jaccard index&#39;,
  column_split = n_split,
  row_split = n_split,
  column_title = NULL,
  row_title = NULL,
  col = col_fun,
  show_column_dend = FALSE,
  show_column_names = FALSE,
  border = TRUE,
  top_annotation = ha_top,
  heatmap_legend_param = list(title_gp = gpar(fontsize = 8),labels_gp = gpar(fontsize = 8)),
  width = unit(7, &quot;cm&quot;))

ht = draw(ht)</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/site-marker-gse-enricher-c2reactome_gobp-jcmatrix-1.png" width="4320" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;\n\n&quot;)</code></pre>
<pre class="r"><code># Extract term order according to hclust from similarity heatmap
row_order &lt;- row_order(ht) %&gt;% set_names(1:n_split)
term_similarity_clusters &lt;- foreach(i=names(row_order), .combine = rbind) %do% {
  data.frame(
    jc_sim_cl = i, 
    row_order = row_order[[i]], 
    Description =  rownames(jc_matrix)[row_order[[i]]]
  )
}

# Add similarity order to data for dot plot
res_top %&lt;&gt;% left_join(term_similarity_clusters)</code></pre>
</div>
<div id="dotplots-1" class="section level4" number="10.1.4">
<h4><span class="header-section-number">10.1.4</span> Dotplots</h4>
<p>The plots below show the top 20 gene sets with an adjusted P value
&lt; 0.01 in each gene group</p>
<pre class="r"><code>res_top %&gt;%
  # Combine similarity clusters
  mutate(
    jc_sim_cl_n = as.numeric(jc_sim_cl),
    jc_sim_cl = &#39;Other&#39;,
    jc_sim_cl = ifelse(jc_sim_cl_n %in% c(1), &#39;Translation&#39;, jc_sim_cl),
    jc_sim_cl = ifelse(jc_sim_cl_n %in% c(3, 4), &#39;Protein folding&#39;, jc_sim_cl),
    jc_sim_cl = ifelse(jc_sim_cl_n %in% c(2), &#39;DNA repair&#39;, jc_sim_cl),
    jc_sim_cl = ifelse(jc_sim_cl_n %in% c(13, 8), &#39;Cell stress&#39;, jc_sim_cl),
    jc_sim_cl = ifelse(jc_sim_cl_n %in% c(6, 7), &#39;Epithelial cell\nprocesses&#39;, jc_sim_cl),
    jc_sim_cl = ifelse(jc_sim_cl_n %in% c(9), &#39;Energy\ngeneration&#39;, jc_sim_cl),
  
    jc_sim_cl = factor(
      jc_sim_cl,
      levels = c(&#39;Translation&#39;, &#39;Protein folding&#39;, &#39;DNA repair&#39;, &#39;Cell stress&#39;, &#39;Epithelial cell\nprocesses&#39;, &#39;Energy\ngeneration&#39;, &#39;Other&#39;)
      ),
    Description = factor(Description, rev(rownames(jc_matrix))),
    # Description = fct_reorder(Description, rownames(jc_matrix)),
    p.adjust = ifelse(p.adjust &gt; 0.05, 1, p.adjust),
    p.adjust = -log10( p.adjust),
    p.adjust = ifelse(p.adjust &gt;= 10, 10, p.adjust),
    group = recode(group, primary_tumor = &quot;PT&quot;, lung = &quot;Lung&quot;, liver = &quot;Liver&quot;),
    group = factor(group, levels = c(&#39;PT&#39;, &#39;Lung&#39;, &#39;Liver&#39;))
  ) %&gt;% 
  ggplot(aes(group, Description, color = GeneProp, size = p.adjust)) +
  geom_point() +
  scale_color_distiller(palette = &quot;Purples&quot;, na.value = &quot;grey70&quot;, direction = 1) +
  scale_y_discrete(labels = default_labeller(70)) +
  scale_x_discrete(position = &quot;top&quot;) +
  scale_size(range = c(0.8, 5)) +
  labs(y = &#39;&#39;, x = &#39;&#39;) +
  # facet_grid(rows = vars(cluster), scale = &#39;free&#39;, space = &#39;free&#39;) +
  # facet_grid(rows = vars(kcluster), scale = &#39;free&#39;, space = &#39;free&#39;) +
  facet_grid(rows = vars(jc_sim_cl), scale = &#39;free&#39;, space = &#39;free&#39;) +
  theme(
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, size = 1),
    # axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    axis.line.x = element_blank(),
    strip.background = element_blank()#,
    # strip.text.y = element_blank()
    )</code></pre>
<p><img src="figure/br16-10x_rnaseq-cancer_cells-main.Rmd/site-marker-gse-enricher-c2reactome_gobp-dotplot-1.png" width="3600" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<p>R version 4.2.2 (2022-10-31) Platform: x86_64-apple-darwin17.0
(64-bit) Running under: macOS Big Sur … 10.16</p>
<p>Matrix products: default BLAS:
/Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK:
/Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</p>
<p>locale: [1]
en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</p>
<p>attached base packages: [1] grid stats4 stats graphics grDevices
utils datasets [8] methods base</p>
<p>other attached packages: [1] SingleR_2.0.0 factoextra_1.0.7<br />
[3] corrplot_0.92 circlize_0.4.15<br />
[5] GSVA_1.46.0 clusterProfiler_4.6.0<br />
[7] ComplexHeatmap_2.14.0 miloR_1.6.0<br />
[9] edgeR_3.40.2 limma_3.54.1<br />
[11] speckle_0.0.3 scDblFinder_1.13.8<br />
[13] bluster_1.8.0 batchelor_1.14.1<br />
[15] BiocSingular_1.14.0 scran_1.26.2<br />
[17] SeuratObject_4.1.3 Seurat_4.3.0<br />
[19] DropletUtils_1.18.1 scater_1.26.1<br />
[21] scuttle_1.8.4 SingleCellExperiment_1.20.0 [23]
SummarizedExperiment_1.28.0 Biobase_2.58.0<br />
[25] GenomicRanges_1.50.2 GenomeInfoDb_1.34.9<br />
[27] IRanges_2.32.0 S4Vectors_0.36.1<br />
[29] BiocGenerics_0.44.0 MatrixGenerics_1.10.0<br />
[31] matrixStats_1.0.0 magrittr_2.0.3<br />
[33] openxlsx_4.2.5.2 patchwork_1.1.3<br />
[35] pheatmap_1.0.12 ggpubr_0.6.0<br />
[37] RColorBrewer_1.1-3 arsenal_3.6.3<br />
[39] ggbeeswarm_0.7.2 colorblindr_0.1.0<br />
[41] colorspace_2.1-1 cowplot_1.1.1<br />
[43] kableExtra_1.3.4 knitr_1.44<br />
[45] DT_0.30 foreach_1.5.2<br />
[47] showtext_0.9-6 showtextdb_3.0<br />
[49] sysfonts_0.8.8 lubridate_1.9.3<br />
[51] forcats_1.0.0 stringr_1.5.1<br />
[53] dplyr_1.1.4 purrr_1.0.2<br />
[55] readr_2.1.4 tidyr_1.3.1<br />
[57] tibble_3.2.1 ggplot2_3.5.0<br />
[59] tidyverse_2.0.0 workflowr_1.7.1</p>
<p>loaded via a namespace (and not attached): [1] rsvd_1.0.5
ica_1.0-3<br />
[3] svglite_2.1.2 ps_1.7.5<br />
[5] Rsamtools_2.14.0 lmtest_0.9-40<br />
[7] rprojroot_2.0.3 crayon_1.5.2<br />
[9] MASS_7.3-60 rhdf5filters_1.10.0<br />
[11] nlme_3.1-163 backports_1.4.1<br />
[13] GOSemSim_2.24.0 rlang_1.1.4<br />
[15] HDO.db_0.99.1 XVector_0.38.0<br />
[17] ROCR_1.0-11 irlba_2.3.5.1<br />
[19] callr_3.7.3 xgboost_1.7.5.1<br />
[21] BiocParallel_1.32.5 rjson_0.2.21<br />
[23] bit64_4.0.5 glue_1.8.0<br />
[25] sctransform_0.4.1 parallel_4.2.2<br />
[27] processx_3.8.2 vipor_0.4.5<br />
[29] spatstat.sparse_3.0-2 AnnotationDbi_1.60.0<br />
[31] DOSE_3.24.2 spatstat.geom_3.2-7<br />
[33] tidyselect_1.2.1 fitdistrplus_1.1-11<br />
[35] XML_3.99-0.14 zoo_1.8-12<br />
[37] GenomicAlignments_1.34.0 org.Mm.eg.db_3.16.0<br />
[39] xtable_1.8-4 evaluate_0.22<br />
[41] cli_3.6.3 zlibbioc_1.44.0<br />
[43] rstudioapi_0.15.0 miniUI_0.1.1.1<br />
[45] sp_2.1-1 whisker_0.4.1<br />
[47] bslib_0.5.1 fastmatch_1.1-4<br />
[49] treeio_1.22.0 shiny_1.7.5.1<br />
[51] xfun_0.40 clue_0.3-65<br />
[53] gson_0.1.0 cluster_2.1.4<br />
[55] tidygraph_1.2.3 KEGGREST_1.38.0<br />
[57] ggrepel_0.9.4 ape_5.7-1<br />
[59] listenv_0.9.0 Biostrings_2.66.0<br />
[61] png_0.1-8 future_1.33.0<br />
[63] withr_3.0.1 bitops_1.0-7<br />
[65] ggforce_0.4.1 plyr_1.8.9<br />
[67] GSEABase_1.60.0 dqrng_0.3.1<br />
[69] pillar_1.9.0 GlobalOptions_0.1.2<br />
[71] cachem_1.0.8 fs_1.6.4<br />
[73] GetoptLong_1.0.5 DelayedMatrixStats_1.20.0 [75] vctrs_0.6.5
ellipsis_0.3.2<br />
[77] generics_0.1.3 tools_4.2.2<br />
[79] beeswarm_0.4.0 munsell_0.5.1<br />
[81] tweenr_2.0.2 fgsea_1.24.0<br />
[83] DelayedArray_0.24.0 fastmap_1.1.1<br />
[85] compiler_4.2.2 abind_1.4-5<br />
[87] httpuv_1.6.8 rtracklayer_1.58.0<br />
[89] plotly_4.10.3 GenomeInfoDbData_1.2.9<br />
[91] gridExtra_2.3 lattice_0.20-45<br />
[93] deldir_1.0-9 utf8_1.2.4<br />
[95] later_1.3.1 jsonlite_1.8.7<br />
[97] scales_1.3.0 graph_1.76.0<br />
[99] ScaledMatrix_1.6.0 tidytree_0.4.2<br />
[101] pbapply_1.7-2 carData_3.0-5<br />
[103] sparseMatrixStats_1.10.0 lazyeval_0.2.2<br />
[105] promises_1.2.1 car_3.1-2<br />
[107] doParallel_1.0.17 R.utils_2.12.2<br />
[109] goftest_1.2-3 spatstat.utils_3.0-3<br />
[111] reticulate_1.34.0 rmarkdown_2.25<br />
[113] statmod_1.5.0 webshot_0.5.5<br />
[115] Rtsne_0.16 downloader_0.4<br />
[117] uwot_0.1.16 igraph_1.5.1<br />
[119] HDF5Array_1.26.0 survival_3.5-7<br />
[121] ResidualMatrix_1.8.0 yaml_2.3.7<br />
[123] systemfonts_1.1.0 htmltools_0.5.6.1<br />
[125] memoise_2.0.1 BiocIO_1.8.0<br />
[127] locfit_1.5-9.8 graphlayouts_1.0.1<br />
[129] viridisLite_0.4.2 digest_0.6.37<br />
[131] mime_0.12 RSQLite_2.3.1<br />
[133] yulab.utils_0.1.7 future.apply_1.11.0<br />
[135] data.table_1.14.8 blob_1.2.4<br />
[137] R.oo_1.25.0 labeling_0.4.3<br />
[139] splines_4.2.2 Rhdf5lib_1.20.0<br />
[141] Cairo_1.6-1 RCurl_1.98-1.12<br />
[143] broom_1.0.5 hms_1.1.3<br />
[145] rhdf5_2.42.0 shape_1.4.6<br />
[147] aplot_0.2.2 sass_0.4.7<br />
[149] Rcpp_1.0.13 RANN_2.6.1<br />
[151] enrichplot_1.18.3 fansi_1.0.6<br />
[153] tzdb_0.4.0 parallelly_1.36.0<br />
[155] R6_2.5.1 ggridges_0.5.4<br />
[157] lifecycle_1.0.4 zip_2.3.0<br />
[159] ggsignif_0.6.4 leiden_0.4.3<br />
[161] jquerylib_0.1.4 Matrix_1.5-3<br />
[163] qvalue_2.30.0 RcppAnnoy_0.0.21<br />
[165] org.Hs.eg.db_3.16.0 iterators_1.0.14<br />
[167] spatstat.explore_3.2-3 htmlwidgets_1.6.2<br />
[169] beachmat_2.14.0 polyclip_1.10-6<br />
[171] crosstalk_1.2.0 shadowtext_0.1.2<br />
[173] gridGraphics_0.5-1 timechange_0.2.0<br />
[175] rvest_1.0.3 globals_0.16.2<br />
[177] spatstat.random_3.2-1 progressr_0.14.0<br />
[179] codetools_0.2-19 GO.db_3.16.0<br />
[181] metapod_1.6.0 gtools_3.9.4<br />
[183] getPass_0.2-2 R.methodsS3_1.8.2<br />
[185] gtable_0.3.5 DBI_1.1.3<br />
[187] git2r_0.32.0 highr_0.10<br />
[189] ggfun_0.1.3 tensor_1.5<br />
[191] httr_1.4.7 KernSmooth_2.23-22<br />
[193] stringi_1.8.4 reshape2_1.4.4<br />
[195] farver_2.1.2 annotate_1.76.0<br />
[197] viridis_0.6.4 magick_2.7.3<br />
[199] ggtree_3.4.4 xml2_1.3.5<br />
[201] BiocNeighbors_1.16.0 restfulr_0.0.15<br />
[203] ggplotify_0.1.2 scattermore_1.2<br />
[205] bit_4.0.5 scatterpie_0.2.1<br />
[207] spatstat.data_3.0-3 ggraph_2.1.0<br />
[209] pkgconfig_2.0.3 rstatix_0.7.2</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
