<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Castro-Giner" />

<meta name="date" content="2025-02-07" />

<title>Comapre 10x RNA-seq analysis of microenvironment mets and normal tissue</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.30/datatables.js"></script>
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.13.4/jszip.min.js"></script>
<link href="site_libs/dt-ext-buttons-1.13.4/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.13.4/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.print.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">met-paths</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheAcetoLab/sznurkowska-met-paths">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Comapre 10x RNA-seq analysis of
microenvironment mets and normal tissue</h1>
<h3 class="subtitle">Differential gene expression analysis of TME and
normal cells from lung and liver normal tissue and metastases, using
pseudobulk</h3>
<h4 class="author">Francesc Castro-Giner</h4>
<h4 class="date">February 07, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-07
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>sznurkowska-met-paths/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250206code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250206)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250206code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250206)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsznurkowskametpathstree30d581dbf1cff0b0af4988bcfb642939a182358btargetblank30d581da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/tree/30d581dbf1cff0b0af4988bcfb642939a182358b" target="_blank">30d581d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsznurkowskametpathstree30d581dbf1cff0b0af4988bcfb642939a182358btargetblank30d581da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/tree/30d581dbf1cff0b0af4988bcfb642939a182358b" target="_blank">30d581d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/patients/
    Ignored:    data/resources/
    Ignored:    data/rnaseq/
    Ignored:    data/seer/
    Ignored:    output/p26532_o26674/
    Ignored:    output/p26532_o28268/
    Ignored:    output/p26532_o28268_o34980/
    Ignored:    output/p26532_o34980/

Untracked files:
    Untracked:  analysis/br16-10x_rnaseq-cancer_cells-main.Rmd
    Untracked:  analysis/br16-10x_rnaseq-qc.Rmd
    Untracked:  analysis/mvt1-tumor-10x_rnaseq-qc.Rmd
    Untracked:  analysis/templates/
    Untracked:  code/R-functions/
    Untracked:  configuration/
    Untracked:  workflowr_update.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   README.md
    Modified:   sznurkowska-met-paths.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd</code>)
and HTML
(<code>docs/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/30d581dbf1cff0b0af4988bcfb642939a182358b/analysis/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd" target="_blank">30d581d</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
<td>
Add nsg normal analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries-additional-functions-and-data"
class="section level2" number="1">
<h2><span class="header-section-number">1</span> Load libraries,
additional functions and data</h2>
<p>Setup environment</p>
<pre class="r"><code>knitr::opts_chunk$set(results=&#39;asis&#39;, echo=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.align = &#39;center&#39;, fig.width = 3.5, fig.asp = 0.618, dpi = 600, dev = c(&quot;png&quot;, &quot;pdf&quot;))

options(stringsAsFactors = FALSE)

use_seed &lt;- 1100101
set.seed(use_seed)

if(!dir.exists(params$output_dir))
  dir.create(params$output_dir, recursive = TRUE, showWarnings = FALSE)</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(tidyverse)
library(showtext)
library(foreach)
library(DT)
library(knitr)
library(kableExtra)
library(cowplot)
library(colorblindr)
library(ggbeeswarm)
library(arsenal)
library(RColorBrewer)
library(ggpubr)
library(pheatmap)
library(patchwork)
library(openxlsx)
library(magrittr)

library(scater)
library(DropletUtils)
library(Seurat)
library(scran)
library(BiocSingular)
library(batchelor)
library(bluster)
library(celldex)
library(SingleR)
library(scDblFinder)
library(speckle)
library(miloR)
library(ComplexHeatmap)
library(clusterProfiler)
library(GSVA)
library(circlize)
library(org.Mm.eg.db)

library(factoextra)</code></pre>
<p>Load ggplot theme</p>
<pre class="r"><code>source(&quot;./configuration/rmarkdown/ggplot_theme.R&quot;)
source(&quot;./configuration/rmarkdown/color_palettes.R&quot;)</code></pre>
<p>Load utils functions</p>
<pre class="r"><code>source(&#39;./code/R-functions/heatmap_scale.R&#39;)
source(&#39;./code/R-functions/dge_wrappers.r&#39;)
source(&#39;./code/R-functions/dge_report.r&#39;)
source(&#39;./code/R-functions/gse_omnibus.r&#39;)
source(&#39;./code/R-functions/gse_report.r&#39;)
source(&#39;./code/R-functions/subchunkify.R&#39;)
source(&#39;./code/R-functions/pairwise_terms_jc.r&#39;)</code></pre>
<p>Load MSigDB gene sets</p>
<pre class="r"><code>gmt_files_symbols &lt;- list(
  msigdb.c2.cp = paste0(params$msigdb_dir, &#39;/&#39;, params$msigdb_vs, &#39;/m2.cp.&#39;, params$msigdb_vs, &#39;.symbols.gmt&#39;),
  msigdb.m2.cp.reactome= paste0(params$msigdb_dir, &#39;/&#39;, params$msigdb_vs, &#39;/m2.cp.reactome.&#39;, params$msigdb_vs, &#39;.symbols.gmt&#39;)
)</code></pre>
<p>Clean files generated in previous runs</p>
<pre class="r"><code>rmd_file &lt;- current_input()
if(!is.null(rmd_file)) {
  figures_dir &lt;- file.path(&#39;./docs/figure&#39;,rmd_file)
  if(dir.exists(figures_dir)) {
    unlink(file.path(figures_dir, &quot;*&quot;))
  }
}</code></pre>
</div>
<div id="configure-analyses" class="section level2" number="2">
<h2><span class="header-section-number">2</span> Configure analyses</h2>
<div id="data-wrangling" class="section level3" number="2.1">
<h3><span class="header-section-number">2.1</span> Data wrangling</h3>
<p>Load original data</p>
<pre class="r"><code>sce_comb &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated.rds&#39;))</code></pre>
<p>Aggregate data</p>
<pre class="r"><code>celltype_label &lt;- &#39;celltype_immgen_pruned.labels&#39;
sce_comb$anatomical_site &lt;- gsub(&quot; .*&quot;, &quot;&quot;, sce_comb$site_caps)
sce_comb$condition &lt;- gsub(&quot;.* &quot;, &quot;&quot;, sce_comb$site_caps)
se &lt;- aggregateAcrossCells(sce_comb, ids = colData(sce_comb)[, c(&quot;anatomical_site&quot;, &quot;condition&quot;, &quot;site&quot;, &quot;celltype_immgen_pruned.labels&quot;, &quot;mouse_id&quot;)])
se &lt;- as(se, &quot;SummarizedExperiment&quot;)
rowData(se) &lt;- rowData(sce_comb)
rownames(se) &lt;- rowData(se)$ensembl_gene
se$sample_alias &lt;- paste(se$anatomical_site, se$condition, se$celltype_immgen_pruned.labels, se$mouse_id, sep = &#39;_&#39;)
colnames(se) &lt;- se$sample_alias</code></pre>
<p>TMM normalization</p>
<pre class="r"><code>norm_method &lt;- &quot;TMM&quot;
exprs_mat_for_norm &lt;- assay(se, &#39;counts&#39;)
norm_factors &lt;- calcNormFactors(exprs_mat_for_norm, method = norm_method)
lib_size &lt;- colSums(exprs_mat_for_norm)/10^6
se$size_factor &lt;- (norm_factors * lib_size)
se &lt;- logNormCounts(se, size.factors =se$size_factor)
se &lt;- logNormCounts(se, size.factors =se$size_factor, log = FALSE)</code></pre>
<p>CPM calculation</p>
<pre class="r"><code>assay(se, &#39;cpm&#39;) &lt;- edgeR::cpm(assay(se, &#39;normcounts&#39;))
assay(se, &#39;logcpm&#39;) &lt;- edgeR::cpm(assay(se, &#39;normcounts&#39;), log = TRUE)</code></pre>
<p>Save aggregated data</p>
<pre class="r"><code>saveRDS(se, file = file.path(params$output_dir, &#39;sce_aggregated.rds&#39;))</code></pre>
<p>Load data</p>
<pre class="r"><code>se &lt;- readRDS(file.path(params$output_dir, &#39;sce_aggregated.rds&#39;))</code></pre>
</div>
<div id="define-of-comparisons" class="section level3" number="2.2">
<h3><span class="header-section-number">2.2</span> Define of
comparisons</h3>
<div id="process-data" class="section level4" number="2.2.1">
<h4><span class="header-section-number">2.2.1</span> Process data</h4>
<pre class="r"><code>x &lt;- colData(se) %&gt;% data.frame %&gt;% 
  mutate(
    cell_type = celltype_immgen_pruned.labels,
    cell_type_site = paste(anatomical_site, cell_type)
  )


comp_list &lt;- list(
  `Lung-DC-T-over-N` = list(
    TME = x %&gt;% filter(anatomical_site == &#39;Lung&#39; &amp; cell_type == &#39;DC&#39; &amp; condition == &#39;TME&#39;) %&gt;% 
      pull(sample_alias),
    Normal = x %&gt;% filter(anatomical_site == &#39;Lung&#39; &amp; cell_type == &#39;DC&#39; &amp; condition == &#39;Normal&#39;) %&gt;% 
      pull(sample_alias)
  ),
  `Lung-Endothelial-T-over-N` = list(
    TME = x %&gt;% filter(anatomical_site == &#39;Lung&#39; &amp; cell_type == &#39;Endothelial cells&#39; &amp; condition == &#39;TME&#39;) %&gt;% 
      pull(sample_alias),
    Normal = x %&gt;% filter(anatomical_site == &#39;Lung&#39; &amp; cell_type == &#39;Endothelial cells&#39; &amp; condition == &#39;Normal&#39;) %&gt;% 
      pull(sample_alias)
  ),
  `Lung-Fibroblasts-T-over-N` = list(
    TME = x %&gt;% filter(anatomical_site == &#39;Lung&#39; &amp; cell_type == &#39;Fibroblasts&#39; &amp; condition == &#39;TME&#39;) %&gt;% 
      pull(sample_alias),
    Normal = x %&gt;% filter(anatomical_site == &#39;Lung&#39; &amp; cell_type == &#39;Fibroblasts&#39; &amp; condition == &#39;Normal&#39;) %&gt;% 
      pull(sample_alias)
  ),
  `Lung-Macrophages-T-over-N` = list(
    TME = x %&gt;% filter(anatomical_site == &#39;Lung&#39; &amp; cell_type == &#39;Macrophages&#39; &amp; condition == &#39;TME&#39;) %&gt;% 
      pull(sample_alias),
    Normal = x %&gt;% filter(anatomical_site == &#39;Lung&#39; &amp; cell_type == &#39;Macrophages&#39; &amp; condition == &#39;Normal&#39;) %&gt;% 
      pull(sample_alias)
  ),
  `Lung-Monocytes-T-over-N` = list(
    TME = x %&gt;% filter(anatomical_site == &#39;Lung&#39; &amp; cell_type == &#39;Monocytes&#39; &amp; condition == &#39;TME&#39;) %&gt;% 
      pull(sample_alias),
    Normal = x %&gt;% filter(anatomical_site == &#39;Lung&#39; &amp; cell_type == &#39;Monocytes&#39; &amp; condition == &#39;Normal&#39;) %&gt;% 
      pull(sample_alias)
  ),
  `Lung-Neutrophils-T-over-N` = list(
    TME = x %&gt;% filter(anatomical_site == &#39;Lung&#39; &amp; cell_type == &#39;Neutrophils&#39; &amp; condition == &#39;TME&#39;) %&gt;% 
      pull(sample_alias),
    Normal = x %&gt;% filter(anatomical_site == &#39;Lung&#39; &amp; cell_type == &#39;Neutrophils&#39; &amp; condition == &#39;Normal&#39;) %&gt;% 
      pull(sample_alias)
  ),
  `Liver-DC-T-over-N` = list(
    TME = x %&gt;% filter(anatomical_site == &#39;Liver&#39; &amp; cell_type == &#39;DC&#39; &amp; condition == &#39;TME&#39;) %&gt;% 
      pull(sample_alias),
    Normal = x %&gt;% filter(anatomical_site == &#39;Liver&#39; &amp; cell_type == &#39;DC&#39; &amp; condition == &#39;Normal&#39;) %&gt;% 
      pull(sample_alias)
  ),
  `Liver-Endothelial-T-over-N` = list(
    TME = x %&gt;% filter(anatomical_site == &#39;Liver&#39; &amp; cell_type == &#39;Endothelial cells&#39; &amp; condition == &#39;TME&#39;) %&gt;% 
      pull(sample_alias),
    Normal = x %&gt;% filter(anatomical_site == &#39;Liver&#39; &amp; cell_type == &#39;Endothelial cells&#39; &amp; condition == &#39;Normal&#39;) %&gt;% 
      pull(sample_alias)
  ),
  `Liver-Macrophages-T-over-N` = list(
    TME = x %&gt;% filter(anatomical_site == &#39;Liver&#39; &amp; cell_type == &#39;Macrophages&#39; &amp; condition == &#39;TME&#39;) %&gt;% 
      pull(sample_alias),
    Normal = x %&gt;% filter(anatomical_site == &#39;Liver&#39; &amp; cell_type == &#39;Macrophages&#39; &amp; condition == &#39;Normal&#39;) %&gt;% 
      pull(sample_alias)
  ),
  `Liver-Monocytes-T-over-N` = list(
    TME = x %&gt;% filter(anatomical_site == &#39;Liver&#39; &amp; cell_type == &#39;Monocytes&#39; &amp; condition == &#39;TME&#39;) %&gt;% 
      pull(sample_alias),
    Normal = x %&gt;% filter(anatomical_site == &#39;Liver&#39; &amp; cell_type == &#39;Monocytes&#39; &amp; condition == &#39;Normal&#39;) %&gt;% 
      pull(sample_alias)
  ),
  `Liver-Neutrophils-T-over-N` = list(
    TME = x %&gt;% filter(anatomical_site == &#39;Liver&#39; &amp; cell_type == &#39;Neutrophils&#39; &amp; condition == &#39;TME&#39;) %&gt;% 
      pull(sample_alias),
    Normal = x %&gt;% filter(anatomical_site == &#39;Liver&#39; &amp; cell_type == &#39;Neutrophils&#39; &amp; condition == &#39;Normal&#39;) %&gt;% 
      pull(sample_alias)
  )
)

comp_list_description &lt;- data.frame(
  comparison = names(comp_list),
  comparison_long = gsub(&quot;-T&quot;, &quot;-TME&quot;, names(comp_list)) %&gt;% 
  gsub(&quot;-N$&quot;, &quot;-Normal&quot;, .) %&gt;% 
  gsub(&quot;Lung-&quot;, &quot;Lung: &quot;, .) %&gt;% 
  gsub(&quot;Liver-&quot;, &quot;Liver: &quot;, .) %&gt;% 
  gsub(&quot;-&quot;, &quot; &quot;, .)
)</code></pre>
</div>
<div id="table-of-comparisons" class="section level4" number="2.2.2">
<h4><span class="header-section-number">2.2.2</span> Table of
comparisons</h4>
<p>We filter comparisons with n &lt; 3 samples in each group</p>
<pre class="r"><code>i = names(comp_list)[1]
comp_df &lt;- foreach(i = names(comp_list), .combine = rbind) %do% {
  icomp &lt;- comp_list[[i]]
  data.frame(
    comparison = i,
    `Sample Group (n)` =  length(icomp[[1]]),
    `Reference Group (n)` = length(icomp[[2]]),
    check.names = FALSE
  )
}

# filter comparisons were we have at least 3 samples in each group
comp_df %&lt;&gt;% 
  mutate(Evaluated = `Sample Group (n)` &gt;= 3 &amp; `Reference Group (n)` &gt;= 3)
kepp_comp &lt;- comp_df %&gt;% filter(Evaluated) %&gt;% pull(comparison)
comp_list &lt;- comp_list[kepp_comp]

# Add long names
comp_df &lt;- comp_df %&gt;% left_join(comp_list_description) %&gt;% 
  mutate(comparison = comparison_long) %&gt;%
  dplyr::select(-comparison_long)

comp_df %&gt;%
  datatable(., 
            rownames = FALSE, 
            filter = &#39;top&#39;, 
            caption = &#39;List of comparisons for differential expression. Comparisons with n &lt; 3 samples in each group won\&#39;t be evaluated&#39;,
            extensions = &#39;Buttons&#39;, 
            options = list(
              dom = &#39;Blfrtip&#39;,
              buttons = c(&#39;csv&#39;, &#39;excel&#39;)
            ))</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-0e41329c1963262c826a" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-0e41329c1963262c826a">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"1\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"1\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"logical\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;true&quot;,&quot;false&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>List of comparisons for differential expression. Comparisons with n &lt; 3 samples in each group won't be evaluated<\/caption>","data":[["Lung: DC TME over Normal","Lung: Endothelial TME over Normal","Lung: Fibroblasts TME over Normal","Lung: Macrophages TME over Normal","Lung: Monocytes TME over Normal","Lung: Neutrophils TME over Normal","Liver: DC TME over Normal","Liver: Endothelial TME over Normal","Liver: Macrophages TME over Normal","Liver: Monocytes TME over Normal","Liver: Neutrophils TME over Normal"],[3,3,3,3,3,3,3,3,3,3,3],[3,3,3,3,3,3,3,3,3,3,3],[true,true,true,true,true,true,true,true,true,true,true]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>comparison<\/th>\n      <th>Sample Group (n)<\/th>\n      <th>Reference Group (n)<\/th>\n      <th>Evaluated<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"className":"dt-right","targets":[1,2]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<div id="data-generation" class="section level2" number="3">
<h2><span class="header-section-number">3</span> Data generation</h2>
<div id="run-differential-expression-analysis" class="section level3"
number="3.1">
<h3><span class="header-section-number">3.1</span> Run differential
expression analysis</h3>
<p>Get Ensembl gene info</p>
<pre class="r"><code>ensembl &lt;-  biomaRt::useEnsembl(biomart=&quot;genes&quot;, dataset=&quot;mmusculus_gene_ensembl&quot;,
                                mirror = &#39;useast&#39;)
gene_desc &lt;- biomaRt::getBM(attributes=c(&#39;external_gene_name&#39;,&#39;description&#39;),
                              filters = &#39;external_gene_name&#39;,
                              values = rowData(se)$gene_name, mart =ensembl) %&gt;%
    dplyr::rename(&#39;gene_name&#39; = &#39;external_gene_name&#39;) %&gt;%
    unique

gene_desc &lt;- gene_desc[!duplicated(gene_desc$gene_name),]
saveRDS(gene_desc, file = file.path(dirname(params$output_dir), &#39;ensembl_gene_desc.rds&#39;))</code></pre>
<pre class="r"><code>gene_desc &lt;- readRDS(file = file.path(dirname(params$output_dir), &#39;ensembl_gene_desc.rds&#39;))</code></pre>
<p>Run differential expression analysis using EdgeR QLF</p>
<pre class="r"><code>use_comp &lt;-  comp_list[[1]]
dge_list &lt;- foreach(use_comp = comp_list) %do%{
  se_cols &lt;- se$sample_alias %in% unlist(use_comp)
  use_se &lt;- se[,se_cols]
  use_se$group &lt;- ifelse(
    use_se$sample_alias %in% use_comp[[1]],
    names(use_comp)[1],
    names(use_comp)[2]
  )
  
  if(is.null(use_comp$batch)) {
    dge &lt;- edgeR_dge(
      use_se,
      # Design configuration for differential expression
      group_var =  &#39;group&#39;,
      group_sample = names(use_comp)[1],
      group_ref = names(use_comp)[2],
      design_formula = &quot;~ group&quot;,
      coef = &#39;last&#39;,
      # Conversion from SingleCellExperiment/Summarized Experiment to DGEList
      assay_to_DGEList = &#39;counts&#39;,
      assay_to_row_filter = &quot;counts&quot;,
      # Feature filtering parameters
      use_filterByExpr = TRUE,
      min_counts = params$min_counts,
      min_present_prop = params$min_present_prop,
      # EdgeR workflow configuration
      run_calcNormFactors = &#39;TMM&#39;,
      estimateDisp_robust = TRUE,
      estimateDisp_trend.method = &quot;locfit&quot;,
      glm_approach = &quot;QLF&quot;,
      # Output configuration
      adjust_method = &#39;BH&#39;
      )
    } else {
      design_formula = paste0(&quot;~ &quot;, use_comp$batch,&quot; + group&quot;)
      dge &lt;- edgeR_dge(
        use_se,
        # Design configuration for differential expression
        group_var =  &#39;group&#39;,
        group_sample = names(use_comp)[1],
        group_ref = names(use_comp)[2],
        batch_vars = use_comp$batch,
        design_formula = design_formula,
        coef = &#39;last&#39;,
        # Conversion from SingleCellExperiment/Summarized Experiment to DGEList
        assay_to_DGEList = &#39;counts&#39;,
        assay_to_row_filter = &quot;counts&quot;,
        # Feature filtering parameters
        use_filterByExpr = TRUE,
        min_counts = params$min_counts,
        min_present_prop = params$min_present_prop,
        # EdgeR workflow configuration
        run_calcNormFactors = &#39;TMM&#39;,
        estimateDisp_robust = TRUE,
        estimateDisp_trend.method = &quot;locfit&quot;,
        glm_approach = &quot;QLF&quot;,
        # Output configuration
        adjust_method = &#39;BH&#39;
        )
    }
  
  # Add gene description
  use_res &lt;- dge$results %&gt;%  left_join(., gene_desc, relationship = &quot;many-to-many&quot;)
  dge$results &lt;- use_res %&gt;% 
    filter(!duplicated(feature)) %&gt;% 
    mutate(rownames = feature) %&gt;% 
    column_to_rownames(&#39;rownames&#39;)
  
   return(dge)
}

names(dge_list) &lt;- names(comp_list)

# Save objects
saveRDS(dge_list, file = file.path(params$output_dir, &#39;dge-edgeR_QLF.rds&#39;))

# ###########
# object = use_se
# # Design configuration for differential expression
# group_var =  &#39;group&#39;
# group_sample = names(use_comp)[1]
# group_ref = names(use_comp)[2]
# design_formula = &quot;~ group&quot;
# coef = &#39;last&#39;,
# # Conversion from SingleCellExperiment/Summarized Experiment to DGEList
# assay_to_DGEList = &#39;counts&#39;
# assay_to_row_filter = &quot;counts&quot;
# # Feature filtering parameters
# use_filterByExpr = TRUE
# min_counts = params$min_count
# min_present_prop = params$min_present_prop,
# # EdgeR workflow configuration
# run_calcNormFactors = &#39;TMM&#39;
# estimateDisp_robust = TRUE
# estimateDisp_trend.method = &quot;locfit&quot;
# glm_approach = &quot;QLF&quot;
# # Output configuration
# adjust_method = &#39;BH&#39;
# 
# use_samples = NULL,
# 
# numeric_covar = NULL,
# batch_vars = NULL,
# contrast = NULL,
# # Conversion from SingleCellExperiment to DGEList
# spike_normalization = FALSE,
# use_colData = NULL,
# use_rowData = NULL,
# # Feature filtering parameters
# use_filterByExpr = TRUE,
# min_counts = 1,
# min_present_prop = 0.40,
# # EdgeR workflow configuration
# run_calcNormFactors = &#39;TMM&#39;,
# estimateDisp_robust = FALSE,
# estimateDisp_trend.method = &quot;locfit&quot;,
# glmQLFit_robust = FALSE,
# glm_approach = &quot;QLF&quot;,
# # Output configuration
# adjust_method = &#39;BH&#39;,
# assays_from_SingleCellExperiment = NULL

# 
# for(use_comp in comp_list) {
#   
#   se_cols &lt;- se$sample_alias %in% unlist(use_comp)
#   use_se &lt;- se[,se_cols]
#   use_se$group &lt;- ifelse(
#     use_se$sample_alias %in% use_comp[[1]],
#     names(use_comp)[1],
#     names(use_comp)[2]
#   )
#   
#   if(is.null(use_comp$batch)) {
#     dge &lt;- edgeR_dge(
#       use_se,
#       # Design configuration for differential expression
#       group_var =  &#39;group&#39;,
#       group_sample = names(use_comp)[1],
#       group_ref = names(use_comp)[2],
#       design_formula = &quot;~ group&quot;,
#       coef = &#39;last&#39;,
#       # Conversion from SingleCellExperiment/Summarized Experiment to DGEList
#       assay_to_DGEList = &#39;counts&#39;,
#       assay_to_row_filter = &quot;counts&quot;,
#       # Feature filtering parameters
#       use_filterByExpr = TRUE,
#       min_counts = params$min_counts,
#       min_present_prop = params$min_present_prop,
#       # EdgeR workflow configuration
#       run_calcNormFactors = &#39;TMM&#39;,
#       estimateDisp_robust = TRUE,
#       estimateDisp_trend.method = &quot;locfit&quot;,
#       glm_approach = &quot;QLF&quot;,
#       # Output configuration
#       adjust_method = &#39;BH&#39;
#       )
#     } else {
#       design_formula = paste0(&quot;~ &quot;, use_comp$batch,&quot; + group&quot;)
#       dge &lt;- edgeR_dge(
#         use_se,
#         # Design configuration for differential expression
#         group_var =  &#39;group&#39;,
#         group_sample = names(use_comp)[1],
#         group_ref = names(use_comp)[2],
#         batch_vars = use_comp$batch,
#         design_formula = design_formula,
#         coef = &#39;last&#39;,
#         # Conversion from SingleCellExperiment/Summarized Experiment to DGEList
#         assay_to_DGEList = &#39;counts&#39;,
#         assay_to_row_filter = &quot;counts&quot;,
#         # Feature filtering parameters
#         use_filterByExpr = TRUE,
#         min_counts = params$min_counts,
#         min_present_prop = params$min_present_prop,
#         # EdgeR workflow configuration
#         run_calcNormFactors = &#39;TMM&#39;,
#         estimateDisp_robust = TRUE,
#         estimateDisp_trend.method = &quot;locfit&quot;,
#         glm_approach = &quot;QLF&quot;,
#         # Output configuration
#         adjust_method = &#39;BH&#39;
#         )
#     }
#   
#   # Add gene description
#   use_res &lt;- dge$results %&gt;%  left_join(., gene_desc, relationship = &quot;many-to-many&quot;)
#   dge$results &lt;- use_res %&gt;% 
#     filter(!duplicated(feature)) %&gt;% 
#     mutate(rownames = feature) %&gt;% 
#     column_to_rownames(&#39;rownames&#39;)
# }</code></pre>
</div>
<div id="run-functional-enrichment" class="section level3" number="3.2">
<h3><span class="header-section-number">3.2</span> Run functional
enrichment</h3>
<pre class="r"><code>dge_list &lt;- readRDS(file.path(params$output_dir, &#39;dge-edgeR_QLF.rds&#39;))

gse_list &lt;- foreach(dge = dge_list) %do%{
  gse_res &lt;- gse_omnibus(
    feature_names = dge$results$gene_name,
    p = dge$results$FDR,
    fc = dge$results$logFC,
    fc_thrs = 0.5,
    gmt_files = gmt_files_symbols,
    
    run_enricher = FALSE,
    run_enrichGO = FALSE,
    run_enrichKEGG = FALSE,
    run_enrichReactome = FALSE,
    
    run_GSEA = TRUE,
    run_gseGO = TRUE,
    run_gseKEGG = FALSE,
    run_gseReactome = FALSE,
    
    go_ontologies = c(&#39;BP&#39;),
    go_simplify_p = 0.05,
    args_gse = list(minGSSize = 10, maxGSSize = 250, 
                    pvalueCutoff = 1, 
                    BPPARAM = BiocParallel::MulticoreParam(1)),
    annot_db = &#39;org.Mm.eg.db&#39;,
    organism = &#39;mmusculus&#39;
    )
  
   return(gse_res)

}
names(gse_list) &lt;- names(dge_list)

# Save objects
saveRDS(gse_list, file = file.path(params$output_dir,&#39;gse_omnibus-edgeR_QLF.rds&#39;))


# 
# gse_list &lt;- list()
# for(dge_names in names(dge_list)) {
# dge &lt;- dge_list[[dge_names]]
# cat(&#39;######&#39;, dge_names, &quot;\n\n\n&quot;)
# cat(&#39;######&#39;, dge$comparison, &quot;\n\n\n&quot;)
# gse_list[[dge_names]] &lt;- gse_omnibus(
#     feature_names = dge$results$gene_name,
#     p = dge$results$FDR,
#     fc = dge$results$logFC,
#     fc_thrs = 0.5,
#     gmt_files = gmt_files_symbols,
#     
#     run_enricher = FALSE,
#     run_enrichGO = FALSE,
#     run_enrichKEGG = FALSE,
#     run_enrichReactome = FALSE,
#     
#     run_GSEA = TRUE,
#     run_gseGO = TRUE,
#     run_gseKEGG = FALSE,
#     run_gseReactome = FALSE,
#     
#     go_ontologies = c(&#39;BP&#39;, &#39;MF&#39;),
#     go_simplify_p = 0.05,
#     args_gse = list(minGSSize = 10, maxGSSize = 250, 
#                     pvalueCutoff = 1, 
#                     BPPARAM = BiocParallel::MulticoreParam(1))
#     )
# 
# }
# # Save objects
# saveRDS(gse_list, file = file.path(params$output_dir,&#39;gse_omnibus-edgeR_QLF.rds&#39;))</code></pre>
</div>
<div id="load-data" class="section level3" number="3.3">
<h3><span class="header-section-number">3.3</span> Load data</h3>
<pre class="r"><code>dge_list &lt;- readRDS(file = file.path(params$output_dir, &#39;dge-edgeR_QLF.rds&#39;))
gse_list &lt;- readRDS(file = file.path(params$output_dir, &#39;gse_omnibus-edgeR_QLF.rds&#39;))</code></pre>
</div>
</div>
<div id="mf-gene-set-enrichment-analysis" class="section level2"
number="4">
<h2><span class="header-section-number">4</span> MF: Gene set enrichment
analysis</h2>
<p>Gene set enrichment analysis</p>
<pre class="r"><code>use_gse_list &lt;- gse_list[c(&quot;Lung-Macrophages-T-over-N&quot;, &quot;Liver-Macrophages-T-over-N&quot;)]
names(use_gse_list) &lt;- c(&#39;Lung&#39;, &#39;Liver&#39;)
test_type &lt;- &#39;GSEA&#39;
gset_collection &lt;- &#39;msigdb.m2.cp.reactome&#39;
p_thrs &lt;- 0.01
topn &lt;- 20

# Load all the results
gse_df &lt;- foreach(i = names(use_gse_list), .combine = rbind) %do% {
  i_1 &lt;- use_gse_list[[i]][[test_type]][[gset_collection]]@result %&gt;% 
    mutate(group = i)
  i_2 &lt;- use_gse_list[[i]][[&#39;gseGO&#39;]][[&#39;BP&#39;]]@result %&gt;% 
    filter(simplify) %&gt;% 
    dplyr::select(-simplify) %&gt;% 
    mutate(group = i)
  rbind(i_1, i_2)
} %&gt;% 
  mutate(
    group = factor(group, c(&#39;Lung&#39;, &#39;Liver&#39;)),
    Description = clean_msigdb_names(Description) %&gt;% tolower() %&gt;% str_to_title,
    Description = ifelse(grepl(&quot;^GO:&quot;, ID), paste0(&quot;GO:&quot;, Description), Description)
  ) 

# Select topn by group in selected gset
top_by_group &lt;- foreach(i = names(use_gse_list), .combine = c) %do% {
  gse_df %&gt;% 
    filter(group == i) %&gt;% 
    filter(p.adjust &lt; p_thrs) %&gt;% 
    arrange(p.adjust) %&gt;% pull(Description) %&gt;%
    head(topn)
}

# Data frame to plot
res_top &lt;- gse_df %&gt;% 
  filter(Description %in% top_by_group) %&gt;% 
  mutate(log.p.adjust = -log10(p.adjust))

# Rows and column orders by hclust
use_mat &lt;- res_top %&gt;% 
  dplyr::select(Description, log.p.adjust, group) %&gt;% 
  pivot_wider(names_from = group, values_from = log.p.adjust) %&gt;% 
  column_to_rownames(&#39;Description&#39;)
use_mat[is.na(use_mat)] &lt;- 0

row_hclust &lt;- hclust(dist(use_mat))
row_hclust_labels &lt;- row_hclust$labels[row_hclust$order]

col_hclust &lt;- hclust(dist(t(use_mat)))
col_hclust_labels &lt;- col_hclust$labels[col_hclust$order]

use_mat &lt;- use_mat[row_hclust_labels,col_hclust_labels]

# GSET clusters by kmeans using silhouette
use_kmax &lt;- ifelse(nrow(use_mat) &lt;= 20, nrow(use_mat)-2, 20)
nb_clust &lt;- fviz_nbclust(use_mat, kmeans, method = &quot;silhouette&quot;, k.max = use_kmax)
silhouette_best_k &lt;- nb_clust$data %&gt;% slice_max(y) %&gt;% pull(clusters) %&gt;% as.character %&gt;% as.numeric
# silhouette_best_k &lt;- 5
kmeans_mat &lt;- kmeans(use_mat, silhouette_best_k, nstart = 25)
kmeans_mat_df &lt;- kmeans_mat$cluster %&gt;%
  data.frame %&gt;%
  set_names(&#39;kcluster&#39;) %&gt;%
  rownames_to_column(&#39;Description&#39;) %&gt;%
  mutate(kcluster = as.character(kcluster))


# Order factors by hclust
res_top %&lt;&gt;% 
  left_join(kmeans_mat_df) %&gt;% 
  mutate(
    Description = factor(Description, levels = row_hclust_labels),
    group = fct_relevel(group, col_hclust_labels)
  )

# Calculate jaccard similarity matrix
gset_collection &lt;- foreach(i = levels(res_top$Description)) %do% {
  res_top %&gt;% filter(Description == i) %&gt;% pull(core_enrichment) %&gt;% 
    paste(collapse = &#39;/&#39;) %&gt;% 
    strsplit(&#39;/&#39;) %&gt;% unlist %&gt;% unique
}
names(gset_collection) &lt;- levels(res_top$Description)

use_terms &lt;- res_top$Description %&gt;% unique
jc_matrix &lt;- pairwise_terms_jc(gset_collection[use_terms])</code></pre>
<div id="results-table" class="section level4" number="4.0.1">
<h4><span class="header-section-number">4.0.1</span> Results table</h4>
<pre class="r"><code>gse_df %&gt;% 
  filter(p.adjust &lt; p_thrs) %&gt;% 
  arrange(desc(p.adjust)) %&gt;% 
  dplyr::select(group, Description:p.adjust, Count:GeneRatio, core_enrichment)  %&gt;%
  dplyr::rename(site = group) %&gt;% 
  datatable(., 
            rownames = FALSE, 
            filter = &#39;top&#39;, 
            caption = &#39;Results from gene-set enrichment analysis&#39;,
            extensions = &#39;Buttons&#39;, 
            options = list(
              dom = &#39;Blfrtip&#39;,
              buttons = c(&#39;csv&#39;, &#39;excel&#39;)
            ))  %&gt;% 
  formatRound(c(&#39;enrichmentScore&#39;, &#39;NES&#39;, &#39;GeneRatio&#39;), digits = 2) %&gt;%
  formatSignif(c(&#39;pvalue&#39;, &#39;p.adjust&#39;), digits = 3)</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-cf74d9caecbce7389320" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-cf74d9caecbce7389320">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Lung&quot;,&quot;Liver&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"13\" data-max=\"239\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-0.781660207372838\" data-max=\"0.83251999237144\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-2.22667597568631\" data-max=\"3.02009903701016\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"1e-10\" data-max=\"0.0006991214438\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"9.1e-09\" data-max=\"0.009787700213198\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"5\" data-max=\"150\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.13095238095238\" data-max=\"0.85\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Results from gene-set enrichment analysis<\/caption>","data":[["Liver","Liver","Liver","Lung","Liver","Lung","Liver","Liver","Liver","Liver","Lung","Liver","Lung","Lung","Liver","Liver","Liver","Lung","Lung","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Lung","Liver","Lung","Lung","Lung","Lung","Liver","Lung","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Lung","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Lung","Lung","Lung","Lung","Lung","Lung","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Lung","Liver","Liver","Liver","Lung","Lung","Lung","Liver","Liver","Lung","Liver","Liver","Liver","Liver","Liver","Lung","Lung","Liver","Lung","Liver","Liver","Liver","Liver","Lung","Lung","Liver","Liver","Lung","Liver","Liver","Liver","Lung","Lung","Liver","Liver","Lung","Lung","Liver","Liver","Liver","Liver","Liver","Liver","Lung","Liver","Lung","Liver","Liver","Liver","Liver","Lung","Liver","Liver","Liver","Lung","Liver","Liver","Liver","Lung","Liver","Lung","Lung","Liver","Lung","Liver","Liver","Liver","Liver","Lung","Lung","Lung","Liver","Liver","Liver","Lung","Liver","Liver","Liver","Liver","Liver","Liver","Lung","Lung","Lung","Lung","Lung","Lung","Lung","Lung","Lung","Lung","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver"],["Uch Proteinases","Antigen Activates B Cell Receptor Bcr Leading To Generation Of Second Messengers","Bile Acid And Bile Salt Metabolism","GO:Negative Regulation Of Peptidase Activity","GO:Regulation Of Neuron Apoptotic Process","Activation Of Atr In Response To Replication Stress","Assembly Of The Pre Replicative Complex","GO:Multicellular Organismal Signaling","GO:Nadh Metabolic Process","GO:Regulation Of Neuron Death","GO:Cell Chemotaxis","Transcriptional Regulation By Runx2","GO:Striated Muscle Contraction","GO:Cell Cycle G2/M Phase Transition","G1 S Dna Damage Checkpoints","GO:Protein Polyubiquitination","GO:Regulation Of Digestive System Process","GO:Positive Regulation Of Angiogenesis","GO:Positive Regulation Of Vasculature Development","GO:Cardiac Chamber Development","GO:Cellular Response To Interferon-Beta","Gli3 Is Processed To Gli3r By The Proteasome","GO:Glial Cell Migration","GO:Positive Regulation Of Inflammatory Response","GO:Cellular Response To Hypoxia","Developmental Biology","GO:Eye Development","GO:Antimicrobial Humoral Response","Stabilization Of P53","GO:Regulation Of Meiotic Cell Cycle","GO:Collagen Catabolic Process","GO:Chromosome Condensation","GO:Kinetochore Organization","Orc1 Removal From Chromatin","Activation Of Apc C And Apc C Cdc20 Mediated Degradation Of Mitotic Proteins","GO:Regulation Of Chemotaxis","GO:Regulation Of Gtpase Activity","Ca Dependent Events","Regulation Of Runx3 Expression And Activity","GO:Regulation Of Collagen Biosynthetic Process","GO:Nadh Regeneration","GO:Glucose Catabolic Process To Pyruvate","Degradation Of The Extracellular Matrix","Dectin 1 Mediated Noncanonical Nf Kb Signaling","Mitotic Metaphase And Anaphase","GO:Cellular Response To Biotic Stimulus","Tnfr2 Non Canonical Nf Kb Pathway","GO:Cardiac Conduction","Activation Of Apc C And Apc C Cdc20 Mediated Degradation Of Mitotic Proteins","GO:Positive Regulation Of Epithelial Cell Apoptotic Process","GO:Cell Surface Receptor Signaling Pathway Involved In Cell-Cell Signaling","GO:Production Of Molecular Mediator Involved In Inflammatory Response","GO:Negative Regulation Of Cell Adhesion","GO:Female Gamete Generation","Apc C Mediated Degradation Of Cell Cycle Proteins","GO:Regulation Of Cardiac Muscle Contraction","GO:Zymogen Activation","GO:Humoral Immune Response","GO:Sodium Ion Transmembrane Transport","GO:Cellular Response To Molecule Of Bacterial Origin","GO:Vesicle-Mediated Transport In Synapse","GO:Ribose Phosphate Biosynthetic Process","GO:Collagen Metabolic Process","GO:Glycolytic Process Through Fructose-6-Phosphate","GO:Glycolytic Process Through Glucose-6-Phosphate","GO:Antimicrobial Humoral Immune Response Mediated By Antimicrobial Peptide","GO:Mitochondrial Gene Expression","Gaba B Receptor Activation","Gaba Receptor Activation","Cdc42 Gtpase Cycle","Scf Skp2 Mediated Degradation Of P27 P21","Ubiquitin Mediated Degradation Of Phosphorylated Cdc25a","Regulation Of Runx2 Expression And Activity","Hedgehog Ligand Biogenesis","Cellular Response To Hypoxia","GO:Circulatory System Process","Potassium Channels","Degradation Of Dvl","Metabolism Of Polyamines","GO:Peptidyl-Tyrosine Phosphorylation","GO:Regulation Of Neurotransmitter Transport","GO:Myeloid Leukocyte Migration","GO:Response To Interleukin-1","GO:Signal Release From Synapse","GO:Protein Folding","Peptide Ligand Binding Receptors","Chemokine Receptors Bind Chemokines","Cyclin A B1 B2 Associated Events During G2 M Transition","GO:Synapse Organization","GO:Dendrite Development","GO:Acute Inflammatory Response","GO:Synapse Assembly","GO:Regulation Of Neurotransmitter Secretion","GO:Sensory Perception Of Chemical Stimulus","GO:Nadh Dehydrogenase Complex Assembly","GO:Mitochondrial Respiratory Chain Complex I Assembly","GO:Chromosome Localization","Regulation Of Apc C Activators Between G1 S And Early Anaphase","GO:Collagen Biosynthetic Process","Cdc42 Gtpase Cycle","Auf1 Hnrnp D0 Binds And Destabilizes Mrna","Regulation Of Mrna Stability By Proteins That Bind Au Rich Elements","The Role Of Gtse1 In G2 M Progression After G2 Checkpoint","GO:Regulation Of Small Gtpase Mediated Signal Transduction","GO:Collagen Metabolic Process","GO:Cardiac Muscle Contraction","GO:Proton Motive Force-Driven Atp Synthesis","Complex I Biogenesis","GO:Protein Localization To Chromosome, Centromeric Region","Rac1 Gtpase Cycle","GO:Cellular Response To Interleukin-1","GO:Forebrain Development","GO:Regulation Of Mitotic Nuclear Division","GO:Response To Chemokine","GO:Regulation Of Neurotransmitter Levels","GO:Neurotransmitter Transport","GO:Response To Interleukin-1","GO:Negative Regulation Of Chromosome Organization","GO:Atp Metabolic Process","Processing Of Capped Intron Containing Pre Mrna","Transmission Across Chemical Synapses","GO:Ras Protein Signal Transduction","GO:Central Nervous System Neuron Differentiation","GO:Nucleoside Triphosphate Metabolic Process","GO:Chemokine-Mediated Signaling Pathway","GO:Ribonucleoprotein Complex Subunit Organization","Rho Gtpase Effectors","GO:Ribonucleoprotein Complex Assembly","Peptide Ligand Binding Receptors","Gpcr Ligand Binding","Mrna Splicing","GO:Cellular Response To Interleukin-1","GO:Regulation Of Leukocyte Migration","GO:Cell Junction Assembly","Chemokine Receptors Bind Chemokines","GO:Regulation Of Nuclear Division","GO:Atp Synthesis Coupled Electron Transport","GO:Mitochondrial Atp Synthesis Coupled Electron Transport","GO:Ribosome Assembly","Rho Gtpases Activate Formins","Neuronal System","GO:Mitotic Cell Cycle Checkpoint Signaling","GO:Negative Regulation Of Mitotic Cell Cycle Phase Transition","GO:Aerobic Respiration","GO:Microtubule Cytoskeleton Organization Involved In Mitosis","GO:Leukocyte Migration","GO:Ribosomal Large Subunit Biogenesis","GO:Response To Chemokine","GO:Cellular Response To Chemokine","Separation Of Sister Chromatids","GO:Regulation Of Chromosome Separation","Mitotic Metaphase And Anaphase","Respiratory Electron Transport Atp Synthesis By Chemiosmotic Coupling And Heat Production By Uncoupling Proteins","Class A 1 Rhodopsin Like Receptors","The Citric Acid Tca Cycle And Respiratory Electron Transport","Mitotic Prometaphase","GO:Rrna Metabolic Process","GO:Cytoplasmic Translation","GO:Cell Chemotaxis","GO:Leukocyte Chemotaxis","GO:Rrna Processing","Respiratory Electron Transport","GO:Meiotic Nuclear Division","GO:Meiotic Cell Cycle Process","GO:Mitotic Sister Chromatid Segregation","GO:Regulation Of Chromosome Segregation","GO:Sister Chromatid Segregation","GO:Meiotic Cell Cycle","GO:Nuclear Chromosome Segregation","Amplification Of Signal From The Kinetochores","Mitotic Spindle Checkpoint","Resolution Of Sister Chromatid Cohesion","Mitochondrial Translation","Formation Of A Pool Of Free 40s Subunits","Eukaryotic Translation Initiation","Translation","Nonsense Mediated Decay Nmd Independent Of The Exon Junction Complex Ejc","Srp Dependent Cotranslational Protein Targeting To Membrane","Major Pathway Of Rrna Processing In The Nucleolus And Cytosol","Nonsense Mediated Decay Nmd","Activation Of The Mrna Upon Binding Of The Cap Binding Complex And Eifs And Subsequent Binding To 43s"],[78,20,17,112,163,33,78,33,28,228,191,60,73,126,58,192,18,112,112,86,41,55,43,90,75,183,160,48,53,30,19,34,21,64,75,119,232,19,53,23,13,13,35,56,173,168,75,19,66,25,239,61,167,86,85,35,44,91,52,147,118,158,50,15,15,29,146,18,18,64,55,48,52,54,66,227,19,52,52,171,51,167,64,71,135,44,25,23,199,159,60,52,46,17,49,49,70,35,25,72,54,82,62,142,60,63,60,55,24,122,53,172,86,46,89,83,74,85,157,224,67,216,79,191,37,156,211,149,34,73,173,62,141,172,18,100,53,53,57,122,89,122,151,148,129,232,67,41,41,175,95,212,112,57,148,181,225,131,169,140,198,94,112,123,169,110,186,163,233,90,107,111,88,92,110,216,84,83,166,104,58],[0.4669032308266243,-0.6734680686707561,-0.7158053492457199,0.5786982230106605,0.3977196746969982,0.725184905559727,0.4738065253110475,-0.6157047376347056,0.6678357084988916,0.3686374240211357,0.5229706167002264,0.517130910224584,-0.4965437554438171,0.5631197778971522,0.5316167532956708,-0.3594524294637572,-0.7216295008993543,0.5838610494449823,0.5838610494449823,-0.462386114593183,-0.5937472502874791,0.5421729296295708,0.5891300312954778,0.4763919388421346,0.5111767581844798,-0.3646919776415108,-0.3893527608367578,0.6972438840294062,0.5335111553752463,0.762029589577217,0.8208986715274619,0.7404556413338398,0.8111806282960143,0.5145321923916498,0.627962849748904,0.4486737673611058,-0.3526212884889288,-0.6994710522348436,0.5378394330951987,0.7082600153277678,0.8060535963131252,0.8060535963131252,0.7407145181090584,0.5288889744588321,0.3895190421029779,0.4043726210558433,0.4960901295703722,-0.7224323953240254,0.5055284279900852,0.6910170137386838,-0.352818578821066,0.5513513476046168,0.4155708822369823,0.6291643660596332,0.6161945068137018,-0.640891209774658,0.7160873348158794,0.6230885121795773,-0.5588257840998151,0.4315188778465758,-0.443668591753746,0.4234150151732131,0.581065273051243,0.7977169757721141,0.7977169757721141,0.6698633082461094,0.4381260149977068,-0.7451934011707315,-0.7451934011707315,-0.5185331711856794,0.56814629731057,0.5796186252234279,0.5650171414912708,0.5624302306357585,0.523341892313135,-0.3639479750469224,-0.7297412156400375,0.5688282658670076,0.57108791340625,-0.3991264401720017,-0.5780049083594017,0.5560582589577213,0.5559609362133879,-0.5358851480585745,0.4562273734370697,0.7343725587340447,0.8043012973011443,0.8145880144708034,-0.3884241572497278,-0.4202731908283234,0.6869007070916801,-0.5936944826443264,-0.6104529651727016,-0.7816602073728375,0.6158040371121234,0.6158040371121234,0.6732180142998472,0.7719893720933773,0.7353426524731689,-0.5434734157958994,0.5867045450093453,0.520015286688839,0.5740679755579389,-0.4369493054873618,0.6958472011366782,-0.5577501676896861,0.5964248730702544,0.6021669181009583,0.8325199923714395,-0.4523478314128415,0.6150251646519085,-0.4266801311670175,0.6639906221392132,0.7393836192587872,-0.5119928281133117,-0.5399432178386622,0.6833691724199884,0.6667212145750984,0.4598370219180883,0.4117826302551044,-0.5538547366288725,-0.4059695946010995,-0.5648441801133604,0.4516289328709305,0.7867508566385608,0.4832197996961512,0.5602483928977859,0.4862330193622605,0.7111111303284774,0.5796819902105156,0.4466381141965146,0.7304933741477199,0.4966400799246576,-0.4561550598959807,0.8215199903561916,0.6722440022310274,0.6690804850600455,0.6690804850600455,0.6537650553712674,0.6462364583733471,-0.5478025540247701,0.6557293312689689,0.6281887777636667,0.508129530955941,0.6548257760690301,0.459417922549447,0.6505036488919022,0.737629361888422,0.737629361888422,0.6175036958166802,0.7084265713452588,0.6027037262607635,0.5654247188316279,0.659320042224123,0.5252714403840195,0.6280325113152466,0.4871418648673388,0.6861757612686807,0.5383230260083087,0.5556358244482049,0.5241563375578245,0.6051365193789817,0.7432166546559538,0.7320368488487378,0.7050169631714229,0.7232858717369156,0.6857749007668281,0.6881057775272551,0.6603151017272968,0.7448860032659526,0.7219090343786962,0.6993870027693544,0.6248232806550584,0.7877976272284304,0.755144815869877,0.6804986137104474,0.7742716058556531,0.7709323308644032,0.6639682459909648,0.7096031280009956,0.735111515407106],[1.742351434583178,-1.982264795983596,-1.993100404427512,1.736434177155437,1.674457949497192,1.826639222448042,1.76811258647538,-2.00516759361969,1.957481862358206,1.612110400365614,1.632978813865071,1.83145162683007,-1.911148757190252,1.706284830876601,1.874854023444622,-1.612326845936743,-2.053331838266699,1.751925685362665,1.751925685362665,-1.84470253203404,-2.019138417361698,1.893332828811678,1.944320155027282,1.826551322716806,1.905081622838356,-1.64817428148769,-1.691163404751853,1.871138496741573,1.848158927085064,1.908000982001085,1.8902910749444,1.905999969268156,1.886463227025189,1.850086217273537,1.792045261532756,1.783709731965129,-1.608265783908195,-2.007432474078666,1.863152699992037,2.004730239040735,2.014380886286198,2.014380886286198,1.896995778865754,1.858136589126379,1.640790760206555,1.703362842387298,1.827799970381095,-2.067175385228643,1.826137313599774,1.982637421679314,-1.621800871260163,1.967077603312066,1.747080751974345,1.814440147836401,1.795519781419557,-2.090513074777408,1.90607649470165,1.810993850424202,-2.02207980527325,1.775137494548106,-1.861515903093806,1.772570773662803,1.988848868320222,2.046307903029299,2.046307903029299,1.992233016582633,1.804828779775779,-2.124014428010168,-2.124014428010168,-1.984968859872226,1.984034940661546,1.971787988758805,1.966408829772395,1.954563092141173,1.890485488862901,-1.669823721144037,-2.094305703243913,1.97967254882352,1.987536718858454,-1.753156114342711,-2.055352570763034,1.7211177208876,1.999095477738416,-2.048856381662659,1.834482064066342,1.963379216953429,1.943519762907332,1.922345428865484,-1.759329556129535,-1.826150367065764,1.896376908500789,-2.127948895599093,-2.122248834890916,-2.189580202374876,2.08644927560537,2.08644927560537,1.886076112101394,1.977091773398645,2.109814710150625,-2.068241632811235,2.03892142918862,1.954486787618549,2.056063115901298,-1.868734002896103,1.921076147479313,-2.044224361427152,2.11454358752068,2.102839024522767,1.999626926017001,-1.917261648948882,2.117114011396852,-1.880355092970295,1.914875202074092,1.974107435885836,-2.049823588362193,-2.130198373564085,1.922790395129319,1.920148361561761,1.923698943626984,1.799764549688462,-2.147402470968879,-1.855436913991698,-2.20608797270383,1.93128614507186,2.04563309276852,2.023942879495262,1.765956167995336,2.010260690396286,2.259830171684067,2.121062848155795,1.881396316270409,2.022547064690198,2.022624753736386,-2.010249429036175,2.219659098554715,1.980773509219961,2.303189773502268,2.303189773502268,2.293512980891038,1.945729906353146,-2.226675975686314,1.982367397124899,1.926624759555097,2.094494396900521,1.987068175282458,2.021712872523729,2.362113629132958,2.406108712502289,2.406108712502289,1.920557813533412,2.070491751558751,1.898239754252445,2.226269597429567,2.317893657490989,2.170647395439099,1.961536899961078,2.124886869208529,2.757756669573485,2.273429718555187,2.265123952169451,2.25130978315539,2.313209666723691,2.230085991039848,2.210432693325683,2.186129121897087,2.162659305849947,2.143113059958338,2.125929778232838,2.08325992011692,2.182592417178059,2.139865113180087,2.072208232161702,2.377457083266282,3.020099037010162,2.963657173536771,2.958750138524622,2.915076017810457,2.900597835143605,2.777522082315958,2.754257743467832,2.592519467826014],[0.0006991214437998186,0.0006438008755598834,0.0006138572631734088,0.0001582547320305132,0.0002547084216499583,0.0001823474588843067,0.0005310286082668193,0.0002240253088289599,0.0002194322277406224,0.0002110789288002559,0.0001216256116075212,0.0004381356656473232,0.0001184973025456546,0.0001152359260450722,0.0004187500665160639,0.0001752741140352552,0.000172515760235813,0.0001080360995525902,0.0001080360995525902,0.0001513782793400706,0.00014248176793609,0.0003412379318465668,0.0001396840411597932,0.0001399748816750884,0.0001285340224706862,0.0003145626628977262,0.0001201631053849839,7.799483803035808e-05,0.000288740456849228,7.434780605883055e-05,7.281389731413544e-05,7.121163782570384e-05,7.132337734885767e-05,0.0002776991743849258,9.657580836167521e-05,0.0001056287707099883,0.0001047598568689134,0.0002562402143293719,0.0002523961392944956,9.777542848382033e-05,9.669804044412436e-05,9.669804044412436e-05,8.160235370675683e-05,0.0002265541690335045,0.0002235162394197995,8.344303758591116e-05,0.0002097084419318175,8.069385595312364e-05,0.0001977065278384668,7.316998390895681e-05,7.063799340716997e-05,6.925917442416773e-05,6.803313185886736e-05,5.011706663287756e-05,5.534069226678636e-05,4.159062390070765e-05,4.022859079727021e-05,3.751803597276502e-05,3.59593300262681e-05,4.681888228234259e-05,4.565912055508865e-05,4.506867040441152e-05,4.349830423980065e-05,4.050018451331434e-05,4.050018451331434e-05,3.888017713278177e-05,3.614617323205889e-05,0.0001117535032690861,0.0001117535032690861,0.0001093868802597172,9.04711676793236e-05,8.317746676500292e-05,8.103195202556848e-05,7.875539042309331e-05,8.46646965458724e-05,2.614680100861686e-05,6.584842078972455e-05,6.728126087243113e-05,6.237031057842139e-05,2.298985883651967e-05,2.236360951206802e-05,1.785816109905591e-05,1.851990703577861e-05,1.75100297949446e-05,1.584585901298996e-05,1.734075794308746e-05,1.841290157935362e-05,1.653852489864788e-05,1.330008411962525e-05,1.210484782458658e-05,1.209737244048643e-05,1.131887421795397e-05,1.091319013064714e-05,9.112777481265697e-06,9.206285286225598e-06,9.206285286225598e-06,9.622339977763641e-06,1.023175112346666e-05,7.247508267181089e-06,7.574736320071434e-06,2.148263603843021e-05,1.941980919671742e-05,1.442577390533501e-05,4.373071237935524e-06,5.118181550565853e-06,4.712939811982921e-06,3.609964457104213e-06,1.052737324023079e-05,3.655108522610448e-06,9.459334217702776e-06,3.072958244314853e-06,2.994241279516292e-06,2.914709535072975e-06,2.432021072157891e-06,2.041699770635669e-06,1.52045384984298e-06,1.552758075631021e-06,1.37077427231248e-06,9.772651889786393e-07,3.930847041523431e-06,2.699369780059035e-06,5.639247134348282e-07,3.833688264115856e-07,3.561484971067716e-07,5.448176291262669e-07,2.694419471402574e-07,5.531439064737628e-07,2.3572438835703e-07,9.296292878433394e-07,1.017759901307138e-06,1.040242933275395e-06,3.313649899715586e-07,1.333386064124195e-07,9.837044881309128e-08,4.576909356035916e-07,1.387121862713749e-07,4.075755452548617e-08,4.075755452548617e-08,3.689891682759979e-08,8.883168300932288e-08,1.114625637409993e-07,2.511582512122228e-08,2.149306079531949e-08,1.325797622935279e-08,1.49489365327194e-08,5.779467627960603e-09,3.58969737154592e-09,2.658626070592756e-09,2.658626070592756e-09,8.678624414642082e-09,2.265744509549462e-09,5.717260884063198e-09,1.394526262251322e-08,1.228614188234843e-08,6.593608263306449e-09,1.414928737584086e-09,1.741673343235695e-10,1e-10,1e-10,1e-10,1e-10,1.354182018941335e-09,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1.003567655375809e-10,2.418757302864012e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10],[0.009787700213197461,0.009189941909952846,0.008937761751804833,0.008601405545306794,0.008360848470104227,0.008259267255348009,0.007889567894249888,0.00751949837517983,0.00751949837517983,0.007394063371555231,0.006684003055675553,0.006645057595651068,0.006585254588660761,0.006551228623206057,0.00648617124305733,0.006377803808383628,0.006326476394897704,0.006213331957989666,0.006213331957989666,0.005730400348566867,0.005437474948715501,0.005400461182267405,0.005385590939203809,0.005385590939203809,0.005157444661416462,0.005088924857545436,0.004991554129885968,0.00483978066807803,0.004777342104232682,0.00471441344573046,0.00470942400273804,0.004703538991566,0.004703538991566,0.00470151160354014,0.004647710777405619,0.004590939349191527,0.004590939349191527,0.00444149704837578,0.00444149704837578,0.004413056358683199,0.004406801959657473,0.004406801959657473,0.004188920823613517,0.004123285876409782,0.004123285876409782,0.004080016858627781,0.004017572255956925,0.003987125893094341,0.003890009520713617,0.003653828770943013,0.003565319796271568,0.003538629591086557,0.003538629591086557,0.003491253684031161,0.00304373807467325,0.002938674654470001,0.002883632030192732,0.002728885381195526,0.002654549944924209,0.002647805222088146,0.002613706242507148,0.002611757270102564,0.002552263001270303,0.002416125948523562,0.002416125948523562,0.002401362519227337,0.002324248454127184,0.002259904177219297,0.002259904177219297,0.002259904177219297,0.001995848790016594,0.001926121846418597,0.001926121846418597,0.001926121846418597,0.001926121846418597,0.001778740346876051,0.001749312782683209,0.001749312782683209,0.001746368696195799,0.001586976432038579,0.001566787806711154,0.001380101012436415,0.00131715823675674,0.001264493536268769,0.001209360113665231,0.00109061032431556,0.00109061032431556,0.00109061032431556,0.001058145675551202,0.0009796578567001619,0.0009650581304942885,0.0009321192206855426,0.0009150357591565281,0.0008153642100668482,0.0008153642100668482,0.0008153642100668482,0.0007932015588336495,0.0007878448365069327,0.0006803960761229606,0.000648060774050556,0.0006255743614390876,0.0005890675456337616,0.0004566071044819082,0.0004462433998015075,0.000444114490335065,0.0004162535769654915,0.0003765594035921595,0.0003483603508585461,0.0003347808657931718,0.0003279235862136962,0.0003278287727003164,0.0003278287727003164,0.0002826696737347242,0.0002405755244578586,0.0002395934680840957,0.0001928921721935932,0.0001745441236834325,0.0001576709197873844,0.0001433525874083041,0.0001430828323114529,0.0001034284842043672,9.453795017368155e-05,7.198133084703931e-05,6.965671022579942e-05,6.572360960142723e-05,5.498958695114644e-05,5.324010099809967e-05,5.029501267944996e-05,4.207204752358263e-05,4.207204752358263e-05,4.207204752358263e-05,4.202387795895714e-05,2.980435326189987e-05,2.430267824887634e-05,2.221326674129431e-05,2.143970229056938e-05,1.275439739617547e-05,1.275439739617547e-05,1.275439739617547e-05,9.771485131025518e-06,5.796053314531966e-06,5.646494138616609e-06,5.594983089139485e-06,5.186078368381833e-06,4.349261181813539e-06,2.466256458695189e-06,1.87222660689295e-06,1.559948846920299e-06,1.559948846920299e-06,1.113756799879067e-06,1.018761122202876e-06,8.804581761457325e-07,7.809347068607403e-07,7.45359274195805e-07,4.363769832442813e-07,2.723737819849366e-07,1.635082934629671e-07,1.1735e-07,1.1735e-07,1.1735e-07,1.1735e-07,9.858445097892921e-08,7.065714285714286e-08,7.065714285714286e-08,7.065714285714286e-08,7.065714285714286e-08,7.065714285714286e-08,7.065714285714286e-08,7.065714285714286e-08,2.57582364879791e-08,2.57582364879791e-08,2.57582364879791e-08,1.956505907205556e-08,9.1e-09,9.1e-09,9.1e-09,9.1e-09,9.1e-09,9.1e-09,9.1e-09,9.1e-09],[53,17,11,23,56,10,48,13,12,72,47,42,20,24,41,56,9,29,29,23,20,41,9,23,26,43,57,22,39,12,7,23,10,45,39,23,85,5,40,7,9,9,12,41,69,22,35,10,45,6,59,19,23,18,41,13,18,34,19,22,34,76,17,10,10,13,90,8,8,37,43,39,41,43,49,52,9,41,40,52,20,49,14,23,71,25,16,9,49,38,28,15,19,6,39,39,14,9,8,33,45,61,48,53,23,18,47,38,10,50,13,48,20,22,27,27,25,19,76,118,20,74,29,89,19,69,46,62,15,19,101,25,29,46,12,24,40,40,30,37,27,27,32,91,33,68,33,22,22,57,20,75,80,19,94,51,98,88,47,42,94,66,35,37,35,25,36,40,43,36,37,40,67,72,85,150,62,70,108,72,49],["activated","supressed","supressed","activated","activated","activated","activated","supressed","activated","activated","activated","activated","supressed","activated","activated","supressed","supressed","activated","activated","supressed","supressed","activated","activated","activated","activated","supressed","supressed","activated","activated","activated","activated","activated","activated","activated","activated","activated","supressed","supressed","activated","activated","activated","activated","activated","activated","activated","activated","activated","supressed","activated","activated","supressed","activated","activated","activated","activated","supressed","activated","activated","supressed","activated","supressed","activated","activated","activated","activated","activated","activated","supressed","supressed","supressed","activated","activated","activated","activated","activated","supressed","supressed","activated","activated","supressed","supressed","activated","activated","supressed","activated","activated","activated","activated","supressed","supressed","activated","supressed","supressed","supressed","activated","activated","activated","activated","activated","supressed","activated","activated","activated","supressed","activated","supressed","activated","activated","activated","supressed","activated","supressed","activated","activated","supressed","supressed","activated","activated","activated","activated","supressed","supressed","supressed","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","supressed","activated","activated","activated","activated","activated","activated","supressed","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated"],[0.6794871794871795,0.85,0.6470588235294118,0.2053571428571428,0.343558282208589,0.303030303030303,0.6153846153846154,0.3939393939393939,0.4285714285714285,0.3157894736842105,0.2460732984293194,0.7,0.273972602739726,0.1904761904761905,0.7068965517241379,0.2916666666666667,0.5,0.2589285714285715,0.2589285714285715,0.2674418604651163,0.4878048780487805,0.7454545454545455,0.2093023255813954,0.2555555555555555,0.3466666666666667,0.2349726775956284,0.35625,0.4583333333333333,0.7358490566037735,0.4,0.3684210526315789,0.6764705882352942,0.4761904761904762,0.703125,0.52,0.1932773109243698,0.3663793103448276,0.2631578947368421,0.7547169811320755,0.3043478260869565,0.6923076923076923,0.6923076923076923,0.3428571428571429,0.7321428571428571,0.3988439306358382,0.130952380952381,0.4666666666666667,0.5263157894736842,0.6818181818181818,0.24,0.2468619246861925,0.3114754098360656,0.1377245508982036,0.2093023255813954,0.4823529411764706,0.3714285714285714,0.4090909090909091,0.3736263736263736,0.3653846153846154,0.1496598639455782,0.288135593220339,0.4810126582278481,0.34,0.6666666666666666,0.6666666666666666,0.4482758620689655,0.6164383561643836,0.4444444444444444,0.4444444444444444,0.578125,0.7818181818181819,0.8125,0.7884615384615384,0.7962962962962963,0.7424242424242424,0.2290748898678414,0.4736842105263158,0.7884615384615384,0.7692307692307693,0.304093567251462,0.392156862745098,0.2934131736526946,0.21875,0.323943661971831,0.5259259259259259,0.5681818181818182,0.64,0.391304347826087,0.2462311557788945,0.2389937106918239,0.4666666666666667,0.2884615384615384,0.4130434782608696,0.3529411764705883,0.7959183673469388,0.7959183673469388,0.2,0.2571428571428571,0.32,0.4583333333333333,0.8333333333333334,0.7439024390243902,0.7741935483870968,0.3732394366197183,0.3833333333333334,0.2857142857142857,0.7833333333333333,0.6909090909090909,0.4166666666666667,0.4098360655737705,0.2452830188679245,0.2790697674418605,0.2325581395348837,0.4782608695652174,0.303370786516854,0.3253012048192771,0.3378378378378378,0.2235294117647059,0.4840764331210191,0.5267857142857143,0.2985074626865671,0.3425925925925926,0.3670886075949367,0.4659685863874345,0.5135135135135135,0.4423076923076923,0.2180094786729858,0.4161073825503356,0.4411764705882353,0.2602739726027397,0.5838150289017341,0.4032258064516129,0.2056737588652482,0.2674418604651163,0.6666666666666666,0.24,0.7547169811320755,0.7547169811320755,0.5263157894736842,0.3032786885245902,0.303370786516854,0.2213114754098361,0.2119205298013245,0.6148648648648649,0.2558139534883721,0.293103448275862,0.4925373134328358,0.5365853658536586,0.5365853658536586,0.3257142857142857,0.2105263157894737,0.3537735849056604,0.7142857142857143,0.3333333333333333,0.6351351351351351,0.281767955801105,0.4355555555555555,0.6717557251908397,0.2781065088757396,0.3,0.4747474747474748,0.7021276595744681,0.3125,0.3008130081300813,0.2071005917159763,0.2272727272727273,0.1935483870967742,0.245398773006135,0.184549356223176,0.4,0.3457943925233645,0.3603603603603603,0.7613636363636364,0.7826086956521739,0.7727272727272727,0.6944444444444444,0.7380952380952381,0.8433734939759037,0.6506024096385542,0.6923076923076923,0.8448275862068966],["Psmd8/H2ac18/Rps27a/Actg1/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/H2ac8/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Psmc3/Psmb3/Psmd13/Ruvbl1/Psma5/Adrm1/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8/Nedd8/H2ac24/Psme1/Uchl5/Psmc6/Psmf1/Psma1/Tfpt/Mcrs1/Psmd10/Psmd9/Mbd6/Psmd3/Ubc/Ino80e/Yy1/Psme3/Ino80c/Psmc2/Psmd7/Ubb","Vav1/Grb2/Pik3cd/Dapp1/Syk/Stim1/Cd79b/Btk/Pik3r1/Itpr1/Sos1/Itpr2/Plcg2/Pik3ap1/Nck1/Blnk/Cd22","Ncoa1/Osbpl9/Osbpl1a/Cyp27a1/Stard5/Rxra/Ncoa2/Abcc3/Alb/Hsd3b7/Ptgis","Ptgs2/Mmp9/Birc5/Il6/Serpinb2/Pi16/Thbs1/Serpina3n/Cstdc5/Cst7/Cstdc4/Serpina3g/Timp1/Vegfa/Ecm1/Cdkn2d/Spink2/Serpine1/Gapdh/Nr4a1/Cd109/Gpi1/Ifi204","Ccl12/Ccl3/Adam8/Vegfa/Lcn2/Tgfb3/Egln3/Ccl2/Prnp/Egr1/Tnf/Mt1/Trem2/Il1b/Jun/Pin1/Coro1a/Btg2/Gpi1/Cx3cr1/Itgam/Msh2/Bag1/Nqo2/Npm1/Nupr1/Wfs1/Ctsz/Retreg1/Pink1/Casp2/Sod2/Sigmar1/Hras/Mcl1/Bcl2/Lgmn/Apoe/Ube2m/Bcl2l11/Set/Park7/Hdgf/Fis1/Casp9/Aars/Tyrobp/Cdc34/C5ar1/Cebpb/Parp1/Zpr1/Aifm1/Hsp90ab1/Casp3/Rhoa","Mcm5/Clspn/Mcm10/Cdc45/Rad9b/Cdc25c/Mcm8/Rfc4/Mcm2/Cdc25a","Psmd8/Rps27a/Mcm5/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Cdt1/Psmc3/Psmb3/Psmd13/Anapc7/Psma5/Psmd2/Psmc5/Psmb10/Anapc16/Psmb2/Psmd4/Psme2/Orc6/Psmb8/Psme1/Psmc6/Psmf1/Psma1/Cdc26/Psmd10/Psmd9/Mcm4/Psmd3/Ubc/Gmnn/Psme3/Mcm3/Psmc2/Psmd7/Ubb","Pak1/Slc9a1/Kcnj2/Tmem65/Scn1b/Ehd3/Jup/Abcc9/Slc8a1/Ank2/Cacna1a/Ryr2/Kcna2","Pgk1/Cbr2/Tpi1/Ldha/Pkm/Gapdh/Eno1/Gpi1/Aldoa/Mdh2/Pfkp/Pfkl","Ccl12/Ccl3/Adam8/Vegfa/Lcn2/Gpnmb/Tgfb3/Egln3/Ccl2/Fos/Ddit4/Prnp/Egr1/Tnf/Mt1/Trem2/Il1b/Jun/Pin1/Coro1a/Fcgr2b/Btg2/Bad/Gpi1/Rack1/Aimp2/Cx3cr1/Gfer/Itgam/Msh2/Sorl1/Bag1/Atp5g1/Nqo2/Npm1/Nupr1/Ccl5/Wfs1/Ctsz/Retreg1/Pink1/Capn2/Casp2/Sod2/Cd34/Sigmar1/Hras/Mcl1/Bcl2/Lgmn/Chmp4b/Apoe/Ube2m/Bcl2l11/Set/Park7/Hdgf/Fis1/Casp9/Aars/Tyrobp/Nenf/Cdc34/Ikbkb/C5ar1/Cebpb/Parp1/Zpr1/Aifm1/Hsp90ab1/Casp3/Rhoa","Saa3/Cxcl3/Vcam1/Ppbp/Tnfsf14/Pgf/Thbs1/Cxcl5/Ackr3/Cxcr2/Epha2/Cxcl1/Ccl24/Cxcl2/Cxcl13/Ccl2/Adam8/Ccl7/Cxcl10/Ccl3/Trem1/Vegfa/Mtus1/Ccrl2/Ccl17/Ccl8/Spp1/Pf4/Ccl4/Ccr1/Itgam/Mif/Trpm2/Trem3/Ccl12/Ninj1/Ccl9/Retnlg/F7/Serpine1/S100a8/Lbp/Bst1/C5ar2/Ednrb/Nr4a1/Hbegf","Psmd8/Rps27a/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Runx2/Rbx1/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8/Psme1/Psmc6/Psmf1/Psma1/Psmd10/Psmd9/Cdk4/Psmd3/Ubc/Psme3/Psmc2/Psmd7/Ubb","Dlg1/Cav1/Pde4d/Tpm1/Kcnj2/Tnnc1/Pde4b/Pik3ca/Agrn/Jup/Stac2/Kcne3/Grcc10/Ank2/Fxyd1/Slc8a1/Ehd3/Atp1b1/Map2k6/Fgf13","Birc5/Ccna2/Ccnb1/Clspn/Ndc80/Plk1/Cdc25b/Dtl/Mrnip/Fbxo5/Foxm1/Kif14/Bard1/Hmga2/Mastl/Cdc25c/Ier3/Cenpf/Ccnb2/Cdc25a/Cdk1/Fam122a/Trim39/Lmnb1","Psmd8/Rps27a/Cdkn1a/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8/Cdkn1b/Psme1/Psmc6/Psmf1/Psma1/Psmd10/Psmd9/Psmd3/Ubc/Psme3/Psmc2/Psmd7/Ubb","Ube2w/Ube4a/Traf3/Rnf14/Ddb2/Tnks/Cbl/Shprh/Trim30a/Fbxw11/Marchf6/Tnks2/Ube2l6/Ube2g1/Rnf8/Fbxo38/Rnf216/Trim34a/Trim21/Lrsam1/Ankib1/Ttc3/Rnf31/Marchf8/Cdc27/Peli1/Smurf1/Rffl/Rnf135/Sash1/Btrc/Nod2/Trim30d/Ube3c/Skp2/Hace1/Smurf2/Ube2v1/Arih1/Nedd4l/Ube2e2/Trim30c/Rnf213/Dzip3/Trim12c/Rnf144b/Traf5/Wwp1/Hamp/Ambra1/Trim14/Trim34b/Fbxl17/Rnf180/Dysf/Prkn","Adrb1/Ppp3ca/Apoa1/Hamp/Nr1h3/Slc22a21/Slc22a5/Apoa2/Stk39","Lrg1/Mmp9/Sphk1/Cela1/Pgf/Adm/Thbs1/Hyal1/F3/Angpt2/Notch4/Cxcr2/Hmga2/Cysltr2/Ccl24/Vegfa/Hmox1/Ecm1/Ninj1/Pkm/Serpine1/Emilin2/Cd40/Grn/Sirt6/Pdcd6/Il1a/Emilin1/Hif1a","Lrg1/Mmp9/Sphk1/Cela1/Pgf/Adm/Thbs1/Hyal1/F3/Angpt2/Notch4/Cxcr2/Hmga2/Cysltr2/Ccl24/Vegfa/Hmox1/Ecm1/Ninj1/Pkm/Serpine1/Emilin2/Cd40/Grn/Sirt6/Pdcd6/Il1a/Emilin1/Hif1a","Ift88/Nphp3/Lrp6/Maml1/Nrp1/Luzp1/Smo/Tmem65/Rxra/Acvr1/Smad6/Pde2a/Rbp4/Frs2/Myh10/Slit3/Hes1/Heg1/Bmpr1a/Nrg1/Ank2/Ryr2/Fgfr2","Gbp2/Igtp/Ifi205/Gbp3/Aim2/Mndal/Ifi203/Gm5431/Ifi208/9930111J21Rik1/Stat1/Pnpt1/Oas1g/Irgm2/Tgtp2/Iigp1/F830016B08Rik/Gbp2b/Gm4951/Gm12185","Psmd8/Rps27a/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Rbx1/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8/Psme1/Psmc6/Psmf1/Psma1/Sufu/Psmd10/Psmd9/Psmd3/Ubc/Psme3/Psmc2/Psmd7/Ubb","Ccl12/Ccl3/Vcan/Ccl2/Ccr2/Trem2/Vim/Tspo/Cx3cr1","Ccl3/Osm/Adam8/Ptgs2/Serpine1/Adora2b/Lgals1/Tnf/Ccr2/Trem2/Il1b/Fem1a/Mefv/Nfkbia/Fcgr1/Alox5ap/Fcgr3/S100a8/Rps19/Lbp/Nupr1/Ccl5/Vamp8","Adam8/Vegfa/Rgcc/Egln3/Suv39h1/Bnip3/Pgk1/Egr1/Higd1a/Bad/Eno1/Eif4ebp1/Acaa2/Tbl2/Pdk1/Pdk3/Nop53/Bnip3l/Sdhd/P4hb/Zfp36l1/Pink1/Cd34/Fam162a/Rbpj/Bcl2","Lyn/Pik3cb/Ranbp9/Vav3/Nrp1/Krt10/Pak1/Rasa1/Rock2/Smad3/Aph1b/Abl1/Tcf12/Gab1/Itgav/Rxra/Plxna1/Nck1/Rps6ka5/Dock1/Numb/Jup/Arhgef12/Frs2/Arhgef11/Farp2/Abl2/Vav2/Arhgap35/Slit3/Akt3/Itsn1/Itga9/Pkp4/Kazn/Gfra2/Ptk2/Ank1/Sema6d/Adam22/Sdc2/Nck2/Plxna4","Nipbl/Arid1a/Spred2/Ikzf1/Cep290/C3/Prkci/Rab3gap1/Tgfbr2/Ift140/Pou2f1/Zfp513/Kmt2c/Abi2/Pds5a/Sos1/Tgfbr1/Ttll5/Kdr/Zeb2/Tdrd7/Tbc1d32/Slc25a25/Hipk2/Ift88/Nf1/Grcc10/Apc/Lrp6/Nrp1/Maf/Pbx1/Smad3/Rxra/Zeb1/Pbx3/Lrp5/Dlg1/Rdh10/Rbp4/Frs2/Mertk/Meis1/Alms1/Scaper/Arhgap35/Myh10/Meis3/Hes1/Stau2/Mitf/Angptl7/Zhx2/Wdpcp/Ptprm/Fgfr2/Thrb","Cxcl3/Ppbp/Pglyrp1/Acod1/Adm/Cxcl5/Cxcl1/Cxcl2/Cxcl13/Cxcl10/Ccl17/Ccl8/Pf4/Ccl12/Gapdh/Wfdc17/Ang2/S100a9/Npy/Ang5/Ccl25/Wfdc21","Psmd8/Rps27a/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8/Psme1/Psmc6/Psmf1/Psma1/Psmd10/Psmd9/Psmd3/Ubc/Psme3/Psmc2/Psmd7/Ubb","Lif/Aspm/Prdm9/Cdc25b/Fbxo5/Psma8/Ttk/Cdc25c/Osm/Cdc20/Cdc25a/Piwil2","Mmp14/Prtn3/Mmp9/Ctsk/Mmp8/Mmp19/Ctsl","Top2a/Ccnb1/Nusap1/Ncaph/Ncapg/Hmga2/Cdca5/H1f5/Cdk1/Ncapg2/Phf13/H1f0/H3f3b/H1f3/Ncapd2/Smc2/Banf1/H1f2/H1f1/H1f4/Dffb/H3f3a/Chmp1a","Cenpe/Cenpk/Ndc80/Nuf2/Cenpn/Cenpa/Cenpw/Dlgap5/Cenpt/Cenps","Psmd8/Rps27a/Mcm5/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Cdt1/Psmc3/Psmb3/Psmd13/Psma5/Rbx1/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Orc6/Psmb8/Psme1/Psmc6/Psmf1/Psma1/Psmd10/Psmd9/Mcm4/Psmd3/Ubc/Psme3/Mcm3/Psmc2/Psmd7/Ubb","Ube2c/Ccna2/Ccnb1/Mad2l1/Plk1/Bub1b/Nek2/Psma8/Cdc20/Cdk1/Psme2b/Psma6/Psmd12/Psmd8/Psmf1/Psmd4/Psmb4/Pttg1/Psmd13/Psmd10/Psmc1/Psma7/Ube2s/Psmb3/Psmb7/Rps27a/Psmd3/Psma4/Psma2/Psmb10/Psmc3/Ube2e1/Psme3/Psmb5/Psmb1/Psmd2/Anapc15/Psmb6/Psmd6","Ccl12/Vegfa/Trem1/F7/Serpine1/Zfp580/Tnfsf14/Ccl2/Ccl7/Megf8/Dusp1/Ccr1/Sell/Ccr2/Cxcl14/Trem2/Il1b/Thbs1/Mif/Cxcl10/Pla2g7/Cxcr4/Bst1","Agap1/Agap3/Prex1/Sos1/Ptk2b/Rabgap1/Snx9/Rundc1/Arhgap22/Arhgap27/Cxcl13/Ripor1/Rapgef1/Tbc1d9b/Flcn/Sh3bp4/Wdr41/Adrb1/Tbc1d1/Nf1/Iqgap2/Tbc1d8b/Rasgrp2/Sipa1l2/Tbc1d9/Evi5/Ralgapa1/Tbck/Vav3/Tbc1d22a/Adap2/Fgd6/Rapgef6/Gapvd1/Dock11/Rasa1/Dab2ip/Tbc1d2b/Sipa1l1/Abr/Rapgef2/Dennd1b/Rap1gap2/Dock7/Dock8/Plxna1/Snx13/Rapgef5/Arhgap6/Pecam1/Tbc1d8/Rasgrf2/Ralgapa2/Arhgap25/Usp6nl/Als2/Arhgap12/Lrch1/Dock9/Rasgef1b/Dock10/Rap1gds1/Vav2/Srgap2/Arhgap29/Pot1b/Arhgap35/Arhgap15/Tbc1d12/Ralgps2/Tbc1d5/Nedd9/Tbc1d4/Sbf2/Arhgef5/Dennd1a/Bcas3/Rasgrp3/Rgl1/Chn2/Pkp4/Arap2/Myo9a/Plxna4/Dgki","Pla2g4a/Adcy4/Adcy9/Adcy3/Pde1c","Runx3/Psmd8/Rps27a/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8/Psme1/Psmc6/Psmf1/Psma1/Psmd10/Psmd9/Psmd3/Ubc/Psme3/Psmc2/Psmd7/Ubb","Rgcc/Serpine1/Tgfb3/Adora2b/Ccl2/Vim/Cygb","Pgk1/Tpi1/Pkm/Gapdh/Eno1/Gpi1/Aldoa/Pfkp/Pfkl","Pgk1/Tpi1/Pkm/Gapdh/Eno1/Gpi1/Aldoa/Pfkp/Pfkl","Mmp14/Mmp9/Ctsk/Mmp8/Dcn/Col18a1/Timp1/Adam8/Mmp19/Spp1/Ctsl/Fn1","Psmd8/Rps27a/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Ube2m/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8/Psme1/Psmc6/Psmf1/Psma1/Psmd10/Nfkb2/Psmd9/Psmd3/Ubc/Psme3/Psmc2/Psmd7/Ubb","Espl1/Lmna/Psmd8/Rps27a/Tuba1c/Cenpf/Psmb5/Tubb6/Psmb4/Psma4/B9d2/Psma6/Gm11808/Psmb6/Ccnb2/Cenpa/Chmp2a/Tuba4a/Psmb7/Ran/Ube2i/Psmc4/Banf1/Psma7/Psma2/Tubb4b/Psmb1/Uba52/Psmd12/Bub3/Psmd6/Psmc3/Psmb3/Psmd13/Anapc7/Chmp6/Psma5/Tuba1a/Cenps/Chmp4b/Lmnb1/Ppp1cc/Vps4a/Psmd2/Nudc/Psmc5/Psmb10/Cenpe/Rps27/Anapc16/Psmb2/Psmd4/Emd/Ppp2r5d/Psme2/Cdc20/Pom121/Psmb8/Pttg1/Sirt2/Psme1/Psmc6/Psmf1/Psma1/Chmp2b/Rps27rt/Cdc26/Psmd10/Smc3","Cxcl2/Ccl12/Cxcl1/Tnip3/Arid5a/Serpine1/Cxcl3/Ccl2/Zfp36/Tnfaip3/Plscr1/Tnf/Cd14/Trem2/Il1b/Vim/Mif/Cxcl10/Fcgr2b/Cxcl16/Cx3cr1/Nfkbia","Tnfsf9/Tnfsf14/Tnf/Psmd8/Tnfrsf12a/Rps27a/Tnfrsf13b/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Tnfsf8/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Tnfrsf14/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Ube2m/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8","Slc9a1/Kcnj2/Tmem65/Scn1b/Ehd3/Jup/Abcc9/Slc8a1/Ank2/Ryr2","Psmd8/Rps27a/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Bub3/Psmd6/Psmc3/Psmb3/Psmd13/Anapc7/Psma5/Psmd2/Psmc5/Psmb10/Anapc16/Psmb2/Psmd4/Psme2/Cdc20/Psmb8/Pttg1/Psme1/Psmc6/Psmf1/Psma1/Cdc26/Psmd10/Psmd9/Psmd3/Ubc/Psme3/Psmc2/Psmd7/Ubb","Ccl12/Rgcc/Ccl2/Thbs1/Bad/Gsn","Trpm4/Nfkb1/Enpp1/Daam1/Tbc1d24/Ptk2b/Fbxw11/Tle1/Tnks2/Zeb2/Rapgef1/Tmem64/Adrb1/Apc/Ppp3ca/Nphp3/Lrrk1/Lrp6/Ccnyl1/Tle4/Dab2ip/Ryk/Macf1/Smo/Btrc/Smad3/Slc29a1/Abl1/Nlk/Invs/Gnaq/Lrp5/Cdk14/Stk3/Jup/Rnf213/Plpp3/Ext1/Bcl9/Ttc21b/Abl2/Ctnnd1/Grk5/Disc1/Mllt3/Arntl/Znrf3/Mitf/Tcf7l2/Ednrb/Wwox/Ppp1r9a/Atxn1/Dmpk/Tcf7l1/Cmah/Prkn/Dgki/Fgfr2","Chil3/Il1r2/Per1/Serpine1/Tarm1/Tnf/Trem2/Slc7a2/Cuedc2/Mefv/Pbxip1/Alox5ap/Zc3h12a/Rps19/Lbp/Vamp8/Pycard/Lilrb4b/Il4ra","Adam8/Vegfa/Rgcc/Serpine1/Arg1/Tarm1/Gpnmb/Spn/Dusp1/Sdc4/Lgals1/Gcnt2/Prnp/Tnfaip3/Runx3/Mmp14/Mmp12/Adam15/Thbs1/Lgals3/Klf4/Anxa1/Mbp","Ptgs2/Cenpe/Top2a/Brca2/Ccnb1/Ndc80/Plk1/Aspm/Prdm9/Cdc25b/Ncaph/Fbxo5/Fosl1/Mastl/Aurka/Ttk/Mcm8/Ccnb2","Ube2c/Ccna2/Ccnb1/Mad2l1/Aurkb/Plk1/Bub1b/Fbxo5/Nek2/Psma8/Aurka/Cdc20/Cdk1/Psme2b/Psma6/Psmd12/Psmd8/Psmf1/Psmd4/Psmb4/Pttg1/Psmd13/Psmd10/Psmc1/Psma7/Ube2s/Psmb3/Psmb7/Rps27a/Psmd3/Psma4/Psma2/Psmb10/Psmc3/Ube2e1/Psme3/Psmb5/Psmb1/Psmd2/Anapc15/Psmb6","Dlg1/Cav1/Pde4d/Kcnj2/Pde4b/Agrn/Jup/Ank2/Fxyd1/Slc8a1/Ehd3/Atp1b1/Fgf13","Mmp14/C1rl/Thbs1/C1ra/C1rb/Hpn/F10/Hp/Pgk1/Ctsl/Eno1/Serpine1/Pycard/Eno1b/Ifi204/Casp6/Lgmn/Plaur","Cxcl3/Ppbp/C1rl/Pglyrp1/Acod1/Adm/C1ra/Cxcl5/Cd46/C1rb/Cxcl1/Cxcl2/Cxcl13/Cxcl10/Ccl17/Ccl8/Fcna/C4b/Pf4/Ccl12/Gapdh/Wfdc17/Cfp/Ang2/S100a9/Npy/Ang5/Ccl25/Wfdc21/Cfb/Fcgr2b/Tnf/Lgals3/Cxcl14","Stk39/Atp2b4/Slc9a4/Slc12a2/Cnksr3/Chp2/Agrn/Ptpn3/Nedd4/Nedd4l/Slc24a3/Slc9a7/Fxyd1/Fxyd2/Slc8a1/Hecw2/Atp1b1/Slc9a9/Fgf13","Cxcl2/Ccl12/Cxcl1/Tnip3/Arid5a/Serpine1/Cxcl3/Ccl2/Zfp36/Tnfaip3/Plscr1/Tnf/Cd14/Trem2/Il1b/Vim/Mif/Cxcl10/Fcgr2b/Cxcl16/Cx3cr1/Nfkbia","Snap47/Fbxl20/Synj1/Tbc1d24/Fcho2/Ppp3cc/Cask/Cyth1/Stx3/Braf/Prepl/Pick1/Stxbp5/Erc1/Prkce/Scamp5/Numb/Pld1/Apba1/Btbd9/Stxbp1/Ston2/Prkcb/Clstn1/Dennd1a/Itsn1/Slc4a8/Nrg1/Rims3/Nr3c2/Prkca/Syn1/Cacna1a/Dgki","Uck2/Aprt/Trem2/Pkm/Eno1/Coasy/Impdh2/Ndufa10/Adssl1/Ndufb6/Nme1/Ndufv3/Aldoa/Ndufa12/Ndufb4/Ndufa6/Pdk1/Ndufv1/Ndufa11/Ndufc2/Pdk3/Atp5d/Atp5g1/Ndufb8/Atp5c1/Ndufs2/Ndufb10/Ada/Ndufb9/Sdhd/Ndufb11/Mmut/Atp5g3/Atp5l/Ak2/Ndufs7/Ndufb7/Cmpk1/Atp5pb/Atp5g2/Ppcs/Prps1/Atp5b/Ndufa13/Ndufa1/Ndufa5/Mpc1/Ndufa2/Sdhb/Mthfd1/Ndufv2/Ndufc1/Adsl/Dnajc30/Ndufs6/Atp5e/Atp5h/Ndufa7/Ndufs8/Ndufa8/Atp5a1/Pdhb/Ndufb5/Ndufb3/Ndufa9/Adss/Tkt/Guk1/Ppat/Mlycd/Nudt2/Atp5j2/Parp1/Papss1/Atp5j/Atp5k","Rgcc/Serpine1/Arg1/Tgfb3/Adora2b/Ccl2/Mmp14/Mmp12/Adam15/Vim/Cygb/Prtn3/Ctsl/Errfi1/Vsir/Prdx5/Retreg1","Pgk1/Tpi1/Pkm/Gapdh/Eno1/Gpi1/Aldoa/Galk1/Pfkp/Pfkl","Pgk1/Tpi1/Pkm/Gapdh/Eno1/Gpi1/Aldoa/Galk1/Pfkp/Pfkl","Cxcl2/Ccl12/Cxcl1/Cxcl3/Ang/Cxcl14/Cxcl10/Lgals3/Gapdh/Fau/Pf4/Rps19/Rpl30","Mrpl54/Mrpl30/Mrpl22/Mrpl9/Mrps18c/Mrpl23/Mrpl57/Mrps11/Mrpl55/Mrpl35/Gatc/Mrps26/Mrpl58/Elac2/Mrpl37/Mtg1/Mrpl15/Mrpl43/Mterf4/Mrpl50/Mrps16/Mrps34/Mrpl47/Aurkaip1/Mrpl20/Gadd45gip1/Chchd10/Mrpl52/Mrps14/Mrpl45/Mrps12/Mrps35/Tufm/Mrpl17/Mrpl13/Mrpl12/Mrps24/Mrpl36/Mrpl33/C1qbp/Mrpl27/Mrpl41/Ndufa7/Mrpl14/Mrps7/Mrpl32/Mrps23/Mrpl18/Mrpl4/Hsd17b10/Mrpl2/Mrpl39/Mrpl28/Uqcc2/Mrps18a/Mrpl40/Mrps33/Mrps17/Mrpl11/Coa3/Mrpl34/Mrps31/Dap3/Hars/Mtg2/Mrps18b/Chchd1/Mrps25/Mrpl51/Mrpl49/Mrps15/Mrps10/Mrpl46/Mpv17l2/Mrpl19/Thap11/Mrps5/Gfm1/Mrpl42/Malsu1/Mrpl24/Noa1/Fastk/Supv3l1/Mrps22/Iars2/Mrpl48/Mrpl21/Nsun3/Gars","Kcnj2/Adcy4/Gabbr1/Adcy9/Kcnj10/Gng10/Adcy3/Kcnj16","Kcnj2/Adcy4/Gabbr1/Adcy9/Kcnj10/Gng10/Adcy3/Kcnj16","Arhgap31/Arhgap17/Pik3r1/Fgd3/Stard8/Arhgap10/Prex1/Arhgap22/Arhgap27/Trio/Fgd4/Spata13/Arhgap4/Vav3/Arhgap21/Dock11/Abr/Dock7/Dock8/Arhgap39/Fam13b/Rasgrf2/Dnmbp/Arhgap5/Dock9/Arhgef12/Arhgef11/Dock10/Vav2/Srgap2/Arhgap29/Arhgap35/Srgap1/Arhgef5/Itsn1/Arap2/Dlc1","Psmd8/Rps27a/Cdkn1a/Psmb5/Psmb4/Psma4/Cks1b/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8/Cdkn1b/Psme1/Psmc6/Psmf1/Psma1/Psmd10/Psmd9/Cdk4/Psmd3/Ubc/Psme3/Psmc2/Psmd7/Ubb","Psmd8/Rps27a/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8/Psme1/Psmc6/Psmf1/Psma1/Psmd10/Psmd9/Psmd3/Ubc/Psme3/Psmc2/Psmd7/Ubb","Psmd8/Rps27a/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Runx2/Rbx1/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8/Psme1/Psmc6/Psmf1/Psma1/Psmd10/Psmd9/Psmd3/Ubc/Psme3/Psmc2/Psmd7/Ubb","Psmd8/Rps27a/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/P4hb/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8/Psme1/Psmc6/Psmf1/Psma1/Psmd10/Psmd9/Erlec1/Psmd3/Ubc/Psme3/Psmc2/Vcp/Psmd7/Ubb/Os9","Egln3/Psmd8/Rps27a/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Rbx1/Elob/Psmd2/Psmc5/Ube2d3/Psmb10/Psmb2/Psmd4/Psme2/Ube2d2a/Psmb8/Egln1/Vhl/Psme1/Psmc6/Psmf1/Psma1/Eloc/Psmd10/Hif1an/Psmd9/Psmd3/Ubc/Psme3/Psmc2/Limd1/Psmd7/Ubb","Dock5/Rock2/Kng1/Trpv4/Smad3/Abr/Slc9a1/Kcnj2/Abl1/Cldn5/Tmem65/Scn1b/Gnaq/Ehd3/Lrp5/Dlg1/Fgg/Per2/Nedd4l/Smad6/Il18/Snta1/Jup/Pde2a/Arhgef12/Ext1/Gas6/Abcc9/Rap1gds1/Arhgap35/Pparg/Cd36/Ptprj/Dock4/Slc22a5/Ednrb/Heg1/Pon1/F5/Ptk2/Dmpk/Abcg3/Cdh5/Slc8a1/Nr3c2/Prkca/Ank2/Immp2l/Stk39/Ryr2/Ptprm/Thrb","Kcnj2/Kcnk13/Abcc9/Gabbr1/Kcnj10/Kcng2/Gng10/Kcnj16/Kcna2","Psmd8/Rps27a/Psmb5/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Psmb7/Psmc4/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Rbx1/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8/Psme1/Psmc6/Dvl3/Psmf1/Psma1/Psmd10/Psmd9/Psmd3/Ubc/Psme3/Psmc2/Psmd7/Ubb","Odc1/Psmd8/Srm/Psmb5/Psmb4/Psma4/Psma6/Psmb6/Psmb7/Amd1/Psmc4/Psma7/Psma2/Psmb1/Psmd12/Oaz1/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8/Oaz2/Psme1/Psmc6/Psmf1/Psma1/Psmd10/Psmd9/Azin1/Psmd3/Psme3/Psmc2/Psmd7","Pkdcc/Plcg2/Kdr/Ptpn4/Pibf1/Abi3/Adam17/Tlr4/Hpx/Dyrk1a/Lyn/Parp9/Aatk/Lrrk1/Clk2/Fbxw7/Nrp1/Parp14/Cd80/Rasa1/Cd81/Vtn/Nod2/Abl1/Pecam1/Acvr1/Prkce/Il18/Insr/Plpp3/Gas6/Sh3bp5/Abl2/Ctnnd1/Cd36/Socs5/Ptprj/Nedd9/Hes1/Adnp/Fer/Lilra5/Ptk2/Il15/Nrg1/Cd4/Kitl/Prkca/Bank1/Pdgfc/Igf1/Fgfr2","Cask/Nf1/Cyth1/Braf/Mctp1/Prepl/Stxbp5/Prkce/Per2/Apba1/Stxbp1/Prkcb/Ppp1r9a/Slc4a8/Rims3/Nr3c2/Prkca/Cacna1a/Prkn/Dgki","Cxcl3/Mmp14/Prtn3/Ppbp/Pgf/Thbs1/Cxcl5/Cxcr2/Ager/Cxcl1/Ccl24/Cxcl2/Cxcl13/Ccl2/Adam8/Ccl7/Cxcl10/Ccl3/Trem1/Vegfa/Mtus1/Ccl17/Ccl8/Spp1/Enpp1/Pf4/Ccl4/Ccr1/Itgam/Mif/Trem3/Ccl12/Ninj1/Ccl9/Retnlg/Serpine1/S100a8/Lbp/Bst1/C5ar2/Ednrb/Dusp1/Il1a/Emilin1/Ccl6/S100a9/Lgmn/Ccl25/Pla2g7","Ccl12/Ccl3/Il1r2/Serpine1/Ccl2/Ccl7/Ccl4/Ccl9/Egr1/Il1b/Ccl6/Anxa1/Rps6ka4/Saa3","Cask/Nf1/Cyth1/Stx3/Braf/Mctp1/Prepl/Pak1/Stxbp5/Erc1/Prkce/Apba1/Stxbp1/Prkcb/Ppp1r9a/Slc4a8/Rims3/Nr3c2/Prkca/Syn1/Cacna1a/Prkn/Dgki","H2-DMb2/Hspa1a/Nudcd2/Fkbp1b/Tbcc/Ero1a/Hspa8/Bag1/Fkbp1a/Dnajc2/Grpel2/Ahsa2/Pdcd5/Qsox1/Fkbp5/Hspa1b/Ppia/Pfdn6/Pfdn4/Dffa/P4hb/Tor2a/Hsp90aa1/Dnlz/Pfdn5/Cct6a/H2-DMa/Tcp1/Cct5/Fkbp4/Chchd4/Ppib/Ppih/Cd74/Ero1b/Ppif/Fkbp8/Hspb1/Cct2/Nudc/Cct8/Sgta/Cdc37/Vbp1/Cct7/Aip/H2-DMb1/Hsp90ab1/Selenof/Entpd5/Ppig/Ahsa1/Grpel1/Dnajb4/Prdx4/Calr/Dnajb11/Pofut2/Tor1a/Cct3/Hspe1/Pfdn2/Ppil3/Ppie/Hspa5/Pdcl3/Hspa9/Cct4/Sdf2l1/Sh3glb1/Tbca","Cxcl3/Ppbp/Fpr2/Fpr1/Cxcl5/Ackr3/Cxcr2/Cxcl1/Cxcl2/Cxcl13/Ccl7/Cxcl10/Ccl3/Ccrl2/Ccl17/Pf4/Ccl4/Ccl12/Ccl9/C5ar2/Ednrb/Ccl6/Npy/Hebp1/Ccl25","Cxcl3/Ppbp/Cxcl5/Ackr3/Cxcr2/Cxcl1/Cxcl2/Cxcl13/Ccl7/Cxcl10/Ccl3/Ccrl2/Ccl17/Pf4/Ccl4/Ccl12","Ccna2/Ccnb1/Plk1/Cdc25b/Foxm1/Cdc25c/Ccnb2/Cdc25a/Cdk1","Cask/Prmt3/Utrn/Homer1/Setd5/Snx27/Nrp1/Pak1/Pick1/Dab2ip/Spg11/Rock2/Wasl/Sipa1l1/Abl1/Arhgap39/Eef2k/Erc1/Numb/Als2/Snta1/Insr/Ppfibp2/Nfia/Dock10/Abl2/Cttn/Srgap2/Myh10/Disc1/Clstn1/Stau2/Adnp/Lrrc4/Itsn1/Zdhhc2/Ppp1r9a/Ptk2/Dmpk/Adgrl3/Nrg1/Rims3/Tanc2/Cadm1/Prkca/Syn1/Cacna1a/Gphn/Fgfr2","Nr3c1/Cask/Prmt3/Ppp3ca/Kidins220/Nrp1/Btbd3/Crtc1/Pak1/Pias2/Arid1b/Dab2ip/Rock2/Wasl/Sipa1l1/Rapgef2/Abl1/Eef2k/Numb/Nedd4l/Mapkapk5/Rere/Dock10/Abl2/Srgap2/Disc1/Scarf1/Stau2/Rbfox2/Itsn1/Ppp1r9a/Cpeb3/Phactr1/Camk1d/Tanc2/Sdc2/Nck2/Cacna1a","Saa3/Ptgs2/Il6/Orm1/F3/Serpina3n/Cxcr2/Sigirr/Cxcl1/Adam8/Hp/Trem1/Il1rn/Fn1/F7/Lbp/Ednrb/Cd163/Il1a/Nupr1/Fcgr1/Npy/Casp6/Gstp1/Fcgr2b/Fcgr3/Tnf/Nlrp3","Sipa1l1/Eef2k/Numb/Srgap2/Clstn1/Stau2/Adnp/Lrrc4/Ppp1r9a/Ptk2/Adgrl3/Nrg1/Cadm1/Prkca/Cacna1a","Cask/Nf1/Cyth1/Braf/Mctp1/Prepl/Stxbp5/Prkce/Apba1/Stxbp1/Prkcb/Ppp1r9a/Slc4a8/Rims3/Nr3c2/Prkca/Cacna1a/Prkn/Dgki","Cd36/Pigr/B3gnt2/Adcy3/Igf1/Pde1c","Acad9/Ndufa10/Ndufb6/Ndufa12/Ndufb4/Ndufa6/Ndufa11/Ndufc2/Ndufb8/Ndufs2/Ndufb10/Ndufaf3/Ndufb9/Ndufaf7/Ndufb11/Ndufaf8/Ndufs7/Ndufb7/Ndufa1/Ndufa5/Ndufa2/Ndufc1/Ndufs6/Ndufs8/Ndufa8/Ndufb5/Ndufb3/Ndufa9/Timm21/Ndufaf1/Ndufs3/Dmac2/Ndufa3/Ndufs5/Dmac1/Ndufab1/Oxa1l/Ndufb2/Ndufaf2","Acad9/Ndufa10/Ndufb6/Ndufa12/Ndufb4/Ndufa6/Ndufa11/Ndufc2/Ndufb8/Ndufs2/Ndufb10/Ndufaf3/Ndufb9/Ndufaf7/Ndufb11/Ndufaf8/Ndufs7/Ndufb7/Ndufa1/Ndufa5/Ndufa2/Ndufc1/Ndufs6/Ndufs8/Ndufa8/Ndufb5/Ndufb3/Ndufa9/Timm21/Ndufaf1/Ndufs3/Dmac2/Ndufa3/Ndufs5/Dmac1/Ndufab1/Oxa1l/Ndufb2/Ndufaf2","Birc5/Cenpe/Brca2/Ccnb1/Ndc80/Aurkb/Kif2c/Kif22/Nuf2/Kif18a/Kif14/Cdca5/Cenpf/Cenpq","Ube2c/Ccna2/Ccnb1/Mad2l1/Plk1/Bub1b/Fbxo5/Cdc20/Cdk1","Rgcc/Serpine1/Arg1/Tgfb3/Adora2b/Ccl2/Vim/Cygb","Arhgap45/Srgap2/Rasgrf2/Arhgef6/Arhgap29/Arhgap27/Abr/Prex1/Plekhg3/Dnmbp/Cav1/Tagap/Arhgef12/Arhgef10/Arap3/Dock8/Dock11/Vav2/Arhgef11/Dock6/Dock10/Ophn1/Spata13/Arhgap35/Vav3/Arhgap39/Prex2/Plekhg1/Srgap3/Arhgap22/Stard13/Dlc1/Arhgap26","Psmd8/Rps27a/Hspa1a/Psmb5/Psmb4/Psma4/Hspa8/Psma6/Gm11808/Psmb6/Psmb7/Hspa1b/Psmc4/Pabpc1/Psma7/Psma2/Psmb1/Uba52/Psmd12/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Hspb1/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Psmb8/Psme1/Psmc6/Psmf1/Psma1/Psmd10/Psmd9/Psmd3/Ubc/Eif4g1/Psme3/Psmc2/Psmd7/Ubb","Zfp36/Psmd8/Rps27a/Hspa1a/Psmb5/Psmb4/Exosc5/Psma4/Hspa8/Psma6/Gm11808/Psmb6/Psmb7/Hspa1b/Psmc4/Pabpc1/Exosc2/Psma7/Psma2/Psmb1/Uba52/Psmd12/Zfp36l1/Psmd6/Psmc3/Psmb3/Psmd13/Psma5/Set/Exosc8/Exosc4/Dis3/Hspb1/Psmd2/Psmc5/Psmb10/Psmb2/Psmd4/Psme2/Exosc1/Parn/Psmb8/Psme1/Psmc6/Ywhaz/Psmf1/Psma1/Psmd10/Ywhab/Psmd9/Anp32a/Psmd3/Ubc/Eif4g1/Psme3/Khsrp/Exosc7/Psmc2/Psmd7/Ubb/Akt1","Psmd8/Rps27a/Tuba1c/Cdkn1a/Psmb5/Tubb6/Psmb4/Psma4/Psma6/Gm11808/Psmb6/Ccnb2/Tuba4a/Psmb7/Psmc4/Psma7/Psma2/Tubb4b/Psmb1/Uba52/Psmd12/Psmd6/Hsp90aa1/Psmc3/Psmb3/Psmd13/Psma5/Tuba1a/Psmd2/Psmc5/Psmb10/Hsp90ab1/Psmb2/Psmd4/Psme2/Psmb8/Psme1/Psmc6/Psmf1/Psma1/Psmd10/Psmd9/Psmd3/Ubc/Psme3/Psmc2/Psmd7/Ubb","Cdc42se2/Ogt/Abca1/Arhgap17/Ralgapb/Shoc2/Stard8/Arhgap18/Cbl/Arhgef18/Cyth3/Sos1/Mapkap1/Ripor1/Rapgef1/Flcn/Nf1/Arfgef2/Cyth1/Sipa1l2/Pik3cb/Camk2d/Ralgapa1/Nrp1/Iqsec2/Madd/Dab2ip/Sipa1l1/Abl1/Rap1gap2/Itgav/Rasgrf2/Ralgapa2/Arhgap25/Arhgef3/Arhgap12/Apoa1/Scai/Arhgap19/Dock2/Abl2/Arhgap35/Auts2/Ralgps2/Dennd1a/Apoc3/Psd3/Heg1/Nrg1/Kitl/Dlc1/Igf1/Dgki","Arg1/Mmp14/Prtn3/Mmp9/Ctsk/Mmp8/Il6/Tgfb3/Creb3l1/Ager/Ccl2/Mmp19/Ctsl/Fn1/Serpine1/Id1/Prdx5/Cygb/Adora2b/Tram2/Emilin1/Hif1a/P3h1","Dlg1/Cav1/Pde4d/Tpm1/Kcnj2/Tnnc1/Pde4b/Pik3ca/Agrn/Jup/Kcne3/Ank2/Fxyd1/Slc8a1/Ehd3/Atp1b1/Map2k6/Fgf13","Ndufa10/Ndufb6/Ndufv3/Ndufa12/Ndufb4/Ndufa6/Ndufv1/Ndufa11/Ndufc2/Atp5d/Atp5g1/Ndufb8/Atp5c1/Ndufs2/Ndufb10/Ndufb9/Sdhd/Ndufb11/Atp5g3/Atp5l/Ndufs7/Ndufb7/Atp5pb/Atp5g2/Atp5b/Ndufa13/Ndufa1/Ndufa5/Ndufa2/Sdhb/Ndufv2/Ndufc1/Dnajc30/Ndufs6/Atp5e/Atp5h/Ndufa7/Ndufs8/Ndufa8/Atp5a1/Ndufb5/Ndufb3/Ndufa9/Atp5j2/Atp5j/Atp5k/Ndufs3","Acad9/Ndufa10/Ndufb6/Ndufv3/Ndufa12/Ndufb4/Ndufa6/Ndufv1/Ndufa11/Ndufc2/Ndufb8/Ndufs2/mt-Nd1/Ndufb10/Ndufaf3/Ndufb9/Ndufaf7/Ndufb11/Ndufs7/Ndufb7/Ndufa13/Ndufa1/Ndufa5/Ndufa2/Ndufv2/Ndufc1/Ndufs6/Ndufa7/Ndufs8/Ndufa8/Ndufb5/Ndufb3/Ndufa9/Ndufaf1/mt-Nd5/Ndufs3/Ndufa3/Ndufs5","Ndc80/Aurkb/Haspin/Bub1b/Zwilch/Ttk/Cenpa/Cdk1/Cenpq/Knl1","Arhgef18/Abi2/Arhgap10/Prex1/Sos1/Pik3ca/Arhgap22/Arhgap27/Trio/Cdc42ep4/Spata13/Iqgap2/Arhgap4/Vav3/Pak1/Arhgap21/Dock5/Dock11/Pld2/Tfrc/Abr/Dock7/Dock8/Arhgap39/Fam13b/Rasgrf2/Dock1/Arhgap25/Als2/Arhgap12/Arhgap5/Dock9/Arhgef11/Dock10/Farp2/Dock2/Abl2/Vav2/Srgap2/Arhgap29/Arhgap35/Arhgap15/Vrk2/Fam13a/Srgap1/Arhgef5/Dock4/Chn2/Arap2/Dlc1","Ccl12/Ccl3/Il1r2/Serpine1/Ccl2/Ccl7/Ccl4/Ccl9/Egr1/Il1b/Ccl6/Rps6ka4/Saa3","P2ry12/Zeb2/Htt/Herc1/Ift88/Nf1/Grcc10/Dync2h1/Wdr47/Mrtfb/Lrp6/Nrp1/Btbd3/Nfib/Pak1/Dab2ip/Ryk/Smo/Atg7/Mcph1/Axl/Rapgef2/Dock7/Gnaq/Fut10/Numb/Ext1/Frs2/Trappc9/Ttc21b/Srgap2/Arhgap35/Gdpd5/Myh10/Disc1/Zswim6/Mrtfa/Syne2/Pgap1/Hes1/Tcf7l2/Bmpr1a/Phactr1/Nrg1/Plxna4/Dlc1/Slc1a2/Fgfr2","Birc5/Cenpe/Sphk1/Spc24/Ccnb1/Mad2l1/Nusap1/Ndc80/Aurkb/Plk1/Bub1/Bub1b/Nuf2/Spc25/Fbxo5/Gen1/Zwilch/Cdca2/Ttk/Mki67","Cxcl3/Ppbp/Cxcl5/Ackr3/Cxcl1/Ccl24/Cxcl2/Cxcl13/Ccl2/Ccl7/Cxcl10/Ccl3/Ccl17/Ccl8/Pf4/Ccl4/Ccr1/Ccl12/Ccl9/Dusp1/Hif1a/Ccl6","Cask/Nf1/Cyth1/Stx3/Braf/Mctp1/Prepl/Pak1/Slc29a1/Stxbp5/Erc1/Prkce/Per2/Apba1/Stxbp1/Prkcb/Kcnj10/Ppp1r9a/Slc4a8/Rims3/Nr3c2/Prkca/Syn1/Cacna1a/Prkn/Dgki/Slc1a2","Cask/Nf1/Cyth1/Stx3/Braf/Mctp1/Prepl/Pak1/Slc29a1/Stxbp5/Erc1/Prkce/Per2/Apba1/Stxbp1/Prkcb/Kcnj10/Ppp1r9a/Slc4a8/Rims3/Nr3c2/Prkca/Syn1/Cacna1a/Prkn/Dgki/Slc1a2","Saa3/Rps6ka4/Il6/Acod1/Hyal1/Nr1d1/Mapk13/Ccl24/Ccl2/Il1r2/Ccl7/Ccl3/Il1rn/Ccl17/Ccl8/Ccl4/Ccl12/Fn1/Zbp1/Ccl9/Serpine1/Cactin/Cd40/Pycard/Ccl6","Birc5/Cenpe/Top2a/Spc24/Ccnb1/Mad2l1/Ndc80/Aurkb/Plk1/Bub1/Bub1b/Nuf2/Spc25/Fbxo5/Gen1/Zwilch/Ttk/Cdc20/Mcm2","Ier3/Fam20c/Ddit4/Pgam1/Pgk1/Trem2/Tpi1/Pkm/Bad/Slc2a6/Gapdh/Tspo/Eno1/Ndufa10/Ndufb6/Gpi1/Ndufv3/Eif6/Aldoa/Galk1/Ndufa12/Ndufb4/Ndufa6/Ndufv1/Ndufa11/Hspa8/Ndufc2/Atp5d/Atp5g1/Ndufb8/Atp5c1/Ndufs2/Ndufb10/Pfkp/Pfkl/Nupr1/Ndufb9/Sdhd/Ndufb11/Atp5g3/Atp5l/Ak2/Ndufs7/Ndufb7/Atp5pb/Atp5g2/Atp5b/Ndufa13/Ndufa1/Aldoc/Ndufa5/Ndufa2/Sdhb/Ndufv2/Ndufc1/Dnajc30/Ndufs6/Atp5e/Atp5h/Ndufa7/Ndufs8/Ndufa8/Atp5a1/Cbfa2t3/Ndufb5/Ndufb3/Ndufa9/Fis1/Guk1/Prxl2c/Atpif1/Nudt2/Atp5j2/Parp1/Atp5j/Atp5k","Clp1/Zmat5/Thoc6/Snrpd2/Gtf2f1/Bud31/Hspa8/Snrnp25/Ddx39a/Snrpc/Snrpe/Snu13/Polr2f/Aaas/Nxt1/Eif4a3/Lsm4/Polr2i/Snrpb/Prpf38a/Hnrnpa1/Ptbp1/Lsm8/Rbm42/Polr2e/Sf3b5/Thoc3/Chtop/Xab2/Srsf3/Cdc5l/Magoh/Cd2bp2/Slu7/Pqbp1/Ybx1/Ppih/Snrpg/Phf5a/Sart1/Wbp4/Cwc22/Puf60/Snrpa1/Zcrb1/Snrpd3/Lsm6/Eif4e/Sf3b2/Hnrnpk/Sf3b6/Snrpf/Txnl4a/Pcbp2/Lsm2/Prpf31/Sf3a1/Pom121/Cwc15/Rbm8a/Hnrnph2/Nudt21/Lsm5/Syf2/Rnps1/Pcbp1/Snrnp27/Thoc7/Polr2d/Dnajc8/Dhx38/Hnrnpa2b1/Polr2g/Lsm7/Sf3b3/Lsm3/Sf3a3/Snrnp40/Ddx46/Hnrnpa3/Slbp/U2af1/Ppil3/Rbm22/Ppie/Nup50/Bcas2/Snrnp200/Cpsf4/Alyref/Snrnp35/Sugp1/Hnrnpf/Cdc40/Thoc5/Hnrnpa0/Prpf19/Magohb/Eftud2/Srrm1/Tpr/Snrpd1/Nup62/Cstf1/Cpsf1/Srsf2/Sec13/Isy1/Srsf9/Polr2c/Sf3b4/Usp39/Prpf40a/Srsf11/Ppil1/Sf3b1/Srrt/Sarnp","Slc38a1/Cask/Camk2d/Aldh2/Pick1/Kcnj2/Dlg1/Apba1/Stxbp1/Adcy4/Prkcb/Gabbr1/Adcy9/Kcnj10/Gng10/Adcy3/Syn1/Cacna1a/Kcnj16/Slc1a2","Cdc42se2/Ogt/Abca1/Arhgap17/Cdc42ep2/Synj2bp/Ralgapb/Shoc2/Stard8/Cbl/Arhgef18/Abi2/Cyth3/Sos1/Mapkap1/Ripor1/Rapgef1/Flcn/Cdc42ep4/Adrb1/Nf1/Arfgef2/Cyth1/Rasgrp2/Pik3cb/Camk2d/Gna12/Nrp1/Iqsec2/Rapgef6/Madd/Dab2ip/Rock2/Usp28/Rapgef2/Abl1/Ksr1/Rapgef5/Pecam1/Rasgrf2/Ralgapa2/Arhgef3/Als2/Mapkapk5/Arhgap5/Apoa1/Scai/Elmo1/Arhgef12/Usp50/Arhgef11/Rasgef1b/Farp2/Dock2/Abl2/Rras2/Arhgap35/Rab39/Auts2/Ralgps2/Srgap1/Rab4b/Dennd1a/Apoc3/Rasgrp3/Rgl1/Psd3/Heg1/Nrg1/Kitl/Dlc1/Igf1/Dgki/Ksr2","Mycbp2/Atxn2/Zeb2/Herc1/Grcc10/B4galt5/Dync2h1/Wdr47/Nrp1/Nfib/Rora/Spg11/Smo/Atg7/Rapgef2/Scn1b/Gnaq/Ttc21b/Arhgap35/Gdpd5/Disc1/Zswim6/Hes1/Ptk2/Bmpr1a/Prkca/Plxna4/Cacna1a/Fgfr2","Ier3/Fam20c/Ddit4/Pgam1/Pgk1/Uck2/Trem2/Tpi1/Pkm/Bad/Slc2a6/Gapdh/Tspo/Eno1/Impdh2/Itpa/Ndufa10/Ndufb6/Gpi1/Nme1/Ndufv3/Eif6/Aldoa/Galk1/Ndufa12/Ndufb4/Ndufa6/Ndufv1/Ndufa11/Hspa8/Ndufc2/Atp5d/Atp5g1/Ndufb8/Atp5c1/Gtpbp1/Ndufs2/Ndufb10/Pfkp/Ran/Ada/Pfkl/Nupr1/Ndufb9/Sdhd/Ndufb11/Atp5g3/Atp5l/Ak2/Ndufs7/Ndufb7/Cmpk1/Nudt16/Atp5pb/Atp5g2/Atp5b/Ndufa13/Ndufa1/Aldoc/Ndufa5/Ndufa2/Sdhb/Dctpp1/Ndufv2/Ndufc1/Dnajc30/Ndufs6/Atp5e/Atp5h/Ndufa7/Ndufs8/Ndufa8/Atp5a1/Cbfa2t3/Ndufb5/Ndufb3/Ndufa9/Fis1/Guk1/Dtymk/Prxl2c/Atpif1/Nudt2/Atp5j2/Parp1/Atp5j/Atp5k/Rhoa/Ndufs3","Cxcl3/Ppbp/Cxcl5/Ackr3/Cxcl1/Ccl24/Cxcl2/Cxcl13/Ccl2/Ccl7/Cxcl10/Ccl3/Ccl17/Ccl8/Pf4/Ccl4/Ccr1/Ccl12/Ccl9","Dnajc17/Clp1/Pwp2/Rps15/Rpl23a/Rps25/Clns1a/Rpl5/Rps14/Rplp0/Rpl11/Rpsa/Rps5/Rpl6/Eif6/Eif3d/Rpl24/Snrpd2/Mrps11/Rpl10/Nopchap1/Znhit3/Rrs1/Nop53/Snrpc/Strap/Eif5/Rps28/Snrpe/Nle1/Ddx28/Snu13/Rps19/Mrto4/Rpl38/Lsm4/Mrpl20/Snrpb/Hsp90aa1/Znhit6/Ramac/Smn1/Ruvbl1/Pih1d1/Xab2/Eif4h/Mrps7/Slu7/Rps2/Rps27l/Gemin2/Snrpg/Puf60/Snrpd3/Snrpf/Rps27/Ddx20/Hsp90ab1/Eif2s3x/Lsm2/Prpf31/Sf3a1/Ruvbl2/Nudt21/Tfip11/Mcts1/Polr2d/Nufip1/Prmt5","Cenpe/Spc24/Prc1/Cenpm/Dlg4/Mad2l1/Cenpk/Ndc80/Aurkb/Kif2c/Plk1/Bub1/Bub1b/Cenpu/Nuf2/Spc25/Dsn1/Ska1/Kif18a/Kif14/Zwilch/Sgo1/Nckap1/Cenpn/Cdc25c/Cenpf/Cenpa/Cdc20/S100a8/Cenpq/Cenpt/Knl1/Ppp2r5d/Cenps/Pmf1/S100a9/Pin1/Cdca8/Sec13/Incenp/Ppp2r5b/Srf/Rangap1/B9d2/Ywhag/Ranbp2","Clp1/Pwp2/Rps15/Rpl23a/Rps25/Clns1a/Rpl5/Rps14/Rplp0/Rpl11/Rpsa/Rps5/Rpl6/Eif6/Eif3d/Rpl24/Snrpd2/Mrps11/Rpl10/Nopchap1/Znhit3/Rrs1/Nop53/Snrpc/Strap/Eif5/Rps28/Snrpe/Nle1/Ddx28/Snu13/Rps19/Mrto4/Rpl38/Lsm4/Mrpl20/Snrpb/Hsp90aa1/Znhit6/Ramac/Smn1/Ruvbl1/Pih1d1/Xab2/Eif4h/Mrps7/Slu7/Rps2/Rps27l/Gemin2/Snrpg/Puf60/Snrpd3/Snrpf/Rps27/Ddx20/Hsp90ab1/Eif2s3x/Lsm2/Prpf31/Sf3a1/Ruvbl2","Cxcl2/Ccl12/Ccl3/Cxcl1/Cxcl3/Ccl7/Ccrl2/Ccl4/Ccl9/Ccl6/Cxcl10/Anxa1/Cxcr4/Cxcl16/Cx3cr1","Cxcl2/Ccl12/Ccl3/Ltb4r1/Cxcl1/Cxcl3/Adora2b/Cnr2/Ccl7/Ccrl2/Ccl4/Ccl9/Ccl6/Gpr35/Cxcl10/Anxa1/Cxcr4/Cxcl16/Cx3cr1","Clp1/Zmat5/Snrpd2/Gtf2f1/Bud31/Hspa8/Snrnp25/Snrpc/Snrpe/Snu13/Polr2f/Eif4a3/Lsm4/Polr2i/Snrpb/Prpf38a/Hnrnpa1/Ptbp1/Lsm8/Rbm42/Polr2e/Sf3b5/Xab2/Srsf3/Cdc5l/Magoh/Cd2bp2/Pqbp1/Ybx1/Ppih/Snrpg/Phf5a/Sart1/Wbp4/Cwc22/Puf60/Snrpa1/Zcrb1/Snrpd3/Lsm6/Sf3b2/Hnrnpk/Sf3b6/Snrpf/Txnl4a/Pcbp2/Lsm2/Prpf31/Sf3a1/Cwc15/Rbm8a/Hnrnph2/Nudt21/Lsm5/Syf2/Rnps1/Pcbp1/Snrnp27/Polr2d/Dnajc8/Dhx38/Hnrnpa2b1/Polr2g/Lsm7/Sf3b3/Lsm3/Sf3a3/Snrnp40/Ddx46/Hnrnpa3/U2af1/Ppil3/Rbm22/Ppie/Bcas2/Snrnp200/Cpsf4/Alyref/Snrnp35/Sugp1/Hnrnpf/Cdc40/Hnrnpa0/Prpf19/Magohb/Eftud2/Srrm1/Snrpd1/Cstf1/Cpsf1/Srsf2/Isy1/Srsf9/Polr2c/Sf3b4/Usp39/Prpf40a/Srsf11/Ppil1/Sf3b1/Srrt","Saa3/Rps6ka4/Il6/Acod1/Hyal1/Nr1d1/Mapk13/Ccl24/Ccl2/Il1r2/Ccl7/Ccl3/Il1rn/Ccl17/Ccl8/Ccl4/Ccl12/Fn1/Zbp1/Ccl9/Serpine1/Cactin/Cd40/Pycard/Ccl6","Ccl12/Adam8/Vegfa/Trem1/F7/Serpine1/Spn/Zfp580/Tnfsf14/Ccl2/Ccl7/Dusp1/Ccr1/Sell/Ccr2/Mmp14/Cxcl14/Trem2/Il1b/Thbs1/Mif/Cxcl10/Coro1a/Lgals3/Anxa1/Ecm1/Cnn2/Pla2g7/Bst1","Rock2/Macf1/Trpv4/Sipa1l1/Smad3/Clasp2/Fmn1/Rapgef2/Abl1/Cldn5/Itgav/Eef2k/Arhgap6/Pecam1/Dlg1/Numb/Nedd4l/Jup/Prkch/Micall2/Cttn/Ctnnd1/Srgap2/Clstn1/Ptprj/Actn1/Stau2/Bcas3/Adnp/Lrrc4/Ppp1r9a/Pkp4/Heg1/Ptk2/Myo9a/Adgrl3/Cdh5/Dst/Nrg1/Ptprk/Cadm1/Wdpcp/Prkca/Cacna1a/Dlc1/Epb41l3","Cxcl2/Ccl12/Ccl3/Cxcl1/Cxcl3/Ccl7/Ccrl2/Ccl4/Cxcl10/Cxcr4/Cxcl16/Cx3cr1","Birc5/Cenpe/Sphk1/Spc24/Lif/Ccnb1/Mad2l1/Nusap1/Ndc80/Aurkb/Plk1/Bub1/Prdm9/Bub1b/Nuf2/Spc25/Fbxo5/Gen1/Zwilch/Psma8/Cdca2/Ttk/Mki67/Osm","Cox4i1/Ndufa10/Uqcrfs1/Ndufb6/Ndufv3/Ndufa12/Uqcrc1/Ndufv1/Ndufc2/Iscu/Ndufb8/Ndufs2/Ndufb9/Sdhd/Cox5b/Ndufs7/Pink1/Sdhb/Uqcr10/Uqcrq/Uqcrh/Ndufv2/Ndufs6/Ndufa7/Uqcrb/Cyc1/Ndufs8/Dnajc15/Ndufa8/Bid/Cox6a1/Cox5a/Coq7/Park7/Ndufaf1/Uqcrc2/Cox7c/Ndufs3/Uqcr11/Chchd2","Cox4i1/Ndufa10/Uqcrfs1/Ndufb6/Ndufv3/Ndufa12/Uqcrc1/Ndufv1/Ndufc2/Iscu/Ndufb8/Ndufs2/Ndufb9/Sdhd/Cox5b/Ndufs7/Pink1/Sdhb/Uqcr10/Uqcrq/Uqcrh/Ndufv2/Ndufs6/Ndufa7/Uqcrb/Cyc1/Ndufs8/Dnajc15/Ndufa8/Bid/Cox6a1/Cox5a/Coq7/Park7/Ndufaf1/Uqcrc2/Cox7c/Ndufs3/Uqcr11/Chchd2","Pwp2/Rps15/Rpl23a/Rps25/Rpl5/Rps14/Rplp0/Rpl11/Mrpl22/Rpsa/Rps5/Rpl6/Eif6/Rpl24/Mrps11/Rpl10/Rrs1/Nop53/Rps28/Nle1/Ddx28/Rps19/Mterf4/Mrto4/Rpl38/Mrpl20/C1qbp/Mrps7/Rps2/Rps27l","Cenpe/Spc24/Cenpm/Mad2l1/Cenpk/Ndc80/Aurkb/Kif2c/Plk1/Bub1/Bub1b/Cenpu/Nuf2/Spc25/Dsn1/Ska1/Kif18a/Zwilch/Sgo1/Cenpn/Cenpf/Cenpa/Cdc20/Cenpq/Cenpt/Knl1/Ppp2r5d/Cenps/Pmf1/Cdca8/Sec13/Incenp/Ppp2r5b/Srf/Rangap1/B9d2/Ranbp2","Slc38a1/Cask/Homer1/Camk2d/Aldh2/Pick1/Kcnj2/Dlg1/Kcnk13/Ppfibp2/Abcc9/Apba1/Stxbp1/Adcy4/Prkcb/Gabbr1/Adcy9/Kcnj10/Kcng2/Gng10/Adcy3/Syn1/Cacna1a/Epb41l3/Kcnj16/Kcna2/Slc1a2","Birc5/Cenpe/Spc24/Ccnb1/Clspn/Mad2l1/Ndc80/Aurkb/Plk1/Bub1/Bub1b/Dtl/Mrnip/Nuf2/Spc25/Bard1/Hmga2/Gen1/Zwilch/Rad9b/Pidd1/Ttk/Ier3/Cdc20/Cdk1/Fam122a/Trim39","Birc5/Cenpe/Spc24/Ccnb1/Clspn/Mad2l1/Ndc80/Aurkb/Plk1/Bub1/Haspin/Bub1b/Dtl/Mrnip/Nuf2/Spc25/Fbxo5/Bard1/Hmga2/Gen1/Zwilch/Pidd1/Ttk/Ccl2/Ier3/Slfn1/Ccl12/Cdc20/Cdk1/Zfp655/Fam122a/Trim39","Bnip3/Cox4i1/Ndufa10/Uqcrfs1/Ndufb6/Ndufv3/Mdh2/Cox7a2l/Ndufa12/Ndufb4/Ndufa6/Uqcrc1/Msh2/Ndufv1/Ndufa11/Ndufc2/Iscu/Nop53/Atp5d/Ndufb8/Cox7a2/Suclg1/Atp5c1/Ndufs2/Fh1/Ndufb10/Nupr1/Ndufb9/Sdhd/Cox5b/Ndufb11/Gadd45gip1/Ndufs7/Ndufb7/Chchd10/Atp5pb/Atp5b/Ndufa13/Fxn/Ndufa1/Pink1/Ndufa5/Ndufa2/Sdhb/Uqcr10/Uqcrq/Uqcrh/Ndufv2/Ndufc1/Adsl/Dnajc30/Ndufs6/Atp5e/Atp5h/Ndufa7/Uqcrb/Cyc1/Ndufs8/Idh3g/Dnajc15/Idh3b/Ndufa8/Atp5a1/Bid/Cox6a1/Pdhb/Cox5a/Cbfa2t3/Ndufb5/Ndufb3/Coq7/Ndufa9/Park7/Ppif/Mtfr1l/Uqcc2/Ndufaf1/Uqcrc2/Atp5j2/Cox7c/Atp5j/Atp5k/Dlst/Rhoa/Ndufs3/Macroh2a1/Ndufa3/Uqcr11/Ndufs5/Chchd2/Mrps36","Birc5/Cenpe/Tpx2/Prc1/Kif23/Gpsm2/Ccnb1/Mad2l1/Nusap1/Ndc80/Aurkb/Plk1/Kif11/Kif4/Nuf2/Spc25/Nek2/Gpsm1/Poc1a/Cep97/Aurka/Mybl2/Plk2/Cenpa/Ints13/Cdc20/Dlgap5/Spry2/Tubg1/Cdca8/Rcc1/Incenp/Racgap1","Cxcl2/Ccl12/Ccl3/Adam8/Cxcl1/Vegfa/Trem1/F7/Serpine1/Cxcl3/Spn/Zfp580/Tnfsf14/Cnr2/Ccl2/Ccl7/Dusp1/Ccl4/Spp1/Ccr1/Sell/Ccl9/Tnf/Ccr2/Mmp14/Cxcl14/Trem2/Jaml/Il1b/Ccl6/Thbs1/Gpr35/Mif/Cxcl10/Coro1a/Lgals3/Anxa1/Ecm1/Slamf9/Cnn2/Pla2g7/Bst1/Prtn3/Cxcl16/Cx3cr1/Itgam/Trem3/Plec/Fcgr3/Ptafr/S100a8/Pf4/Eps8/Jagn1/Rps19/Ada/Lbp/Fadd/Ccl5/Slc37a4/Ptpn22/Ppia/Lyve1/Padi2/Tnfrsf14/C3ar1/Aimp1/Pycard","Gtf3a/Rpl23a/Rpl10a/Rpl5/Rplp0/Rpl11/Rpl7a/Rpl14/Rpl6/Eif6/Rpl24/Rpl26/Rpl10/Nhp2/Nop16/Znhit3/Rrs1/Rpl7/Nop53/Pak1ip1/Nle1/Ddx28/Snu13/Rrp15/Npm1/Mrto4/Rpl38/Rpl35a/Rpl35/Mrpl20/Rsl24d1/Zfp622/Znhit6","Cxcl2/Ccl12/Ccl3/Cxcl1/Cxcl3/Ccl2/Ccl7/Dusp1/Ccl4/Ccr1/Ccl9/Ccr2/Trem2/Ccl6/Gpr35/Cxcl10/Cxcr4/Cx3cr1/Rbm15/Pf4/Zc3h12a/Cib1","Cxcl2/Ccl12/Ccl3/Cxcl1/Cxcl3/Ccl2/Ccl7/Dusp1/Ccl4/Ccr1/Ccl9/Ccr2/Trem2/Ccl6/Gpr35/Cxcl10/Cxcr4/Cx3cr1/Rbm15/Pf4/Zc3h12a/Cib1","Ube2c/Cenpe/Spc24/Cenpm/Mad2l1/Cenpk/Ndc80/Aurkb/Kif2c/Plk1/Bub1/Bub1b/Cenpu/Nuf2/Spc25/Dsn1/Ska1/Kif18a/Zwilch/Psma8/Sgo1/Cdca5/Cenpn/Cenpf/Cenpa/Cdc20/Cenpq/Psme2b/Cenpt/Knl1/Ppp2r5d/Cenps/Psma6/Pmf1/Psmd12/Psmd8/Cdca8/Sec13/Psmf1/Incenp/Psmd4/Ppp2r5b/Rangap1/B9d2/Psmb4/Ranbp2/Pttg1/Psmd13/Tuba4a/Psmd10/Psmc1/Psma7/Tubb4b/Ube2s/Itgb3bp/Psmb3/Psmb7","Birc5/Ube2c/Cenpe/Spc24/Ccnb1/Mad2l1/Ndc80/Aurkb/Plk1/Bub1/Bub1b/Ncaph/Cit/Nuf2/Spc25/Fbxo5/Ncapg/Gen1/Zwilch/Ttk","Ube2c/Cenpe/Spc24/Cenpm/Ccnb1/Mad2l1/Cenpk/Ndc80/Aurkb/Kif2c/Plk1/Bub1/Bub1b/Cenpu/Nuf2/Spc25/Fbxo5/Dsn1/Ska1/Kif18a/Zwilch/Psma8/Sgo1/Cdca5/Cenpn/Cenpf/Cenpa/Ccnb2/Cdc20/Cdk1/Cenpq/Psme2b/Cenpt/Knl1/Ppp2r5d/Pom121/Cenps/Psma6/Lmnb1/Pmf1/Psmd12/Psmd8/Cdca8/Sec13/Rcc1/Psmf1/Incenp/Psmd4/Ppp2r5b/Rangap1/B9d2/Psmb4/Ranbp2/Chmp4b/Pttg1/Psmd13/Tuba4a/Psmd10/Psmc1/Psma7/Chmp2a/Tubb4b/Ube2s/Itgb3bp/Psmb3/Psmb7/Ran/Cenph/Rps27a/Ube2i/Psmd3/Cenpl/Psma4/Banf1/Nup37","Cox11/Cox4i1/Acad9/Ndufa10/Uqcrfs1/Ndufb6/Ndufv3/Cox7a2l/Ndufa12/Ndufb4/Ndufa6/Uqcrc1/Ndufv1/Ndufa11/Ndufc2/Atp5d/Atp5g1/Ndufb8/Atp5c1/Ndufs2/mt-Nd1/Ndufb10/Ndufaf3/Ndufb9/Cox5b/Ndufaf7/Ndufb11/Atp5g3/Atp5l/Ucp2/Ndufs7/Cox8a/Ndufb7/mt-Co3/Atp5pb/Atp5b/Ndufa13/Ndufa1/Ndufa5/Ndufa2/Uqcr10/Uqcrq/Uqcrh/Ndufv2/Coq10b/Ndufc1/Ndufs6/Atp5e/Surf1/Atp5h/Cox6b1/Ndufa7/Uqcrb/Cox7b/Cyc1/Ndufs8/Ndufa8/Atp5a1/Cox6a1/mt-Cytb/Cox5a/Ndufb5/Ndufb3/Ndufa9/mt-Co1/Ndufaf1/Uqcrc2/mt-Nd5/Atp5j2/mt-Co2/Cox7c/Atp5j/Atp5k/Cox20/Ndufs3/Etfb/Ndufa3/Uqcr11/Ndufs5/Ndufa4","Cxcl2/Ccl12/Ccl3/Ltb4r1/Cxcl1/Cxcl3/Adora2b/Cnr2/Ccl7/Ccrl2/Ccl4/Ccl9/Ccl6/Gpr35/Cxcl10/Anxa1/Cxcr4/Cxcl16/Cx3cr1","Cox11/Ldha/Cox4i1/Acad9/Ndufa10/Uqcrfs1/Ndufb6/Ndufv3/Mdh2/Cox7a2l/Ndufa12/Ndufb4/Ndufa6/Uqcrc1/Pdk1/Ndufv1/Ndufa11/Ndufc2/Pdk3/Atp5d/Atp5g1/Ndufb8/Suclg1/Atp5c1/Ndufs2/mt-Nd1/Fh1/Ndufb10/Ndufaf3/Ndufb9/Sdhd/Cox5b/Ndufaf7/Ndufb11/Atp5g3/Atp5l/Ucp2/Ndufs7/Cox8a/Ndufb7/mt-Co3/Atp5pb/Atp5b/Ndufa13/Ndufa1/Ndufa5/Ndufa2/Sdhb/Uqcr10/Uqcrq/Uqcrh/Ndufv2/Coq10b/Ndufc1/Ndufs6/Atp5e/Surf1/Atp5h/Vdac1/Cox6b1/Ldhb/Ndufa7/Uqcrb/Cox7b/Cyc1/Ndufs8/Idh3g/Idh3b/Ndufa8/Atp5a1/Cox6a1/Pdhb/mt-Cytb/Cox5a/Ndufb5/Ndufb3/Ndufa9/mt-Co1/Ndufaf1/Uqcrc2/mt-Nd5/Atp5j2/mt-Co2/Cox7c/Atp5j/Atp5k/Dlst/Cox20/Ndufs3/Etfb/Ndufa3/Uqcr11/Ndufs5/Ndufa4","Cenpe/Spc24/Cenpm/Ccnb1/Mad2l1/Cenpk/Ndc80/Aurkb/Kif2c/Plk1/Bub1/Cep76/Bub1b/Cenpu/Ncaph/Nuf2/Spc25/Ncapg/Nek2/Dsn1/Ska1/Kif18a/Cep72/Zwilch/Sgo1/Cdca5/Cenpn/Cenpf/Cenpa/Ccnb2/Cdc20/Haus4/Cdk1/Cenpq/Cenpt/Knl1/Ppp2r5d/Cenps/Haus7/Pmf1/Haus8/Tubg1/Cdca8/Sec13/Incenp/Ppp2r5b/Ccp110/Rangap1/B9d2/Ywhag/Ranbp2","Ang/Suv39h1/Pwp2/Npm3/Rps15/Isg20/Rps25/Rrp9/Rpl10a/Rpl5/Rps6/Rps14/Rrp1b/Rps16/Rpl11/Rpl7a/Rpl14/Rpp30/Imp4/Eif6/Rps17/Krr1/Rps7/Rps8/Rpl26/Pop4/Lyar/Mrps11/Wdr46/Nhp2/Wdr3/Exosc5/Znhit3/Rrs1/Rpl27/Nol11/Imp3/Ddx49/Rbfa/Rpl7/Nop53/Rps24/Ddx51/Rps28/Snu13/Rpp40/Rps19/Mterf4/Rrp15/Npm1/Mrto4/Rpl35a/Ddx54/Exosc2/Rpl35/Pwp1/Rrp1/Tsr3/Pa2g4/Mars1/Znhit6/Utp4/Fbl/Nob1/Ncl/Tsr1/Pih1d1/Pes1/Pop5/Dkc1/Bysl/Emg1/Usp36/Rps21/Wdr74/Gar1/Sart1/Exosc8/Exosc4/Pin4/Fam207a/Dis3/Gtf2h5/Lsm6/Ngdn/Esf1/Utp14a/Nsa2/Rps27/Fcf1/Bud23/Trir/Dimt1/C1d/Macroh2a1/Frg1/Rpl7l1/Exosc1","Rpl29/Rps15/Rpl23a/Pkm/Rpl12/Rps25/Rpl18/Rps9/Rwdd1/Rpl10a/Rpl5/Eif4ebp1/Rpl17/Rps6/Rpl31/Rps14/Rplp0/Rps27a/Rps3a1/Rps16/Rpl11/Rpl7a/Rps12/Rpl32/Rpl41/Rpl14/Rpsa/Rps5/Rpl6/Rpl8/Rpl13/Rpl3/Rpl4/Eif3d/Rpl24/Rps17/Eef2/Rps7/Rpl15/Rps13/Rps8/Rpl26/Eif3i/Rpl28/Rpl18a/Rplp2/Rps4x/Fau/Rps11/Rpl19/Rpl9/Rps23/Eif3m/Rpl27a/Rpl27/Rps26/Rpl7/Rps3/Rpl23/Rps24/Rpl36/Rpl9-ps6/Rplp1/Eif5/Rps28/Rps15a/Rpl22/Rps19/Rpl30/Rpl36a/Rps20/Rpl38/Drg2/Rpl35a/Pabpc1/Rpl34/Uba52/Dph3/Rpl37a/Rpl21/Cnbp/Rpl39/Rps18/Eif4h/Rps21/Ybx1/Rps2/Mif4gd","Cxcl2/Ccl12/Ccl3/Adam8/Cxcl1/Vegfa/Trem1/F7/Serpine1/Cxcl3/Zfp580/Tnfsf14/Cnr2/Ccl2/Ccl7/Ccrl2/Dusp1/Ccl4/Spp1/Ccr1/Sell/Ccl9/Ccr2/Cxcl14/Jaml/Il1b/Ccl6/Thbs1/Gpr35/Nr4a1/Mif/Cxcl10/Coro1a/Lgals3/Anxa1/Slamf9/Pla2g7/Cxcr4/Bst1/Cxcl16/Cx3cr1/Hmgb2/Saa3/Itgam/Trem3/Plec/Fcgr3","Cxcl2/Ccl12/Ccl3/Adam8/Cxcl1/Vegfa/Trem1/F7/Serpine1/Cxcl3/Zfp580/Tnfsf14/Cnr2/Ccl2/Ccl7/Dusp1/Ccl4/Spp1/Ccr1/Sell/Ccl9/Ccr2/Cxcl14/Jaml/Il1b/Ccl6/Thbs1/Gpr35/Mif/Cxcl10/Coro1a/Lgals3/Anxa1/Slamf9/Pla2g7/Bst1/Cxcl16/Cx3cr1/Itgam/Trem3/Plec/Fcgr3","Suv39h1/Pwp2/Npm3/Rps15/Isg20/Rps25/Rrp9/Rpl10a/Rpl5/Rps6/Rps14/Rrp1b/Rps16/Rpl11/Rpl7a/Rpl14/Rpp30/Imp4/Eif6/Rps17/Krr1/Rps7/Rps8/Rpl26/Pop4/Lyar/Mrps11/Wdr46/Nhp2/Wdr3/Exosc5/Znhit3/Rrs1/Rpl27/Nol11/Imp3/Ddx49/Rbfa/Rpl7/Nop53/Rps24/Ddx51/Rps28/Snu13/Rpp40/Rps19/Mterf4/Rrp15/Npm1/Mrto4/Rpl35a/Ddx54/Exosc2/Rpl35/Pwp1/Rrp1/Tsr3/Pa2g4/Znhit6/Utp4/Fbl/Nob1/Ncl/Tsr1/Pih1d1/Pes1/Pop5/Dkc1/Bysl/Emg1/Usp36/Rps21/Wdr74/Gar1/Sart1/Exosc8/Exosc4/Pin4/Fam207a/Dis3/Gtf2h5/Lsm6/Ngdn/Esf1/Utp14a/Nsa2/Rps27/Fcf1/Bud23/Dimt1/C1d/Frg1/Rpl7l1/Exosc1","Cox11/Cox4i1/Acad9/Ndufa10/Uqcrfs1/Ndufb6/Ndufv3/Cox7a2l/Ndufa12/Ndufb4/Ndufa6/Uqcrc1/Ndufv1/Ndufa11/Ndufc2/Ndufb8/Ndufs2/mt-Nd1/Ndufb10/Ndufaf3/Ndufb9/Cox5b/Ndufaf7/Ndufb11/Ndufs7/Cox8a/Ndufb7/mt-Co3/Ndufa13/Ndufa1/Ndufa5/Ndufa2/Uqcr10/Uqcrq/Uqcrh/Ndufv2/Coq10b/Ndufc1/Ndufs6/Surf1/Cox6b1/Ndufa7/Uqcrb/Cox7b/Cyc1/Ndufs8/Ndufa8/Cox6a1/mt-Cytb/Cox5a/Ndufb5/Ndufb3/Ndufa9/mt-Co1/Ndufaf1/Uqcrc2/mt-Nd5/mt-Co2/Cox7c/Cox20/Ndufs3/Etfb/Ndufa3/Uqcr11/Ndufs5/Ndufa4","Birc5/Cenpe/Top2a/Brca2/Lif/Ndc80/Plk1/Aspm/Bub1/Haspin/Prdm9/Bub1b/Cdc25b/Ccne1/Ncaph/Nuf2/Dnmt3l/Tesmin/Fbxo5/Fanca/Mnd1/Suv39h2/Kif18a/Rad54l/Brip1/Mastl/Psma8/Aurka/Sgo1/Ttk/Ccne2/Chtf18/Osm/Ccnb2/Cdc20","Birc5/Cenpe/Top2a/Brca2/Lif/Ndc80/Plk1/Aspm/Bub1/Haspin/Prdm9/Bub1b/Cdc25b/Ccne1/Ncaph/Nuf2/Dnmt3l/Tesmin/Fbxo5/Fanca/Mnd1/Suv39h2/Kif18a/Rad54l/Brip1/Mastl/Psma8/Aurka/Sgo1/Ttk/Cdc25c/Ccne2/Chtf18/Osm/Ccnb2/Cdc20/Cdc25a","Birc5/Ube2c/Cenpe/Spc24/Prc1/Kif23/Knstrn/Ccnb1/Mad2l1/Nusap1/Cenpk/Ndc80/Aurkb/Kif2c/Plk1/Bub1/Haspin/Kif22/Bub1b/Kif11/Ncaph/Cit/Nuf2/Spc25/Fbxo5/Ncapg/Nek2/Dsn1/Kif18a/Kif14/Gen1/Zwilch/Sgo1/Cdca5/Ttk","Birc5/Ube2c/Cenpe/Spc24/Rad18/Ccnb1/Mad2l1/Ndc80/Aurkb/Kif2c/Plk1/Bub1/Bub1b/Ncaph/Cit/Nuf2/Spc25/Fbxo5/Ncapg/Gen1/Zwilch/Cdca2/Ttk/Chtf18/Mki67","Birc5/Ube2c/Cenpe/Top2a/Spc24/Prc1/Kif23/Knstrn/Ccnb1/Mad2l1/Nusap1/Cenpk/Ndc80/Aurkb/Kif2c/Plk1/Bub1/Haspin/Kif22/Bub1b/Kif11/Ncaph/Cit/Nuf2/Spc25/Fbxo5/Ncapg/Nek2/Dsn1/Kif18a/Kif14/Gen1/Zwilch/Sgo1/Cdca5/Ttk","Birc5/Cenpe/Top2a/Brca2/Lif/Ndc80/Plk1/Aspm/Bub1/Haspin/Prdm9/Bub1b/Cdc25b/Ccne1/Ncaph/Nuf2/Dnmt3l/Tesmin/Fbxo5/Fanca/Nek2/Mnd1/Suv39h2/Kif18a/Rad54l/Brip1/Hmga2/Mastl/Psma8/Aurka/Sgo1/Ttk/Cdc25c/Ccne2/Chtf18/Mki67/Osm/Ccnb2/Cdc20/Cdc25a","Birc5/Ube2c/Cenpe/Top2a/Spc24/Prc1/Brca2/Kif23/Knstrn/Ccnb1/Mad2l1/Nusap1/Cenpk/Ndc80/Aurkb/Kif2c/Plk1/Bub1/Haspin/Prdm9/Kif22/Bub1b/Ccne1/Kif11/Ncaph/Kif4/Cit/Nuf2/Spc25/Fbxo5/Ncapg/Nek2/Dsn1/Kif18a/Brip1/Kif14/Gen1/Zwilch/Sgo1/Cdca5/Ttk/Ccne2/Cenpf","Cenpe/Spc24/Cenpm/Mad2l1/Cenpk/Ndc80/Aurkb/Kif2c/Plk1/Bub1/Bub1b/Cenpu/Nuf2/Spc25/Dsn1/Ska1/Kif18a/Zwilch/Sgo1/Cenpn/Cenpf/Cenpa/Cdc20/Cenpq/Cenpt/Knl1/Ppp2r5d/Cenps/Pmf1/Cdca8/Sec13/Incenp/Ppp2r5b/Rangap1/B9d2/Ranbp2","Ube2c/Cenpe/Spc24/Cenpm/Mad2l1/Cenpk/Ndc80/Aurkb/Kif2c/Plk1/Bub1/Bub1b/Cenpu/Nuf2/Spc25/Dsn1/Ska1/Kif18a/Zwilch/Sgo1/Cenpn/Cenpf/Cenpa/Cdc20/Cenpq/Cenpt/Knl1/Ppp2r5d/Cenps/Pmf1/Cdca8/Sec13/Incenp/Ppp2r5b/Rangap1/B9d2/Ranbp2","Cenpe/Spc24/Cenpm/Ccnb1/Mad2l1/Cenpk/Ndc80/Aurkb/Kif2c/Plk1/Bub1/Bub1b/Cenpu/Nuf2/Spc25/Dsn1/Ska1/Kif18a/Zwilch/Sgo1/Cdca5/Cenpn/Cenpf/Cenpa/Ccnb2/Cdc20/Cdk1/Cenpq/Cenpt/Knl1/Ppp2r5d/Cenps/Pmf1/Cdca8/Sec13/Incenp/Ppp2r5b/Rangap1/B9d2/Ranbp2","Mrpl54/Mrpl30/Mrpl22/Mrpl9/Mrps18c/Mrpl23/Mrpl57/Mrps11/Mrpl55/Mrpl35/Mrps26/Mrpl58/Mrpl37/Mrpl15/Mrpl43/Mrpl50/Mrps16/Mrps34/Mrpl47/Aurkaip1/Mrpl20/Gadd45gip1/Mrpl52/Mrps14/Mrpl45/Mrps12/Mrps35/Mrpl17/Mrpl13/Mrpl12/Mrps24/Mrpl36/Mrpl33/Mrpl27/Mrpl41/Mrpl14/Mrps7/Mrpl32/Mrps23/Mrpl18/Mrpl4/Mrpl2/Mrpl39/Mrpl28/Mrps18a/Mrpl40/Mrps33/Mrps17/Mrpl11/Mrpl34/Mrps31/Dap3/Mrps18b/Chchd1/Mrps36/Mrps25/Mrpl51/Mrpl49/Mrps15/Mrps10/Mrpl46/Oxa1l/Mrpl19/Mrps5/Gfm1/Mrpl42/Mrpl24","Rpl29/Rps15/Rpl23a/Rpl12/Rps25/Rpl18/Rps9/Rpl5/Rps6/Rpl31/Rps14/Rplp0/Rps3a1/Rps16/Rpl11/Rpl32/Rpl14/Rpsa/Eif3j2/Rpl6/Rpl8/Rpl13/Rpl3/Rpl4/Eif3d/Rpl24/Rps17/Rps7/Rpl15/Rps13/Rps8/Rpl26/Eif3f/Eif3i/Rpl28/Rpl18a/Eif3e/Rplp2/Rps4x/Rpl10/Rps11/Eif3b/Rpl19/Rpl9/Eif3l/Rps23/Eif3m/Rpl27a/Rpl27/Rps26/Gm11808/Rpl7/Rps3/Rpl23/Rpl10-ps3/Rps24/Eif3k/Rplp1/Eif3h/Rps28/Rps15a/Rpl22/Rps19/Rpl37/Rpl30/Rpl36a/Rps20/Rpl38/Rpl35a/Rpl34/Eif1ax/Rpl35","Rpl29/Eif2b2/Rps15/Rpl23a/Rpl12/Rps25/Rpl18/Rps9/Rpl5/Eif4ebp1/Rps6/Rpl31/Rps14/Rplp0/Rps3a1/Rps16/Rpl11/Rpl32/Rpl14/Rpsa/Eif3j2/Rpl6/Rpl8/Rpl13/Rpl3/Rpl4/Eif3d/Rpl24/Rps17/Rps7/Rpl15/Rps13/Rps8/Rpl26/Eif3f/Eif3i/Rpl28/Rpl18a/Eif3e/Rplp2/Rps4x/Rpl10/Rps11/Eif3b/Rpl19/Rpl9/Eif3l/Rps23/Eif3m/Rpl27a/Rpl27/Rps26/Gm11808/Rpl7/Rps3/Rpl23/Rpl10-ps3/Rps24/Eif3k/Rplp1/Eif3h/Eif5/Rps28/Rps15a/Rpl22/Rps19/Rpl37/Rpl30/Rpl36a/Rps20/Rpl38/Rpl35a/Pabpc1/Rpl34/Eif1ax/Rpl35/Uba52/Rpl37a/Rpl39/Rps18/Eif4h/Rps21/Rps2/Eif3c/Rps27l","Rpl29/Eif2b2/Rps15/Rpl23a/Rpl12/Rps25/Mrpl54/Rpl18/Rps9/Rpl5/Eif4ebp1/Rps6/Rpl31/Mrpl30/Rps14/Rplp0/Rps3a1/Rps16/Rpl11/Mrpl22/Rpl32/Rpl14/Rpsa/Eif3j2/Rpl6/Rpl8/Rpl13/Rpl3/Mrpl9/Rpl4/Eif3d/Rpl24/Rps17/Eef2/Rps7/Rpl15/Rps13/Rps8/Rpl26/Eif3f/Eif3i/Rpl28/Rpl18a/Eif3e/Mrps18c/Mrpl23/Eef1b2/Rplp2/Rps4x/Mrpl57/Mrps11/Rpl10/Rps11/Eif3b/Mrpl55/Rpl19/Rpl9/Eif3l/Mrpl35/Rps23/Eif3m/Rpl27a/Rpl27/Mrps26/Rps26/Gm11808/Eef1a1/Rpl7/Rps3/Rpl23/Rpl10-ps3/Rps24/Eef1g/Eif3k/Srp9/Mrpl58/Rplp1/Eif3h/Eif5/Rps28/Mrpl37/Rps15a/Mrpl15/Rpl22/Rps19/Mrpl43/Rpl37/Mrpl50/Rpl30/Rpl36a/Mrps16/Mrps34/Rps20/Rpl38/Rpl35a/Pabpc1/Rpl34/Eif1ax/Ppa1/Rpl35/Mrpl47/Aurkaip1/Mrpl20/Gadd45gip1/Mrpl52/Uba52/Mrps14/Mrpl45/Mrps12/Mrps35/Mrpl17/Mrpl13/Mrpl12/Mrps24/Rpl37a/Mrpl36/Mrpl33/Mrpl27/Rpl39/Rps18/Srp19/Mrpl41/Eif4h/Mrpl14/Mrps7/Rps21/Mrpl32/Rps2/Mrps23/Eif3c/Mrpl18/Mrpl4/Rps27l/Mrpl2/Mrpl39/Mrpl28/Srp54a/Mrps18a/Mrpl40/Eif4e/Mrps33/Eif4a1/Mrps17/Mrpl11/Rpl22l1/Eif3g/Rps27/Mrpl34/Mrps31/Dap3","Rpl29/Rps15/Rpl23a/Rpl12/Rps25/Rpl18/Rps9/Rpl5/Rps6/Rpl31/Rps14/Rplp0/Rps3a1/Rps16/Rpl11/Rpl32/Rpl14/Rpsa/Rpl6/Rpl8/Rpl13/Rpl3/Rpl4/Rpl24/Rps17/Rps7/Rpl15/Rps13/Rps8/Rpl26/Rpl28/Rpl18a/Rplp2/Rps4x/Rpl10/Rps11/Rpl19/Rpl9/Rps23/Rpl27a/Rpl27/Rps26/Gm11808/Rpl7/Rps3/Rpl23/Rpl10-ps3/Rps24/Rplp1/Rps28/Rps15a/Rpl22/Rps19/Rpl37/Rpl30/Rpl36a/Rps20/Rpl38/Rpl35a/Pabpc1/Rpl34/Rpl35","Rpl29/Rps15/Rpl23a/Rpl12/Rps25/Rpl18/Rps9/Rpl5/Rps6/Rpl31/Rps14/Rplp0/Rps3a1/Rps16/Rpl11/Rpl32/Rpl14/Rpsa/Rpl6/Rpl8/Rpl13/Rpl3/Rpl4/Rpl24/Rps17/Rps7/Rpl15/Rps13/Rps8/Rpl26/Rpl28/Rpl18a/Rplp2/Rps4x/Rpl10/Rps11/Rpl19/Rpl9/Rps23/Rpl27a/Rpl27/Rps26/Gm11808/Rpl7/Rps3/Rpl23/Rpl10-ps3/Rps24/Srp9/Rplp1/Rps28/Rps15a/Rpl22/Rps19/Rpl37/Rpl30/Rpl36a/Rps20/Rpl38/Rpl35a/Rpl34/Rpl35/Uba52/Rpl37a/Rpl39/Rps18/Srp19/Rps21/Rps2/Rps27l","Pwp2/Rpl29/Rps15/Rpl23a/Rpl12/Rps25/Rrp9/Rpl18/Rps9/Rpl5/Rps6/Rpl31/Rps14/Rplp0/Rps3a1/Rps16/Rpl11/Rpl32/Rpl14/Rpsa/Rpl6/Rpl8/Rpp30/Imp4/Rpl13/Rpl3/Rpl4/Rpl24/Rps17/Krr1/Rps7/Rpl15/Rps13/Rps8/Rpl26/Rpl28/Rpl18a/Rplp2/Rps4x/Wdr46/Rpl10/Rps11/Wdr3/Rpl19/Rpl9/Exosc5/Rps23/Rpl27a/Rpl27/Nol11/Imp3/Ddx49/Rps26/Gm11808/Rpl7/Rps3/Rpl23/Rpl10-ps3/Rps24/Rplp1/Rps28/Snu13/Rpp40/Rps15a/Rpl22/Rps19/Rpl37/Rpl30/Rpl36a/Rps20/Rpl38/Rpl35a/Rpl34/Exosc2/Rpl35/Ltv1/Uba52/Rpl37a/Utp4/Fbl/Pno1/Nob1/Ncl/Rpl39/Tsr1/Rps18/Pes1/Bysl/Emg1/Rps21/Rps2/Noc4l/Rps27l/Gnl3/Exosc8/Rpp21/Exosc4/Nop58/Dis3/Utp14a/Rpl22l1/Rps27/Fcf1/Bud23/C1d/Nip7/Rps29/Exosc1","Rpl29/Rps15/Rpl23a/Rpl12/Rps25/Rpl18/Rps9/Rpl5/Rps6/Rpl31/Rps14/Rplp0/Rps3a1/Rps16/Rpl11/Rpl32/Rpl14/Rpsa/Rpl6/Rpl8/Rpl13/Rpl3/Rpl4/Rpl24/Rps17/Rps7/Rpl15/Rps13/Rps8/Rpl26/Rpl28/Rpl18a/Rplp2/Rps4x/Rpl10/Rps11/Rpl19/Rpl9/Rps23/Rpl27a/Rpl27/Rps26/Gm11808/Rpl7/Rps3/Rpl23/Rpl10-ps3/Rps24/Rplp1/Rps28/Rps15a/Rpl22/Rps19/Rpl37/Rpl30/Rpl36a/Eif4a3/Rps20/Rpl38/Rpl35a/Pabpc1/Rpl34/Rpl35/Uba52/Upf3a/Rpl37a/Rpl39/Rps18/Magoh/Rps21/Rps2/Rps27l","Rps15/Rps25/Rps9/Eif4ebp1/Rps6/Rps14/Rps3a1/Rps16/Rpsa/Eif3j2/Eif3d/Rps17/Rps7/Rps13/Rps8/Eif3f/Eif3i/Eif3e/Rps4x/Rps11/Eif3b/Eif3l/Rps23/Eif3m/Rps26/Gm11808/Rps3/Rps24/Eif3k/Eif3h/Rps28/Rps15a/Rps19/Rps20/Pabpc1/Eif1ax/Uba52/Rps18/Eif4h/Rps21/Rps2/Eif3c/Rps27l/Eif4e/Eif4a1/Eif3g/Rps27/Eif2s3x/Rps29"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>site<\/th>\n      <th>Description<\/th>\n      <th>setSize<\/th>\n      <th>enrichmentScore<\/th>\n      <th>NES<\/th>\n      <th>pvalue<\/th>\n      <th>p.adjust<\/th>\n      <th>Count<\/th>\n      <th>.sign<\/th>\n      <th>GeneRatio<\/th>\n      <th>core_enrichment<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":9,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[2,3,4,5,6,7,9]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render"],"jsHooks":[]}</script>
</div>
<div id="number-of-gene-sets" class="section level4" number="4.0.2">
<h4><span class="header-section-number">4.0.2</span> Number of gene
sets</h4>
<pre class="r"><code>gse_df %&gt;% 
  filter(p.adjust &lt; p_thrs) %&gt;% 
  group_by(group) %&gt;% 
  summarise(n = n()) %&gt;% 
  ggplot(aes(x=group, y=n)) +
  geom_bar(stat=&quot;identity&quot;, color = &#39;black&#39;)+
  geom_text(aes(label=n), vjust=1.4, color=&quot;white&quot;, size=geom_text_size)+
  labs(
    x = &#39;Site&#39;,
    y = &#39;Number of gene sets&#39;,
    title = &#39;Number of significant gene sets by site&#39;,
    caption = paste(&quot;Adjusted P value &lt;&quot;, p_thrs)
    
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/mf-marker-gsea-c2reactome_gobp-ngset-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-set-similarity-matrix" class="section level4"
number="4.0.3">
<h4><span class="header-section-number">4.0.3</span> Gene-set similarity
matrix</h4>
<p>Heatmap showing the pair-wise similarity matrix of top 20 gene sets
with an adjusted P value &lt; 0.01 in each gene group. Heatmap colors
represent the Jaccard similarity coefficient.</p>
<pre class="r"><code>col_fun &lt;- colorRamp2(seq(0, 1, length.out = 4), brewer.pal(4, &quot;GnBu&quot;))
n_split &lt;- 9

use_mat &lt;- jc_matrix
colnames(use_mat) &lt;- colnames(use_mat) %&gt;% clean_msigdb_names %&gt;% tolower() %&gt;% str_to_title %&gt;% ep_str_wrap(70)
rownames(use_mat) &lt;- rownames(use_mat) %&gt;% clean_msigdb_names %&gt;% tolower() %&gt;% str_to_title %&gt;% ep_str_wrap(70)

ha_top &lt;- HeatmapAnnotation(
  foo = anno_block(
    # labels = c(&quot;Translation&quot;, &quot;Cell division&quot;),
    labels = 1:n_split,
    labels_gp = gpar(col = &quot;black&quot;, fontsize = 8),
    gp = gpar(lwd = 0, lty = 0))
  )

ht &lt;- Heatmap(
  use_mat,
  name = &#39;Jaccard index&#39;,
  column_split = n_split,
  row_split = n_split,
  column_title = NULL,
  row_title = NULL,
  col = col_fun,
  show_column_dend = FALSE,
  show_column_names = FALSE,
  border = TRUE,
  top_annotation = ha_top,
  heatmap_legend_param = list(title_gp = gpar(fontsize = 8),labels_gp = gpar(fontsize = 8)),
  width = unit(7, &quot;cm&quot;))

ht = draw(ht)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/mf-marker-gsea-c2reactome_gobp-jcmatrix-1.png" width="4320" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;\n\n&quot;)</code></pre>
<pre class="r"><code># Extract term order according to hclust from similarity heatmap
row_order &lt;- row_order(ht) %&gt;% set_names(1:n_split)
term_similarity_clusters &lt;- foreach(i=names(row_order), .combine = rbind) %do% {
  data.frame(
    jc_sim_cl = i, 
    row_order = row_order[[i]], 
    Description =  rownames(jc_matrix)[row_order[[i]]]
  )
}

# Add similarity order to data for dot plot
res_top %&lt;&gt;% left_join(term_similarity_clusters)</code></pre>
</div>
<div id="dotplots" class="section level4" number="4.0.4">
<h4><span class="header-section-number">4.0.4</span> Dotplots</h4>
<p>The plots below show the top 20 gene sets with an adjusted P value
&lt; 0.01 in each gene group</p>
<pre class="r"><code>use_linewidth &lt;- 1/2.141959
x_max &lt;- res_top$NES %&gt;% abs %&gt;% max()

res_top %&gt;%
  # Combine similarity clusters
  mutate(
    # jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;1&#39;), &#39;Translation&#39;, jc_sim_cl),
    # jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;3&#39;,&#39;5&#39;,&#39;6&#39;), &#39;Energy\ngeneration&#39;, jc_sim_cl),
    # jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;4&#39;, &#39;8&#39;), &#39;Cell stress&#39;, jc_sim_cl),
    # jc_sim_cl = factor(
    #   jc_sim_cl,
    #   levels = c(&#39;Translation&#39;, &#39;Energy\ngeneration&#39;, &#39;Cell stress&#39;, &#39;2&#39;, &#39;7&#39;, &#39;9&#39;)
    #   ),

    p.adjust = ifelse(p.adjust &gt; 0.05, 1, p.adjust),
    p.adjust = -log10( p.adjust),
    p.adjust = ifelse(p.adjust &gt;= 10, 10, p.adjust),
    group = recode(group, primary_tumor = &quot;PT&quot;, lung = &quot;Lung&quot;, liver = &quot;Liver&quot;),
    group = factor(group, levels = c(&#39;PT&#39;, &#39;Lung&#39;, &#39;Liver&#39;))
  ) %&gt;% 
  ggplot(aes(NES, Description, color = NES, size = p.adjust)) +
  geom_point() +
  facet_grid(rows = vars(jc_sim_cl), cols = vars(group), scale = &#39;free_y&#39;, space = &#39;free_y&#39;) +
  scale_x_continuous(limits = c(-x_max, x_max),
                        expand = expansion(mult = c(0.08, 0.08))) +
  scale_y_discrete(labels = default_labeller(50)) +
  geom_vline(xintercept = 0, lty = 3) +
  scale_size_continuous(range=c(0.3, 2.5)) +
  scale_color_gradient2(trans = &quot;reverse&quot;) +
  labs(
    size = expression(paste(&quot;lo&quot;, g[10],&quot;(Adjusted P value)&quot;)),
    color = &quot;NES&quot;
  ) +
  theme(
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, size = use_linewidth/2),
    
    axis.line = element_blank(),
    axis.ticks = element_line(color = &quot;black&quot;, size = use_linewidth/2),
    axis.text.y = element_text(size = 4),
    axis.text.x = element_text(size = 4),
    
    legend.text = element_text(size = 4), 
    legend.title = element_text(size = 4),
    legend.key.width = unit(0.2, &quot;cm&quot;),
    
    strip.background = element_blank()
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/mf-marker-gsea-c2reactome_gobp-dotplot-1.png" width="2520" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="endothelial-gene-set-enrichment-analysis"
class="section level2" number="5">
<h2><span class="header-section-number">5</span> Endothelial: Gene set
enrichment analysis</h2>
<p>Gene set enrichment analysis</p>
<pre class="r"><code>use_gse_list &lt;- gse_list[c(&quot;Lung-Endothelial-T-over-N&quot;, &quot;Liver-Endothelial-T-over-N&quot;)]
names(use_gse_list) &lt;- c(&#39;Lung&#39;, &#39;Liver&#39;)
test_type &lt;- &#39;GSEA&#39;
gset_collection &lt;- &#39;msigdb.m2.cp.reactome&#39;
p_thrs &lt;- 0.01
topn &lt;- 40

# Load all the results
gse_df &lt;- foreach(i = names(use_gse_list), .combine = rbind) %do% {
  i_1 &lt;- use_gse_list[[i]][[test_type]][[gset_collection]]@result %&gt;% 
    mutate(group = i)
  i_2 &lt;- use_gse_list[[i]][[&#39;gseGO&#39;]][[&#39;BP&#39;]]@result %&gt;% 
    filter(simplify) %&gt;% 
    dplyr::select(-simplify) %&gt;% 
    mutate(group = i)
  rbind(i_1, i_2)
} %&gt;% 
  mutate(
    group = factor(group, c(&#39;Lung&#39;, &#39;Liver&#39;)),
    Description = clean_msigdb_names(Description) %&gt;% tolower() %&gt;% str_to_title,
    Description = ifelse(grepl(&quot;^GO:&quot;, ID), paste0(&quot;GO:&quot;, Description), Description)
  ) 

# Select topn by group in selected gset
top_by_group &lt;- foreach(i = names(use_gse_list), .combine = c) %do% {
  gse_df %&gt;% 
    filter(group == i) %&gt;% 
    filter(p.adjust &lt; p_thrs) %&gt;% 
    arrange(p.adjust) %&gt;% pull(Description) %&gt;%
    head(topn)
}

# Data frame to plot
res_top &lt;- gse_df %&gt;% 
  filter(Description %in% top_by_group) %&gt;% 
  mutate(log.p.adjust = -log10(p.adjust))

# Rows and column orders by hclust
use_mat &lt;- res_top %&gt;% 
  dplyr::select(Description, log.p.adjust, group) %&gt;% 
  pivot_wider(names_from = group, values_from = log.p.adjust) %&gt;% 
  column_to_rownames(&#39;Description&#39;)
use_mat[is.na(use_mat)] &lt;- 0

row_hclust &lt;- hclust(dist(use_mat))
row_hclust_labels &lt;- row_hclust$labels[row_hclust$order]

col_hclust &lt;- hclust(dist(t(use_mat)))
col_hclust_labels &lt;- col_hclust$labels[col_hclust$order]

use_mat &lt;- use_mat[row_hclust_labels,col_hclust_labels]

# GSET clusters by kmeans using silhouette
use_kmax &lt;- ifelse(nrow(use_mat) &lt;= 20, nrow(use_mat)-5, 20)
nb_clust &lt;- fviz_nbclust(use_mat, kmeans, method = &quot;silhouette&quot;, k.max = 7)
silhouette_best_k &lt;- nb_clust$data %&gt;% slice_max(y) %&gt;% pull(clusters) %&gt;% as.character %&gt;% as.numeric
# silhouette_best_k &lt;- 5
kmeans_mat &lt;- kmeans(use_mat, silhouette_best_k, nstart = 25)
kmeans_mat_df &lt;- kmeans_mat$cluster %&gt;%
  data.frame %&gt;%
  set_names(&#39;kcluster&#39;) %&gt;%
  rownames_to_column(&#39;Description&#39;) %&gt;%
  mutate(kcluster = as.character(kcluster))


# Order factors by hclust
res_top %&lt;&gt;% 
  left_join(kmeans_mat_df) %&gt;% 
  mutate(
    Description = factor(Description, levels = row_hclust_labels),
    group = fct_relevel(group, col_hclust_labels)
  )

# Calculate jaccard similarity matrix
gset_collection &lt;- foreach(i = levels(res_top$Description)) %do% {
  res_top %&gt;% filter(Description == i) %&gt;% pull(core_enrichment) %&gt;% 
    paste(collapse = &#39;/&#39;) %&gt;% 
    strsplit(&#39;/&#39;) %&gt;% unlist %&gt;% unique
}
names(gset_collection) &lt;- levels(res_top$Description)

use_terms &lt;- res_top$Description %&gt;% unique
jc_matrix &lt;- pairwise_terms_jc(gset_collection[use_terms])</code></pre>
<div id="results-table-1" class="section level4" number="5.0.1">
<h4><span class="header-section-number">5.0.1</span> Results table</h4>
<pre class="r"><code>gse_df %&gt;% 
  filter(p.adjust &lt; p_thrs) %&gt;% 
  arrange(desc(p.adjust)) %&gt;% 
  dplyr::select(group, Description:p.adjust, Count:GeneRatio, core_enrichment)  %&gt;%
  dplyr::rename(site = group) %&gt;% 
  datatable(., 
            rownames = FALSE, 
            filter = &#39;top&#39;, 
            caption = &#39;Results from gene-set enrichment analysis&#39;,
            extensions = &#39;Buttons&#39;, 
            options = list(
              dom = &#39;Blfrtip&#39;,
              buttons = c(&#39;csv&#39;, &#39;excel&#39;)
            ))  %&gt;% 
  formatRound(c(&#39;enrichmentScore&#39;, &#39;NES&#39;, &#39;GeneRatio&#39;), digits = 2) %&gt;%
  formatSignif(c(&#39;pvalue&#39;, &#39;p.adjust&#39;), digits = 3)</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-684d012f49e01d513276" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-684d012f49e01d513276">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Lung&quot;,&quot;Liver&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"13\" data-max=\"246\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-0.733162154166808\" data-max=\"0.847385568702863\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-1.99818190412097\" data-max=\"3.09016968175297\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"1e-10\" data-max=\"0.00036950160608\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"9.7125e-09\" data-max=\"0.009577017783933\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"6\" data-max=\"129\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.154166666666666\" data-max=\"0.827586206896552\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Results from gene-set enrichment analysis<\/caption>","data":[["Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver","Liver"],["Nuclear Receptor Transcription Pathway","Signaling By Gpcr","GO:Tissue Remodeling","GO:Regulation Of Endothelial Cell Proliferation","GO:Negative Regulation Of Mitotic Nuclear Division","GO:Mammary Gland Epithelium Development","GO:Cell Activation Involved In Immune Response","GO:Mammary Gland Duct Morphogenesis","GO:Intrinsic Apoptotic Signaling Pathway In Response To Dna Damage","GO:Regulation Of Vasculature Development","GO:T Cell Differentiation","GO:Antifungal Innate Immune Response","GO:Positive Regulation Of Phosphatidylinositol 3-Kinase Signaling","GO:Regulation Of Collagen Biosynthetic Process","GO:Glial Cell Migration","GO:Regulation Of Calcium Ion Import","GO:Response To Interferon-Gamma","GO:Leukocyte Homeostasis","GO:Biological Process Involved In Symbiotic Interaction","GO:Antimicrobial Humoral Response","GO:Chemical Synaptic Transmission, Postsynaptic","GO:Positive Regulation Of Peptidase Activity","GO:Regulation Of Antigen Processing And Presentation","GO:Collagen Biosynthetic Process","GO:Alpha-Beta T Cell Activation","GO:Regulation Of Cytokine Production Involved In Immune Response","GO:Ribonucleoprotein Complex Assembly","GO:Dendrite Development","GO:Cytokine Production Involved In Immune Response","GO:Regulation Of Peptidyl-Tyrosine Phosphorylation","GO:Regulation Of Lymphocyte Proliferation","GO:Ribosome Assembly","GO:Neuroinflammatory Response","GO:Cellular Response To Tumor Necrosis Factor","GO:Leukocyte Mediated Cytotoxicity","GO:Production Of Molecular Mediator Of Immune Response","GO:Calcium Ion Transport","GO:Positive Regulation Of Cell Adhesion Mediated By Integrin","GO:Regulation Of Endothelial Cell Apoptotic Process","GO:Leukocyte Adhesion To Vascular Endothelial Cell","GO:Acute Inflammatory Response","GO:Ribonucleoprotein Complex Subunit Organization","GO:Regulation Of Endopeptidase Activity","GO:Endothelial Cell Apoptotic Process","Rhob Gtpase Cycle","GO:Regulation Of Postsynaptic Membrane Potential","GO:Regulation Of Adaptive Immune Response","GO:Regulation Of Cell Junction Assembly","GO:Phagocytosis","GO:Negative Regulation Of T Cell Activation","GO:Response To Tumor Necrosis Factor","Cell Surface Interactions At The Vascular Wall","Gpcr Ligand Binding","GO:Xenobiotic Metabolic Process","GO:Positive Regulation Of Immune Effector Process","GO:Positive Regulation Of Reactive Oxygen Species Metabolic Process","Cdc42 Gtpase Cycle","GO:Rrna Processing","GO:Cell Adhesion Mediated By Integrin","GO:Regulation Of Chemotaxis","GO:Osteoclast Differentiation","GO:B Cell Mediated Immunity","Integrin Cell Surface Interactions","Degradation Of The Extracellular Matrix","GO:Antigen Processing And Presentation Of Exogenous Peptide Antigen","Rhoc Gtpase Cycle","GO:Positive Regulation Of Neuron Apoptotic Process","GO:Humoral Immune Response","GO:T Cell Proliferation","GO:Synapse Assembly","GO:Ribosomal Large Subunit Biogenesis","GO:Cellular Response To Biotic Stimulus","GO:Positive Regulation Of Leukocyte Activation","GO:Collagen Metabolic Process","GO:Positive Regulation Of T Cell Activation","GO:Negative Regulation Of Cell Activation","Immunoregulatory Interactions Between A Lymphoid And A Non Lymphoid Cell","GO:Regulation Of T Cell Activation","Rhoa Gtpase Cycle","GO:Positive Regulation Of Neuron Death","GO:Positive Regulation Of Cytosolic Calcium Ion Concentration","GO:Regulation Of Leukocyte Proliferation","Class A 1 Rhodopsin Like Receptors","Chemokine Receptors Bind Chemokines","GO:Cell Killing","GO:Positive Regulation Of Apoptotic Signaling Pathway","GO:Myeloid Leukocyte Activation","GO:Reactive Oxygen Species Metabolic Process","Response To Elevated Platelet Cytosolic Ca2","GO:Production Of Molecular Mediator Involved In Inflammatory Response","Peptide Ligand Binding Receptors","GO:Cellular Response To Interleukin-1","Extracellular Matrix Organization","GO:Response To Interleukin-1","GO:Positive Regulation Of Erk1 And Erk2 Cascade","GO:Erk1 And Erk2 Cascade","GO:Response To Molecule Of Bacterial Origin","GO:Response To Chemokine","GO:Cellular Response To Chemokine","GO:Lymphocyte Mediated Immunity","GO:Positive Regulation Of Leukocyte Differentiation","GO:Positive Regulation Of Hemopoiesis","GO:Leukocyte Proliferation","GO:Tumor Necrosis Factor Production","GO:Adaptive Immune Response Based On Somatic Recombination Of Immune Receptors Built From Immunoglobulin Superfamily Domains","GO:Regulation Of Leukocyte Cell-Cell Adhesion","GO:Regulation Of Tumor Necrosis Factor Production","GO:Positive Regulation Of Leukocyte Migration","GO:Positive Regulation Of Inflammatory Response","GO:Cytoplasmic Translation","GO:Leukocyte Chemotaxis","GO:Granulocyte Migration","GO:Mononuclear Cell Migration","GO:Cell Chemotaxis","GO:Leukocyte Migration","GO:Myeloid Leukocyte Migration","Nonsense Mediated Decay Nmd Independent Of The Exon Junction Complex Ejc","Formation Of A Pool Of Free 40s Subunits","Srp Dependent Cotranslational Protein Targeting To Membrane","Eukaryotic Translation Initiation","Nonsense Mediated Decay Nmd","Major Pathway Of Rrna Processing In The Nucleolus And Cytosol","Activation Of The Mrna Upon Binding Of The Cap Binding Complex And Eifs And Subsequent Binding To 43s","Translation"],[34,240,130,105,47,48,194,23,97,220,222,13,47,32,48,19,102,94,180,49,50,131,15,39,122,83,156,203,96,167,156,57,47,115,83,180,226,13,39,35,59,163,235,42,61,52,156,127,178,89,131,64,97,49,198,59,75,211,57,144,79,99,41,45,25,65,68,105,155,95,67,172,243,71,155,155,53,243,115,100,100,176,76,24,119,127,163,169,109,61,51,60,128,73,132,206,207,48,48,202,144,144,228,133,228,240,128,107,94,136,148,101,125,193,246,160,84,92,83,110,104,167,58,217],[-0.6552587442550414,0.3357415667713591,0.4067242070038706,0.4359319242584312,-0.5938697168204534,-0.5968606540768295,0.3539346459733095,-0.7331621541668074,0.4505024985368248,0.3372669830591303,0.3461492478646612,0.8130824041470821,0.5628054187620606,0.6323863355582985,0.5618367407661008,0.7533053193922866,0.4339269571857485,0.4672432158442823,0.3788144259690177,0.5624723915303325,-0.618481178922948,0.4110070202261049,0.7869917482216469,0.6196114369387575,0.4318334085315544,0.4829265476052941,0.4003074484368307,-0.4601105321821133,0.4606075276191463,0.3916186701582943,0.4054718056797583,0.549859540024296,0.5886665105844664,0.4389839662849297,0.4974787931629458,0.3916509604840417,0.3607715526661327,0.8473855687028622,0.6455146040254361,0.6501579870251533,0.5635656847651747,0.3951016211012108,0.3740068803613739,0.6239424011560695,-0.6216328580221849,-0.6374590158272215,0.4184212840826243,-0.5140897708751209,0.4045612432274897,0.5089755894420502,0.4446990270382252,0.5470558306337315,0.5033315807939571,-0.6592824310488291,0.3949761876760579,0.5850036598737584,-0.6087787552182062,0.3886847013741306,0.5927893464174263,0.4428929271604528,0.5492791936713993,0.5126219328354918,0.6571634181254579,0.6483207155058215,0.7658381025505948,-0.6357644983590324,0.568421986836722,0.508850349412676,0.4384506119729419,-0.581923747131617,0.5795783043199583,0.428569422000006,0.390716205043049,0.5784431470992362,0.4457342135298429,0.4468125454472244,0.6288777782201942,0.396482706225408,-0.565725123281353,0.5162154496893846,0.5182635214372595,0.4388506596197814,0.585848816314659,0.789201694895825,0.5073603147994699,0.4940262071322921,0.4550364427830911,0.4568212579587349,0.524965468914769,0.6292802225939061,0.6677946784275586,0.6441735391741602,0.5029188768258074,0.6118976237152755,0.5066792029377567,0.4479732221324073,0.4443417323076609,0.7028531083640656,0.7028531083640656,0.4603725710270155,0.5054398722754062,0.5054398722754062,0.4451493230268073,0.5323145440649882,0.4471674439054515,0.4411555761127392,0.5361843015646046,0.5793945012213122,0.6155655325406595,0.6755111544059942,0.5811202398098474,0.6171073243498703,0.5861831211285844,0.5441425413350721,0.5328441911032706,0.5619496062722121,0.7901388147909806,0.7745361033530218,0.7890723692107,0.7438345450970393,0.7240820745655129,0.6223089958607628,0.7399254689955614,0.5989664423691693],[-1.862769179316367,1.501683289915288,1.69477207183998,1.745432544877607,-1.788726439760684,-1.803743548034843,1.584299209990531,-1.942553276147741,1.799819847807841,1.507684511145439,1.558981747961883,2.047287147893644,1.933310301932599,2.008037473832399,1.933775184000132,2.089750328432936,1.733500969385284,1.849677136019407,1.6630936033648,1.926195045235363,-1.877792556183429,1.708657911210006,2.055212176239649,2.016444420247189,1.78315756992971,1.872649740608548,1.72031025620611,-1.667781377095385,1.830851153487799,1.70178900571194,1.74250393950232,1.977519830519129,2.022146538352383,1.783128866208477,1.929079147945833,1.719449319984351,1.633720747800566,2.133660223450234,2.10074289123204,2.084747329602481,2.040757647879468,1.707046072073967,1.676438165045438,2.090644642269992,-1.95061494254039,-1.954576564577936,1.798153966989843,-1.777246416069491,1.777754480834307,1.990898647317072,1.848723922618767,2.03735443308127,1.981370833664065,-1.998181904120966,1.757853491610893,2.118387838717154,-1.952470552516358,1.734624605658378,2.131912975101126,1.880083124503308,2.093341630531044,2.035074752928141,2.210892864058604,2.244887275574021,2.249181927579314,-1.996454440332767,2.1091199946461,2.037391415753945,1.880259861355608,-1.941933241460402,2.148183481801052,1.853661552479753,1.772391809753385,2.168525835074013,1.911494995438152,1.916119334339967,2.245506934663164,1.798550180802828,-1.935154493545253,2.046551401794992,2.054671042749388,1.933160304989994,2.249656969678635,2.344503085613807,2.082324295650542,2.064524488720741,1.965995912996788,1.981330639000106,2.11990691993149,2.303712914047092,2.362209460296101,2.343376033109621,2.102232188169936,2.31317276142555,2.1109291250894,1.999850491108476,1.986158482398141,2.419136735519438,2.419136735519438,2.055565445723361,2.145595280576319,2.145595280576319,2.00947699185347,2.220146514500271,2.018587120214084,1.998611055775225,2.247223970086887,2.327557539067915,2.436841141084706,2.832714885941774,2.47554482221521,2.45726544287141,2.439449480794878,2.425134472676524,2.423187785658774,2.410731207028179,3.090169681752973,3.055510970398605,3.050531437830426,3.015177901846798,2.897399419511511,2.690561091405296,2.67370048990332,2.664498193453705],[0.0002958152211253392,0.0002834666002181305,0.0003695016060792363,0.0003538867314743963,0.0003461582493128497,0.0003392070470638058,0.0003088028680118802,0.0002947668702386286,0.0002835774101664672,0.0002650545054949175,0.0002474708414602726,0.0002404761005383885,0.0002262198320544112,0.0002151670937839455,0.0002117250258337475,0.0001753937913879554,0.0001596091156988741,0.0001542508636090112,0.0001432813305598462,0.0001414176316695309,0.0001365570808298309,0.0001283712639734303,0.0001264097918093962,0.0001206422608747041,0.0001139086577950777,0.0001036393810638574,0.0001020129629438413,9.947967373013612e-05,8.634405284493475e-05,7.447682527947564e-05,6.993881022311142e-05,6.79549593393442e-05,6.729449357975852e-05,5.654452422792041e-05,4.685058957484122e-05,4.50478427957154e-05,4.234891231663774e-05,3.734505218233939e-05,3.727648715400571e-05,3.553257829721414e-05,3.658687369208432e-05,3.616249008291368e-05,3.554259863226194e-05,2.75914978703028e-05,3.404774643916786e-05,2.586060359122959e-05,2.485553684039164e-05,1.952234537407311e-05,1.815904892435868e-05,1.388004073732741e-05,1.356569495308358e-05,1.764682634668645e-05,1.654904679257253e-05,1.087170302488976e-05,1.03153777249249e-05,8.457252202101683e-06,1.132895939045545e-05,7.276986522735606e-06,5.286599272586394e-06,4.44974403896546e-06,3.8931665551085e-06,3.38929912997179e-06,5.427564263362987e-06,4.737362025758848e-06,2.918633420589512e-06,3.889341397673139e-06,2.46878387619079e-06,2.344805377593386e-06,2.338307660195148e-06,2.187945733842503e-06,1.787050067260516e-06,1.625951135934981e-06,1.567571406732997e-06,1.503882805871295e-06,1.187520638011156e-06,1.066961261109127e-06,1.675299774412495e-06,8.093653787499216e-07,1.259861053638464e-06,7.016136276098394e-07,5.722404792401607e-07,4.229775332663359e-07,6.259413725934191e-07,4.783513585010627e-07,2.832194130723158e-07,2.611406109045147e-07,2.168801927292058e-07,1.798288283287494e-07,2.650971942898134e-07,1.695600989058461e-07,2.308613922944301e-07,8.40448211392006e-08,9.01696254927945e-08,3.537680856671896e-08,3.57985078237503e-08,2.850898228831923e-08,1.784938525575149e-08,1.575593679668116e-08,1.575593679668116e-08,1.278015120071954e-08,5.85063497818077e-09,5.85063497818077e-09,4.708343440810084e-09,3.790184648580954e-09,3.539397005679216e-09,3.171539217852767e-09,2.360491952526594e-09,1.146605903395285e-09,3.463382934373536e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10,1e-10],[0.009577017783932856,0.009576241233455973,0.00929058741225961,0.008986953545792294,0.008895536842273578,0.008789962204270762,0.008254788245433367,0.007963409223095715,0.007743492829223046,0.007316368659829815,0.006906068152619365,0.006785433970191528,0.00645489060114806,0.006209282212094656,0.006180180495457491,0.005240147449761325,0.004883462039967358,0.004748122038001017,0.004508032423610956,0.004489125945309673,0.004362096940469881,0.00417947211359649,0.004169060601298204,0.004004849954134785,0.003806197848297367,0.003485989512737296,0.003454158925278465,0.003390988341445378,0.003024423754478783,0.002645229339821376,0.002537280122308449,0.002483044881183664,0.002476729948489808,0.00212733065595265,0.001802682912504686,0.001759984565841835,0.001680391606689087,0.001542077398651234,0.001542077398651234,0.001535741582496663,0.001535741582496663,0.001535741582496663,0.001535741582496663,0.00125122515788632,0.001202504499237883,0.001194054596725955,0.001177263270039816,0.0009443237348087366,0.0008868250912194015,0.0006979873950978803,0.000689001646667115,0.0006529325748273988,0.0006429304678914429,0.000569251336736238,0.000557359611328655,0.0004720261970821368,0.0004632948129675729,0.0004176802558859072,0.0003274468012861744,0.0002825031246738196,0.0002672080126134604,0.0002390868094600934,0.0002342898573685022,0.000216525311412625,0.0002094722526175014,0.0001888761416245018,0.0001871485568234779,0.0001804434320120728,0.0001804434320120728,0.0001736340059716574,0.0001440702744700978,0.0001353804232690781,0.0001326949195799482,0.0001294613690003442,0.0001058143389554151,9.676957580666532e-05,8.678052831456723e-05,7.612531034575652e-05,6.992228847693474e-05,6.723576631378065e-05,5.589248834732262e-05,4.212358610705334e-05,3.741203434654512e-05,3.09732504629438e-05,2.876942797988583e-05,2.706802373028633e-05,2.294863539315909e-05,1.943299189535571e-05,1.8725501814835e-05,1.87216465726694e-05,1.793793018127722e-05,1.016342015633333e-05,7.784644334211258e-06,5.042124123043451e-06,5.042124123043451e-06,4.524910032574168e-06,3.126104403929719e-06,2.860526176360106e-06,2.860526176360106e-06,2.596415517938182e-06,1.415017859722864e-06,1.415017859722864e-06,1.281505289879481e-06,1.132373401773098e-06,1.123537336990296e-06,1.073883179164947e-06,8.563527590630407e-07,4.47970106411127e-07,1.858050319699088e-07,7.255714285714286e-08,7.255714285714286e-08,7.255714285714286e-08,7.255714285714286e-08,7.255714285714286e-08,7.255714285714286e-08,7.255714285714286e-08,9.7125e-09,9.7125e-09,9.7125e-09,9.7125e-09,9.7125e-09,9.7125e-09,9.7125e-09,9.7125e-09],[11,37,26,27,16,16,50,11,26,39,44,6,15,12,10,8,21,26,34,15,14,38,8,13,36,21,59,57,26,36,40,30,18,26,29,45,39,7,13,14,20,63,58,14,23,16,40,32,40,25,30,23,27,16,48,20,35,90,21,34,22,31,18,24,12,23,21,31,41,36,27,37,61,25,43,41,24,65,40,29,28,45,26,15,35,37,42,40,21,18,23,21,42,22,36,46,46,22,22,63,39,39,60,39,65,71,38,36,36,82,48,33,37,55,79,46,64,76,64,87,65,78,48,129],["supressed","activated","activated","activated","supressed","supressed","activated","supressed","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","supressed","activated","activated","activated","activated","activated","activated","supressed","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","supressed","supressed","activated","supressed","activated","activated","activated","activated","activated","supressed","activated","activated","supressed","activated","activated","activated","activated","activated","activated","activated","activated","supressed","activated","activated","activated","supressed","activated","activated","activated","activated","activated","activated","activated","activated","supressed","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated","activated"],[0.3235294117647059,0.1541666666666667,0.2,0.2571428571428571,0.3404255319148936,0.3333333333333333,0.2577319587628866,0.4782608695652174,0.2680412371134021,0.1772727272727273,0.1981981981981982,0.4615384615384616,0.3191489361702128,0.375,0.2083333333333333,0.4210526315789473,0.2058823529411765,0.2765957446808511,0.1888888888888889,0.3061224489795918,0.28,0.2900763358778626,0.5333333333333333,0.3333333333333333,0.2950819672131147,0.2530120481927711,0.3782051282051282,0.2807881773399015,0.2708333333333333,0.2155688622754491,0.2564102564102564,0.5263157894736842,0.3829787234042553,0.2260869565217391,0.3493975903614458,0.25,0.1725663716814159,0.5384615384615384,0.3333333333333333,0.4,0.3389830508474576,0.3865030674846626,0.2468085106382979,0.3333333333333333,0.3770491803278688,0.3076923076923077,0.2564102564102564,0.2519685039370079,0.2247191011235955,0.2808988764044944,0.2290076335877863,0.359375,0.2783505154639175,0.3265306122448979,0.2424242424242424,0.3389830508474576,0.4666666666666667,0.4265402843601896,0.3684210526315789,0.2361111111111111,0.2784810126582278,0.3131313131313131,0.4390243902439024,0.5333333333333333,0.48,0.3538461538461539,0.3088235294117647,0.2952380952380952,0.2645161290322581,0.3789473684210526,0.4029850746268657,0.2151162790697674,0.2510288065843622,0.352112676056338,0.2774193548387097,0.2645161290322581,0.4528301886792453,0.2674897119341564,0.3478260869565217,0.29,0.28,0.2556818181818182,0.3421052631578947,0.625,0.2941176470588235,0.2913385826771653,0.2576687116564417,0.2366863905325444,0.1926605504587156,0.2950819672131147,0.4509803921568628,0.35,0.328125,0.3013698630136986,0.2727272727272727,0.2233009708737864,0.2222222222222222,0.4583333333333333,0.4583333333333333,0.3118811881188119,0.2708333333333333,0.2708333333333333,0.2631578947368421,0.2932330827067669,0.2850877192982456,0.2958333333333333,0.296875,0.3364485981308411,0.3829787234042553,0.6029411764705882,0.3243243243243243,0.3267326732673267,0.296,0.2849740932642487,0.3211382113821138,0.2875,0.7619047619047619,0.8260869565217391,0.7710843373493976,0.7909090909090909,0.625,0.4670658682634731,0.8275862068965517,0.5944700460829493],["Nr1h4/Nr4a2/Nr3c2/Esr1/Ar/Rxrg/Nr6a1/Ppara/Rora/Thrb/Esrrg","Rgs1/Ccl12/Ccl3/Edn1/Ccl7/Cxcl13/Cxcl2/C3ar1/Ccl4/Ccr5/Ccl9/Apln/Adm/Ccr1/Ccl6/Rgs2/Pde4c/Nmb/Gpr143/Adcy7/Cxcl1/Ccl19/Prkcd/Fgd2/Cysltr1/S1pr1/Btk/Gnal/Ccrl2/Cxcr4/Fn1/F2r/Pf4/Ednrb/Ackr3/Ccl5/Ackr2","Tmem119/Spp1/Itgb3/Timp1/Ccr2/Csf1r/Angpt2/Rassf2/Ctss/Syk/Fcgr4/Acp5/Lrp1/S1pr1/Adam8/F2r/Klf6/Gja1/Car2/Ptk2b/Mmp14/Foxc1/Cd24a/Il18/Fgfr3/Snx10","Lrg1/Ccl12/Ccl2/Itgb3/Pdgfb/Thbs1/Apln/Cnmd/Ager/Arg1/Bmp4/Jun/Nrarp/Gja1/Xdh/Zfp580/Tnf/Egf/Pik3cd/Fgfr3/Pold4/Pdcd6/Jcad/Rgcc/Aimp1/Bmpr2/Itga4","Bub1b/Cdk5rap2/Cenpe/Anapc15/Aurkb/Ndc80/Birc5/Mtbp/Cdca8/Tom1l2/Mad1l1/Tex14/Esr1/Spdl1/Kntc1/Fgfr2","Lbh/Pml/Etv5/Ncoa3/Ccnd1/Rreb1/Csf1/Btrc/Ntn1/Foxf1/Esr1/Ar/Robo1/Prlr/Csmd1/Fgfr2","Trem2/Ccr2/Itgam/Lgals3/Lat2/Il4ra/Slc11a1/Syk/Clec4e/Shld3/Pck1/Rps6/Nlrp3/Lcp1/Itgb2/Ccl19/Nckap1l/Lbp/Lrp1/Tyrobp/Icosl/Prkcz/Gadd45g/Ptpn6/Ptprc/Irf8/Icam1/Lilrb4a/Ptk2b/Cd74/Dock10/Ercc1/Fgr/Sbno2/H2-DMb1/Apbb1ip/Il18/Spi1/Itgal/Nfkbiz/Il6ra/Mfng/Il33/Tnfaip3/Stx11/Mad2l2/Psen2/Cplx2/Coro1a/Vamp2","Pml/Etv5/Ncoa3/Csf1/Btrc/Ntn1/Esr1/Ar/Robo1/Csmd1/Fgfr2","Ddit4/Bcl2a1b/Clu/Ier3/Myc/Nupr1/Rps3/Cdkn1a/Mif/Ackr3/Rpl26/Tpt1/Bcl2l11/Cd44/Cd74/Tnf/Pnp/Skil/Sod2/Mcl1/Plscr1/Tnfrsf1a/Htra2/Bid/Xpa/Cdkn2d","Lrg1/Serpine1/Ccl2/Itgb3/C3ar1/Ccr2/Thbs1/Adm/Angpt2/Lgals3/Cnmd/Emilin2/Runx1/Tnfrsf12a/Itgb2/Cyp1b1/Cysltr1/Glul/Cxcr4/Pgk1/Ccl5/Ninj1/Col4a2/Btg1/Ptk2b/Tnf/Foxc1/Pik3cd/Dcn/Gadd45a/Ctsh/Rhob/Foxo4/Il1b/Pkm/Pdcd6/Jcad/Rgcc/Ecscr","Zbtb16/Runx3/Ccr2/Ptpn22/Lilrb4b/Ctla2a/Runx1/Mafb/Il4ra/Syk/Clec4e/Pck1/Rps6/Nlrp3/Ccl19/Nckap1l/Tnfsf9/Prkcz/Bmp4/Rhoh/Crtam/Adam8/Gadd45g/Egr1/Ctsl/Nrarp/Ptprc/Lilrb4a/Gimap5/Cd44/Cd74/Pnp/Ikzf1/H2-Aa/Il18/Spi1/Psap/Il1b/Nfkbiz/Il6ra/Hsp90aa1/Fas/Zfp36l1/Gba","Clec4a2/Clec4n/Clec4e/Clec4a1/Spi1/Clec4a3","Trem2/Pdgfb/Selp/Hcst/Insr/Osm/Ptpn6/Fn1/F2r/Ccl5/Egf/Dcn/Fgr/Il18/Pik3ap1","Serpine1/Ccl2/Ccn2/Pdgfb/Cygb/Ager/Bmp4/Fn1/F2r/Il18/Rgcc/Il6ra","Ccl12/Ccl2/Ccl3/Trem2/Ccr2/Col3a1/Vcan/Lrp1/Fn1/Adgrg1","Ccl12/Serpine1/Ccl2/Pdgfb/Ace/Lgals3/Cxcr4/Egf","Ccl12/Ccl8/Ccl2/Ccl3/Ccl7/Capg/Ccl4/Ccl9/Ccl6/Slc11a1/Arg1/Ccl19/Ifitm6/Ccl5/Irf8/Cd74/H2-Aa/Cdc42ep4/H2-Ab1/Ifitm2/Gsn","Ccl2/Ccr2/Itgam/Pirb/Rps6/Nckap1l/Cfh/Mif/F2r/Tnfrsf13b/Lilrb4a/Gimap5/Bcl2l11/Cd44/Cd74/Dock10/Pik3cd/Cd24a/Il18/Skil/Lipa/Fas/Tnfaip3/Axl/Coro1a/Ikbkb","Ctsj/Ccl12/Ccl8/Itgb3/Siglec1/Csf1r/Ch25h/Insr/Ncf1/Cxcl1/Arg1/Slc3a2/Hspa8/Tfap4/Fn1/Jun/Ctsl/Ifitm6/Ccl5/Icam1/Trem1/Fuca2/Bcl2l11/Cd74/Gpx1/Ctsb/Hpn/Chmp4b/Ifitm2/Nectin2/Gsn/Inpp5k/Vapa/Axl","Ccl12/Ccl8/Cxcl13/Cxcl2/Adm/Lgals3/Cxcl14/Cxcl1/Ccl19/Wfdc17/Pf4/Fau/Rps19/Lgals4/Rpl30","Unc13b/Lrrk2/Ppp3ca/Abat/Stx1a/Atxn1/Cux2/Npff/Tmem108/Insyn2a/Prkn/Slc8a3/Anks1b/Grid2","Ccn2/Ccn1/Laptm5/Myc/Aph1b/Syk/Rps3/Ager/Nlrp3/Tfap4/Rack1/Acer2/F2r/Xdh/Picalm/Bcl2l11/S100a8/Tnf/Pdcd2/Ctsh/Pdcd6/S100a9/Gsn/Ifi207/Efna1/Fas/Anp32b/Aph1c/Ddrgk1/Fis1/Htra2/Bid/Gramd4/Rest/Lgmn/Casp8ap2/Aph1a/Psenen","Trem2/Thbs1/Cd68/Slc11a1/Ccl19/Cd74/H2-Ab1/Hfe","Serpine1/Ccl2/Ccn2/Pdgfb/Cygb/Ager/Arg1/Bmp4/Fn1/F2r/Il18/Rgcc/Il6ra","Zbtb16/Runx3/Ccr2/Clec4a2/Ptpn22/Lilrb4b/Runx1/Il4ra/Myc/Syk/Ager/Clec4a1/Nlrp3/Ccl19/Nckap1l/Prkcz/Crtam/Gadd45g/Ctsl/Ptprc/Cebpb/Lilrb4a/Gimap5/Cd44/Pnp/Cd24a/Il18/Psap/H2-Ab1/Nfkbiz/Il6ra/Clec4a3/Rpl22/Hfe/Zap70/Twsg1","Ccr2/Lilrb4b/Gprc5b/Arid5a/Laptm5/Syk/Arg1/Nlrp3/Acp5/Prkcz/Mif/Lilrb4a/Gimap5/Litaf/Cd74/Tnf/Il18/Il1b/Axl/Nlrx1/Hfe","Rps15/Rpl23a/Nop53/Rps25/Rpl5/Rpl6/Rpsa/Rps14/Rps5/Eif5/Rpl24/Rpl11/Rpl10/Celf2/Rps19/Rps28/Rplp0/Eif6/Hsp90aa1/Sf3b1/Mrto4/Snu13/Nopchap1/Rpl38/Prpf39/Prmt5/Znhit3/Eif3d/Rps2/Eif3a/Snrpd2/Mrps11/Prpf19/Eif4b/Mcts2/Eif4h/Luc7l/Nol3/Strap/Wdr77/Mrps7/Scaf11/Rps27/Eif2s2/Ddx28/Hsp90ab1/Mrpl20/Naf1/Ramac/Zrsr2/Tarbp2/Snrpb/Cpsf7/Dhx9/Snrpc/Lsm4/Rrp7a/Luc7l2/Ythdc1","Strn/Stau2/Lzts1/Hecw2/Trak1/Prex2/Ankrd27/Sipa1l1/Cdkl3/Phactr1/Dynlt1b/Abl2/Klf7/Pias2/Plk2/Eef2k/Myo6/Cdk5r1/Nrp1/Rere/Shank3/Itsn1/Ache/Rtn4ip1/Asap1/Pdlim5/Cpeb3/Rapgef4/Lrrk2/Arid1b/Srgap2/Cask/Ppp3ca/Kalrn/Numb/Cdkl5/Dynlt1c/Ntn1/Fyn/Tanc2/Rbfox2/Nedd4l/Cacna1a/Mapkapk5/Farp1/Rapgef2/Cux2/Tnik/Dnm3/Robo1/Sdk1/Disc1/Ptprd/Ppfia2/Prkg1/Ctnnd2/Camk1d","Ccr2/Lilrb4b/Gprc5b/Arid5a/Laptm5/Slc11a1/Syk/Arg1/Nlrp3/Acp5/Prkcz/Mif/Lilrb4a/Gimap5/Crlf2/Trem1/Litaf/Cd74/Tnf/Il18/Il1b/Gba/Axl/Psen2/Nlrx1/Hfe","Itgb3/Trem2/Pdgfb/Ace/Ptpn22/Csf1r/Socs3/Lilrb4b/Gprc5b/Syk/Clec4n/Prnp/Osm/Itgb2/Prkcd/Prkcz/Ptpn6/Mif/Ptprc/Ccl5/Icam1/Lilrb4a/Ehd4/Cd44/Ptk2b/Cd74/Egf/Cd24a/Samsn1/Il18/Fgfr3/Gprc5a/Il6st/Il6ra/Hgf/Efna1","Ccr2/Ptpn22/Lilrb4b/Lgals3/Laptm5/Syk/Prnp/Rps3/Ager/Arg1/Ccl19/Nckap1l/Cdkn1a/Tyrobp/Icosl/Tnfsf9/Btk/Bmp4/Crtam/Ptpn6/Aif1/Mif/Rassf5/Ptprc/Tnfrsf13b/Slfn1/Ccl5/Cebpb/Lilrb4a/Cd44/Lrrc32/Cd74/Sdc4/Pnp/H2-Aa/Cd24a/H2-DMb1/Il18/Il1b/Itgal","Rps15/Rpl23a/Nop53/Rps25/Rpl5/Rpl6/Rpsa/Rps14/Rps5/Rpl24/Rpl11/Rpl10/Rps19/Rps28/Mpv17l2/Rplp0/Eif6/Mrto4/Rpl38/Rps2/Ddx3x/Nip7/C1qbp/Mrps11/Mrps7/Rps27/Sbds/Ddx28/Mrpl20/Ngrn","Trem2/Itgam/Ulk4/C1qa/Clu/Nupr1/Ager/Nr1d1/Ctsc/Lrp1/Tyrobp/Aif1/Jun/Myo5a/Tnf/Il1b/Il33/Psen2","Ccl12/Ccl8/Ccl2/Ccl3/Ccl7/Ccl4/Ccl9/Ccl6/Laptm5/Syk/Rps3/Ccl19/Nr1d1/Dbn1/Fabp4/Zfp36/Ccl5/Nfkbia/Birc2/Ptk2b/Tnf/Plvap/Adamts7/Inpp5k/Zfp36l1/Gba","Ccl2/Itgam/Fcgr1/Fcgr3/Ncf1/Ager/Cxcl1/Arg1/H2-T22/Gm8909/Nckap1l/Ctsc/Tyrobp/Crtam/Hspa8/Ptpn6/Ptprc/Gimap5/Trem1/Pnp/Il18/Ctsh/Spi1/Nectin2/Pvr/Stx11/Coro1a/H2-Q10/Gfus","Ccr2/Ptpn22/Lilrb4b/Gprc5b/Arid5a/Laptm5/Il4ra/Slc11a1/Syk/Shld3/Arg1/Nlrp3/Acp5/Icosl/Prkcz/Btk/Mif/Ptprc/Lilrb4a/Gimap5/Crlf2/Trem1/Litaf/Cd74/Tnf/Ercc1/Lgals4/Pou2f2/H2-Aa/H2-DMb1/Il18/Il1b/Dnajb9/H2-Ab1/Il33/Fas/Gba/Tnfaip3/Axl/H2-Eb1/Mad2l2/Psen2/Nlrx1/Hfe/Phb2","Ccl12/Ccl8/Serpine1/Ccl2/Edn1/Ccn2/Cacng1/Itgb3/Pdgfb/Ace/Ccr5/Ptpn22/Ccr1/Lgals3/Cd33/Trpm2/Prnp/Gem/Ccl19/Gja4/Cysltr1/Bmp4/Ptpn6/Cxcr4/Myo5a/Kcne3/F2r/Ednrb/Ptprc/Gja1/Ccl5/Icam1/Tmem37/Gimap5/Tmem38b/Ptk2b/Fkbp1b/Egf/Gsto1","Cxcl13/Itgb3/Syk/Nckap1l/Ptpn6/Ccl5/Cd24a","Ccl12/Serpine1/Ccl2/Thbs1/Angptl4/Ager/Bmp4/Icam1/Tnf/Rgcc/Ecscr/Tnfaip3/Itga4","Lrg1/Ccr2/Itgam/Selp/Sele/Itgb2/Itgb7/Icam1/Tnf/Selplg/Irak1/Chst2/Jam2/Itga4","Saa3/Serpina3n/Ccr5/Fcgr1/Fcgr3/Cd163/Alox5ap/Nupr1/Cxcl1/Nlrp3/Lbp/Adam8/Fn1/Ednrb/Ccl5/Icam1/Trem1/Tnf/Ctnnbip1/Il1b","Rps15/Rpl23a/Nop53/Rps25/Rpl5/Rpl6/Rpsa/Rps14/Rps5/Eif5/Rpl24/Rpl11/Rpl10/Celf2/Rps19/Rps28/Rplp0/Eif6/Hsp90aa1/Sf3b1/Mrto4/Snu13/Nopchap1/Rpl38/Prpf39/Prmt5/Znhit3/Eif3d/Rps2/Eif3a/Tfip11/Snrpd2/Mrps11/Vcp/Prpf19/Eif4b/Mcts2/Eif4h/Luc7l/Nol3/Strap/Wdr77/Mrps7/Scaf11/Rps27/Eif2s2/Ddx28/Hsp90ab1/Mrpl20/Naf1/Zfand1/Ramac/Zrsr2/Tarbp2/Snrpb/Cpsf7/Dhx9/Snrpc/Kif5b/Lsm4/Rrp7a/Luc7l2/Ythdc1","Serpina3n/A2m/Serpine1/Ccn2/Timp1/Ccn1/Thbs1/Laptm5/Myc/Serpina3g/Aph1b/Syk/Serpina3f/Rps3/Ager/Nlrp3/Serpina3i/Plaur/Tfap4/Rack1/Acer2/F2r/Xdh/Picalm/Bcl2l11/S100a8/Cd44/Herpud1/Serpinb8/Tnf/Timp3/Gpx1/Pdcd2/Pdcd6/S100a9/Gsn/Hgf/Ifi207/Efna1/Fas/Anp32b/Hspa1b/Aph1c/Cst3/Fis1/Htra2/Bid/Gramd4/Rest/Igbp1/Cdkn2d/Lgmn/Klf4/Furin/Casp8ap2/Atp13a2/Aph1a/Psenen","Ccl12/Serpine1/Ccl2/Thbs1/Angptl4/Ager/Bmp4/Icam1/Tnf/Rgcc/Ecscr/Bmpr2/Tnfaip3/Itga4","Bcr/Anln/Arhgap32/Arhgef10/Rhpn2/Pkn3/Arhgap35/Tjp2/Cav1/Prex1/Arhgef10l/Ophn1/Arhgef25/Stk10/Arhgap39/Racgap1/Dlc1/Stard13/Arhgef3/Ect2/Diaph3/Depdc1b/Arhgap26","Unc13b/Lrrk2/Ppp3ca/Abat/Stx1a/Rgs7bp/Atxn1/Nr3c2/Cux2/Npff/Tmem108/Insyn2a/Prkn/Slc8a3/Fgf14/Grid2","Trem2/Ccr2/Fcgr1/Lilrb4b/Fcgr3/Arid5a/Il4ra/Slc11a1/Adcy7/Clec4n/Shld3/Ager/Arg1/Nlrp3/H2-T22/Gm8909/Ccl19/Nckap1l/Prkcz/Hspa8/Ptpn6/Mif/Ptprc/Cd48/Lilrb4a/Gimap5/Cd44/Cd74/Tnf/Pnp/Cd24a/Samsn1/Il18/Il1b/H2-Ab1/Nfkbiz/Nectin2/Il33/Pvr/Tnfaip3","Peak1/Lims1/Mdga1/Phldb2/Vstm5/Pdlim5/Cav1/Rapgef1/Adgrl2/Il1rap/Ptk2/Ntn1/Flrt3/Smad3/Nedd4l/Dlc1/Farp1/Adnp/St8sia2/Rapgef2/Cux2/Prkca/Arhgap6/Wdpcp/Bcas3/Ptprd/Aplnr/Flrt2/Ntng2/Lrrtm3/Asic2/Grid2","Ccl2/Itgb3/Trem2/Ccr2/Itgam/Cd300lf/Thbs1/Fcgr1/Fcgr3/Abca1/Ncf4/Slc11a1/Syk/Pld4/Ager/Itgb2/Nckap1l/Lbp/Lrp1/Tyrobp/Rhoh/Myo1g/Hspa8/Aif1/Rack1/Cfp/Ptprc/Irf8/Rab34/Mfge8/Marco/Hck/Tnf/Tgm2/Fgr/Il1b/Itgal/Ncf2/Gsn/C4b","Runx3/Ptpn22/Lilrb4b/Lgals3/Laptm5/Runx1/Il4ra/Prnp/Tnfaip8l2/Arg1/Nckap1l/Bmp4/Crtam/Ptpn6/Nrarp/Slfn1/Cebpb/Lilrb4a/Gimap5/Cd44/Lrrc32/Cd74/Sdc4/H2-Aa/Cd24a","Ccl12/Ccl8/Ccl2/Ccl3/Ccl7/Ccl4/Ccl9/Ccl6/Laptm5/Cd14/Syk/Insr/Rps3/Ccl19/Nr1d1/Dbn1/Fabp4/Zfp36/Ccl5/Aff3/Nfkbia/Birc2/Ptk2b/Tnf/Map2k7/Plvap/Adamts7/Inpp5k/Zfp36l1/Gba","Itgb3/Jaml/Itgam/Selp/Angpt2/Sele/Thbd/Itgb2/Tnfrsf10b/Ptpn6/Fn1/Mif/Procr/Pf4/Cd48/Trem1/Cd44/Cd74/Sdc4/Esam/Selplg/Itgal/Jam2","Ccl12/Ccl3/Edn1/Ccl7/Cxcl13/Cxcl2/C3ar1/Ccl4/Ccr5/Ccl9/Apln/Adm/Ccl6/Nmb/Gpr143/Cxcl1/Ccl19/Cysltr1/S1pr1/Ccrl2/Cxcr4/F2r/Pf4/Ednrb/Ackr3/Ccl5/Ackr2","Acaa1b/Gsta3/Cyp2d10/Gstt1/Cyp2c50/Cyp2j5/Cyp2r1/Fmo2/Cyp2c70/Cyp2c39/Cyp1a1/Rora/Cyp2c29/Cyp2c40/Sult2a8/Abcb11","Ccl2/Trem2/Ccr2/Itgam/Fcgr1/Ptpn22/Fcgr3/Gprc5b/Arid5a/Laptm5/Il4ra/Syk/Shld3/Cxcl1/Pck1/Arg1/Nlrp3/H2-T22/Itgb2/Gm8909/Ccl19/Lbp/Tyrobp/Prkcz/Crtam/Hspa8/Ptpn6/Mif/Cfp/Ptprc/Lilrb4a/Gimap5/Rps19/Cd74/Tnf/Pnp/Fgr/Cd24a/H2-DMb1/Il18/Spi1/Il1b/Dnajb9/Nfkbiz/Nectin2/C4b/Il33/Pvr","Lcn2/Pdgfb/Itgam/Thbs1/Syk/Ager/Cxcl1/Itgb2/Prkcd/Cdkn1a/Cyp1b1/Tyrobp/Xdh/Ptk2b/Nnt/Gadd45a/Il18/Sod2/Tspo/Dcxr","Prex2/Arap1/Fam13b/Fgd4/Arhgap45/Arhgef11/Dock8/Bcr/Arhgap32/Arhgef10/Arhgef9/Itsn1/Trio/Arap2/Arhgap35/Cav1/Prex1/Dock7/Dnmbp/Arhgap10/Srgap2/Ophn1/Arhgef25/Arhgap39/Srgap1/Racgap1/Dlc1/Stard13/Vav3/Ect2/Farp1/Plekhg1/Rasgrf2/Depdc1b/Arhgap26","Rps6/Rps15/Nop53/Rps25/Rpl10a/Rpl5/Rps16/Rps14/Rps7/Rpl7/Rps17/Rpl14/Rps8/Las1l/Rpl26/Rpl11/Rps19/Imp3/Rps28/Rnasel/Rps24/Pop4/Rpl7a/Npm1/Tsr3/Rpl35a/Eif6/Mrto4/Snu13/Rpl35/Suv39h1/Helb/Rpl27/Nhp2/Znhit3/Nol8/Tsr1/Rpusd2/Rpusd3/Ncl/Ftsj3/Rpp40/Npm3/Fbl/Kri1/Rps21/Ddx49/Rps27rt/Mrps11/Rrp1/Gar1/Ddx47/Esf1/Lyar/Ddx54/Pa2g4/Rpf1/Ddx27/Rps27/Sbds/Sart1/Naf1/Ddx51/Pes1/Nop14/Pop7/Exosc4/Ddx56/Gtf2h5/Rpp30/Pop5/Riok3/Mphosph6/Imp4/Wdr46/Riok1/Nsa2/Exosc9/Fdxacb1/Utp3/Tsr2/Dkc1/Rrp7a/Wdr18/Pin4/Mphosph10/Utp15/Exosc5/Nsun5/Rsl1d1","Serpine1/Cxcl13/Itgb3/Itgam/Itga2b/Itga3/Syk/Itgb2/Nckap1l/Cyp1b1/Fermt3/Ptpn6/Acer2/Itgb7/Ccl5/Icam1/Itgb5/Cd24a/Itgal/Efna1/Itga4","Ccl12/Serpine1/Ccl2/Edn1/Ccl7/Cxcl13/C3ar1/Trem2/Pdgfb/Ccr2/Thbs1/Csf1r/Angpt2/Ccr1/Cxcl14/Sema3a/Ager/Sema6b/Ccl19/Nckap1l/Lbp/S1pr1/Aif1/Cxcr4/Fn1/Mif/Ccl5/Sema3f/Ninj1/Zfp580/Trem1/Ptk2b/Gpsm3/Cd74","Ccl3/Itgb3/Trem2/Cd300lf/Csf1r/Lilrb4b/Ccr1/Rassf2/Fos/Mafb/Junb/Tyrobp/Adam8/Ccl5/Cebpb/Lilrb4a/Ninj1/Car2/Tnf/Sbno2/Fgfr3/Snx10","Csf2rb/Csf2rb2/Trem2/Fcgr1/Fcgr3/Pirb/C1qa/C1qc/C1qb/Il4ra/Shld3/Prkcd/Icosl/Btk/Ptpn6/Cfh/Ptprc/Gimap5/Crlf2/Cd74/Tnf/Ercc1/Pou2f2/H2-Aa/H2-DMb1/H2-Ab1/Nectin2/C4b/Fas/H2-Eb1/Mad2l2","Spp1/Itgb3/Itgam/Itga2b/Thbs1/Itga3/Col3a1/Itgb2/Fn1/Itgb7/Icam1/Col4a2/Cd44/Col13a1/Itgal/Jam2/Col4a1/Itga4","A2m/Capn12/Spp1/Timp1/Ctss/Adamts4/Col3a1/Mmp19/Adam8/Fn1/Ctsl/Mmp17/Adamts5/Col4a2/Cd44/Mmp14/Dcn/Col13a1/Ctsb/Phykpl/Nid1/Col4a1/Capn5/Adam15","Clec4a2/Fcgr1/Fcgr3/Ctss/Ifi30/Clec4a1/Cd74/H2-Aa/H2-DMb1/H2-Ab1/Clec4a3/H2-Eb1","Fmnl3/Bcr/Anln/Arhgap32/Arhgef10/Fmnl2/Pkn3/Arhgap35/Tjp2/Cav1/Prex1/Arhgef10l/Arhgap18/Ophn1/Arhgef25/Stk10/Arhgap39/Racgap1/Dlc1/Stard13/Diaph3/Depdc1b/Arhgap26","Ccl3/Lcn2/Itgam/Nupr1/Prnp/Ager/Tyrobp/Egr1/Jun/Bcl2l11/Tnf/Il18/Fgfr3/Map2k7/Il1b/Ctsz/Ncf2/Mcl1/Fas/Fis1/Tnfrsf1a","Ccl12/Ccl8/A2m/Cxcl13/Cxcl2/Trem2/Ccr2/Adm/C1qa/Lgals3/C1qc/C1qb/Cxcl14/Cxcl1/Ccl19/Ptpn6/Cfh/Wfdc17/Cfp/Pf4/Ptprc/Fau/Gimap5/Rps19/Tnf/Lgals4/Rpl30/Il1b/Rgcc/S100a9/C4b","Ccr2/Itgam/Ptpn22/Lilrb4b/Lgals3/Laptm5/Myc/Slc11a1/Syk/Prnp/Rps3/Ager/Arg1/Rps6/Itgb2/Ccl19/Nckap1l/Icosl/Tnfsf9/Bmp4/Crtam/Ptpn6/Aif1/Cxcr4/Ptprc/Gja1/Slfn1/Ccl5/Cebpb/Lilrb4a/Cd44/Lrrc32/Fkbp1b/Sdc4/Pnp/H2-Aa/Cd24a/H2-DMb1/Il18/Il1b/Itgal","Eef2k/Myo6/Nptn/Gpm6a/Shank3/Arhgef9/Ache/Mdga1/Vstm5/Pdlim5/Kirrel3/Adgrl2/Il1rap/Rab29/Srgap2/Numb/Ptk2/Ntn1/Flrt3/Nrg1/Cacna1a/Farp1/Adnp/St8sia2/Cux2/Prkca/Dnm3/Sdk1/Ptprd/Flrt2/Ntng2/Lrrc4/Lrrtm3/Asic2/Cntn5/Grid2","Rpl23a/Nop53/Rpl10a/Rpl5/Rpl6/Rpl7/Rpl14/Las1l/Rpl26/Rpl24/Rpl11/Rpl10/Rpl7a/Rplp0/Npm1/Rpl35a/Eif6/Mrto4/Snu13/Rpl35/Zfp622/Rpl38/Nhp2/Znhit3/Malsu1/Gtf3a/Ftsj3","Ccl12/Serpine1/Ccl2/Cxcl13/Cxcl2/Trem2/Ptpn22/Arid5a/Cd14/Abca1/Plscr2/Cd68/Syk/Ly86/Cxcl1/Nlrp3/Fcgr4/Nr1d1/Lbp/Cfh/Zfp36/Mif/Pf4/Ccl5/Irf8/Cebpb/Lilrb4a/Nfkbia/Bcl2l11/Litaf/Tnf/Sbno2/Il18/Map2k7/Spi1/Irak1/Il1b","Zbtb16/Runx3/Ccl2/Trem2/Ccr2/Itgam/Thbs1/Ptpn22/Lilrb4b/Runx1/Il4ra/Syk/Shld3/Rps3/Ager/Pck1/Nlrp3/Itgb2/Ccl19/Nckap1l/Lbp/Cdkn1a/Ctsc/Tyrobp/Icosl/Tnfsf9/Prkcz/Rhoh/Adam8/Aif1/Mif/Ptprc/Ccl5/Lilrb4a/Gimap5/Crlf2/Cd74/Mmp14/Pnp/Fgr/H2-Aa/Cd24a/H2-DMb1/Jund/Il18/Il1b/Itgal/H2-Ab1/Nfkbiz/Nectin2/Il6st/Il33/Hsp90aa1/Axl/H2-Eb1/Mad2l2/Coro1a/Zap70/Cd47/Zbtb1/Tfrc","Ctsj/Serpine1/Ccl2/Ccn2/Pdgfb/Cygb/Ctss/Ager/Arg1/Mmp19/Prkcd/Bmp4/Fn1/Ctsl/F2r/Mmp17/Mmp14/Retreg1/Il18/Fgfr3/Ctsb/Rgcc/Il6ra/Retreg2/Adam15","Zbtb16/Runx3/Ccl2/Ccr2/Ptpn22/Lilrb4b/Runx1/Il4ra/Syk/Rps3/Ager/Pck1/Nlrp3/Ccl19/Nckap1l/Icosl/Tnfsf9/Prkcz/Rhoh/Adam8/Aif1/Ptprc/Ccl5/Lilrb4a/Gimap5/Cd74/Pnp/H2-Aa/Cd24a/H2-DMb1/Il18/Il1b/Itgal/H2-Ab1/Nfkbiz/Il6st/Hsp90aa1/H2-Eb1/Coro1a/Zap70/Cd47/Zbtb1/Tfrc","Runx3/Trem2/Pdgfb/Cygb/Ccr2/Cd300lf/Ptpn22/Lilrb4b/Lgals3/Laptm5/Runx1/Il4ra/Prnp/Tnfaip8l2/Arg1/Nr1d1/Nckap1l/Prkcd/Tyrobp/Btk/Bmp4/Crtam/Ptpn6/Hmgb3/Rassf5/Nrarp/Tnfrsf13b/Slfn1/Cebpb/Lilrb4a/Gimap5/Cd44/Lrrc32/C1qtnf1/Cd74/Sdc4/Fgr/H2-Aa/Cd24a/Samsn1/Spi1","Trem2/Jaml/Siglec1/Cd33/Hcst/Lair1/Col3a1/H2-T22/Fcgr4/Itgb2/Gm8909/Cd300c2/Crtam/Ifitm6/Itgb7/Icam1/Pilra/Trem1/Itgal/Ifitm2/Nectin2/Pianp/Pvr/Itga4","Zbtb16/Runx3/Ccl2/Ccr2/Ptpn22/Lilrb4b/Lgals3/Ctla2a/Laptm5/Runx1/Il4ra/Syk/Prnp/Tnfaip8l2/Rps3/Ager/Pck1/Arg1/Nlrp3/Ccl19/Nckap1l/Icosl/Tnfsf9/Prkcz/Bmp4/Rhoh/Crtam/Adam8/Ptpn6/Aif1/Nrarp/Ptprc/Slfn1/Ccl5/Cebpb/Lilrb4a/Gimap5/Cd44/Lrrc32/Cd74/Sdc4/Pnp/H2-Aa/Cd24a/H2-DMb1/Il18/Il1b/Itgal/H2-Ab1/Nfkbiz/Il6st/Hsp90aa1/Fas/H2-Eb1/Adrm1/Hfe/Bid/Coro1a/Zap70/Twsg1/Cd47/Zbtb1/Tfrc/Fgl1/Sh2b3","Arhgap45/Arhgef11/Plekhg5/Fmnl3/Bcr/Anln/Arhgef18/Arhgap32/Arhgef10/Rhpn2/Pkn3/Aaas/Trio/Arap2/Arhgap35/Tjp2/Cav1/Prex1/Arhgap23/Arhgap10/Arhgef10l/Arhgap18/Kalrn/Ophn1/Arhgef25/Stk10/Arhgap39/Srgap1/Racgap1/Dlc1/Stard13/Arhgef3/Vav3/Ect2/Farp1/Diaph3/Arhgap6/Rasgrf2/Depdc1b/Arhgap26","Ccl3/Lcn2/Itgam/C1qa/Ddit4/Fos/Clu/Trpm2/Nupr1/Prnp/Ager/Tyrobp/Egr1/Jun/Picalm/Bcl2l11/Tnf/Il18/Fgfr3/Map2k7/Il1b/Ctsz/Ncf2/Mcl1/Fas/Fis1/Tnfrsf1a/Ikbkb/Rest","Ccl3/Edn1/Cxcl13/Cxcl2/C3ar1/Ccr2/Ccr5/Adm/Ccr1/Nmb/Cd52/Cxcl1/Lrp1/S1pr1/Bmp4/Synpo/Ccrl2/Cxcr4/F2r/Ednrb/Ackr3/Gja1/Ackr2/Gimap5/C1qtnf1/Ptk2b/Fkbp1b/Cd24a","Csf2rb/Ccl12/Csf2rb2/Ccr2/Ptpn22/Csf1r/Lilrb4b/Lgals3/Laptm5/Syk/Prnp/Rps3/Ager/Arg1/Ccl19/Nckap1l/Cdkn1a/Tyrobp/Icosl/Tnfsf9/Btk/Bmp4/Crtam/Ptpn6/Aif1/Mif/Rassf5/Ptprc/Tnfrsf13b/Slfn1/Ccl5/Cebpb/Lilrb4a/Cd44/Lrrc32/Cd74/Csf2ra/Sdc4/Pnp/H2-Aa/Cd24a/H2-DMb1/Il18/Il1b/Itgal","Ccl12/Ccl3/Edn1/Ccl7/Cxcl13/Cxcl2/C3ar1/Ccl4/Ccr5/Ccl9/Apln/Ccl6/Nmb/Gpr143/Cxcl1/Ccl19/Cysltr1/S1pr1/Ccrl2/Cxcr4/F2r/Pf4/Ednrb/Ackr3/Ccl5/Ackr2","Ccl12/Ccl3/Ccl7/Cxcl13/Cxcl2/Ccl4/Ccr5/Cxcl1/Ccl19/Ccrl2/Cxcr4/Pf4/Ackr3/Ccl5/Ackr2","Ccl12/Ccl8/Ccl2/Cxcl13/Itgam/Fcgr1/Fcgr3/Lgals3/Cxcl14/Syk/Ncf1/Ager/Cxcl1/Arg1/H2-T22/Gm8909/Ccl19/Nckap1l/Lyz2/Ctsc/Tyrobp/Crtam/Hspa8/Ptpn6/Cfh/Ptprc/Gimap5/Rps19/Trem1/Bcl2l11/Pnp/Rpl30/Il18/Ctsh/Spi1","Runx3/Inhbb/Siglec1/Thbs1/Clu/Myc/Nupr1/Tnfrsf12a/Osm/Mal/Rps3/Prkcd/Ctsc/Serinc3/Itm2c/Rps7/Rack1/Ptprc/Rpl26/Rpl11/Dedd2/Atf3/Bcl2l11/S100a8/Tnf/Timp3/Ctsh/Skil/S100a9/Bclaf1/Mcl1/Fas/Plscr1/Fis1/Psen2/Htra2/Bid","Trem2/Ccr2/Itgam/Cd300lf/Thbs1/Lilrb4b/Fcgr3/Pirb/C1qa/Cd33/Clu/Lat2/Il4ra/Plscr2/Slc11a1/Syk/Ager/Fcgr4/Itgb2/Nr1d1/Prkcd/Lbp/Ctsc/Tyrobp/Tnfsf9/Rhoh/Ptpn6/Aif1/Fn1/Jun/Ndrg1/Cd48/Ccl5/Lilrb4a/Gimap5/Crlf2/Tnf/Fgr/Sbno2/Jund/Il18/Spi1","Edn1/Lcn2/Ccn2/Hba-a1/Ccn1/Pdgfb/Itgam/Thbs1/Ddit4/Gpx3/Sh3pxd2b/Ncf4/Ier3/Syk/Insr/Ncf1/Ager/Cxcl1/Cybb/Itgb2/Prkcd/Cdkn1a/Cyp1b1/Acp5/Tyrobp/Bnip3/Pink1/Hbp1/Eef1a1/Xdh/Birc2/Ptk2b/Tnf/Nnt/Gadd45a/Gpx1/Il18/Prdx6/Sod2/Eif6","A2m/Serpine1/Itgb3/Timp1/F13a1/Pdgfb/Itga2b/Thbs1/Selp/Itih3/Qsox1/Clu/Cd63/Serpina3i/Fermt3/Actn1/Fn1/Pf4/Tuba4a/Egf/Timp3","Chil3/Serpine1/Il17ra/Trem2/Lilrb4b/Per1/Alox5ap/Il4ra/Syk/Adcy7/Pld4/Ncf1/Lbp/Il1r2/Lilrb4a/Rps19/Gpsm3/Tnf","Ccl12/Ccl3/Edn1/Ccl7/Cxcl13/Cxcl2/C3ar1/Ccl4/Ccr5/Ccl9/Apln/Ccl6/Nmb/Cxcl1/Ccl19/Ccrl2/Cxcr4/F2r/Pf4/Ednrb/Ackr3/Ccl5/Ackr2","Saa3/Ccl12/Ccl8/Serpine1/Ccl2/Ccl3/Ccl7/Ccl4/Ccl9/Ccl6/Rps6ka4/Ccl19/Nr1d1/Egr1/Fn1/Il1r2/Ccl5/Cebpb/Map2k7/Irak1/Il1b","A2m/Capn12/Serpine1/Spp1/Itgb3/Timp1/Itgam/Itga2b/Thbs1/Itga3/Ctss/Adamts4/Emilin2/Col3a1/Itgb2/Vcan/Mmp19/Bmp4/Adam8/Fn1/Ctsl/Itgb7/Mmp17/Adamts5/Icam1/Fbln2/Col4a2/Cd44/Mmp14/Sdc4/Dcn/Col13a1/Pcolce/Ctsb/Phykpl/Nid1/Itgal/Jam2/Col4a1/Capn5/Adam15/Itga4","Saa3/Ccl12/Ccl8/Serpine1/Ccl2/Ccl3/Ccl7/Ccl4/Ccl9/Sele/Ccl6/Rps6ka4/Ccl19/Nr1d1/Egr1/Fn1/Il1r2/Ccl5/Cebpb/Map2k7/Irak1/Il1b","Ccl12/Ccl8/Ccl2/Ccl3/Ccn2/Ccl7/Itgb3/Trem2/Pdgfb/Ccl4/Ccl9/Ptpn22/Csf1r/Ccr1/Ccl6/Ager/Ccl19/Lrp1/Prkcz/Bmp4/Cxcr4/Jun/Mif/F2r/Ackr3/Ptprc/Ccl5/Icam1/Cd44/Map3k12/Ptk2b/Cd74/Egf/Fgfr3/Map2k7/Il1b","Ccl12/Ccl8/Ccl2/Ccl3/Edn1/Ccn2/Ccl7/Itgb3/Ccn1/Trem2/Pdgfb/Ccl4/Ccl9/Ptpn22/Csf1r/Ccr1/Ccl6/Myc/Syk/Ager/Ccl19/Spred3/Lrp1/Prkcz/Bmp4/Ptpn6/Cxcr4/Fn1/Jun/Mif/F2r/Ackr3/Ptprc/Ccl5/Icam1/Atf3/Cd44/Map3k12/Ptk2b/Cd74/Egf/Timp3/Fgfr3/Ctsh/Map2k7/Il1b","Ccl12/Serpine1/Ccl2/Cxcl13/Cxcl2/Kcnj8/Trem2/Ace/Ptpn22/Adm/Arid5a/Cd14/Abca1/Plscr2/Cd68/Slc11a1/Ly86/Cxcl1/Nlrp3/Fcgr4/Nr1d1/Lbp/Acp5/Mgst1/Cfh/Zfp36/Mif/F2r/Pf4/Ednrb/Irf5/Ccl5/Irf8/Cebpb/Lilrb4a/Nfkbia/S100a8/Litaf/Tnf/Sbno2/Cd24a/Il18/Map2k7/Spi1/Irak1/Il1b","Ccl12/Ccl8/Ccl2/Ccl3/Edn1/Ccl7/Cxcl13/Cxcl2/Trem2/Ccr2/Ccl4/Ccl9/Ccr1/Ccl6/Cxcl1/Ccl19/Cxcr4/Pf4/Ackr3/Ccl5/Ptk2b/Foxc1","Ccl12/Ccl8/Ccl2/Ccl3/Edn1/Ccl7/Cxcl13/Cxcl2/Trem2/Ccr2/Ccl4/Ccl9/Ccr1/Ccl6/Cxcl1/Ccl19/Cxcr4/Pf4/Ackr3/Ccl5/Ptk2b/Foxc1","Csf2rb/Csf2rb2/Trem2/Ccr2/Fcgr1/Lilrb4b/Fcgr3/Pirb/C1qa/C1qc/C1qb/Arid5a/Il4ra/Slc11a1/Shld3/Ager/Arg1/Nlrp3/H2-T22/Gm8909/Nckap1l/Prkcd/Ctsc/Icosl/Prkcz/Btk/Crtam/Myo1g/Hspa8/Ptpn6/Cfh/Ptprc/Icam1/Lilrb4a/Gimap5/Crlf2/Cd74/Tnf/Ercc1/Pnp/Pou2f2/H2-Aa/Cd24a/H2-DMb1/Il18/Ctsh/Il1b/H2-Ab1/Nectin2/C4b/Pvr/Fas/Gba/Stx11/H2-Eb1/Mad2l2/Hfe/Coro1a/Zbtb1/Tfrc/H2-Q10/Gfus/Cfi","Zbtb16/Runx3/Ccl3/Itgb3/Trem2/Ccr2/Itgam/Csf1r/Lilrb4b/Ccr1/Fos/Runx1/Il4ra/Syk/Ager/Pck1/Nlrp3/Ccl19/Nckap1l/Tyrobp/Tnfsf9/Prkcz/Rhoh/Adam8/Jun/Ptprc/Ccl5/Lilrb4a/Ninj1/Gimap5/Car2/Cd74/Tnf/Mmp14/Pnp/Ikzf1/H2-Aa/Il18/Ctnnbip1","Zbtb16/Runx3/Ccl3/Itgb3/Trem2/Ccr2/Itgam/Csf1r/Lilrb4b/Ccr1/Fos/Runx1/Il4ra/Syk/Ager/Pck1/Nlrp3/Ccl19/Nckap1l/Tyrobp/Tnfsf9/Prkcz/Rhoh/Adam8/Jun/Ptprc/Ccl5/Lilrb4a/Ninj1/Gimap5/Car2/Cd74/Tnf/Mmp14/Pnp/Ikzf1/H2-Aa/Il18/Ctnnbip1","Csf2rb/Ccl12/Csf2rb2/Trem2/Ccr2/Itgam/Ptpn22/Csf1r/Lilrb4b/Lgals3/Laptm5/Clu/Myc/Slc11a1/Syk/Prnp/Junb/Rps3/Ager/Arg1/Rps6/Itgb2/Ccl19/Nckap1l/Prkcd/Cdkn1a/Tyrobp/Icosl/Tnfsf9/Btk/Bmp4/Crtam/Ptpn6/Aif1/Cxcr4/Mif/Rassf5/Ptprc/Gja1/Tnfrsf13b/Slfn1/Ccl5/Cebpb/Lilrb4a/Cd44/Lrrc32/Cd74/Csf2ra/Fkbp1b/Sdc4/Pnp/H2-Aa/Cd24a/H2-DMb1/Il18/Il1b/Itgal/Impdh2/Il6st/Il33","Ccl2/Ccl3/Ccn1/Trem2/Ccr2/Clec4a2/Ccl4/Ccr5/Thbs1/Ptpn22/Lilrb4b/Fcgr3/Cd33/Arid5a/Cd14/Lpl/Clu/Syk/Ager/Clec4a1/Ccl19/Lbp/Acp5/Tyrobp/Ptpn6/Zfp36/Mif/Ptprc/Lilrb4a/Cd24a/Il18/Clec4a3/Il33/Tspo/Tnfaip3/Axl/Psen2/Tnfrsf1a/Selenos","Csf2rb/Csf2rb2/Il17ra/C3ar1/Trem2/Ccr2/Fcgr1/Lilrb4b/Fcgr3/Pirb/C1qa/C1qc/C1qb/Arid5a/Il4ra/Slc11a1/Clec4n/Shld3/Ager/Arg1/Nlrp3/H2-T22/Gm8909/Ccl19/Nckap1l/Prkcd/Ctsc/Icosl/Prkcz/Btk/Gadd45g/Myo1g/Hspa8/Ptpn6/Cfh/Ptprc/Icam1/Lilrb4a/Gimap5/Crlf2/Cd74/Tnf/Ercc1/Pnp/Pou2f2/H2-Aa/Cd24a/H2-DMb1/Il18/Ctsh/Il1b/H2-Ab1/Nfkbiz/Nectin2/Il6ra/C4b/Il18bp/Il33/Pvr/Fas/Gba/Tnfaip3/Stx11/H2-Eb1/Mad2l2","Zbtb16/Runx3/Ccl2/Ccr2/Ptpn22/Selp/Lilrb4b/Sele/Lgals3/Laptm5/Runx1/Il4ra/Syk/Prnp/Tnfaip8l2/Ass1/Rps3/Ager/Pck1/Arg1/Nlrp3/Itgb2/Ccl19/Nckap1l/Icosl/Tnfsf9/Prkcz/Bmp4/Rhoh/Crtam/Adam8/Ptpn6/Aif1/Nrarp/Ptprc/Slfn1/Ccl5/Icam1/Cebpb/Lilrb4a/Gimap5/Cd44/Lrrc32/Cd74/Tnf/Sdc4/Pnp/H2-Aa/Cd24a/H2-DMb1/Il18/Irak1/Chst2/Il1b/Itgal/H2-Ab1/Nfkbiz/Il6st/Hsp90aa1/Itga4/H2-Eb1/Hfe/Coro1a/Zap70/Twsg1/Cd47/Zbtb1/Tfrc/Fgl1/Klf4/Nkap","Ccl2/Ccl3/Ccn1/Trem2/Ccr2/Clec4a2/Ccl4/Ccr5/Thbs1/Ptpn22/Lilrb4b/Fcgr3/Cd33/Arid5a/Cd14/Lpl/Clu/Syk/Ager/Clec4a1/Ccl19/Lbp/Acp5/Tyrobp/Ptpn6/Zfp36/Mif/Ptprc/Lilrb4a/Cd24a/Il18/Clec4a3/Il33/Tspo/Tnfaip3/Axl/Tnfrsf1a/Selenos","Ccl12/Serpine1/Ccl2/Edn1/Ccl7/Cxcl13/Itgb3/C3ar1/Trem2/Ccr2/Thbs1/Csf1r/Selp/Sele/Ccr1/Lgals3/Cxcl14/Ager/Ccl19/Lyve1/Nckap1l/Lbp/Adam8/Aif1/Ccl5/Icam1/Zfp580/Trem1/Ptk2b/Gpsm3/Cd74/Mmp14/Spi1/Il1b/Plvap/Jam2","Serpine1/Ccl3/Il17ra/Trem2/Ccr2/Ace/Ccr5/Fcgr1/Fcgr3/Gprc5b/Ctss/Alox5ap/Lpl/Nupr1/Osm/Ager/Nlrp3/Lbp/Ctsc/Adam8/Camk2n1/Fabp4/Ccl5/Cebpb/Ninj1/Rps19/Nfkbia/S100a8/Gpsm3/Tnf/Cd24a/Il18/Il1b/Nfkbiz/S100a9/Il33","Rpl29/Rpl12/Rps3/Rps6/Rps15/Rpl23a/Rps25/Rpl10a/Rpl5/Rps9/Rpl17/Rps16/Pabpc1/Rpl6/Rpsa/Rpl18/Rps27a/Rps14/Rps5/Rps3a1/Rps7/Rpl3/Rps4x/Rps12/Rpl7/Rpl32/Rps17/Eef2/Rpl14/Rpl8/Rpl23/Rps8/Eif5/Rpl31/Rpl13/Rpl26/Rpl4/Rpl18a/Rps13/Rpl24/Rpl9/Rpl11/Rpl27a/Rpl28/Rps11/Fau/Rps19/Rpl41/Rpl15/Rpl19/Rps20/Rplp1/Rps28/Rpl22l1/Rps23/Gm3550/Rps24/Rpl30/Rpl36/Rpl7a/Rplp0/Rpl21/Pkm/Mif4gd/Rpl35a/Rpl36a/Rps15a/Eif3i/Rplp2/Rpl34/Eif4ebp1/Eif3m/Rpl39/Rpl22/Rpl37a/Uba52/Csde1/Rpl38/Rps26/Rpl27/Rpl9-ps6/Dph3","Ccl12/Ccl8/Serpine1/Ccl2/Ccl3/Edn1/Ccl7/Spp1/Cxcl13/Il17ra/Cxcl2/C3ar1/Jaml/Pdgfb/Ccr2/Itgam/Ccl4/Ccl9/Thbs1/Csf1r/Fcgr3/Ccr1/Ch25h/Lgals3/Ccl6/Cxcl14/Trpm2/Syk/Cxcl1/Itgb2/Ccl19/Nckap1l/Lbp/S1pr1/Adam8/Aif1/Mif/Pf4/Ednrb/Ccl5/Ninj1/Rps19/Zfp580/Trem1/S100a8/Ptk2b/Gpsm3/Cd74","Ccl12/Ccl8/Ccl2/Ccl3/Edn1/Ccl7/Spp1/Cxcl13/Il17ra/Cxcl2/C3ar1/Jaml/Itgam/Ccl4/Ccl9/Thbs1/Csf1r/Fcgr3/Lgals3/Ccl6/Syk/Cxcl1/Itgb2/Ccl19/Nckap1l/Lbp/Rhoh/Adam8/Pf4/Ccl5/Trem1/S100a8/Cd74","Ccl12/Ccl8/Serpine1/Ccl2/Ccl3/Ccl7/Cxcl13/Itgb3/C3ar1/Jaml/Pdgfb/Ccr2/Ccl4/Ccl9/Thbs1/Eps8/Csf1r/Ccr1/Ch25h/Lgals3/Ccl6/Cxcl14/Trpm2/Ager/Ccl19/S1pr1/Crtam/Adam8/Myo1g/Aif1/Itgb7/Ccl5/Icam1/Ninj1/Rps19/Trem1/Ptk2b","Saa3/Ccl12/Ccl8/Serpine1/Ccl2/Ccl3/Edn1/Ccl7/Spp1/Cxcl13/Il17ra/Cxcl2/C3ar1/Jaml/Pdgfb/Ccr2/Itgam/Ccl4/Ccr5/Ccl9/Thbs1/Csf1r/Fcgr3/Ccr1/Ch25h/Lgals3/Ccl6/Cxcl14/Trpm2/Syk/Cxcl1/Itgb2/Ccl19/Nckap1l/Prkcd/Lbp/S1pr1/Adam8/Ccrl2/Aif1/Cxcr4/Mif/Pf4/Ednrb/Ackr3/Ccl5/Ackr2/Ninj1/Rps19/Zfp580/Trem1/S100a8/Ptk2b/Gpsm3/Cd74","Ccl12/Ccl8/Serpine1/Ccl2/Ccl3/Edn1/Ccl7/Spp1/Cxcl13/Itgb3/Il17ra/Cxcl2/C3ar1/Trem2/Jaml/Pdgfb/Ccr2/Itgam/Ccl4/Ccl9/Thbs1/Eps8/Ptpn22/Itga3/Csf1r/Selp/Fcgr3/Sele/Ccr1/Ch25h/Lgals3/Ccl6/Cxcl14/Trpm2/Syk/Ager/Cxcl1/Itgb2/Ccl19/Lyve1/Nckap1l/Lbp/S1pr1/Rhoh/Crtam/Adam8/Myo1g/Aif1/Itgb7/Mif/Pf4/Ednrb/Ccl5/Icam1/Ninj1/Rps19/Zfp580/Trem1/S100a8/Ptk2b/Gpsm3/Cd74/Tnf/Mmp14/Cd24a/Selplg/Spi1/Chst2/Il1b/Plvap/Itgal/S100a9/Jam2/Aimp1/Il33/Retnlg/Dusp1/Gba/Itga4","Ccl12/Ccl8/Serpine1/Ccl2/Ccl3/Edn1/Ccl7/Spp1/Cxcl13/Il17ra/Cxcl2/C3ar1/Trem2/Jaml/Pdgfb/Ccr2/Itgam/Ccl4/Ccl9/Thbs1/Csf1r/Fcgr3/Ccr1/Lgals3/Ccl6/Syk/Ager/Cxcl1/Itgb2/Ccl19/Nckap1l/Lbp/Rhoh/Adam8/Aif1/Mif/Pf4/Ednrb/Ccl5/Ninj1/Rps19/Trem1/S100a8/Ptk2b/Cd74/Mmp14","Rpl29/Rpl12/Rps3/Rps6/Rps15/Rpl23a/Rps25/Rpl5/Rps9/Rps16/Pabpc1/Rpl6/Rpsa/Rpl18/Rps14/Rps3a1/Rps7/Rpl3/Rps4x/Rpl7/Rpl32/Rps17/Rpl14/Rpl8/Rpl23/Rps8/Rpl31/Rpl13/Rpl26/Rpl4/Rpl18a/Rps13/Rpl24/Rpl9/Rpl11/Rpl10/Rpl27a/Rpl28/Rps11/Rps19/Rpl15/Rpl19/Rps20/Rplp1/Rps28/Rpl22l1/Rps23/Rps24/Rpl30/Rplp0/Rpl35a/Rpl36a/Rps15a/Rplp2/Rpl34/Rpl37/Rpl39/Rpl22/Rpl37a/Uba52/Rpl35/Rpl38/Rps26/Rpl27","Rpl29/Rpl12/Rps3/Rps6/Rps15/Rpl23a/Rps25/Rpl5/Rps9/Rps16/Rpl6/Rpsa/Rpl18/Rps14/Rps3a1/Rps7/Rpl3/Rps4x/Rpl7/Rpl32/Rps17/Rpl14/Rpl8/Rpl23/Rps8/Rpl31/Rpl13/Eif3f/Rpl26/Rpl4/Rpl18a/Rps13/Rpl24/Rpl9/Rpl11/Rpl10/Rpl27a/Rpl28/Rps11/Rps19/Rpl15/Rpl19/Rps20/Rplp1/Rps28/Rpl22l1/Rps23/Rps24/Rpl30/Rplp0/Eif3k/Rpl35a/Rpl36a/Rps15a/Eif3i/Rplp2/Rpl34/Rpl37/Eif3e/Eif3m/Rpl39/Rpl22/Rpl37a/Uba52/Rpl35/Eif3h/Rpl38/Rps26/Rpl27/Eif3c/Eif3d/Rps2/Rps21/Rpl10-ps3/Eif3l/Eif3a","Rpl29/Rpl12/Rps3/Rps6/Rps15/Rpl23a/Rps25/Rpl5/Rps9/Rps16/Rpl6/Rpsa/Rpl18/Rps14/Rps3a1/Rps7/Rpl3/Rps4x/Rpl7/Rpl32/Rps17/Rpl14/Rpl8/Rpl23/Rps8/Rpl31/Rpl13/Rpl26/Rpl4/Rpl18a/Rps13/Rpl24/Rpl9/Rpl11/Rpl10/Rpl27a/Rpl28/Rps11/Rps19/Rpl15/Rpl19/Rps20/Rplp1/Rps28/Rpl22l1/Rps23/Rps24/Rpl30/Rplp0/Rpl35a/Rpl36a/Rps15a/Rplp2/Srp54a/Rpl34/Rpl37/Rpl39/Rpl22/Rpl37a/Uba52/Rpl35/Rpl38/Rps26/Rpl27","Rpl29/Rpl12/Rps3/Rps6/Rps15/Rpl23a/Rps25/Rpl5/Rps9/Rps16/Pabpc1/Rpl6/Rpsa/Rpl18/Rps14/Rps3a1/Rps7/Rpl3/Rps4x/Rpl7/Rpl32/Rps17/Rpl14/Rpl8/Rpl23/Rps8/Eif5/Rpl31/Rpl13/Eif3f/Rpl26/Rpl4/Rpl18a/Rps13/Rpl24/Rpl9/Rpl11/Rpl10/Rpl27a/Rpl28/Rps11/Rps19/Rpl15/Rpl19/Rps20/Rplp1/Rps28/Rpl22l1/Rps23/Rps24/Rpl30/Rplp0/Eif3k/Rpl35a/Rpl36a/Rps15a/Eif3i/Rplp2/Rpl34/Rpl37/Eif3e/Eif4ebp1/Eif3m/Rpl39/Rpl22/Rpl37a/Uba52/Rpl35/Eif3h/Eif2b2/Rpl38/Rps26/Rpl27/Eif3c/Eif3d/Rps2/Rps21/Rpl10-ps3/Eif3l/Eif3a/Eif4a1/Rps27rt/Eif5b/Eif4b/Eif3b/Eif4h/Eif3g","Rpl29/Rpl12/Rps3/Rps6/Rps15/Rpl23a/Rps25/Rpl5/Rps9/Rps16/Pabpc1/Rpl6/Rpsa/Rpl18/Rps14/Rps3a1/Rps7/Rpl3/Rps4x/Rpl7/Rpl32/Rps17/Rpl14/Rpl8/Rpl23/Rps8/Rpl31/Rpl13/Rpl26/Rpl4/Rpl18a/Rps13/Rpl24/Rpl9/Rpl11/Rpl10/Rpl27a/Rpl28/Rps11/Rps19/Rpl15/Rpl19/Rps20/Rplp1/Rps28/Rpl22l1/Rps23/Rps24/Rpl30/Rplp0/Rpl35a/Rpl36a/Rps15a/Rplp2/Rpl34/Rpl37/Rpl39/Rpl22/Rpl37a/Uba52/Rpl35/Upf3b/Rpl38/Rps26/Rpl27","Rpl29/Rpl12/Rps3/Rps6/Rps15/Rpl23a/Rps25/Rpl5/Rps9/Rps16/Rpl6/Rpsa/Rpl18/Rps14/Rps3a1/Rps7/Rpl3/Rps4x/Rpl7/Rpl32/Rps17/Rpl14/Rpl8/Rpl23/Rps8/Rpl31/Rpl13/Las1l/Rpl26/Rpl4/Rpl18a/Rps13/Rpl24/Rpl9/Rpl11/Rpl10/Rpl27a/Rpl28/Rps11/Rps19/Imp3/Rpl15/Rpl19/Rps20/Rplp1/Rps28/Rpl22l1/Rps23/Rps24/Rpl30/Rplp0/Rpl35a/Gnl3/Rpl36a/Rps15a/Rplp2/Rpl34/Rpl37/Rpl39/Rpl22/Rpl37a/Snu13/Uba52/Rpl35/Rpl38/Rps26/Rpl27/Nop56/Tsr1/Ncl/Ftsj3/Rpp40/Fbl/Rps2/Rps21/Rpl10-ps3/Ddx49/Nip7","Rps3/Rps6/Rps15/Rps25/Rps9/Rps16/Pabpc1/Rpsa/Rps14/Rps3a1/Rps7/Rps4x/Rps17/Rps8/Eif3f/Rps13/Rps11/Rps19/Rps20/Rps28/Rps23/Rps24/Eif3k/Rps15a/Eif3i/Eif3e/Eif4ebp1/Eif3m/Uba52/Eif3h/Rps26/Eif3c/Eif3d/Rps2/Rps21/Eif3l/Eif3a/Eif4a1/Rps27rt/Eif4b/Eif3b/Eif4h/Eif3g/Rps18/Rps27/Eif2s2/Rps29/Gm11808","Rpl29/Rpl12/Rps3/Rps6/Rps15/Rpl23a/Rps25/Rpl5/Rps9/Rps16/Pabpc1/Rpl6/Eef1b2/Rpsa/Rpl18/Rps14/Rps3a1/Rps7/Rpl3/Rps4x/Rpl7/Rpl32/Rps17/Eef2/Rpl14/Rpl8/Rpl23/Rps8/Eif5/Rpl31/Rpl13/Eif3f/Rpl26/Rpl4/Rpl18a/Eef1a1/Rps13/Rpl24/Rpl9/Rpl11/Rpl10/Rpl27a/Rpl28/Rps11/Rps19/Rpl15/Rpl19/Rps20/Rplp1/Rps28/Rpl22l1/Rps23/Rps24/Rpl30/Rplp0/Eef1g/Eif3k/Rpl35a/Rpl36a/Rps15a/Eif3i/Rplp2/Srp54a/Rpl34/Rpl37/Mrpl30/Eif3e/Eif4ebp1/Eif3m/Eef1d/Rpl39/Rpl22/Rpl37a/Mrps34/Uba52/Rpl35/Eif3h/Eif2b2/Rpl38/Rps26/Rpl27/Eif3c/Srp9/Eif3d/Mrrf/Mrpl2/Rps2/Mrps26/Rps21/Mrpl12/Rpl10-ps3/Mrpl52/Eif3l/Eif3a/Eif4a1/Mrpl16/Oxa1l/Rps27rt/Mrpl51/Eif5b/Mrps11/Srp72/Mrps18c/Eif4b/Mrps24/Mrps30/Mrpl9/Eif3b/Mrpl11/Eif4h/Mrpl41/Eif3g/Mrps18b/Mrpl23/Aurkaip1/Gadd45gip1/Mrps7/Rps18/Rps27/Mrps33/Eif2s2/Mrpl20/Mrpl43/Mrpl13/Mrpl50/Mrpl57/Mrpl32/Rps29/Gm11808"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>site<\/th>\n      <th>Description<\/th>\n      <th>setSize<\/th>\n      <th>enrichmentScore<\/th>\n      <th>NES<\/th>\n      <th>pvalue<\/th>\n      <th>p.adjust<\/th>\n      <th>Count<\/th>\n      <th>.sign<\/th>\n      <th>GeneRatio<\/th>\n      <th>core_enrichment<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":9,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[2,3,4,5,6,7,9]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render"],"jsHooks":[]}</script>
</div>
<div id="number-of-gene-sets-1" class="section level4" number="5.0.2">
<h4><span class="header-section-number">5.0.2</span> Number of gene
sets</h4>
<pre class="r"><code>gse_df %&gt;% 
  filter(p.adjust &lt; p_thrs) %&gt;% 
  group_by(group) %&gt;% 
  summarise(n = n()) %&gt;% 
  ggplot(aes(x=group, y=n)) +
  geom_bar(stat=&quot;identity&quot;, color = &#39;black&#39;)+
  geom_text(aes(label=n), vjust=1.4, color=&quot;white&quot;, size=geom_text_size)+
  labs(
    x = &#39;Site&#39;,
    y = &#39;Number of gene sets&#39;,
    title = &#39;Number of significant gene sets by site&#39;,
    caption = paste(&quot;Adjusted P value &lt;&quot;, p_thrs)
    
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/endothelial-marker-gsea-c2reactome_gobp-ngset-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-set-similarity-matrix-1" class="section level4"
number="5.0.3">
<h4><span class="header-section-number">5.0.3</span> Gene-set similarity
matrix</h4>
<p>Heatmap showing the pair-wise similarity matrix of top 40 gene sets
with an adjusted P value &lt; 0.01 in each gene group. Heatmap colors
represent the Jaccard similarity coefficient.</p>
<pre class="r"><code>col_fun &lt;- colorRamp2(seq(0, 1, length.out = 4), brewer.pal(4, &quot;GnBu&quot;))
n_split &lt;- 9

use_mat &lt;- jc_matrix
colnames(use_mat) &lt;- colnames(use_mat) %&gt;% clean_msigdb_names %&gt;% tolower() %&gt;% str_to_title %&gt;% ep_str_wrap(70)
rownames(use_mat) &lt;- rownames(use_mat) %&gt;% clean_msigdb_names %&gt;% tolower() %&gt;% str_to_title %&gt;% ep_str_wrap(70)

ha_top &lt;- HeatmapAnnotation(
  foo = anno_block(
    # labels = c(&quot;Translation&quot;, &quot;Cell division&quot;),
    labels = 1:n_split,
    labels_gp = gpar(col = &quot;black&quot;, fontsize = 8),
    gp = gpar(lwd = 0, lty = 0))
  )

ht &lt;- Heatmap(
  use_mat,
  name = &#39;Jaccard index&#39;,
  column_split = n_split,
  row_split = n_split,
  column_title = NULL,
  row_title = NULL,
  col = col_fun,
  show_column_dend = FALSE,
  show_column_names = FALSE,
  border = TRUE,
  top_annotation = ha_top,
  heatmap_legend_param = list(title_gp = gpar(fontsize = 8),labels_gp = gpar(fontsize = 8)),
  width = unit(7, &quot;cm&quot;))

ht = draw(ht)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/endothelial-marker-gsea-c2reactome_gobp-jcmatrix-1.png" width="4320" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;\n\n&quot;)</code></pre>
<pre class="r"><code># Extract term order according to hclust from similarity heatmap
row_order &lt;- row_order(ht) %&gt;% set_names(1:n_split)
term_similarity_clusters &lt;- foreach(i=names(row_order), .combine = rbind) %do% {
  data.frame(
    jc_sim_cl = i, 
    row_order = row_order[[i]], 
    Description =  rownames(jc_matrix)[row_order[[i]]]
  )
}

# Add similarity order to data for dot plot
res_top %&lt;&gt;% left_join(term_similarity_clusters)</code></pre>
</div>
<div id="dotplots-1" class="section level4" number="5.0.4">
<h4><span class="header-section-number">5.0.4</span> Dotplots</h4>
<p>The plots below show the top 40 gene sets with an adjusted P value
&lt; 0.01 in each gene group</p>
<pre class="r"><code>use_linewidth &lt;- 1/2.141959
x_max &lt;- res_top$NES %&gt;% abs %&gt;% max()

res_top %&gt;%
  # Combine similarity clusters
  mutate(
    # jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;1&#39;), &#39;Translation&#39;, jc_sim_cl),
    # jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;3&#39;,&#39;5&#39;,&#39;6&#39;), &#39;Energy\ngeneration&#39;, jc_sim_cl),
    # jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;4&#39;, &#39;8&#39;), &#39;Cell stress&#39;, jc_sim_cl),
    # jc_sim_cl = factor(
    #   jc_sim_cl,
    #   levels = c(&#39;Translation&#39;, &#39;Energy\ngeneration&#39;, &#39;Cell stress&#39;, &#39;2&#39;, &#39;7&#39;, &#39;9&#39;)
    #   ),

    p.adjust = ifelse(p.adjust &gt; 0.05, 1, p.adjust),
    p.adjust = -log10( p.adjust),
    p.adjust = ifelse(p.adjust &gt;= 10, 10, p.adjust),
    group = recode(group, primary_tumor = &quot;PT&quot;, lung = &quot;Lung&quot;, liver = &quot;Liver&quot;),
    group = factor(group, levels = c(&#39;PT&#39;, &#39;Lung&#39;, &#39;Liver&#39;))
  ) %&gt;% 
  filter(group == &#39;Liver&#39;) %&gt;% 
  ggplot(aes(NES, Description, color = NES, size = p.adjust)) +
  geom_point() +
  facet_grid(rows = vars(jc_sim_cl), cols = vars(group), scale = &#39;free_y&#39;, space = &#39;free_y&#39;) +
  scale_x_continuous(limits = c(-x_max, x_max),
                        expand = expansion(mult = c(0.08, 0.08))) +
  scale_y_discrete(labels = default_labeller(50)) +
  geom_vline(xintercept = 0, lty = 3) +
  scale_size_continuous(range=c(0.3, 2.5)) +
  scale_color_gradient2(trans = &quot;reverse&quot;) +
  labs(
    size = expression(paste(&quot;lo&quot;, g[10],&quot;(Adjusted P value)&quot;)),
    color = &quot;NES&quot;
  ) +
  theme(
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, size = use_linewidth/2),
    
    axis.line = element_blank(),
    axis.ticks = element_line(color = &quot;black&quot;, size = use_linewidth/2),
    axis.text.y = element_text(size = 4),
    axis.text.x = element_text(size = 4),
    
    legend.text = element_text(size = 4), 
    legend.title = element_text(size = 4),
    legend.key.width = unit(0.2, &quot;cm&quot;),
    
    strip.background = element_blank()
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/endothelial-marker-gsea-c2reactome_gobp-dotplot-1.png" width="2520" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="knitrtemplate" class="section level2" number="6">
<h2><span class="header-section-number">6</span> Knitr:template</h2>
<pre class="r"><code># knit template: https://workflowr.github.io/workflowr/articles/wflow-07-common-code.html
src &lt;- knit_expand(
  file = here::here(&quot;analysis/templates/rnaseq-deg-edger.Rmd&quot;)
  )</code></pre>
<!-- To be compatible with workflowr, we need to specify the absolute directory under which the plots are generated, otherwise they will be saved into figure/<basename>.Rmd/<chunk-name>.<extension. and not visible in the html document: https://github.com/workflowr/workflowr/issues/111 -->
</div>
<div id="download-results" class="section level2" number="7">
<h2><span class="header-section-number">7</span> Download results</h2>
<!-- Write DGE results to xlsx file -->
<pre class="r"><code># File name summary
rmd_file &lt;- current_input()
if(is.null(rmd_file))
  rmd_file &lt;- &#39;tmp&#39;
file_xlsx &lt;- file.path(&#39;./docs/file&#39;,rmd_file, &#39;dge.xlsx&#39;)

dir.create(dirname(file_xlsx), recursive = TRUE, showWarnings = FALSE)

# Generate workbook
wb &lt;- createWorkbook()
for(i in names(dge_list)) {
  addWorksheet(wb, i)
  res &lt;- dge_list[[i]]$results %&gt;% 
    dplyr::select(feature, gene_name, logFC, PValue, FDR)
  writeData(wb, i, res)
}
saveWorkbook(wb, file_xlsx, TRUE)

dge_file_xlsx &lt;- file_xlsx</code></pre>
<!-- Write GSE results to xlsx file -->
<pre class="r"><code>gse_collections &lt;- list(
  GSEA = names(gmt_files_symbols),
  gseGO = c(&quot;BP&quot;, &quot;MF&quot;)
)

# File name summary
rmd_file &lt;- current_input()
if(is.null(rmd_file))
  rmd_file &lt;- &#39;tmp&#39;

file_xlsx_list &lt;- list()
for(sr in names(gse_collections)) {
  file_xlsx_list[[sr]] &lt;- list()
  for(gc in gse_collections[[sr]]) {
    # Generate file
    file_xlsx &lt;- file.path(&#39;./docs/file&#39;, 
                           rmd_file, 
                           paste0(&#39;gse_&#39;, tolower(sr), &#39;_&#39;, tolower(gc), &#39;.xlsx&#39;)
                           )
    file_xlsx_list[[sr]][[gc]] &lt;- file_xlsx
    dir.create(dirname(file_xlsx), recursive = TRUE, showWarnings = FALSE)
    
    # Generate workbook
    wb &lt;- createWorkbook()
    for(i in names(gse_list)) {
      addWorksheet(wb, i)
      if(!is.null(gse_list[[i]][[sr]][[gc]])) {
        res &lt;- gse_list[[i]][[sr]][[gc]]@result
        writeData(wb, i, res)
      }
    }
    saveWorkbook(wb, file_xlsx, TRUE)
  }
}</code></pre>
<!-- Change dge_list names after writing to xlsx -->
<pre class="r"><code>comp_names &lt;- comp_list_description$comparison_long %&gt;% 
  set_names(comp_list_description$comparison)
names(comp_list) &lt;- comp_names[names(comp_list)]
names(dge_list) &lt;- comp_names[names(dge_list)]
names(gse_list) &lt;- comp_names[names(gse_list)]</code></pre>
<p>The tables of results can be downloaded using the following
links:</p>
<ul>
<li><p><a
href="./file/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge.xlsx"><strong>Differential
gene expression</strong></a>
<!-- Write links to download GSE results --></p></li>
<li><p><a
href="./file/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse_gsea_msigdb.c2.cp.xlsx"><strong>Gene
set enrichment analysis: msigdb.c2.cp</strong></a></p></li>
<li><p><a
href="./file/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse_gsea_msigdb.m2.cp.reactome.xlsx"><strong>Gene
set enrichment analysis: msigdb.m2.cp.reactome</strong></a></p></li>
<li><p><a
href="./file/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse_gsego_bp.xlsx"><strong>Gene
set enrichment analysis: GO:BP</strong></a></p></li>
<li><p><a
href="./file/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse_gsego_mf.xlsx"><strong>Gene
set enrichment analysis: GO:MF</strong></a></p></li>
</ul>
</div>
<div id="differential-gene-expression-analysis" class="section level2"
number="8">
<h2><span class="header-section-number">8</span> Differential gene
expression analysis</h2>
<div id="download-results-1" class="section level3" number="8.1">
<h3><span class="header-section-number">8.1</span> Download results</h3>
<ul>
<li><a
href="./file/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge.xlsx"><strong>Differential
gene expression</strong></a></li>
</ul>
</div>
<div id="summary-of-results" class="section level3" number="8.2">
<h3><span class="header-section-number">8.2</span> Summary of
results</h3>
<div id="fdr-distribution" class="section level4" number="8.2.1">
<h4><span class="header-section-number">8.2.1</span> FDR
distribution</h4>
<pre class="r"><code>i &lt;- names(dge_list[1])
res &lt;- foreach(i = names(dge_list), .combine = rbind) %do% {
  x &lt;- dge_list[[i]]$results
  c(
    `FDR &lt; 0.1` = x %&gt;% filter(FDR &lt;= 0.1) %&gt;% nrow,
    `FDR &lt; 0.05` = x %&gt;% filter(FDR &lt;= 0.05) %&gt;% nrow,
    `FDR &lt; 0.01` = x %&gt;% filter(FDR &lt;= 0.01) %&gt;% nrow,
    `FDR &lt; 0.001` = x %&gt;% filter(FDR &lt;= 0.001) %&gt;% nrow,
    `FDR &lt; 0.0001` = x %&gt;% filter(FDR &lt;= 0.0001) %&gt;% nrow
  )
}
if(class(res)[1] == &#39;integer&#39;){
  res &lt;- data.frame(res) %&gt;% t() 
}

rownames(res) &lt;- names(dge_list)
res %&gt;% 
  data.frame(check.names = F) %&gt;% 
  rownames_to_column(&#39;comparison&#39;) %&gt;% 
  kbl(caption = &#39;Number of genes at different FDR thresholds&#39;) %&gt;%
  kable_paper(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;), full_width = F) %&gt;%
  column_spec(1, bold = TRUE)</code></pre>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Number of genes at different FDR thresholds
</caption>
<thead>
<tr>
<th style="text-align:left;">
comparison
</th>
<th style="text-align:right;">
FDR &lt; 0.1
</th>
<th style="text-align:right;">
FDR &lt; 0.05
</th>
<th style="text-align:right;">
FDR &lt; 0.01
</th>
<th style="text-align:right;">
FDR &lt; 0.001
</th>
<th style="text-align:right;">
FDR &lt; 0.0001
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: DC TME over Normal
</td>
<td style="text-align:right;">
1934
</td>
<td style="text-align:right;">
1449
</td>
<td style="text-align:right;">
773
</td>
<td style="text-align:right;">
365
</td>
<td style="text-align:right;">
176
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Endothelial TME over Normal
</td>
<td style="text-align:right;">
1360
</td>
<td style="text-align:right;">
875
</td>
<td style="text-align:right;">
445
</td>
<td style="text-align:right;">
228
</td>
<td style="text-align:right;">
146
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Fibroblasts TME over Normal
</td>
<td style="text-align:right;">
1144
</td>
<td style="text-align:right;">
882
</td>
<td style="text-align:right;">
523
</td>
<td style="text-align:right;">
304
</td>
<td style="text-align:right;">
205
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Macrophages TME over Normal
</td>
<td style="text-align:right;">
2498
</td>
<td style="text-align:right;">
1801
</td>
<td style="text-align:right;">
966
</td>
<td style="text-align:right;">
493
</td>
<td style="text-align:right;">
292
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Monocytes TME over Normal
</td>
<td style="text-align:right;">
5341
</td>
<td style="text-align:right;">
4640
</td>
<td style="text-align:right;">
3433
</td>
<td style="text-align:right;">
2035
</td>
<td style="text-align:right;">
1082
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Neutrophils TME over Normal
</td>
<td style="text-align:right;">
2480
</td>
<td style="text-align:right;">
2074
</td>
<td style="text-align:right;">
1407
</td>
<td style="text-align:right;">
768
</td>
<td style="text-align:right;">
390
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: DC TME over Normal
</td>
<td style="text-align:right;">
1585
</td>
<td style="text-align:right;">
1203
</td>
<td style="text-align:right;">
659
</td>
<td style="text-align:right;">
331
</td>
<td style="text-align:right;">
184
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Endothelial TME over Normal
</td>
<td style="text-align:right;">
5402
</td>
<td style="text-align:right;">
4325
</td>
<td style="text-align:right;">
2497
</td>
<td style="text-align:right;">
984
</td>
<td style="text-align:right;">
315
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Macrophages TME over Normal
</td>
<td style="text-align:right;">
3783
</td>
<td style="text-align:right;">
3094
</td>
<td style="text-align:right;">
1971
</td>
<td style="text-align:right;">
1163
</td>
<td style="text-align:right;">
714
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Monocytes TME over Normal
</td>
<td style="text-align:right;">
3169
</td>
<td style="text-align:right;">
2472
</td>
<td style="text-align:right;">
1502
</td>
<td style="text-align:right;">
803
</td>
<td style="text-align:right;">
440
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Neutrophils TME over Normal
</td>
<td style="text-align:right;">
845
</td>
<td style="text-align:right;">
674
</td>
<td style="text-align:right;">
375
</td>
<td style="text-align:right;">
163
</td>
<td style="text-align:right;">
59
</td>
</tr>
</tbody>
</table>
</div>
<div id="number-of-genes-at-fdr-0.05" class="section level4"
number="8.2.2">
<h4><span class="header-section-number">8.2.2</span> Number of genes at
FDR 0.05</h4>
<pre class="r"><code>i &lt;- names(dge_list[1])
res &lt;- foreach(i = names(dge_list), .combine = rbind) %do% {
  x &lt;- dge_list[[i]]$results %&gt;% 
    filter(FDR &lt;= 0.05)
  c(
    `Up` = x %&gt;% filter(logFC &gt; 0) %&gt;% nrow,
    `Down` = x %&gt;% filter(logFC &lt; 0) %&gt;% nrow
  )
}
if(class(res)[1] == &#39;integer&#39;){
  res &lt;- data.frame(res) %&gt;% t() 
}

rownames(res) &lt;- names(dge_list)
res %&gt;% 
  data.frame(check.names = F) %&gt;% 
  rownames_to_column(&#39;comparison&#39;) %&gt;% 
  kbl(caption = &#39;Number of genes at at FDR &lt; 0&#39;) %&gt;%
  kable_paper(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;), full_width = F) %&gt;%
  column_spec(1, bold = TRUE)</code></pre>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Number of genes at at FDR
</caption>
<thead>
<tr>
<th style="text-align:left;">
comparison
</th>
<th style="text-align:right;">
Up
</th>
<th style="text-align:right;">
Down
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: DC TME over Normal
</td>
<td style="text-align:right;">
567
</td>
<td style="text-align:right;">
882
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Endothelial TME over Normal
</td>
<td style="text-align:right;">
149
</td>
<td style="text-align:right;">
726
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Fibroblasts TME over Normal
</td>
<td style="text-align:right;">
552
</td>
<td style="text-align:right;">
330
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Macrophages TME over Normal
</td>
<td style="text-align:right;">
933
</td>
<td style="text-align:right;">
868
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Monocytes TME over Normal
</td>
<td style="text-align:right;">
2276
</td>
<td style="text-align:right;">
2364
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Neutrophils TME over Normal
</td>
<td style="text-align:right;">
924
</td>
<td style="text-align:right;">
1150
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: DC TME over Normal
</td>
<td style="text-align:right;">
440
</td>
<td style="text-align:right;">
763
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Endothelial TME over Normal
</td>
<td style="text-align:right;">
1897
</td>
<td style="text-align:right;">
2428
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Macrophages TME over Normal
</td>
<td style="text-align:right;">
1626
</td>
<td style="text-align:right;">
1468
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Monocytes TME over Normal
</td>
<td style="text-align:right;">
1113
</td>
<td style="text-align:right;">
1359
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Neutrophils TME over Normal
</td>
<td style="text-align:right;">
291
</td>
<td style="text-align:right;">
383
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="volcano-plots" class="section level3" number="8.3">
<h3><span class="header-section-number">8.3</span> Volcano-plots</h3>
<!-- Top-20 genes based on adjusted P-values are labeled. -->
<pre class="r"><code>i=names(dge_list)[1]
p_thrs &lt;- 0.05
for(i in names(dge_list)) {
  cat(&quot;#### Comparison:&quot;, i, &quot;\n\n&quot;)
  # cat(dge_list[[i]]$comparison, &quot;\n\n&quot;)
  x &lt;- dge_list[[i]]$results
  res &lt;- plot.volcano(
    x$logFC, 
    x$FDR, 
    label = x$gene_name, 
    fill.palette = c(&quot;gray80&quot;, &quot;red&quot;), 
    top.labels = 20, 
    pval.thr = p_thrs, 
    hline.color =  &#39;grey60&#39;, hline.type = 2, hline.size = 0.5) +
    geom_vline(xintercept = 0, linetype = 2, linewidth = 0.5, color = &#39;grey60&#39;)
  print(res)
  cat(&quot;\n\n&quot;)
}</code></pre>
<pre class="r"><code>i=names(dge_list)[1]
p_thrs &lt;- 0.05
for(i in names(dge_list)) {
  cat(&quot;#### Comparison:&quot;, i, &quot;\n\n&quot;)
  # cat(dge_list[[i]]$comparison, &quot;\n\n&quot;)
  x &lt;- dge_list[[i]]$results %&gt;% 
    mutate(
      direction = ifelse(FDR&lt;0.05 &amp; logFC &gt; 0, &#39;Up&#39;, &#39;none&#39;),
      direction = ifelse(FDR&lt;0.05 &amp; logFC &lt; 0, &#39;Down&#39;, direction)
      )
  use_xlim &lt;- xlim(-max(abs(x$logFC), na.rm = TRUE), max(abs(x$logFC), na.rm = TRUE))
  res &lt;- x %&gt;% 
    ggplot(aes(logFC, -log10(FDR), fill = direction)) +
    geom_point(size = 2, stroke = 0.01, shape = 21,
                colour = &#39;white&#39;, alpha = 0.3) +
    # geom_point() +
    use_xlim +
    scale_fill_manual(values = c(Up = &#39;red&#39;, none = &#39;gray80&#39;, Down = &#39;blue&#39;)) +
    geom_hline(yintercept = -log10(0.05), linetype = 2, linewidth = 0.5, color = &#39;grey60&#39;) +
    geom_vline(xintercept = 0, linetype = 2, linewidth = 0.5, color = &#39;grey60&#39;) +
    labs(
      fill = NULL,
      x = expression(paste(&quot;lo&quot;, g[2],&quot;(Fold change)&quot;)),
      y = expression(paste(&quot;lo&quot;, g[10],&quot;(Adjusted P value)&quot;)),
    ) +
    theme(legend.position=&quot;none&quot;)
  print(res)

  cat(&quot;\n\n&quot;)
}</code></pre>
<div id="comparison-lung-dc-tme-over-normal" class="section level4"
number="8.3.1">
<h4><span class="header-section-number">8.3.1</span> Comparison: Lung:
DC TME over Normal</h4>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-volcano-colored-plot-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-lung-endothelial-tme-over-normal"
class="section level4" number="8.3.2">
<h4><span class="header-section-number">8.3.2</span> Comparison: Lung:
Endothelial TME over Normal</h4>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-volcano-colored-plot-2.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-lung-fibroblasts-tme-over-normal"
class="section level4" number="8.3.3">
<h4><span class="header-section-number">8.3.3</span> Comparison: Lung:
Fibroblasts TME over Normal</h4>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-volcano-colored-plot-3.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-lung-macrophages-tme-over-normal"
class="section level4" number="8.3.4">
<h4><span class="header-section-number">8.3.4</span> Comparison: Lung:
Macrophages TME over Normal</h4>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-volcano-colored-plot-4.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-lung-monocytes-tme-over-normal"
class="section level4" number="8.3.5">
<h4><span class="header-section-number">8.3.5</span> Comparison: Lung:
Monocytes TME over Normal</h4>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-volcano-colored-plot-5.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-lung-neutrophils-tme-over-normal"
class="section level4" number="8.3.6">
<h4><span class="header-section-number">8.3.6</span> Comparison: Lung:
Neutrophils TME over Normal</h4>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-volcano-colored-plot-6.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-liver-dc-tme-over-normal" class="section level4"
number="8.3.7">
<h4><span class="header-section-number">8.3.7</span> Comparison: Liver:
DC TME over Normal</h4>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-volcano-colored-plot-7.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-liver-endothelial-tme-over-normal"
class="section level4" number="8.3.8">
<h4><span class="header-section-number">8.3.8</span> Comparison: Liver:
Endothelial TME over Normal</h4>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-volcano-colored-plot-8.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-liver-macrophages-tme-over-normal"
class="section level4" number="8.3.9">
<h4><span class="header-section-number">8.3.9</span> Comparison: Liver:
Macrophages TME over Normal</h4>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-volcano-colored-plot-9.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-liver-monocytes-tme-over-normal"
class="section level4" number="8.3.10">
<h4><span class="header-section-number">8.3.10</span> Comparison: Liver:
Monocytes TME over Normal</h4>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-volcano-colored-plot-10.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-liver-neutrophils-tme-over-normal"
class="section level4" number="8.3.11">
<h4><span class="header-section-number">8.3.11</span> Comparison: Liver:
Neutrophils TME over Normal</h4>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-volcano-colored-plot-11.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="heatmaps" class="section level3" number="8.4">
<h3><span class="header-section-number">8.4</span> Heatmaps</h3>
<pre class="r"><code>i &lt;- names(dge_list)[1]
p_thrs &lt;- 0.05
lfc_thrs &lt;- 1
topn &lt;- 50
max_color_range &lt;- 3 # put the color range to -3 to 3

for(i in names(dge_list)) {
  cat(&quot;#### Comparison&quot;, i, &quot;\n\n&quot;)
  use_comp &lt;- comp_list[[i]]
  
  # Extract samples from comparisons
  use_samples &lt;- unlist(use_comp[1:2])
  use_se &lt;- se[,use_samples]
  
  # Arrange samples by condition (first control and then sample)
  use_annot &lt;- colData(use_se) %&gt;% data.frame %&gt;% 
    mutate(
      condition_new = ifelse(sample_alias %in% use_comp[[1]], 
                             names(use_comp)[1], 
                             names(use_comp)[2]),
      condition_new = factor(condition_new, levels = names(use_comp)[2:1]),
      condition_num = as.numeric(condition_new)
    ) %&gt;% 
    arrange(condition_num)
  use_samples &lt;- use_annot$sample_alias
  
  # Extract DGE results
  use_dge &lt;- dge_list[[i]]$results %&gt;% 
    filter(FDR &lt; p_thrs &amp; abs(logFC) &gt; lfc_thrs) %&gt;% 
    mutate(direction = ifelse(logFC &gt; 0, &#39;Up&#39;, &#39;Down&#39;)) %&gt;% 
    arrange(FDR) %&gt;% 
    head(topn) %&gt;% 
    arrange(logFC)
  if(nrow(use_dge) &gt; 50){
    cat(paste0(&quot;Showing &quot;, nrow(use_dge),&quot; features with FDR &lt; &quot;, p_thrs, &quot; and absolute log2 fold-change &gt; &quot;, lfc_thrs, &quot;\n&quot;))
  } else {
    cat(paste0(&quot;Only &quot;, nrow(use_dge),&quot; features with FDR &lt; &quot;, p_thrs, &quot; and absolute log2 fold-change &gt; &quot;, lfc_thrs, &quot;. Showing top-&quot;, topn,&quot; features based on P value and absolute log2 fold-change &gt;&quot;, lfc_thrs,&quot;\n&quot;))
    use_dge &lt;- dge_list[[i]]$results %&gt;% 
      filter(abs(logFC) &gt; lfc_thrs) %&gt;% 
      arrange(PValue) %&gt;% 
      mutate(direction = ifelse(logFC &gt; 0, &#39;Up&#39;, &#39;Down&#39;)) %&gt;% 
      head(topn) %&gt;% 
      arrange(logFC)
  }
  
  # Extract matrix of expression
  mat &lt;- assay(use_se, &#39;logcounts&#39;)[rownames(use_dge), use_samples]
  rownames(mat) &lt;- make.names(use_dge$gene_name, unique = TRUE)
  zmat &lt;- t(apply(mat, 1, scale, center = TRUE, scale = TRUE))
  colnames(zmat) &lt;- colnames(mat)
  zmat &lt;- zmat[,rownames(use_annot)]
  
  # Col Annotation
  use_annot_levels &lt;-levels(use_annot$condition_new)
  column_ha &lt;-  HeatmapAnnotation(
    condition = anno_block(
      # gp = gpar(fill = c(&#39;#fdbf6f&#39;, &#39;#6a3d9a&#39;) %&gt;% set_names(use_annot_levels)),
      labels = use_annot_levels, 
      labels_gp = gpar(col = &quot;black&quot;, fontsize = 8)
    ),
    `Condition` = use_annot$site_caps,
    col = list(`Condition` = site_type_palette)
  )
    
  # Row annotation (right)
  row_annot_right &lt;-  rowAnnotation(
    log2FC = anno_numeric(
      round(use_dge$logFC,1), 
      labels_gp = gpar(fontsize = 6), 
      bg_gp = gpar(
        fill = c(&quot;#74add1&quot;, &quot;#f46d43&quot;), 
        col = &#39;white&#39;
      ),
      bar_width = unit(1, &quot;npc&quot;) - unit(1.5, &quot;pt&quot;)
    ), 
    annotation_name_rot = 0)
  
  # Row annotation (left)
  direction_palette &lt;- c(&#39;Up&#39; = &quot;#f46d43&quot;, &#39;Down&#39; = &quot;#74add1&quot;)
  direction_palette &lt;- direction_palette[names(direction_palette) %in% use_dge$direction]
  
  row_annot_left &lt;-  rowAnnotation(
    direction = anno_block(
      gp = gpar(fill = direction_palette),
      labels = names(direction_palette), 
      labels_gp = gpar(col = &quot;white&quot;, fontsize = 8)
    )
  )
  
  # Heatmap color
  zmax &lt;- zmat %&gt;% abs %&gt;% max
  if(!is.null(max_color_range))
    zmax &lt;- max_color_range
  col_fun &lt;-  colorRamp2(
    seq(-zmax, zmax, length.out = 11),
    rev(brewer.pal(11, &quot;RdBu&quot;))
  )
  
  # Generate Heatmap
  htz &lt;- Heatmap(
    zmat,
    name= &#39;z-score&#39;,
    col = col_fun,
    row_names_gp = gpar(fontsize = 8),
    column_names_gp = gpar(fontsize = 8),
    
    top_annotation = column_ha,
    column_split = use_annot$condition_num,
    #cluster_column_slices = turn off the clustering on slices, and now you can precisely control the order of slices
    cluster_column_slices = FALSE, 
    column_title = NULL,
    
    right_annotation = row_annot_right,
    row_split = use_dge$direction,
    # left_annotation = row_annot_left,
    row_title = NULL
  )
  
  draw(
    htz, 
    # heatmap_legend_side = &quot;bottom&quot;, 
    annotation_legend_side = &quot;bottom&quot;
  )
    
  cat(&quot;\n\n&quot;)
  
}</code></pre>
<div id="comparison-lung-dc-tme-over-normal-1" class="section level4"
number="8.4.1">
<h4><span class="header-section-number">8.4.1</span> Comparison Lung: DC
TME over Normal</h4>
<p>Only 50 features with FDR &lt; 0.05 and absolute log2 fold-change
&gt; 1. Showing top-50 features based on P value and absolute log2
fold-change &gt;1
<img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-heatmap-plot-1.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-lung-endothelial-tme-over-normal-1"
class="section level4" number="8.4.2">
<h4><span class="header-section-number">8.4.2</span> Comparison Lung:
Endothelial TME over Normal</h4>
<p>Only 50 features with FDR &lt; 0.05 and absolute log2 fold-change
&gt; 1. Showing top-50 features based on P value and absolute log2
fold-change &gt;1
<img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-heatmap-plot-2.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-lung-fibroblasts-tme-over-normal-1"
class="section level4" number="8.4.3">
<h4><span class="header-section-number">8.4.3</span> Comparison Lung:
Fibroblasts TME over Normal</h4>
<p>Only 50 features with FDR &lt; 0.05 and absolute log2 fold-change
&gt; 1. Showing top-50 features based on P value and absolute log2
fold-change &gt;1
<img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-heatmap-plot-3.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-lung-macrophages-tme-over-normal-1"
class="section level4" number="8.4.4">
<h4><span class="header-section-number">8.4.4</span> Comparison Lung:
Macrophages TME over Normal</h4>
<p>Only 50 features with FDR &lt; 0.05 and absolute log2 fold-change
&gt; 1. Showing top-50 features based on P value and absolute log2
fold-change &gt;1
<img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-heatmap-plot-4.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-lung-monocytes-tme-over-normal-1"
class="section level4" number="8.4.5">
<h4><span class="header-section-number">8.4.5</span> Comparison Lung:
Monocytes TME over Normal</h4>
<p>Only 50 features with FDR &lt; 0.05 and absolute log2 fold-change
&gt; 1. Showing top-50 features based on P value and absolute log2
fold-change &gt;1
<img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-heatmap-plot-5.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-lung-neutrophils-tme-over-normal-1"
class="section level4" number="8.4.6">
<h4><span class="header-section-number">8.4.6</span> Comparison Lung:
Neutrophils TME over Normal</h4>
<p>Only 50 features with FDR &lt; 0.05 and absolute log2 fold-change
&gt; 1. Showing top-50 features based on P value and absolute log2
fold-change &gt;1
<img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-heatmap-plot-6.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-liver-dc-tme-over-normal-1" class="section level4"
number="8.4.7">
<h4><span class="header-section-number">8.4.7</span> Comparison Liver:
DC TME over Normal</h4>
<p>Only 50 features with FDR &lt; 0.05 and absolute log2 fold-change
&gt; 1. Showing top-50 features based on P value and absolute log2
fold-change &gt;1
<img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-heatmap-plot-7.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-liver-endothelial-tme-over-normal-1"
class="section level4" number="8.4.8">
<h4><span class="header-section-number">8.4.8</span> Comparison Liver:
Endothelial TME over Normal</h4>
<p>Only 50 features with FDR &lt; 0.05 and absolute log2 fold-change
&gt; 1. Showing top-50 features based on P value and absolute log2
fold-change &gt;1
<img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-heatmap-plot-8.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-liver-macrophages-tme-over-normal-1"
class="section level4" number="8.4.9">
<h4><span class="header-section-number">8.4.9</span> Comparison Liver:
Macrophages TME over Normal</h4>
<p>Only 50 features with FDR &lt; 0.05 and absolute log2 fold-change
&gt; 1. Showing top-50 features based on P value and absolute log2
fold-change &gt;1
<img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-heatmap-plot-9.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-liver-monocytes-tme-over-normal-1"
class="section level4" number="8.4.10">
<h4><span class="header-section-number">8.4.10</span> Comparison Liver:
Monocytes TME over Normal</h4>
<p>Only 50 features with FDR &lt; 0.05 and absolute log2 fold-change
&gt; 1. Showing top-50 features based on P value and absolute log2
fold-change &gt;1
<img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-heatmap-plot-10.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-liver-neutrophils-tme-over-normal-1"
class="section level4" number="8.4.11">
<h4><span class="header-section-number">8.4.11</span> Comparison Liver:
Neutrophils TME over Normal</h4>
<p>Only 50 features with FDR &lt; 0.05 and absolute log2 fold-change
&gt; 1. Showing top-50 features based on P value and absolute log2
fold-change &gt;1
<img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/dge-heatmap-plot-11.png" width="4320" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="functional-enrichment-analysis" class="section level2"
number="9">
<h2><span class="header-section-number">9</span> Functional enrichment
analysis</h2>
<pre class="r"><code>gse_collections &lt;- list(
  GSEA = names(gmt_files_symbols),
  gseGO = c(&quot;BP&quot;)
)</code></pre>
<div id="download-results-2" class="section level3" number="9.1">
<h3><span class="header-section-number">9.1</span> Download results</h3>
<!-- Write links to download GSE results -->
<ul>
<li><p><a
href="./file/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse_gsea_msigdb.c2.cp.xlsx"><strong>Gene
set enrichment analysis: msigdb.c2.cp</strong></a></p></li>
<li><p><a
href="./file/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse_gsea_msigdb.m2.cp.reactome.xlsx"><strong>Gene
set enrichment analysis: msigdb.m2.cp.reactome</strong></a></p></li>
<li><p><a
href="./file/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse_gsego_bp.xlsx"><strong>Gene
set enrichment analysis: GO:BP</strong></a></p></li>
<li><p><a
href="./file/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse_gsego_mf.xlsx"><strong>Gene
set enrichment analysis: GO:MF</strong></a></p></li>
</ul>
</div>
<div id="summary-of-results-1" class="section level3" number="9.2">
<h3><span class="header-section-number">9.2</span> Summary of
results</h3>
<pre class="r"><code>res &lt;- foreach(i = names(gse_list), .combine = rbind) %do% {
  foreach(sr = names(gse_collections), .combine = rbind) %do% {
    foreach(gc = gse_collections[[sr]], .combine = rbind) %do% {
      if(sr == &#39;GSEA&#39;) {
        x &lt;- gse_list[[i]][[sr]][[gc]]@result %&gt;% 
          filter(p.adjust &lt;= 0.05)
        if(gc == &#39;msigdb.c2.cp&#39;) {
          x %&lt;&gt;% 
            filter(grepl(&#39;REACTOME|KEGG|WP&#39;, ID))
        }
      }
      if(sr == &#39;gseGO&#39;) {
        x &lt;- gse_list[[i]][[sr]][[gc]]@result %&gt;% 
          filter(p.adjust &lt;= 0.05)
      }
      c(
        Comparison = i,
        `Collection` = gc,
        `Total` = nrow(x),
        `Supressed` = x %&gt;% filter(NES &lt; 0) %&gt;% nrow,
        `Activated` = x %&gt;% filter(NES &gt; 0) %&gt;% nrow
      )
    }
  }
}
res &lt;- res %&gt;% data.frame(check.names = F)</code></pre>
<div id="table" class="section level4" number="9.2.1">
<h4><span class="header-section-number">9.2.1</span> Table</h4>
<pre class="r"><code>res %&gt;% 
  data.frame(check.names = F) %&gt;% 
  kbl(caption = &#39;Number of gene sets at FDR &lt; 0.05&#39;,
              row.names = F) %&gt;%
  kable_paper(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;), 
              full_width = F) %&gt;%
  column_spec(1, bold = TRUE)</code></pre>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Number of gene sets at FDR
</caption>
<thead>
<tr>
<th style="text-align:left;">
Comparison
</th>
<th style="text-align:left;">
Collection
</th>
<th style="text-align:left;">
Total
</th>
<th style="text-align:left;">
Supressed
</th>
<th style="text-align:left;">
Activated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: DC TME over Normal
</td>
<td style="text-align:left;">
msigdb.c2.cp
</td>
<td style="text-align:left;">
60
</td>
<td style="text-align:left;">
18
</td>
<td style="text-align:left;">
42
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: DC TME over Normal
</td>
<td style="text-align:left;">
msigdb.m2.cp.reactome
</td>
<td style="text-align:left;">
49
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
38
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: DC TME over Normal
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
99
</td>
<td style="text-align:left;">
45
</td>
<td style="text-align:left;">
54
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Endothelial TME over Normal
</td>
<td style="text-align:left;">
msigdb.c2.cp
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Endothelial TME over Normal
</td>
<td style="text-align:left;">
msigdb.m2.cp.reactome
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Endothelial TME over Normal
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Fibroblasts TME over Normal
</td>
<td style="text-align:left;">
msigdb.c2.cp
</td>
<td style="text-align:left;">
17
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
16
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Fibroblasts TME over Normal
</td>
<td style="text-align:left;">
msigdb.m2.cp.reactome
</td>
<td style="text-align:left;">
13
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
13
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Fibroblasts TME over Normal
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
276
</td>
<td style="text-align:left;">
42
</td>
<td style="text-align:left;">
234
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Macrophages TME over Normal
</td>
<td style="text-align:left;">
msigdb.c2.cp
</td>
<td style="text-align:left;">
35
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
32
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Macrophages TME over Normal
</td>
<td style="text-align:left;">
msigdb.m2.cp.reactome
</td>
<td style="text-align:left;">
36
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
33
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Macrophages TME over Normal
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
182
</td>
<td style="text-align:left;">
23
</td>
<td style="text-align:left;">
159
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Monocytes TME over Normal
</td>
<td style="text-align:left;">
msigdb.c2.cp
</td>
<td style="text-align:left;">
133
</td>
<td style="text-align:left;">
36
</td>
<td style="text-align:left;">
97
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Monocytes TME over Normal
</td>
<td style="text-align:left;">
msigdb.m2.cp.reactome
</td>
<td style="text-align:left;">
111
</td>
<td style="text-align:left;">
31
</td>
<td style="text-align:left;">
80
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Monocytes TME over Normal
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
294
</td>
<td style="text-align:left;">
62
</td>
<td style="text-align:left;">
232
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Neutrophils TME over Normal
</td>
<td style="text-align:left;">
msigdb.c2.cp
</td>
<td style="text-align:left;">
124
</td>
<td style="text-align:left;">
32
</td>
<td style="text-align:left;">
92
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Neutrophils TME over Normal
</td>
<td style="text-align:left;">
msigdb.m2.cp.reactome
</td>
<td style="text-align:left;">
111
</td>
<td style="text-align:left;">
27
</td>
<td style="text-align:left;">
84
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Lung: Neutrophils TME over Normal
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
255
</td>
<td style="text-align:left;">
60
</td>
<td style="text-align:left;">
195
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: DC TME over Normal
</td>
<td style="text-align:left;">
msigdb.c2.cp
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
7
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: DC TME over Normal
</td>
<td style="text-align:left;">
msigdb.m2.cp.reactome
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
7
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: DC TME over Normal
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
47
</td>
<td style="text-align:left;">
24
</td>
<td style="text-align:left;">
23
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Endothelial TME over Normal
</td>
<td style="text-align:left;">
msigdb.c2.cp
</td>
<td style="text-align:left;">
59
</td>
<td style="text-align:left;">
18
</td>
<td style="text-align:left;">
41
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Endothelial TME over Normal
</td>
<td style="text-align:left;">
msigdb.m2.cp.reactome
</td>
<td style="text-align:left;">
42
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
26
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Endothelial TME over Normal
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
381
</td>
<td style="text-align:left;">
46
</td>
<td style="text-align:left;">
335
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Macrophages TME over Normal
</td>
<td style="text-align:left;">
msigdb.c2.cp
</td>
<td style="text-align:left;">
110
</td>
<td style="text-align:left;">
30
</td>
<td style="text-align:left;">
80
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Macrophages TME over Normal
</td>
<td style="text-align:left;">
msigdb.m2.cp.reactome
</td>
<td style="text-align:left;">
99
</td>
<td style="text-align:left;">
27
</td>
<td style="text-align:left;">
72
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Macrophages TME over Normal
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
274
</td>
<td style="text-align:left;">
103
</td>
<td style="text-align:left;">
171
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Monocytes TME over Normal
</td>
<td style="text-align:left;">
msigdb.c2.cp
</td>
<td style="text-align:left;">
112
</td>
<td style="text-align:left;">
33
</td>
<td style="text-align:left;">
79
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Monocytes TME over Normal
</td>
<td style="text-align:left;">
msigdb.m2.cp.reactome
</td>
<td style="text-align:left;">
100
</td>
<td style="text-align:left;">
32
</td>
<td style="text-align:left;">
68
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Monocytes TME over Normal
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
143
</td>
<td style="text-align:left;">
38
</td>
<td style="text-align:left;">
105
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Neutrophils TME over Normal
</td>
<td style="text-align:left;">
msigdb.c2.cp
</td>
<td style="text-align:left;">
62
</td>
<td style="text-align:left;">
15
</td>
<td style="text-align:left;">
47
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Neutrophils TME over Normal
</td>
<td style="text-align:left;">
msigdb.m2.cp.reactome
</td>
<td style="text-align:left;">
56
</td>
<td style="text-align:left;">
15
</td>
<td style="text-align:left;">
41
</td>
</tr>
<tr>
<td style="text-align:left;font-weight: bold;">
Liver: Neutrophils TME over Normal
</td>
<td style="text-align:left;">
BP
</td>
<td style="text-align:left;">
28
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
25
</td>
</tr>
</tbody>
</table>
</div>
<div id="barplot" class="section level4" number="9.2.2">
<h4><span class="header-section-number">9.2.2</span> Barplot</h4>
<pre class="r"><code>res %&gt;% 
  pivot_longer(cols = c(&#39;Total&#39;, &#39;Supressed&#39;, &#39;Activated&#39;), names_to = &#39;Direction&#39;, values_to = &#39;Count&#39;) %&gt;% 
  mutate(
    comb_names = paste(Comparison, Collection),
    Count = as.numeric(Count),
    Comparison = gsub(&quot;-over-&quot;, &quot; over\n&quot;, Comparison),
    Comparison = gsub(&quot; corrected&quot;, &quot;\ncorrected&quot;, Comparison)
    ) %&gt;% 
  ggplot(aes(Collection, Count, fill = Direction)) +
  geom_col(position = &quot;dodge&quot;) +
  coord_flip() +
  facet_grid(rows = vars(Comparison)) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.05))) +
  scale_fill_manual(values = c(Supressed = &#39;#74add1&#39;, 
                               Activated = &#39;#f46d43&#39;,
                               Total = &#39;grey80&#39;)
                    ) +
  labs(x = &#39;&#39;, fill = &#39;&#39;) +
  theme_facet +
  theme(
    axis.line = element_blank(),
    axis.ticks = element_line(linewidth = one_pt/4, color = &#39;black&#39;),
    strip.text = element_text(size = 5)
    
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-summary-fdr-barplot-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="dotplots-2" class="section level3" number="9.3">
<h3><span class="header-section-number">9.3</span> Dotplots</h3>
<!-- # Subchunkify dotplots -->
<!-- ## chunk options need to be set to echo=FALSE, results='asis' -->
<!-- ## Cause results='asis',we must use raw HTML to print text and headings -->
<h4>
<span class="header-section-number"> 8.2.1 </span>Comparison Lung: DC
TME over Normal
</h4>
<h5>
<span class="header-section-number"> 8.2.1.1 </span>Collection
msigdb.c2.cp
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-dc-tme-over-normal_GSEA_msigdb.c2.cp-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.1.2 </span>Collection
msigdb.m2.cp.reactome
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-dc-tme-over-normal_GSEA_msigdb.m2.cp.reactome-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.1.3 </span>Collection BP
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-dc-tme-over-normal_gseGO_BP-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h4>
<span class="header-section-number"> 8.2.2 </span>Comparison Lung:
Endothelial TME over Normal
</h4>
<h4>
<span class="header-section-number"> 8.2.3 </span>Comparison Lung:
Fibroblasts TME over Normal
</h4>
<h5>
<span class="header-section-number"> 8.2.3.4 </span>Collection
msigdb.c2.cp
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-fibroblasts-tme-over-normal_GSEA_msigdb.c2.cp-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.3.5 </span>Collection
msigdb.m2.cp.reactome
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-fibroblasts-tme-over-normal_GSEA_msigdb.m2.cp.reactome-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.3.6 </span>Collection BP
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-fibroblasts-tme-over-normal_gseGO_BP-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h4>
<span class="header-section-number"> 8.2.4 </span>Comparison Lung:
Macrophages TME over Normal
</h4>
<h5>
<span class="header-section-number"> 8.2.4.7 </span>Collection
msigdb.c2.cp
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-macrophages-tme-over-normal_GSEA_msigdb.c2.cp-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.4.8 </span>Collection
msigdb.m2.cp.reactome
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-macrophages-tme-over-normal_GSEA_msigdb.m2.cp.reactome-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.4.9 </span>Collection BP
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-macrophages-tme-over-normal_gseGO_BP-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h4>
<span class="header-section-number"> 8.2.5 </span>Comparison Lung:
Monocytes TME over Normal
</h4>
<h5>
<span class="header-section-number"> 8.2.5.10 </span>Collection
msigdb.c2.cp
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-monocytes-tme-over-normal_GSEA_msigdb.c2.cp-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.5.11 </span>Collection
msigdb.m2.cp.reactome
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-monocytes-tme-over-normal_GSEA_msigdb.m2.cp.reactome-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.5.12 </span>Collection BP
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-monocytes-tme-over-normal_gseGO_BP-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h4>
<span class="header-section-number"> 8.2.6 </span>Comparison Lung:
Neutrophils TME over Normal
</h4>
<h5>
<span class="header-section-number"> 8.2.6.13 </span>Collection
msigdb.c2.cp
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-neutrophils-tme-over-normal_GSEA_msigdb.c2.cp-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.6.14 </span>Collection
msigdb.m2.cp.reactome
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-neutrophils-tme-over-normal_GSEA_msigdb.m2.cp.reactome-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.6.15 </span>Collection BP
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-lung:-neutrophils-tme-over-normal_gseGO_BP-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h4>
<span class="header-section-number"> 8.2.7 </span>Comparison Liver: DC
TME over Normal
</h4>
<h5>
<span class="header-section-number"> 8.2.7.16 </span>Collection
msigdb.c2.cp
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-dc-tme-over-normal_GSEA_msigdb.c2.cp-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.7.17 </span>Collection
msigdb.m2.cp.reactome
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-dc-tme-over-normal_GSEA_msigdb.m2.cp.reactome-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.7.18 </span>Collection BP
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-dc-tme-over-normal_gseGO_BP-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h4>
<span class="header-section-number"> 8.2.8 </span>Comparison Liver:
Endothelial TME over Normal
</h4>
<h5>
<span class="header-section-number"> 8.2.8.19 </span>Collection
msigdb.c2.cp
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-endothelial-tme-over-normal_GSEA_msigdb.c2.cp-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.8.20 </span>Collection
msigdb.m2.cp.reactome
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-endothelial-tme-over-normal_GSEA_msigdb.m2.cp.reactome-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.8.21 </span>Collection BP
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-endothelial-tme-over-normal_gseGO_BP-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h4>
<span class="header-section-number"> 8.2.9 </span>Comparison Liver:
Macrophages TME over Normal
</h4>
<h5>
<span class="header-section-number"> 8.2.9.22 </span>Collection
msigdb.c2.cp
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-macrophages-tme-over-normal_GSEA_msigdb.c2.cp-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.9.23 </span>Collection
msigdb.m2.cp.reactome
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-macrophages-tme-over-normal_GSEA_msigdb.m2.cp.reactome-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.9.24 </span>Collection BP
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-macrophages-tme-over-normal_gseGO_BP-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h4>
<span class="header-section-number"> 8.2.10 </span>Comparison Liver:
Monocytes TME over Normal
</h4>
<h5>
<span class="header-section-number"> 8.2.10.25 </span>Collection
msigdb.c2.cp
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-monocytes-tme-over-normal_GSEA_msigdb.c2.cp-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.10.26 </span>Collection
msigdb.m2.cp.reactome
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-monocytes-tme-over-normal_GSEA_msigdb.m2.cp.reactome-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.10.27 </span>Collection BP
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-monocytes-tme-over-normal_gseGO_BP-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h4>
<span class="header-section-number"> 8.2.11 </span>Comparison Liver:
Neutrophils TME over Normal
</h4>
<h5>
<span class="header-section-number"> 8.2.11.28 </span>Collection
msigdb.c2.cp
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-neutrophils-tme-over-normal_GSEA_msigdb.c2.cp-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.11.29 </span>Collection
msigdb.m2.cp.reactome
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-neutrophils-tme-over-normal_GSEA_msigdb.m2.cp.reactome-1.png" width="2100" style="display: block; margin: auto;" /></p>
<h5>
<span class="header-section-number"> 8.2.11.30 </span>Collection BP
</h5>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd/gse-dotplot-liver:-neutrophils-tme-over-normal_gseGO_BP-1.png" width="2100" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<p>R version 4.2.2 (2022-10-31) Platform: x86_64-apple-darwin17.0
(64-bit) Running under: macOS Big Sur … 10.16</p>
<p>Matrix products: default BLAS:
/Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK:
/Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</p>
<p>locale: [1]
en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</p>
<p>attached base packages: [1] grid stats4 stats graphics grDevices
utils datasets [8] methods base</p>
<p>other attached packages: [1] factoextra_1.0.7
org.Mm.eg.db_3.16.0<br />
[3] AnnotationDbi_1.60.0 circlize_0.4.15<br />
[5] GSVA_1.46.0 clusterProfiler_4.6.0<br />
[7] ComplexHeatmap_2.14.0 miloR_1.6.0<br />
[9] edgeR_3.40.2 limma_3.54.1<br />
[11] speckle_0.0.3 scDblFinder_1.13.8<br />
[13] SingleR_2.0.0 celldex_1.8.0<br />
[15] bluster_1.8.0 batchelor_1.14.1<br />
[17] BiocSingular_1.14.0 scran_1.26.2<br />
[19] SeuratObject_4.1.3 Seurat_4.3.0<br />
[21] DropletUtils_1.18.1 scater_1.26.1<br />
[23] scuttle_1.8.4 SingleCellExperiment_1.20.0 [25]
SummarizedExperiment_1.28.0 Biobase_2.58.0<br />
[27] GenomicRanges_1.50.2 GenomeInfoDb_1.34.9<br />
[29] IRanges_2.32.0 S4Vectors_0.36.1<br />
[31] BiocGenerics_0.44.0 MatrixGenerics_1.10.0<br />
[33] matrixStats_1.0.0 magrittr_2.0.3<br />
[35] openxlsx_4.2.5.2 patchwork_1.1.3<br />
[37] pheatmap_1.0.12 ggpubr_0.6.0<br />
[39] RColorBrewer_1.1-3 arsenal_3.6.3<br />
[41] ggbeeswarm_0.7.2 colorblindr_0.1.0<br />
[43] colorspace_2.1-1 cowplot_1.1.1<br />
[45] kableExtra_1.3.4 knitr_1.44<br />
[47] DT_0.30 foreach_1.5.2<br />
[49] showtext_0.9-6 showtextdb_3.0<br />
[51] sysfonts_0.8.8 lubridate_1.9.3<br />
[53] forcats_1.0.0 stringr_1.5.1<br />
[55] dplyr_1.1.4 purrr_1.0.2<br />
[57] readr_2.1.4 tidyr_1.3.1<br />
[59] tibble_3.2.1 ggplot2_3.5.0<br />
[61] tidyverse_2.0.0 workflowr_1.7.1</p>
<p>loaded via a namespace (and not attached): [1] rsvd_1.0.5
ica_1.0-3<br />
[3] svglite_2.1.2 ps_1.7.5<br />
[5] Rsamtools_2.14.0 lmtest_0.9-40<br />
[7] rprojroot_2.0.3 crayon_1.5.2<br />
[9] MASS_7.3-60 rhdf5filters_1.10.0<br />
[11] nlme_3.1-163 backports_1.4.1<br />
[13] GOSemSim_2.24.0 rlang_1.1.4<br />
[15] HDO.db_0.99.1 XVector_0.38.0<br />
[17] ROCR_1.0-11 irlba_2.3.5.1<br />
[19] callr_3.7.3 filelock_1.0.2<br />
[21] xgboost_1.7.5.1 BiocParallel_1.32.5<br />
[23] rjson_0.2.21 bit64_4.0.5<br />
[25] glue_1.8.0 sctransform_0.4.1<br />
[27] parallel_4.2.2 processx_3.8.2<br />
[29] vipor_0.4.5 spatstat.sparse_3.0-2<br />
[31] DOSE_3.24.2 spatstat.geom_3.2-7<br />
[33] tidyselect_1.2.1 fitdistrplus_1.1-11<br />
[35] XML_3.99-0.14 zoo_1.8-12<br />
[37] GenomicAlignments_1.34.0 xtable_1.8-4<br />
[39] evaluate_0.22 cli_3.6.3<br />
[41] zlibbioc_1.44.0 rstudioapi_0.15.0<br />
[43] miniUI_0.1.1.1 sp_2.1-1<br />
[45] whisker_0.4.1 bslib_0.5.1<br />
[47] fastmatch_1.1-4 treeio_1.22.0<br />
[49] shiny_1.7.5.1 xfun_0.40<br />
[51] clue_0.3-65 gson_0.1.0<br />
[53] cluster_2.1.4 tidygraph_1.2.3<br />
[55] KEGGREST_1.38.0 interactiveDisplayBase_1.36.0 [57] ggrepel_0.9.4
ape_5.7-1<br />
[59] listenv_0.9.0 Biostrings_2.66.0<br />
[61] png_0.1-8 future_1.33.0<br />
[63] withr_3.0.1 ggforce_0.4.1<br />
[65] bitops_1.0-7 plyr_1.8.9<br />
[67] GSEABase_1.60.0 dqrng_0.3.1<br />
[69] pillar_1.9.0 GlobalOptions_0.1.2<br />
[71] cachem_1.0.8 fs_1.6.4<br />
[73] GetoptLong_1.0.5 DelayedMatrixStats_1.20.0<br />
[75] vctrs_0.6.5 ellipsis_0.3.2<br />
[77] generics_0.1.3 tools_4.2.2<br />
[79] beeswarm_0.4.0 tweenr_2.0.2<br />
[81] munsell_0.5.1 fgsea_1.24.0<br />
[83] DelayedArray_0.24.0 fastmap_1.1.1<br />
[85] compiler_4.2.2 abind_1.4-5<br />
[87] httpuv_1.6.8 rtracklayer_1.58.0<br />
[89] ExperimentHub_2.6.0 plotly_4.10.3<br />
[91] GenomeInfoDbData_1.2.9 gridExtra_2.3<br />
[93] lattice_0.20-45 deldir_1.0-9<br />
[95] utf8_1.2.4 later_1.3.1<br />
[97] BiocFileCache_2.6.0 jsonlite_1.8.7<br />
[99] scales_1.3.0 graph_1.76.0<br />
[101] ScaledMatrix_1.6.0 tidytree_0.4.2<br />
[103] pbapply_1.7-2 carData_3.0-5<br />
[105] sparseMatrixStats_1.10.0 lazyeval_0.2.2<br />
[107] promises_1.2.1 doParallel_1.0.17<br />
[109] car_3.1-2 R.utils_2.12.2<br />
[111] goftest_1.2-3 spatstat.utils_3.0-3<br />
[113] reticulate_1.34.0 rmarkdown_2.25<br />
[115] statmod_1.5.0 webshot_0.5.5<br />
[117] Rtsne_0.16 downloader_0.4<br />
[119] uwot_0.1.16 igraph_1.5.1<br />
[121] HDF5Array_1.26.0 survival_3.5-7<br />
[123] ResidualMatrix_1.8.0 yaml_2.3.7<br />
[125] systemfonts_1.1.0 htmltools_0.5.6.1<br />
[127] memoise_2.0.1 BiocIO_1.8.0<br />
[129] locfit_1.5-9.8 here_1.0.1<br />
[131] graphlayouts_1.0.1 viridisLite_0.4.2<br />
[133] digest_0.6.37 mime_0.12<br />
[135] rappdirs_0.3.3 RSQLite_2.3.1<br />
[137] yulab.utils_0.1.7 future.apply_1.11.0<br />
[139] data.table_1.14.8 blob_1.2.4<br />
[141] R.oo_1.25.0 labeling_0.4.3<br />
[143] splines_4.2.2 Rhdf5lib_1.20.0<br />
[145] Cairo_1.6-1 AnnotationHub_3.6.0<br />
[147] RCurl_1.98-1.12 broom_1.0.5<br />
[149] hms_1.1.3 rhdf5_2.42.0<br />
[151] BiocManager_1.30.22 shape_1.4.6<br />
[153] aplot_0.2.2 sass_0.4.7<br />
[155] Rcpp_1.0.13 RANN_2.6.1<br />
[157] enrichplot_1.18.3 fansi_1.0.6<br />
[159] tzdb_0.4.0 parallelly_1.36.0<br />
[161] R6_2.5.1 ggridges_0.5.4<br />
[163] lifecycle_1.0.4 zip_2.3.0<br />
[165] curl_5.1.0 ggsignif_0.6.4<br />
[167] leiden_0.4.3 jquerylib_0.1.4<br />
[169] Matrix_1.5-3 qvalue_2.30.0<br />
[171] RcppAnnoy_0.0.21 org.Hs.eg.db_3.16.0<br />
[173] iterators_1.0.14 spatstat.explore_3.2-3<br />
[175] htmlwidgets_1.6.2 beachmat_2.14.0<br />
[177] polyclip_1.10-6 crosstalk_1.2.0<br />
[179] shadowtext_0.1.2 gridGraphics_0.5-1<br />
[181] timechange_0.2.0 rvest_1.0.3<br />
[183] globals_0.16.2 spatstat.random_3.2-1<br />
[185] progressr_0.14.0 codetools_0.2-19<br />
[187] GO.db_3.16.0 metapod_1.6.0<br />
[189] gtools_3.9.4 getPass_0.2-2<br />
[191] dbplyr_2.3.4 R.methodsS3_1.8.2<br />
[193] gtable_0.3.5 DBI_1.1.3<br />
[195] git2r_0.32.0 highr_0.10<br />
[197] ggfun_0.1.3 tensor_1.5<br />
[199] httr_1.4.7 KernSmooth_2.23-22<br />
[201] stringi_1.8.4 farver_2.1.2<br />
[203] reshape2_1.4.4 annotate_1.76.0<br />
[205] viridis_0.6.4 magick_2.7.3<br />
[207] ggtree_3.4.4 xml2_1.3.5<br />
[209] BiocNeighbors_1.16.0 restfulr_0.0.15<br />
[211] ggplotify_0.1.2 scattermore_1.2<br />
[213] BiocVersion_3.16.0 bit_4.0.5<br />
[215] scatterpie_0.2.1 spatstat.data_3.0-3<br />
[217] ggraph_2.1.0 pkgconfig_2.0.3<br />
[219] rstatix_0.7.2</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
