<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Castro-Giner" />

<meta name="date" content="2025-02-07" />

<title>10x RNA-seq analysis of MVT1 cancer and microenvironment in tumor and mets</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.30/datatables.js"></script>
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.13.4/jszip.min.js"></script>
<link href="site_libs/dt-ext-buttons-1.13.4/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.13.4/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.print.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">met-paths</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheAcetoLab/sznurkowska-met-paths">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">10x RNA-seq analysis of MVT1 cancer and
microenvironment in tumor and mets</h1>
<h3 class="subtitle">Analysis of cancer cells in primary tumor, lung and
liver metastases</h3>
<h4 class="author">Francesc Castro-Giner</h4>
<h4 class="date">February 07, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-07
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>sznurkowska-met-paths/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250206code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250206)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250206code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250206)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsznurkowskametpathstree542194747834f4d6d18783e043df241c0a9a743btargetblank5421947a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/tree/542194747834f4d6d18783e043df241c0a9a743b" target="_blank">5421947</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsznurkowskametpathstree542194747834f4d6d18783e043df241c0a9a743btargetblank5421947a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/tree/542194747834f4d6d18783e043df241c0a9a743b" target="_blank">5421947</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    data/patients/
    Ignored:    data/resources/
    Ignored:    data/rnaseq/
    Ignored:    data/seer/
    Ignored:    output/p26532_o26674/
    Ignored:    output/p26532_o28268/
    Ignored:    output/p26532_o28268_o34980/
    Ignored:    output/p26532_o34980/

Untracked files:
    Untracked:  analysis/br16-10x_rnaseq-cancer_cells-main.Rmd
    Untracked:  analysis/br16-10x_rnaseq-qc.Rmd
    Untracked:  analysis/mvt1-tumor-10x_rnaseq-pll-lr-cellphonedb.Rmd
    Untracked:  analysis/mvt1-tumor-10x_rnaseq-pll-lr-combined.Rmd
    Untracked:  analysis/mvt1-tumor-10x_rnaseq-pll-lr-lrdb.Rmd
    Untracked:  analysis/mvt1-tumor-10x_rnaseq-pll-tme-main.Rmd
    Untracked:  analysis/mvt1-tumor-10x_rnaseq-pll-tme-myeloid.Rmd
    Untracked:  analysis/mvt1-tumor-10x_rnaseq-qc.Rmd
    Untracked:  analysis/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd
    Untracked:  analysis/mvt1-tumor-nsg-normal-10x_rnaseq-pseudobulk-deg.Rmd
    Untracked:  analysis/nsg-normal-10x_rnaseq-main.Rmd
    Untracked:  analysis/nsg-normal-10x_rnaseq-qc.Rmd
    Untracked:  analysis/templates/
    Untracked:  code/R-functions/
    Untracked:  configuration/
    Untracked:  workflowr_update.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   sznurkowska-met-paths.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd</code>)
and HTML
(<code>docs/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/542194747834f4d6d18783e043df241c0a9a743b/analysis/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd" target="_blank">5421947</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
<td>
Add mvt1 cancer cells analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/TheAcetoLab/sznurkowska-met-paths/39b365c03bef848fb12a4a5aee47a955aecb169a/docs/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.html" target="_blank">39b365c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/9d6ed46f16dd51295783ddf80945aa0f40166c4a/analysis/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd" target="_blank">9d6ed46</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
<td>
Add mvt1 cancel cells analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries-additional-functions-and-data"
class="section level2" number="1">
<h2><span class="header-section-number">1</span> Load libraries,
additional functions and data</h2>
<p>Setup environment</p>
<pre class="r"><code>knitr::opts_chunk$set(results=&#39;asis&#39;, echo=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.align = &#39;center&#39;, fig.width = 3.5, fig.asp = 0.618, dpi = 600, dev = c(&quot;png&quot;, &quot;pdf&quot;), fig.showtext = FALSE)

options(stringsAsFactors = FALSE)

use_seed &lt;- 1100101
set.seed(use_seed)

if(!dir.exists(params$output_dir))
  dir.create(params$output_dir, recursive = TRUE, showWarnings = FALSE)</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(tidyverse)
library(showtext)
library(foreach)
library(DT)
library(knitr)
library(kableExtra)
library(cowplot)
library(colorblindr)
library(ggbeeswarm)
library(arsenal)
library(RColorBrewer)
library(ggpubr)
library(pheatmap)
library(patchwork)
library(openxlsx)
library(magrittr)

library(scater)
library(DropletUtils)
library(Seurat)
library(scran)
library(BiocSingular)
library(batchelor)
library(bluster)
library(scDblFinder)
library(speckle)
library(miloR)
library(ComplexHeatmap)
library(clusterProfiler)
library(GSVA)
library(circlize)
library(corrplot)
library(factoextra)</code></pre>
<p>Load ggplot theme</p>
<pre class="r"><code>source(&quot;./configuration/rmarkdown/ggplot_theme.R&quot;)
source(&quot;./configuration/rmarkdown/color_palettes.R&quot;)</code></pre>
<p>Load custom functions</p>
<pre class="r"><code>source(&#39;./code/R-functions/gse_omnibus.r&#39;)
source(&#39;./code/R-functions/gse_report.r&#39;)
clean_msigdb_names &lt;- function(x) x %&gt;% gsub(&#39;REACTOME_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;WP_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;BIOCARTA_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;KEGG_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;PID_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;GOBP_&#39;, &#39;&#39;, .) %&gt;% gsub(&#39;_&#39;, &#39; &#39;, .)
source(&#39;./code/R-functions/pairwise_terms_jc.r&#39;)</code></pre>
<p>Clean files generated in previous runs</p>
<pre class="r"><code>rmd_file &lt;- current_input()
if(!is.null(rmd_file)) {
  figures_dir &lt;- file.path(&#39;./docs/figure&#39;,rmd_file)
  if(dir.exists(figures_dir)) {
    unlink(file.path(figures_dir, &quot;*&quot;))
  }
}</code></pre>
</div>
<div id="data-processing" class="section level2" number="2">
<h2><span class="header-section-number">2</span> Data processing</h2>
<div id="batch-integration" class="section level3" number="2.1">
<h3><span class="header-section-number">2.1</span> Batch
integration</h3>
<p>Here we prepare the data for batch correction (next section). First,
we combine cells from all SCE objects into a single SCE and return
pre-computed per-gene variance at batch level. Then we adjust for batch
level library size. Returned object will contain log-normalized
expression values after adjusting the size factors for systematic
differences in coverage between</p>
<pre class="r"><code># File paths to SCE files
sce_files &lt;- system(paste(&quot;find&quot;, params$cellranger_count_dir, &quot;-maxdepth 2 -type f -regex &#39;.*/sce_normalized[.]rds$&#39;&quot;), intern = TRUE) %&gt;% sort

# Define list of anatomical sites to keep
keep_sites &lt;- c(&#39;liver&#39;, &#39;primary_tumor&#39;, &#39;lung&#39;)

# Combine all SCE files into sce_comb and generate an combine gene  with pre-computed gene variance 
if(exists(&#39;sce_comb&#39;)) 
  rm(sce_comb)
sce_file &lt;-  sce_files[7]
precomputed_geneVar &lt;- foreach(sce_file = sce_files) %do% {
  use_sce &lt;- readRDS(sce_file)
  # Select cancer cells
  use_sce &lt;- use_sce[,use_sce$cell_type == &#39;cancer_cell&#39;]
  # Remove cells from anatomical sites not selected
  use_sce &lt;- use_sce[,use_sce$site %in% keep_sites]
   
  use_sce$cluster_id_batch &lt;- as.character(use_sce$cluster_id_batch)
  if(ncol(use_sce) == 0) {
    NULL
  } else {
    # Combine sce
    if(!exists(&#39;sce_comb&#39;)) {
      sce_comb &lt;- use_sce
    } else {
      new_counts &lt;- cbind(counts(sce_comb), counts(use_sce))
      new_logcounts &lt;- cbind(logcounts(sce_comb), logcounts(use_sce))
      new_colData &lt;- rbind(colData(sce_comb), colData(use_sce))
      sce_comb &lt;- SingleCellExperiment(
        assays = list(counts = new_counts, logcounts = new_logcounts),
        colData = new_colData,
        rowData = rowData(sce_comb)
      )
    }
    metadata(use_sce)$dec
  }
}
metadata(sce_comb)$precomputed_geneVar &lt;- compact(precomputed_geneVar)
rm(use_sce)

# Modify coldata and rownames
sce_comb$batch &lt;-  gsub(&quot;_.*&quot;, &quot;&quot;, sce_comb$Sample)
sce_comb$cluster_id_batch &lt;-  paste(sce_comb$batch, sce_comb$cluster_id_batch, sep = &#39;_&#39;) %&gt;% as.factor()

rownames(sce_comb) &lt;- make.names(rowData(sce_comb)$gene_name, unique = TRUE)

# Rescale each batch to adjust for differences in sequencing depth between batches. The multiBatchNorm() function recomputes log-normalized expression values after adjusting the size factors for systematic differences in coverage between SingleCellExperiment objects. (Size factors only remove biases between cells within a single batch.) This improves the quality of the correction by removing one aspect of the technical differences between batches.
sce_comb &lt;- multiBatchNorm(sce_comb, batch = sce_comb$Sample, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))

# Identifying a set of HVGs using stats from all batches.
combined_dec &lt;- combineVar(metadata(sce_comb)$precomputed_geneVar)
# combined_hvg &lt;- getTopHVGs(combined_dec, n=5000, row.names = FALSE)
chosen_hvgs &lt;- combined_dec$bio &gt; 0 &amp; !rowData(sce_comb)$is.mito &amp; !rowData(sce_comb)$is.ribo

# Dimensionality reduction
sce_comb &lt;- runPCA(sce_comb, subset_row=chosen_hvgs, BSPARAM=BiocSingular::RandomParam(), BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
sce_comb$cluster_id  &lt;- clusterCells(sce_comb, use.dimred=&quot;PCA&quot;, BLUSPARAM=SNNGraphParam(k=25, type=&quot;rank&quot;, cluster.fun=&quot;louvain&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)))
tab &lt;- table(Cluster=sce_comb$cluster_id, Batch=sce_comb$Sample)
sce_comb &lt;- runTSNE(sce_comb, dimred=&quot;PCA&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
sce_comb &lt;- runUMAP(sce_comb, dimred=&quot;PCA&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))

saveRDS(sce_comb, file.path(params$output_dir, &#39;sce_integrated.rds&#39;))</code></pre>
</div>
<div id="additional-doublet-detection" class="section level3"
number="2.2">
<h3><span class="header-section-number">2.2</span> Additional doublet
detection</h3>
<p><a href="https://github.com/plger/scDblFinder">scDblFinder</a> : We
suggest using the cluster-based approach when the datasets are
segregated into clear clusters, and the random one for the rest
(e.g. developmental trajectories).</p>
<pre class="r"><code>sce_comb &lt;- scDblFinder(
  sce_comb, 
  samples =sce_comb$Sample, 
  BPPARAM=BiocParallel::MulticoreParam(params$ncores/2, RNGseed=use_seed))

saveRDS(sce_comb, file.path(params$output_dir, &#39;sce_integrated.rds&#39;))</code></pre>
</div>
<div id="batch-correction-and-dimensionality-reduction"
class="section level3" number="2.3">
<h3><span class="header-section-number">2.3</span> Batch correction and
dimensionality reduction</h3>
<p>Here, we actually perform the correction itself. We also perform the
clustering.</p>
<pre class="r"><code>if(!exists(&#39;sce_comb&#39;))
  sce_comb &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated.rds&#39;))

# Remove previous clustering
sce_comb$cluster_id &lt;- NULL
sce_comb$cluster_id_batch &lt;- NULL

# Remove genes for HVG
mito_genes &lt;- rownames(sce_comb)[rowData(sce_comb)$is.mito]
ribo_genes &lt;- rownames(sce_comb)[rowData(sce_comb)$is.ribo]
use_genes &lt;- rownames(sce_comb)[!rownames(sce_comb) %in% c(mito_genes, ribo_genes)]

# Select HVG
dec &lt;- modelGeneVar(sce_comb[use_genes,], block = sce_comb$batch) 
chosen_hvgs &lt;- getTopHVGs(dec, fdr.threshold=0.05)

# MNN correction
sce_mnn &lt;- fastMNN(sce_comb, 
                       batch = sce_comb$batch, 
                       auto.merge = TRUE, 
                       subset.row=chosen_hvgs, 
                       BSPARAM=RandomParam(deferred=TRUE), 
                       BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed),
                       d=30, k=20)
colData(sce_mnn) &lt;- colData(sce_comb)
reducedDim(sce_comb, &#39;corrected&#39;) &lt;- reducedDim(sce_mnn, &#39;corrected&#39;)


# Dimensionality reduction
sce_mnn &lt;- runUMAP(
  sce_mnn, dimred=&quot;corrected&quot;, 
  BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)
)
reducedDim(sce_comb, &#39;UMAP_corrected&#39;) &lt;- reducedDim(sce_mnn, &#39;UMAP&#39;)


# MNN-related diagnostic examining the variance in the differences in expression between MNN pairs.
metadata(sce_mnn)$mnnDeltaVariance &lt;- mnnDeltaVariance(
  sce_comb,
  pairs=metadata(sce_mnn)$merge.info$pairs, 
  BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)
  )

# Save SCE objects
saveRDS(sce_mnn, file.path(params$output_dir, &#39;sce_mnn.rds&#39;))
saveRDS(sce_comb, file.path(params$output_dir, &#39;sce_integrated.rds&#39;))</code></pre>
</div>
<div id="clustering" class="section level3" number="2.4">
<h3><span class="header-section-number">2.4</span> Clustering</h3>
<div id="clustering-parameter-sweeps" class="section level4"
number="2.4.1">
<h4><span class="header-section-number">2.4.1</span> Clustering
parameter sweeps</h4>
<pre class="r"><code>if(!exists(&#39;sce_comb&#39;))
  sce_comb &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated.rds&#39;))
if(!exists(&#39;sce_mnn&#39;))
  sce_mnn &lt;- readRDS(file.path(params$output_dir, &#39;sce_mnn.rds&#39;))

# ClusterSweep
csweep &lt;- clusterSweep(
  reducedDim(sce_mnn, &quot;corrected&quot;), 
  SNNGraphParam(), 
  k=as.integer(c(5, 10, 15, 20, 25, 30, 35, 40)),
  type=&quot;rank&quot;,
  cluster.fun=c(&quot;walktrap&quot;),
  BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)
  )

saveRDS(csweep, file.path(params$output_dir, &#39;clusterSweep.rds&#39;))</code></pre>
<p>Evaluating <a
href="https://bioconductor.org/books/release/OSCA.advanced/clustering-redux.html#clustering-parameter-sweeps">clustering
parameters sweep</a></p>
<pre class="r"><code>if(!exists(&#39;sce_mnn&#39;))
  sce_mnn &lt;- readRDS(file.path(params$output_dir, &#39;sce_mnn.rds&#39;))
if(!exists(&#39;csweep&#39;))
  csweep &lt;- readRDS(file.path(params$output_dir, &#39;clusterSweep.rds&#39;))

df &lt;- as.data.frame(csweep$parameters)
df$num.clusters &lt;- vapply(as.list(csweep$clusters), function(cluster) { 
    length(unique(cluster))
}, 0L)

all.sil &lt;- lapply(as.list(csweep$clusters), function(cluster) {
    sil &lt;- approxSilhouette(reducedDim(sce_mnn, &#39;corrected&#39;), cluster)
    mean(sil$width)
})
df$silhouette &lt;- unlist(all.sil)

all.pur &lt;- foreach(cname=colnames(csweep$clusters)) %do% {
  use_k &lt;- gsub(&quot;k.&quot;, &quot;&quot;, cname) %&gt;% gsub(&quot;_.*&quot;, &quot;&quot;, .) %&gt;% as.numeric
  pur &lt;- neighborPurity(reducedDim(sce_mnn, &#39;corrected&#39;), csweep$clusters[[cname]], k=use_k, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
  mean(pur$width)
}
df$purity &lt;- unlist(all.pur)

all.wcss &lt;- lapply(as.list(csweep$clusters), function(cluster) {
    sum(clusterRMSD(reducedDim(sce_mnn, &#39;corrected&#39;), cluster, sum=TRUE), na.rm=TRUE)
})
df$wcss &lt;- unlist(all.wcss)

gridExtra::grid.arrange(
    ggplot(df, aes(x=k, y=num.clusters)) + 
        geom_line(lwd=2) ,
    ggplot(df, aes(x=k, y=silhouette)) + 
        geom_line(lwd=2),
    ggplot(df, aes(x=k, y=purity)) + 
        geom_line(lwd=2),
    ggplot(df, aes(x=k, y=wcss)) + 
        geom_line(lwd=2),
    ncol=2
)</code></pre>
</div>
<div id="clustering-with-best-parameters" class="section level4"
number="2.4.2">
<h4><span class="header-section-number">2.4.2</span> Clustering with
best parameters</h4>
<pre class="r"><code>if(!exists(&#39;sce_comb&#39;))
  sce_comb &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated.rds&#39;))
if(!exists(&#39;sce_mnn&#39;))
  sce_mnn &lt;- readRDS(file.path(params$output_dir, &#39;sce_mnn.rds&#39;))
 
# Clustering using corrected values
sce_mnn$cluster_mnn  &lt;- clusterCells(
  sce_mnn,
  use.dimred=&quot;corrected&quot;,
  BLUSPARAM=SNNGraphParam(k=20, type=&quot;rank&quot;, cluster.fun=&quot;walktrap&quot;,
                          BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
)
sce_comb$cluster_mnn &lt;- sce_mnn$cluster_mnn

# Save SCE objects
saveRDS(sce_mnn, file.path(params$output_dir, &#39;sce_mnn.rds&#39;))
saveRDS(sce_comb, file.path(params$output_dir, &#39;sce_integrated.rds&#39;))</code></pre>
</div>
</div>
<div id="markers-detection" class="section level3" number="2.5">
<h3><span class="header-section-number">2.5</span> Markers
detection</h3>
<p>We identify the genes that drive separation between clusters and cell
types. These marker genes allow us to assign biological meaning to each
cluster based on their functional annotation. We added batch as a
blocking factor to the model. The block argument works for all effect
sizes shown above and is robust to differences in the log-fold changes
or variance between batches. However, it assumes that each pair of
clusters is present in at least one batch. In scenarios where cells from
two clusters never co-occur in the same batch, the associated pairwise
comparison will be impossible and is ignored during calculation of
summary statistics.</p>
<pre class="r"><code>marker_info &lt;- list()

if(!exists(&#39;sce_comb&#39;))
  sce_comb &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated.rds&#39;))


# Cluster markers
marker_info$cluster_markers &lt;- scoreMarkers(
  sce_comb, 
  groups = sce_comb$cluster_mnn %&gt;% as.character, 
  block = sce_comb$batch,
  lfc=1, 
  full.stats=TRUE,
  row.data=rowData(sce_comb)[,c(&#39;gene_name&#39;, &#39;is.mito&#39;, &#39;is.ribo&#39;),drop=FALSE],
  BPPARAM=BiocParallel::MulticoreParam(1, RNGseed=use_seed)
)


# Site markers
marker_info$site_markers &lt;- scoreMarkers(
  sce_comb, 
  groups = sce_comb$site, 
  block = sce_comb$batch,
  lfc=1, 
  full.stats=TRUE,
  row.data=rowData(sce_comb)[,c(&#39;gene_name&#39;, &#39;is.mito&#39;, &#39;is.ribo&#39;),drop=FALSE],
  BPPARAM=BiocParallel::MulticoreParam(1, RNGseed=use_seed)
)

saveRDS(marker_info, file.path(params$output_dir, &#39;scoreMarkers.rds&#39;))</code></pre>
</div>
<div id="functional-enrichment" class="section level3" number="2.6">
<h3><span class="header-section-number">2.6</span> Functional
enrichment</h3>
<p>Run functional enrichment based on results from cluster markers
detection</p>
<pre class="r"><code># Load markers
score_markers &lt;- readRDS(file.path(params$output_dir, &#39;scoreMarkers.rds&#39;))
marker_info &lt;- score_markers$cluster_markers

# Load MSigDB files
gmt_files_symbols &lt;- list(
  msigdb.m2.cp = &#39;./data/resources/MSigDB/v2022.1.Mm/m2.cp.v2022.1.Mm.symbols.gmt&#39;,
  msigdb.m2.cp.reactome = &#39;./data/resources/MSigDB/v2022.1.Mm/m2.cp.reactome.v2022.1.Mm.symbols.gmt&#39;
)

# Load mouse annotation for Go 
library(org.Mm.eg.db)

# Run ORA using mean.AUC
i &lt;- marker_info[[1]]
gse_res&lt;- foreach(i=marker_info) %do% {
  x &lt;- i %&gt;% data.frame %&gt;% rownames_to_column(&#39;features&#39;) %&gt;% 
    mutate(
      p = 0
    )
  gse_omnibus(
    feature_names = x$features,
    p = x$p,
    fc = x$mean.AUC,
    fc_thrs = 0.25,
    return_sets = c(&#39;abs&#39;),

    gmt_files = gmt_files_symbols,

    annot_db = &#39;org.Mm.eg.db&#39;,
    organism = &#39;mmusculus&#39;,
    kegg_organism = &#39;mmu&#39;,
    reactome_organism = &#39;mouse&#39;,

    run_all_ora = FALSE,
    run_all_gsea = FALSE,
    run_GSEA = FALSE,
    run_enricher = TRUE,
    run_enrichGO = TRUE,

    args_gse = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1),
    args_enricher = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1, qvalueCutoff = 1),

    go_simplify_p = 0.1,
    go_simplify_cutoff = 0.4
    )
} 
names(gse_res) &lt;- names(marker_info)
saveRDS(gse_res, file.path(params$output_dir, &#39;scoreMarkers_cluster_gse_auc.rds&#39;))


# # Run GSEA using Cohen&#39;s logFC
# i &lt;- marker_info[[1]]
# gse_res&lt;- foreach(i=marker_info) %do% {
#   x &lt;- i %&gt;% data.frame %&gt;% rownames_to_column(&#39;features&#39;) %&gt;%
#     mutate(
#       p = 0,
#       mean.logFC.cohen = ifelse(mean.logFC.cohen &lt;= -5, -5, mean.logFC.cohen),
#       mean.logFC.cohen = ifelse(mean.logFC.cohen &gt;= 5, 5, mean.logFC.cohen)
#     )
# 
#   gse_omnibus(
#     feature_names = x$features,
#     p = x$p,
#     fc = x$mean.logFC.cohen,
# 
#     gmt_files = gmt_files_symbols,
# 
#     annot_db = &#39;org.Mm.eg.db&#39;,
#     organism = &#39;mmusculus&#39;,
#     kegg_organism = &#39;mmu&#39;,
#     reactome_organism = &#39;mouse&#39;,
# 
#     run_all_ora = FALSE,
#     run_all_gsea = FALSE,
#     run_GSEA = TRUE,
#     run_gseGO = TRUE,
# 
#     args_gse = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1)
#     )
# }
# names(gse_res) &lt;- names(marker_info)
# saveRDS(gse_res, file.path(params$output_dir, &#39;scoreMarkers_cluster_gse_cohen.rds&#39;))</code></pre>
<p>Run functional enrichment based on results from site markers
detection</p>
<pre class="r"><code># Load markers
score_markers &lt;- readRDS(file.path(params$output_dir, &#39;scoreMarkers.rds&#39;))
marker_info &lt;- score_markers$site_markers

# Load MSigDB files
gmt_files_symbols &lt;- list(
  msigdb.m2.cp = &#39;./data/resources/MSigDB/v2022.1.Mm/m2.cp.v2022.1.Mm.symbols.gmt&#39;,
  msigdb.m2.cp.reactome = &#39;./data/resources/MSigDB/v2022.1.Mm/m2.cp.reactome.v2022.1.Mm.symbols.gmt&#39;
)

# Load mouse annotation for Go 
library(org.Mm.eg.db)

# Run ORA using mean.AUC
i &lt;- marker_info[[1]]
gse_res&lt;- foreach(i=marker_info) %do% {
  x &lt;- i %&gt;% data.frame %&gt;% rownames_to_column(&#39;features&#39;) %&gt;% 
    mutate(
      p = 0
    )
  gse_omnibus(
    feature_names = x$features,
    p = x$p,
    fc = x$mean.AUC,
    fc_thrs = 0.25,
    return_sets = c(&#39;abs&#39;),
    
    gmt_files = gmt_files_symbols, 

    annot_db = &#39;org.Mm.eg.db&#39;,
    organism = &#39;mmusculus&#39;,
    kegg_organism = &#39;mmu&#39;,
    reactome_organism = &#39;mouse&#39;,
    
    run_all_ora = FALSE,
    run_all_gsea = FALSE,
    run_GSEA = FALSE,
    run_enricher = TRUE,
    run_enrichGO = TRUE,

    args_gse = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1),
    args_enricher = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1, qvalueCutoff = 1),
    
    go_simplify_p = 0.1,
    go_simplify_cutoff = 0.4
    )
} 
names(gse_res) &lt;- names(marker_info)
saveRDS(gse_res, file.path(params$output_dir, &#39;scoreMarkers_site_gse_auc.rds&#39;))

# # Run GSEA using Cohen&#39;s logFC
# i &lt;- marker_info[[1]]
# gse_res&lt;- foreach(i=marker_info) %do% {
#   x &lt;- i %&gt;% data.frame %&gt;% rownames_to_column(&#39;features&#39;) %&gt;% 
#     mutate(
#       p = 0,
#       mean.logFC.cohen = ifelse(mean.logFC.cohen &lt;= -5, -5, mean.logFC.cohen),
#       mean.logFC.cohen = ifelse(mean.logFC.cohen &gt;= 5, 5, mean.logFC.cohen)
#     )
#   
#   gse_omnibus(
#     feature_names = x$features,
#     p = x$p,
#     fc = x$mean.logFC.cohen,
#     
#     gmt_files = gmt_files_symbols, 
# 
#     annot_db = &#39;org.Mm.eg.db&#39;,
#     organism = &#39;mmusculus&#39;,
#     kegg_organism = &#39;mmu&#39;,
#     reactome_organism = &#39;mouse&#39;,
#     
#     run_all_ora = FALSE,
#     run_all_gsea = FALSE,
#     run_GSEA = TRUE,
#     run_gseGO = TRUE,
# 
#     args_gse = list(minGSSize = 10, maxGSSize = 500, pvalueCutoff = 1)
#     )
# }
# names(gse_res) &lt;- names(marker_info)
# saveRDS(gse_res, file.path(params$output_dir, &#39;scoreMarkers_site_gse_cohen.rds&#39;))</code></pre>
</div>
<div id="load-data" class="section level3" number="2.7">
<h3><span class="header-section-number">2.7</span> Load data</h3>
<pre class="r"><code>sce_comb &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated.rds&#39;))
sce_mnn &lt;- readRDS(file.path(params$output_dir, &#39;sce_mnn.rds&#39;))
score_markers &lt;- readRDS(file.path(params$output_dir, &#39;scoreMarkers.rds&#39;))
gse_cl_auc &lt;- readRDS( file.path(params$output_dir, &#39;scoreMarkers_cluster_gse_auc.rds&#39;))
gse_site_auc &lt;- readRDS( file.path(params$output_dir, &#39;scoreMarkers_site_gse_auc.rds&#39;))

# Cluster MNN levels
cl_levels_sort &lt;- sce_comb$cluster_mnn %&gt;% as.character %&gt;% as.numeric %&gt;% unique %&gt;% sort %&gt;% as.character()
cl_levels_rev &lt;- sce_comb$cluster_mnn %&gt;% as.character %&gt;% as.numeric %&gt;% unique %&gt;% sort(decreasing = TRUE) %&gt;% as.character()
sce_comb$cluster_mnn %&lt;&gt;% as.character %&gt;% factor(levels = cl_levels_sort)
sce_comb$cluster_mnn_rev &lt;- sce_comb$cluster_mnn %&gt;% as.character %&gt;% factor(levels = cl_levels_rev)

# Selected MNN clusters to show
selected_clusters_to_show_list &lt;- list(
  `Primary tumor` = c(&#39;11&#39;, &#39;4&#39;),
  Liver = c(&#39;7&#39;, &#39;12&#39;),
  Lung = c(&#39;8&#39;, &#39;6&#39;)
)

selected_clusters_to_show &lt;- unlist(selected_clusters_to_show_list)

selected_clusters_to_df &lt;- rbind(
  c(site_caps = &#39;Primary tumor&#39;, cluster = &#39;11&#39;),
  c(site_caps = &#39;Primary tumor&#39;, cluster = &#39;4&#39;),
  c(site_caps = &#39;Liver&#39;, cluster = &#39;7&#39;),
  c(site_caps = &#39;Liver&#39;, cluster = &#39;12&#39;),
  c(site_caps = &#39;Lung&#39;, cluster = &#39;8&#39;),
  c(site_caps = &#39;Lung&#39;, cluster = &#39;6&#39;)
) %&gt;% 
  data.frame %&gt;% 
  mutate(
    site_caps = factor(site_caps, levels = c(&#39;Primary tumor&#39;, &#39;Liver&#39;, &#39;Lung&#39;)),
  )

site_caps_ord &lt;- c(&#39;Primary tumor&#39;, &#39;Liver&#39;, &#39;Lung&#39;)
site_ord &lt;- c(&#39;primary_tumor&#39;, &#39;liver&#39;, &#39;lung&#39;)

# Add new colData vars
colData(sce_comb) &lt;- colData(sce_comb) %&gt;% data.frame %&gt;% 
  mutate(
    site_caps = ifelse(site == &#39;liver&#39;, &#39;Liver&#39;, site),
    site_caps = ifelse(site == &#39;lung&#39;, &#39;Lung&#39;, site_caps),
    site_caps = ifelse(site == &#39;primary_tumor&#39;, &#39;Primary tumor&#39;, site_caps),
    site_caps = factor(site_caps, levels = c(&#39;Primary tumor&#39;, &#39;Liver&#39;, &#39;Lung&#39;)),
    site = factor(site, levels = c(&#39;primary_tumor&#39;, &#39;liver&#39;, &#39;lung&#39;))
  ) %&gt;% 
  DataFrame

# Coordinate colData between sce_comb and sce_mnn
colData(sce_mnn) &lt;- colData(sce_comb)
assay(sce_mnn, &#39;logcounts&#39;) &lt;- logcounts(sce_comb[rownames(sce_mnn),])
reducedDim(sce_comb, &#39;UMAP_corrected&#39;) &lt;- reducedDim(sce_mnn, &#39;UMAP&#39;)</code></pre>
</div>
</div>
<div id="data-description" class="section level2" number="3">
<h2><span class="header-section-number">3</span> Data description</h2>
<div id="site-distribution-per-batch" class="section level3"
number="3.1">
<h3><span class="header-section-number">3.1</span> Site distribution per
batch</h3>
<pre class="r"><code>table(sce_comb$Sample, sce_comb$site) %&gt;% 
  kbl(caption = &#39;Number of cells per batch and site.&#39;) %&gt;%
  kable_paper(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;), full_width = F)</code></pre>
<table class=" lightable-paper" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Number of cells per batch and site.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
primary_tumor
</th>
<th style="text-align:right;">
liver
</th>
<th style="text-align:right;">
lung
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
01_22_04_25_PT_c_Lung_c_Liver_c
</td>
<td style="text-align:right;">
4737
</td>
<td style="text-align:right;">
2795
</td>
<td style="text-align:right;">
2937
</td>
</tr>
<tr>
<td style="text-align:left;">
04_22_04_26_cancer_PT_Lung_Liver_BM_Brain
</td>
<td style="text-align:right;">
2718
</td>
<td style="text-align:right;">
3620
</td>
<td style="text-align:right;">
3395
</td>
</tr>
<tr>
<td style="text-align:left;">
06_22_04_28_cancer_PT_Lung_Liver_BM_Brain_Ovaries
</td>
<td style="text-align:right;">
2253
</td>
<td style="text-align:right;">
3027
</td>
<td style="text-align:right;">
2623
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>sites_selected &lt;- sce_comb$site %&gt;% table %&gt;% data.frame %&gt;% dplyr::rename(site = &quot;.&quot;) %&gt;% filter(Freq &gt;= 500) %&gt;% collect %&gt;% .[[&#39;site&#39;]] %&gt;% as.character</code></pre>
</div>
</div>
<div id="umap" class="section level2" number="4">
<h2><span class="header-section-number">4</span> UMAP</h2>
<div id="site" class="section level3" number="4.1">
<h3><span class="header-section-number">4.1</span> Site</h3>
<pre class="r"><code>plotUMAP(sce_mnn, colour_by=&quot;site_caps&quot;, point_alpha = 0.2, point_size = 0.05) + 
  scale_color_manual(values = site_palette) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 3))) +
  labs(
    color = &#39;&#39;,
    title = paste0(&#39;UMAP of all cancer cells (&#39;,scales::number(ncol(sce_mnn), big.mark = &quot;&#39;&quot;),&#39; cells)&#39;)
  ) +
  theme(
    plot.title = element_text(size = 8, face = &#39;plain&#39;),
    axis.line = element_line(linewidth = rel(0.5)),
    axis.text=element_blank(),
    axis.ticks=element_blank()#,
    # axis.line =element_blank()
  )</code></pre>
<p><img src="figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/final-figures-umap-site-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-final-figures-umap-site-1">
Past versions of final-figures-umap-site-1.png
</button>
</p>
<div id="fig-final-figures-umap-site-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/39b365c03bef848fb12a4a5aee47a955aecb169a/docs/figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/final-figures-umap-site-1.png" target="_blank">39b365c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster" class="section level3" number="4.2">
<h3><span class="header-section-number">4.2</span> Cluster</h3>
<pre class="r"><code>plotUMAP(sce_mnn, colour_by=&quot;cluster_mnn&quot;, text_by=&quot;cluster_mnn&quot;, point_alpha = 0.2, point_size = 0.05, text_size = 2) + 
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 3))) +
  labs(
    color = &#39;&#39;,
    title = paste0(&#39;UMAP of all cancer cells (&#39;,scales::number(ncol(sce_mnn), big.mark = &quot;&#39;&quot;),&#39; cells)&#39;)
  ) +
  theme(
    plot.title = element_text(size = 8, face = &#39;plain&#39;),
    axis.line = element_line(linewidth = rel(0.5)),
    axis.text=element_blank(),
    axis.ticks=element_blank()#,
    # axis.line =element_blank()
  )</code></pre>
<p><img src="figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/final-figures-umap-cluster-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-final-figures-umap-cluster-1">
Past versions of final-figures-umap-cluster-1.png
</button>
</p>
<div id="fig-final-figures-umap-cluster-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/39b365c03bef848fb12a4a5aee47a955aecb169a/docs/figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/final-figures-umap-cluster-1.png" target="_blank">39b365c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="differential-abundance-of-clusters" class="section level2"
number="5">
<h2><span class="header-section-number">5</span> Differential abundance
of clusters</h2>
<p>Differential abundance of clusters by site</p>
<div id="statistical-analysis" class="section level3" number="5.1">
<h3><span class="header-section-number">5.1</span> Statistical
analysis</h3>
<p>Differential abundance using <code>speckle::propeller</code>
package</p>
<pre class="r"><code>ct_label &lt;- &#39;cluster_mnn&#39;
use_data &lt;- colData(sce_comb) %&gt;% data.frame %&gt;% 
  mutate(
    sample = paste(mouse_id, site_caps, sep = &#39;_&#39;)
  )

# Run propeller Global
propeller_global &lt;- propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$site_caps) %&gt;% 
  rownames_to_column(&#39;cluster&#39;)

# Run pairwise propeller to obtain P-values
site_levels &lt;- use_data$site_caps %&gt;% unique
site_combn &lt;- combn(site_levels, 2, simplify = FALSE)
sc &lt;-  site_combn[[1]]
propeller_pairwise &lt;-foreach(sc = site_combn, .combine = rbind) %do% {
  cat(sc)
  use_data &lt;- colData(sce_comb) %&gt;% data.frame %&gt;% 
    filter(site_caps %in% sc) %&gt;% 
    mutate(
      sample = paste(mouse_id, site_caps, sep = &#39;_&#39;),
      group = ifelse(site_caps == sc[1], &#39;group1&#39;, &#39;group2&#39;)
    )
  propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$group) %&gt;% 
    mutate(
      group1 = sc[1],
      group2 = sc[2],
      FDR.sign = symnum(FDR, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001,0.001, 0.01, 0.05, 1), symbols = c(&quot;****&quot;, &quot;***&quot;, &quot;**&quot;, &quot;*&quot;, &quot;&quot;)),
      p.sign = symnum(P.Value, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c(&quot;****&quot;, &quot;***&quot;, &quot;**&quot;, &quot;*&quot;, &quot;&quot;))
    ) %&gt;% 
    dplyr::rename(cluster = BaselineProp.clusters)
}</code></pre>
<p>3 23 12 1</p>
<p>Table of results with pairwise comparison between sites for cluster
abundance.</p>
<pre class="r"><code>propeller_pairwise %&gt;%
  arrange(FDR) %&gt;% 
  datatable(., 
            rownames = FALSE, 
            filter = &#39;top&#39;, 
            caption = &#39;Pairwise comparison between sites for cluster abundance using speckle::propeller&#39;,
            extensions = &#39;Buttons&#39;, 
            options = list(
              dom = &#39;Blfrtip&#39;,
              buttons = c(&#39;csv&#39;, &#39;excel&#39;)
            )) %&gt;% 
  formatPercentage(names(propeller_pairwise)[2:4], digits = 3) %&gt;% 
  formatRound(names(propeller_pairwise)[5:6], digits = 3, interval = -1000) %&gt;% 
  formatSignif(c(&#39;P.Value&#39;, &#39;FDR&#39;), digits = 3)</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-262c826acf74d9caecbc" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-262c826acf74d9caecbc">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;1&quot;,&quot;2&quot;,&quot;3&quot;,&quot;4&quot;,&quot;5&quot;,&quot;6&quot;,&quot;7&quot;,&quot;8&quot;,&quot;9&quot;,&quot;10&quot;,&quot;11&quot;,&quot;12&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.000869707017448\" data-max=\"0.183759791122716\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.000312321092974\" data-max=\"0.293299665100307\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.000312321092974\" data-max=\"0.286104242445068\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.00179726540792\" data-max=\"39.1314316899209\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-9.77955643876308\" data-max=\"7.77393148814995\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"7.6316988e-08\" data-max=\"0.995857137698942\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"4.86474037e-07\" data-max=\"0.995857137698942\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Primary tumor&quot;,&quot;Liver&quot;,&quot;Lung&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Primary tumor&quot;,&quot;Liver&quot;,&quot;Lung&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Pairwise comparison between sites for cluster abundance using speckle::propeller<\/caption>","data":[["4","11","8","6","11","7","4","9","8","4","6","12","7","9","10","12","7","1","1","2","12","5","10","11","2","10","9","5","2","3","3","1","5","3","6","8"],[0.0983289817232376,0.1021932114882507,0.1523617981192586,0.07185954231668207,0.1053957027273214,0.1370757180156658,0.005598738924824699,0.1416187989556136,0.1496543963992927,0.1056636124953116,0.07046026898140706,0.07827676240208878,0.1581779637984454,0.152119166264802,0.02807694368536677,0.02620157530943578,0.02212934683598564,0.06649520441515297,0.07984334203655352,0.04156657963446475,0.101429580909931,0.1404576833179323,0.03712561830733271,0.000869707017448497,0.02496919037668113,0.01237597911227154,0.01820949067782791,0.1252210255585919,0.05674838288851443,0.1301842691743219,0.123613566950651,0.1269772245474806,0.1837597911227154,0.1037075718015666,0.005900783289817232,0.01535248041775457],[0.0007078247712742834,0.000312321092974768,0.2932996651003065,0.1426524711061757,0.001551773860621287,0.2861042424450679,0.0111562269218399,0.01129545857729071,0.2932996651003065,0.0111562269218399,0.1426524711061757,0.1467606678302349,0.03855255235029773,0.02506269302965999,0.05323474827556617,0.04754558136424235,0.03855255235029775,0.1156786145876277,0.1360094382614941,0.0691984326070815,0.04754558136424235,0.07803816679670487,0.05323474827556617,0.001551773860621287,0.03856124167612782,0.01985797523184477,0.02506269302965999,0.07803816679670486,0.03856124167612783,0.1546662649308302,0.1546662649308302,0.1156786145876277,0.1996494240622475,0.1078641722798698,0.005813970085990871,0.01642607275462905],[0.2121857792352096,0.1737757214924726,0.01642607275462912,0.005813970085990889,0.1737757214924726,0.007311366594306329,0.0007078247712742874,0.2412064406818472,0.01788310003081474,0.2121857792352096,0.006542261725068173,0.007388577859480831,0.2861042424450679,0.2412064406818472,0.005056413753790854,0.007388577859480831,0.007311366594306329,0.02726316488930342,0.02726316488930342,0.01369342860858962,0.1467606678302348,0.1996494240622476,0.01985797523184477,0.0003123210929747682,0.01369342860858962,0.005056413753790854,0.01129545857729071,0.178465541650851,0.06919843260708151,0.1078641722798698,0.1092282034782657,0.136009438261494,0.178465541650851,0.1092282034782657,0.006542261725068173,0.01788310003081474],[0.003335872808373525,0.001797265407920039,17.85573882945637,24.5361549846817,0.008929750642344504,39.13143168992092,15.7612835472761,0.04682900898234925,16.40094080975422,0.05257763721042366,21.80476371949017,19.86318214701026,0.1347500198557867,0.1039055713388592,10.52816301586385,6.435011211695239,5.272961197202211,4.2430368982221,4.988761899571563,5.053404416456713,0.3239667825662979,0.3908759925717281,2.68077423070795,4.968520844497212,2.816039925306881,3.927284474486885,2.218829174412452,0.4372730224268071,0.5572559987750592,1.433898408171393,1.415992024089326,0.8505190232844136,1.118700126732815,0.9875120971053297,0.8886789202751267,0.9185249048724741],[-9.779556438763077,-9.733815126620151,7.773931488149949,7.528045698200305,-8.103165756205227,6.122510292741914,5.439765422931941,-5.493129982761154,5.825537621456945,-5.81546002797096,5.694410685971986,5.086089924747784,-4.654349603150482,-4.569147463029586,4.050068151393076,3.728130275763893,3.392403448175806,3.005199016967331,3.019576764363666,2.888379621074923,-2.741692606680814,-2.73175469646885,2.380203029083436,2.231889496134817,2.20413638464385,2.219625896934875,1.967729778203539,-1.88851608890018,-1.297628720897025,0.9946067820985329,0.8761004271109479,-0.3755674723828257,0.25891401893557,0.1120228081214383,-0.0853746019885169,-0.005280835210014094],[7.631698805913867e-08,8.107900622886115e-08,2.294667449037646e-07,5.334542894149173e-07,1.729525950528496e-06,2.081715476297516e-05,3.471933619857881e-05,6.519271917365057e-05,5.522652581065556e-05,5.616652398104955e-05,6.886554864150484e-05,0.0001403090274779109,0.0001914478426694336,0.0005039542163921386,0.001330541433930317,0.00246085944939094,0.004705492170695225,0.009970168287648115,0.008738168897872958,0.01259970909213314,0.01283997261515865,0.01312057498057574,0.02841731879501904,0.03840783097792146,0.05194003961262817,0.04252629351548503,0.06363005824212475,0.08108240575229787,0.209703506432378,0.3322286998447596,0.396606280782584,0.7113324738444464,0.7992725989289156,0.9123124740468102,0.9331092863723522,0.9958571376989414],[4.864740373731669e-07,4.864740373731669e-07,2.753600938845175e-06,3.200725736489504e-06,2.075431140634195e-05,8.326861905190066e-05,0.0001388773447943152,0.0001955781575209517,0.0002065966459245145,0.0002065966459245145,0.0002065966459245145,0.0003367416659469861,0.0005743435280083007,0.001209490119341133,0.002661082867860634,0.004218616198955897,0.007058238256042838,0.01329355771686415,0.01747633779574592,0.02159950130079967,0.02624114996115149,0.02624114996115149,0.04871540364860406,0.05761174646688219,0.0623280475351538,0.06378944027322754,0.08484007765616633,0.08845353354796132,0.2516442077188537,0.3624313089215558,0.396606280782584,0.7113324738444464,0.9958571376989414,0.9958571376989414,0.9958571376989414,0.9958571376989414],["Liver","Liver","Lung","Lung","Lung","Liver","Lung","Liver","Lung","Lung","Lung","Liver","Lung","Lung","Lung","Lung","Lung","Lung","Liver","Liver","Lung","Lung","Lung","Lung","Lung","Liver","Lung","Lung","Lung","Lung","Lung","Lung","Liver","Liver","Liver","Liver"],["Primary tumor","Primary tumor","Liver","Liver","Primary tumor","Primary tumor","Liver","Primary tumor","Primary tumor","Primary tumor","Primary tumor","Primary tumor","Liver","Primary tumor","Primary tumor","Primary tumor","Primary tumor","Primary tumor","Primary tumor","Primary tumor","Liver","Liver","Liver","Liver","Primary tumor","Primary tumor","Liver","Primary tumor","Liver","Liver","Primary tumor","Liver","Primary tumor","Primary tumor","Primary tumor","Primary tumor"],["****","****","****","****","****","****","***","***","***","***","***","***","***","**","**","**","**","*","*","*","*","*","*","","","","","","","","","","","","",""],["****","****","****","****","****","****","****","****","****","****","****","***","***","***","**","**","**","**","**","*","*","*","*","*","","*","","","","","","","","","",""]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>cluster<\/th>\n      <th>BaselineProp.Freq<\/th>\n      <th>PropMean.group1<\/th>\n      <th>PropMean.group2<\/th>\n      <th>PropRatio<\/th>\n      <th>Tstatistic<\/th>\n      <th>P.Value<\/th>\n      <th>FDR<\/th>\n      <th>group1<\/th>\n      <th>group2<\/th>\n      <th>FDR.sign<\/th>\n      <th>p.sign<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":7,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, -1000, \",\", \".\", null);\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, -1000, \",\", \".\", null);\n  }"},{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[1,2,3,4,5,6,7]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render","options.columnDefs.5.render","options.columnDefs.6.render"],"jsHooks":[]}</script>
</div>
<div id="barplots-by-site-and-mouse" class="section level3"
number="5.2">
<h3><span class="header-section-number">5.2</span> Barplots by site and
mouse</h3>
<pre class="r"><code># Create data
use_data &lt;- colData(sce_comb) %&gt;% 
  data.frame %&gt;%
  dplyr::rename(&#39;celltype_label&#39; = ct_label) %&gt;% 
  group_by(site_caps, mouse_id, celltype_label) %&gt;% 
  summarise(freq = n()) %&gt;% 
  dplyr::rename(cell_type = celltype_label) %&gt;% 
  na.omit()

# Remove low abundant cell types and sites
cell_type_freq &lt;- use_data %&gt;% group_by(cell_type) %&gt;% summarise(totalfreq = sum(freq))
site_filtered &lt;- use_data %&gt;% group_by(site_caps) %&gt;% summarise(freq = sum(freq)) %&gt;% filter(freq &gt;= 0)
use_data &lt;- use_data %&gt;% 
  filter(site_caps %in% site_filtered$site_caps) %&gt;% 
  left_join(cell_type_freq) %&gt;% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data &lt;- use_data %&gt;% 
  arrange(site_caps, cell_type)%&gt;% 
  group_by(site_caps, mouse_id) %&gt;%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion &lt; 0.03, &#39;&#39;, cell_type_label)
    
  )</code></pre>
</div>
<div id="stripchart-boxplot" class="section level3" number="5.3">
<h3><span class="header-section-number">5.3</span>
Stripchart-Boxplot</h3>
<pre class="r"><code>max_prop_cell_type &lt;- arranged_data %&gt;% group_by(cell_type) %&gt;% summarise(max_proportion = max(proportion))
stat_test &lt;- propeller_pairwise %&gt;% 
  filter(group1 %in% arranged_data$site_caps &amp; group2 %in%  arranged_data$site_caps) %&gt;%
  left_join(max_prop_cell_type, by = c(&#39;cluster&#39;=&#39;cell_type&#39;)) %&gt;%
  arrange(cluster, group1, group2) %&gt;% 
  na.omit() %&gt;% 
  filter(FDR &lt; 0.05) %&gt;% 
  mutate(cell_type = cluster)
  
add_count_y &lt;- stat_test %&gt;% group_by(cluster) %&gt;% summarise(group1 = group1, group2 = group2, n = 1:n())
stat_test &lt;- stat_test %&gt;% left_join(add_count_y) %&gt;% mutate(y.position = max_proportion + (n*max_proportion/20))

arranged_data %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, color = site_caps)) +
  geom_boxplot(alpha = 0.4, color = &#39;black&#39;, aes(fill = site_caps), 
               show.legend = FALSE, size = 0.25, fatten = 1) +
  # geom_point(alpha = 1, size = 2, shape = 21, color = &#39;black&#39;, aes(fill = site_caps)) +
  geom_quasirandom(method=&quot;smiley&quot;, width = 0.1, alpha = 1, size = 1, 
                   shape = 21, color = &#39;black&#39;, aes(fill = site_caps), stroke = 0.4) +
  scale_fill_manual(values = site_palette) +
  scale_color_manual(values = site_palette) +
  # scale_y_continuous(expand = c(0, 0)) +
  scale_y_continuous(expand = c(0.02, 0, 0.05, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  facet_wrap(vars(cell_type), scales = &#39;free_y&#39;) +
  labs(
    fill = &#39;Site&#39;, 
    x = NULL,
    y = &#39;Proportion of cells&#39;
  ) +
  stat_pvalue_manual(stat_test, label = &#39;FDR.sign&#39;)</code></pre>
<p><img src="figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/da-stripchart-site-and-mouse-cluster_mnn-1.png" width="4320" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-da-stripchart-site-and-mouse-cluster_mnn-1">
Past versions of da-stripchart-site-and-mouse-cluster_mnn-1.png
</button>
</p>
<div id="fig-da-stripchart-site-and-mouse-cluster_mnn-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/39b365c03bef848fb12a4a5aee47a955aecb169a/docs/figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/da-stripchart-site-and-mouse-cluster_mnn-1.png" target="_blank">39b365c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="stripchart-boxplot-selected-clusters-equal-breaks"
class="section level3" number="5.4">
<h3><span class="header-section-number">5.4</span> Stripchart-Boxplot
selected clusters equal breaks</h3>
<pre class="r"><code>sub_arranged_data &lt;- arranged_data %&gt;% 
  filter(cell_type %in% selected_clusters_to_show) %&gt;% 
  mutate(
    cell_type = as.character(cell_type)
  )

# add missing categories
for(ct in unique(sub_arranged_data$cell_type)) {
  for(st in unique(sub_arranged_data$site_caps)) {
    for(mi in unique(sub_arranged_data$mouse_id)){
      nmatch &lt;- sub_arranged_data %&gt;% 
        filter(cell_type == ct) %&gt;%  
        filter(site_caps == st) %&gt;%  
        filter(mouse_id == mi) %&gt;% 
        nrow
      if(nmatch == 0){
        add_df &lt;- sub_arranged_data[1,] %&gt;% 
          mutate_all(function(x) NA) %&gt;% 
          mutate(
            site_caps = st, 
            mouse_id = mi,
            cell_type = ct,
            proportion = 0) 
        sub_arranged_data &lt;- rbind(sub_arranged_data, add_df)
      }
    }
  }
}

sub_arranged_data %&lt;&gt;% 
  mutate(site_caps = factor(site_caps, levels = levels(arranged_data$site_caps)))


# stat test
max_prop_cell_type &lt;- sub_arranged_data %&gt;% group_by(cell_type) %&gt;% summarise(max_proportion = max(proportion))
stat_test &lt;- propeller_pairwise %&gt;% 
  filter(group1 %in% sub_arranged_data$site_caps &amp; group2 %in%  sub_arranged_data$site_caps) %&gt;%
  left_join(max_prop_cell_type, by = c(&#39;cluster&#39;=&#39;cell_type&#39;)) %&gt;%
  arrange(cluster, group1, group2) %&gt;% 
  na.omit() %&gt;% 
  filter(FDR &lt; 0.05) %&gt;% 
  mutate(cell_type = cluster)
  
add_count_y &lt;- stat_test %&gt;% group_by(cluster) %&gt;% summarise(group1 = group1, group2 = group2, n = 1:n())
stat_test &lt;- stat_test %&gt;% left_join(add_count_y) %&gt;% mutate(y.position = max_proportion + (n*max_proportion/3))

# add site info
use_cl_info &lt;- selected_clusters_to_df %&gt;% 
  mutate(site_cl = paste(site_caps, cluster))

sub_arranged_data %&lt;&gt;% 
  left_join(use_cl_info %&gt;% rename(site_caps_g = site_caps), 
            by = join_by(cell_type == cluster)) %&gt;% 
  mutate(cell_type = factor(site_cl, levels = use_cl_info$site_cl)) 

stat_test %&lt;&gt;% 
  left_join(use_cl_info) %&gt;% 
  mutate(cell_type = factor(site_cl, levels = use_cl_info$site_cl)) 

equal_breaks &lt;- function(n = 3, s = 0.05, ...){
  function(x){
    # rescaling
    d &lt;- s * diff(range(x)) / (1+2*s)
    seq(0, max(x)-d, length=n) %&gt;% round(2)
    # res &lt;- seq(min(x)+d, max(x)-d, length=n)
    # floor(10*res) / 10
  }
}

stat_test$cell_type &lt;- recode_factor(
      stat_test$cell_type, 
      `Primary tumor 11` = &quot;PT 11&quot;, 
      `Primary tumor 4` = &quot;PT 4&quot;)

sub_arranged_data %&gt;% 
  mutate(
    cell_type = recode_factor(
      cell_type, 
      `Primary tumor 11` = &quot;PT 11&quot;, 
      `Primary tumor 4` = &quot;PT 4&quot;)
  ) %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, color = site_caps)) +
  geom_boxplot(alpha = 0.4, color = &#39;black&#39;, aes(fill = site_caps), 
               show.legend = FALSE, size = 0.25, fatten = 1) +
  # geom_point(alpha = 1, size = 2, shape = 21, color = &#39;black&#39;, aes(fill = site_caps)) +
  geom_quasirandom(method=&quot;smiley&quot;, width = 0.2, alpha = 1, size = 1, 
                   shape = 21, color = &#39;black&#39;, aes(fill = site_caps), stroke = 0.4) +
  scale_fill_manual(values = site_palette) +
  scale_color_manual(values = site_palette) +
  # scale_y_continuous(expand = c(0, 0)) +
  scale_y_continuous(expand = c(0.06, 0, 0.10, 0),
                     # n.breaks = 4,
                     breaks=equal_breaks(n=4, s=0.1)
                     # limits = c(0, 0.55)
                     ) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), scales = &#39;free_y&#39;, ncol = 2) +
  guides(
    # fill = guide_legend(override.aes = list(alpha = 1, size = 3)),
    fill = &#39;none&#39;,
    color = &#39;none&#39;,
    ) +
  labs(
    fill = &#39;Site&#39;, 
    x = NULL,
    y = &#39;Proportion of cells&#39;
  ) +
  stat_pvalue_manual(stat_test, label = &#39;FDR.sign&#39;, size = 2.88) +
    theme(
      # axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)
      axis.text.x = element_blank()
      # axis.text.x = element_blank(),
      # panel.border = element_rect(color = &quot;black&quot;, fill = NA, linewidth = 1),
      # axis.line = element_blank(),
      # axis.ticks.x=element_blank(),
      # axis.line.x = element_blank(),
      # strip.background = element_blank()
      )</code></pre>
<p><img src="figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/da-stripchart-site-and-mouse-cluster_mnn-selected-equal-breaks-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-da-stripchart-site-and-mouse-cluster_mnn-selected-equal-breaks-1">
Past versions of
da-stripchart-site-and-mouse-cluster_mnn-selected-equal-breaks-1.png
</button>
</p>
<div
id="fig-da-stripchart-site-and-mouse-cluster_mnn-selected-equal-breaks-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/39b365c03bef848fb12a4a5aee47a955aecb169a/docs/figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/da-stripchart-site-and-mouse-cluster_mnn-selected-equal-breaks-1.png" target="_blank">39b365c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="cluster-markers-enrichment" class="section level2" number="6">
<h2><span class="header-section-number">6</span> Cluster markers
enrichment</h2>
<div id="ora---c2.cp.reactome-and-gobp" class="section level3"
number="6.1">
<h3><span class="header-section-number">6.1</span> ORA - C2.CP.REACTOME
and <a href="GO:BP" class="uri">GO:BP</a></h3>
<p>Over-enrichment analysis using genes with a mean AUC ≥ 0.25 in the
score markers results.</p>
<pre class="r"><code>gse_list &lt;- gse_cl_auc
test_type &lt;- &#39;enricher&#39;
gset_collection &lt;- &#39;msigdb.m2.cp.reactome&#39;
direction &lt;- &#39;abs&#39;
p_thrs &lt;- 0.01
topn &lt;- 20

# Load all the results
gse_df &lt;- foreach(i = names(gse_list), .combine = rbind) %do% {
  i_1 &lt;- gse_list[[i]][[test_type]][[direction]][[gset_collection]]@result %&gt;% 
    mutate(group = i)
  i_2 &lt;- gse_list[[i]][[&#39;enrichGO&#39;]][[direction]][[&#39;BP&#39;]]@result %&gt;% 
    filter(simplify) %&gt;% 
    dplyr::select(-simplify) %&gt;% 
    mutate(group = i)
  rbind(i_1, i_2)
} %&gt;% 
  mutate(
    group_n = as.numeric(group),
    group = fct_reorder(group, group_n),
    Description = clean_msigdb_names(Description) %&gt;% tolower() %&gt;% str_to_title,
    Description = ifelse(grepl(&quot;^GO:&quot;, ID), paste0(&quot;GO:&quot;, Description), Description)
  ) %&gt;% 
  dplyr::select(-group_n)

# Select topn by group in selected gset
top_by_group &lt;- foreach(i = names(gse_list), .combine = c) %do% {
  gse_df %&gt;% 
    filter(group == i) %&gt;% 
    filter(p.adjust &lt; p_thrs) %&gt;% 
    arrange(p.adjust) %&gt;% pull(Description) %&gt;%
    head(topn)
}

# Data frame to plot
res_top &lt;- gse_df %&gt;% 
  filter(Description %in% top_by_group) %&gt;% 
  mutate(log.p.adjust = -log10(p.adjust))

# Rows and column orders by hclust
use_mat &lt;- res_top %&gt;% 
  dplyr::select(Description, log.p.adjust, group) %&gt;% 
  pivot_wider(names_from = group, values_from = log.p.adjust) %&gt;% 
  column_to_rownames(&#39;Description&#39;)
use_mat[is.na(use_mat)] &lt;- 0

row_hclust &lt;- hclust(dist(use_mat))
row_hclust_labels &lt;- row_hclust$labels[row_hclust$order]

col_hclust &lt;- hclust(dist(t(use_mat)))
col_hclust_labels &lt;- col_hclust$labels[col_hclust$order]

use_mat &lt;- use_mat[row_hclust_labels,col_hclust_labels]

# GSET clusters by kmeans using silhouette
use_kmax &lt;- ifelse(nrow(use_mat) &lt;= 20, nrow(use_mat)-2, 20)
nb_clust &lt;- fviz_nbclust(use_mat, kmeans, method = &quot;silhouette&quot;, k.max = use_kmax)
silhouette_best_k &lt;- nb_clust$data %&gt;% slice_max(y) %&gt;% pull(clusters) %&gt;% as.character %&gt;% as.numeric
# silhouette_best_k &lt;- 5
kmeans_mat &lt;- kmeans(use_mat, silhouette_best_k, nstart = 25)
kmeans_mat_df &lt;- kmeans_mat$cluster %&gt;%
  data.frame %&gt;%
  set_names(&#39;kcluster&#39;) %&gt;%
  rownames_to_column(&#39;Description&#39;) %&gt;%
  mutate(kcluster = as.character(kcluster))


# Order factors by hclust
res_top %&lt;&gt;% 
  left_join(kmeans_mat_df) %&gt;% 
  mutate(
    Description = factor(Description, levels = row_hclust_labels),
    group = fct_relevel(group, col_hclust_labels)
  )

# Calculate jaccard similarity matrix
gset_collection &lt;- foreach(i = levels(res_top$Description)) %do% {
  res_top %&gt;% filter(Description == i) %&gt;% pull(geneID) %&gt;% 
    paste(collapse = &#39;/&#39;) %&gt;% 
    strsplit(&#39;/&#39;) %&gt;% unlist %&gt;% unique
}
names(gset_collection) &lt;- levels(res_top$Description)

use_terms &lt;- res_top$Description %&gt;% unique
jc_matrix &lt;- pairwise_terms_jc(gset_collection[use_terms])</code></pre>
<div id="results-table" class="section level4" number="6.1.1">
<h4><span class="header-section-number">6.1.1</span> Results table</h4>
<pre class="r"><code>gse_df %&gt;% 
  filter(p.adjust &lt; p_thrs) %&gt;% 
  arrange(desc(p.adjust)) %&gt;% 
  dplyr::select(group, ID:p.adjust, Count:GeneProp, geneID)  %&gt;%
  dplyr::rename(cluster = group) %&gt;% 
  datatable(., 
            rownames = FALSE, 
            filter = &#39;top&#39;, 
            caption = &#39;Results from over enrichment analysis&#39;,
            extensions = &#39;Buttons&#39;, 
            options = list(
              dom = &#39;Blfrtip&#39;,
              buttons = c(&#39;csv&#39;, &#39;excel&#39;)
            ))  %&gt;% 
  formatRound(c(&#39;GeneProp&#39;), digits = 2) %&gt;%
  formatSignif(c(&#39;pvalue&#39;, &#39;p.adjust&#39;), digits = 3)</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-e7389320684d012f49e0" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e7389320684d012f49e0">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;1&quot;,&quot;2&quot;,&quot;3&quot;,&quot;4&quot;,&quot;5&quot;,&quot;6&quot;,&quot;7&quot;,&quot;8&quot;,&quot;9&quot;,&quot;10&quot;,&quot;11&quot;,&quot;12&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"0.002456547701158\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"0.009637250844528\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"3\" data-max=\"57\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"10\" data-max=\"492\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.026584867075664\" data-max=\"0.583333333333334\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Results from over enrichment analysis<\/caption>","data":[["4","11","1","6","1","1","1","6","2","12","12","12","2","2","2","11","11","11","11","11","4","1","12","6","2","10","4","6","6","6","6","10","10","1","6","11","6","6","12","6","2","12","1","6","6","1","1","9","6","4","1","4","12","12","1","2","1","1","2","6","6","6","6","1","11","2","6","6","5","1","6","4","4","2","6","8","6","6","7","10","6","1","12","10","10","12","6","2","12","1","1","1","6","6","8","8","6","12","11","1","2","1","12","1","6","7","9","6","6","12","11","10","2","10","2","7","7","6","6","6","10","6","2","6","6","10","10","1","6","10","2","2","2","1","11","10","2","12","6","6","6","6","6","12","10","2","1","6","2","2","6","2","6","6","2","6","6","6","12","6","6","7","7","12","12","6","12","12","1","6","6","1","10","6","12","2","6","6","6","6","6","6","6","4","4","8","10","11","6","6","6","10","2","6","6","6","6","10","1","2","2","2","7","7","7","5","6","1","6","2","6","9","1","1","5","5","5","11","12","6","6","6","6","11","1","6","6","6","11","2","6","6","10","9","9","10","10","2","5","6","11","6","8","12","12","9","12","1","5","1","1","2","10","12","5","2","12","6","6","6","2","1","2","9","2","6","12","2","12","12","10","6","6","2","6","6","10","10","2","1","1","1","6","2","2","1","12","1","6","1","1","9","2","7","2","12","2","12","1","10","1","1","1","10","1","1","2","2","1","2","1","12","2","2","6","6","1","1","1","1","11","2","6","1","1","2","2","2","1","1","1","5","12","1","6","1","2","12","12","1","2","6","6","2","2","2","2","1","2","1","1","1","2","1","10","2","2","2","12","2","2","1","2","10","12","2","1","2","1","1","1","2","2","2","12","1","2","2","2","10","12","12","2","1","1","1","1","1","1","1","1","1","1","12","2","1","12","1","12","1","1","1","2","12","12","1","1","2","12","1","1","1","1","1","12","2","10","1","2","12","12","1","1","12","12","1","2","12","1","12","1","1","12","12","12","1","9","2","2","12","12","12","1","12","2","1","12","12","12","12","1","1","12","12","7","12","2","12","12","1","2","2","2","2","1","2","12","12","12","2","1","12","12","12","12","12","10","12","1","12","12","12","12","12","12","12","12","1","1","12","12","12","12","2","12","12","12","10","12","2","9","12","12","12","1","2","12","12","1","2","12","12","10","10","2","12","12","2","1","1","12","2","2","12","2","1","2","2","12","1","1","2","1","1","1","2","1","1","2","2","12","4","1","1","1","2","2","1","1","1","1","2","1","12","2","4","4","2","2","11","2","4","11","1","4","4","4","4","2","4","2","11","1","1","11","11","11","11","11","11"],["REACTOME_RESPONSE_TO_ELEVATED_PLATELET_CYTOSOLIC_CA2","GO:0032328","REACTOME_CYCLIN_A_B1_B2_ASSOCIATED_EVENTS_DURING_G2_M_TRANSITION","REACTOME_SWITCHING_OF_ORIGINS_TO_A_POST_REPLICATIVE_STATE","REACTOME_DEUBIQUITINATION","REACTOME_APOPTOSIS_INDUCED_DNA_FRAGMENTATION","REACTOME_DEPOLYMERISATION_OF_THE_NUCLEAR_LAMINA","REACTOME_DISSOLUTION_OF_FIBRIN_CLOT","REACTOME_MHC_CLASS_II_ANTIGEN_PRESENTATION","REACTOME_ANTIGEN_PROCESSING_UBIQUITINATION_PROTEASOME_DEGRADATION","REACTOME_MAPK_FAMILY_SIGNALING_CASCADES","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_TP53","REACTOME_DEADENYLATION_DEPENDENT_MRNA_DECAY","REACTOME_PROCESSIVE_SYNTHESIS_ON_THE_C_STRAND_OF_THE_TELOMERE","REACTOME_TRANSPORT_OF_CONNEXONS_TO_THE_PLASMA_MEMBRANE","GO:0006090","REACTOME_PINK1_PRKN_MEDIATED_MITOPHAGY","REACTOME_SELECTIVE_AUTOPHAGY","REACTOME_DOWNREGULATION_OF_ERBB4_SIGNALING","REACTOME_REGULATION_OF_PTEN_LOCALIZATION","GO:0046034","REACTOME_INTRINSIC_PATHWAY_FOR_APOPTOSIS","REACTOME_CITRIC_ACID_CYCLE_TCA_CYCLE","REACTOME_GAP_JUNCTION_TRAFFICKING_AND_REGULATION","REACTOME_APC_CDC20_MEDIATED_DEGRADATION_OF_NEK2A","REACTOME_DISSOLUTION_OF_FIBRIN_CLOT","REACTOME_REGULATION_OF_INSULIN_LIKE_GROWTH_FACTOR_IGF_TRANSPORT_AND_UPTAKE_BY_INSULIN_LIKE_GROWTH_FACTOR_BINDING_PROTEINS_IGFBPS","REACTOME_INSULIN_RECEPTOR_RECYCLING","REACTOME_SEALING_OF_THE_NUCLEAR_ENVELOPE_NE_BY_ESCRT_III","REACTOME_APC_C_MEDIATED_DEGRADATION_OF_CELL_CYCLE_PROTEINS","REACTOME_CASPASE_MEDIATED_CLEAVAGE_OF_CYTOSKELETAL_PROTEINS","REACTOME_COPI_DEPENDENT_GOLGI_TO_ER_RETROGRADE_TRAFFIC","REACTOME_SMOOTH_MUSCLE_CONTRACTION","REACTOME_SLBP_DEPENDENT_PROCESSING_OF_REPLICATION_DEPENDENT_HISTONE_PRE_MRNAS","REACTOME_NEDDYLATION","REACTOME_REGULATION_OF_MRNA_STABILITY_BY_PROTEINS_THAT_BIND_AU_RICH_ELEMENTS","REACTOME_MITOTIC_METAPHASE_AND_ANAPHASE","REACTOME_M_PHASE","REACTOME_TETRAHYDROBIOPTERIN_BH4_SYNTHESIS_RECYCLING_SALVAGE_AND_REGULATION","REACTOME_DOWNSTREAM_TCR_SIGNALING","REACTOME_PKMTS_METHYLATE_HISTONE_LYSINES","GO:0051099","REACTOME_SIGNALING_BY_INTERLEUKINS","REACTOME_METABOLISM_OF_AMINO_ACIDS_AND_DERIVATIVES","REACTOME_G2_M_CHECKPOINTS","REACTOME_INTRAFLAGELLAR_TRANSPORT","REACTOME_ANTIGEN_PROCESSING_UBIQUITINATION_PROTEASOME_DEGRADATION","GO:0006334","REACTOME_ASSEMBLY_OF_THE_PRE_REPLICATIVE_COMPLEX","GO:0009144","REACTOME_INTRA_GOLGI_AND_RETROGRADE_GOLGI_TO_ER_TRAFFIC","REACTOME_PEXOPHAGY","REACTOME_PROGRAMMED_CELL_DEATH","REACTOME_DETOXIFICATION_OF_REACTIVE_OXYGEN_SPECIES","REACTOME_TETRAHYDROBIOPTERIN_BH4_SYNTHESIS_RECYCLING_SALVAGE_AND_REGULATION","REACTOME_APC_C_CDC20_MEDIATED_DEGRADATION_OF_CYCLIN_B","REACTOME_INHIBITION_OF_THE_PROTEOLYTIC_ACTIVITY_OF_APC_C_REQUIRED_FOR_THE_ONSET_OF_ANAPHASE_BY_MITOTIC_SPINDLE_CHECKPOINT_COMPONENTS","REACTOME_HSP90_CHAPERONE_CYCLE_FOR_STEROID_HORMONE_RECEPTORS_SHR_IN_THE_PRESENCE_OF_LIGAND","REACTOME_COPI_DEPENDENT_GOLGI_TO_ER_RETROGRADE_TRAFFIC","GO:0051098","REACTOME_MITOTIC_G1_PHASE_AND_G1_S_TRANSITION","REACTOME_MAP2K_AND_MAPK_ACTIVATION","REACTOME_DEGRADATION_OF_BETA_CATENIN_BY_THE_DESTRUCTION_COMPLEX","REACTOME_SIGNALING_BY_WNT","GO:0055076","REACTOME_AGGREPHAGY","REACTOME_RESPONSE_TO_ELEVATED_PLATELET_CYTOSOLIC_CA2","REACTOME_TETRAHYDROBIOPTERIN_BH4_SYNTHESIS_RECYCLING_SALVAGE_AND_REGULATION","GO:0010634","REACTOME_CYTOKINE_SIGNALING_IN_IMMUNE_SYSTEM","REACTOME_RHO_GTPASES_ACTIVATE_PKNS","REACTOME_DOWNREGULATION_OF_ERBB4_SIGNALING","REACTOME_REGULATION_OF_PTEN_LOCALIZATION","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX2","REACTOME_FCERI_MEDIATED_NF_KB_ACTIVATION","GO:0035994","REACTOME_APOPTOTIC_CLEAVAGE_OF_CELLULAR_PROTEINS","REACTOME_ACTIVATION_OF_APC_C_AND_APC_C_CDC20_MEDIATED_DEGRADATION_OF_MITOTIC_PROTEINS","GO:0055076","REACTOME_RHO_GTPASES_ACTIVATE_KTN1","REACTOME_PYROPTOSIS","REACTOME_TCF_DEPENDENT_SIGNALING_IN_RESPONSE_TO_WNT","GO:0098761","REACTOME_ASPARAGINE_N_LINKED_GLYCOSYLATION","REACTOME_IRON_UPTAKE_AND_TRANSPORT","REACTOME_SELECTIVE_AUTOPHAGY","REACTOME_MAPK_FAMILY_SIGNALING_CASCADES","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_E2F6","REACTOME_ACTIVATION_OF_BH3_ONLY_PROTEINS","REACTOME_METABOLISM_OF_AMINO_ACIDS_AND_DERIVATIVES","REACTOME_COPI_INDEPENDENT_GOLGI_TO_ER_RETROGRADE_TRAFFIC","REACTOME_GAP_JUNCTION_ASSEMBLY","REACTOME_APC_C_CDH1_MEDIATED_DEGRADATION_OF_CDC20_AND_OTHER_APC_C_CDH1_TARGETED_PROTEINS_IN_LATE_MITOSIS_EARLY_G1","REACTOME_S_PHASE","GO:0016032","GO:0048525","REACTOME_CDK_MEDIATED_PHOSPHORYLATION_AND_REMOVAL_OF_CDC6","REACTOME_GLUCOSE_METABOLISM","REACTOME_P75NTR_SIGNALS_VIA_NF_KB","GO:0051438","REACTOME_SUMOYLATION_OF_DNA_DAMAGE_RESPONSE_AND_REPAIR_PROTEINS","REACTOME_CONDENSATION_OF_PROPHASE_CHROMOSOMES","REACTOME_CYTOKINE_SIGNALING_IN_IMMUNE_SYSTEM","REACTOME_APOPTOSIS","REACTOME_ORC1_REMOVAL_FROM_CHROMATIN","GO:0070997","GO:0042118","REACTOME_REGULATED_NECROSIS","REACTOME_CELLULAR_RESPONSE_TO_HYPOXIA","REACTOME_METABOLISM_OF_NUCLEOTIDES","REACTOME_CELLULAR_RESPONSES_TO_STIMULI","REACTOME_MAP2K_AND_MAPK_ACTIVATION","REACTOME_INHIBITION_OF_THE_PROTEOLYTIC_ACTIVITY_OF_APC_C_REQUIRED_FOR_THE_ONSET_OF_ANAPHASE_BY_MITOTIC_SPINDLE_CHECKPOINT_COMPONENTS","REACTOME_PLATELET_ACTIVATION_SIGNALING_AND_AGGREGATION","REACTOME_DEPOLYMERISATION_OF_THE_NUCLEAR_LAMINA","GO:0050678","GO:0010634","REACTOME_PTEN_REGULATION","REACTOME_RUNX1_REGULATES_TRANSCRIPTION_OF_GENES_INVOLVED_IN_DIFFERENTIATION_OF_HSCS","REACTOME_CELL_CELL_COMMUNICATION","REACTOME_INTRA_GOLGI_AND_RETROGRADE_GOLGI_TO_ER_TRAFFIC","REACTOME_PCP_CE_PATHWAY","REACTOME_SUMOYLATION_OF_DNA_REPLICATION_PROTEINS","REACTOME_ACTIVATION_OF_NF_KAPPAB_IN_B_CELLS","REACTOME_CELL_JUNCTION_ORGANIZATION","REACTOME_RESPONSE_TO_ELEVATED_PLATELET_CYTOSOLIC_CA2","REACTOME_ROS_AND_RNS_PRODUCTION_IN_PHAGOCYTES","REACTOME_ORGANELLE_BIOGENESIS_AND_MAINTENANCE","REACTOME_EPH_EPHRIN_SIGNALING","REACTOME_METABOLISM_OF_CARBOHYDRATES","REACTOME_CHROMATIN_MODIFYING_ENZYMES","REACTOME_ESTABLISHMENT_OF_SISTER_CHROMATID_COHESION","REACTOME_SLBP_DEPENDENT_PROCESSING_OF_REPLICATION_DEPENDENT_HISTONE_PRE_MRNAS","REACTOME_ANCHORING_OF_THE_BASAL_BODY_TO_THE_PLASMA_MEMBRANE","REACTOME_NF_KB_IS_ACTIVATED_AND_SIGNALS_SURVIVAL","REACTOME_NERVOUS_SYSTEM_DEVELOPMENT","REACTOME_RHO_GTPASES_ACTIVATE_IQGAPS","REACTOME_INTRINSIC_PATHWAY_FOR_APOPTOSIS","REACTOME_G1_S_TRANSITION","REACTOME_INTERCONVERSION_OF_NUCLEOTIDE_DI_AND_TRIPHOSPHATES","REACTOME_ASYMMETRIC_LOCALIZATION_OF_PCP_PROTEINS","REACTOME_SIGNALING_BY_THE_B_CELL_RECEPTOR_BCR","REACTOME_HEDGEHOG_ON_STATE","REACTOME_MRNA_DECAY_BY_5_TO_3_EXORIBONUCLEASE","GO:1902410","REACTOME_FACTORS_INVOLVED_IN_MEGAKARYOCYTE_DEVELOPMENT_AND_PLATELET_PRODUCTION","REACTOME_DEADENYLATION_DEPENDENT_MRNA_DECAY","REACTOME_BETA_CATENIN_INDEPENDENT_WNT_SIGNALING","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX1","REACTOME_PROCESSING_AND_ACTIVATION_OF_SUMO","REACTOME_ABC_FAMILY_PROTEINS_MEDIATED_TRANSPORT","GO:0051438","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX3","REACTOME_GLI3_IS_PROCESSED_TO_GLI3R_BY_THE_PROTEASOME","REACTOME_OXIDATIVE_STRESS_INDUCED_SENESCENCE","REACTOME_DEGRADATION_OF_DVL","REACTOME_TNFR2_NON_CANONICAL_NF_KB_PATHWAY","REACTOME_UCH_PROTEINASES","REACTOME_AGGREPHAGY","REACTOME_INTRINSIC_PATHWAY_FOR_APOPTOSIS","REACTOME_STABILIZATION_OF_P53","GO:0046034","GO:0006735","REACTOME_APOPTOSIS","GO:0006979","REACTOME_DEGRADATION_OF_AXIN","GO:0050657","GO:0050658","REACTOME_FACTORS_INVOLVED_IN_MEGAKARYOCYTE_DEVELOPMENT_AND_PLATELET_PRODUCTION","REACTOME_REGULATION_OF_RUNX3_EXPRESSION_AND_ACTIVITY","REACTOME_CLEC7A_DECTIN_1_SIGNALING","REACTOME_RECYCLING_PATHWAY_OF_L1","REACTOME_TRANSFERRIN_ENDOCYTOSIS_AND_RECYCLING","REACTOME_TP53_REGULATES_METABOLIC_GENES","REACTOME_MRNA_SPLICING_MINOR_PATHWAY","REACTOME_FORMATION_OF_SENESCENCE_ASSOCIATED_HETEROCHROMATIN_FOCI_SAHF","REACTOME_RHO_GTPASES_ACTIVATE_PAKS","REACTOME_ANTIGEN_PROCESSING_CROSS_PRESENTATION","REACTOME_C_TYPE_LECTIN_RECEPTORS_CLRS","REACTOME_REGULATION_OF_RUNX2_EXPRESSION_AND_ACTIVITY","REACTOME_UBIQUITIN_MEDIATED_DEGRADATION_OF_PHOSPHORYLATED_CDC25A","REACTOME_REGULATION_OF_PTEN_STABILITY_AND_ACTIVITY","REACTOME_MITOTIC_G2_G2_M_PHASES","GO:0010035","GO:1990000","GO:0052547","REACTOME_GOLGI_TO_ER_RETROGRADE_TRANSPORT","REACTOME_PEXOPHAGY","REACTOME_REGULATION_OF_RAS_BY_GAPS","REACTOME_HEDGEHOG_OFF_STATE","REACTOME_DETOXIFICATION_OF_REACTIVE_OXYGEN_SPECIES","REACTOME_RHO_GTPASES_ACTIVATE_IQGAPS","REACTOME_RUNX1_REGULATES_GENES_INVOLVED_IN_MEGAKARYOCYTE_DIFFERENTIATION_AND_PLATELET_FUNCTION","REACTOME_CROSS_PRESENTATION_OF_SOLUBLE_EXOGENOUS_ANTIGENS_ENDOSOMES","REACTOME_REGULATION_OF_MRNA_STABILITY_BY_PROTEINS_THAT_BIND_AU_RICH_ELEMENTS","REACTOME_DEVELOPMENTAL_BIOLOGY","REACTOME_G1_S_DNA_DAMAGE_CHECKPOINTS","REACTOME_FCGAMMA_RECEPTOR_FCGR_DEPENDENT_PHAGOCYTOSIS","REACTOME_SELECTIVE_AUTOPHAGY","REACTOME_PROGRAMMED_CELL_DEATH","REACTOME_ORC1_REMOVAL_FROM_CHROMATIN","REACTOME_MITOTIC_G2_G2_M_PHASES","GO:0001666","GO:1903706","GO:0036293","GO:1901216","REACTOME_HEDGEHOG_LIGAND_BIOGENESIS","REACTOME_DEPOSITION_OF_NEW_CENPA_CONTAINING_NUCLEOSOMES_AT_THE_CENTROMERE","REACTOME_APOPTOSIS","REACTOME_KINESINS","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX2","GO:0010632","REACTOME_UB_SPECIFIC_PROCESSING_PROTEASES","REACTOME_NUCLEAR_ENVELOPE_BREAKDOWN","GO:0009205","GO:0009144","GO:0009199","GO:1901216","REACTOME_TCF_DEPENDENT_SIGNALING_IN_RESPONSE_TO_WNT","REACTOME_FCGAMMA_RECEPTOR_FCGR_DEPENDENT_PHAGOCYTOSIS","REACTOME_CELL_EXTRACELLULAR_MATRIX_INTERACTIONS","REACTOME_DOWNSTREAM_SIGNALING_EVENTS_OF_B_CELL_RECEPTOR_BCR","REACTOME_FC_EPSILON_RECEPTOR_FCERI_SIGNALING","GO:0097193","REACTOME_COPI_DEPENDENT_GOLGI_TO_ER_RETROGRADE_TRAFFIC","REACTOME_SIGNALING_BY_HEDGEHOG","REACTOME_DECTIN_1_MEDIATED_NONCANONICAL_NF_KB_SIGNALING","REACTOME_KEAP1_NFE2L2_PATHWAY","GO:0000028","REACTOME_PROCESSIVE_SYNTHESIS_ON_THE_LAGGING_STRAND","REACTOME_NERVOUS_SYSTEM_DEVELOPMENT","REACTOME_MAPK6_MAPK4_SIGNALING","REACTOME_INSULIN_RECEPTOR_RECYCLING","GO:0032680","GO:0006110","REACTOME_GLUCONEOGENESIS","REACTOME_RHO_GTPASE_EFFECTORS","REACTOME_TRANSPORT_OF_MATURE_TRANSCRIPT_TO_CYTOPLASM","GO:0046034","REACTOME_CYCLIN_A_CDK2_ASSOCIATED_EVENTS_AT_S_PHASE_ENTRY","GO:0006110","GO:0052547","GO:0043620","REACTOME_MITOCHONDRIAL_TRANSLATION","REACTOME_INTRACELLULAR_SIGNALING_BY_SECOND_MESSENGERS","GO:0050727","REACTOME_SIGNALING_BY_WNT","REACTOME_CHROMOSOME_MAINTENANCE","GO:0010035","REACTOME_RHO_GTPASES_ACTIVATE_IQGAPS","REACTOME_GOLGI_TO_ER_RETROGRADE_TRANSPORT","REACTOME_DUAL_INCISION_IN_TC_NER","REACTOME_GLUCOSE_METABOLISM","REACTOME_DEUBIQUITINATION","GO:0043620","REACTOME_TRANSCRIPTION_COUPLED_NUCLEOTIDE_EXCISION_REPAIR_TC_NER","REACTOME_METABOLISM_OF_AMINO_ACIDS_AND_DERIVATIVES","REACTOME_EPHB_MEDIATED_FORWARD_SIGNALING","REACTOME_RECYCLING_PATHWAY_OF_L1","REACTOME_L1CAM_INTERACTIONS","REACTOME_THE_ROLE_OF_GTSE1_IN_G2_M_PROGRESSION_AFTER_G2_CHECKPOINT","REACTOME_TRANSPORT_OF_CONNEXONS_TO_THE_PLASMA_MEMBRANE","REACTOME_APOPTOSIS","GO:0007178","REACTOME_DNA_DAMAGE_TELOMERE_STRESS_INDUCED_SENESCENCE","GO:0009611","REACTOME_SIGNALING_BY_INTERLEUKINS","REACTOME_MRNA_SPLICING_MINOR_PATHWAY","REACTOME_COMPLEX_I_BIOGENESIS","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX1","GO:0071496","GO:0032970","GO:0043254","REACTOME_RMTS_METHYLATE_HISTONE_ARGININES","REACTOME_SCF_SKP2_MEDIATED_DEGRADATION_OF_P27_P21","REACTOME_RHO_GTPASES_ACTIVATE_WASPS_AND_WAVES","REACTOME_CELLULAR_RESPONSES_TO_STIMULI","REACTOME_GLYCOLYSIS","REACTOME_HDR_THROUGH_SINGLE_STRAND_ANNEALING_SSA","REACTOME_AGGREPHAGY","REACTOME_REGULATION_OF_APC_C_ACTIVATORS_BETWEEN_G1_S_AND_EARLY_ANAPHASE","REACTOME_METABOLISM_OF_NUCLEOTIDES","REACTOME_PROGRAMMED_CELL_DEATH","REACTOME_POLYMERASE_SWITCHING_ON_THE_C_STRAND_OF_THE_TELOMERE","REACTOME_HDR_THROUGH_HOMOLOGOUS_RECOMBINATION_HRR","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX1","GO:0033108","REACTOME_MRNA_SPLICING_MINOR_PATHWAY","REACTOME_AUF1_HNRNP_D0_BINDS_AND_DESTABILIZES_MRNA","REACTOME_PTEN_REGULATION","REACTOME_MRNA_DECAY_BY_5_TO_3_EXORIBONUCLEASE","GO:1901216","REACTOME_SWITCHING_OF_ORIGINS_TO_A_POST_REPLICATIVE_STATE","GO:0010035","REACTOME_CELLULAR_RESPONSES_TO_STIMULI","REACTOME_UB_SPECIFIC_PROCESSING_PROTEASES","REACTOME_ACTIVATION_OF_APC_C_AND_APC_C_CDC20_MEDIATED_DEGRADATION_OF_MITOTIC_PROTEINS","REACTOME_NEDDYLATION","REACTOME_MITOTIC_TELOPHASE_CYTOKINESIS","GO:0019058","REACTOME_RECRUITMENT_OF_NUMA_TO_MITOTIC_CENTROSOMES","REACTOME_PCP_CE_PATHWAY","REACTOME_MHC_CLASS_II_ANTIGEN_PRESENTATION","REACTOME_EPH_EPHRIN_SIGNALING","REACTOME_TCR_SIGNALING","REACTOME_MITOTIC_PROPHASE","REACTOME_TRANSLESION_SYNTHESIS_BY_Y_FAMILY_DNA_POLYMERASES_BYPASSES_LESIONS_ON_DNA_TEMPLATE","REACTOME_APOPTOTIC_EXECUTION_PHASE","REACTOME_INTERLEUKIN_1_FAMILY_SIGNALING","REACTOME_MRNA_DECAY_BY_5_TO_3_EXORIBONUCLEASE","REACTOME_ABC_FAMILY_PROTEINS_MEDIATED_TRANSPORT","REACTOME_INTERCONVERSION_OF_NUCLEOTIDE_DI_AND_TRIPHOSPHATES","GO:0140719","REACTOME_ASSEMBLY_OF_THE_PRE_REPLICATIVE_COMPLEX","REACTOME_RHO_GTPASE_EFFECTORS","REACTOME_CELLULAR_RESPONSE_TO_CHEMICAL_STRESS","REACTOME_CELLULAR_RESPONSE_TO_CHEMICAL_STRESS","REACTOME_HEDGEHOG_ON_STATE","REACTOME_TNFR2_NON_CANONICAL_NF_KB_PATHWAY","REACTOME_UCH_PROTEINASES","GO:0035914","REACTOME_NUCLEOTIDE_EXCISION_REPAIR","REACTOME_METABOLISM_OF_POLYAMINES","REACTOME_DEGRADATION_OF_BETA_CATENIN_BY_THE_DESTRUCTION_COMPLEX","REACTOME_TRANSPORT_OF_MATURE_TRANSCRIPT_TO_CYTOPLASM","REACTOME_GLOBAL_GENOME_NUCLEOTIDE_EXCISION_REPAIR_GG_NER","REACTOME_REGULATION_OF_APC_C_ACTIVATORS_BETWEEN_G1_S_AND_EARLY_ANAPHASE","REACTOME_TRANSLESION_SYNTHESIS_BY_POLH","REACTOME_INITIATION_OF_NUCLEAR_ENVELOPE_NE_REFORMATION","GO:0046385","REACTOME_KEAP1_NFE2L2_PATHWAY","GO:0035914","REACTOME_GLUCONEOGENESIS","REACTOME_BETA_CATENIN_INDEPENDENT_WNT_SIGNALING","REACTOME_RHO_GTPASES_ACTIVATE_IQGAPS","REACTOME_MAPK6_MAPK4_SIGNALING","REACTOME_CONDENSATION_OF_PROPHASE_CHROMOSOMES","REACTOME_CELL_CYCLE_CHECKPOINTS","REACTOME_TP53_REGULATES_METABOLIC_GENES","REACTOME_SIGNALING_BY_HEDGEHOG","REACTOME_APC_C_MEDIATED_DEGRADATION_OF_CELL_CYCLE_PROTEINS","REACTOME_THE_ROLE_OF_GTSE1_IN_G2_M_PROGRESSION_AFTER_G2_CHECKPOINT","REACTOME_CELLULAR_RESPONSES_TO_STIMULI","REACTOME_REGULATION_OF_TP53_ACTIVITY","REACTOME_TRANSLESION_SYNTHESIS_BY_POLK","REACTOME_DUAL_INCISION_IN_GG_NER","REACTOME_TERMINATION_OF_TRANSLESION_DNA_SYNTHESIS","REACTOME_KINESINS","GO:0034502","REACTOME_ANTIGEN_PROCESSING_CROSS_PRESENTATION","REACTOME_DOWNSTREAM_TCR_SIGNALING","REACTOME_REGULATION_OF_PLK1_ACTIVITY_AT_G2_M_TRANSITION","GO:0140694","REACTOME_REGULATION_OF_PTEN_STABILITY_AND_ACTIVITY","GO:0022604","REACTOME_SUMOYLATION","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_TP53","REACTOME_CELLULAR_SENESCENCE","REACTOME_PTEN_REGULATION","REACTOME_DNA_DAMAGE_BYPASS","REACTOME_MITOTIC_PROPHASE","REACTOME_RUNX1_REGULATES_TRANSCRIPTION_OF_GENES_INVOLVED_IN_DIFFERENTIATION_OF_HSCS","REACTOME_GAP_FILLING_DNA_REPAIR_SYNTHESIS_AND_LIGATION_IN_GG_NER","REACTOME_RHO_GTPASES_ACTIVATE_WASPS_AND_WAVES","REACTOME_MITOTIC_G2_G2_M_PHASES","REACTOME_REGULATION_OF_TP53_ACTIVITY_THROUGH_PHOSPHORYLATION","REACTOME_ACTIVATION_OF_NF_KAPPAB_IN_B_CELLS","GO:0006403","REACTOME_FCERI_MEDIATED_NF_KB_ACTIVATION","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX3","REACTOME_HEDGEHOG_LIGAND_BIOGENESIS","GO:0009411","REACTOME_G2_M_DNA_DAMAGE_CHECKPOINT","REACTOME_NUCLEAR_ENVELOPE_NE_REASSEMBLY","REACTOME_SIGNALING_BY_HEDGEHOG","REACTOME_ASYMMETRIC_LOCALIZATION_OF_PCP_PROTEINS","REACTOME_RNA_POLYMERASE_II_TRANSCRIPTION_TERMINATION","REACTOME_PROCESSING_OF_DNA_DOUBLE_STRAND_BREAK_ENDS","REACTOME_POLYMERASE_SWITCHING","REACTOME_EPHB_MEDIATED_FORWARD_SIGNALING","REACTOME_M_PHASE","REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA","REACTOME_METABOLISM_OF_NUCLEOTIDES","REACTOME_ORC1_REMOVAL_FROM_CHROMATIN","REACTOME_DECTIN_1_MEDIATED_NONCANONICAL_NF_KB_SIGNALING","REACTOME_GLI3_IS_PROCESSED_TO_GLI3R_BY_THE_PROTEASOME","REACTOME_REGULATION_OF_MRNA_STABILITY_BY_PROTEINS_THAT_BIND_AU_RICH_ELEMENTS","GO:0006403","REACTOME_AURKA_ACTIVATION_BY_TPX2","REACTOME_CELLULAR_RESPONSE_TO_HYPOXIA","REACTOME_DEGRADATION_OF_DVL","REACTOME_ASSEMBLY_OF_THE_PRE_REPLICATIVE_COMPLEX","REACTOME_STABILIZATION_OF_P53","REACTOME_TNFR2_NON_CANONICAL_NF_KB_PATHWAY","REACTOME_TELOMERE_C_STRAND_LAGGING_STRAND_SYNTHESIS","REACTOME_DEGRADATION_OF_AXIN","REACTOME_HEDGEHOG_ON_STATE","REACTOME_G2_M_CHECKPOINTS","REACTOME_G2_M_CHECKPOINTS","REACTOME_REGULATION_OF_RUNX3_EXPRESSION_AND_ACTIVITY","REACTOME_DNA_REPLICATION","REACTOME_G1_S_DNA_DAMAGE_CHECKPOINTS","REACTOME_APOPTOSIS_INDUCED_DNA_FRAGMENTATION","REACTOME_DEGRADATION_OF_BETA_CATENIN_BY_THE_DESTRUCTION_COMPLEX","REACTOME_DNA_REPLICATION","REACTOME_CLEC7A_DECTIN_1_SIGNALING","REACTOME_HEDGEHOG_OFF_STATE","REACTOME_INITIATION_OF_NUCLEAR_ENVELOPE_NE_REFORMATION","REACTOME_MRNA_SPLICING","REACTOME_SWITCHING_OF_ORIGINS_TO_A_POST_REPLICATIVE_STATE","GO:0140694","REACTOME_RNA_POLYMERASE_II_TRANSCRIPTION_TERMINATION","REACTOME_REGULATION_OF_RUNX2_EXPRESSION_AND_ACTIVITY","REACTOME_UBIQUITIN_MEDIATED_DEGRADATION_OF_PHOSPHORYLATED_CDC25A","REACTOME_HEDGEHOG_OFF_STATE","REACTOME_RECOGNITION_OF_DNA_DAMAGE_BY_PCNA_CONTAINING_REPLICATION_COMPLEX","GO:0070972","REACTOME_SIGNALING_BY_THE_B_CELL_RECEPTOR_BCR","REACTOME_DNA_REPLICATION_PRE_INITIATION","REACTOME_INTERLEUKIN_1_FAMILY_SIGNALING","REACTOME_TCR_SIGNALING","REACTOME_CELLULAR_RESPONSES_TO_STIMULI","REACTOME_CDK_MEDIATED_PHOSPHORYLATION_AND_REMOVAL_OF_CDC6","REACTOME_SEPARATION_OF_SISTER_CHROMATIDS","REACTOME_MAPK6_MAPK4_SIGNALING","REACTOME_CROSS_PRESENTATION_OF_SOLUBLE_EXOGENOUS_ANTIGENS_ENDOSOMES","REACTOME_INTERCONVERSION_OF_NUCLEOTIDE_DI_AND_TRIPHOSPHATES","REACTOME_APC_C_MEDIATED_DEGRADATION_OF_CELL_CYCLE_PROTEINS","REACTOME_INTERCONVERSION_OF_NUCLEOTIDE_DI_AND_TRIPHOSPHATES","REACTOME_CDK_MEDIATED_PHOSPHORYLATION_AND_REMOVAL_OF_CDC6","REACTOME_MITOTIC_G1_PHASE_AND_G1_S_TRANSITION","REACTOME_DNA_REPLICATION_PRE_INITIATION","REACTOME_ABC_FAMILY_PROTEINS_MEDIATED_TRANSPORT","REACTOME_S_PHASE","REACTOME_MITOTIC_G1_PHASE_AND_G1_S_TRANSITION","REACTOME_C_TYPE_LECTIN_RECEPTORS_CLRS","GO:0035914","REACTOME_MITOTIC_G1_PHASE_AND_G1_S_TRANSITION","REACTOME_G1_S_TRANSITION","REACTOME_DNA_REPLICATION_PRE_INITIATION","REACTOME_UCH_PROTEINASES","REACTOME_ANTIGEN_PROCESSING_CROSS_PRESENTATION","REACTOME_AUF1_HNRNP_D0_BINDS_AND_DESTABILIZES_MRNA","REACTOME_REGULATION_OF_PTEN_STABILITY_AND_ACTIVITY","REACTOME_PRC2_METHYLATES_HISTONES_AND_DNA","REACTOME_REGULATION_OF_RAS_BY_GAPS","REACTOME_BETA_CATENIN_INDEPENDENT_WNT_SIGNALING","REACTOME_CELLULAR_RESPONSES_TO_STIMULI","REACTOME_REGULATION_OF_RAS_BY_GAPS","REACTOME_RUNX1_REGULATES_TRANSCRIPTION_OF_GENES_INVOLVED_IN_DIFFERENTIATION_OF_HSCS","REACTOME_NUCLEAR_ENVELOPE_NE_REASSEMBLY","REACTOME_FC_EPSILON_RECEPTOR_FCERI_SIGNALING","REACTOME_G1_S_DNA_DAMAGE_CHECKPOINTS","REACTOME_SWITCHING_OF_ORIGINS_TO_A_POST_REPLICATIVE_STATE","GO:0043620","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX3","REACTOME_EPIGENETIC_REGULATION_OF_GENE_EXPRESSION","REACTOME_ACTIVATION_OF_APC_C_AND_APC_C_CDC20_MEDIATED_DEGRADATION_OF_MITOTIC_PROTEINS","REACTOME_SIGNALING_BY_THE_B_CELL_RECEPTOR_BCR","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX2","REACTOME_ACTIVATION_OF_THE_PRE_REPLICATIVE_COMPLEX","REACTOME_DNA_REPAIR","REACTOME_BASE_EXCISION_REPAIR","REACTOME_DNA_DOUBLE_STRAND_BREAK_REPAIR","REACTOME_S_PHASE","REACTOME_RHO_GTPASE_EFFECTORS","REACTOME_PCP_CE_PATHWAY","REACTOME_APC_C_CDH1_MEDIATED_DEGRADATION_OF_CDC20_AND_OTHER_APC_C_CDH1_TARGETED_PROTEINS_IN_LATE_MITOSIS_EARLY_G1","REACTOME_ASYMMETRIC_LOCALIZATION_OF_PCP_PROTEINS","REACTOME_DNA_STRAND_ELONGATION","REACTOME_CYCLIN_A_CDK2_ASSOCIATED_EVENTS_AT_S_PHASE_ENTRY","REACTOME_DOWNSTREAM_TCR_SIGNALING","REACTOME_REGULATION_OF_MRNA_STABILITY_BY_PROTEINS_THAT_BIND_AU_RICH_ELEMENTS","REACTOME_C_TYPE_LECTIN_RECEPTORS_CLRS","REACTOME_ORC1_REMOVAL_FROM_CHROMATIN","REACTOME_ASSEMBLY_OF_THE_PRE_REPLICATIVE_COMPLEX","GO:0034976","REACTOME_CELLULAR_RESPONSE_TO_HYPOXIA","REACTOME_METABOLISM_OF_POLYAMINES","REACTOME_DECTIN_1_MEDIATED_NONCANONICAL_NF_KB_SIGNALING","REACTOME_GLI3_IS_PROCESSED_TO_GLI3R_BY_THE_PROTEASOME","REACTOME_MITOTIC_METAPHASE_AND_ANAPHASE","REACTOME_DEGRADATION_OF_DVL","REACTOME_STABILIZATION_OF_P53","REACTOME_G1_S_TRANSITION","REACTOME_FCERI_MEDIATED_NF_KB_ACTIVATION","REACTOME_DEGRADATION_OF_AXIN","REACTOME_DOWNSTREAM_SIGNALING_EVENTS_OF_B_CELL_RECEPTOR_BCR","REACTOME_G1_S_TRANSITION","REACTOME_ACTIVATION_OF_NF_KAPPAB_IN_B_CELLS","REACTOME_REGULATION_OF_RUNX3_EXPRESSION_AND_ACTIVITY","REACTOME_HEDGEHOG_LIGAND_BIOGENESIS","REACTOME_KEAP1_NFE2L2_PATHWAY","REACTOME_TELOMERE_MAINTENANCE","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX2","REACTOME_REGULATION_OF_RUNX2_EXPRESSION_AND_ACTIVITY","REACTOME_UBIQUITIN_MEDIATED_DEGRADATION_OF_PHOSPHORYLATED_CDC25A","GO:0006735","REACTOME_CYCLIN_A_CDK2_ASSOCIATED_EVENTS_AT_S_PHASE_ENTRY","REACTOME_HOMOLOGY_DIRECTED_REPAIR","GO:0043620","REACTOME_CLEC7A_DECTIN_1_SIGNALING","REACTOME_CROSS_PRESENTATION_OF_SOLUBLE_EXOGENOUS_ANTIGENS_ENDOSOMES","REACTOME_FC_EPSILON_RECEPTOR_FCERI_SIGNALING","REACTOME_RHO_GTPASE_EFFECTORS","REACTOME_LAGGING_STRAND_SYNTHESIS","REACTOME_DOWNSTREAM_SIGNALING_EVENTS_OF_B_CELL_RECEPTOR_BCR","REACTOME_AUF1_HNRNP_D0_BINDS_AND_DESTABILIZES_MRNA","REACTOME_SCF_SKP2_MEDIATED_DEGRADATION_OF_P27_P21","REACTOME_DEPOSITION_OF_NEW_CENPA_CONTAINING_NUCLEOSOMES_AT_THE_CENTROMERE","REACTOME_THE_ROLE_OF_GTSE1_IN_G2_M_PROGRESSION_AFTER_G2_CHECKPOINT","REACTOME_RESPIRATORY_ELECTRON_TRANSPORT","GO:0043254","GO:0061620","REACTOME_ACTIVATION_OF_ATR_IN_RESPONSE_TO_REPLICATION_STRESS","REACTOME_SCF_SKP2_MEDIATED_DEGRADATION_OF_P27_P21","REACTOME_METABOLISM_OF_POLYAMINES","REACTOME_AMPLIFICATION_OF_SIGNAL_FROM_THE_KINETOCHORES","REACTOME_APC_C_CDH1_MEDIATED_DEGRADATION_OF_CDC20_AND_OTHER_APC_C_CDH1_TARGETED_PROTEINS_IN_LATE_MITOSIS_EARLY_G1","REACTOME_MITOTIC_SPINDLE_CHECKPOINT","REACTOME_CELLULAR_RESPONSE_TO_CHEMICAL_STRESS","REACTOME_RHO_GTPASES_ACTIVATE_FORMINS","REACTOME_MITOTIC_SPINDLE_CHECKPOINT","REACTOME_RESPIRATORY_ELECTRON_TRANSPORT_ATP_SYNTHESIS_BY_CHEMIOSMOTIC_COUPLING_AND_HEAT_PRODUCTION_BY_UNCOUPLING_PROTEINS","REACTOME_PCNA_DEPENDENT_LONG_PATCH_BASE_EXCISION_REPAIR","REACTOME_AMPLIFICATION_OF_SIGNAL_FROM_THE_KINETOCHORES","REACTOME_RESOLUTION_OF_AP_SITES_VIA_THE_MULTIPLE_NUCLEOTIDE_PATCH_REPLACEMENT_PATHWAY","REACTOME_G2_M_CHECKPOINTS","REACTOME_THE_CITRIC_ACID_TCA_CYCLE_AND_RESPIRATORY_ELECTRON_TRANSPORT","REACTOME_RHO_GTPASES_ACTIVATE_FORMINS","REACTOME_MRNA_SPLICING","REACTOME_DNA_REPLICATION","REACTOME_ACTIVATION_OF_APC_C_AND_APC_C_CDC20_MEDIATED_DEGRADATION_OF_MITOTIC_PROTEINS","REACTOME_APC_C_MEDIATED_DEGRADATION_OF_CELL_CYCLE_PROTEINS","REACTOME_THE_ROLE_OF_GTSE1_IN_G2_M_PROGRESSION_AFTER_G2_CHECKPOINT","REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA","REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA","REACTOME_MITOTIC_G2_G2_M_PHASES","REACTOME_MRNA_SPLICING","REACTOME_SEPARATION_OF_SISTER_CHROMATIDS","GO:0009141","REACTOME_ACTIVATION_OF_THE_MRNA_UPON_BINDING_OF_THE_CAP_BINDING_COMPLEX_AND_EIFS_AND_SUBSEQUENT_BINDING_TO_43S","GO:0000375","GO:0000377","GO:0000398","REACTOME_MITOTIC_PROMETAPHASE","REACTOME_S_PHASE","GO:0007059","GO:0140014","REACTOME_MITOTIC_PROMETAPHASE","REACTOME_RESOLUTION_OF_SISTER_CHROMATID_COHESION","REACTOME_RESOLUTION_OF_SISTER_CHROMATID_COHESION","REACTOME_CELL_CYCLE_CHECKPOINTS","GO:0006091","REACTOME_CHROMOSOME_MAINTENANCE","REACTOME_MAJOR_PATHWAY_OF_RRNA_PROCESSING_IN_THE_NUCLEOLUS_AND_CYTOSOL","REACTOME_TRANSLATION","REACTOME_MITOTIC_METAPHASE_AND_ANAPHASE","REACTOME_M_PHASE","REACTOME_MAJOR_PATHWAY_OF_RRNA_PROCESSING_IN_THE_NUCLEOLUS_AND_CYTOSOL","REACTOME_CELL_CYCLE_CHECKPOINTS","REACTOME_NONSENSE_MEDIATED_DECAY_NMD","REACTOME_ACTIVATION_OF_THE_MRNA_UPON_BINDING_OF_THE_CAP_BINDING_COMPLEX_AND_EIFS_AND_SUBSEQUENT_BINDING_TO_43S","REACTOME_SEPARATION_OF_SISTER_CHROMATIDS","GO:0002181","REACTOME_SRP_DEPENDENT_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE","REACTOME_NONSENSE_MEDIATED_DECAY_NMD_INDEPENDENT_OF_THE_EXON_JUNCTION_COMPLEX_EJC","REACTOME_FORMATION_OF_A_POOL_OF_FREE_40S_SUBUNITS","GO:0006260","REACTOME_EUKARYOTIC_TRANSLATION_INITIATION","GO:0007059","REACTOME_TRANSLATION","REACTOME_MITOTIC_METAPHASE_AND_ANAPHASE","REACTOME_M_PHASE","REACTOME_NONSENSE_MEDIATED_DECAY_NMD","REACTOME_SRP_DEPENDENT_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE","REACTOME_NONSENSE_MEDIATED_DECAY_NMD_INDEPENDENT_OF_THE_EXON_JUNCTION_COMPLEX_EJC","REACTOME_FORMATION_OF_A_POOL_OF_FREE_40S_SUBUNITS","GO:0002181","REACTOME_EUKARYOTIC_TRANSLATION_INITIATION"],["Response To Elevated Platelet Cytosolic Ca2","GO:Alanine Transport","Cyclin A B1 B2 Associated Events During G2 M Transition","Switching Of Origins To A Post Replicative State","Deubiquitination","Apoptosis Induced Dna Fragmentation","Depolymerisation Of The Nuclear Lamina","Dissolution Of Fibrin Clot","Mhc Class Ii Antigen Presentation","Antigen Processing Ubiquitination Proteasome Degradation","Mapk Family Signaling Cascades","Transcriptional Regulation By Tp53","Deadenylation Dependent Mrna Decay","Processive Synthesis On The C Strand Of The Telomere","Transport Of Connexons To The Plasma Membrane","GO:Pyruvate Metabolic Process","Pink1 Prkn Mediated Mitophagy","Selective Autophagy","Downregulation Of Erbb4 Signaling","Regulation Of Pten Localization","GO:Atp Metabolic Process","Intrinsic Pathway For Apoptosis","Citric Acid Cycle Tca Cycle","Gap Junction Trafficking And Regulation","Apc Cdc20 Mediated Degradation Of Nek2a","Dissolution Of Fibrin Clot","Regulation Of Insulin Like Growth Factor Igf Transport And Uptake By Insulin Like Growth Factor Binding Proteins Igfbps","Insulin Receptor Recycling","Sealing Of The Nuclear Envelope Ne By Escrt Iii","Apc C Mediated Degradation Of Cell Cycle Proteins","Caspase Mediated Cleavage Of Cytoskeletal Proteins","Copi Dependent Golgi To Er Retrograde Traffic","Smooth Muscle Contraction","Slbp Dependent Processing Of Replication Dependent Histone Pre Mrnas","Neddylation","Regulation Of Mrna Stability By Proteins That Bind Au Rich Elements","Mitotic Metaphase And Anaphase","M Phase","Tetrahydrobiopterin Bh4 Synthesis Recycling Salvage And Regulation","Downstream Tcr Signaling","Pkmts Methylate Histone Lysines","GO:Positive Regulation Of Binding","Signaling By Interleukins","Metabolism Of Amino Acids And Derivatives","G2 M Checkpoints","Intraflagellar Transport","Antigen Processing Ubiquitination Proteasome Degradation","GO:Nucleosome Assembly","Assembly Of The Pre Replicative Complex","GO:Purine Nucleoside Triphosphate Metabolic Process","Intra Golgi And Retrograde Golgi To Er Traffic","Pexophagy","Programmed Cell Death","Detoxification Of Reactive Oxygen Species","Tetrahydrobiopterin Bh4 Synthesis Recycling Salvage And Regulation","Apc C Cdc20 Mediated Degradation Of Cyclin B","Inhibition Of The Proteolytic Activity Of Apc C Required For The Onset Of Anaphase By Mitotic Spindle Checkpoint Components","Hsp90 Chaperone Cycle For Steroid Hormone Receptors Shr In The Presence Of Ligand","Copi Dependent Golgi To Er Retrograde Traffic","GO:Regulation Of Binding","Mitotic G1 Phase And G1 S Transition","Map2k And Mapk Activation","Degradation Of Beta Catenin By The Destruction Complex","Signaling By Wnt","GO:Transition Metal Ion Homeostasis","Aggrephagy","Response To Elevated Platelet Cytosolic Ca2","Tetrahydrobiopterin Bh4 Synthesis Recycling Salvage And Regulation","GO:Positive Regulation Of Epithelial Cell Migration","Cytokine Signaling In Immune System","Rho Gtpases Activate Pkns","Downregulation Of Erbb4 Signaling","Regulation Of Pten Localization","Transcriptional Regulation By Runx2","Fceri Mediated Nf Kb Activation","GO:Response To Muscle Stretch","Apoptotic Cleavage Of Cellular Proteins","Activation Of Apc C And Apc C Cdc20 Mediated Degradation Of Mitotic Proteins","GO:Transition Metal Ion Homeostasis","Rho Gtpases Activate Ktn1","Pyroptosis","Tcf Dependent Signaling In Response To Wnt","GO:Cellular Response To Interleukin-7","Asparagine N Linked Glycosylation","Iron Uptake And Transport","Selective Autophagy","Mapk Family Signaling Cascades","Transcriptional Regulation By E2f6","Activation Of Bh3 Only Proteins","Metabolism Of Amino Acids And Derivatives","Copi Independent Golgi To Er Retrograde Traffic","Gap Junction Assembly","Apc C Cdh1 Mediated Degradation Of Cdc20 And Other Apc C Cdh1 Targeted Proteins In Late Mitosis Early G1","S Phase","GO:Viral Process","GO:Negative Regulation Of Viral Process","Cdk Mediated Phosphorylation And Removal Of Cdc6","Glucose Metabolism","P75ntr Signals Via Nf Kb","GO:Regulation Of Ubiquitin-Protein Transferase Activity","Sumoylation Of Dna Damage Response And Repair Proteins","Condensation Of Prophase Chromosomes","Cytokine Signaling In Immune System","Apoptosis","Orc1 Removal From Chromatin","GO:Neuron Death","GO:Endothelial Cell Activation","Regulated Necrosis","Cellular Response To Hypoxia","Metabolism Of Nucleotides","Cellular Responses To Stimuli","Map2k And Mapk Activation","Inhibition Of The Proteolytic Activity Of Apc C Required For The Onset Of Anaphase By Mitotic Spindle Checkpoint Components","Platelet Activation Signaling And Aggregation","Depolymerisation Of The Nuclear Lamina","GO:Regulation Of Epithelial Cell Proliferation","GO:Positive Regulation Of Epithelial Cell Migration","Pten Regulation","Runx1 Regulates Transcription Of Genes Involved In Differentiation Of Hscs","Cell Cell Communication","Intra Golgi And Retrograde Golgi To Er Traffic","Pcp Ce Pathway","Sumoylation Of Dna Replication Proteins","Activation Of Nf Kappab In B Cells","Cell Junction Organization","Response To Elevated Platelet Cytosolic Ca2","Ros And Rns Production In Phagocytes","Organelle Biogenesis And Maintenance","Eph Ephrin Signaling","Metabolism Of Carbohydrates","Chromatin Modifying Enzymes","Establishment Of Sister Chromatid Cohesion","Slbp Dependent Processing Of Replication Dependent Histone Pre Mrnas","Anchoring Of The Basal Body To The Plasma Membrane","Nf Kb Is Activated And Signals Survival","Nervous System Development","Rho Gtpases Activate Iqgaps","Intrinsic Pathway For Apoptosis","G1 S Transition","Interconversion Of Nucleotide Di And Triphosphates","Asymmetric Localization Of Pcp Proteins","Signaling By The B Cell Receptor Bcr","Hedgehog On State","Mrna Decay By 5 To 3 Exoribonuclease","GO:Mitotic Cytokinetic Process","Factors Involved In Megakaryocyte Development And Platelet Production","Deadenylation Dependent Mrna Decay","Beta Catenin Independent Wnt Signaling","Transcriptional Regulation By Runx1","Processing And Activation Of Sumo","Abc Family Proteins Mediated Transport","GO:Regulation Of Ubiquitin-Protein Transferase Activity","Transcriptional Regulation By Runx3","Gli3 Is Processed To Gli3r By The Proteasome","Oxidative Stress Induced Senescence","Degradation Of Dvl","Tnfr2 Non Canonical Nf Kb Pathway","Uch Proteinases","Aggrephagy","Intrinsic Pathway For Apoptosis","Stabilization Of P53","GO:Atp Metabolic Process","GO:Nadh Regeneration","Apoptosis","GO:Response To Oxidative Stress","Degradation Of Axin","GO:Nucleic Acid Transport","GO:Rna Transport","Factors Involved In Megakaryocyte Development And Platelet Production","Regulation Of Runx3 Expression And Activity","Clec7a Dectin 1 Signaling","Recycling Pathway Of L1","Transferrin Endocytosis And Recycling","Tp53 Regulates Metabolic Genes","Mrna Splicing Minor Pathway","Formation Of Senescence Associated Heterochromatin Foci Sahf","Rho Gtpases Activate Paks","Antigen Processing Cross Presentation","C Type Lectin Receptors Clrs","Regulation Of Runx2 Expression And Activity","Ubiquitin Mediated Degradation Of Phosphorylated Cdc25a","Regulation Of Pten Stability And Activity","Mitotic G2 G2 M Phases","GO:Response To Inorganic Substance","GO:Amyloid Fibril Formation","GO:Regulation Of Peptidase Activity","Golgi To Er Retrograde Transport","Pexophagy","Regulation Of Ras By Gaps","Hedgehog Off State","Detoxification Of Reactive Oxygen Species","Rho Gtpases Activate Iqgaps","Runx1 Regulates Genes Involved In Megakaryocyte Differentiation And Platelet Function","Cross Presentation Of Soluble Exogenous Antigens Endosomes","Regulation Of Mrna Stability By Proteins That Bind Au Rich Elements","Developmental Biology","G1 S Dna Damage Checkpoints","Fcgamma Receptor Fcgr Dependent Phagocytosis","Selective Autophagy","Programmed Cell Death","Orc1 Removal From Chromatin","Mitotic G2 G2 M Phases","GO:Response To Hypoxia","GO:Regulation Of Hemopoiesis","GO:Response To Decreased Oxygen Levels","GO:Positive Regulation Of Neuron Death","Hedgehog Ligand Biogenesis","Deposition Of New Cenpa Containing Nucleosomes At The Centromere","Apoptosis","Kinesins","Transcriptional Regulation By Runx2","GO:Regulation Of Epithelial Cell Migration","Ub Specific Processing Proteases","Nuclear Envelope Breakdown","GO:Purine Ribonucleoside Triphosphate Metabolic Process","GO:Purine Nucleoside Triphosphate Metabolic Process","GO:Ribonucleoside Triphosphate Metabolic Process","GO:Positive Regulation Of Neuron Death","Tcf Dependent Signaling In Response To Wnt","Fcgamma Receptor Fcgr Dependent Phagocytosis","Cell Extracellular Matrix Interactions","Downstream Signaling Events Of B Cell Receptor Bcr","Fc Epsilon Receptor Fceri Signaling","GO:Intrinsic Apoptotic Signaling Pathway","Copi Dependent Golgi To Er Retrograde Traffic","Signaling By Hedgehog","Dectin 1 Mediated Noncanonical Nf Kb Signaling","Keap1 Nfe2l2 Pathway","GO:Ribosomal Small Subunit Assembly","Processive Synthesis On The Lagging Strand","Nervous System Development","Mapk6 Mapk4 Signaling","Insulin Receptor Recycling","GO:Regulation Of Tumor Necrosis Factor Production","GO:Regulation Of Glycolytic Process","Gluconeogenesis","Rho Gtpase Effectors","Transport Of Mature Transcript To Cytoplasm","GO:Atp Metabolic Process","Cyclin A Cdk2 Associated Events At S Phase Entry","GO:Regulation Of Glycolytic Process","GO:Regulation Of Peptidase Activity","GO:Regulation Of Dna-Templated Transcription In Response To Stress","Mitochondrial Translation","Intracellular Signaling By Second Messengers","GO:Regulation Of Inflammatory Response","Signaling By Wnt","Chromosome Maintenance","GO:Response To Inorganic Substance","Rho Gtpases Activate Iqgaps","Golgi To Er Retrograde Transport","Dual Incision In Tc Ner","Glucose Metabolism","Deubiquitination","GO:Regulation Of Dna-Templated Transcription In Response To Stress","Transcription Coupled Nucleotide Excision Repair Tc Ner","Metabolism Of Amino Acids And Derivatives","Ephb Mediated Forward Signaling","Recycling Pathway Of L1","L1cam Interactions","The Role Of Gtse1 In G2 M Progression After G2 Checkpoint","Transport Of Connexons To The Plasma Membrane","Apoptosis","GO:Transmembrane Receptor Protein Serine/Threonine Kinase Signaling Pathway","Dna Damage Telomere Stress Induced Senescence","GO:Response To Wounding","Signaling By Interleukins","Mrna Splicing Minor Pathway","Complex I Biogenesis","Transcriptional Regulation By Runx1","GO:Cellular Response To External Stimulus","GO:Regulation Of Actin Filament-Based Process","GO:Regulation Of Protein-Containing Complex Assembly","Rmts Methylate Histone Arginines","Scf Skp2 Mediated Degradation Of P27 P21","Rho Gtpases Activate Wasps And Waves","Cellular Responses To Stimuli","Glycolysis","Hdr Through Single Strand Annealing Ssa","Aggrephagy","Regulation Of Apc C Activators Between G1 S And Early Anaphase","Metabolism Of Nucleotides","Programmed Cell Death","Polymerase Switching On The C Strand Of The Telomere","Hdr Through Homologous Recombination Hrr","Transcriptional Regulation By Runx1","GO:Mitochondrial Respiratory Chain Complex Assembly","Mrna Splicing Minor Pathway","Auf1 Hnrnp D0 Binds And Destabilizes Mrna","Pten Regulation","Mrna Decay By 5 To 3 Exoribonuclease","GO:Positive Regulation Of Neuron Death","Switching Of Origins To A Post Replicative State","GO:Response To Inorganic Substance","Cellular Responses To Stimuli","Ub Specific Processing Proteases","Activation Of Apc C And Apc C Cdc20 Mediated Degradation Of Mitotic Proteins","Neddylation","Mitotic Telophase Cytokinesis","GO:Viral Life Cycle","Recruitment Of Numa To Mitotic Centrosomes","Pcp Ce Pathway","Mhc Class Ii Antigen Presentation","Eph Ephrin Signaling","Tcr Signaling","Mitotic Prophase","Translesion Synthesis By Y Family Dna Polymerases Bypasses Lesions On Dna Template","Apoptotic Execution Phase","Interleukin 1 Family Signaling","Mrna Decay By 5 To 3 Exoribonuclease","Abc Family Proteins Mediated Transport","Interconversion Of Nucleotide Di And Triphosphates","GO:Constitutive Heterochromatin Formation","Assembly Of The Pre Replicative Complex","Rho Gtpase Effectors","Cellular Response To Chemical Stress","Cellular Response To Chemical Stress","Hedgehog On State","Tnfr2 Non Canonical Nf Kb Pathway","Uch Proteinases","GO:Skeletal Muscle Cell Differentiation","Nucleotide Excision Repair","Metabolism Of Polyamines","Degradation Of Beta Catenin By The Destruction Complex","Transport Of Mature Transcript To Cytoplasm","Global Genome Nucleotide Excision Repair Gg Ner","Regulation Of Apc C Activators Between G1 S And Early Anaphase","Translesion Synthesis By Polh","Initiation Of Nuclear Envelope Ne Reformation","GO:Deoxyribose Phosphate Biosynthetic Process","Keap1 Nfe2l2 Pathway","GO:Skeletal Muscle Cell Differentiation","Gluconeogenesis","Beta Catenin Independent Wnt Signaling","Rho Gtpases Activate Iqgaps","Mapk6 Mapk4 Signaling","Condensation Of Prophase Chromosomes","Cell Cycle Checkpoints","Tp53 Regulates Metabolic Genes","Signaling By Hedgehog","Apc C Mediated Degradation Of Cell Cycle Proteins","The Role Of Gtse1 In G2 M Progression After G2 Checkpoint","Cellular Responses To Stimuli","Regulation Of Tp53 Activity","Translesion Synthesis By Polk","Dual Incision In Gg Ner","Termination Of Translesion Dna Synthesis","Kinesins","GO:Protein Localization To Chromosome","Antigen Processing Cross Presentation","Downstream Tcr Signaling","Regulation Of Plk1 Activity At G2 M Transition","GO:Non-Membrane-Bounded Organelle Assembly","Regulation Of Pten Stability And Activity","GO:Regulation Of Cell Morphogenesis","Sumoylation","Transcriptional Regulation By Tp53","Cellular Senescence","Pten Regulation","Dna Damage Bypass","Mitotic Prophase","Runx1 Regulates Transcription Of Genes Involved In Differentiation Of Hscs","Gap Filling Dna Repair Synthesis And Ligation In Gg Ner","Rho Gtpases Activate Wasps And Waves","Mitotic G2 G2 M Phases","Regulation Of Tp53 Activity Through Phosphorylation","Activation Of Nf Kappab In B Cells","GO:Rna Localization","Fceri Mediated Nf Kb Activation","Transcriptional Regulation By Runx3","Hedgehog Ligand Biogenesis","GO:Response To Uv","G2 M Dna Damage Checkpoint","Nuclear Envelope Ne Reassembly","Signaling By Hedgehog","Asymmetric Localization Of Pcp Proteins","Rna Polymerase Ii Transcription Termination","Processing Of Dna Double Strand Break Ends","Polymerase Switching","Ephb Mediated Forward Signaling","M Phase","Processing Of Capped Intron Containing Pre Mrna","Metabolism Of Nucleotides","Orc1 Removal From Chromatin","Dectin 1 Mediated Noncanonical Nf Kb Signaling","Gli3 Is Processed To Gli3r By The Proteasome","Regulation Of Mrna Stability By Proteins That Bind Au Rich Elements","GO:Rna Localization","Aurka Activation By Tpx2","Cellular Response To Hypoxia","Degradation Of Dvl","Assembly Of The Pre Replicative Complex","Stabilization Of P53","Tnfr2 Non Canonical Nf Kb Pathway","Telomere C Strand Lagging Strand Synthesis","Degradation Of Axin","Hedgehog On State","G2 M Checkpoints","G2 M Checkpoints","Regulation Of Runx3 Expression And Activity","Dna Replication","G1 S Dna Damage Checkpoints","Apoptosis Induced Dna Fragmentation","Degradation Of Beta Catenin By The Destruction Complex","Dna Replication","Clec7a Dectin 1 Signaling","Hedgehog Off State","Initiation Of Nuclear Envelope Ne Reformation","Mrna Splicing","Switching Of Origins To A Post Replicative State","GO:Non-Membrane-Bounded Organelle Assembly","Rna Polymerase Ii Transcription Termination","Regulation Of Runx2 Expression And Activity","Ubiquitin Mediated Degradation Of Phosphorylated Cdc25a","Hedgehog Off State","Recognition Of Dna Damage By Pcna Containing Replication Complex","GO:Protein Localization To Endoplasmic Reticulum","Signaling By The B Cell Receptor Bcr","Dna Replication Pre Initiation","Interleukin 1 Family Signaling","Tcr Signaling","Cellular Responses To Stimuli","Cdk Mediated Phosphorylation And Removal Of Cdc6","Separation Of Sister Chromatids","Mapk6 Mapk4 Signaling","Cross Presentation Of Soluble Exogenous Antigens Endosomes","Interconversion Of Nucleotide Di And Triphosphates","Apc C Mediated Degradation Of Cell Cycle Proteins","Interconversion Of Nucleotide Di And Triphosphates","Cdk Mediated Phosphorylation And Removal Of Cdc6","Mitotic G1 Phase And G1 S Transition","Dna Replication Pre Initiation","Abc Family Proteins Mediated Transport","S Phase","Mitotic G1 Phase And G1 S Transition","C Type Lectin Receptors Clrs","GO:Skeletal Muscle Cell Differentiation","Mitotic G1 Phase And G1 S Transition","G1 S Transition","Dna Replication Pre Initiation","Uch Proteinases","Antigen Processing Cross Presentation","Auf1 Hnrnp D0 Binds And Destabilizes Mrna","Regulation Of Pten Stability And Activity","Prc2 Methylates Histones And Dna","Regulation Of Ras By Gaps","Beta Catenin Independent Wnt Signaling","Cellular Responses To Stimuli","Regulation Of Ras By Gaps","Runx1 Regulates Transcription Of Genes Involved In Differentiation Of Hscs","Nuclear Envelope Ne Reassembly","Fc Epsilon Receptor Fceri Signaling","G1 S Dna Damage Checkpoints","Switching Of Origins To A Post Replicative State","GO:Regulation Of Dna-Templated Transcription In Response To Stress","Transcriptional Regulation By Runx3","Epigenetic Regulation Of Gene Expression","Activation Of Apc C And Apc C Cdc20 Mediated Degradation Of Mitotic Proteins","Signaling By The B Cell Receptor Bcr","Transcriptional Regulation By Runx2","Activation Of The Pre Replicative Complex","Dna Repair","Base Excision Repair","Dna Double Strand Break Repair","S Phase","Rho Gtpase Effectors","Pcp Ce Pathway","Apc C Cdh1 Mediated Degradation Of Cdc20 And Other Apc C Cdh1 Targeted Proteins In Late Mitosis Early G1","Asymmetric Localization Of Pcp Proteins","Dna Strand Elongation","Cyclin A Cdk2 Associated Events At S Phase Entry","Downstream Tcr Signaling","Regulation Of Mrna Stability By Proteins That Bind Au Rich Elements","C Type Lectin Receptors Clrs","Orc1 Removal From Chromatin","Assembly Of The Pre Replicative Complex","GO:Response To Endoplasmic Reticulum Stress","Cellular Response To Hypoxia","Metabolism Of Polyamines","Dectin 1 Mediated Noncanonical Nf Kb Signaling","Gli3 Is Processed To Gli3r By The Proteasome","Mitotic Metaphase And Anaphase","Degradation Of Dvl","Stabilization Of P53","G1 S Transition","Fceri Mediated Nf Kb Activation","Degradation Of Axin","Downstream Signaling Events Of B Cell Receptor Bcr","G1 S Transition","Activation Of Nf Kappab In B Cells","Regulation Of Runx3 Expression And Activity","Hedgehog Ligand Biogenesis","Keap1 Nfe2l2 Pathway","Telomere Maintenance","Transcriptional Regulation By Runx2","Regulation Of Runx2 Expression And Activity","Ubiquitin Mediated Degradation Of Phosphorylated Cdc25a","GO:Nadh Regeneration","Cyclin A Cdk2 Associated Events At S Phase Entry","Homology Directed Repair","GO:Regulation Of Dna-Templated Transcription In Response To Stress","Clec7a Dectin 1 Signaling","Cross Presentation Of Soluble Exogenous Antigens Endosomes","Fc Epsilon Receptor Fceri Signaling","Rho Gtpase Effectors","Lagging Strand Synthesis","Downstream Signaling Events Of B Cell Receptor Bcr","Auf1 Hnrnp D0 Binds And Destabilizes Mrna","Scf Skp2 Mediated Degradation Of P27 P21","Deposition Of New Cenpa Containing Nucleosomes At The Centromere","The Role Of Gtse1 In G2 M Progression After G2 Checkpoint","Respiratory Electron Transport","GO:Regulation Of Protein-Containing Complex Assembly","GO:Glycolytic Process Through Glucose-6-Phosphate","Activation Of Atr In Response To Replication Stress","Scf Skp2 Mediated Degradation Of P27 P21","Metabolism Of Polyamines","Amplification Of Signal From The Kinetochores","Apc C Cdh1 Mediated Degradation Of Cdc20 And Other Apc C Cdh1 Targeted Proteins In Late Mitosis Early G1","Mitotic Spindle Checkpoint","Cellular Response To Chemical Stress","Rho Gtpases Activate Formins","Mitotic Spindle Checkpoint","Respiratory Electron Transport Atp Synthesis By Chemiosmotic Coupling And Heat Production By Uncoupling Proteins","Pcna Dependent Long Patch Base Excision Repair","Amplification Of Signal From The Kinetochores","Resolution Of Ap Sites Via The Multiple Nucleotide Patch Replacement Pathway","G2 M Checkpoints","The Citric Acid Tca Cycle And Respiratory Electron Transport","Rho Gtpases Activate Formins","Mrna Splicing","Dna Replication","Activation Of Apc C And Apc C Cdc20 Mediated Degradation Of Mitotic Proteins","Apc C Mediated Degradation Of Cell Cycle Proteins","The Role Of Gtse1 In G2 M Progression After G2 Checkpoint","Processing Of Capped Intron Containing Pre Mrna","Processing Of Capped Intron Containing Pre Mrna","Mitotic G2 G2 M Phases","Mrna Splicing","Separation Of Sister Chromatids","GO:Nucleoside Triphosphate Metabolic Process","Activation Of The Mrna Upon Binding Of The Cap Binding Complex And Eifs And Subsequent Binding To 43s","GO:Rna Splicing, Via Transesterification Reactions","GO:Rna Splicing, Via Transesterification Reactions With Bulged Adenosine As Nucleophile","GO:Mrna Splicing, Via Spliceosome","Mitotic Prometaphase","S Phase","GO:Chromosome Segregation","GO:Mitotic Nuclear Division","Mitotic Prometaphase","Resolution Of Sister Chromatid Cohesion","Resolution Of Sister Chromatid Cohesion","Cell Cycle Checkpoints","GO:Generation Of Precursor Metabolites And Energy","Chromosome Maintenance","Major Pathway Of Rrna Processing In The Nucleolus And Cytosol","Translation","Mitotic Metaphase And Anaphase","M Phase","Major Pathway Of Rrna Processing In The Nucleolus And Cytosol","Cell Cycle Checkpoints","Nonsense Mediated Decay Nmd","Activation Of The Mrna Upon Binding Of The Cap Binding Complex And Eifs And Subsequent Binding To 43s","Separation Of Sister Chromatids","GO:Cytoplasmic Translation","Srp Dependent Cotranslational Protein Targeting To Membrane","Nonsense Mediated Decay Nmd Independent Of The Exon Junction Complex Ejc","Formation Of A Pool Of Free 40s Subunits","GO:Dna Replication","Eukaryotic Translation Initiation","GO:Chromosome Segregation","Translation","Mitotic Metaphase And Anaphase","M Phase","Nonsense Mediated Decay Nmd","Srp Dependent Cotranslational Protein Targeting To Membrane","Nonsense Mediated Decay Nmd Independent Of The Exon Junction Complex Ejc","Formation Of A Pool Of Free 40s Subunits","GO:Cytoplasmic Translation","Eukaryotic Translation Initiation"],["7/90","3/207","4/204","7/167","14/204","3/204","3/204","3/167","10/282","16/209","16/209","16/209","7/282","4/282","4/282","7/207","4/125","6/125","3/125","3/125","9/139","5/204","4/209","5/167","5/282","4/244","7/90","4/167","4/167","7/167","3/167","10/244","6/244","3/204","12/167","7/125","12/167","17/167","3/209","7/167","8/282","11/292","15/204","13/167","10/167","6/204","16/204","5/156","7/167","10/139","12/204","3/90","10/209","5/209","3/204","5/282","4/204","6/204","10/282","16/230","9/167","5/167","7/167","14/204","8/207","6/282","9/167","3/167","6/86","20/204","4/167","3/90","3/90","8/282","7/167","4/131","5/167","7/167","7/146","4/244","4/167","11/204","4/292","19/244","8/244","7/209","15/167","5/282","4/209","15/204","6/204","5/204","7/167","10/167","12/131","7/131","7/167","8/209","4/125","6/288","9/282","7/204","21/209","9/204","7/167","13/146","3/156","6/167","7/167","9/209","18/125","7/244","5/282","19/244","4/282","13/146","8/146","9/167","7/167","8/167","16/244","8/167","7/282","7/167","7/167","13/244","7/244","14/204","7/167","20/244","17/282","4/282","4/282","9/204","4/125","20/244","6/282","6/209","9/167","5/167","7/167","9/167","8/167","4/209","5/357","13/282","7/204","10/167","17/282","4/282","9/167","7/369","8/167","7/167","7/282","7/167","9/167","9/167","6/209","6/167","7/167","10/146","4/146","10/209","19/292","7/167","11/292","11/292","11/204","7/167","9/167","6/204","7/244","8/167","7/209","5/282","5/167","8/167","10/167","7/167","7/167","8/167","13/167","16/139","5/139","15/131","14/244","4/125","8/167","10/167","6/167","7/244","11/282","7/167","9/167","22/167","8/167","11/244","8/204","14/282","10/282","17/282","11/146","15/146","11/146","7/86","8/167","9/204","10/167","9/282","8/167","11/156","14/204","7/204","9/86","9/86","9/86","10/207","13/209","9/167","5/167","9/167","11/167","15/207","10/204","12/167","8/167","9/167","5/207","5/282","17/167","9/167","7/244","10/156","6/156","8/244","21/244","11/282","9/86","9/167","7/207","20/230","7/131","10/209","18/209","14/156","17/209","12/204","13/86","6/204","12/204","10/282","12/244","18/209","6/86","11/282","18/209","7/167","7/167","9/167","11/282","5/204","13/282","15/156","7/282","22/230","19/209","9/282","8/209","16/209","19/357","21/230","21/230","11/282","9/167","7/167","32/244","11/244","8/282","7/204","7/204","11/204","13/167","7/282","10/282","16/204","10/292","8/204","9/167","12/204","5/204","10/156","13/282","18/146","33/282","16/209","12/282","18/209","5/204","19/357","11/204","11/204","12/204","12/244","12/204","13/204","9/282","10/282","13/204","6/282","12/204","7/209","5/369","13/282","19/167","15/167","15/204","11/204","12/204","12/204","10/207","15/282","10/167","11/204","11/204","13/282","9/282","7/282","6/204","6/288","11/204","8/86","8/209","14/204","8/167","11/204","12/282","22/209","11/209","15/204","14/282","12/167","28/167","19/282","7/282","10/282","9/282","10/204","13/369","11/204","12/204","12/204","25/369","11/204","24/357","20/282","28/282","15/282","14/209","11/282","17/282","11/204","8/282","10/244","18/209","15/282","11/204","17/369","12/204","12/204","11/204","16/369","15/282","13/282","16/209","11/204","13/282","15/282","7/282","11/244","28/209","21/209","16/282","12/204","11/204","11/204","13/204","16/288","12/204","12/204","11/204","13/204","11/204","14/209","10/282","11/204","13/209","18/204","18/209","11/204","16/204","12/204","7/282","13/209","16/209","14/204","15/204","8/282","19/209","14/204","24/288","12/204","11/204","11/204","15/209","10/282","13/357","15/204","17/282","16/209","15/209","32/204","13/204","20/209","13/209","11/204","10/282","14/209","9/204","13/209","17/204","15/204","15/209","18/209","17/209","16/204","11/156","20/282","18/282","15/209","15/209","13/209","12/204","13/209","16/282","13/204","17/209","33/209","13/209","13/209","13/204","17/204","13/209","15/209","10/146","14/209","19/282","14/209","16/209","13/204","11/282","32/282","12/282","23/282","19/204","29/282","15/209","14/209","13/209","10/282","14/204","15/209","15/209","17/209","14/209","15/209","25/357","14/209","13/204","13/209","13/209","24/209","13/209","13/209","17/209","15/209","13/209","15/204","17/204","14/209","13/209","14/209","15/209","14/282","14/209","13/209","13/209","9/357","15/209","22/282","11/156","17/209","13/209","19/209","26/204","10/282","16/209","14/209","14/204","18/282","16/209","17/209","35/357","10/357","13/282","15/209","15/209","20/282","16/204","19/204","23/209","24/282","22/282","20/209","11/282","18/204","12/282","27/282","25/209","23/204","26/204","26/282","19/204","20/204","19/204","36/282","31/204","28/204","32/282","33/282","30/292","14/90","31/288","31/288","31/288","35/282","31/282","36/288","33/288","31/204","26/204","30/282","38/204","42/292","32/282","23/90","26/90","43/282","57/282","28/125","50/282","22/90","20/125","37/204","24/139","22/90","22/90","23/90","45/369","25/90","51/369","35/125","44/204","55/204","29/125","28/125","29/125","30/125","34/207","33/125"],["128/8764","14/18646","25/8764","92/8764","254/8764","12/8764","12/8764","13/8764","104/8764","297/8764","297/8764","297/8764","55/8764","17/8764","17/8764","120/18646","23/8764","64/8764","10/8764","10/8764","213/18646","39/8764","22/8764","44/8764","27/8764","13/8764","121/8764","26/8764","26/8764","87/8764","12/8764","96/8764","34/8764","11/8764","226/8764","85/8764","225/8764","389/8764","10/8764","85/8764","67/8764","191/18646","270/8764","253/8764","162/8764","54/8764","297/8764","72/18646","83/8764","244/18646","187/8764","11/8764","132/8764","34/8764","10/8764","25/8764","21/8764","52/8764","96/8764","401/18646","130/8764","39/8764","80/8764","235/8764","134/18646","36/8764","128/8764","10/8764","159/18646","402/8764","22/8764","10/8764","10/8764","63/8764","78/8764","21/18646","37/8764","77/8764","134/18646","11/8764","21/8764","154/8764","17/18646","266/8764","56/8764","64/8764","297/8764","23/8764","18/8764","253/8764","48/8764","32/8764","74/8764","147/8764","347/18646","100/18646","73/8764","81/8764","17/8764","52/18646","74/8764","64/8764","402/8764","103/8764","71/8764","446/18646","11/18646","50/8764","70/8764","97/8764","447/8764","39/8764","21/8764","253/8764","12/8764","431/18646","159/18646","113/8764","67/8764","88/8764","187/8764","87/8764","42/8764","65/8764","65/8764","128/8764","36/8764","210/8764","64/8764","264/8764","204/8764","11/8764","11/8764","95/8764","14/8764","262/8764","29/8764","39/8764","105/8764","28/8764","62/8764","104/8764","82/8764","14/8764","21/18646","127/8764","55/8764","125/8764","197/8764","10/8764","100/8764","52/18646","78/8764","58/8764","38/8764","57/8764","98/8764","98/8764","36/8764","39/8764","56/8764","213/18646","17/18646","103/8764","401/18646","55/8764","146/18646","146/18646","127/8764","54/8764","93/8764","36/8764","31/8764","72/8764","48/8764","16/8764","22/8764","70/8764","112/8764","52/8764","52/8764","69/8764","183/8764","489/18646","24/18646","441/18646","129/8764","11/8764","67/8764","108/8764","34/8764","29/8764","86/8764","48/8764","85/8764","443/8764","66/8764","78/8764","64/8764","132/8764","71/8764","183/8764","212/18646","421/18646","219/18646","133/18646","64/8764","81/8764","103/8764","57/8764","63/8764","244/18646","184/8764","47/8764","238/18646","244/18646","244/18646","133/18646","154/8764","78/8764","18/8764","79/8764","121/8764","308/18646","96/8764","140/8764","58/8764","76/8764","19/18646","14/8764","262/8764","74/8764","26/8764","182/18646","49/18646","35/8764","245/8764","80/8764","213/18646","72/8764","49/18646","441/18646","55/18646","89/8764","260/8764","355/18646","235/8764","130/8764","489/18646","29/8764","129/8764","64/8764","81/8764","254/8764","55/18646","76/8764","253/8764","36/8764","36/8764","67/8764","75/8764","17/8764","103/8764","377/18646","28/8764","492/18646","270/8764","48/8764","50/8764","197/8764","277/18646","414/18646","422/18646","71/8764","60/8764","32/8764","447/8764","61/8764","35/8764","36/8764","36/8764","97/8764","132/8764","25/8764","56/8764","197/8764","87/18646","48/8764","56/8764","113/8764","14/8764","133/18646","92/8764","489/18646","447/8764","184/8764","77/8764","226/8764","13/8764","253/18646","88/8764","87/8764","104/8764","64/8764","103/8764","120/8764","39/8764","49/8764","119/8764","14/8764","100/8764","28/8764","10/18646","83/8764","245/8764","154/8764","154/8764","82/8764","98/8764","98/8764","82/18646","108/8764","58/8764","80/8764","80/8764","81/8764","36/8764","19/8764","17/8764","18/18646","76/8764","82/18646","35/8764","125/8764","29/8764","74/8764","64/8764","284/8764","72/8764","140/8764","87/8764","75/8764","447/8764","155/8764","17/8764","41/8764","32/8764","57/8764","104/18646","70/8764","85/8764","85/8764","374/18646","69/8764","334/18646","165/8764","297/8764","94/8764","113/8764","48/8764","120/8764","67/8764","22/8764","32/8764","183/8764","91/8764","65/8764","171/18646","78/8764","78/8764","64/8764","149/18646","89/8764","65/8764","140/8764","62/8764","64/8764","87/8764","14/8764","36/8764","389/8764","231/8764","97/8764","71/8764","58/8764","58/8764","85/8764","171/18646","70/8764","70/8764","57/8764","83/8764","56/8764","98/8764","32/8764","55/8764","82/8764","162/8764","162/8764","54/8764","126/8764","66/8764","12/8764","80/8764","126/8764","93/8764","108/8764","17/8764","177/8764","92/8764","374/18646","64/8764","52/8764","52/8764","108/8764","30/8764","75/18646","104/8764","98/8764","119/8764","103/8764","447/8764","73/8764","187/8764","74/8764","48/8764","28/8764","87/8764","28/8764","73/8764","130/8764","98/8764","100/8764","147/8764","130/8764","112/8764","82/18646","130/8764","105/8764","98/8764","98/8764","70/8764","56/8764","69/8764","80/8764","67/8764","125/8764","447/8764","67/8764","67/8764","65/8764","121/8764","66/8764","92/8764","55/18646","78/8764","111/8764","77/8764","104/8764","63/8764","31/8764","298/8764","38/8764","159/8764","147/8764","245/8764","87/8764","74/8764","62/8764","23/8764","72/8764","85/8764","85/8764","112/8764","71/8764","83/8764","257/18646","70/8764","58/8764","58/8764","58/8764","225/8764","57/8764","56/8764","105/8764","78/8764","55/8764","79/8764","105/8764","65/8764","54/8764","64/8764","76/8764","49/8764","63/8764","52/8764","52/8764","17/18646","72/8764","130/8764","55/18646","93/8764","48/8764","121/8764","245/8764","19/8764","79/8764","56/8764","60/8764","81/8764","75/8764","87/8764","422/18646","19/18646","35/8764","60/8764","58/8764","94/8764","74/8764","111/8764","154/8764","135/8764","111/8764","108/8764","20/8764","94/8764","24/8764","162/8764","156/8764","135/8764","177/8764","126/8764","77/8764","87/8764","75/8764","231/8764","231/8764","183/8764","177/8764","187/8764","262/18646","58/8764","283/18646","283/18646","283/18646","193/8764","147/8764","364/18646","303/18646","193/8764","122/8764","122/8764","284/8764","444/18646","130/8764","172/8764","222/8764","225/8764","389/8764","172/8764","284/8764","108/8764","58/8764","187/8764","144/18646","87/8764","88/8764","95/8764","280/18646","113/8764","364/18646","222/8764","225/8764","389/8764","108/8764","87/8764","88/8764","95/8764","144/18646","113/8764"],[0.0003288300813093274,0.0004484759082276236,0.002456547701157755,0.001811220753305292,0.002401739244122655,0.00234098375454664,0.00234098375454664,0.001688700751242982,0.001837463030743715,0.00194464121753996,0.00194464121753996,0.00194464121753996,0.001791169295036395,0.001792627827825129,0.001792627827825129,0.0003873481202859961,0.0002830723235048774,0.0002846151134735573,0.0003159377348251191,0.0003159377348251191,3.179203773306493e-05,0.001967612423969234,0.001640299243201682,0.001409079522761952,0.00150150237821755,0.000343794233986099,0.0002327591781974692,0.001370314275463322,0.001370314275463322,0.001307789272134712,0.001317350357691662,0.0003210003274522945,0.0003059842383996048,0.001786203925364169,0.001252782443829983,0.0001992428885769356,0.001205871953133673,0.001164239097470195,0.001417410138345215,0.001140309949910477,0.001306274149259521,0.0002197981355503579,0.001565901187075979,0.00106709252512925,0.001034567721677454,0.001504949541661244,0.001514149921649081,0.000342458574048912,0.0009901529624754021,1.49107058044452e-05,0.00137811618447994,0.0001628190240698429,0.001191682031741079,0.001164386775479952,0.001321624766810011,0.001041551756780309,0.001249960568087885,0.001232839347106424,0.0009946310813440289,4.052982560295039e-05,0.0008189784884192366,0.000806504963565462,0.0007945703146128557,0.001155589180108331,0.0001270310204687996,0.0009082686982970873,0.00073242889165502,0.0007390267628434914,9.370034659642848e-05,0.001072620895556365,0.0007114791207310549,0.0001192997900414037,0.0001192997900414037,0.000866560569097724,0.0006821483640053498,1.269368213654743e-05,0.0006296180989351697,0.0006308833435067358,8.951680132166119e-05,0.0001658177255399021,0.0005908315333657902,0.0009125132730397631,0.0001193816346473053,0.0001522104566624359,0.0001439379231327569,0.000778255991427299,0.0005294549176774337,0.0006953431427206821,0.0007394139194485717,0.0008130517615560026,0.000802655367520936,0.0007851544727166352,0.0004951859263771052,0.0004853067819492049,6.231820924861517e-06,6.691837178289706e-06,0.0004555374197479783,0.0006609494065206175,8.12959083213749e-05,0.000144485029625148,0.0005745673535840955,0.0006742384311987413,0.0005873303041136676,0.0006372694748312222,0.0003838450545468268,4.998173462770974e-05,9.022992295577769e-05,0.0003472776134091344,0.0003515562596279985,0.0004883029012368408,5.874236059317349e-05,8.543508959135674e-05,0.0004432937462312114,7.840400567112376e-05,0.00042343069372369,3.519121366114319e-05,3.649662882776842e-05,0.0002924983374432317,0.0002675233464715068,0.0002587878040354069,6.306391105948448e-05,0.0002390813070416432,0.0003400220992368519,0.000221102790284716,0.000221102790284716,5.40427795670464e-05,4.97908267140043e-05,0.0003797619070128773,0.0002004633958936321,4.477727505209272e-05,0.0003018773969963468,0.0002895871985189549,0.0002895871985189549,0.0003501090908558331,3.534619532885305e-05,4.018918553125092e-05,0.0002676006031557272,0.0002894096758253719,0.0001676405319545414,0.0001630260755815626,0.000163845412862184,0.0001557803241056302,0.0001580622708890544,0.0002607094929151824,3.954638728904472e-05,0.0002143877534050746,0.0002627793232535245,0.0001297911653690804,0.0001991207433723541,0.0001890534692809152,0.0001150952949434998,6.981185818265291e-05,0.0001109095641636553,0.0001067766399451033,0.0001775599167308765,9.540140885184077e-05,9.83600191740192e-05,9.83600191740192e-05,0.0001833780805987343,8.476779587189219e-05,8.503304118085972e-05,7.023581728207721e-06,7.930998234679652e-06,0.0001639108452267873,2.004484257746781e-05,7.560229028686369e-05,1.938388105636308e-05,1.938388105636308e-05,0.0001744891228837186,6.70431008712921e-05,6.520820462220094e-05,0.0001606574303607386,1.766403673062166e-05,6.237227053328886e-05,0.000127296175880282,0.0001086144168315586,4.791739022210063e-05,5.080963078948924e-05,5.121339293136716e-05,5.229082917222153e-05,5.229082917222153e-05,4.573552246462147e-05,4.433037497904602e-05,7.658603743093175e-07,8.122687849765988e-07,4.985611604003487e-07,1.306722151762053e-05,1.204636763068272e-05,3.684747362073022e-05,3.740947082209631e-05,3.78078396721072e-05,1.099559407803468e-05,9.016094131333214e-05,3.068005800877337e-05,3.179274472029674e-05,3.26991883505394e-05,3.297630175803477e-05,9.312738138257341e-06,0.000108638872573809,8.37367460020966e-05,8.253425920421124e-05,8.063753196717373e-05,9.066181396680704e-07,1.128976960752607e-06,1.247652066034361e-06,2.696004223215612e-06,2.624729670091076e-05,0.0001029035941165839,2.473319404731137e-05,7.539809062379074e-05,2.334063540422852e-05,6.75234514291483e-06,9.520273098765405e-05,9.531657242096224e-05,1.513621395500817e-06,1.858927819089705e-06,1.858927819089705e-06,2.357934238494504e-06,7.538350141563366e-05,1.578125458523339e-05,1.658428130422876e-05,1.752260862886326e-05,1.806838496611464e-05,1.915253136903469e-06,7.397560575287418e-05,1.3623421571344e-05,1.252472517235651e-05,1.273678967958033e-05,1.646078771808016e-06,5.249419339079591e-05,9.61565284230545e-06,1.0208277445107e-05,4.977431288128927e-06,3.140199472431205e-06,3.235696901344623e-06,3.951087432065238e-06,4.310848496174689e-06,4.58188707415797e-05,6.009640310482126e-07,8.121619064931397e-06,1.086330749120294e-06,6.128918495070363e-07,1.102203430612712e-07,4.720892549945771e-05,4.66484548911455e-05,1.765568999424931e-06,4.393329692359469e-05,4.747736459782622e-05,3.599736183636676e-07,4.48783996657029e-05,4.398114909041592e-05,3.302312683416311e-05,2.132007805103199e-06,3.426637788991608e-05,1.951109800564083e-07,2.807320008952852e-05,3.251557075438915e-05,4.215637215747658e-06,4.215637215747658e-06,4.426409270656209e-06,2.470530713803973e-05,3.205366658012236e-05,2.403847178762893e-05,6.629445819941876e-07,2.195154157073661e-05,2.084658586368913e-07,2.272149628205442e-05,1.811522482090995e-05,2.07593789779927e-05,1.801682933617e-05,1.783553733248972e-06,4.867158299003459e-08,6.735615697188876e-08,1.447767686357061e-05,1.715452342114762e-06,1.812444150824018e-06,7.198491595434911e-07,7.600986719915597e-07,1.14914183065504e-05,1.571579248880292e-05,1.571579248880292e-05,1.433994768726059e-05,1.235471812745524e-06,9.683135164159716e-06,9.743996804451086e-06,1.330014315560127e-05,1.050958206829753e-06,1.274047965698303e-05,9.384099782508034e-07,1.150946396742102e-05,1.097939716286363e-05,1.758843971710036e-07,6.881829038783382e-06,5.431943662531831e-08,6.207718887565499e-06,7.591311209466533e-06,5.529941070570353e-06,6.978482967576558e-06,7.193881259291244e-06,4.548201604558955e-07,5.540132634651579e-06,4.947171458858446e-06,4.836757767288809e-06,1.506201103470033e-07,4.366815142600969e-06,3.943639194225802e-06,2.951820910555837e-06,2.744511928253779e-06,3.589017634561725e-06,2.543023862968106e-06,3.188973257925357e-06,3.068275866655178e-06,6.861584121479054e-07,2.112942883514204e-06,1.830285665735168e-07,2.092801764936026e-07,2.716711045650132e-06,2.738287183307851e-06,2.568821959801967e-06,2.568821959801967e-06,2.441733874674364e-08,1.678738826716928e-06,1.143408557535838e-07,2.134041668742718e-06,2.134041668742718e-06,1.588476705429446e-06,1.427530461421411e-06,1.198626065872163e-06,1.4768384150948e-06,2.0472020252797e-07,1.265371053244047e-06,3.966746615729559e-09,1.237545959817324e-06,1.107089868387652e-06,4.487358864857772e-08,9.61857128757902e-07,7.180269579232613e-07,1.013356470084714e-06,9.214541922874864e-07,8.0540835446394e-07,6.061162549074852e-07,1.424782767647396e-08,1.976163962748923e-08,5.005538664355574e-07,4.890352916243227e-07,4.702801980236591e-07,4.763294551718958e-07,6.234486008052436e-07,1.394357269764856e-07,5.401748866839608e-07,5.424412189902213e-07,5.424412189902213e-07,1.275721738971072e-07,4.64653144009788e-07,3.195125673135239e-08,2.990952082638167e-07,2.835874790589473e-07,2.672643997984827e-07,4.262629263137256e-07,2.46754967088245e-07,2.517325602060907e-07,3.410017240998596e-07,2.248366338290916e-07,8.76630265542772e-09,3.262758207309191e-07,1.719886849274905e-07,2.473852394849149e-07,5.366518673570225e-08,2.065697600716474e-07,2.065697600716474e-07,2.097479294984878e-07,4.490447788979095e-08,1.268402148943798e-07,1.102477660926633e-07,1.941290183077741e-07,1.493293245534346e-07,9.078152863189651e-08,9.270586315852905e-08,9.380142506245538e-08,2.021708435202333e-09,1.417034398970564e-07,1.266127771453036e-07,6.553585562324319e-08,7.025076541830609e-08,7.257417030230808e-08,7.257417030230808e-08,6.804804777009087e-08,1.002381466305275e-08,5.958197875741098e-08,5.958197875741098e-08,6.002047131046413e-08,5.075505341411775e-08,4.94345268294781e-08,6.951021588001929e-08,3.498656702081602e-08,4.054183618091805e-08,5.82205750685598e-08,3.459116204819775e-08,5.035143380739822e-08,3.310110444893968e-08,3.09349166009554e-08,2.991727223271312e-08,2.288447010302321e-08,4.289174614131474e-08,4.355037244395165e-08,2.581396386290054e-08,2.484921950537953e-08,1.967391680834913e-08,3.700223903316719e-08,2.238175056916453e-08,4.42601691076754e-09,2.079115345711311e-08,2.175656658384492e-08,2.175656658384492e-08,3.440757950932827e-08,1.725212798376308e-08,1.623045969634974e-09,1.469470956362535e-08,1.147848933505364e-08,1.901799899345143e-08,1.782116764726201e-08,1.011932090593401e-08,1.015757750361506e-08,1.691329305526261e-08,1.615699668084336e-08,8.829753192872735e-09,7.978589314233909e-09,1.452082192843878e-08,7.920710029469529e-09,1.360127118357026e-08,7.156893414279948e-09,6.363779472424945e-09,1.177827051426226e-08,1.081617893821204e-08,1.032360445366778e-08,5.494019710897162e-09,8.280152107877704e-11,4.996275813940724e-09,5.122115797663484e-09,8.85958871137054e-09,8.85958871137054e-09,7.970040844653338e-09,4.162647164080484e-09,6.62764035420462e-09,3.648439184783784e-09,3.377579517351741e-09,5.623564182495312e-09,4.982486013503258e-09,4.537277850644803e-09,4.537277850644803e-09,2.277371252829363e-09,2.333412586921263e-09,3.734534559745945e-09,3.604166396011438e-09,1.373727429644659e-11,3.320906584304546e-09,1.946594339690162e-09,2.782740690141034e-09,2.571733885998327e-09,1.512488084783374e-09,1.506197808816051e-09,1.545932258058996e-09,1.251502210431774e-09,1.179135998122341e-09,1.092043902806687e-09,9.909301079330867e-10,1.608610454934318e-09,1.609020773288541e-09,1.650321247605337e-09,8.024585790895534e-10,7.996082678743285e-10,1.146282619248951e-09,1.146282619248951e-09,9.939875735965051e-10,9.046875882223611e-10,8.087383228975807e-10,2.966842515059853e-11,7.417305114677726e-10,5.050528728001007e-10,6.817574649986138e-10,6.817574649986138e-10,5.998902663725902e-10,5.401795335925893e-10,4.258334815405488e-10,3.512331962808177e-10,3.225537453936771e-10,3.339213941180774e-10,2.771605375937752e-10,2.401779633716034e-10,2.600461834982142e-10,2.604084099493095e-10,2.083715640003312e-10,2.187838462743097e-10,2.272812345061536e-10,1.662492303157139e-10,1.556245030557619e-10,1.556245030557619e-10,6.642973439465343e-12,9.680065196417147e-11,1.238501661971285e-10,8.583842269616068e-13,4.755939538384787e-11,5.142685306902083e-11,5.338431072739102e-11,7.050045441541658e-11,7.265078898252087e-11,3.535255244251123e-11,2.990448956682378e-11,5.934834770633173e-11,5.862032540370008e-11,1.524484477930366e-11,1.547264345223117e-11,3.22420977325167e-13,4.629179398159011e-13,2.358714807131516e-11,5.735518509533642e-12,3.345775899591841e-12,1.147421312493826e-11,8.455954812968496e-12,7.20662508009386e-12,1.318017726563684e-12,5.726756158671118e-12,4.700005220339851e-12,6.62643622301563e-13,4.09905456627913e-12,3.617624747287827e-12,1.872006701110205e-12,1.270972510181967e-12,2.437319418066059e-14,4.317890774578276e-14,3.383290380088063e-14,1.771994627282482e-14,6.280328079353868e-15,5.193184104344868e-15,3.695469996810512e-15,1.523712671550861e-15,1.251687995002006e-15,1.088516444087824e-15,8.070586725600815e-16,5.848952652874828e-16,1.486306438243293e-17,3.577471216581735e-16,1.012611248612551e-17,1.012611248612551e-17,1.012611248612551e-17,2.907240986091806e-17,2.376727191533753e-17,5.881793254536835e-19,1.000781567374663e-18,6.487065510904828e-18,2.494063903260909e-18,7.831856549271272e-19,5.478616855778709e-19,5.018544515324692e-21,4.841719550685116e-20,6.249566292377718e-20,5.056194156525946e-21,5.448666067409554e-22,5.467662761649758e-23,2.171440105580242e-22,7.944532447597448e-24,2.860517144939031e-23,2.968763736508901e-23,1.333651964238654e-24,6.994762813787149e-26,1.565810583980426e-25,2.07211734867929e-25,3.339266229483267e-26,6.381344473855367e-28,2.074952222664137e-27,1.220638329837598e-28,1.549728195385448e-27,4.960786054153256e-29,7.343542395752173e-29,6.549785945347772e-30,2.16473835197048e-31,7.335863541292111e-33,2.374729610342888e-33,8.888199046533719e-36,1.965790247234329e-35],[0.009637250844527212,0.009412053994525426,0.009384779889579238,0.009376840774924271,0.009247641656503765,0.009085246475978625,0.009085246475978625,0.008834571298608021,0.008765878678777353,0.008741445473019431,0.008741445473019431,0.008741445473019431,0.008631171022861733,0.008631171022861733,0.008631171022861733,0.008468870045194463,0.008455980549731131,0.008455980549731131,0.008455980549731131,0.008455980549731131,0.008357773475159069,0.007759374800975447,0.007594585496023787,0.007450133221411599,0.007436011777839296,0.00740479888585444,0.007390103907769648,0.007323077364572806,0.007323077364572806,0.007194759645854463,0.007194759645854463,0.007190407334931396,0.007139632229324113,0.007101249752057551,0.006995874995320244,0.006973501100192747,0.006810435917129949,0.006650883120030883,0.006628897919735702,0.00658993075704078,0.006531370746297604,0.006359899755508041,0.006276440003935686,0.006239352764579259,0.006121192353258269,0.006119167865176862,0.006119167865176862,0.006021883914455587,0.005928988221087649,0.005879788322219558,0.005663015245467987,0.005639458924600923,0.005630089598939994,0.005557846155125956,0.005476902635339793,0.005258319548793794,0.005224194169187829,0.005197055523577943,0.005070668257832304,0.005026073702149083,0.004963808643223909,0.004948555146815242,0.004936268079532367,0.004913766165851945,0.004676805638061656,0.004676234882321638,0.004649320267509053,0.004649320267509053,0.004617084578539013,0.004600979104623353,0.004592274324718627,0.004545322000577481,0.004545322000577481,0.004506114959308164,0.004460891275140248,0.004450042280441057,0.004180653622971302,0.004180653622971302,0.004083424890721047,0.004037301143580225,0.004022510576476681,0.003948840624039328,0.003886120919508635,0.003874447987771095,0.003838344616873518,0.003753463791987911,0.003654709640078952,0.003652307416310654,0.003603669944259881,0.003549842066079333,0.003536022294754393,0.003490368519622133,0.003466301484639737,0.003445678151839355,0.003284353687104588,0.003284353687104588,0.003281189820503554,0.003255527395947297,0.003082469857185465,0.003063715409204781,0.003048724733303364,0.003024794429873252,0.002924020761340087,0.002885414566596923,0.00280545576632019,0.002796037037114821,0.002639436064033497,0.002607812851270377,0.002607812851270377,0.00245743742687671,0.002429797642717631,0.002392182508557989,0.002376420082888968,0.002310854903991016,0.002293582924336654,0.002263584392296159,0.002263584392296159,0.002236487287835172,0.002077485987442795,0.002041548231834877,0.001961988344072851,0.001916506606446721,0.001861173595822768,0.00180144404543449,0.00180144404543449,0.001780232738679175,0.001742678934990151,0.00173554740681586,0.001688649284053138,0.001671684935278128,0.001669960068490429,0.001619197239030716,0.001619197239030716,0.001615125900268891,0.001608251887462814,0.001607567421250037,0.001529146303747012,0.001472490987990628,0.001436505937610467,0.001428617020921148,0.001428617020921148,0.00142830815694291,0.00142830815694291,0.001341205502441438,0.00126137156897137,0.001238684797451542,0.001223800848294985,0.001217098286574206,0.001163402096108136,0.001117134136659954,0.001100045415133066,0.001084341399235037,0.001080824576261504,0.001061359801054327,0.001061277663219032,0.0009976516230507662,0.0009976516230507662,0.0009976516230507662,0.0009539781046877975,0.0009187265536279844,0.0009187265536279844,0.0009050855185416419,0.0009050855185416419,0.0008623945606818469,0.0008580840144463961,0.0008539622334675285,0.0008413142819602031,0.0008413142819602031,0.0008204344335590229,0.0007748935147216784,0.0007716304213627111,0.0007627328489941863,0.0007609123514729331,0.0007560736208547454,0.0006774497635927653,0.0006567383343303542,0.0006497135524648526,0.0006497135524648526,0.0006497135524648526,0.0006497135524648526,0.0006497135524648526,0.0006494444189976249,0.0006480057754289962,0.0006406093150848776,0.0006406093150848776,0.0006117345438112278,0.0006098036708222914,0.0006090108079956267,0.00056940897930416,0.00056940897930416,0.00056940897930416,0.0005597756985181291,0.0005515728174462671,0.000546307399124776,0.000546307399124776,0.000546307399124776,0.000546307399124776,0.0005215133357424111,0.0005208275361626724,0.0005183703323939314,0.000517082105857709,0.0005113599588162237,0.0005085073349137189,0.0005085073349137189,0.0005085073349137189,0.0005063085543678601,0.000501727171552025,0.0004982164111189064,0.0004916958976605501,0.0004840371249922368,0.0004833456581625656,0.0004803372794846231,0.0004660980391385053,0.0004660980391385053,0.000457993341428226,0.000457993341428226,0.000457993341428226,0.0004151274223216157,0.0004058437343655626,0.0003904342316590858,0.0003904342316590858,0.0003904342316590858,0.0003904342316590858,0.0003793398244279433,0.0003691231756444436,0.0003563600274188405,0.0003516769150417459,0.0003516769150417459,0.0003477615751906403,0.0003412122570401734,0.0003170946181386362,0.0003170946181386362,0.0003097068357057999,0.0003069009485214749,0.0003069009485214749,0.0003017593947322282,0.0003017593947322282,0.0003015925669066005,0.0002961250262990068,0.0002883174768050646,0.0002868818453301843,0.0002740502127081463,0.0002704807218723595,0.0002571497941911638,0.0002571218406500044,0.0002511923167363652,0.0002450736924774018,0.0002393446524570827,0.0002365026672649296,0.0002285993482971741,0.0002263871779496146,0.0002201541788944208,0.0001989873951429652,0.0001934796702808676,0.0001922818708455904,0.0001895852473578549,0.0001858606081392861,0.0001692250313473951,0.0001692250313473951,0.0001692250313473951,0.0001690363119971139,0.0001667472655072323,0.0001666667377275606,0.0001550261509767895,0.0001542540759024735,0.000148577971722025,0.00013150065973239,0.0001290399576283996,0.0001216657274279826,0.0001069460510595732,0.0001061347572308008,0.0001054123856610059,0.0001054123856610059,0.0001045609995702322,9.007847429595371e-05,9.007847429595371e-05,8.513105126305468e-05,8.513105126305468e-05,8.41625002733269e-05,8.263465082822182e-05,8.263465082822182e-05,7.7057521088686e-05,7.675368636681569e-05,7.238397626163664e-05,7.238397626163664e-05,7.226411114543355e-05,7.139661731180388e-05,7.000106238499667e-05,6.662710845580704e-05,6.395599863714637e-05,6.171178405333694e-05,6.11686847939157e-05,5.262575147304939e-05,5.165778423067772e-05,4.817931076916507e-05,4.5646455714065e-05,4.356923267722096e-05,4.251365281563088e-05,4.090474343945836e-05,3.2970326904321e-05,3.187205715699556e-05,2.879960527835452e-05,2.849607889402684e-05,2.811575393144061e-05,2.604112932599846e-05,2.380789587625207e-05,2.36145672844467e-05,2.229915941706195e-05,2.193787029125854e-05,2.099003823402247e-05,1.973934079905696e-05,1.894148968348463e-05,1.800156775399799e-05,1.772145644237719e-05,1.733537461955342e-05,1.733537461955342e-05,1.716695426458384e-05,1.716695426458384e-05,1.652834129398897e-05,1.652834129398897e-05,1.547570929768612e-05,1.431056049004595e-05,1.420685132738279e-05,1.410197805425932e-05,1.410197805425932e-05,1.376679811372186e-05,1.258162440574803e-05,1.074630265954353e-05,1.003019423585218e-05,9.010790732481105e-06,8.715020352624492e-06,7.818457579602961e-06,7.743024045884065e-06,7.733813509165168e-06,7.434057852781041e-06,6.816639651632088e-06,6.550421370527997e-06,6.427178707523596e-06,5.925462375404253e-06,5.791833607836275e-06,5.628222366998076e-06,4.910767447431073e-06,4.910767447431073e-06,4.732509282663453e-06,4.709228734160144e-06,4.673421069611052e-06,4.673421069611052e-06,4.550244265578568e-06,4.185753265582577e-06,4.01899630433664e-06,4.01899630433664e-06,4.01899630433664e-06,3.900758080276664e-06,3.606593451123593e-06,3.442963799673027e-06,3.049598201905582e-06,2.949309782213052e-06,2.836275263167571e-06,2.779714575820492e-06,2.727102735565983e-06,2.727102735565983e-06,2.689513598142441e-06,2.541631512850601e-06,2.454564743519761e-06,2.158081499977365e-06,1.987424803606557e-06,1.98313741160858e-06,1.782369287115557e-06,1.709445625412676e-06,1.709445625412676e-06,1.709445625412676e-06,1.557686444132526e-06,1.499020721479034e-06,1.333228799260114e-06,1.302633847485498e-06,1.281088415905781e-06,1.161350976963733e-06,1.161350976963733e-06,1.161350976963733e-06,1.132156723713307e-06,9.648337157696631e-07,8.749509823623217e-07,8.738114083099092e-07,6.337280228183688e-07,6.337280228183688e-07,6.337280228183688e-07,6.278395350863101e-07,5.709643450229064e-07,5.644232782849415e-07,5.644232782849415e-07,5.644232782849415e-07,5.065147167245628e-07,5.036142420753081e-07,4.876246962492262e-07,4.787635487059034e-07,4.218076147333814e-07,4.147096347191259e-07,3.677190922080152e-07,3.642611539503965e-07,3.596986683451446e-07,3.437994140424361e-07,3.402220028324818e-07,3.216195798262722e-07,3.200606736753907e-07,3.200606736753907e-07,3.005482935466278e-07,2.9637239849099e-07,2.841787983428207e-07,2.80853060202564e-07,2.736169007080363e-07,2.735655133144618e-07,2.72793873320517e-07,2.72793873320517e-07,2.72793873320517e-07,2.655118218803165e-07,2.563173300444801e-07,2.396697881827644e-07,1.996031382392443e-07,1.755533663008204e-07,1.492429412536952e-07,1.422620796669364e-07,1.419158685505076e-07,1.419158685505076e-07,1.37383415519063e-07,1.335837404148299e-07,1.308408882216596e-07,1.257232255576252e-07,1.222389191430392e-07,1.210383501378312e-07,1.166183066295006e-07,1.128942219220288e-07,1.037296054005266e-07,1.02893193360442e-07,9.630559323831099e-08,9.372213454996438e-08,9.264053926305904e-08,8.638958699219071e-08,8.323438171203163e-08,8.323438171203163e-08,8.20397914672912e-08,8.20397914672912e-08,7.687768564738533e-08,7.269765940126275e-08,6.528930817014338e-08,6.323961253625226e-08,6.117171792537042e-08,5.660239601076803e-08,5.126424498337797e-08,4.774453738292145e-08,4.774453738292145e-08,4.388610596171145e-08,4.388610596171145e-08,4.116879764672316e-08,4.070070832569014e-08,3.919244356776213e-08,3.843949371332512e-08,3.490445022892704e-08,3.303612665475125e-08,3.133454708466383e-08,3.081694472746124e-08,2.871017050680994e-08,2.871017050680994e-08,2.503004420863549e-08,2.452602876094469e-08,2.32178029770639e-08,2.147015233855021e-08,2.065131723354787e-08,2.065131723354787e-08,2.065131723354787e-08,1.814254178811164e-08,1.777311104502485e-08,1.560967213859601e-08,1.560967213859601e-08,1.438175770547443e-08,1.351194688215978e-08,1.248152811671933e-08,1.199110037586436e-08,1.184211126929582e-08,1.176051689520234e-08,1.127334665336993e-08,1.127334665336993e-08,1.068266128194266e-08,1.000412496213475e-08,8.215037581386421e-09,7.070476951218199e-09,7.027527521666811e-09,7.027527521666811e-09,6.776575144167804e-09,6.181422320458634e-09,6.028454690326515e-09,6.028454690326515e-09,5.6276067125003e-09,5.6276067125003e-09,5.372101906509085e-09,4.81083710226097e-09,4.803609660987851e-09,4.803609660987851e-09,4.41425585052472e-09,3.447592450723953e-09,3.066766020119373e-09,2.686742630389829e-09,2.05974465556517e-09,2.05974465556517e-09,2.05974465556517e-09,1.915262344952151e-09,1.888920513545543e-09,1.818692420098078e-09,1.730722333679926e-09,1.707137766376248e-09,1.604345747890739e-09,1.02340484548329e-09,1.02340484548329e-09,9.228269130229988e-10,9.228269130229988e-10,6.814064998379936e-10,5.311090139828153e-10,3.872735603777556e-10,3.509759308804644e-10,2.584351189713496e-10,2.349359776110598e-10,2.034140691329953e-10,1.861195751568113e-10,1.629335143051148e-10,1.534019985628118e-10,1.52250598176082e-10,1.263584643874105e-10,7.488026804440819e-11,5.507547544121857e-11,1.128478890564585e-11,1.624191222129829e-12,1.378690829885886e-12,8.376701874426277e-13,2.791891300730947e-13,2.53946702702464e-13,2.007872031600378e-13,7.923305892064476e-14,7.650942869449761e-14,7.604064873699228e-14,4.663005663680471e-14,3.801819224368638e-14,1.814853950800278e-14,1.703770666897051e-14,5.883271354438921e-15,5.883271354438921e-15,5.883271354438921e-15,2.159664732525342e-15,2.059830232662586e-15,1.453635226611699e-15,1.453635226611699e-15,5.286958391387435e-16,2.439194497389169e-16,8.145130811242123e-17,6.697609106189472e-17,1.568295161038966e-17,6.294235415890651e-18,3.401549653422729e-18,3.210683289393975e-19,9.444354516843228e-20,1.421592318028937e-20,1.235006560048763e-20,4.131156872750673e-21,2.179714064443542e-21,1.929696428730786e-21,2.173852701709005e-22,1.65496088174204e-22,1.973691774617024e-23,1.973691774617024e-23,6.361302167165623e-24,9.961278723688227e-25,7.905567968350362e-25,3.810832865752982e-25,1.175210548167298e-25,1.795496115761406e-26,1.795496115761406e-26,5.960305210266473e-28,2.462389875366421e-29,1.112605970429303e-30,5.40250986353007e-31,2.816670277846535e-32,8.9443456249162e-33],[7,3,4,7,14,3,3,3,10,16,16,16,7,4,4,7,4,6,3,3,9,5,4,5,5,4,7,4,4,7,3,10,6,3,12,7,12,17,3,7,8,11,15,13,10,6,16,5,7,10,12,3,10,5,3,5,4,6,10,16,9,5,7,14,8,6,9,3,6,20,4,3,3,8,7,4,5,7,7,4,4,11,4,19,8,7,15,5,4,15,6,5,7,10,12,7,7,8,4,6,9,7,21,9,7,13,3,6,7,9,18,7,5,19,4,13,8,9,7,8,16,8,7,7,7,13,7,14,7,20,17,4,4,9,4,20,6,6,9,5,7,9,8,4,5,13,7,10,17,4,9,7,8,7,7,7,9,9,6,6,7,10,4,10,19,7,11,11,11,7,9,6,7,8,7,5,5,8,10,7,7,8,13,16,5,15,14,4,8,10,6,7,11,7,9,22,8,11,8,14,10,17,11,15,11,7,8,9,10,9,8,11,14,7,9,9,9,10,13,9,5,9,11,15,10,12,8,9,5,5,17,9,7,10,6,8,21,11,9,9,7,20,7,10,18,14,17,12,13,6,12,10,12,18,6,11,18,7,7,9,11,5,13,15,7,22,19,9,8,16,19,21,21,11,9,7,32,11,8,7,7,11,13,7,10,16,10,8,9,12,5,10,13,18,33,16,12,18,5,19,11,11,12,12,12,13,9,10,13,6,12,7,5,13,19,15,15,11,12,12,10,15,10,11,11,13,9,7,6,6,11,8,8,14,8,11,12,22,11,15,14,12,28,19,7,10,9,10,13,11,12,12,25,11,24,20,28,15,14,11,17,11,8,10,18,15,11,17,12,12,11,16,15,13,16,11,13,15,7,11,28,21,16,12,11,11,13,16,12,12,11,13,11,14,10,11,13,18,18,11,16,12,7,13,16,14,15,8,19,14,24,12,11,11,15,10,13,15,17,16,15,32,13,20,13,11,10,14,9,13,17,15,15,18,17,16,11,20,18,15,15,13,12,13,16,13,17,33,13,13,13,17,13,15,10,14,19,14,16,13,11,32,12,23,19,29,15,14,13,10,14,15,15,17,14,15,25,14,13,13,13,24,13,13,17,15,13,15,17,14,13,14,15,14,14,13,13,9,15,22,11,17,13,19,26,10,16,14,14,18,16,17,35,10,13,15,15,20,16,19,23,24,22,20,11,18,12,27,25,23,26,26,19,20,19,36,31,28,32,33,30,14,31,31,31,35,31,36,33,31,26,30,38,42,32,23,26,43,57,28,50,22,20,37,24,22,22,23,45,25,51,35,44,55,29,28,29,30,34,33],[128,14,25,92,254,12,12,13,104,297,297,297,55,17,17,120,23,64,10,10,213,39,22,44,27,13,121,26,26,87,12,96,34,11,226,85,225,389,10,85,67,191,270,253,162,54,297,72,83,244,187,11,132,34,10,25,21,52,96,401,130,39,80,235,134,36,128,10,159,402,22,10,10,63,78,21,37,77,134,11,21,154,17,266,56,64,297,23,18,253,48,32,74,147,347,100,73,81,17,52,74,64,402,103,71,446,11,50,70,97,447,39,21,253,12,431,159,113,67,88,187,87,42,65,65,128,36,210,64,264,204,11,11,95,14,262,29,39,105,28,62,104,82,14,21,127,55,125,197,10,100,52,78,58,38,57,98,98,36,39,56,213,17,103,401,55,146,146,127,54,93,36,31,72,48,16,22,70,112,52,52,69,183,489,24,441,129,11,67,108,34,29,86,48,85,443,66,78,64,132,71,183,212,421,219,133,64,81,103,57,63,244,184,47,238,244,244,133,154,78,18,79,121,308,96,140,58,76,19,14,262,74,26,182,49,35,245,80,213,72,49,441,55,89,260,355,235,130,489,29,129,64,81,254,55,76,253,36,36,67,75,17,103,377,28,492,270,48,50,197,277,414,422,71,60,32,447,61,35,36,36,97,132,25,56,197,87,48,56,113,14,133,92,489,447,184,77,226,13,253,88,87,104,64,103,120,39,49,119,14,100,28,10,83,245,154,154,82,98,98,82,108,58,80,80,81,36,19,17,18,76,82,35,125,29,74,64,284,72,140,87,75,447,155,17,41,32,57,104,70,85,85,374,69,334,165,297,94,113,48,120,67,22,32,183,91,65,171,78,78,64,149,89,65,140,62,64,87,14,36,389,231,97,71,58,58,85,171,70,70,57,83,56,98,32,55,82,162,162,54,126,66,12,80,126,93,108,17,177,92,374,64,52,52,108,30,75,104,98,119,103,447,73,187,74,48,28,87,28,73,130,98,100,147,130,112,82,130,105,98,98,70,56,69,80,67,125,447,67,67,65,121,66,92,55,78,111,77,104,63,31,298,38,159,147,245,87,74,62,23,72,85,85,112,71,83,257,70,58,58,58,225,57,56,105,78,55,79,105,65,54,64,76,49,63,52,52,17,72,130,55,93,48,121,245,19,79,56,60,81,75,87,422,19,35,60,58,94,74,111,154,135,111,108,20,94,24,162,156,135,177,126,77,87,75,231,231,183,177,187,262,58,283,283,283,193,147,364,303,193,122,122,284,444,130,172,222,225,389,172,284,108,58,187,144,87,88,95,280,113,364,222,225,389,108,87,88,95,144,113],[0.0546875,0.2142857142857143,0.16,0.07608695652173914,0.05511811023622047,0.25,0.25,0.2307692307692308,0.09615384615384616,0.05387205387205387,0.05387205387205387,0.05387205387205387,0.1272727272727273,0.2352941176470588,0.2352941176470588,0.05833333333333333,0.1739130434782609,0.09375,0.3,0.3,0.04225352112676056,0.1282051282051282,0.1818181818181818,0.1136363636363636,0.1851851851851852,0.3076923076923077,0.05785123966942149,0.1538461538461539,0.1538461538461539,0.08045977011494253,0.25,0.1041666666666667,0.1764705882352941,0.2727272727272727,0.05309734513274336,0.08235294117647059,0.05333333333333334,0.04370179948586118,0.3,0.08235294117647059,0.1194029850746269,0.05759162303664921,0.05555555555555555,0.05138339920948617,0.06172839506172839,0.1111111111111111,0.05387205387205387,0.06944444444444445,0.08433734939759036,0.04098360655737705,0.06417112299465241,0.2727272727272727,0.07575757575757576,0.1470588235294118,0.3,0.2,0.1904761904761905,0.1153846153846154,0.1041666666666667,0.0399002493765586,0.06923076923076923,0.1282051282051282,0.08749999999999999,0.05957446808510639,0.05970149253731343,0.1666666666666667,0.0703125,0.3,0.03773584905660377,0.04975124378109453,0.1818181818181818,0.3,0.3,0.126984126984127,0.08974358974358974,0.1904761904761905,0.1351351351351351,0.09090909090909091,0.05223880597014925,0.3636363636363636,0.1904761904761905,0.07142857142857142,0.2352941176470588,0.07142857142857142,0.1428571428571428,0.109375,0.0505050505050505,0.2173913043478261,0.2222222222222222,0.05928853754940711,0.125,0.15625,0.0945945945945946,0.06802721088435375,0.0345821325648415,0.07000000000000001,0.0958904109589041,0.09876543209876543,0.2352941176470588,0.1153846153846154,0.1216216216216216,0.109375,0.05223880597014925,0.08737864077669903,0.09859154929577464,0.02914798206278027,0.2727272727272727,0.12,0.1,0.09278350515463918,0.04026845637583892,0.1794871794871795,0.2380952380952381,0.07509881422924901,0.3333333333333333,0.03016241299303944,0.05031446540880503,0.07964601769911504,0.1044776119402985,0.09090909090909091,0.0855614973262032,0.09195402298850575,0.1666666666666667,0.1076923076923077,0.1076923076923077,0.1015625,0.1944444444444444,0.06666666666666667,0.109375,0.07575757575757576,0.08333333333333333,0.3636363636363636,0.3636363636363636,0.09473684210526316,0.2857142857142857,0.07633587786259542,0.2068965517241379,0.1538461538461539,0.08571428571428572,0.1785714285714286,0.1129032258064516,0.08653846153846154,0.0975609756097561,0.2857142857142857,0.2380952380952381,0.1023622047244094,0.1272727272727273,0.08,0.08629441624365482,0.4,0.09,0.1346153846153846,0.1025641025641026,0.1206896551724138,0.1842105263157895,0.1228070175438596,0.09183673469387756,0.09183673469387756,0.1666666666666667,0.1538461538461539,0.125,0.04694835680751173,0.2352941176470588,0.0970873786407767,0.04738154613466334,0.1272727272727273,0.07534246575342465,0.07534246575342465,0.08661417322834646,0.1296296296296296,0.09677419354838709,0.1666666666666667,0.2258064516129032,0.1111111111111111,0.1458333333333333,0.3125,0.2272727272727273,0.1142857142857143,0.08928571428571429,0.1346153846153846,0.1346153846153846,0.1159420289855072,0.07103825136612021,0.032719836400818,0.2083333333333333,0.03401360544217687,0.1085271317829457,0.3636363636363636,0.1194029850746269,0.09259259259259259,0.1764705882352941,0.2413793103448276,0.1279069767441861,0.1458333333333333,0.1058823529411765,0.04966139954853273,0.1212121212121212,0.141025641025641,0.125,0.1060606060606061,0.1408450704225352,0.09289617486338798,0.05188679245283019,0.03562945368171021,0.0502283105022831,0.05263157894736842,0.125,0.1111111111111111,0.0970873786407767,0.1578947368421053,0.126984126984127,0.04508196721311476,0.07608695652173914,0.148936170212766,0.03781512605042017,0.03688524590163934,0.03688524590163934,0.07518796992481203,0.08441558441558442,0.1153846153846154,0.2777777777777778,0.1139240506329114,0.09090909090909091,0.0487012987012987,0.1041666666666667,0.08571428571428572,0.1379310344827586,0.1184210526315789,0.2631578947368421,0.3571428571428572,0.0648854961832061,0.1216216216216216,0.2692307692307692,0.05494505494505494,0.1224489795918367,0.2285714285714286,0.08571428571428572,0.1375,0.04225352112676056,0.125,0.1428571428571428,0.04535147392290249,0.1272727272727273,0.1123595505617977,0.06923076923076923,0.03943661971830986,0.07234042553191489,0.09230769230769231,0.02658486707566462,0.2068965517241379,0.09302325581395349,0.15625,0.1481481481481481,0.07086614173228346,0.1090909090909091,0.1447368421052632,0.07114624505928854,0.1944444444444444,0.1944444444444444,0.1343283582089552,0.1466666666666667,0.2941176470588235,0.1262135922330097,0.03978779840848806,0.25,0.04471544715447155,0.07037037037037037,0.1875,0.16,0.08121827411167512,0.06859205776173286,0.05072463768115942,0.04976303317535545,0.1549295774647887,0.15,0.21875,0.07158836689038031,0.180327868852459,0.2285714285714286,0.1944444444444444,0.1944444444444444,0.1134020618556701,0.09848484848484848,0.28,0.1785714285714286,0.08121827411167512,0.1149425287356322,0.1666666666666667,0.1607142857142857,0.1061946902654867,0.3571428571428572,0.07518796992481203,0.1413043478260869,0.03680981595092025,0.0738255033557047,0.08695652173913043,0.1558441558441558,0.07964601769911504,0.3846153846153846,0.07509881422924901,0.125,0.1264367816091954,0.1153846153846154,0.1875,0.116504854368932,0.1083333333333333,0.2307692307692308,0.2040816326530612,0.1092436974789916,0.4285714285714285,0.12,0.25,0.5,0.1566265060240964,0.07755102040816327,0.09740259740259741,0.09740259740259741,0.1341463414634146,0.1224489795918367,0.1224489795918367,0.1219512195121951,0.1388888888888889,0.1724137931034483,0.1375,0.1375,0.1604938271604938,0.25,0.3684210526315789,0.3529411764705883,0.3333333333333333,0.1447368421052632,0.0975609756097561,0.2285714285714286,0.112,0.2758620689655172,0.1486486486486487,0.1875,0.07746478873239436,0.1527777777777778,0.1071428571428571,0.1609195402298851,0.16,0.06263982102908278,0.1225806451612903,0.4117647058823529,0.2439024390243902,0.28125,0.1754385964912281,0.125,0.1571428571428571,0.1411764705882353,0.1411764705882353,0.06684491978609626,0.1594202898550725,0.0718562874251497,0.1212121212121212,0.09427609427609428,0.1595744680851064,0.1238938053097345,0.2291666666666667,0.1416666666666667,0.1641791044776119,0.3636363636363636,0.3125,0.09836065573770492,0.1648351648351648,0.1692307692307692,0.09941520467836257,0.1538461538461539,0.1538461538461539,0.171875,0.1073825503355705,0.1685393258426966,0.2,0.1142857142857143,0.1774193548387097,0.203125,0.1724137931034483,0.5,0.3055555555555556,0.07197943444730077,0.09090909090909091,0.1649484536082474,0.1690140845070423,0.1896551724137931,0.1896551724137931,0.1529411764705882,0.0935672514619883,0.1714285714285714,0.1714285714285714,0.1929824561403509,0.1566265060240964,0.1964285714285714,0.1428571428571428,0.3125,0.2,0.1585365853658537,0.1111111111111111,0.1111111111111111,0.2037037037037037,0.126984126984127,0.1818181818181818,0.5833333333333334,0.1625,0.126984126984127,0.1505376344086022,0.1388888888888889,0.4705882352941176,0.1073446327683616,0.1521739130434783,0.06417112299465241,0.1875,0.2115384615384615,0.2115384615384615,0.1388888888888889,0.3333333333333333,0.1733333333333333,0.1442307692307692,0.173469387755102,0.134453781512605,0.145631067961165,0.07158836689038031,0.1780821917808219,0.106951871657754,0.1756756756756757,0.2291666666666667,0.3571428571428572,0.1609195402298851,0.3214285714285715,0.1780821917808219,0.1307692307692308,0.1530612244897959,0.15,0.1224489795918367,0.1307692307692308,0.1428571428571428,0.1341463414634146,0.1538461538461539,0.1714285714285714,0.1530612244897959,0.1530612244897959,0.1857142857142857,0.2142857142857143,0.1884057971014493,0.2,0.1940298507462687,0.136,0.0738255033557047,0.1940298507462687,0.1940298507462687,0.2,0.140495867768595,0.196969696969697,0.1630434782608696,0.1818181818181818,0.1794871794871795,0.1711711711711712,0.1818181818181818,0.1538461538461539,0.2063492063492063,0.3548387096774194,0.1073825503355705,0.3157894736842105,0.1446540880503145,0.1292517006802721,0.1183673469387755,0.1724137931034483,0.1891891891891892,0.2096774193548387,0.4347826086956522,0.1944444444444444,0.1764705882352941,0.1764705882352941,0.1517857142857143,0.1971830985915493,0.1807228915662651,0.09727626459143969,0.2,0.2241379310344828,0.2241379310344828,0.2241379310344828,0.1066666666666667,0.2280701754385965,0.2321428571428572,0.1619047619047619,0.1923076923076923,0.2363636363636364,0.189873417721519,0.1619047619047619,0.2153846153846154,0.2407407407407407,0.21875,0.1973684210526316,0.2857142857142857,0.2222222222222222,0.25,0.25,0.5294117647058824,0.2083333333333333,0.1692307692307692,0.2,0.1827956989247312,0.2708333333333333,0.1570247933884298,0.1061224489795918,0.5263157894736842,0.2025316455696203,0.25,0.2333333333333333,0.2222222222222222,0.2133333333333333,0.1954022988505747,0.08293838862559241,0.5263157894736842,0.3714285714285714,0.25,0.2586206896551724,0.2127659574468085,0.2162162162162162,0.1711711711711712,0.1493506493506493,0.1777777777777778,0.1981981981981982,0.1851851851851852,0.55,0.1914893617021277,0.5,0.1666666666666667,0.1602564102564103,0.1703703703703704,0.1468926553672316,0.2063492063492063,0.2467532467532468,0.2298850574712644,0.2533333333333334,0.1558441558441558,0.1341991341991342,0.1530054644808743,0.1807909604519774,0.1764705882352941,0.1145038167938931,0.2413793103448276,0.1095406360424028,0.1095406360424028,0.1095406360424028,0.1813471502590674,0.2108843537414966,0.0989010989010989,0.1089108910891089,0.1606217616580311,0.2131147540983606,0.2459016393442623,0.1338028169014084,0.0945945945945946,0.2461538461538462,0.1337209302325581,0.1171171171171171,0.1911111111111111,0.1465295629820051,0.1627906976744186,0.176056338028169,0.2037037037037037,0.3448275862068966,0.1978609625668449,0.1666666666666667,0.2528735632183908,0.25,0.2421052631578947,0.1607142857142857,0.2212389380530974,0.1401098901098901,0.1576576576576577,0.1955555555555556,0.141388174807198,0.2685185185185185,0.3218390804597701,0.3295454545454545,0.3157894736842105,0.2361111111111111,0.2920353982300885],["Ecm1/Aplp2/Psap/Sparc/Clu/App/Tmsb4x","Slc6a6/Slc38a2/Slc3a2","Ccna2/Plk1/Ccnb2/Ccnb1","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Psmd1/Psmd14/Psmc3/Ccna2/Cdc20/Ruvbl1/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Hmgb1/Casp3/Hmgb2","Lmna/Ccnb1/Lmnb1","S100a10/Plaur/Serpinb6a","Tubb4b/Cenpe/Ap1s1/Kif15/Dynll2/Tuba1b/Tuba1c/Racgap1/Tubb6/Kif11","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Ube2l3/Elob","Psmb7/Psmd14/Psmc3/Psmb2/Calm3/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2/Hbegf","Pcna/Txn1/Prdx1/Ywhah/Cox6a1/Ndufa4/Cycs/Gsr/Prdx2/Cox5a/Ywhaq/Prelid1/Gtf2h5/Elob/Cox7b/Rbbp7","Exosc8/Lsm5/Lsm3/Lsm4/Lsm6/Lsm7/Lsm2","Pcna/Rpa3/Lig1/Fen1","Tubb4b/Tuba1b/Tuba1c/Tubb6","Nupr1/Ddit4/Myc/Zbtb20/App/Ier3/Ogt","Ubc/Uba52/Map1lc3b/Sqstm1","Tubb4b/Ubc/Uba52/Map1lc3b/Sqstm1/Tuba1a","Ubc/Uba52/Nedd4","Ubc/Uba52/Nedd4","Nupr1/Ddit4/Guk1/Ndufs6/Ndufs4/Myc/Ndufa6/App/Ogt","Ywhah/Dynll1/Cycs/Casp3/Dynll2","Fh1/Sdhb/Mdh2/Suclg1","Tubb4b/Actb/Actg1/Tuba1a/Tuba1c","Ube2c/Cdc20/Anapc5/Mad2l1/Bub3","S100a10/Plaur/Anxa2/Serpinb6a","Cst3/Cp/Mxra8/Igfbp7/Pnpla2/Aplp2/App","Atp6v1g1/Atp6v0b/Atp6v1f/Atp6v0c","Tubb4b/Chmp3/Tuba1a/Tuba1c","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Gsn/Casp3/Plec","Tmed2/Kdelr2/Napa/Cope/Use1/Rab1a/Arf1/Arf4/Tuba1a/Kif5b","Tpm2/Tln1/Cald1/Tpm4/Myl6/Vcl","Snrpe/Snrpb/Snrpd3","Psmd1/Psmb7/Psmd14/Psma7/Vcp/Ube2m/Psmd8/Psmb3/Psmb5/Commd6/Uchl3/Ddb1","Ubc/Hspb1/Zfp36/Uba52/Zfp36l1/Pabpc1/Psmb8","Psmd1/Tubb4b/Psmb7/Psmd14/Psma7/Lmna/Chmp3/Psmd8/Psmb3/Psmb5/Tuba1a/Tuba1c","Psmd1/Tubb4b/Psmb7/Psmd14/Psma7/Lmna/Rab2a/Ywhag/Chmp3/H2aj/Psmd8/Psmb3/Hsp90aa1/H2bc4/Psmb5/Tuba1a/Tuba1c","Gchfr/Calm3/Calm1","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Rbbp4/Kmt5a/Ezh2/Suz12/H3c6/H4c4/H3c3/Rbbp7","S100a10/Txn1/Stmn1/Eno1/Ran/Hmgb1/Hmgb2/Anxa2/Nme1/Eif2s1/Calm1","Psmd1/Psmd14/Psmc3/Hmgb1/Psmc4/Psma1/Casp3/Psmd7/Hsp90b1/Ube2n/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Ethe1/Psmd8/Txnrd1/Psmb3/Odc1/Psmb5/Slc6a8/Sat1/Sms","Psmd1/Psmb7/Psmd14/Psma7/Ywhah/Ywhag/Psmd8/Psmb3/H2bc4/Psmb5","Tubb4b/Dynll1/Dynll2/Tuba1b/Tuba1a/Tubb6","Eloc/Psmd1/Psmd14/Psmc3/Ube2c/Cdc20/Psmc4/Psma1/Psmd7/Ube2n/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5/Ube2l3","Sox9/H3f3b/H2bc4/H1f2/Brd2","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Samhd1/Nupr1/Ddit4/Guk1/Ndufs6/Ndufs4/Myc/Ndufa6/App/Ogt","Tubb4b/Cenpe/Dynll1/Kif22/Kif23/Dynll2/Tuba1b/Tuba1a/Racgap1/Kif20a/Tubb6/Kif20b","Gm11808/Uba52/Map1lc3b","Vim/Lmna/Ywhah/Dynll1/Hmgb1/Cycs/Casp3/Hmgb2/Dynll2/Ywhaq","Txn1/Prdx1/Cycs/Gsr/Prdx2","Gchfr/Calm3/Calm1","Ube2c/Cdc20/Anapc5/Cdk1/Ccnb1","Bub1b/Ube2c/Cdc20/Mad2l1","Tubb4b/Dynll1/Dynll2/Tuba1b/Tuba1a/Tubb6","Tubb4b/Cenpe/Kdelr2/Kif15/Tuba1b/Tuba1c/Racgap1/Tubb6/Kif20b/Kif11","Actr3/Edf1/Slpi/Ctsz/S100a10/Txn1/Actb/Cycs/Camk1/Plaur/Bax/Nme1/Calm1/Eif4g1/Ppp1r2/Cdkn1a","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Ccnd1/Psmb3/Psmb5/Cdkn1a","Actb/Kras/Iqgap1/Map2k2/Actg1","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Psmd1/Psmd14/Psmc3/Kras/Calm3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Calm1/Psmd6/Psmc6/Psmb5","Lcn2/B2m/Cp/Mt2/Mt1/Aplp2/Myc/App","Tubb4b/Ube2n/Dynll2/Tuba1b/Tuba1c/Tubb6","Tagln2/Anxa5/Tex264/Manf/Actg1/Calm1/Timp1/Flna/Tmsb4x","Gchfr/Calm1/Hsp90aa1","Lcn2/Jun/Hspb1/Gpi1/Sparc/Rhob","Eloc/Psmd1/Psmd14/Psmc3/Eif4e/Hmgb1/Kras/Psmc4/Psma1/Hras/Casp3/Psmd7/Hsp90b1/Ube2n/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5/Tnfrsf12a","Ywhah/Ywhag/Rac1/Rhob","Gm11808/Uba52/Nedd4","Gm11808/Uba52/Nedd4","Psmc3/Psma1/Ccnd1/Cdk1/Cdk4/Psmd12/Ccnb1/Stub1","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Gsn/Jun/Nfkbia/Fos","Gsn/Lmna/Casp3/Dsp/Plec","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Lcn2/Cp/Ftl1/Mt2/Mt1/Aplp2/Myc","Cdc42/Rac1/Rhoa/Kif5b","Cycs/Chmp3/Bax/Casp3","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Hspd1/Ybx1/Eno1/Hmgb1","St6galnac4/Pdia3/Ctsz/Capzb/Tmed2/Kdelr2/Napa/Cope/Calr/Nus1/Rab1a/Sar1b/Arf1/Lman2/Arf4/Derl1/St3gal1/Pmm1/Tuba1a","Atp6v1g1/Atp6v0b/Atp6v1f/Atp6v1e1/Atp6v1b2/Hmox1/Atp6v1a/Atp6v0c","Vim/Tomm5/Dynll1/Ube2n/Dynll2/Tuba1b/Tuba1c","Psmd1/Psmb7/Psmd14/Psma7/Hspb1/Actb/Rac1/Kras/Psmd8/Iqgap1/Map2k2/Psmb3/Actg1/Calm1/Psmb5","Rbbp4/Ezh2/Tfdp1/Suz12/Rbbp7","Ywhah/Dynll1/Dynll2/Ywhaq","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Echs1/Psmd7/Psmb3/Psmd12/Odc1/Psmd6/Psmc6/Psmb5/Psat1/Sms","Tubb4b/Dynll1/Dynll2/Tuba1b/Tuba1a/Tubb6","Tubb4b/Gja1/Tuba1b/Tuba1a/Tubb6","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Ccnd1/Psmb3/Psmb5/Cdkn1a/Pold4","Gsn/Slpi/Jun/Isg15/Zfp36/Bax/Isg20/Ifitm3/Bst2/Ifi27/Ctsb/Lgals1","Gsn/Slpi/Isg15/Zfp36/Isg20/Ifitm3/Bst2","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Eno1/Mdh2/Tpi1/Gapdh/Mdh1/Eno3/Pgam1/Pgk1","Ubc/Uba52/Sqstm1/Nfkbia","Bub1b/Cdc20/Mad2l1/Plk1/Ube2n/Ube2l3","Parp1/Nsmce1/Nsmce4a/Sumo3/Sumo2/Smc6/Rad21/Smc3/Smc1a","Smc4/H2az1/Smc2/Plk1/H2ax/H2az2/Ccnb1","Psmb7/Psmd14/Psmc3/Jun/Psmb2/Hmgb1/Psma1/Casp3/Hsp90b1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tnfrsf12a/Elob","Lmna/Ywhah/Dynll1/Hmgb1/Cycs/Casp3/Hmgb2/Dynll2/Lmnb1","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Btg2/Lcn2/Nr4a2/Cebpb/Jun/Gapdh/Gpi1/Bnip3/Mt1/Ddit4/Ccl2/Fos/Egr1","Foxp1/Tcim/Tcf4","Cycs/Chmp3/Bax/Casp3/Pdcd6ip/Hsp90aa1","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Dtymk/Uck2/Dut/Txn1/Ak2/Gsr/Impdh2/Nme1/Rrm2","Tubb4b/Hspa5/Id1/Jun/Ubc/Uba52/Dnajb1/Mt2/Mt1/Map1lc3b/Cited2/Sqstm1/Fos/Tuba1a/Fkbp5/Cdkn1a/Psmb8/Cox7a2l","Tln1/Actb/Iqgap1/Map2k2/Rap1b/Actg1/Vcl","Ube2c/Cdc20/Anapc5/Mad2l1/Bub3","Tagln2/Anxa5/Tln1/Cdc42/Wdr1/Rac1/Gng12/Aldoa/Manf/Rhoa/Srgn/Rap1b/Actg1/Rhob/Vcl/Cyrib/Vegfa/Timp1/Tmsb4x","Lmna/Cdk1/Ccnb1/Lmnb1","Glul/Id1/Jun/Errfi1/Cav1/Zfp36/Sparc/Ccl2/Zfp36l1/Myc/Nr4a1/Hes1/Maged1","Lcn2/Jun/Hspb1/Gpi1/Bsg/Sparc/Rhob/Hbegf","Psmd1/Psmb7/Psmd14/Psma7/Mkrn1/Psmd8/Chd3/Psmb3/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Dst/Actb/Flnc/Vasp/Iqgap1/Actg1/Plec/Flna","Capzb/Tmed2/Kdelr2/Tgoln1/Napa/Pafah1b3/Cope/Use1/Arl1/Rab1a/Arf1/Pafah1b1/Arf4/Tuba1a/Kif5b/Rab18","Psmd1/Psmb7/Psmd14/Psma7/Rac1/Psmd8/Psmb3/Psmb5","Pcna/Cdca8/Sumo3/Aurkb/Top2a/Sumo2/Incenp","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Dst/Actb/Flnc/Vasp/Actg1/Plec/Flna","Tagln2/Anxa5/Tln1/Wdr1/Aldoa/Manf/Srgn/Actg1/Vcl/Cyrib/Vegfa/Timp1/Tmsb4x","Atp6v1g1/Atp6v0b/Atp6v1f/Atp6v1e1/Atp6v1b2/Atp6v1a/Atp6v0c","Tubb4b/Ssna1/Ckap5/Mapre1/Atp5k/Dynll1/Cycs/Plk1/Dynll2/Tuba1b/Tuba1a/Tubb5/Tubb6/Haus1","Actr3/Arpc5/Actb/Rac1/Arpc4/Actr2/Actg1","Cd44/Sdc4/Pgm1/Slc2a1/Eno1/Hk2/Tpi1/Gapdh/Dera/Gpi1/Aldoa/Hexa/Pkm/Pfkl/Galk1/Pfkp/St3gal1/Pgam1/Got1/Pgk1","Hat1/H2bc18/Rbbp4/Kmt5a/Ezh2/Ccnd1/Smarcc1/Cdk4/Suz12/H2ac15/H2ac24/H3c6/H2ac8/H4c4/H3c3/Pbrm1/Rbbp7","Esco2/Rad21/Smc3/Smc1a","Snrpe/Snrpb/Slbp/Snrpd3","Tubb4b/Ssna1/Ckap5/Mapre1/Dynll1/Plk1/Tuba1a/Tubb5/Haus1","Ubc/Uba52/Sqstm1/Nfkbia","Arpc2/Actr3/Arpc5/Rhoc/Sdcbp/Tln1/Cdc42/Arpc3/Actb/Rac1/Arpc4/Vasp/Rhoa/Map2k2/Actr2/Actg1/Rhob/Tuba1a/Ezr/Msn","Tubb4b/Calm3/Calm1/Tuba1b/Tuba1c/Tubb6","Ywhah/Dynll1/Cycs/Casp3/Dynll2/Ywhaq","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Ccnd1/Psmb3/Psmb5/Cdkn1a","Txn1/Cmpk1/Gsr/Txnrd1/Nme1","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Kras/Psmd8/Psmb3/Calm1/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Spop/Psmb3/Psmb5","Lsm5/Lsm4/Lsm6/Lsm7","Chmp4b/Chmp5/Rhoa/Pdcd6ip/Arf1","Tubb4b/Cenpe/Kif15/H3f3b/H3c6/H3c3/Tuba1b/Tuba1c/Racgap1/Cbx5/Tubb6/Kif20b/Kif11","Exosc8/Eif4e/Lsm5/Lsm3/Lsm4/Lsm6/Lsm2","Psmd1/Psmb7/Psmd14/Psma7/Rac1/Kras/Psmd8/Psmb3/Calm1/Psmb5","Psmc3/H2bc18/H2az1/Psma1/Ccnd1/H2ax/Smarcc1/H2az2/Psmd12/H3f3b/H2ac15/H2ac24/H3c6/H2ac8/H4c4/H3c3/Pbrm1","Sae1/Uba2/Sumo3/Sumo2","Psmd1/Psmb7/Psmd14/Eif2s2/Psma7/Vcp/Psmd8/Psmb3/Psmb5","Cdc20/Mad2l1/Bub3/Pin1/Fbxo5/Ube2n/Stub1","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Ccnd1/Psmb3/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Txn1/Rbbp4/Ezh2/Cdk4/Trp53/Suz12/Rbbp7","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Ube2m/Psmd8/Psmb3/Psmb5/Tnfrsf12a","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Actg1/Psmb5/Uchl3","Vim/Dynll1/Ube2n/Dynll2/Tuba1b/Tuba1c","Ywhah/Ywhag/Cycs/Bax/Casp3/Nmt1","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Cfh/Tpi1/Gapdh/Gpi1/Aldoa/Ddit4/Myc/Ndufa6/Ndufv3/Ier3","Tpi1/Gapdh/Gpi1/Aldoa","Vim/Lmna/Ywhah/Dynll1/Hmgb1/Cycs/Casp3/Hmgb2/Dynll2/Ywhaq","Ndufs2/Pycr2/Txn1/Psip1/Jun/Prdx1/Akr1b3/Chchd4/Ercc1/Gsr/Casp3/Prdx2/Eif2s1/Macroh2a1/Psmb5/Hnrnpm/Cfl1/Anxa1/Prdx4","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Hnrnpa3/Ybx1/Ran/Hnrnpa2b1/Ddx39a/Eif5a/Thoc7/Hnrnpa1/Mrpl18/Srsf3/Srsf7","Hnrnpa3/Ybx1/Ran/Hnrnpa2b1/Ddx39a/Eif5a/Thoc7/Hnrnpa1/Mrpl18/Srsf3/Srsf7","Tubb4b/Cenpe/Kif22/Kif23/H3f3b/Tuba1b/Tuba1a/Racgap1/Kif20a/Tubb6/Kif20b","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Ube2m/Psmd8/Psmb3/Calm1/Psmb5","Tubb4b/Rdx/Tuba1b/Tuba1a/Ezr/Tubb6","Atp6v1g1/Atp6v0b/Atp6v1f/Atp6v1e1/Atp6v1b2/Atp6v1a/Atp6v0c","Txn1/Prdx1/Ywhah/Ywhag/Cycs/Gsr/Txnrd1/Prdx5","Snrpb/Ybx1/Sf3b5/Snrpd3/Srsf7/Snrpd1/Txnl4a","Trp53/H1f5/H1f3/H1f4/Lmnb1","Rac1/Myl6/Calm1/Myh9/Flna","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Calr/Psmb3/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Kras/Ube2m/Psmd8/Psmb3/Calm1/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Mkrn1/Psmd8/Psmb3/Psmb5","Psmd1/Tubb4b/Psmb7/Psmd14/Psma7/Ywhag/Psmd8/Psmb3/Hsp90aa1/Psmb5/Tuba1a/Tuba1c/Cdkn1a","Ass1/Lcn2/B2m/Cp/Selenow/Fosb/Mt2/Psap/Fos/Lgmn/Crip1/Pdcd6/Cpne8/Slc38a2/App/Atrx","B2m/Hspg2/Clu/Pfdn5/App","Gsn/Slpi/Wfdc2/Bax/Ctsd/Perp/Cstb/Grn/Timp2/Serpinb6a/Anxa8/Psme2/Ctsb/Fam162a/Bin1","Capzb/Tmed2/Kdelr2/Napa/Pafah1b3/Cope/Use1/Rab1a/Arf1/Pafah1b1/Arf4/Tuba1a/Kif5b/Rab18","Ubc/Uba52/Map1lc3b/Sqstm1","Psmd1/Psmb7/Psmd14/Psma7/Kras/Psmd8/Psmb3/Psmb5","Psmd1/Tubb4b/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5/Tuba1a/Tuba1c","Txn1/Prdx1/Cycs/Gsr/Txnrd1/Prdx5","Cdc42/Actb/Rac1/Iqgap1/Actg1/Tuba1a/Ctnna1","H2bc18/H2az1/H2ax/H2az2/H3f3b/H2ac15/H2ac24/H3c6/H2ac8/H4c4/H3c3","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Hspb1/Psmd8/Psmb3/Psmb5/Eif4g1","Actr3/Arpc5/Tubb4b/Actb/Rac1/Arpc4/Kras/Vasp/Perp/Map2k2/Actr2/Actg1/Rhob/Hsp90aa1/Dsp/Tuba1a/Tuba1c/Krt8/Krt18/Ezr/Cdkn1a/Msn","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Psmb5/Cdkn1a","Arpc2/Actr3/Arpc5/Cdc42/Arpc3/Actb/Rac1/Arpc4/Actr2/Actg1/Myh9","Tubb4b/Dynll1/Tomm40/Ube2n/Dynll2/Tuba1b/Tuba1a/Tubb6","Lmna/Clspn/Ywhah/Hmgb1/Cycs/Casp3/Hmgb2/Dynll2/Kpnb1/H1f5/H1f3/H1f4/Stub1/Lmnb1","Mcm3/Mcm6/Psmc3/Ccna2/Mcm7/Psma1/Mcm5/Orc6/Psmd12/Mcm4","Tubb4b/Ssna1/Psmc3/Tpx2/Ccna2/Psma1/Ccnb2/Cdk1/Hmmr/Trp53/Psmd12/Ccnb1/Tuba1b/Tuba1c/Tubb5/Tubb6/Haus1","Nr4a2/Cav1/Ucp2/Bnip3/Cited2/Ddit4/Zfp36l1/Bnip3l/Myc/Egr1/Acaa2","Cebpb/Jun/Hspb1/Tmem176b/Tmem176a/Zfp36/Klf13/Tcim/Nfkbia/Zfp36l1/Fos/Myc/Tob2/Nfkbiz/Zfp36l2","Nr4a2/Cav1/Ucp2/Bnip3/Cited2/Ddit4/Zfp36l1/Bnip3l/Myc/Egr1/Acaa2","Lcn2/Jun/Nupr1/Ddit4/Fos/Clu/Egr1","Psmd1/Psmb7/Psmd14/Psma7/Vcp/Psmd8/Psmb3/Psmb5","Knl1/H2az1/Cenpa/Ruvbl1/H2ax/Cenpw/H2az2/Cenpx/Cenpm","Gsn/Lmna/Ywhah/Ywhag/Cycs/Bax/Casp3/Nmt1/Dsp/Plec","Tubb4b/Cenpe/Kif15/Tuba1b/Tuba1c/Racgap1/Tubb6/Kif20b/Kif11","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Ccnd1/Psmb3/Psmb5","Glul/Lcn2/Klf4/Jun/Hspb1/Serpine1/Gadd45a/Foxp1/Sox9/Rhob/Hbegf","Psmd1/Psmd14/Psmc3/Ccna2/Cdc20/Ruvbl1/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Lmna/Plk1/Ccnb2/Nup37/Ccnb1/Lmnb1/Banf1","Tpi1/Gpi1/Nupr1/Aldoa/Ddit4/Myc/Ndufa6/Ndufv3/Uqcc3","Tpi1/Gpi1/Nupr1/Aldoa/Ddit4/Myc/Ndufa6/Ndufv3/Uqcc3","Tpi1/Gpi1/Nupr1/Aldoa/Ddit4/Myc/Ndufa6/Ndufv3/Uqcc3","Lcn2/Mcl1/Jun/Nupr1/Ddit4/Fos/Clu/Atf4/App/Egr1","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Actr3/Arpc5/Actb/Rac1/Arpc4/Actr2/Actg1/Hsp90aa1/Myh9","Actb/Flnc/Vasp/Actg1/Flna","Psmd1/Psmb7/Psmd14/Psma7/Kras/Psmd8/Psmb3/Calm1/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Lat2/Rac1/Kras/Psmd8/Psmb3/Calm1/Psmb5","Serinc3/Mcl1/Hspb1/Nupr1/Herpud1/Ddit4/Ddx5/Rps7/Crip1/Clu/Myc/Atf4/Cdkn1a/Ier3/Ddx3x","Tubb4b/Cenpe/Kif22/Kif23/Tuba1b/Tuba1a/Racgap1/Kif20a/Tubb6/Kif20b","Psmd1/Tubb4b/Psmb7/Psmd14/Psma7/Vcp/Psmd8/Spop/Psmb3/Psmb5/Tuba1a/Tuba1c","Psmd1/Psmb7/Psmd14/Psma7/Ube2m/Psmd8/Psmb3/Psmb5","Psmd1/Psmb7/Psmd14/Psma7/Vcp/Prdx1/Psmd8/Psmb3/Psmb5","Rps27/Rps5/Rps19/Rps28/Rps14","Pcna/Rpa3/Lig1/Prim1/Fen1","Actr3/Arpc5/Tubb4b/Actb/Rac1/Arpc4/Kras/Vasp/Map2k2/Actr2/Actg1/Rhob/Hsp90aa1/Tuba1a/Tuba1c/Ezr/Msn","Psmd1/Psmb7/Psmd14/Psma7/Hspb1/Rac1/Psmd8/Psmb3/Psmb5","Atp6v1g1/Atp6v0b/Atp6v1f/Atp6v1e1/Atp6v1b2/Atp6v1a/Atp6v0c","Lbp/Ccn1/Errfi1/Hspb1/Foxp1/Zfp36/Ccl2/Clu/Zbtb20/App","Nupr1/Ddit4/Myc/Zbtb20/App/Ier3","Eno1/Tpi1/Gapdh/Gpi1/Aldoa/Pgam1/Got1/Pgk1","Arpc2/Actr3/Arpc5/Rhoc/Cdc42/Arpc3/Actb/Rac1/Arpc4/Iqgap1/Rhoa/Myl6/Actr2/Ppp2ca/Pafah1b1/Actg1/Rhob/Myh9/Tuba1a/Kif5b/Ctnna1","Nxt1/Magoh/Slbp/Ddx39a/Srsf2/Alyref/Thoc7/Rnps1/Srsf3/U2af1/Srsf7","Tpi1/Gpi1/Nupr1/Aldoa/Ddit4/Myc/Ndufa6/Ndufv3/Uqcc3","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Ccnd1/Psmb3/Psmb5/Cdkn1a","Nupr1/Ddit4/Myc/Zbtb20/App/Ier3/Ogt","Aim2/Gsn/Cd44/Slpi/Vcp/Cycs/Plaur/Bax/Ctsd/Cryab/Perp/Cstb/Grn/Serpinb6a/Cast/Anxa8/Ctsb/Dap/Bin1/Timp1","Atf3/Jun/Ppp1r15a/Tcim/Klf2/Dnajb1/Egr1","Mrps14/Mrps21/Mrpl20/Mrps25/Mrpl54/Mrpl42/Mrps24/Mrpl12/Mrpl18/Mrpl14","Psmb7/Psmd14/Psmc3/Psmb2/Calm3/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2/Ier3/Hbegf/Rbbp7","Lbp/Cebpb/Serpine1/Foxp1/Rps19/Zfp36/Nupr1/Socs3/Nfkbia/Ctla2a/Nfkbiz/App/Ier3/Pdcd4","Psmb7/Psmd14/Psmc3/Psmb2/Ap2s1/Calm3/Psma1/Psmb6/Pfn1/Psmb3/Psmd12/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2","Knl1/Ccna2/H2az1/Cenpa/Rpa3/Ruvbl1/H2ax/Cenpw/H2az2/Nhp2/Cenpx/Cenpm","Lcn2/Cp/Jun/Fosb/Ucp2/Bnip3/Jund/Junb/Mt2/Mt1/Selenom/Rhob/Fos","Tubb4b/Calm3/Calm1/Tuba1b/Tuba1a/Tubb6","Tubb4b/Cenpe/Dynll1/Kif22/Kif23/Dynll2/Tuba1b/Tuba1a/Racgap1/Kif20a/Tubb6/Kif20b","Pcna/Pole3/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Ercc1/Rfc4/Hmgn1","Eno1/Hk2/Tpi1/Gapdh/Gpi1/Aldoa/Pkm/Pfkl/Pfkp/Pgam1/Got1/Pgk1","Psmb7/Psmd14/Psmc3/Vcp/Psmb2/Ruvbl1/Psma1/Vdac3/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2/Nedd8","Atf3/Jun/Tcim/Klf2/Cited2/Egr1","Pcna/Pole3/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Lig1/Ercc1/Rfc4/Hmgn1","Pycr2/Psmb7/Psmd14/Psmc3/Psmb2/Srm/Psma1/Acat1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Psat1/Sms","Actr3/Arpc5/Actb/Rac1/Arpc4/Actr2/Actg1","Tubb4b/Actb/Actg1/Tuba1a/Tuba1c/Ezr/Msn","Tubb4b/Actb/Rac1/Map2k2/Actg1/Tuba1a/Tuba1c/Ezr/Msn","Tubb4b/Psmc3/Psma1/Ccnb2/Cdk1/Trp53/Psmd12/Ccnb1/Tuba1b/Tuba1c/Tubb6","Tubb4b/Gja1/Tuba1b/Tuba1a/Tubb6","Lmna/Clspn/Ywhah/Hmgb1/Cycs/Casp3/Hmgb2/Dynll2/Kpnb1/H1f5/H1f3/H1f4/Lmnb1","Hspa5/Id1/Ccn1/Jun/Cited2/Ccl2/Tob1/Ddx5/Fos/Trps1/Scx/Hes1/Egr1/Ltbp3/Pdcd4","Ccna2/Rad50/Trp53/H1f5/H1f3/H1f4/Lmnb1","Dst/Cd44/Anxa5/S100a10/Plaur/Bax/Casp3/Plet1/Tpm1/Map2k2/Ccl2/Grn/Actg1/Dsp/Anxa8/Plec/Myh9/Tnfrsf12a/Cdkn1a/Myof/Timp1/Flna","Psmb7/Psmd14/Psmc3/Jun/Psmb2/Hmgb1/Psma1/Casp3/Hsp90b1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2","Snrpe/Snrpb/Ybx1/Snrnp40/Sf3b5/Snrpd3/Srsf2/Srsf7/Snrpd1","Ndufs2/Ndufc1/Ndufb6/Ndufa5/Ndufc2/Ndufb7/Ndufv2/Ndufb8","Psmb7/Psmd14/Psmc3/H2az1/Psmb2/Psma1/Ccnd1/H2az2/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Atf3/Hspa5/Pdia3/Sh3glb1/Aqp3/Plin2/Slc2a1/Rac1/Asns/Gabarapl1/Ldha/Higd1a/Cnn2/Cd68/Gabarap/Sox9/Dap/Atf4/Got1","Actr3/Arpc5/Gsn/Dstn/S100a10/Rac1/Capg/Tmsb10/Arpc4/Arhgdib/Vasp/Iqgap1/Tpm1/Actg1/Rhob/Dsp/Gmfb/Myh9/Bin1/Flna/Tmsb4x","Actr3/Arpc5/Gsn/Vcp/Isg15/Rac1/Capg/Tmsb10/Arpc4/Vasp/Bax/Eif4ebp1/Cryab/Tpm1/Hsp90aa1/Lgals3/Eif4g1/Riok3/Bin1/Msn/Tmsb4x","Ccnd1/Smarcc1/Cdk4/H2ac15/H2ac24/H3c6/H2ac8/H4c4/H3c3/Pbrm1/Rbbp7","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Ccnd1/Psmb3/Psmb5/Cdkn1a","Actr3/Arpc5/Actb/Rac1/Arpc4/Actr2/Actg1","Hspa5/H13/Eif2s2/Psmd4/Ube2d3/Atp6v1g1/Prdx1/Atp6v0b/Capzb/Dnajb6/Hsph1/Atp6v1f/Cycs/Atp6v1e1/Psmd8/Atp6v1b2/Sin3b/Hmox1/Mt2/Mt1/Cryab/Cited2/Sqstm1/P4hb/Ero1a/Psmc6/St13/Tuba1a/Atp6v1a/Atp6v0c/Hspa9/Prdx5","Eno1/Hk2/Tpi1/Gapdh/Gpi1/Aldoa/Pkm/Pfkl/Pfkp/Pgam1/Pgk1","Rfc5/Rfc2/Rfc3/Rpa3/Ercc1/Topbp1/Rad50/Rfc4","Tubb4b/Dynll1/Ube2n/Dynll2/Tuba1b/Tuba1a/Tubb6","Bub1b/Ube2c/Ccna2/Cdc20/Mad2l1/Plk1/Ccnb1","Dtymk/Dut/Txn1/Tyms/Rrm1/Gsr/Impdh2/Nme1/Tk1/Rrm2/Ak6","Gsn/Lmna/Ywhah/Ywhag/Cycs/Chmp3/Bax/Casp3/Pdcd6ip/Nmt1/Hsp90aa1/Dsp/Plec","Pcna/Rfc1/Rfc5/Rfc2/Rfc3/Prim1/Rfc4","Pcna/Pole3/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Rad51ap1/Rad50/Rfc4","Psmd1/Psmd14/Psmc3/H2az1/Psmc4/Psma1/Ccnd1/Psmd7/H2ax/H2az2/Psmb3/Psmd12/H3f3b/Psmd6/Psmc6/Psmb5","Ndufs2/Ndufc1/Ndufb6/Ndufa5/Chchd4/Ndufc2/Ndufb7/Coa3/Uqcc2/Ndufb8","Snrpe/Snrpb/Sf3b5/Snrpd3/Srsf2/Zcrb1/Srsf7/Snrpd1","Psmd1/Psmb7/Psmd14/Psma7/Hspb1/Psmd8/Psmb3/Psmb5/Eif4g1","Psmd1/Psmd14/Psmc3/Ezh2/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Lsm5/Lsm3/Lsm4/Lsm6/Lsm2","Lcn2/Mcl1/Jun/Nupr1/Ddit4/Fos/Clu/Atf4/App/Egr1","Mcm3/Mcm6/Psmc3/Ube2c/Ccna2/Anapc5/Mcm7/Psma1/Mcm5/Orc6/Psmd12/Gmnn/Mcm4","Lcn2/Nr4a2/Cp/Jun/Cav1/Gapdh/Selenow/Fosb/Ucp2/Bnip3/Jund/Junb/Mt2/Mt1/Rhob/Fos/Nr4a1/Krt8","Eloc/Hspa14/Tubb4b/Psmc3/Ube2c/Ccna2/Txn1/Rbbp4/Anapc5/Rpa3/Ndufa4/Ezh2/Cycs/Hikeshi/Psma1/Gsr/Cox5a/Cdk4/Ptges3/Rad50/Trp53/Suz12/Dynll2/Psmd12/Eif2s1/H1f5/H1f3/H1f4/Tuba1b/Tuba1c/Lmnb1/Tubb6/Rbbp7","Psmb7/Psmd14/Psmc3/Psmb2/Ruvbl1/Psma1/Vdac3/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmc3/Ube2c/Ccna2/Cdc20/Anapc5/Mad2l1/Psma1/Bub3/Cdk1/Pttg1/Psmd12/Ccnb1","Psmb7/Psmd14/Psmc3/Vcp/Psmb2/Psma1/Commd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Nedd8/Elob/Rbbp7","Plk1/Kif23/Rad21/Kif20a/Smc1a","Chmp4b/Slpi/Chmp5/Cdc42/Isg15/Vamp8/Rab7/Arl8b/Isg20/Pdcd6ip/Snx3/Bsg/Rab1a/P4hb/Ifi27/Ctsb/Lgals1/Vapa/Slc3a2","Tubb4b/Ssna1/Ckap5/Mapre1/Dynll1/Plk1/Tuba1b/Tuba1a/Tubb5/Tubb6/Haus1","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Tubb4b/Cenpe/Dynll1/Ap1s1/Kif22/Kif23/Dynll2/Tuba1b/Tuba1a/Racgap1/Kif20a/Tubb6","Arpc2/Actr3/Arpc5/Sdcbp/Cdc42/Arpc3/Actb/Rac1/Arpc4/Rhoa/Actr2/Actg1","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Ube2n/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Smc4/Lmna/H2az1/Smc2/Plk1/H2ax/Ccnb2/Arpp19/Nup37/H2az2/Ccnb1/Lmnb1/Banf1","Pcna/Pole3/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Pclaf/Rfc4","Lmna/Clspn/Hmgb1/Casp3/Hmgb2/Kpnb1/H1f5/H1f3/H1f4/Lmnb1","Psmd1/Psmd14/Psmc3/Hmgb1/Psmc4/Psma1/Psmd7/Ube2n/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Lsm5/Lsm3/Lsm4/Lsm6/Lsm7/Lsm2","Psmd1/Psmd14/Psmc3/Eif2s2/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Dtymk/Dut/Txn1/Ak2/Gsr/Nme1/Rrm2","Hat1/Ezh2/H3f3b/Smchd1/Hells","Mcm3/Mcm6/Psmc3/Ube2c/Anapc5/Mcm7/Psma1/Mcm5/Orc6/Kpnb1/Psmd12/Gmnn/Mcm4","Actr3/Arpc5/Tubb4b/Ywhah/Ywhag/Actb/Rac1/Arpc4/Iqgap1/Myl6/Actr2/Actg1/Rhob/Calm1/Myh9/Tuba1a/Tuba1c/Kif5b/Flna","Psmd1/Psmb7/Psmd14/H13/Psma7/Vcp/Txn1/Prdx1/Cycs/Psmd8/Gsr/Txnrd1/Psmb3/Psmb5/Prdx5","Psmd1/Psmd14/Psmc3/Txn1/Cycs/Psmc4/Psma1/Gsr/Psmd7/Cox5a/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5/Tnfrsf12a","Psmd1/Psmd14/Psmc3/Ruvbl1/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Gpc1/Btg2/Atf3/Nupr1/Cited2/Ddx5/Fos/Myc/Maff/Egr1","Parp1/Pcna/Pole3/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Lig1/Ercc1/Sumo3/Ube2n/Sumo2/Rfc4/Hmgn1","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Odc1/Psmb5/Sat1/Sms","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Rbm8a/Eif4e/Magoh/Ddx39a/Nup37/Sarnp/Srsf2/Alyref/Thoc7/Srsf3/Srsf7","Parp1/Pcna/Pole3/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Ercc1/Sumo3/Ube2n/Sumo2/Rfc4","Ube2c/Ccna2/Cdc20/Anapc5/Mad2l1/Bub3/Fbxo5/Cdk1/Ccnb1","Pcna/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Rfc4","Lbr/Lmna/Ccnb2/Ccnb1/Lmnb1/Banf1","Dtymk/Dut/Tyms/Rrm1/Nme1/Rrm2","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Btg2/Atf3/Nupr1/Cited2/Fos/Myc/Scx/Egr1","Eno1/Mdh2/Tpi1/Gapdh/Mdh1/Eno3/Pgam1/Pgk1","Psmd1/Psmd14/Psmc3/Kras/Calm3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Calm1/Psmd6/Psmc6/Psmb5","Tubb4b/Actb/Rac1/Iqgap1/Actg1/Calm1/Tuba1a/Tuba1c","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Smc4/H2bc18/H2az1/Smc2/H2ax/Cdk1/H2az2/H2ac15/H2ac24/H2ac8/H4c4/Ccnb1","Mcm6/Psmb7/Psmd14/Psmc3/Psmb2/Cenpa/Ywhah/Dynll1/Psma1/Spc24/Ccnb2/Ube2n/Psmb6/Dynll2/Psmb3/Psmd12/Ywhaq/Psma3/Psma2/Psmd6/Psmb5/Psme2","Txn1/Prdx1/Ywhah/Cox6a1/Ndufa4/Cycs/Gsr/Prdx2/Cox5a/Ywhaq/Cox7b","Psmd1/Tubb4b/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5/Tuba1b/Tuba1a/Tubb6","Psmc3/Ube2c/Ccna2/Cdc20/Anapc5/Mad2l1/Psma1/Bub3/Fbxo5/Cdk1/Pttg1/Aurkb/Psmd12/Ccnb1","Psmd1/Tubb4b/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Hsp90aa1/Psmb5/Tuba1a/Tuba1c/Cdkn1a","Psmd1/Tubb4b/Psmb7/Psmd14/H13/Eif2s2/Psma7/Dnaja1/Vcp/Txn1/Atp6v1g1/Prdx1/Atp6v0b/Atp6v1f/Cycs/Psmd8/Gsr/Hsbp1/Cryab/Txnrd1/Psmb3/Hsp90aa1/Psmb5/Tuba1a/Tuba1c/Atp6v0c/Cdkn1a/Prdx5","Ssrp1/Tpx2/Ccna2/Rbbp4/Rfc5/Kmt5a/Rfc2/Rfc3/Rpa3/Pin1/Topbp1/Cdk1/Rad50/Aurkb/Trp53/Ak6/Supt16/Rfc4/Rbbp7","Pcna/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Rfc4","Parp1/Pcna/Pole3/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Ercc1/Rfc4","Pcna/Pole3/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Pclaf/Rfc4","Tubb4b/Cenpe/Kif22/Kif23/Tuba1b/Tuba1a/Racgap1/Kif20a/Tubb6/Kif20b","Knl1/Cenpa/Ezh2/Bub3/Ctcf/Cdk1/Aurkb/H1f5/Macroh2a1/Esco2/Rad21/Cenpq/Dkc1","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Ube2n/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Tubb4b/Ssna1/Ckap5/Mapre1/Aurka/Dynll1/Plk1/Ccnb2/Ccnb1/Tuba1a/Tubb5/Haus1","Tpx2/Cenpe/Cdc20/Cdca8/Cenpa/Lsm3/Prc1/Lsm4/Ccnb2/Fbxo5/Cenpw/G3bp1/Aurkb/Kpnb1/Birc5/Cenpx/Eif2s1/Cenph/Racgap1/Tubb5/Haus1/Incenp/Kif11/Smc3/Smc1a","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Arpc2/Actr3/Cd44/S100a10/Rhoc/Sh3glb1/Cdc42/Capzb/Wdr1/Rac1/Aldoa/Calr/Rhoa/Actr2/Pafah1b1/P4hb/Rhob/Fermt2/Myh9/Ezr/Vegfa/Myl12a/Anxa1/Msn","Nop58/Parp1/Pcna/Cdca8/Sae1/Uba2/Nsmce1/Nsmce4a/Dnmt1/Sumo3/Aurkb/Suz12/Top2a/Sumo2/Smc6/H4c4/Rad21/Incenp/Smc3/Smc1a","Eloc/Ssrp1/Pcna/Tpx2/Ccna2/Txn1/Rbbp4/Ywhah/Rfc5/Kmt5a/Rfc2/Rfc3/Rpa3/Ndufa4/Cycs/Gsr/Pin1/Cox5a/Topbp1/Cdk1/Rad50/Aurkb/Trp53/Ak6/Ccnb1/Supt16/Rfc4/Rbbp7","Ube2c/Ccna2/Txn1/Rbbp4/Anapc5/Ezh2/Cdk4/Rad50/Trp53/Suz12/H1f5/H1f3/H1f4/Lmnb1/Rbbp7","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Rbbp7","Pcna/Pole3/Usp1/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Rad18/Pclaf/Rfc4","Smc4/Lmna/H2bc18/H2az1/Smc2/H2ax/Ccnb2/Arpp19/Cdk1/H2az2/H2ac15/H2ac24/H2ac8/H4c4/Ccnb1/Lmnb1/Banf1","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Pcna/Pole3/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Rfc4","Arpc2/Actr3/Arpc5/Cdc42/Arpc3/Actb/Rac1/Arpc4/Actr2/Actg1","Psmb7/Psmd14/Psmc3/Psmb2/Dynll1/Psma1/Ccnb2/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c/Tubb5","Ssrp1/Tpx2/Ccna2/Rfc5/Rfc2/Rfc3/Rpa3/Pin1/Topbp1/Rad50/Aurkb/Trp53/Ak6/Supt16/Rfc4","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Nop58/Hnrnpa3/Nxt1/Magoh/Ybx1/Slbp/Ran/Hnrnpa2b1/Fbl/Ddx39a/Kpnb1/Alyref/Thoc7/Mrpl18/Srsf3/Srsf7/Dkc1","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Ube2n/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Ccnd1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Parp1/Pcna/Usp1/Rad18/Ercc1/Ccnd1/Casp3/Cdkn2d/Tipin/Pclaf/Aurkb/Trp53/Eif2s1/Pbk/Hmgn1/Timp1","H2bc18/Ywhah/Rfc5/Rfc2/Rfc3/Rpa3/H2ax/Topbp1/Cdk1/Ube2n/Rad50/Trp53/H4c4/Ccnb1/Rfc4","Lbr/Tubb4b/Lmna/Ran/Ccnb2/Cdk1/Kpnb1/Ccnb1/Tuba1b/Tuba1c/Lmnb1/Tubb6/Banf1","Psmb7/Psmd14/Psmc3/Vcp/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Snrpe/Snrpb/Magoh/Slbp/Ddx39a/Snrpd3/Srsf2/Alyref/Thoc7/Rnps1/Srsf3/U2af1/Srsf7","Ccna2/H2bc18/Clspn/Rfc5/Rfc2/Rfc3/Rpa3/H2ax/Tipin/Topbp1/Ube2n/Rad50/Sumo2/H4c4/Rfc4","Pcna/Rfc1/Rfc5/Rfc2/Rfc3/Prim1/Rfc4","Arpc2/Actr3/Arpc5/Cdc42/Arpc3/Actb/Rac1/Arpc4/Rhoa/Actr2/Actg1","Psmb7/Psmd14/Psmc3/Lmna/H2az1/Psmb2/Cenpa/Dynll1/Ran/Psma1/Spc24/Ccnb2/Arpp19/H2az2/Pttg1/Psmb6/Dynll2/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c/Tubb5/Banf1","Hnrnpa3/Snrpb/Ybx1/Hnrnpd/Hnrnpa2b1/Lsm5/Snrpa/Snrpa1/Lsm4/Lsm6/Ddx39a/Sf3b5/Snrpd3/Lsm7/Thoc7/Hnrnpa1/Snrpc/Srsf3/Srsf7/Snrpd1/Txnl4a","Dtymk/Uck2/Nudt5/Dut/Gmps/Txn1/Ak2/Tyms/Paics/Rrm1/Gsr/Impdh2/Nme1/Tk1/Rrm2/Ak6","Psmd1/Psmd14/Psmc3/Ccna2/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Exosc8/Hnrnpd/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Nop58/Hnrnpu/Hnrnpa3/Ckap5/Rbm8a/Magoh/Ran/Hnrnpa2b1/Ddx39a/Nup37/Sarnp/Alyref/Thoc7/Mrpl18/Srsf3/Srsf7","Tubb4b/Ssna1/Ckap5/Tpx2/Mapre1/Aurka/Dynll1/Plk1/Hmmr/Tuba1a/Tubb5/Haus1","Eloc/Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Ube2c/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Gmnn/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tnfrsf12a","Pcna/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Lig1/Prim1/Rfc4/Fen1","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmd1/Psmd14/Psmc3/Dbf4/Ywhah/Rpa3/Psmc4/Psma1/Psmd7/H2ax/Ccnb2/Ube2n/Psmb3/Psmd12/Ccnb1/Psmd6/Psmc6/Psmb5","Mcm6/Psmb7/Psmd14/Psmc3/Psmb2/Ywhah/Psma1/Ccnb2/Ube2n/Psmb6/Psmb3/Psmd12/Ywhaq/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Ube2c/Ccna2/Dbf4/Rpa3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Gmnn/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Ccna2/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Hmgb1/Casp3/Hmgb2/Kpnb1/H1f5/H1f3/H1f4","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Pcna/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Gmnn/Psmd6/Psmb5/Psme2","Psmd1/Psmd14/Psmc3/Calm3/Psmc4/Psma1/Psmd7/Ube2n/Psmb3/Psmd12/Calm1/Psmd6/Psmc6/Psmb5","Psmd1/Tubb4b/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5/Tuba1b/Tuba1a/Tubb6","Lbr/Lmna/Ccnb2/Cdk1/Kpnb1/Ccnb1/Lmnb1/Banf1","Hnrnpa3/Snrpb/Ybx1/Hnrnpd/Hnrnpa2b1/Lsm5/Snrpa/Snrpa1/Lsm4/Lsm6/Sf3b5/Snrpd3/Lsm7/Hnrnpa1/Snrpc/Srsf3/Srsf7/Snrpd1/Txnl4a","Psmd1/Psmd14/Psmc3/Ube2c/Ccna2/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Gmnn/Psmd6/Psmc6/Psmb5","Aspm/Hnrnpu/Tpx2/Mapre1/Aurka/Cenpe/Cdc20/Cdca8/Mrpl20/Cenpa/Lsm3/Plk1/Lsm4/Kif23/Ccnb2/Cenpw/Birc5/Cenpx/Dlgap5/Racgap1/Ezr/Tubb5/Haus1/Smc1a","Snrpe/Snrpb/Rbm8a/Magoh/Ddx39a/Snrpd3/Sarnp/Srsf2/Alyref/Thoc7/Srsf3/Srsf7","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c","Pcna/Pole3/Usp1/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Rad18/Rfc4","Hspa5/Chmp4b/Ssr3/Sec61b/Srp72/Kdelr2/Ppp1r15a/Spcs2/Gja1/Sec61g/Rtn4/Rab10/Vapa","Psmd1/Psmd14/Psmc3/Kras/Calm3/Psmc4/Psma1/Hras/Psmd7/Psmb3/Psmd12/Calm1/Psmd6/Psmc6/Psmb5","Mcm3/Mcm6/Psmc3/Ube2c/Pole3/Dbf4/Anapc5/Mcm7/Rpa3/Psma1/Mcm5/Orc6/Prim1/Kpnb1/Psmd12/Gmnn/Mcm4","Psmb7/Psmd14/Psmc3/Psmb2/Hmgb1/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2","Eloc/Psmd1/Tubb4b/Psmd14/Psmc3/Eif2s2/Ube2c/Ccna2/Txn1/Dnajc2/Dynll1/Rpa3/Ezh2/Cycs/Psmc4/Hikeshi/Psma1/Gsr/Psmd7/Cox5a/Nup37/Hspa4/Dynll2/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5/Tuba1b/Tuba1a/Lmnb1/Tubb6","Psmd1/Psmd14/Psmc3/Ube2c/Ccna2/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmb7/Psmd14/Psmc3/Psmb2/Cenpa/Dynll1/Psma1/Spc24/Pttg1/Psmb6/Dynll2/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Dtymk/Dut/Txn1/Ak2/Tyms/Rrm1/Gsr/Nme1/Rrm2/Ak6","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Pttg1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Dtymk/Dut/Txn1/Tyms/Rrm1/Gsr/Nme1/Rrm2/Ak6","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmd1/Psmd14/Psmc3/Ccna2/Cks1b/Dbf4/Rpa3/Psmc4/Psma1/Ccnd1/Psmd7/Psmb3/Psmd12/Gmnn/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Ube2c/Dbf4/Rpa3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Gmnn/Psmd6/Psmc6/Psmb5","Psmb7/Psmd14/Psmc3/Vcp/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Eif2s1/Psma2/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Pcna/Cks1b/Psmb2/Psma1/Ccnd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Gmnn/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Cks1b/Psmb2/Psma1/Ccnd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Gmnn/Psmd6/Psmb5/Psme2","Psmd1/Psmd14/Psmc3/Kras/Calm3/Psmc4/Psma1/Hras/Psmd7/Ube2n/Psmb3/Psmd12/Calm1/Psmd6/Psmc6/Psmb5","Btg2/Atf3/Nupr1/Cited2/Ddx5/Fos/Myc/Scx/Maff/Nr4a1/Egr1","Mcm3/Mcm6/Psmc3/Ccna2/Cks1b/Pole3/Rbbp4/Dbf4/Mcm7/Rpa3/Psma1/Ccnd1/Tfdp1/Mcm5/Orc6/Cdk4/Prim1/Psmd12/Gmnn/Mcm4","Mcm3/Mcm6/Psmc3/Ccna2/Cks1b/Pole3/Dbf4/Mcm7/Rpa3/Psma1/Ccnd1/Mcm5/Orc6/Cdk4/Prim1/Psmd12/Gmnn/Mcm4","Mcm6/Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Gmnn/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Ruvbl1/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Nedd8","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmd1/Psmd14/Psmc3/Hnrnpd/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","H2bc18/H2az1/Rbbp4/Ezh2/Dnmt1/H2ax/H2az2/Suz12/H3f3b/H2ac15/H2ac24/H3c6/H2ac8/H4c4/H3c3/Rbbp7","Psmd1/Psmd14/Psmc3/Kras/Psmc4/Psma1/Hras/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmb7/Psmd14/Psmc3/Psmb2/Ap2s1/Calm3/Psma1/Psmb6/Pfn1/Psmb3/Psmd12/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Vcp/Txn1/Jun/Prdx1/Psmb2/Dynll1/Cox6a1/Ndufa4/Cycs/Psma1/Gsr/Prdx2/Mt2/Mt1/Cox5a/Psmb6/Dynll2/Psmb3/Psmd12/Psma3/Eif2s1/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c/Elob/Cox7b/Rbbp7","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Lbr/Tubb4b/Lmna/Ran/Ccnb2/Nup37/Ccnb1/Rangap1/Tuba1b/Tuba1a/Lmnb1/Tubb6/Banf1","Psmd1/Psmd14/Psmc3/Lat2/Kras/Calm3/Psmc4/Psma1/Hras/Psmd7/Ube2n/Psmb3/Psmd12/Calm1/Psmd6/Psmc6/Psmb5","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Gmnn/Psmd6/Psmb5/Psme2","Atf3/Cebpb/Jun/Ppp1r15a/Tcim/Klf2/Dnajb1/Nedd4/Cited2/Egr1","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Ccnd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","H2bc18/H2az1/Rbbp4/Baz1b/Ezh2/Smarca5/Dnmt1/H2ax/H2az2/Suz12/H3f3b/H2ac15/H2ac24/H3c6/H2ac8/H4c4/H3c3/Dek/Rbbp7","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Pttg1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Calm3/Psma1/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Ccnd1/Psmd7/Psmb3/Psmd12/Ccnb1/Psmd6/Psmc6/Psmb5","Mcm3/Mcm6/Pole3/Dbf4/Mcm7/Rpa3/Mcm5/Orc6/Prim1/Gmnn/Mcm4","Ube2t/Parp1/Pcna/Ccna2/H2bc18/Pole3/Usp1/Clspn/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Rad18/Rad51ap1/Lig1/Ercc1/H2ax/Tipin/Pclaf/Topbp1/Sumo3/Ube2n/Rad50/Trp53/Sumo2/Cenpx/H4c4/Apex1/Rfc4/Hmgn1/Fen1","Parp1/Pcna/Pole3/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Lig1/Apex1/Rfc4/Fen1","Parp1/Pcna/Ccna2/H2bc18/Pole3/Clspn/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Rad51ap1/Ercc1/H2ax/Tipin/Topbp1/Ube2n/Rad50/Trp53/Sumo2/H4c4/Rfc4/Fen1","Psmd1/Psmd14/Psmc3/Ube2c/Ccna2/Cks1b/Rpa3/Psmc4/Psma1/Ccnd1/Psmd7/Psmb3/Psmd12/Gmnn/Psmd6/Psmc6/Psmb5/Rad21/Smc1a","Sgo2a/Cenpf/Tubb4b/Spc25/Knl1/Pmf1/Cenpe/Cdc20/Cdca8/Nudc/Cenpa/Ywhah/Ppp1cc/Mad2l1/Calm3/Prc1/Bub3/Pin1/Spc24/Aurkb/Dynll2/Calm1/Cenph/Cenpm/Tuba1b/Tuba1c/Cenpq/Tubb6/Incenp","Psmb7/Psmd14/Psmc3/Psmb2/Ap2s1/Psma1/Psmb6/Pfn1/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Pttg1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Pcna/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Lig1/Prim1/Rfc4/Fen1","Psmd1/Psmd14/Psmc3/Ccna2/Cks1b/Psmc4/Psma1/Ccnd1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Hnrnpd/Psma1/Anp32a/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Calm3/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Gmnn/Psmd6/Psmb5/Psme2","Ubxn4/Ufc1/Atf3/Hspa5/Pdia3/H13/Serp1/Sec61b/Tmed2/Pdia4/Ppp1r15a/Selenos/Nupr1/Calr/Manf/Hsp90b1/Ddit3/P4hb/Pdia6/Selenok/Ero1a/Psmc6/Derl1/Atf4/Sdf2l1","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Elob","Psmd1/Psmd14/Psmc3/Psmc4/Psma1/Psmd7/Psmb3/Psmd12/Odc1/Psmd6/Psmc6/Psmb5/Sms","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Lmna/Psmb2/Cenpa/Dynll1/Ran/Psma1/Spc24/Ccnb2/Pttg1/Psmb6/Dynll2/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c/Banf1","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Cks1b/Psmb2/Psma1/Ccnd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Gmnn/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmd1/Psmd14/Psmc3/Kras/Calm3/Psmc4/Psma1/Hras/Psmd7/Psmb3/Psmd12/Calm1/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Ccna2/Cks1b/Dbf4/Rpa3/Psmc4/Psma1/Ccnd1/Psmd7/Psmb3/Psmd12/Gmnn/Psmd6/Psmc6/Psmb5","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Vcp/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Vcp/Prdx1/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Pcna/Ccna2/Gar1/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Lig1/Prim1/Nhp2/Rfc4/Fen1/Dkc1","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Ccnd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Eno1/Tpi1/Gapdh/Gpi1/Aldoa/Pkm/Pfkl/Pfkp/Pgk1","Psmb7/Psmd14/Psmc3/Cks1b/Psmb2/Psma1/Ccnd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Parp1/Pcna/Ccna2/H2bc18/Pole3/Clspn/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Rad51ap1/Ercc1/H2ax/Tipin/Topbp1/Ube2n/Rad50/Sumo2/H4c4/Rfc4/Fen1","Atf3/Cebpb/Jun/Ppp1r15a/Tcim/Klf2/Dnajb1/Nedd4/Cited2/Atf4/Egr1","Psmb7/Psmd14/Psmc3/Psmb2/Calm3/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Jun/Psmb2/Lat2/Calm3/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2","Cenpf/Tubb4b/Ckap5/Bub1b/Knl1/Bub1/Mapre1/Cenpe/Cdc20/Cdca8/Cenpa/Ywhah/Dynll1/Mad2l1/Calm3/Plk1/Spc24/Nup37/Dynll2/Calm1/Diaph3/Rangap1/Cenpm/Tuba1b/Tuba1a/Tubb6","Pcna/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Lig1/Prim1/Rfc4/Fen1","Psmb7/Psmd14/Psmc3/Psmb2/Calm3/Psma1/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Hnrnpd/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmd1/Psmd14/Psmc3/Ccna2/Cks1b/Psmc4/Psma1/Ccnd1/Psmd7/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Knl1/H2bc18/H2az1/Rbbp4/Cenpa/Smarca5/H2ax/Cenpw/H2az2/Cenpx/H2ac15/H2ac24/H2ac8/H4c4/Cenph/Cenpm/Cenpq/Rbbp7","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Ccnb2/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c","Ndufs2/Ndufc1/Ndufb6/Cox6a1/Ndufa4/Ndufa5/Cycs/Ndufc2/Uqcrc2/Ndufb7/Cox5a/Uqcrc1/Uqcr10/Cyc1/Ndufv2/Ndufb8/Cox7b","Arpc2/Actr3/Arpc5/Rhoc/Sh3glb1/Cdc42/Capzb/Isg15/Arpc3/Rac1/Mtpn/Capg/Tmsb10/Arpc4/Napa/Vasp/Eif4ebp1/Vps35/Cryab/Rhoa/Sar1a/Rap1b/Sar1b/Sox9/H3f3b/Vdac2/Psmc6/Fermt2/Lgals3/Cyrib/Vegfa/Riok3/Capn1/Msn/Tmsb4x","Eno1/Tpi1/Gapdh/Gpi1/Aldoa/Pkm/Pfkl/Galk1/Pfkp/Pgk1","Mcm3/Mcm6/Clspn/Dbf4/Rfc5/Rfc2/Mcm7/Rfc3/Rpa3/Mcm5/Orc6/Mcm4/Rfc4","Psmb7/Psmd14/Psmc3/Cks1b/Psmb2/Psma1/Ccnd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Srm/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Sms","Sgo2a/Cenpf/Spc25/Knl1/Pmf1/Cenpe/Cdc20/Cdca8/Nudc/Cenpa/Ppp1cc/Mad2l1/Bub3/Spc24/Aurkb/Dynll2/Cenph/Cenpm/Cenpq/Incenp","Psmd1/Psmd14/Psmc3/Ube2c/Aurka/Cdc20/Psmc4/Psma1/Plk1/Psmd7/Pttg1/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5","Cenpf/Ckap5/Bub1b/Knl1/Bub1/Mapre1/Ube2c/Cenpe/Cdc20/Cdca8/Cenpa/Dynll1/Mad2l1/Plk1/Spc24/Nup37/Dynll2/Rangap1/Cenpm","Psmb7/Psmd14/Psmc3/Vcp/Txn1/Prdx1/Psmb2/Cox6a1/Ndufa4/Cycs/Psma1/Gsr/Prdx2/Cox5a/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Cox7b","Sgo2a/Cenpf/Tubb4b/Spc25/Knl1/Pmf1/Cenpe/Cdc20/Cdca8/Nudc/Cenpa/Ppp1cc/Mad2l1/Bub3/Spc24/Aurkb/Dynll2/Cenph/Cenpm/Tuba1b/Tuba1c/Cenpq/Tubb6/Incenp","Sgo2a/Cenpf/Spc25/Knl1/Ube2c/Pmf1/Cenpe/Cdc20/Cdca8/Nudc/Cenpa/Ppp1cc/Anapc5/Mad2l1/Bub3/Spc24/Aurkb/Dynll2/Cenph/Cenpm/Cenpq/Incenp","Ndufs2/Atp5g3/Ndufc1/Ndufb6/Atp5k/Cox6a1/Ndufa4/Ndufa5/Cycs/Ndufc2/Uqcrc2/Ndufb7/Cox5a/Uqcrc1/Uqcr10/Atp5g1/Cyc1/Ndufv2/Ndufb8/Cox7b","Pcna/Pole3/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Lig1/Apex1/Rfc4/Fen1","Cenpf/Ckap5/Bub1b/Knl1/Bub1/Mapre1/Cenpe/Cdc20/Cdca8/Cenpa/Dynll1/Mad2l1/Plk1/Spc24/Nup37/Dynll2/Rangap1/Cenpm","Parp1/Pcna/Pole3/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Lig1/Apex1/Rfc4/Fen1","Mcm3/Mcm6/Psmc3/H2bc18/Clspn/Dbf4/Ywhah/Rfc5/Rfc2/Mcm7/Rfc3/Rpa3/Psma1/Mcm5/Orc6/H2ax/Ccnb2/Topbp1/Cdk1/Ube2n/Rad50/Trp53/Psmd12/H4c4/Ccnb1/Mcm4/Rfc4","Ndufs2/Fh1/Atp5g3/Ndufc1/Ndufb6/Sdhb/Atp5k/Cox6a1/Mdh2/Ndufa4/Ndufa5/Cycs/Suclg1/Ldha/Ndufc2/Uqcrc2/Ndufb7/Cox5a/Uqcrc1/Uqcr10/Atp5g1/Cyc1/Ndufv2/Ndufb8/Cox7b","Cenpf/Tubb4b/Ckap5/Bub1b/Knl1/Bub1/Mapre1/Cenpe/Cdc20/Cdca8/Cenpa/Dynll1/Mad2l1/Plk1/Spc24/Nup37/Dynll2/Diaph3/Rangap1/Cenpm/Tuba1b/Tuba1a/Tubb6","Snrpe/Hnrnpu/Hnrnpa3/Snrpb/Rbm8a/Magoh/Hnrnpd/Lsm8/Hnrnpa2b1/Lsm5/Lsm3/Snrpa/Snrpa1/Lsm4/Lsm6/Cwc15/Sf3b5/Snrpd3/Srsf2/Alyref/Zcrb1/Snrpc/Srsf3/Lsm2/Srsf7/Snrpd1","Mcm3/Mcm6/Psmc3/Pcna/Ube2c/Ccna2/Pole3/Dbf4/Rfc1/Rfc5/Anapc5/Rfc2/Mcm7/Rfc3/Rpa3/Lig1/Psma1/Mcm5/Orc6/Prim1/Kpnb1/Psmd12/Gmnn/Mcm4/Rfc4/Fen1","Psmd1/Psmd14/Psmc3/Bub1b/Ube2c/Ccna2/Cdc20/Mad2l1/Psmc4/Psma1/Plk1/Psmd7/Pttg1/Psmb3/Psmd12/Ccnb1/Psmd6/Psmc6/Psmb5","Psmd1/Psmd14/Psmc3/Bub1b/Ube2c/Aurka/Ccna2/Cdc20/Mad2l1/Psmc4/Psma1/Plk1/Psmd7/Pttg1/Psmb3/Psmd12/Ccnb1/Psmd6/Psmc6/Psmb5","Psmd1/Tubb4b/Psmd14/Psmc3/Mapre1/Psmc4/Psma1/Plk1/Psmd7/Ccnb2/Psmb3/Psmd12/Ccnb1/Psmd6/Psmc6/Psmb5/Tuba1b/Tuba1a/Tubb6","Snrpe/Hnrnpa3/Snrpb/Nxt1/Magoh/Ybx1/Snrnp40/Slbp/Hnrnpd/Lsm8/Hnrnpa2b1/Lsm5/Lsm3/Hnrnpf/Snrpa1/Fus/Lsm4/Lsm6/Ddx39a/U2surp/Sf3b5/Snrpd3/Ptbp1/Lsm7/Srsf2/Alyref/Thoc7/Phf5a/Snu13/Tra2b/Rnps1/Srsf3/U2af1/Lsm2/Srsf7/Snrpd1","Snrpe/Hnrnpu/Hnrnpa3/Snrpb/Rbm8a/Eif4e/Magoh/Hnrnpd/Lsm8/Hnrnpa2b1/Lsm5/Lsm3/Snrpa/Snrpa1/Lsm4/Lsm6/Ddx39a/Cwc15/Sf3b5/Snrpd3/Nup37/Sarnp/Srsf2/Alyref/Thoc7/Zcrb1/Snrpc/Srsf3/Lsm2/Srsf7/Snrpd1","Psmd1/Tubb4b/Ssna1/Psmd14/Psmc3/Ckap5/Tpx2/Mapre1/Aurka/Ccna2/Dynll1/Psmc4/Psma1/Plk1/Psmd7/Ccnb2/Hmmr/Psmb3/Psmd12/Ccnb1/Psmd6/Psmc6/Psmb5/Tuba1b/Tuba1a/Tubb5/Tubb6/Haus1","Snrpe/Hnrnpa3/Snrpb/Magoh/Ybx1/Snrnp40/Hnrnpd/Lsm8/Hnrnpa2b1/Lsm5/Lsm3/Hnrnpf/Snrpa1/Fus/Lsm4/Lsm6/U2surp/Sf3b5/Snrpd3/Ptbp1/Lsm7/Srsf2/Alyref/Phf5a/Snu13/Tra2b/Rnps1/Srsf3/U2af1/Lsm2/Srsf7/Snrpd1","Sgo2a/Cenpf/Tubb4b/Spc25/Psmc3/Knl1/Ube2c/Pmf1/Cenpe/Cdc20/Cdca8/Nudc/Cenpa/Ppp1cc/Anapc5/Mad2l1/Psma1/Bub3/Spc24/Pttg1/Aurkb/Dynll2/Psmd12/Cenph/Rad21/Cenpm/Tuba1b/Tuba1c/Cenpq/Tubb6/Incenp/Smc3/Smc1a","Dtymk/Uck2/Ndufs2/Ola1/Atp5g3/Dut/Ndufc1/Ndufb6/Vcp/Ak2/Sdhb/Eno1/Atp5k/Ran/Ndufa5/Tpi1/Gapdh/Ndufc2/Dctpp1/Ndufb7/Impdh2/Eno3/Nme1/Atp5g1/Galk1/Ier3/Ndufv2/Pgam1/Ndufb8/Pgk1","Rps27/Rps20/Gm11808/Rps9/Rps19/Uba52/Rps29/Rps24/Eif3e/Eif4b/Eif4a2/Rps10/Rps28/Rps18","Snrpe/Hnrnpu/Hnrnpa3/Snrpb/Srsf6/Rbm8a/Psip1/Magoh/Sfpq/Lsm8/Hnrnpa2b1/Lsm5/Lsm3/Snrpa/Snrpa1/Lsm4/Lsm6/Ddx39a/Cwc15/Sf3b5/Snrpd3/Dazap1/Luc7l3/Srsf2/Zcrb1/Snrpc/Srsf3/Hnrnpm/Lsm2/Srsf7/Snrpd1","Snrpe/Hnrnpu/Hnrnpa3/Snrpb/Srsf6/Rbm8a/Psip1/Magoh/Sfpq/Lsm8/Hnrnpa2b1/Lsm5/Lsm3/Snrpa/Snrpa1/Lsm4/Lsm6/Ddx39a/Cwc15/Sf3b5/Snrpd3/Dazap1/Luc7l3/Srsf2/Zcrb1/Snrpc/Srsf3/Hnrnpm/Lsm2/Srsf7/Snrpd1","Snrpe/Hnrnpu/Hnrnpa3/Snrpb/Srsf6/Rbm8a/Psip1/Magoh/Sfpq/Lsm8/Hnrnpa2b1/Lsm5/Lsm3/Snrpa/Snrpa1/Lsm4/Lsm6/Ddx39a/Cwc15/Sf3b5/Snrpd3/Dazap1/Luc7l3/Srsf2/Zcrb1/Snrpc/Srsf3/Hnrnpm/Lsm2/Srsf7/Snrpd1","Sgo2a/Cenpf/Tubb4b/Ssna1/Spc25/Knl1/Smc4/Pmf1/Cenpe/Smc2/Cdc20/Cdca8/Nudc/Cenpa/Ppp1cc/Mad2l1/Bub3/Spc24/Ccnb2/Cdk1/Aurkb/Dynll2/Cenph/Ccnb1/Rad21/Cenpm/Tuba1b/Tuba1c/Tubb5/Cenpq/Tubb6/Haus1/Incenp/Smc3/Smc1a","Mcm3/Mcm6/Psmc3/Pcna/Ube2c/Ccna2/Cks1b/Pole3/Rfc1/Rfc5/Anapc5/Rfc2/Mcm7/Rfc3/Rpa3/Lig1/Psma1/Ccnd1/Mcm5/Orc6/Cdk4/Prim1/Psmd12/Gmnn/Esco2/Rad21/Mcm4/Rfc4/Fen1/Smc3/Smc1a","Hnrnpu/Cenpf/Bub1b/Knstrn/Knl1/Bub1/Mapre1/Top1/Ube2c/Smc4/Cenpe/Smc2/Cdc20/Cdca8/Sfpq/Tacc3/Ran/Mad2l1/Plk1/Kif22/Mki67/Spc24/Kif23/Cenpw/Nup37/Pttg1/Birc5/Cenpx/Ccnb1/Dlgap5/Rad21/Racgap1/Naa50/Banf1/Slc25a5/Smc1a","Hnrnpu/Bub1b/Knstrn/Bub1/Tpx2/Mapre1/Ube2c/Aurka/Smc4/Cenpe/Smc2/Cdc20/Cdca8/Ywhah/Tacc3/Ran/Mad2l1/Plk1/Kif22/Mki67/Spc24/Kif23/Pttg1/Birc5/Ccnb1/Reep4/Rad21/Racgap1/Ranbp1/Naa50/Banf1/Kif20b/Smc1a","Cenpf/Tubb4b/Ssna1/Ckap5/Bub1b/Knl1/Bub1/Mapre1/Smc4/Cenpe/Smc2/Cdc20/Cdca8/Cenpa/Dynll1/Mad2l1/Plk1/Spc24/Ccnb2/Nup37/Dynll2/Ccnb1/Rad21/Rangap1/Cenpm/Tuba1b/Tuba1a/Tubb5/Tubb6/Haus1/Smc1a","Cenpf/Tubb4b/Ckap5/Bub1b/Knl1/Bub1/Mapre1/Cenpe/Cdc20/Cdca8/Cenpa/Dynll1/Mad2l1/Plk1/Spc24/Ccnb2/Nup37/Dynll2/Ccnb1/Rad21/Rangap1/Cenpm/Tuba1b/Tuba1a/Tubb6/Smc1a","Sgo2a/Cenpf/Tubb4b/Spc25/Knl1/Pmf1/Cenpe/Cdc20/Cdca8/Nudc/Cenpa/Ppp1cc/Mad2l1/Bub3/Spc24/Ccnb2/Cdk1/Aurkb/Dynll2/Cenph/Ccnb1/Rad21/Cenpm/Tuba1b/Tuba1c/Cenpq/Tubb6/Incenp/Smc3/Smc1a","Psmd1/Cenpf/Psmd14/Psmc3/Ckap5/Bub1b/Knl1/Bub1/Mapre1/Ube2c/Ccna2/Cenpe/Cdc20/Cdca8/Dbf4/Cenpa/Ywhah/Dynll1/Rpa3/Mad2l1/Psmc4/Psma1/Plk1/Psmd7/Spc24/H2ax/Ccnb2/Nup37/Ube2n/Dynll2/Psmb3/Psmd12/Ccnb1/Psmd6/Psmc6/Psmb5/Rangap1/Cenpm","Ndufs2/Fh1/Ndufc1/Ndufb6/Vcp/Sdhb/Eno1/Atp5k/Cox6a1/Mdh2/Hmgb1/Ndufa4/Ndufa5/Cycs/Suclg1/Chchd4/Tpi1/Gapdh/Ldha/Ndufc2/Uqcrc2/Ndufb7/Acat1/Cox5a/Uqcrc1/Cisd1/Uqcr10/Mdh1/Eno3/Galk1/Prelid1/Macroh2a1/Dnajc15/Oxct1/Cyc1/Uqcc2/Ier3/Ndufv2/Ppp1ca/Pgam1/Ndufb8/Pgk1","Knl1/Pcna/Ccna2/H2bc18/Gar1/H2az1/Rbbp4/Cenpa/Rfc1/Rfc5/Rfc2/Rfc3/Rpa3/Lig1/Smarca5/H2ax/Cenpw/Prim1/H2az2/Nhp2/Cenpx/H2ac15/H2ac24/H2ac8/H4c4/Cenph/Cenpm/Rfc4/Cenpq/Fen1/Dkc1/Rbbp7","Rpl37a/Gm2000/Rpl12/Rps27/Rps20/Gm11808/Rpl22/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Rps29/Rps24/Gnl3/Rpl37/Rpl30/Rpl3/Rps10/Rps28/Rps18/Rpl39/Rpl36a","Rpl37a/Gm2000/Rpl12/Rps27/Rps20/Gm11808/Rpl22/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Eef2/Rps29/Rps24/Rpl37/Rpl30/Eif3e/Rpl3/Eif4b/Eif4a2/Rps10/Rps28/Rps18/Rpl39/Rpl36a","Sgo2a/Lbr/Cenpf/Tubb4b/Spc25/Psmc3/Knl1/Ube2c/Pmf1/Lmna/Cenpe/Cdc20/Cdca8/Nudc/Cenpa/Ppp1cc/Anapc5/Ran/Mad2l1/Psma1/Bub3/Spc24/Ccnb2/Fbxo5/Cdk1/Pttg1/Aurkb/Dynll2/Kpnb1/Psmd12/Cenph/Ccnb1/Rad21/Cenpm/Tuba1b/Tuba1c/Cenpq/Lmnb1/Tubb6/Banf1/Incenp/Smc3/Smc1a","Sgo2a/Lbr/Cenpf/Tubb4b/Ssna1/Spc25/Psmc3/Knl1/Ube2c/Smc4/Pmf1/Lmna/H2bc18/Cenpe/H2az1/Smc2/Cdc20/Cdca8/Nudc/Cenpa/Ppp1cc/Anapc5/Ran/Mad2l1/Psma1/Bub3/Spc24/H2ax/Ccnb2/Arpp19/Fbxo5/Cdk1/H2az2/Pttg1/Aurkb/Dynll2/Kpnb1/Psmd12/H2ac15/H2ac24/H2ac8/H4c4/Cenph/Ccnb1/Rad21/Cenpm/Tuba1b/Tuba1c/Tubb5/Cenpq/Lmnb1/Tubb6/Haus1/Banf1/Incenp/Smc3/Smc1a","Rpl7/Rpl37a/Rpl12/Rps3a1/Rps27/Rps20/Rpl22/Rpl9/Rps9/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Rps7/Rps29/Rps24/Rpl37/Rpl30/Rpl8/Rpl3/Rps10/Rps28/Rps18/Rps14/Rpl39/Rps4x/Rpl36a","Mcm3/Sgo2a/Mcm6/Cenpf/Spc25/Psmc3/Knl1/Ube2c/Ccna2/Pmf1/H2bc18/Cenpe/Cdc20/Cdca8/Clspn/Nudc/Dbf4/Cenpa/Ywhah/Rfc5/Ppp1cc/Anapc5/Rfc2/Mcm7/Rfc3/Rpa3/Mad2l1/Psma1/Bub3/Mcm5/Orc6/Spc24/H2ax/Ccnb2/Topbp1/Cdk1/Ube2n/Rad50/Aurkb/Trp53/Dynll2/Psmd12/H4c4/Cenph/Ccnb1/Cenpm/Mcm4/Rfc4/Cenpq/Incenp","Rpl37a/Gm2000/Rpl12/Rps27/Rps20/Gm11808/Rpl22/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Rps29/Rps24/Rpl37/Rpl30/Rpl3/Rps10/Rps28/Rps18/Rpl39/Rpl36a","Rps3a1/Rps27/Rps20/Rps9/Rps19/Rps15a/Uba52/Rps7/Rps29/Rps24/Pabpc1/Eif3e/Eif3h/Eif4b/Eif4a2/Rps10/Rps28/Rps18/Rps14/Rps4x","Psmd1/Cenpf/Tubb4b/Psmd14/Psmc3/Ckap5/Bub1b/Knl1/Bub1/Mapre1/Ube2c/Cenpe/Cdc20/Cdca8/Cenpa/Dynll1/Mad2l1/Psmc4/Psma1/Plk1/Psmd7/Spc24/Nup37/Pttg1/Dynll2/Psmb3/Psmd12/Psmd6/Psmc6/Psmb5/Rad21/Rangap1/Cenpm/Tuba1b/Tuba1a/Tubb6/Smc1a","Rpl37a/Rpl12/Rps20/Rpl22/Rpl21/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Eef2/Rps29/Rps24/Rpl30/Rpl3/Eif4b/Eif4a2/Rps10/Rpl10a/Rps28/Rps18/Rpl36/Rpl39/Rpl36a","Rpl37a/Gm2000/Rpl12/Rps27/Rps20/Gm11808/Rpl22/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Rps29/Rps24/Rpl37/Rpl30/Rpl3/Rps10/Rps28/Rps18/Rpl39/Rpl36a","Rpl37a/Gm2000/Rpl12/Rps27/Rps20/Gm11808/Rpl22/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Rps29/Rps24/Rpl37/Rpl30/Rpl3/Rps10/Rps28/Rps18/Rpl39/Rpl36a","Rpl37a/Gm2000/Rpl12/Rps27/Rps20/Gm11808/Rpl22/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Rps29/Rps24/Rpl37/Rpl30/Eif3e/Rpl3/Rps10/Rps28/Rps18/Rpl39/Rpl36a","Mcm3/Mcm6/Nucks1/Uchl5/Cacybp/Ssrp1/Pcna/Ccna2/Pole3/Nasp/Rbbp4/Dbf4/Rfc1/Rfc5/Rfc2/Mcm7/Rfc3/Rpa3/Lig1/Rrm1/Mcm5/Smarca5/Rnaseh2a/Orc6/Tipin/Pclaf/Topbp1/Fbxo5/Cdk1/Nfic/Prim1/Rad50/Trp53/Alyref/Cenpx/Rrm2/Gmnn/Supt16/Esco2/Mcm4/Rfc4/Fen1/Fam111a/Smc3/Rbbp7","Rpl37a/Gm2000/Rpl12/Rps27/Rps20/Gm11808/Rpl22/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Rps29/Rps24/Rpl37/Rpl30/Eif3e/Rpl3/Eif4b/Eif4a2/Rps10/Rps28/Rps18/Rpl39/Rpl36a","Sgo2a/Cenpf/Spc25/Knstrn/Knl1/Nusap1/Ube2c/Smc4/Pmf1/Cenpe/Smc2/Cdc20/Cdca8/Nudc/Anapc5/Ran/Baz1b/Mad2l1/Rad18/Prc1/Bub3/Mki67/Smarca5/Ctcf/Spc24/Smarcc1/Fbxo5/Cenpw/Pttg1/Aurkb/Kpnb1/Top2a/Birc5/Cenpx/Smc6/Macroh2a1/Cenph/Ccnb1/Pbrm1/Esco2/Rad21/Racgap1/Naa50/Cenpq/Banf1/Incenp/Fen1/Kif11/Smc3/Slc25a5/Smc1a","Rpl7/Eef1b2/Rpl37a/Rpl12/Rps3a1/Rps27/Rps20/Rpl22/Rpl9/Rps9/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Eef2/Rps7/Rps29/Rps24/Rpl37/Rpl30/Pabpc1/Eif3e/Eif3h/Rpl8/Rpl3/Eif4b/Eif4a2/Rps10/Rps28/Rps18/Rps14/Rpl39/Rps4x/Rpl36a","Psmd1/Lbr/Cenpf/Tubb4b/Psmd14/Psmc3/Ckap5/Bub1b/Knl1/Bub1/Mapre1/Ube2c/Lmna/Cenpe/Cdc20/Cdca8/Cenpa/Dynll1/Ran/Mad2l1/Psmc4/Psma1/Plk1/Psmd7/Spc24/Ccnb2/Nup37/Pttg1/Dynll2/Psmb3/Psmd12/Ccnb1/Psmd6/Psmc6/Psmb5/Rad21/Rangap1/Cenpm/Tuba1b/Tuba1a/Lmnb1/Tubb6/Banf1/Smc1a","Psmd1/Lbr/Cenpf/Tubb4b/Ssna1/Psmd14/Psmc3/Ckap5/Bub1b/Knl1/Bub1/Mapre1/Ube2c/Smc4/Lmna/Cenpe/H2az1/Smc2/Cdc20/Cdca8/Cenpa/Dynll1/Ran/Mad2l1/Psmc4/Psma1/Plk1/Psmd7/Spc24/H2ax/Kif23/Ccnb2/Arpp19/Nup37/H2az2/Pttg1/Dynll2/Psmb3/Psmd12/Ccnb1/Psmd6/Psmc6/Psmb5/Rad21/Rangap1/Cenpm/Tuba1b/Tuba1a/Tubb5/Kif20a/Lmnb1/Tubb6/Haus1/Banf1/Smc1a","Rpl7/Rpl37a/Rpl12/Rps3a1/Rps27/Rps20/Rpl22/Rpl9/Rps9/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Rps7/Rps29/Rps24/Rpl37/Rpl30/Pabpc1/Rpl8/Rpl3/Rps10/Rps28/Rps18/Rps14/Rpl39/Rps4x/Rpl36a","Rpl7/Rpl37a/Rpl12/Rps3a1/Rps27/Rps20/Rpl22/Rpl9/Rps9/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Rps7/Rps29/Rps24/Rpl37/Rpl30/Rpl8/Rpl3/Rps10/Rps28/Rps18/Rps14/Rpl39/Rps4x/Rpl36a","Rpl7/Rpl37a/Rpl12/Rps3a1/Rps27/Rps20/Rpl22/Rpl9/Rps9/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Rps7/Rps29/Rps24/Rpl37/Rpl30/Pabpc1/Rpl8/Rpl3/Rps10/Rps28/Rps18/Rps14/Rpl39/Rps4x/Rpl36a","Rpl7/Rpl37a/Rpl12/Rps3a1/Rps27/Rps20/Rpl22/Rpl9/Rps9/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Rps7/Rps29/Rps24/Rpl37/Rpl30/Eif3e/Eif3h/Rpl8/Rpl3/Rps10/Rps28/Rps18/Rps14/Rpl39/Rps4x/Rpl36a","Rpl7/Rpl37a/Rpl12/Rps3a1/Rps20/Rpl22/Rpl9/Rpl21/Rps9/Rps5/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Eef2/Rps7/Rps29/Rps24/Rpl30/Pabpc1/Rpl8/Rpl3/Eif4b/Eif4a2/Rps10/Rpl10a/Rps28/Rps18/Rps14/Rpl17/Rpl39/Rps4x/Rpl36a","Rpl7/Rpl37a/Rpl12/Rps3a1/Rps27/Rps20/Rpl22/Rpl9/Rps9/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Rps7/Rps29/Rps24/Rpl37/Rpl30/Pabpc1/Eif3e/Eif3h/Rpl8/Rpl3/Eif4b/Eif4a2/Rps10/Rps28/Rps18/Rps14/Rpl39/Rps4x/Rpl36a"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>cluster<\/th>\n      <th>ID<\/th>\n      <th>Description<\/th>\n      <th>GeneRatio<\/th>\n      <th>BgRatio<\/th>\n      <th>pvalue<\/th>\n      <th>p.adjust<\/th>\n      <th>Count<\/th>\n      <th>setSize<\/th>\n      <th>GeneProp<\/th>\n      <th>geneID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":9,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[5,6,7,8,9]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render"],"jsHooks":[]}</script>
</div>
<div id="number-of-gene-sets" class="section level4" number="6.1.2">
<h4><span class="header-section-number">6.1.2</span> Number of gene
sets</h4>
<pre class="r"><code>gse_df %&gt;% 
  filter(p.adjust &lt; p_thrs) %&gt;% 
  group_by(group) %&gt;% 
  summarise(n = n()) %&gt;% 
  ggplot(aes(x=group, y=n)) +
  geom_bar(stat=&quot;identity&quot;, color = &#39;black&#39;)+
  geom_text(aes(label=n), vjust=1.4, color=&quot;white&quot;, size=geom_text_size)+
  labs(
    x = &#39;Clusters&#39;,
    y = &#39;Number of gene sets&#39;,
    title = &#39;Number of significant gene sets by cluster&#39;,
    caption = paste(&quot;Adjusted P value &lt;&quot;, p_thrs)
    
  )</code></pre>
<p><img src="figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/cluster-marker-gse-enricher-c2reactome_gobp-ngset-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cluster-marker-gse-enricher-c2reactome_gobp-ngset-1">
Past versions of cluster-marker-gse-enricher-c2reactome_gobp-ngset-1.png
</button>
</p>
<div id="fig-cluster-marker-gse-enricher-c2reactome_gobp-ngset-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/39b365c03bef848fb12a4a5aee47a955aecb169a/docs/figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/cluster-marker-gse-enricher-c2reactome_gobp-ngset-1.png" target="_blank">39b365c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gene-set-similarity-matrix" class="section level4"
number="6.1.3">
<h4><span class="header-section-number">6.1.3</span> Gene-set similarity
matrix</h4>
<p>Heatmap showing the pair-wise similarity matrix of top 20 gene sets
with an adjusted P value &lt; 0.01 in each gene group. Heatmap colors
represent the Jaccard similarity coefficient.</p>
<pre class="r"><code>col_fun &lt;- colorRamp2(seq(0, 1, length.out = 4), brewer.pal(4, &quot;GnBu&quot;))
n_split &lt;- 10

use_mat &lt;- jc_matrix
colnames(use_mat) &lt;- colnames(use_mat) %&gt;% clean_msigdb_names %&gt;% tolower() %&gt;% str_to_title %&gt;% ep_str_wrap(70)
rownames(use_mat) &lt;- rownames(use_mat) %&gt;% clean_msigdb_names %&gt;% tolower() %&gt;% str_to_title %&gt;% ep_str_wrap(70)

ha_top &lt;- HeatmapAnnotation(
  foo = anno_block(
    # labels = c(&quot;Translation&quot;, &quot;Cell division&quot;),
    labels = 1:n_split,
    labels_gp = gpar(col = &quot;black&quot;, fontsize = 8),
    gp = gpar(lwd = 0, lty = 0))
  )

ht &lt;- Heatmap(
  use_mat,
  name = &#39;Jaccard index&#39;,
  column_split = n_split,
  row_split = n_split,
  column_title = NULL,
  row_title = NULL,
  col = col_fun,
  show_column_dend = FALSE,
  show_column_names = FALSE,
  border = TRUE,
  top_annotation = ha_top,
  heatmap_legend_param = list(title_gp = gpar(fontsize = 8),labels_gp = gpar(fontsize = 8)),
  width = unit(7, &quot;cm&quot;))

ht = draw(ht)</code></pre>
<p><img src="figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/cluster-marker-gse-enricher-c2reactome_gobp-jcmatrix-1.png" width="4320" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cluster-marker-gse-enricher-c2reactome_gobp-jcmatrix-1">
Past versions of
cluster-marker-gse-enricher-c2reactome_gobp-jcmatrix-1.png
</button>
</p>
<div id="fig-cluster-marker-gse-enricher-c2reactome_gobp-jcmatrix-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/39b365c03bef848fb12a4a5aee47a955aecb169a/docs/figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/cluster-marker-gse-enricher-c2reactome_gobp-jcmatrix-1.png" target="_blank">39b365c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n&quot;)</code></pre>
<pre class="r"><code># Extract term order according to hclust from similarity heatmap
row_order &lt;- row_order(ht) %&gt;% set_names(1:n_split)
term_similarity_clusters &lt;- foreach(i=names(row_order), .combine = rbind) %do% {
  data.frame(
    jc_sim_cl = i, 
    row_order = row_order[[i]], 
    Description =  rownames(jc_matrix)[row_order[[i]]]
  )
}

# Add similarity order to data for dot plot
res_top %&lt;&gt;% left_join(term_similarity_clusters)</code></pre>
</div>
<div id="dotplots" class="section level4" number="6.1.4">
<h4><span class="header-section-number">6.1.4</span> Dotplots</h4>
<p>The plots below show the top 20 gene sets with an adjusted P value
&lt; 0.01 in each gene group</p>
<pre class="r"><code>res_top %&gt;%
  # Combine similarity clusters
  mutate(
    # jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;5&#39;, &#39;10&#39;), &#39;5-10&#39;, jc_sim_cl),
    # jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;2&#39;, &#39;4&#39;), &#39;2-4&#39;, jc_sim_cl),
    # jc_sim_cl = factor(
    #   jc_sim_cl,
    #   levels = c(&#39;1&#39;, &#39;3&#39;, &#39;5-10&#39;, &#39;6&#39;, &#39;8&#39;, &#39;9&#39;, &#39;2-4&#39;, &#39;7&#39;)
    #   ),

    p.adjust = ifelse(p.adjust &gt; 0.05, 1, p.adjust),
    p.adjust = -log10( p.adjust),
    p.adjust = ifelse(p.adjust &gt;= 10, 10, p.adjust),
  ) %&gt;% 
  ggplot(aes(group, Description, color = GeneProp, size = p.adjust)) +
  geom_point() +
  scale_color_distiller(palette = &quot;Purples&quot;, na.value = &quot;grey70&quot;, direction = 1) +
  scale_y_discrete(labels = default_labeller(70)) +
  scale_x_discrete(position = &quot;top&quot;) +
  labs(y = &#39;&#39;, x = &#39;&#39;) +
  # facet_grid(rows = vars(cluster), scale = &#39;free&#39;, space = &#39;free&#39;) +
  # facet_grid(rows = vars(kcluster), scale = &#39;free&#39;, space = &#39;free&#39;) +
  facet_grid(rows = vars(jc_sim_cl), scale = &#39;free&#39;, space = &#39;free&#39;) +
  theme(
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, size = 1),
    # axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    axis.line.x = element_blank(),
    panel.grid.major.y = element_line(size = 0.2, colour=&quot;grey60&quot;, linetype = &#39;longdash&#39;),
    panel.grid.major.x = element_line(size = 0.2, colour=&quot;grey60&quot;, linetype = &#39;longdash&#39;),
    strip.background = element_blank()#,
    # strip.text.y = element_blank()
    )</code></pre>
<p><img src="figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/cluster-marker-gse-enricher-c2reactome_gobp-dotplot-1.png" width="4320" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cluster-marker-gse-enricher-c2reactome_gobp-dotplot-1">
Past versions of
cluster-marker-gse-enricher-c2reactome_gobp-dotplot-1.png
</button>
</p>
<div id="fig-cluster-marker-gse-enricher-c2reactome_gobp-dotplot-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/39b365c03bef848fb12a4a5aee47a955aecb169a/docs/figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/cluster-marker-gse-enricher-c2reactome_gobp-dotplot-1.png" target="_blank">39b365c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="ora---c2.cp.reactome-and-gobp-for-selected-clusters"
class="section level3" number="6.2">
<h3><span class="header-section-number">6.2</span> ORA - C2.CP.REACTOME
and <a href="GO:BP" class="uri">GO:BP</a> for selected clusters</h3>
<p>Over-enrichment analysis using genes with a mean AUC ≥ 0.25 in the
score markers results.</p>
<pre class="r"><code>gse_list &lt;- gse_cl_auc
test_type &lt;- &#39;enricher&#39;
gset_collection &lt;- &#39;msigdb.m2.cp.reactome&#39;
direction &lt;- &#39;abs&#39;
p_thrs &lt;- 0.0001
topn &lt;- 20
selected_clusters &lt;- data.frame(
  site = c(&#39;PT&#39;, &#39;PT&#39;, &#39;Lung&#39;, &#39;Lung&#39;, &#39;Liver&#39;, &#39;Liver&#39;),
  group = c(&#39;11&#39;, &#39;4&#39;, &#39;8&#39;, &#39;6&#39;, &#39;7&#39;, &#39;12&#39;)
)

# Load all the results
gse_df &lt;- foreach(i = selected_clusters$group, .combine = rbind) %do% {
  i_1 &lt;- gse_list[[i]][[test_type]][[direction]][[gset_collection]]@result %&gt;% 
    mutate(group = i)
  i_2 &lt;- gse_list[[i]][[&#39;enrichGO&#39;]][[direction]][[&#39;BP&#39;]]@result %&gt;% 
    filter(simplify) %&gt;% 
    dplyr::select(-simplify) %&gt;% 
    mutate(group = i)
  rbind(i_1, i_2)
} %&gt;% 
  mutate(
    group_n = as.numeric(group),
    group = fct_reorder(group, group_n),
    Description = clean_msigdb_names(Description) %&gt;% tolower() %&gt;% str_to_title,
    Description = ifelse(grepl(&quot;^GO:&quot;, ID), paste0(&quot;GO:&quot;, Description), Description)
  ) %&gt;% 
  dplyr::select(-group_n)

# Select topn by group in selected gset
top_by_group &lt;- foreach(i = selected_clusters$group, .combine = c) %do% {
  gse_df %&gt;% 
    filter(group == i) %&gt;% 
    filter(p.adjust &lt; p_thrs) %&gt;% 
    arrange(p.adjust) %&gt;% pull(Description) %&gt;%
    head(topn)
}

# Data frame to plot
res_top &lt;- gse_df %&gt;% 
  filter(Description %in% top_by_group) %&gt;% 
  mutate(log.p.adjust = -log10(p.adjust))

# Rows and column orders by hclust
use_mat &lt;- res_top %&gt;% 
  dplyr::select(Description, log.p.adjust, group) %&gt;% 
  pivot_wider(names_from = group, values_from = log.p.adjust) %&gt;% 
  column_to_rownames(&#39;Description&#39;)
use_mat[is.na(use_mat)] &lt;- 0

row_hclust &lt;- hclust(dist(use_mat))
row_hclust_labels &lt;- row_hclust$labels[row_hclust$order]

col_hclust &lt;- hclust(dist(t(use_mat)))
col_hclust_labels &lt;- col_hclust$labels[col_hclust$order]

use_mat &lt;- use_mat[row_hclust_labels,col_hclust_labels]

# GSET clusters by kmeans using silhouette
use_kmax &lt;- ifelse(nrow(use_mat) &lt;= 20, nrow(use_mat)-2, 20)
nb_clust &lt;- fviz_nbclust(use_mat, kmeans, method = &quot;silhouette&quot;, k.max = use_kmax)
silhouette_best_k &lt;- nb_clust$data %&gt;% slice_max(y) %&gt;% pull(clusters) %&gt;% as.character %&gt;% as.numeric
# silhouette_best_k &lt;- 5
kmeans_mat &lt;- kmeans(use_mat, silhouette_best_k, nstart = 25)
kmeans_mat_df &lt;- kmeans_mat$cluster %&gt;%
  data.frame %&gt;%
  set_names(&#39;kcluster&#39;) %&gt;%
  rownames_to_column(&#39;Description&#39;) %&gt;%
  mutate(kcluster = as.character(kcluster))


# Order factors by hclust
res_top %&lt;&gt;% 
  left_join(kmeans_mat_df) %&gt;% 
  mutate(
    Description = factor(Description, levels = row_hclust_labels),
    group = fct_relevel(group, col_hclust_labels)
  )

# Calculate jaccard similarity matrix
gset_collection &lt;- foreach(i = levels(res_top$Description)) %do% {
  res_top %&gt;% filter(Description == i) %&gt;% 
    pull(geneID) %&gt;% 
    paste(collapse = &#39;/&#39;) %&gt;% 
    strsplit(&#39;/&#39;) %&gt;% unlist %&gt;% unique
}
names(gset_collection) &lt;- levels(res_top$Description)

use_terms &lt;- res_top$Description %&gt;% unique
jc_matrix &lt;- pairwise_terms_jc(gset_collection[use_terms])</code></pre>
<div id="results-table-1" class="section level4" number="6.2.1">
<h4><span class="header-section-number">6.2.1</span> Results table</h4>
<pre class="r"><code>gse_df %&gt;% 
  filter(p.adjust &lt; p_thrs) %&gt;% 
  arrange(desc(p.adjust)) %&gt;% 
  dplyr::select(group, ID:p.adjust, Count:GeneProp, geneID)  %&gt;%
  dplyr::rename(cluster = group) %&gt;% 
  datatable(., 
            rownames = FALSE, 
            filter = &#39;top&#39;, 
            caption = &#39;Results from over enrichment analysis&#39;,
            extensions = &#39;Buttons&#39;, 
            options = list(
              dom = &#39;Blfrtip&#39;,
              buttons = c(&#39;csv&#39;, &#39;excel&#39;)
            ))  %&gt;% 
  formatRound(c(&#39;GeneProp&#39;), digits = 2) %&gt;%
  formatSignif(c(&#39;pvalue&#39;, &#39;p.adjust&#39;), digits = 3)</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-1d513276832cd5f68000" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-1d513276832cd5f68000">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;4&quot;,&quot;6&quot;,&quot;7&quot;,&quot;8&quot;,&quot;11&quot;,&quot;12&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"7.59131121e-06\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"9.0078474296e-05\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"7\" data-max=\"42\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"28\" data-max=\"489\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.03680981595092\" data-max=\"0.344827586206897\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Results from over enrichment analysis<\/caption>","data":[["6","6","6","12","6","7","12","12","12","6","6","11","6","12","6","12","12","6","6","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","7","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","4","12","4","4","11","4","11","4","4","4","4","4","11","11","11","11","11","11","11"],["REACTOME_SCF_SKP2_MEDIATED_DEGRADATION_OF_P27_P21","REACTOME_RHO_GTPASES_ACTIVATE_WASPS_AND_WAVES","REACTOME_PROGRAMMED_CELL_DEATH","GO:0033108","REACTOME_AUF1_HNRNP_D0_BINDS_AND_DESTABILIZES_MRNA","GO:0010035","REACTOME_UB_SPECIFIC_PROCESSING_PROTEASES","REACTOME_NEDDYLATION","REACTOME_INTERCONVERSION_OF_NUCLEOTIDE_DI_AND_TRIPHOSPHATES","REACTOME_RHO_GTPASE_EFFECTORS","REACTOME_CELLULAR_RESPONSE_TO_CHEMICAL_STRESS","GO:0035914","REACTOME_METABOLISM_OF_POLYAMINES","REACTOME_GLUCONEOGENESIS","REACTOME_RHO_GTPASES_ACTIVATE_IQGAPS","REACTOME_CELL_CYCLE_CHECKPOINTS","REACTOME_TP53_REGULATES_METABOLIC_GENES","REACTOME_THE_ROLE_OF_GTSE1_IN_G2_M_PROGRESSION_AFTER_G2_CHECKPOINT","REACTOME_CELLULAR_RESPONSES_TO_STIMULI","REACTOME_PTEN_REGULATION","REACTOME_MITOTIC_G2_G2_M_PHASES","REACTOME_SIGNALING_BY_HEDGEHOG","REACTOME_M_PHASE","REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA","REACTOME_TNFR2_NON_CANONICAL_NF_KB_PATHWAY","REACTOME_HEDGEHOG_ON_STATE","REACTOME_G2_M_CHECKPOINTS","REACTOME_DEGRADATION_OF_BETA_CATENIN_BY_THE_DESTRUCTION_COMPLEX","REACTOME_DNA_REPLICATION","REACTOME_MRNA_SPLICING","REACTOME_HEDGEHOG_OFF_STATE","REACTOME_INTERLEUKIN_1_FAMILY_SIGNALING","REACTOME_TCR_SIGNALING","REACTOME_SEPARATION_OF_SISTER_CHROMATIDS","REACTOME_MAPK6_MAPK4_SIGNALING","REACTOME_APC_C_MEDIATED_DEGRADATION_OF_CELL_CYCLE_PROTEINS","REACTOME_CDK_MEDIATED_PHOSPHORYLATION_AND_REMOVAL_OF_CDC6","REACTOME_ABC_FAMILY_PROTEINS_MEDIATED_TRANSPORT","REACTOME_S_PHASE","REACTOME_MITOTIC_G1_PHASE_AND_G1_S_TRANSITION","REACTOME_DNA_REPLICATION_PRE_INITIATION","REACTOME_UCH_PROTEINASES","REACTOME_ANTIGEN_PROCESSING_CROSS_PRESENTATION","REACTOME_REGULATION_OF_PTEN_STABILITY_AND_ACTIVITY","REACTOME_BETA_CATENIN_INDEPENDENT_WNT_SIGNALING","REACTOME_CELLULAR_RESPONSES_TO_STIMULI","REACTOME_REGULATION_OF_RAS_BY_GAPS","REACTOME_RUNX1_REGULATES_TRANSCRIPTION_OF_GENES_INVOLVED_IN_DIFFERENTIATION_OF_HSCS","REACTOME_G1_S_DNA_DAMAGE_CHECKPOINTS","REACTOME_SWITCHING_OF_ORIGINS_TO_A_POST_REPLICATIVE_STATE","GO:0043620","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX3","REACTOME_ACTIVATION_OF_APC_C_AND_APC_C_CDC20_MEDIATED_DEGRADATION_OF_MITOTIC_PROTEINS","REACTOME_SIGNALING_BY_THE_B_CELL_RECEPTOR_BCR","REACTOME_PCP_CE_PATHWAY","REACTOME_APC_C_CDH1_MEDIATED_DEGRADATION_OF_CDC20_AND_OTHER_APC_C_CDH1_TARGETED_PROTEINS_IN_LATE_MITOSIS_EARLY_G1","REACTOME_ASYMMETRIC_LOCALIZATION_OF_PCP_PROTEINS","REACTOME_DOWNSTREAM_TCR_SIGNALING","REACTOME_REGULATION_OF_MRNA_STABILITY_BY_PROTEINS_THAT_BIND_AU_RICH_ELEMENTS","REACTOME_C_TYPE_LECTIN_RECEPTORS_CLRS","REACTOME_ORC1_REMOVAL_FROM_CHROMATIN","REACTOME_ASSEMBLY_OF_THE_PRE_REPLICATIVE_COMPLEX","REACTOME_CELLULAR_RESPONSE_TO_HYPOXIA","REACTOME_DECTIN_1_MEDIATED_NONCANONICAL_NF_KB_SIGNALING","REACTOME_GLI3_IS_PROCESSED_TO_GLI3R_BY_THE_PROTEASOME","REACTOME_MITOTIC_METAPHASE_AND_ANAPHASE","REACTOME_DEGRADATION_OF_DVL","REACTOME_STABILIZATION_OF_P53","REACTOME_G1_S_TRANSITION","REACTOME_FCERI_MEDIATED_NF_KB_ACTIVATION","REACTOME_DEGRADATION_OF_AXIN","REACTOME_ACTIVATION_OF_NF_KAPPAB_IN_B_CELLS","REACTOME_REGULATION_OF_RUNX3_EXPRESSION_AND_ACTIVITY","REACTOME_HEDGEHOG_LIGAND_BIOGENESIS","REACTOME_KEAP1_NFE2L2_PATHWAY","REACTOME_TRANSCRIPTIONAL_REGULATION_BY_RUNX2","REACTOME_REGULATION_OF_RUNX2_EXPRESSION_AND_ACTIVITY","REACTOME_UBIQUITIN_MEDIATED_DEGRADATION_OF_PHOSPHORYLATED_CDC25A","REACTOME_CYCLIN_A_CDK2_ASSOCIATED_EVENTS_AT_S_PHASE_ENTRY","REACTOME_CLEC7A_DECTIN_1_SIGNALING","REACTOME_CROSS_PRESENTATION_OF_SOLUBLE_EXOGENOUS_ANTIGENS_ENDOSOMES","REACTOME_FC_EPSILON_RECEPTOR_FCERI_SIGNALING","REACTOME_DOWNSTREAM_SIGNALING_EVENTS_OF_B_CELL_RECEPTOR_BCR","REACTOME_AUF1_HNRNP_D0_BINDS_AND_DESTABILIZES_MRNA","REACTOME_THE_ROLE_OF_GTSE1_IN_G2_M_PROGRESSION_AFTER_G2_CHECKPOINT","REACTOME_RESPIRATORY_ELECTRON_TRANSPORT","REACTOME_SCF_SKP2_MEDIATED_DEGRADATION_OF_P27_P21","REACTOME_METABOLISM_OF_POLYAMINES","REACTOME_CELLULAR_RESPONSE_TO_CHEMICAL_STRESS","REACTOME_RESPIRATORY_ELECTRON_TRANSPORT_ATP_SYNTHESIS_BY_CHEMIOSMOTIC_COUPLING_AND_HEAT_PRODUCTION_BY_UNCOUPLING_PROTEINS","REACTOME_THE_CITRIC_ACID_TCA_CYCLE_AND_RESPIRATORY_ELECTRON_TRANSPORT","GO:0009141","REACTOME_ACTIVATION_OF_THE_MRNA_UPON_BINDING_OF_THE_CAP_BINDING_COMPLEX_AND_EIFS_AND_SUBSEQUENT_BINDING_TO_43S","GO:0006091","REACTOME_MAJOR_PATHWAY_OF_RRNA_PROCESSING_IN_THE_NUCLEOLUS_AND_CYTOSOL","REACTOME_TRANSLATION","REACTOME_MAJOR_PATHWAY_OF_RRNA_PROCESSING_IN_THE_NUCLEOLUS_AND_CYTOSOL","REACTOME_NONSENSE_MEDIATED_DECAY_NMD","REACTOME_ACTIVATION_OF_THE_MRNA_UPON_BINDING_OF_THE_CAP_BINDING_COMPLEX_AND_EIFS_AND_SUBSEQUENT_BINDING_TO_43S","GO:0002181","REACTOME_SRP_DEPENDENT_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE","REACTOME_NONSENSE_MEDIATED_DECAY_NMD_INDEPENDENT_OF_THE_EXON_JUNCTION_COMPLEX_EJC","REACTOME_FORMATION_OF_A_POOL_OF_FREE_40S_SUBUNITS","REACTOME_EUKARYOTIC_TRANSLATION_INITIATION","REACTOME_TRANSLATION","REACTOME_NONSENSE_MEDIATED_DECAY_NMD","REACTOME_SRP_DEPENDENT_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE","REACTOME_NONSENSE_MEDIATED_DECAY_NMD_INDEPENDENT_OF_THE_EXON_JUNCTION_COMPLEX_EJC","REACTOME_FORMATION_OF_A_POOL_OF_FREE_40S_SUBUNITS","GO:0002181","REACTOME_EUKARYOTIC_TRANSLATION_INITIATION"],["Scf Skp2 Mediated Degradation Of P27 P21","Rho Gtpases Activate Wasps And Waves","Programmed Cell Death","GO:Mitochondrial Respiratory Chain Complex Assembly","Auf1 Hnrnp D0 Binds And Destabilizes Mrna","GO:Response To Inorganic Substance","Ub Specific Processing Proteases","Neddylation","Interconversion Of Nucleotide Di And Triphosphates","Rho Gtpase Effectors","Cellular Response To Chemical Stress","GO:Skeletal Muscle Cell Differentiation","Metabolism Of Polyamines","Gluconeogenesis","Rho Gtpases Activate Iqgaps","Cell Cycle Checkpoints","Tp53 Regulates Metabolic Genes","The Role Of Gtse1 In G2 M Progression After G2 Checkpoint","Cellular Responses To Stimuli","Pten Regulation","Mitotic G2 G2 M Phases","Signaling By Hedgehog","M Phase","Processing Of Capped Intron Containing Pre Mrna","Tnfr2 Non Canonical Nf Kb Pathway","Hedgehog On State","G2 M Checkpoints","Degradation Of Beta Catenin By The Destruction Complex","Dna Replication","Mrna Splicing","Hedgehog Off State","Interleukin 1 Family Signaling","Tcr Signaling","Separation Of Sister Chromatids","Mapk6 Mapk4 Signaling","Apc C Mediated Degradation Of Cell Cycle Proteins","Cdk Mediated Phosphorylation And Removal Of Cdc6","Abc Family Proteins Mediated Transport","S Phase","Mitotic G1 Phase And G1 S Transition","Dna Replication Pre Initiation","Uch Proteinases","Antigen Processing Cross Presentation","Regulation Of Pten Stability And Activity","Beta Catenin Independent Wnt Signaling","Cellular Responses To Stimuli","Regulation Of Ras By Gaps","Runx1 Regulates Transcription Of Genes Involved In Differentiation Of Hscs","G1 S Dna Damage Checkpoints","Switching Of Origins To A Post Replicative State","GO:Regulation Of Dna-Templated Transcription In Response To Stress","Transcriptional Regulation By Runx3","Activation Of Apc C And Apc C Cdc20 Mediated Degradation Of Mitotic Proteins","Signaling By The B Cell Receptor Bcr","Pcp Ce Pathway","Apc C Cdh1 Mediated Degradation Of Cdc20 And Other Apc C Cdh1 Targeted Proteins In Late Mitosis Early G1","Asymmetric Localization Of Pcp Proteins","Downstream Tcr Signaling","Regulation Of Mrna Stability By Proteins That Bind Au Rich Elements","C Type Lectin Receptors Clrs","Orc1 Removal From Chromatin","Assembly Of The Pre Replicative Complex","Cellular Response To Hypoxia","Dectin 1 Mediated Noncanonical Nf Kb Signaling","Gli3 Is Processed To Gli3r By The Proteasome","Mitotic Metaphase And Anaphase","Degradation Of Dvl","Stabilization Of P53","G1 S Transition","Fceri Mediated Nf Kb Activation","Degradation Of Axin","Activation Of Nf Kappab In B Cells","Regulation Of Runx3 Expression And Activity","Hedgehog Ligand Biogenesis","Keap1 Nfe2l2 Pathway","Transcriptional Regulation By Runx2","Regulation Of Runx2 Expression And Activity","Ubiquitin Mediated Degradation Of Phosphorylated Cdc25a","Cyclin A Cdk2 Associated Events At S Phase Entry","Clec7a Dectin 1 Signaling","Cross Presentation Of Soluble Exogenous Antigens Endosomes","Fc Epsilon Receptor Fceri Signaling","Downstream Signaling Events Of B Cell Receptor Bcr","Auf1 Hnrnp D0 Binds And Destabilizes Mrna","The Role Of Gtse1 In G2 M Progression After G2 Checkpoint","Respiratory Electron Transport","Scf Skp2 Mediated Degradation Of P27 P21","Metabolism Of Polyamines","Cellular Response To Chemical Stress","Respiratory Electron Transport Atp Synthesis By Chemiosmotic Coupling And Heat Production By Uncoupling Proteins","The Citric Acid Tca Cycle And Respiratory Electron Transport","GO:Nucleoside Triphosphate Metabolic Process","Activation Of The Mrna Upon Binding Of The Cap Binding Complex And Eifs And Subsequent Binding To 43s","GO:Generation Of Precursor Metabolites And Energy","Major Pathway Of Rrna Processing In The Nucleolus And Cytosol","Translation","Major Pathway Of Rrna Processing In The Nucleolus And Cytosol","Nonsense Mediated Decay Nmd","Activation Of The Mrna Upon Binding Of The Cap Binding Complex And Eifs And Subsequent Binding To 43s","GO:Cytoplasmic Translation","Srp Dependent Cotranslational Protein Targeting To Membrane","Nonsense Mediated Decay Nmd Independent Of The Exon Junction Complex Ejc","Formation Of A Pool Of Free 40s Subunits","Eukaryotic Translation Initiation","Translation","Nonsense Mediated Decay Nmd","Srp Dependent Cotranslational Protein Targeting To Membrane","Nonsense Mediated Decay Nmd Independent Of The Exon Junction Complex Ejc","Formation Of A Pool Of Free 40s Subunits","GO:Cytoplasmic Translation","Eukaryotic Translation Initiation"],["9/167","7/167","13/167","10/292","9/167","18/146","16/209","18/209","7/209","19/167","15/167","10/207","10/167","8/209","8/167","22/209","11/209","12/167","28/167","14/209","18/209","16/209","28/209","21/209","14/209","13/209","18/209","13/209","16/209","19/209","15/209","16/209","15/209","20/209","13/209","14/209","13/209","15/209","18/209","17/209","15/209","15/209","13/209","13/209","17/209","33/209","13/209","13/209","13/209","15/209","10/146","14/209","14/209","16/209","15/209","14/209","13/209","15/209","15/209","17/209","14/209","15/209","14/209","13/209","13/209","24/209","13/209","13/209","17/209","15/209","13/209","14/209","13/209","14/209","15/209","14/209","13/209","13/209","15/209","17/209","13/209","19/209","16/209","14/209","16/209","17/209","15/209","15/209","23/209","20/209","25/209","30/292","14/90","42/292","23/90","26/90","28/125","22/90","20/125","24/139","22/90","22/90","23/90","25/90","35/125","29/125","28/125","29/125","30/125","34/207","33/125"],["60/8764","32/8764","132/8764","87/18646","56/8764","489/18646","184/8764","226/8764","28/8764","245/8764","154/8764","82/18646","58/8764","35/8764","29/8764","284/8764","72/8764","75/8764","447/8764","113/8764","183/8764","140/8764","389/8764","231/8764","98/8764","82/8764","162/8764","80/8764","126/8764","177/8764","108/8764","119/8764","103/8764","187/8764","74/8764","87/8764","73/8764","100/8764","147/8764","130/8764","98/8764","98/8764","70/8764","69/8764","125/8764","447/8764","67/8764","67/8764","66/8764","92/8764","55/18646","78/8764","77/8764","104/8764","87/8764","74/8764","62/8764","85/8764","85/8764","112/8764","71/8764","83/8764","70/8764","58/8764","58/8764","225/8764","57/8764","56/8764","105/8764","78/8764","55/8764","65/8764","54/8764","64/8764","76/8764","63/8764","52/8764","52/8764","72/8764","93/8764","48/8764","121/8764","79/8764","56/8764","75/8764","87/8764","60/8764","58/8764","154/8764","108/8764","156/8764","262/18646","58/8764","444/18646","172/8764","222/8764","172/8764","108/8764","58/8764","144/18646","87/8764","88/8764","95/8764","113/8764","222/8764","108/8764","87/8764","88/8764","95/8764","144/18646","113/8764"],[1.715452342114762e-06,1.812444150824018e-06,1.235471812745524e-06,1.050958206829753e-06,9.384099782508034e-07,5.431943662531831e-08,7.591311209466533e-06,6.978482967576558e-06,3.068275866655178e-06,1.830285665735168e-07,2.092801764936026e-07,2.441733874674364e-08,1.143408557535838e-07,1.237545959817324e-06,4.487358864857772e-08,1.013356470084714e-06,9.214541922874864e-07,1.424782767647396e-08,1.976163962748923e-08,4.262629263137256e-07,3.262758207309191e-07,1.941290183077741e-07,1.417034398970564e-07,1.266127771453036e-07,6.951021588001929e-08,5.82205750685598e-08,5.035143380739822e-08,4.289174614131474e-08,4.355037244395165e-08,3.700223903316719e-08,3.440757950932827e-08,1.901799899345143e-08,1.782116764726201e-08,1.691329305526261e-08,1.615699668084336e-08,1.452082192843878e-08,1.360127118357026e-08,1.177827051426226e-08,1.081617893821204e-08,1.032360445366778e-08,8.85958871137054e-09,8.85958871137054e-09,7.970040844653338e-09,6.62764035420462e-09,5.623564182495312e-09,4.982486013503258e-09,4.537277850644803e-09,4.537277850644803e-09,3.734534559745945e-09,3.604166396011438e-09,1.373727429644659e-11,3.320906584304546e-09,2.782740690141034e-09,2.571733885998327e-09,1.608610454934318e-09,1.609020773288541e-09,1.650321247605337e-09,1.146282619248951e-09,1.146282619248951e-09,9.939875735965051e-10,9.046875882223611e-10,8.087383228975807e-10,7.417305114677726e-10,6.817574649986138e-10,6.817574649986138e-10,5.998902663725902e-10,5.401795335925893e-10,4.258334815405488e-10,3.512331962808177e-10,3.225537453936771e-10,3.339213941180774e-10,2.600461834982142e-10,2.604084099493095e-10,2.083715640003312e-10,2.187838462743097e-10,1.662492303157139e-10,1.556245030557619e-10,1.556245030557619e-10,9.680065196417147e-11,4.755939538384787e-11,5.142685306902083e-11,5.338431072739102e-11,3.535255244251123e-11,2.990448956682378e-11,1.524484477930366e-11,1.547264345223117e-11,5.735518509533642e-12,3.345775899591841e-12,1.318017726563684e-12,6.62643622301563e-13,2.437319418066059e-14,1.486306438243293e-17,3.577471216581735e-16,5.018544515324692e-21,6.249566292377718e-20,5.056194156525946e-21,2.171440105580242e-22,2.860517144939031e-23,2.968763736508901e-23,6.994762813787149e-26,1.565810583980426e-25,2.07211734867929e-25,3.339266229483267e-26,2.074952222664137e-27,1.549728195385448e-27,6.549785945347772e-30,2.16473835197048e-31,7.335863541292111e-33,2.374729610342888e-33,8.888199046533719e-36,1.965790247234329e-35],[9.007847429595371e-05,9.007847429595371e-05,7.675368636681569e-05,7.139661731180388e-05,6.662710845580704e-05,5.165778423067772e-05,4.5646455714065e-05,4.251365281563088e-05,1.894148968348463e-05,1.733537461955342e-05,1.733537461955342e-05,1.547570929768612e-05,1.420685132738279e-05,7.743024045884065e-06,7.434057852781041e-06,6.427178707523596e-06,5.925462375404253e-06,4.910767447431073e-06,4.910767447431073e-06,2.779714575820492e-06,2.158081499977365e-06,1.302633847485498e-06,9.648337157696631e-07,8.749509823623217e-07,4.876246962492262e-07,4.147096347191259e-07,3.642611539503965e-07,3.200606736753907e-07,3.200606736753907e-07,2.80853060202564e-07,2.655118218803165e-07,1.492429412536952e-07,1.422620796669364e-07,1.37383415519063e-07,1.335837404148299e-07,1.222389191430392e-07,1.166183066295006e-07,1.02893193360442e-07,9.630559323831099e-08,9.372213454996438e-08,8.20397914672912e-08,8.20397914672912e-08,7.687768564738533e-08,6.528930817014338e-08,5.660239601076803e-08,5.126424498337797e-08,4.774453738292145e-08,4.774453738292145e-08,4.116879764672316e-08,4.070070832569014e-08,3.919244356776213e-08,3.843949371332512e-08,3.303612665475125e-08,3.133454708466383e-08,2.065131723354787e-08,2.065131723354787e-08,2.065131723354787e-08,1.560967213859601e-08,1.560967213859601e-08,1.438175770547443e-08,1.351194688215978e-08,1.248152811671933e-08,1.184211126929582e-08,1.127334665336993e-08,1.127334665336993e-08,1.068266128194266e-08,1.000412496213475e-08,8.215037581386421e-09,7.070476951218199e-09,7.027527521666811e-09,7.027527521666811e-09,6.028454690326515e-09,6.028454690326515e-09,5.6276067125003e-09,5.6276067125003e-09,4.81083710226097e-09,4.803609660987851e-09,4.803609660987851e-09,3.447592450723953e-09,2.05974465556517e-09,2.05974465556517e-09,2.05974465556517e-09,1.818692420098078e-09,1.730722333679926e-09,1.02340484548329e-09,1.02340484548329e-09,5.311090139828153e-10,3.872735603777556e-10,2.034140691329953e-10,1.534019985628118e-10,1.128478890564585e-11,1.814853950800278e-14,1.703770666897051e-14,1.568295161038966e-17,3.401549653422729e-18,3.210683289393975e-19,1.235006560048763e-20,2.179714064443542e-21,1.929696428730786e-21,1.65496088174204e-22,1.973691774617024e-23,1.973691774617024e-23,6.361302167165623e-24,7.905567968350362e-25,1.175210548167298e-25,5.960305210266473e-28,2.462389875366421e-29,1.112605970429303e-30,5.40250986353007e-31,2.816670277846535e-32,8.9443456249162e-33],[9,7,13,10,9,18,16,18,7,19,15,10,10,8,8,22,11,12,28,14,18,16,28,21,14,13,18,13,16,19,15,16,15,20,13,14,13,15,18,17,15,15,13,13,17,33,13,13,13,15,10,14,14,16,15,14,13,15,15,17,14,15,14,13,13,24,13,13,17,15,13,14,13,14,15,14,13,13,15,17,13,19,16,14,16,17,15,15,23,20,25,30,14,42,23,26,28,22,20,24,22,22,23,25,35,29,28,29,30,34,33],[60,32,132,87,56,489,184,226,28,245,154,82,58,35,29,284,72,75,447,113,183,140,389,231,98,82,162,80,126,177,108,119,103,187,74,87,73,100,147,130,98,98,70,69,125,447,67,67,66,92,55,78,77,104,87,74,62,85,85,112,71,83,70,58,58,225,57,56,105,78,55,65,54,64,76,63,52,52,72,93,48,121,79,56,75,87,60,58,154,108,156,262,58,444,172,222,172,108,58,144,87,88,95,113,222,108,87,88,95,144,113],[0.15,0.21875,0.09848484848484848,0.1149425287356322,0.1607142857142857,0.03680981595092025,0.08695652173913043,0.07964601769911504,0.25,0.07755102040816327,0.09740259740259741,0.1219512195121951,0.1724137931034483,0.2285714285714286,0.2758620689655172,0.07746478873239436,0.1527777777777778,0.16,0.06263982102908278,0.1238938053097345,0.09836065573770492,0.1142857142857143,0.07197943444730077,0.09090909090909091,0.1428571428571428,0.1585365853658537,0.1111111111111111,0.1625,0.126984126984127,0.1073446327683616,0.1388888888888889,0.134453781512605,0.145631067961165,0.106951871657754,0.1756756756756757,0.1609195402298851,0.1780821917808219,0.15,0.1224489795918367,0.1307692307692308,0.1530612244897959,0.1530612244897959,0.1857142857142857,0.1884057971014493,0.136,0.0738255033557047,0.1940298507462687,0.1940298507462687,0.196969696969697,0.1630434782608696,0.1818181818181818,0.1794871794871795,0.1818181818181818,0.1538461538461539,0.1724137931034483,0.1891891891891892,0.2096774193548387,0.1764705882352941,0.1764705882352941,0.1517857142857143,0.1971830985915493,0.1807228915662651,0.2,0.2241379310344828,0.2241379310344828,0.1066666666666667,0.2280701754385965,0.2321428571428572,0.1619047619047619,0.1923076923076923,0.2363636363636364,0.2153846153846154,0.2407407407407407,0.21875,0.1973684210526316,0.2222222222222222,0.25,0.25,0.2083333333333333,0.1827956989247312,0.2708333333333333,0.1570247933884298,0.2025316455696203,0.25,0.2133333333333333,0.1954022988505747,0.25,0.2586206896551724,0.1493506493506493,0.1851851851851852,0.1602564102564103,0.1145038167938931,0.2413793103448276,0.0945945945945946,0.1337209302325581,0.1171171171171171,0.1627906976744186,0.2037037037037037,0.3448275862068966,0.1666666666666667,0.2528735632183908,0.25,0.2421052631578947,0.2212389380530974,0.1576576576576577,0.2685185185185185,0.3218390804597701,0.3295454545454545,0.3157894736842105,0.2361111111111111,0.2920353982300885],["Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Ccnd1/Psmb3/Psmb5/Cdkn1a","Actr3/Arpc5/Actb/Rac1/Arpc4/Actr2/Actg1","Gsn/Lmna/Ywhah/Ywhag/Cycs/Chmp3/Bax/Casp3/Pdcd6ip/Nmt1/Hsp90aa1/Dsp/Plec","Ndufs2/Ndufc1/Ndufb6/Ndufa5/Chchd4/Ndufc2/Ndufb7/Coa3/Uqcc2/Ndufb8","Psmd1/Psmb7/Psmd14/Psma7/Hspb1/Psmd8/Psmb3/Psmb5/Eif4g1","Lcn2/Nr4a2/Cp/Jun/Cav1/Gapdh/Selenow/Fosb/Ucp2/Bnip3/Jund/Junb/Mt2/Mt1/Rhob/Fos/Nr4a1/Krt8","Psmb7/Psmd14/Psmc3/Psmb2/Ruvbl1/Psma1/Vdac3/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Vcp/Psmb2/Psma1/Commd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Nedd8/Elob/Rbbp7","Dtymk/Dut/Txn1/Ak2/Gsr/Nme1/Rrm2","Actr3/Arpc5/Tubb4b/Ywhah/Ywhag/Actb/Rac1/Arpc4/Iqgap1/Myl6/Actr2/Actg1/Rhob/Calm1/Myh9/Tuba1a/Tuba1c/Kif5b/Flna","Psmd1/Psmb7/Psmd14/H13/Psma7/Vcp/Txn1/Prdx1/Cycs/Psmd8/Gsr/Txnrd1/Psmb3/Psmb5/Prdx5","Gpc1/Btg2/Atf3/Nupr1/Cited2/Ddx5/Fos/Myc/Maff/Egr1","Psmd1/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Odc1/Psmb5/Sat1/Sms","Eno1/Mdh2/Tpi1/Gapdh/Mdh1/Eno3/Pgam1/Pgk1","Tubb4b/Actb/Rac1/Iqgap1/Actg1/Calm1/Tuba1a/Tuba1c","Mcm6/Psmb7/Psmd14/Psmc3/Psmb2/Cenpa/Ywhah/Dynll1/Psma1/Spc24/Ccnb2/Ube2n/Psmb6/Dynll2/Psmb3/Psmd12/Ywhaq/Psma3/Psma2/Psmd6/Psmb5/Psme2","Txn1/Prdx1/Ywhah/Cox6a1/Ndufa4/Cycs/Gsr/Prdx2/Cox5a/Ywhaq/Cox7b","Psmd1/Tubb4b/Psmb7/Psmd14/Psma7/Psmd8/Psmb3/Hsp90aa1/Psmb5/Tuba1a/Tuba1c/Cdkn1a","Psmd1/Tubb4b/Psmb7/Psmd14/H13/Eif2s2/Psma7/Dnaja1/Vcp/Txn1/Atp6v1g1/Prdx1/Atp6v0b/Atp6v1f/Cycs/Psmd8/Gsr/Hsbp1/Cryab/Txnrd1/Psmb3/Hsp90aa1/Psmb5/Tuba1a/Tuba1c/Atp6v0c/Cdkn1a/Prdx5","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Rbbp7","Psmb7/Psmd14/Psmc3/Psmb2/Dynll1/Psma1/Ccnb2/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c/Tubb5","Psmb7/Psmd14/Psmc3/Vcp/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c","Psmb7/Psmd14/Psmc3/Lmna/H2az1/Psmb2/Cenpa/Dynll1/Ran/Psma1/Spc24/Ccnb2/Arpp19/H2az2/Pttg1/Psmb6/Dynll2/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c/Tubb5/Banf1","Hnrnpa3/Snrpb/Ybx1/Hnrnpd/Hnrnpa2b1/Lsm5/Snrpa/Snrpa1/Lsm4/Lsm6/Ddx39a/Sf3b5/Snrpd3/Lsm7/Thoc7/Hnrnpa1/Snrpc/Srsf3/Srsf7/Snrpd1/Txnl4a","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tnfrsf12a","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Psmb2/Ywhah/Psma1/Ccnb2/Ube2n/Psmb6/Psmb3/Psmd12/Ywhaq/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Pcna/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Gmnn/Psmd6/Psmb5/Psme2","Hnrnpa3/Snrpb/Ybx1/Hnrnpd/Hnrnpa2b1/Lsm5/Snrpa/Snrpa1/Lsm4/Lsm6/Sf3b5/Snrpd3/Lsm7/Hnrnpa1/Snrpc/Srsf3/Srsf7/Snrpd1/Txnl4a","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c","Psmb7/Psmd14/Psmc3/Psmb2/Hmgb1/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Cenpa/Dynll1/Psma1/Spc24/Pttg1/Psmb6/Dynll2/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Pttg1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Vcp/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Eif2s1/Psma2/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Pcna/Cks1b/Psmb2/Psma1/Ccnd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Gmnn/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Cks1b/Psmb2/Psma1/Ccnd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Gmnn/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Gmnn/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Ruvbl1/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Nedd8","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Ap2s1/Calm3/Psma1/Psmb6/Pfn1/Psmb3/Psmd12/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Vcp/Txn1/Jun/Prdx1/Psmb2/Dynll1/Cox6a1/Ndufa4/Cycs/Psma1/Gsr/Prdx2/Mt2/Mt1/Cox5a/Psmb6/Dynll2/Psmb3/Psmd12/Psma3/Eif2s1/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c/Elob/Cox7b/Rbbp7","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Gmnn/Psmd6/Psmb5/Psme2","Atf3/Cebpb/Jun/Ppp1r15a/Tcim/Klf2/Dnajb1/Nedd4/Cited2/Egr1","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Ccnd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Pttg1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Calm3/Psma1/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Ap2s1/Psma1/Psmb6/Pfn1/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Pttg1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Hnrnpd/Psma1/Anp32a/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Calm3/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Gmnn/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Elob","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Lmna/Psmb2/Cenpa/Dynll1/Ran/Psma1/Spc24/Ccnb2/Pttg1/Psmb6/Dynll2/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c/Banf1","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Mcm6/Psmb7/Psmd14/Psmc3/Cks1b/Psmb2/Psma1/Ccnd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Gmnn/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Vcp/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Vcp/Prdx1/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Ccnd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Cks1b/Psmb2/Psma1/Ccnd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Calm3/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Jun/Psmb2/Lat2/Calm3/Psma1/Ube2n/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Calm3/Psma1/Psmb6/Psmb3/Psmd12/Nfkbia/Psma3/Calm1/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Hnrnpd/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Psma1/Ccnb2/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Tuba1b/Tuba1c","Ndufs2/Ndufc1/Ndufb6/Cox6a1/Ndufa4/Ndufa5/Cycs/Ndufc2/Uqcrc2/Ndufb7/Cox5a/Uqcrc1/Uqcr10/Cyc1/Ndufv2/Ndufb8/Cox7b","Psmb7/Psmd14/Psmc3/Cks1b/Psmb2/Psma1/Ccnd1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2","Psmb7/Psmd14/Psmc3/Psmb2/Srm/Psma1/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Sms","Psmb7/Psmd14/Psmc3/Vcp/Txn1/Prdx1/Psmb2/Cox6a1/Ndufa4/Cycs/Psma1/Gsr/Prdx2/Cox5a/Psmb6/Psmb3/Psmd12/Psma3/Psma2/Psmd6/Psmb5/Psme2/Cox7b","Ndufs2/Atp5g3/Ndufc1/Ndufb6/Atp5k/Cox6a1/Ndufa4/Ndufa5/Cycs/Ndufc2/Uqcrc2/Ndufb7/Cox5a/Uqcrc1/Uqcr10/Atp5g1/Cyc1/Ndufv2/Ndufb8/Cox7b","Ndufs2/Fh1/Atp5g3/Ndufc1/Ndufb6/Sdhb/Atp5k/Cox6a1/Mdh2/Ndufa4/Ndufa5/Cycs/Suclg1/Ldha/Ndufc2/Uqcrc2/Ndufb7/Cox5a/Uqcrc1/Uqcr10/Atp5g1/Cyc1/Ndufv2/Ndufb8/Cox7b","Dtymk/Uck2/Ndufs2/Ola1/Atp5g3/Dut/Ndufc1/Ndufb6/Vcp/Ak2/Sdhb/Eno1/Atp5k/Ran/Ndufa5/Tpi1/Gapdh/Ndufc2/Dctpp1/Ndufb7/Impdh2/Eno3/Nme1/Atp5g1/Galk1/Ier3/Ndufv2/Pgam1/Ndufb8/Pgk1","Rps27/Rps20/Gm11808/Rps9/Rps19/Uba52/Rps29/Rps24/Eif3e/Eif4b/Eif4a2/Rps10/Rps28/Rps18","Ndufs2/Fh1/Ndufc1/Ndufb6/Vcp/Sdhb/Eno1/Atp5k/Cox6a1/Mdh2/Hmgb1/Ndufa4/Ndufa5/Cycs/Suclg1/Chchd4/Tpi1/Gapdh/Ldha/Ndufc2/Uqcrc2/Ndufb7/Acat1/Cox5a/Uqcrc1/Cisd1/Uqcr10/Mdh1/Eno3/Galk1/Prelid1/Macroh2a1/Dnajc15/Oxct1/Cyc1/Uqcc2/Ier3/Ndufv2/Ppp1ca/Pgam1/Ndufb8/Pgk1","Rpl37a/Gm2000/Rpl12/Rps27/Rps20/Gm11808/Rpl22/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Rps29/Rps24/Gnl3/Rpl37/Rpl30/Rpl3/Rps10/Rps28/Rps18/Rpl39/Rpl36a","Rpl37a/Gm2000/Rpl12/Rps27/Rps20/Gm11808/Rpl22/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Eef2/Rps29/Rps24/Rpl37/Rpl30/Eif3e/Rpl3/Eif4b/Eif4a2/Rps10/Rps28/Rps18/Rpl39/Rpl36a","Rpl7/Rpl37a/Rpl12/Rps3a1/Rps27/Rps20/Rpl22/Rpl9/Rps9/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Rps7/Rps29/Rps24/Rpl37/Rpl30/Rpl8/Rpl3/Rps10/Rps28/Rps18/Rps14/Rpl39/Rps4x/Rpl36a","Rpl37a/Gm2000/Rpl12/Rps27/Rps20/Gm11808/Rpl22/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Rps29/Rps24/Rpl37/Rpl30/Rpl3/Rps10/Rps28/Rps18/Rpl39/Rpl36a","Rps3a1/Rps27/Rps20/Rps9/Rps19/Rps15a/Uba52/Rps7/Rps29/Rps24/Pabpc1/Eif3e/Eif3h/Eif4b/Eif4a2/Rps10/Rps28/Rps18/Rps14/Rps4x","Rpl37a/Rpl12/Rps20/Rpl22/Rpl21/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Eef2/Rps29/Rps24/Rpl30/Rpl3/Eif4b/Eif4a2/Rps10/Rpl10a/Rps28/Rps18/Rpl36/Rpl39/Rpl36a","Rpl37a/Gm2000/Rpl12/Rps27/Rps20/Gm11808/Rpl22/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Rps29/Rps24/Rpl37/Rpl30/Rpl3/Rps10/Rps28/Rps18/Rpl39/Rpl36a","Rpl37a/Gm2000/Rpl12/Rps27/Rps20/Gm11808/Rpl22/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Rps29/Rps24/Rpl37/Rpl30/Rpl3/Rps10/Rps28/Rps18/Rpl39/Rpl36a","Rpl37a/Gm2000/Rpl12/Rps27/Rps20/Gm11808/Rpl22/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Rps29/Rps24/Rpl37/Rpl30/Eif3e/Rpl3/Rps10/Rps28/Rps18/Rpl39/Rpl36a","Rpl37a/Gm2000/Rpl12/Rps27/Rps20/Gm11808/Rpl22/Rps9/Rps19/Uba52/Rpl18a/Rpl29/Rps29/Rps24/Rpl37/Rpl30/Eif3e/Rpl3/Eif4b/Eif4a2/Rps10/Rps28/Rps18/Rpl39/Rpl36a","Rpl7/Eef1b2/Rpl37a/Rpl12/Rps3a1/Rps27/Rps20/Rpl22/Rpl9/Rps9/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Eef2/Rps7/Rps29/Rps24/Rpl37/Rpl30/Pabpc1/Eif3e/Eif3h/Rpl8/Rpl3/Eif4b/Eif4a2/Rps10/Rps28/Rps18/Rps14/Rpl39/Rps4x/Rpl36a","Rpl7/Rpl37a/Rpl12/Rps3a1/Rps27/Rps20/Rpl22/Rpl9/Rps9/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Rps7/Rps29/Rps24/Rpl37/Rpl30/Pabpc1/Rpl8/Rpl3/Rps10/Rps28/Rps18/Rps14/Rpl39/Rps4x/Rpl36a","Rpl7/Rpl37a/Rpl12/Rps3a1/Rps27/Rps20/Rpl22/Rpl9/Rps9/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Rps7/Rps29/Rps24/Rpl37/Rpl30/Rpl8/Rpl3/Rps10/Rps28/Rps18/Rps14/Rpl39/Rps4x/Rpl36a","Rpl7/Rpl37a/Rpl12/Rps3a1/Rps27/Rps20/Rpl22/Rpl9/Rps9/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Rps7/Rps29/Rps24/Rpl37/Rpl30/Pabpc1/Rpl8/Rpl3/Rps10/Rps28/Rps18/Rps14/Rpl39/Rps4x/Rpl36a","Rpl7/Rpl37a/Rpl12/Rps3a1/Rps27/Rps20/Rpl22/Rpl9/Rps9/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Rps7/Rps29/Rps24/Rpl37/Rpl30/Eif3e/Eif3h/Rpl8/Rpl3/Rps10/Rps28/Rps18/Rps14/Rpl39/Rps4x/Rpl36a","Rpl7/Rpl37a/Rpl12/Rps3a1/Rps20/Rpl22/Rpl9/Rpl21/Rps9/Rps5/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Eef2/Rps7/Rps29/Rps24/Rpl30/Pabpc1/Rpl8/Rpl3/Eif4b/Eif4a2/Rps10/Rpl10a/Rps28/Rps18/Rps14/Rpl17/Rpl39/Rps4x/Rpl36a","Rpl7/Rpl37a/Rpl12/Rps3a1/Rps27/Rps20/Rpl22/Rpl9/Rps9/Rps19/Rps15a/Uba52/Rpl18a/Rpl4/Rps7/Rps29/Rps24/Rpl37/Rpl30/Pabpc1/Eif3e/Eif3h/Rpl8/Rpl3/Eif4b/Eif4a2/Rps10/Rps28/Rps18/Rps14/Rpl39/Rps4x/Rpl36a"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>cluster<\/th>\n      <th>ID<\/th>\n      <th>Description<\/th>\n      <th>GeneRatio<\/th>\n      <th>BgRatio<\/th>\n      <th>pvalue<\/th>\n      <th>p.adjust<\/th>\n      <th>Count<\/th>\n      <th>setSize<\/th>\n      <th>GeneProp<\/th>\n      <th>geneID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":9,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[5,6,7,8,9]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render"],"jsHooks":[]}</script>
</div>
<div id="number-of-gene-sets-1" class="section level4" number="6.2.2">
<h4><span class="header-section-number">6.2.2</span> Number of gene
sets</h4>
<pre class="r"><code>gse_df %&gt;% 
  filter(p.adjust &lt; p_thrs) %&gt;% 
  group_by(group) %&gt;% 
  summarise(n = n()) %&gt;% 
  ggplot(aes(x=group, y=n)) +
  geom_bar(stat=&quot;identity&quot;, color = &#39;black&#39;)+
  geom_text(aes(label=n), vjust=1.4, color=&quot;white&quot;, size=geom_text_size)+
  labs(
    x = &#39;Clusters&#39;,
    y = &#39;Number of gene sets&#39;,
    title = &#39;Number of significant gene sets by cluster&#39;,
    caption = paste(&quot;Adjusted P value &lt;&quot;, p_thrs)
    
  )</code></pre>
<p><img src="figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/selected-cluster-marker-gse-enricher-c2reactome_gobp-ngset-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-selected-cluster-marker-gse-enricher-c2reactome_gobp-ngset-1">
Past versions of
selected-cluster-marker-gse-enricher-c2reactome_gobp-ngset-1.png
</button>
</p>
<div
id="fig-selected-cluster-marker-gse-enricher-c2reactome_gobp-ngset-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/39b365c03bef848fb12a4a5aee47a955aecb169a/docs/figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/selected-cluster-marker-gse-enricher-c2reactome_gobp-ngset-1.png" target="_blank">39b365c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gene-set-similarity-matrix-1" class="section level4"
number="6.2.3">
<h4><span class="header-section-number">6.2.3</span> Gene-set similarity
matrix</h4>
<p>Heatmap showing the pair-wise similarity matrix of top 20 gene sets
with an adjusted P value &lt; 10^{-4} in each gene group. Heatmap colors
represent the Jaccard similarity coefficient.</p>
<pre class="r"><code>col_fun &lt;- colorRamp2(seq(0, 1, length.out = 4), brewer.pal(4, &quot;GnBu&quot;))
n_split &lt;- 7

use_mat &lt;- jc_matrix
colnames(use_mat) &lt;- colnames(use_mat) %&gt;% clean_msigdb_names %&gt;% tolower() %&gt;% str_to_title %&gt;% ep_str_wrap(70)
rownames(use_mat) &lt;- rownames(use_mat) %&gt;% clean_msigdb_names %&gt;% tolower() %&gt;% str_to_title %&gt;% ep_str_wrap(70)

ha_top &lt;- HeatmapAnnotation(
  foo = anno_block(
    # labels = c(&quot;Translation&quot;, &quot;Cell division&quot;),
    labels = 1:n_split,
    labels_gp = gpar(col = &quot;black&quot;, fontsize = 8),
    gp = gpar(lwd = 0, lty = 0))
  )

ht &lt;- Heatmap(
  use_mat,
  name = &#39;Jaccard index&#39;,
  column_split = n_split,
  row_split = n_split,
  column_title = NULL,
  row_title = NULL,
  col = col_fun,
  show_column_dend = FALSE,
  show_column_names = FALSE,
  border = TRUE,
  top_annotation = ha_top,
  heatmap_legend_param = list(title_gp = gpar(fontsize = 8),labels_gp = gpar(fontsize = 8)),
  width = unit(7, &quot;cm&quot;))

ht = draw(ht)</code></pre>
<p><img src="figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/selected-cluster-marker-gse-enricher-c2reactome_gobp-jcmatrix-1.png" width="4320" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-selected-cluster-marker-gse-enricher-c2reactome_gobp-jcmatrix-1">
Past versions of
selected-cluster-marker-gse-enricher-c2reactome_gobp-jcmatrix-1.png
</button>
</p>
<div
id="fig-selected-cluster-marker-gse-enricher-c2reactome_gobp-jcmatrix-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/39b365c03bef848fb12a4a5aee47a955aecb169a/docs/figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/selected-cluster-marker-gse-enricher-c2reactome_gobp-jcmatrix-1.png" target="_blank">39b365c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n&quot;)</code></pre>
<pre class="r"><code># Extract term order according to hclust from similarity heatmap
row_order &lt;- row_order(ht) %&gt;% set_names(1:n_split)
term_similarity_clusters &lt;- foreach(i=names(row_order), .combine = rbind) %do% {
  data.frame(
    jc_sim_cl = i, 
    row_order = row_order[[i]], 
    Description =  rownames(jc_matrix)[row_order[[i]]]
  )
}

# Add similarity order to data for dot plot
res_top %&lt;&gt;% left_join(term_similarity_clusters)</code></pre>
</div>
<div id="dotplots-1" class="section level4" number="6.2.4">
<h4><span class="header-section-number">6.2.4</span> Dotplots</h4>
<p>The plots below show the top 20 gene sets with an adjusted P value
&lt; 10^{-4} in each gene group</p>
<pre class="r"><code>res_top %&gt;%
  left_join(selected_clusters) %&gt;%
   # Combine similarity clusters
  mutate(
    jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;3&#39;), &#39;Translation&#39;, jc_sim_cl),
    jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;4&#39;), &#39;Energy\ngeneration&#39;, jc_sim_cl),
    jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;6&#39;), &#39;Cell\nStress&#39;, jc_sim_cl),
    jc_sim_cl = factor(
      jc_sim_cl,
      levels = c(&#39;Translation&#39;, &#39;Energy\ngeneration&#39;, &#39;Cell\nStress&#39;, &#39;1&#39;, &#39;2&#39;, &#39;5&#39;, &#39;7&#39;)
      ),
    p.adjust = ifelse(p.adjust &gt; 0.05, 1, p.adjust),
    p.adjust = -log10( p.adjust),
    p.adjust = ifelse(p.adjust &gt;= 10, 10, p.adjust),
    site = factor(site, levels = c(&#39;PT&#39;, &#39;Lung&#39;, &#39;Liver&#39;)),
    group = factor(group, levels = selected_clusters$group)
  ) %&gt;% 
  ggplot(aes(group, Description, color = GeneProp, size = p.adjust)) +
  geom_point() +
  scale_color_distiller(palette = &quot;Purples&quot;, na.value = &quot;grey70&quot;, direction = 1) +
  scale_y_discrete(labels = default_labeller(70)) +
  scale_x_discrete(position = &quot;top&quot;) +
  scale_size(range = c(1, 5)) +
  labs(y = &#39;&#39;, x = &#39;&#39;) +
  facet_grid(cols = vars(site), rows = vars(jc_sim_cl), scale = &#39;free&#39;, space = &#39;free&#39;) +
  # facet_grid(rows = vars(jc_sim_cl), scale = &#39;free&#39;, space = &#39;free&#39;) +
  theme(
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, size = 1),
    # axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    axis.line.x = element_blank(),
    strip.background = element_blank()#,
    # strip.text.y = element_blank()
    )</code></pre>
<p><img src="figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/selected-cluster-marker-gse-enricher-c2reactome_gobp-dotplot-1.png" width="3600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-selected-cluster-marker-gse-enricher-c2reactome_gobp-dotplot-1">
Past versions of
selected-cluster-marker-gse-enricher-c2reactome_gobp-dotplot-1.png
</button>
</p>
<div
id="fig-selected-cluster-marker-gse-enricher-c2reactome_gobp-dotplot-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/39b365c03bef848fb12a4a5aee47a955aecb169a/docs/figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/selected-cluster-marker-gse-enricher-c2reactome_gobp-dotplot-1.png" target="_blank">39b365c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="site-markers-enrichment" class="section level2" number="7">
<h2><span class="header-section-number">7</span> Site markers
enrichment</h2>
<div id="ora---c2.cp.reactome-and-gobp-1" class="section level3"
number="7.1">
<h3><span class="header-section-number">7.1</span> ORA - C2.CP.REACTOME
and <a href="GO:BP" class="uri">GO:BP</a></h3>
<p>Over-enrichment analysis using genes with a mean AUC ≥ 0.25 in the
score markers results.</p>
<pre class="r"><code>gse_list &lt;- gse_site_auc
test_type &lt;- &#39;enricher&#39;
gset_collection &lt;- &#39;msigdb.m2.cp.reactome&#39;
direction &lt;- &#39;abs&#39;
p_thrs &lt;- 0.01
topn &lt;- 20

# Load all the results
gse_df &lt;- foreach(i = names(gse_list), .combine = rbind) %do% {
  i_1 &lt;- gse_list[[i]][[test_type]][[direction]][[gset_collection]]@result %&gt;% 
    mutate(group = i)
  i_2 &lt;- gse_list[[i]][[&#39;enrichGO&#39;]][[direction]][[&#39;BP&#39;]]@result %&gt;% 
    filter(simplify) %&gt;% 
    dplyr::select(-simplify) %&gt;% 
    mutate(group = i)
  rbind(i_1, i_2)
} %&gt;% 
  mutate(
    group = factor(group, c(&#39;primary_tumor&#39;, &#39;lung&#39;, &#39;liver&#39;)),
    Description = clean_msigdb_names(Description) %&gt;% tolower() %&gt;% str_to_title,
    Description = ifelse(grepl(&quot;^GO:&quot;, ID), paste0(&quot;GO:&quot;, Description), Description)
  ) 

# Select topn by group in selected gset
top_by_group &lt;- foreach(i = names(gse_list), .combine = c) %do% {
  gse_df %&gt;% 
    filter(group == i) %&gt;% 
    filter(p.adjust &lt; p_thrs) %&gt;% 
    arrange(p.adjust) %&gt;% pull(Description) %&gt;%
    head(topn)
}

# Data frame to plot
res_top &lt;- gse_df %&gt;% 
  filter(Description %in% top_by_group) %&gt;% 
  mutate(log.p.adjust = -log10(p.adjust))

# Rows and column orders by hclust
use_mat &lt;- res_top %&gt;% 
  dplyr::select(Description, log.p.adjust, group) %&gt;% 
  pivot_wider(names_from = group, values_from = log.p.adjust) %&gt;% 
  column_to_rownames(&#39;Description&#39;)
use_mat[is.na(use_mat)] &lt;- 0

row_hclust &lt;- hclust(dist(use_mat))
row_hclust_labels &lt;- row_hclust$labels[row_hclust$order]

col_hclust &lt;- hclust(dist(t(use_mat)))
col_hclust_labels &lt;- col_hclust$labels[col_hclust$order]

use_mat &lt;- use_mat[row_hclust_labels,col_hclust_labels]

# GSET clusters by kmeans using silhouette
use_kmax &lt;- ifelse(nrow(use_mat) &lt;= 20, nrow(use_mat)-2, 20)
nb_clust &lt;- fviz_nbclust(use_mat, kmeans, method = &quot;silhouette&quot;, k.max = use_kmax)
silhouette_best_k &lt;- nb_clust$data %&gt;% slice_max(y) %&gt;% pull(clusters) %&gt;% as.character %&gt;% as.numeric
# silhouette_best_k &lt;- 5
kmeans_mat &lt;- kmeans(use_mat, silhouette_best_k, nstart = 25)
kmeans_mat_df &lt;- kmeans_mat$cluster %&gt;%
  data.frame %&gt;%
  set_names(&#39;kcluster&#39;) %&gt;%
  rownames_to_column(&#39;Description&#39;) %&gt;%
  mutate(kcluster = as.character(kcluster))


# Order factors by hclust
res_top %&lt;&gt;% 
  left_join(kmeans_mat_df) %&gt;% 
  mutate(
    Description = factor(Description, levels = row_hclust_labels),
    group = fct_relevel(group, col_hclust_labels)
  )

# Calculate jaccard similarity matrix
gset_collection &lt;- foreach(i = levels(res_top$Description)) %do% {
  res_top %&gt;% filter(Description == i) %&gt;% pull(geneID) %&gt;% 
    paste(collapse = &#39;/&#39;) %&gt;% 
    strsplit(&#39;/&#39;) %&gt;% unlist %&gt;% unique
}
names(gset_collection) &lt;- levels(res_top$Description)

use_terms &lt;- res_top$Description %&gt;% unique
jc_matrix &lt;- pairwise_terms_jc(gset_collection[use_terms])</code></pre>
<div id="results-table-2" class="section level4" number="7.1.1">
<h4><span class="header-section-number">7.1.1</span> Results table</h4>
<pre class="r"><code>gse_df %&gt;% 
  filter(p.adjust &lt; p_thrs) %&gt;% 
  arrange(desc(p.adjust)) %&gt;% 
  dplyr::select(group, ID:p.adjust, Count:GeneProp, geneID)  %&gt;%
  dplyr::rename(site = group) %&gt;% 
  datatable(., 
            rownames = FALSE, 
            filter = &#39;top&#39;, 
            caption = &#39;Results from over enrichment analysis&#39;,
            extensions = &#39;Buttons&#39;, 
            options = list(
              dom = &#39;Blfrtip&#39;,
              buttons = c(&#39;csv&#39;, &#39;excel&#39;)
            ))  %&gt;% 
  formatRound(c(&#39;GeneProp&#39;), digits = 2) %&gt;%
  formatSignif(c(&#39;pvalue&#39;, &#39;p.adjust&#39;), digits = 3)</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-3d8dd78c0ab17af95663" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-3d8dd78c0ab17af95663">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;primary_tumor&quot;,&quot;lung&quot;,&quot;liver&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"0.000568702310133\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"0.009992912020897\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"3\" data-max=\"34\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"14\" data-max=\"492\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.030487804878048\" data-max=\"0.257142857142858\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Results from over enrichment analysis<\/caption>","data":[["primary_tumor","lung","primary_tumor","liver","primary_tumor","primary_tumor","primary_tumor","lung","lung","lung","primary_tumor","primary_tumor","primary_tumor","lung","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","lung","primary_tumor","primary_tumor","liver","lung","lung","liver","primary_tumor","primary_tumor","primary_tumor","lung","liver","liver","liver","primary_tumor","liver","primary_tumor","primary_tumor","liver","lung","liver","liver","liver","liver","liver","liver","primary_tumor","primary_tumor","liver","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","primary_tumor","liver"],["REACTOME_RESPONSE_TO_ELEVATED_PLATELET_CYTOSOLIC_CA2","REACTOME_RHO_GTPASES_ACTIVATE_IQGAPS","GO:0055076","GO:0050863","REACTOME_INACTIVATION_OF_CSF3_G_CSF_SIGNALING","REACTOME_FC_EPSILON_RECEPTOR_FCERI_SIGNALING","REACTOME_TOLL_LIKE_RECEPTOR_4_TLR4_CASCADE","REACTOME_APOPTOTIC_EXECUTION_PHASE","REACTOME_DEGRADATION_OF_THE_EXTRACELLULAR_MATRIX","REACTOME_TP53_REGULATES_METABOLIC_GENES","REACTOME_P75NTR_SIGNALS_VIA_NF_KB","GO:0048145","GO:0006090","REACTOME_PROGRAMMED_CELL_DEATH","GO:0032328","REACTOME_OXIDATIVE_STRESS_INDUCED_SENESCENCE","REACTOME_TAK1_DEPENDENT_IKK_AND_NF_KAPPA_B_ACTIVATION","REACTOME_NF_KB_IS_ACTIVATED_AND_SIGNALS_SURVIVAL","REACTOME_MYD88_INDEPENDENT_TLR4_CASCADE","REACTOME_DDX58_IFIH1_MEDIATED_INDUCTION_OF_INTERFERON_ALPHA_BETA","REACTOME_DETOXIFICATION_OF_REACTIVE_OXYGEN_SPECIES","REACTOME_TOLL_LIKE_RECEPTOR_9_TLR9_CASCADE","GO:0071559","GO:0048144","REACTOME_APOPTOSIS","GO:0051014","GO:0032370","REACTOME_REGULATION_OF_INSULIN_LIKE_GROWTH_FACTOR_IGF_TRANSPORT_AND_UPTAKE_BY_INSULIN_LIKE_GROWTH_FACTOR_BINDING_PROTEINS_IGFBPS","REACTOME_MYD88_MAL_TIRAP_CASCADE_INITIATED_ON_PLASMA_MEMBRANE","GO:0034976","GO:0009611","GO:0051098","GO:0008637","REACTOME_CITRIC_ACID_CYCLE_TCA_CYCLE","GO:1901216","REACTOME_METABOLISM_OF_CARBOHYDRATES","GO:0006110","GO:0035914","REACTOME_RESPIRATORY_ELECTRON_TRANSPORT","GO:0052547","REACTOME_RESPIRATORY_ELECTRON_TRANSPORT_ATP_SYNTHESIS_BY_CHEMIOSMOTIC_COUPLING_AND_HEAT_PRODUCTION_BY_UNCOUPLING_PROTEINS","REACTOME_PYRUVATE_METABOLISM_AND_CITRIC_ACID_TCA_CYCLE","REACTOME_GLYCOLYSIS","REACTOME_GLUCOSE_METABOLISM","GO:0010035","REACTOME_GLUCONEOGENESIS","REACTOME_ACTIVATION_OF_THE_MRNA_UPON_BINDING_OF_THE_CAP_BINDING_COMPLEX_AND_EIFS_AND_SUBSEQUENT_BINDING_TO_43S","REACTOME_TRANSLATION","REACTOME_THE_CITRIC_ACID_TCA_CYCLE_AND_RESPIRATORY_ELECTRON_TRANSPORT","GO:0002181","REACTOME_MAJOR_PATHWAY_OF_RRNA_PROCESSING_IN_THE_NUCLEOLUS_AND_CYTOSOL","REACTOME_NONSENSE_MEDIATED_DECAY_NMD","REACTOME_FORMATION_OF_A_POOL_OF_FREE_40S_SUBUNITS","REACTOME_EUKARYOTIC_TRANSLATION_INITIATION","REACTOME_SRP_DEPENDENT_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE","REACTOME_NONSENSE_MEDIATED_DECAY_NMD_INDEPENDENT_OF_THE_EXON_JUNCTION_COMPLEX_EJC","GO:0006091"],["Response To Elevated Platelet Cytosolic Ca2","Rho Gtpases Activate Iqgaps","GO:Transition Metal Ion Homeostasis","GO:Regulation Of T Cell Activation","Inactivation Of Csf3 G Csf Signaling","Fc Epsilon Receptor Fceri Signaling","Toll Like Receptor 4 Tlr4 Cascade","Apoptotic Execution Phase","Degradation Of The Extracellular Matrix","Tp53 Regulates Metabolic Genes","P75ntr Signals Via Nf Kb","GO:Regulation Of Fibroblast Proliferation","GO:Pyruvate Metabolic Process","Programmed Cell Death","GO:Alanine Transport","Oxidative Stress Induced Senescence","Tak1 Dependent Ikk And Nf Kappa B Activation","Nf Kb Is Activated And Signals Survival","Myd88 Independent Tlr4 Cascade","Ddx58 Ifih1 Mediated Induction Of Interferon Alpha Beta","Detoxification Of Reactive Oxygen Species","Toll Like Receptor 9 Tlr9 Cascade","GO:Response To Transforming Growth Factor Beta","GO:Fibroblast Proliferation","Apoptosis","GO:Actin Filament Severing","GO:Positive Regulation Of Lipid Transport","Regulation Of Insulin Like Growth Factor Igf Transport And Uptake By Insulin Like Growth Factor Binding Proteins Igfbps","Myd88 Mal Tirap Cascade Initiated On Plasma Membrane","GO:Response To Endoplasmic Reticulum Stress","GO:Response To Wounding","GO:Regulation Of Binding","GO:Apoptotic Mitochondrial Changes","Citric Acid Cycle Tca Cycle","GO:Positive Regulation Of Neuron Death","Metabolism Of Carbohydrates","GO:Regulation Of Glycolytic Process","GO:Skeletal Muscle Cell Differentiation","Respiratory Electron Transport","GO:Regulation Of Peptidase Activity","Respiratory Electron Transport Atp Synthesis By Chemiosmotic Coupling And Heat Production By Uncoupling Proteins","Pyruvate Metabolism And Citric Acid Tca Cycle","Glycolysis","Glucose Metabolism","GO:Response To Inorganic Substance","Gluconeogenesis","Activation Of The Mrna Upon Binding Of The Cap Binding Complex And Eifs And Subsequent Binding To 43s","Translation","The Citric Acid Tca Cycle And Respiratory Electron Transport","GO:Cytoplasmic Translation","Major Pathway Of Rrna Processing In The Nucleolus And Cytosol","Nonsense Mediated Decay Nmd","Formation Of A Pool Of Free 40s Subunits","Eukaryotic Translation Initiation","Srp Dependent Cotranslational Protein Targeting To Membrane","Nonsense Mediated Decay Nmd Independent Of The Exon Junction Complex Ejc","GO:Generation Of Precursor Metabolites And Energy"],["6/71","4/89","6/119","12/182","3/71","6/71","6/71","5/89","7/89","6/89","3/71","6/119","6/119","8/89","3/119","4/71","4/71","3/71","6/71","4/71","5/89","6/71","9/119","8/182","8/89","4/128","7/182","7/71","6/71","10/119","15/128","14/182","8/182","5/133","8/119","15/133","6/119","7/119","9/133","16/128","11/133","8/133","9/133","10/133","21/182","9/133","9/71","16/71","19/133","15/119","16/71","15/71","15/71","16/71","15/71","15/71","34/182"],["128/8764","29/8764","134/18646","362/18646","19/8764","121/8764","118/8764","49/8764","108/8764","72/8764","17/8764","121/18646","120/18646","132/8764","14/18646","38/8764","38/8764","14/8764","100/8764","34/8764","34/8764","94/8764","238/18646","142/18646","103/8764","17/18646","100/18646","121/8764","84/8764","257/18646","492/18646","401/18646","119/18646","22/8764","133/18646","264/8764","49/18646","82/18646","87/8764","441/18646","108/8764","48/8764","61/8764","81/8764","489/18646","35/8764","58/8764","222/8764","156/8764","144/18646","172/8764","108/8764","95/8764","113/8764","87/8764","88/8764","444/18646"],[0.0005687023101323379,0.000194397796161974,0.0002198615875303447,0.000237205680361289,0.0004498779203389228,0.0004213058255220607,0.0003681598492164354,0.0001293450707615478,0.0001066463679293955,8.447215426436845e-05,0.000319395259743315,0.0001258976763700817,0.0001202748384615098,5.432229274290985e-05,8.764226560688356e-05,0.0002371208328656688,0.0002371208328656688,0.0001739801830399455,0.0001493193446682555,0.0001526594177984161,2.127525246075482e-05,0.0001059558246356018,2.211411546616285e-05,7.789515026294473e-05,8.909103454585423e-06,4.705191319293114e-06,5.606730696372273e-05,5.142458033238809e-05,5.636949643498053e-05,5.922074895750751e-06,1.452834693901451e-06,4.111342478168994e-05,2.202494171825701e-05,1.597562584078881e-05,2.189087231344481e-06,1.031154955750573e-05,6.652375416862727e-07,9.239835217902989e-07,6.138190584702452e-06,6.041594266542165e-08,6.408030242724955e-07,5.158791626765434e-07,2.894663809593835e-07,3.472642016902322e-07,1.394173353414895e-08,1.64448969022855e-09,6.92381311749383e-10,1.561672801683248e-11,1.610606893960884e-12,2.058902043966965e-14,3.014303744344743e-13,4.373041300606049e-15,5.963903925359462e-16,3.486611560165106e-16,1.492888585348199e-16,1.788782377139276e-16,6.459742337636622e-21],[0.009992912020896796,0.009303323102037329,0.00889502397960539,0.008567983490640295,0.008300247630253124,0.00818220261145476,0.007547276908936926,0.007221766450853086,0.007145306651269501,0.007074542919640858,0.006932755932075485,0.006252374569059672,0.006231740067786976,0.0060659893562916,0.005735955646429458,0.005468599207964486,0.005468599207964486,0.004585620538695706,0.004333178859047349,0.004333178859047349,0.003563604787176432,0.003554336299139733,0.003467118587591833,0.002993970006260363,0.002984549657286117,0.002956036446345899,0.00221170771417422,0.002080034418450782,0.002080034418450782,0.001841025033216515,0.001825486792887174,0.001665649290479817,0.0009710408128137427,0.0008325076132588837,0.000799176616712785,0.0006045145928087733,0.0004595894037384947,0.0004595894037384947,0.0004112587691750643,0.0001518252639182046,5.008943639730007e-05,4.838946545905977e-05,4.071672764817972e-05,4.071672764817972e-05,8.195552379485989e-07,3.85632832358595e-07,3.193608800444029e-08,8.232246626015981e-10,7.553746332676544e-10,5.120489383345842e-11,1.853796802772017e-11,3.227304479847264e-13,5.501701371144103e-14,4.28853221900308e-14,3.300303485821965e-14,3.300303485821965e-14,1.936630752823459e-17],[6,4,6,12,3,6,6,5,7,6,3,6,6,8,3,4,4,3,6,4,5,6,9,8,8,4,7,7,6,10,15,14,8,5,8,15,6,7,9,16,11,8,9,10,21,9,9,16,19,15,16,15,15,16,15,15,34],[128,29,134,362,19,121,118,49,108,72,17,121,120,132,14,38,38,14,100,34,34,94,238,142,103,17,100,121,84,257,492,401,119,22,133,264,49,82,87,441,108,48,61,81,489,35,58,222,156,144,172,108,95,113,87,88,444],[0.046875,0.1379310344827586,0.04477611940298507,0.03314917127071823,0.1578947368421053,0.04958677685950413,0.05084745762711865,0.1020408163265306,0.06481481481481481,0.08333333333333333,0.1764705882352941,0.04958677685950413,0.05,0.06060606060606061,0.2142857142857143,0.1052631578947368,0.1052631578947368,0.2142857142857143,0.06,0.1176470588235294,0.1470588235294118,0.06382978723404255,0.03781512605042017,0.05633802816901409,0.07766990291262135,0.2352941176470588,0.07000000000000001,0.05785123966942149,0.07142857142857142,0.03891050583657588,0.03048780487804878,0.03491271820448878,0.06722689075630252,0.2272727272727273,0.06015037593984962,0.05681818181818182,0.1224489795918367,0.08536585365853659,0.103448275862069,0.036281179138322,0.1018518518518518,0.1666666666666667,0.1475409836065574,0.1234567901234568,0.04294478527607362,0.2571428571428571,0.1551724137931035,0.07207207207207207,0.1217948717948718,0.1041666666666667,0.09302325581395349,0.1388888888888889,0.1578947368421053,0.1415929203539823,0.1724137931034483,0.1704545454545454,0.07657657657657657],["Ecm1/Serpine1/Aplp2/Clu/App/Tmsb4x","Tubb4b/Calm1/Tuba1a/Tuba1c","Lcn2/B2m/Cp/Aplp2/Myc/App","Sdc4/Hspb1/Cav1/Lgals7/Casp3/Prdx2/Ccl2/Prelid1/Lgals3/Lgals1/Hes1/Anxa1","Ubc/Uba52/Socs3","Jun/Ubc/Uba52/Nfkbia/Fos/Psmb8","Jun/Ubc/Uba52/Nfkbia/Fos/App","Gsn/Lmna/Hmgb2/Dsp/Plec","Capn2/Cd44/Spp1/Ctsd/Col18a1/Cast/Ctsb","Txn1/Prdx1/Ywhah/Cycs/Gsr/Prdx5","Ubc/Uba52/Nfkbia","Jun/Serpine1/Nupr1/Ifi30/Myc/Cdkn1a","Nupr1/Ddit4/Myc/Zbtb20/App/Ier3","Gsn/Lmna/Ywhah/Cycs/Bax/Hmgb2/Dsp/Plec","Slc6a6/Slc38a2/Slc3a2","Jun/Ubc/Uba52/Fos","Ubc/Uba52/Nfkbia/App","Ubc/Uba52/Nfkbia","Jun/Ubc/Uba52/Nfkbia/Fos/App","Ubc/Uba52/Nfkbia/App","Txn1/Prdx1/Cycs/Gsr/Prdx5","Jun/Ubc/Uba52/Nfkbia/Fos/App","Hspa5/Jun/Serpine1/Cited2/Fos/Fermt2/Scx/Itgb5/Zfp36l2","Cks1b/Jun/Cav1/Ifi30/Anxa2/Mif/Cdk4/Cdkn1a","Gsn/Lmna/Ywhah/Cycs/Bax/Hmgb2/Dsp/Plec","Gsn/Dstn/Capg/Myh9","Dbi/Spp1/Cav1/Anxa2/Mif/Nfkbia/Prelid1","Cst3/Cp/Ccn1/Mxra8/Igfbp7/Aplp2/App","Jun/Ubc/Uba52/Nfkbia/Fos/App","Atf3/Hspa5/Jun/Ppp1r15a/Nupr1/Herpud1/Marcks/Clu/Atf4/App","Cd44/S100a10/Bax/Plet1/Map2k2/Ccl2/Grn/Dsp/Anxa8/Plec/Myh9/Tnfrsf12a/Cdkn1a/Anxa1/Myof","Nucks1/Id1/Slpi/S100a10/Jun/Stmn1/Eno1/Cav1/Hmgb2/Anxa2/Tle5/Calm1/Hes1/Cdkn1a","Jun/Bnip3/Higd1a/Siva1/Prelid1/Bnip3l/Fam162a/Ier3","Sdhc/Fh1/Suclg1/Idh2/Sdhd","Lcn2/Jun/Nupr1/Ddit4/Fos/Clu/Atf4/App","Sdc4/Eno1/Akr1b3/Tpi1/Gapdh/Gpi1/Aldoa/Pkm/Mdh1/Eno3/Galk1/Calm1/Tkt/Pgam1/Pgk1","Nupr1/Ddit4/Myc/Zbtb20/App/Ier3","Btg2/Atf3/Nupr1/Cited2/Fos/Myc/Scx","Ndufs2/Cox6a1/Ndufc2/Uqcrc2/Cox5a/Uqcrfs1/Ndufv3/Ndufv2/Cox7b","Gsn/Cd44/Slpi/Cycs/Bax/Ctsd/Cryab/Perp/Cstb/Grn/Serpinb6a/Cast/Anxa8/Ctsb/Dap/Bin1","Ndufs2/Atp5c1/Atp5g3/Cox6a1/Ndufc2/Uqcrc2/Cox5a/Uqcrfs1/Ndufv3/Ndufv2/Cox7b","Sdhc/Fh1/Suclg1/Ldha/Idh2/Sdhd/Vdac1/Glo1","Eno1/Tpi1/Gapdh/Gpi1/Aldoa/Pkm/Eno3/Pgam1/Pgk1","Eno1/Tpi1/Gapdh/Gpi1/Aldoa/Pkm/Mdh1/Eno3/Pgam1/Pgk1","Anxa5/Jun/Cav1/Akr1b3/Gapdh/Selenow/Fosb/Bnip3/Casp3/Prdx2/Junb/Mt2/Mt1/Cdk4/Rhob/Fos/Calm1/Krt8/Hnrnpa1/Ppp1ca/Anxa1","Eno1/Tpi1/Gapdh/Gpi1/Aldoa/Mdh1/Eno3/Pgam1/Pgk1","Rps27/Rps20/Rps9/Rps19/Uba52/Rps24/Eif4b/Rps28/Rps4x","Rpl37a/Rpl12/Rps27/Rps20/Rpl22/Rps9/Rps19/Uba52/Rps24/Rpl37/Rpl30/Rpl3/Eif4b/Rps28/Rps4x/Rpl36a","Sdhc/Ndufs2/Fh1/Atp5c1/Atp5g3/Cox6a1/Suclg1/Ldha/Idh2/Ndufc2/Uqcrc2/Sdhd/Cox5a/Vdac1/Uqcrfs1/Glo1/Ndufv3/Ndufv2/Cox7b","Rpl37a/Rpl12/Rps20/Rpl22/Rps9/Rps19/Uba52/Rps24/Rpl30/Rpl3/Eif4b/Rpl10a/Rps28/Rps4x/Rpl36a","Rpl37a/Rpl12/Rps27/Rps20/Rpl22/Rps9/Rps19/Uba52/Rps24/Gnl3/Rpl37/Rpl30/Rpl3/Rps28/Rps4x/Rpl36a","Rpl37a/Rpl12/Rps27/Rps20/Rpl22/Rps9/Rps19/Uba52/Rps24/Rpl37/Rpl30/Rpl3/Rps28/Rps4x/Rpl36a","Rpl37a/Rpl12/Rps27/Rps20/Rpl22/Rps9/Rps19/Uba52/Rps24/Rpl37/Rpl30/Rpl3/Rps28/Rps4x/Rpl36a","Rpl37a/Rpl12/Rps27/Rps20/Rpl22/Rps9/Rps19/Uba52/Rps24/Rpl37/Rpl30/Rpl3/Eif4b/Rps28/Rps4x/Rpl36a","Rpl37a/Rpl12/Rps27/Rps20/Rpl22/Rps9/Rps19/Uba52/Rps24/Rpl37/Rpl30/Rpl3/Rps28/Rps4x/Rpl36a","Rpl37a/Rpl12/Rps27/Rps20/Rpl22/Rps9/Rps19/Uba52/Rps24/Rpl37/Rpl30/Rpl3/Rps28/Rps4x/Rpl36a","Sdhc/Ndufs2/Fh1/Atp5c1/Eno1/Cox6a1/Suclg1/Tpi1/Gapdh/Gpi1/Ldha/Idh2/Ndufc2/Cavin3/Uqcrc2/Aldoa/Bnip3/Sdhd/Acat1/Cox5a/Pkm/Cisd1/Mdh1/Eno3/Galk1/Uqcrfs1/Prelid1/Tkt/Ndufv3/Ier3/Ndufv2/Ppp1ca/Pgam1/Pgk1"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>site<\/th>\n      <th>ID<\/th>\n      <th>Description<\/th>\n      <th>GeneRatio<\/th>\n      <th>BgRatio<\/th>\n      <th>pvalue<\/th>\n      <th>p.adjust<\/th>\n      <th>Count<\/th>\n      <th>setSize<\/th>\n      <th>GeneProp<\/th>\n      <th>geneID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":9,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[5,6,7,8,9]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render"],"jsHooks":[]}</script>
</div>
<div id="number-of-gene-sets-2" class="section level4" number="7.1.2">
<h4><span class="header-section-number">7.1.2</span> Number of gene
sets</h4>
<pre class="r"><code>gse_df %&gt;% 
  filter(p.adjust &lt; p_thrs) %&gt;% 
  group_by(group) %&gt;% 
  summarise(n = n()) %&gt;% 
  ggplot(aes(x=group, y=n)) +
  geom_bar(stat=&quot;identity&quot;, color = &#39;black&#39;)+
  geom_text(aes(label=n), vjust=1.4, color=&quot;white&quot;, size=geom_text_size)+
  labs(
    x = &#39;Site&#39;,
    y = &#39;Number of gene sets&#39;,
    title = &#39;Number of significant gene sets by site&#39;,
    caption = paste(&quot;Adjusted P value &lt;&quot;, p_thrs)
    
  )</code></pre>
<p><img src="figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/site-marker-gse-enricher-c2reactome_gobp-ngset-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-site-marker-gse-enricher-c2reactome_gobp-ngset-1">
Past versions of site-marker-gse-enricher-c2reactome_gobp-ngset-1.png
</button>
</p>
<div id="fig-site-marker-gse-enricher-c2reactome_gobp-ngset-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/39b365c03bef848fb12a4a5aee47a955aecb169a/docs/figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/site-marker-gse-enricher-c2reactome_gobp-ngset-1.png" target="_blank">39b365c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gene-set-similarity-matrix-2" class="section level4"
number="7.1.3">
<h4><span class="header-section-number">7.1.3</span> Gene-set similarity
matrix</h4>
<p>Heatmap showing the pair-wise similarity matrix of top 20 gene sets
with an adjusted P value &lt; 0.01 in each gene group. Heatmap colors
represent the Jaccard similarity coefficient.</p>
<pre class="r"><code>col_fun &lt;- colorRamp2(seq(0, 1, length.out = 4), brewer.pal(4, &quot;GnBu&quot;))
n_split &lt;- 9

use_mat &lt;- jc_matrix
colnames(use_mat) &lt;- colnames(use_mat) %&gt;% clean_msigdb_names %&gt;% tolower() %&gt;% str_to_title %&gt;% ep_str_wrap(70)
rownames(use_mat) &lt;- rownames(use_mat) %&gt;% clean_msigdb_names %&gt;% tolower() %&gt;% str_to_title %&gt;% ep_str_wrap(70)

ha_top &lt;- HeatmapAnnotation(
  foo = anno_block(
    # labels = c(&quot;Translation&quot;, &quot;Cell division&quot;),
    labels = 1:n_split,
    labels_gp = gpar(col = &quot;black&quot;, fontsize = 8),
    gp = gpar(lwd = 0, lty = 0))
  )

ht &lt;- Heatmap(
  use_mat,
  name = &#39;Jaccard index&#39;,
  column_split = n_split,
  row_split = n_split,
  column_title = NULL,
  row_title = NULL,
  col = col_fun,
  show_column_dend = FALSE,
  show_column_names = FALSE,
  border = TRUE,
  top_annotation = ha_top,
  heatmap_legend_param = list(title_gp = gpar(fontsize = 8),labels_gp = gpar(fontsize = 8)),
  width = unit(7, &quot;cm&quot;))

ht = draw(ht)</code></pre>
<p><img src="figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/site-marker-gse-enricher-c2reactome_gobp-jcmatrix-1.png" width="4320" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-site-marker-gse-enricher-c2reactome_gobp-jcmatrix-1">
Past versions of site-marker-gse-enricher-c2reactome_gobp-jcmatrix-1.png
</button>
</p>
<div id="fig-site-marker-gse-enricher-c2reactome_gobp-jcmatrix-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/39b365c03bef848fb12a4a5aee47a955aecb169a/docs/figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/site-marker-gse-enricher-c2reactome_gobp-jcmatrix-1.png" target="_blank">39b365c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n&quot;)</code></pre>
<pre class="r"><code># Extract term order according to hclust from similarity heatmap
row_order &lt;- row_order(ht) %&gt;% set_names(1:n_split)
term_similarity_clusters &lt;- foreach(i=names(row_order), .combine = rbind) %do% {
  data.frame(
    jc_sim_cl = i, 
    row_order = row_order[[i]], 
    Description =  rownames(jc_matrix)[row_order[[i]]]
  )
}

# Add similarity order to data for dot plot
res_top %&lt;&gt;% left_join(term_similarity_clusters)</code></pre>
</div>
<div id="dotplots-2" class="section level4" number="7.1.4">
<h4><span class="header-section-number">7.1.4</span> Dotplots</h4>
<p>The plots below show the top 20 gene sets with an adjusted P value
&lt; 0.01 in each gene group</p>
<pre class="r"><code>res_top %&gt;%
  # Combine similarity clusters
  mutate(
    jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;1&#39;), &#39;Translation&#39;, jc_sim_cl),
    jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;3&#39;,&#39;5&#39;,&#39;6&#39;), &#39;Energy\ngeneration&#39;, jc_sim_cl),
    jc_sim_cl = ifelse(jc_sim_cl %in% c(&#39;4&#39;, &#39;8&#39;), &#39;Cell stress&#39;, jc_sim_cl),
    jc_sim_cl = factor(
      jc_sim_cl,
      levels = c(&#39;Translation&#39;, &#39;Energy\ngeneration&#39;, &#39;Cell stress&#39;, &#39;2&#39;, &#39;7&#39;, &#39;9&#39;)
      ),

    p.adjust = ifelse(p.adjust &gt; 0.05, 1, p.adjust),
    p.adjust = -log10( p.adjust),
    p.adjust = ifelse(p.adjust &gt;= 10, 10, p.adjust),
    group = recode(group, primary_tumor = &quot;PT&quot;, lung = &quot;Lung&quot;, liver = &quot;Liver&quot;),
    group = factor(group, levels = c(&#39;PT&#39;, &#39;Lung&#39;, &#39;Liver&#39;))
  ) %&gt;% 
  ggplot(aes(group, Description, color = GeneProp, size = p.adjust)) +
  geom_point() +
  scale_color_distiller(palette = &quot;Purples&quot;, na.value = &quot;grey70&quot;, direction = 1) +
  scale_y_discrete(labels = default_labeller(70)) +
  scale_x_discrete(position = &quot;top&quot;) +
  scale_size(range = c(1, 5)) +
  labs(y = &#39;&#39;, x = &#39;&#39;) +
  # facet_grid(rows = vars(cluster), scale = &#39;free&#39;, space = &#39;free&#39;) +
  # facet_grid(rows = vars(kcluster), scale = &#39;free&#39;, space = &#39;free&#39;) +
  facet_grid(rows = vars(jc_sim_cl), scale = &#39;free&#39;, space = &#39;free&#39;) +
  theme(
    panel.border = element_rect(color = &quot;black&quot;, fill = NA, size = 1),
    # axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    axis.line.x = element_blank(),
    strip.background = element_blank()#,
    # strip.text.y = element_blank()
    )</code></pre>
<p><img src="figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/site-marker-gse-enricher-c2reactome_gobp-dotplot-1.png" width="3600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-site-marker-gse-enricher-c2reactome_gobp-dotplot-1">
Past versions of site-marker-gse-enricher-c2reactome_gobp-dotplot-1.png
</button>
</p>
<div id="fig-site-marker-gse-enricher-c2reactome_gobp-dotplot-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/39b365c03bef848fb12a4a5aee47a955aecb169a/docs/figure/mvt1-tumor-10x_rnaseq-pll-cancer_cells-main.Rmd/site-marker-gse-enricher-c2reactome_gobp-dotplot-1.png" target="_blank">39b365c</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<p>R version 4.2.2 (2022-10-31) Platform: x86_64-apple-darwin17.0
(64-bit) Running under: macOS Big Sur … 10.16</p>
<p>Matrix products: default BLAS:
/Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK:
/Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</p>
<p>locale: [1]
en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</p>
<p>attached base packages: [1] grid stats4 stats graphics grDevices
utils datasets [8] methods base</p>
<p>other attached packages: [1] factoextra_1.0.7 corrplot_0.92<br />
[3] circlize_0.4.15 GSVA_1.46.0<br />
[5] clusterProfiler_4.6.0 ComplexHeatmap_2.14.0<br />
[7] miloR_1.6.0 edgeR_3.40.2<br />
[9] limma_3.54.1 speckle_0.0.3<br />
[11] scDblFinder_1.13.8 bluster_1.8.0<br />
[13] batchelor_1.14.1 BiocSingular_1.14.0<br />
[15] scran_1.26.2 SeuratObject_4.1.3<br />
[17] Seurat_4.3.0 DropletUtils_1.18.1<br />
[19] scater_1.26.1 scuttle_1.8.4<br />
[21] SingleCellExperiment_1.20.0 SummarizedExperiment_1.28.0 [23]
Biobase_2.58.0 GenomicRanges_1.50.2<br />
[25] GenomeInfoDb_1.34.9 IRanges_2.32.0<br />
[27] S4Vectors_0.36.1 BiocGenerics_0.44.0<br />
[29] MatrixGenerics_1.10.0 matrixStats_1.0.0<br />
[31] magrittr_2.0.3 openxlsx_4.2.5.2<br />
[33] patchwork_1.1.3 pheatmap_1.0.12<br />
[35] ggpubr_0.6.0 RColorBrewer_1.1-3<br />
[37] arsenal_3.6.3 ggbeeswarm_0.7.2<br />
[39] colorblindr_0.1.0 colorspace_2.1-1<br />
[41] cowplot_1.1.1 kableExtra_1.3.4<br />
[43] knitr_1.44 DT_0.30<br />
[45] foreach_1.5.2 showtext_0.9-6<br />
[47] showtextdb_3.0 sysfonts_0.8.8<br />
[49] lubridate_1.9.3 forcats_1.0.0<br />
[51] stringr_1.5.1 dplyr_1.1.4<br />
[53] purrr_1.0.2 readr_2.1.4<br />
[55] tidyr_1.3.1 tibble_3.2.1<br />
[57] ggplot2_3.5.0 tidyverse_2.0.0<br />
[59] workflowr_1.7.1</p>
<p>loaded via a namespace (and not attached): [1] rsvd_1.0.5
ica_1.0-3<br />
[3] svglite_2.1.2 ps_1.7.5<br />
[5] Rsamtools_2.14.0 lmtest_0.9-40<br />
[7] rprojroot_2.0.3 crayon_1.5.2<br />
[9] MASS_7.3-60 rhdf5filters_1.10.0<br />
[11] nlme_3.1-163 backports_1.4.1<br />
[13] GOSemSim_2.24.0 rlang_1.1.4<br />
[15] HDO.db_0.99.1 XVector_0.38.0<br />
[17] ROCR_1.0-11 irlba_2.3.5.1<br />
[19] callr_3.7.3 xgboost_1.7.5.1<br />
[21] BiocParallel_1.32.5 rjson_0.2.21<br />
[23] bit64_4.0.5 glue_1.8.0<br />
[25] sctransform_0.4.1 parallel_4.2.2<br />
[27] processx_3.8.2 vipor_0.4.5<br />
[29] spatstat.sparse_3.0-2 AnnotationDbi_1.60.0<br />
[31] DOSE_3.24.2 spatstat.geom_3.2-7<br />
[33] tidyselect_1.2.1 fitdistrplus_1.1-11<br />
[35] XML_3.99-0.14 zoo_1.8-12<br />
[37] GenomicAlignments_1.34.0 org.Mm.eg.db_3.16.0<br />
[39] xtable_1.8-4 evaluate_0.22<br />
[41] cli_3.6.3 zlibbioc_1.44.0<br />
[43] rstudioapi_0.15.0 miniUI_0.1.1.1<br />
[45] sp_2.1-1 whisker_0.4.1<br />
[47] bslib_0.5.1 fastmatch_1.1-4<br />
[49] treeio_1.22.0 shiny_1.7.5.1<br />
[51] xfun_0.40 clue_0.3-65<br />
[53] gson_0.1.0 cluster_2.1.4<br />
[55] tidygraph_1.2.3 KEGGREST_1.38.0<br />
[57] ggrepel_0.9.4 ape_5.7-1<br />
[59] listenv_0.9.0 Biostrings_2.66.0<br />
[61] png_0.1-8 future_1.33.0<br />
[63] withr_3.0.1 bitops_1.0-7<br />
[65] ggforce_0.4.1 plyr_1.8.9<br />
[67] GSEABase_1.60.0 dqrng_0.3.1<br />
[69] pillar_1.9.0 GlobalOptions_0.1.2<br />
[71] cachem_1.0.8 fs_1.6.4<br />
[73] GetoptLong_1.0.5 DelayedMatrixStats_1.20.0 [75] vctrs_0.6.5
ellipsis_0.3.2<br />
[77] generics_0.1.3 tools_4.2.2<br />
[79] beeswarm_0.4.0 munsell_0.5.1<br />
[81] tweenr_2.0.2 fgsea_1.24.0<br />
[83] DelayedArray_0.24.0 fastmap_1.1.1<br />
[85] compiler_4.2.2 abind_1.4-5<br />
[87] httpuv_1.6.8 rtracklayer_1.58.0<br />
[89] plotly_4.10.3 GenomeInfoDbData_1.2.9<br />
[91] gridExtra_2.3 lattice_0.20-45<br />
[93] deldir_1.0-9 utf8_1.2.4<br />
[95] later_1.3.1 jsonlite_1.8.7<br />
[97] scales_1.3.0 graph_1.76.0<br />
[99] ScaledMatrix_1.6.0 tidytree_0.4.2<br />
[101] pbapply_1.7-2 carData_3.0-5<br />
[103] sparseMatrixStats_1.10.0 lazyeval_0.2.2<br />
[105] promises_1.2.1 car_3.1-2<br />
[107] doParallel_1.0.17 R.utils_2.12.2<br />
[109] goftest_1.2-3 spatstat.utils_3.0-3<br />
[111] reticulate_1.34.0 rmarkdown_2.25<br />
[113] statmod_1.5.0 webshot_0.5.5<br />
[115] Rtsne_0.16 downloader_0.4<br />
[117] uwot_0.1.16 igraph_1.5.1<br />
[119] HDF5Array_1.26.0 survival_3.5-7<br />
[121] ResidualMatrix_1.8.0 yaml_2.3.7<br />
[123] systemfonts_1.1.0 htmltools_0.5.6.1<br />
[125] memoise_2.0.1 BiocIO_1.8.0<br />
[127] locfit_1.5-9.8 graphlayouts_1.0.1<br />
[129] viridisLite_0.4.2 digest_0.6.37<br />
[131] mime_0.12 RSQLite_2.3.1<br />
[133] yulab.utils_0.1.7 future.apply_1.11.0<br />
[135] data.table_1.14.8 blob_1.2.4<br />
[137] R.oo_1.25.0 labeling_0.4.3<br />
[139] splines_4.2.2 Rhdf5lib_1.20.0<br />
[141] Cairo_1.6-1 RCurl_1.98-1.12<br />
[143] broom_1.0.5 hms_1.1.3<br />
[145] rhdf5_2.42.0 shape_1.4.6<br />
[147] aplot_0.2.2 sass_0.4.7<br />
[149] Rcpp_1.0.13 RANN_2.6.1<br />
[151] enrichplot_1.18.3 fansi_1.0.6<br />
[153] tzdb_0.4.0 parallelly_1.36.0<br />
[155] R6_2.5.1 ggridges_0.5.4<br />
[157] lifecycle_1.0.4 zip_2.3.0<br />
[159] ggsignif_0.6.4 leiden_0.4.3<br />
[161] jquerylib_0.1.4 Matrix_1.5-3<br />
[163] qvalue_2.30.0 RcppAnnoy_0.0.21<br />
[165] org.Hs.eg.db_3.16.0 iterators_1.0.14<br />
[167] spatstat.explore_3.2-3 htmlwidgets_1.6.2<br />
[169] beachmat_2.14.0 polyclip_1.10-6<br />
[171] crosstalk_1.2.0 shadowtext_0.1.2<br />
[173] gridGraphics_0.5-1 timechange_0.2.0<br />
[175] rvest_1.0.3 globals_0.16.2<br />
[177] spatstat.random_3.2-1 progressr_0.14.0<br />
[179] codetools_0.2-19 GO.db_3.16.0<br />
[181] metapod_1.6.0 gtools_3.9.4<br />
[183] getPass_0.2-2 R.methodsS3_1.8.2<br />
[185] gtable_0.3.5 DBI_1.1.3<br />
[187] git2r_0.32.0 highr_0.10<br />
[189] ggfun_0.1.3 tensor_1.5<br />
[191] httr_1.4.7 KernSmooth_2.23-22<br />
[193] stringi_1.8.4 reshape2_1.4.4<br />
[195] farver_2.1.2 annotate_1.76.0<br />
[197] viridis_0.6.4 magick_2.7.3<br />
[199] ggtree_3.4.4 xml2_1.3.5<br />
[201] BiocNeighbors_1.16.0 restfulr_0.0.15<br />
[203] ggplotify_0.1.2 scattermore_1.2<br />
[205] bit_4.0.5 scatterpie_0.2.1<br />
[207] spatstat.data_3.0-3 ggraph_2.1.0<br />
[209] pkgconfig_2.0.3 rstatix_0.7.2</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
