<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Francesc Castro-Giner" />

<meta name="date" content="2025-02-07" />

<title>Comapre 10x RNA-seq analysis of microenvironment mets and normal tissue</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.30/datatables.js"></script>
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.13.4/jszip.min.js"></script>
<link href="site_libs/dt-ext-buttons-1.13.4/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.13.4/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.print.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">met-paths</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheAcetoLab/sznurkowska-met-paths">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Comapre 10x RNA-seq analysis of
microenvironment mets and normal tissue</h1>
<h3 class="subtitle">Analysis of TME and normal cells from lung and
liver normal tissue and metastases</h3>
<h4 class="author">Francesc Castro-Giner</h4>
<h4 class="date">February 07, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-07
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>sznurkowska-met-paths/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250206code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250206)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250206code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250206)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsznurkowskametpathstree30d581dbf1cff0b0af4988bcfb642939a182358btargetblank30d581da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/tree/30d581dbf1cff0b0af4988bcfb642939a182358b" target="_blank">30d581d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomTheAcetoLabsznurkowskametpathstree30d581dbf1cff0b0af4988bcfb642939a182358btargetblank30d581da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/tree/30d581dbf1cff0b0af4988bcfb642939a182358b" target="_blank">30d581d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/patients/
    Ignored:    data/resources/
    Ignored:    data/rnaseq/
    Ignored:    data/seer/
    Ignored:    output/p26532_o26674/
    Ignored:    output/p26532_o28268/
    Ignored:    output/p26532_o28268_o34980/
    Ignored:    output/p26532_o34980/

Untracked files:
    Untracked:  analysis/br16-10x_rnaseq-cancer_cells-main.Rmd
    Untracked:  analysis/br16-10x_rnaseq-qc.Rmd
    Untracked:  analysis/mvt1-tumor-10x_rnaseq-qc.Rmd
    Untracked:  analysis/templates/
    Untracked:  code/R-functions/
    Untracked:  configuration/
    Untracked:  workflowr_update.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   sznurkowska-met-paths.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd</code>)
and HTML
(<code>docs/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheAcetoLab/sznurkowska-met-paths/blob/30d581dbf1cff0b0af4988bcfb642939a182358b/analysis/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd" target="_blank">30d581d</a>
</td>
<td>
Francesc Castro-Giner
</td>
<td>
2025-02-07
</td>
<td>
Add nsg normal analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries-additional-functions-and-data"
class="section level2" number="1">
<h2><span class="header-section-number">1</span> Load libraries,
additional functions and data</h2>
<p>Setup environment</p>
<pre class="r"><code>knitr::opts_chunk$set(results=&#39;asis&#39;, echo=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.align = &#39;center&#39;, fig.width = 3.5, fig.asp = 0.618, dpi = 600, dev = c(&quot;png&quot;, &quot;pdf&quot;))

options(stringsAsFactors = FALSE)

use_seed &lt;- 1100101
set.seed(use_seed)

if(!dir.exists(params$output_dir))
  dir.create(params$output_dir, recursive = TRUE, showWarnings = FALSE)</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(tidyverse)
library(showtext)
library(foreach)
library(DT)
library(knitr)
library(kableExtra)
library(cowplot)
library(colorblindr)
library(ggbeeswarm)
library(arsenal)
library(RColorBrewer)
library(ggpubr)
library(pheatmap)
library(patchwork)
library(openxlsx)
library(magrittr)

library(scater)
library(DropletUtils)
library(Seurat)
library(scran)
library(BiocSingular)
library(batchelor)
library(bluster)
library(celldex)
library(SingleR)
library(scDblFinder)
library(speckle)
library(miloR)
library(ComplexHeatmap)
library(clusterProfiler)
library(GSVA)
library(circlize)</code></pre>
<p>Load ggplot theme</p>
<pre class="r"><code>source(&quot;./configuration/rmarkdown/ggplot_theme.R&quot;)
source(&quot;./configuration/rmarkdown/color_palettes.R&quot;)</code></pre>
<p>Load utils functions</p>
<pre class="r"><code>source(&#39;./code/R-functions/heatmap_scale.R&#39;)</code></pre>
<p>Clean files generated in previous runs</p>
<pre class="r"><code>rmd_file &lt;- current_input()
if(!is.null(rmd_file)) {
  figures_dir &lt;- file.path(&#39;./docs/figure&#39;,rmd_file)
  if(dir.exists(figures_dir)) {
    unlink(file.path(figures_dir, &quot;*&quot;))
  }
}</code></pre>
</div>
<div id="data-processing" class="section level2" number="2">
<h2><span class="header-section-number">2</span> Data processing</h2>
<div id="original-data" class="section level3" number="2.1">
<h3><span class="header-section-number">2.1</span> Original data</h3>
<p>Data processing original TME sce data</p>
<pre class="r"><code>sce_comb &lt;- readRDS(file.path(params$output_dir_tme, &#39;sce_integrated-tme.rds&#39;))

sr_immgen &lt;- readRDS(file.path(params$output_dir_tme, &#39;SingleR_ImmGenData_annotation-tme.rds&#39;))

# Add cell type annotation
sce_comb$celltype_immgen_pruned.labels &lt;- sr_immgen$pred$pruned.labels
sce_comb$celltype_immgen.labels &lt;- sr_immgen$pred$labels
sce_comb$celltype_immgen_pruned.labels.fine &lt;- sr_immgen$pred_fine$pruned.labels
sce_comb$celltype_immgen.labels.fine &lt;- sr_immgen$pred_fine$labels

# Add cluster level annotation using immgen database
sr_immgen$pred_cluster_simplified %&lt;&gt;%
  dplyr::rename(
    celltype_immgen_cluster_labels = celltype_cluster_labels,
    celltype_immgen_cluster_pruned.labels = celltype_cluster_pruned.labels
  )
colData(sce_comb) &lt;- colData(sce_comb) %&gt;% data.frame %&gt;%
  mutate(cluster_mnn = as.character(cluster_mnn)) %&gt;%
  left_join(sr_immgen$pred_cluster_simplified) %&gt;%
  mutate(cluster_mnn = factor(cluster_mnn)) %&gt;%
  DataFrame


# Select sites
sce_comb &lt;- sce_comb[,sce_comb$site %in% c(&#39;liver&#39;, &#39;lung&#39;)]

# Remove doublets
sce_comb &lt;- sce_comb[,sce_comb$scDblFinder.class != &#39;doublet&#39;]

# Site levels order
site_caps_ord &lt;- c(&#39;Liver TME&#39;, &#39;Lung TME&#39;)
site_ord &lt;- c(&#39;liver_tme&#39;, &#39;lung_tme&#39;)


# Add new colData vars
colData(sce_comb) &lt;- colData(sce_comb) %&gt;% data.frame %&gt;% 
  mutate(
    site_caps = ifelse(site == &#39;liver&#39;, &#39;Liver&#39;, site),
    site_caps = ifelse(site == &#39;lung&#39;, &#39;Lung&#39;, site_caps),
    
    site = paste0(site, &#39;_tme&#39;),
    site_caps = paste(site_caps, &#39;TME&#39;),
    
    site = factor(site, levels = site_ord),
    site_caps = factor(site_caps, levels = site_caps_ord),
    
    site_mouse = paste(site, mouse_id, sep = &#39;_&#39;)
  ) %&gt;% 
  DataFrame

sce_tme &lt;- sce_comb
rm(sce_comb)</code></pre>
<p>Data processing original Normal sce data</p>
<pre class="r"><code>sce_comb &lt;- readRDS(file.path(params$output_dir_normal, &#39;sce_integrated.rds&#39;))
sr_immgen &lt;- readRDS(file.path(params$output_dir_normal, &#39;SingleR_ImmGenData_annotation.rds&#39;))

# Add cell type annotation
sce_comb$celltype_immgen_pruned.labels &lt;- sr_immgen$pred$pruned.labels
sce_comb$celltype_immgen.labels &lt;- sr_immgen$pred$labels
sce_comb$celltype_immgen_pruned.labels.fine &lt;- sr_immgen$pred_fine$pruned.labels
sce_comb$celltype_immgen.labels.fine &lt;- sr_immgen$pred_fine$labels

# Add cluster level annotation using immgen database
sr_immgen$pred_cluster_simplified %&lt;&gt;%
  dplyr::rename(
    celltype_immgen_cluster_labels = celltype_cluster_labels,
    celltype_immgen_cluster_pruned.labels = celltype_cluster_pruned.labels
  )
colData(sce_comb) &lt;- colData(sce_comb) %&gt;% data.frame %&gt;%
  mutate(cluster = as.character(cluster)) %&gt;%
  left_join(sr_immgen$pred_cluster_simplified) %&gt;%
  mutate(cluster = factor(cluster)) %&gt;%
  DataFrame

# Remove doublets
sce_comb &lt;- sce_comb[,sce_comb$scDblFinder.class != &#39;doublet&#39;]


# Site levels order
site_caps_ord &lt;- c(&#39;Liver Normal&#39;, &#39;Lung Normal&#39;)
site_ord &lt;- c(&#39;liver_normal&#39;, &#39;lung_normal&#39;)


# Add new colData vars
colData(sce_comb) &lt;- colData(sce_comb) %&gt;% data.frame %&gt;% 
  mutate(
    site_caps = ifelse(site == &#39;liver&#39;, &#39;Liver&#39;, site),
    site_caps = ifelse(site == &#39;lung&#39;, &#39;Lung&#39;, site_caps),
    
    site = paste0(site, &#39;_normal&#39;),
    site_caps = paste(site_caps, &#39;Normal&#39;),
    
    site = factor(site, levels = site_ord),
    site_caps = factor(site_caps, levels = site_caps_ord),
    
    mouse_id = mouse_hto,
    site_mouse = paste(site, mouse_id, sep = &#39;_&#39;)
  ) %&gt;% 
  DataFrame

sce_normal&lt;- sce_comb
rm(sce_comb)</code></pre>
</div>
<div id="batch-integration" class="section level3" number="2.2">
<h3><span class="header-section-number">2.2</span> Batch
integration</h3>
<p>Here we prepare the data for batch correction (next section). First,
we combine cells from all SCE objects into a single SCE and return
pre-computed per-gene variance at batch level. Then we adjust for batch
level library size. Returned object will contain log-normalized
expression values after adjusting the size factors for systematic
differences in coverage between</p>
<pre class="r"><code># Combine all SCE files into sce_comb and generate an combine gene  with pre-computed gene variance 
if(exists(&#39;sce_comb&#39;)) 
  rm(sce_comb)

# Combine count data
use_rows &lt;- intersect(rownames(sce_tme), rownames(sce_normal))
new_counts &lt;- cbind(counts(sce_tme[use_rows,]), counts(sce_normal[use_rows,]))
new_logcounts &lt;- cbind(logcounts(sce_tme[use_rows,]), logcounts(sce_normal[use_rows,]))

# Combine colData
use_cols &lt;- intersect(names(colData(sce_tme)), names(colData(sce_normal)))
use_cols &lt;- use_cols[use_cols != &#39;cluster_id_batch&#39;]
new_colData &lt;- rbind(colData(sce_tme)[,use_cols], colData(sce_normal)[,use_cols])

# Combined SCE
sce_comb &lt;- SingleCellExperiment(
  assays = list(counts = new_counts, logcounts = new_logcounts),
  colData = new_colData,
  rowData = rowData(sce_normal)[use_rows,]
)

# Modify coldata and rownames
sce_comb$batch &lt;-  gsub(&quot;_.*&quot;, &quot;&quot;, sce_comb$Sample)
sce_comb$origin &lt;- ifelse(sce_comb$batch == &#39;huNSG&#39;, &#39;normal&#39;, &#39;tme&#39;)
# sce_comb$site &lt;- 
sce_comb$site&lt;- factor(sce_comb$site, c(&#39;liver_normal&#39;, &#39;lung_normal&#39;, &#39;liver_tme&#39;, &#39;lung_tme&#39;))
# sce_comb$site_caps&lt;- recode(as.character(sce_comb$site), 
#                             liver_normal = &quot;Liver Normal&quot;, 
#                             lung_normal = &quot;Lung Normal&quot;, 
#                             liver_tme = &quot;Liver TME&quot;, 
#                             lung_tme = &quot;Lung TME&quot;)
sce_comb$site_caps&lt;- factor(sce_comb$site_caps, c(&#39;Liver Normal&#39;, &#39;Lung Normal&#39;, &#39;Liver TME&#39;, &#39;Lung TME&#39;))

# Recalculate per-gene variance at batch level
metadata(sce_comb)$precomputed_geneVar &lt;- foreach(i = unique(sce_comb$Sample)) %do%{
  use_sce &lt;- sce_comb[,sce_comb$Sample == i]
  modelGeneVar(use_sce, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
}


# Rescale each batch to adjust for differences in sequencing depth between batches. The multiBatchNorm() function recomputes log-normalized expression values after adjusting the size factors for systematic differences in coverage between SingleCellExperiment objects. (Size factors only remove biases between cells within a single batch.) This improves the quality of the correction by removing one aspect of the technical differences between batches.
sce_comb &lt;- multiBatchNorm(sce_comb, batch = sce_comb$Sample, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))

# Identifying a set of HVGs using stats from all batches.
combined_dec &lt;- combineVar(metadata(sce_comb)$precomputed_geneVar)
chosen_hvgs &lt;- combined_dec$bio &gt; 0 &amp; !rowData(sce_comb)$is.mito &amp; !rowData(sce_comb)$is.ribo

# Dimensionality reduction
sce_comb &lt;- runPCA(sce_comb, subset_row=chosen_hvgs, BSPARAM=BiocSingular::RandomParam(), BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
sce_comb$cluster_id  &lt;- clusterCells(sce_comb, use.dimred=&quot;PCA&quot;, BLUSPARAM=SNNGraphParam(k=25, type=&quot;rank&quot;, cluster.fun=&quot;louvain&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)))
tab &lt;- table(Cluster=sce_comb$cluster_id, Batch=sce_comb$Sample)
sce_comb &lt;- runTSNE(sce_comb, dimred=&quot;PCA&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
sce_comb &lt;- runUMAP(sce_comb, dimred=&quot;PCA&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))

saveRDS(sce_comb, file.path(params$output_dir, &#39;sce_integrated.rds&#39;))</code></pre>
</div>
<div id="batch-correction-using-mnn" class="section level3"
number="2.3">
<h3><span class="header-section-number">2.3</span> Batch correction
using MNN</h3>
<p>Here, we actually perform the correction itself. We also perform the
clustering.</p>
<pre class="r"><code>if(!exists(&#39;sce_comb&#39;))
  sce_comb &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated.rds&#39;))

# Identifying a set of HVGs using stats from all batches.
combined_dec &lt;- combineVar(metadata(sce_comb)$precomputed_geneVar)
# combined_hvg &lt;- getTopHVGs(combined_dec, n=2000, row.names = FALSE)
chosen_hvgs &lt;- combined_dec$bio &gt; 0 &amp; !rowData(sce_comb)$is.mito &amp; !rowData(sce_comb)$is.ribo

# MNN correction
sce_mnn &lt;- fastMNN(sce_comb,
                   batch = sce_comb$Sample, 
                   auto.merge = TRUE, 
                      subset.row=chosen_hvgs, 
                      BSPARAM=RandomParam(deferred=TRUE), 
                      BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed),
                      d=50, k=20)
# colData(sce_mnn) &lt;- colData(sce_comb)
rowData(sce_mnn) &lt;-  rowData(sce_comb)[chosen_hvgs,]

# Dimensionality reduction
sce_mnn &lt;- runTSNE(sce_mnn, dimred=&quot;corrected&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
sce_mnn &lt;- runUMAP(sce_mnn, dimred=&quot;corrected&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))

# Clustering using corrected values
sce_mnn$cluster_mnn  &lt;- clusterCells(sce_mnn, use.dimred=&quot;corrected&quot;, BLUSPARAM=SNNGraphParam(k=25, type=&quot;rank&quot;, cluster.fun=&quot;louvain&quot;, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)))

# Add corrected info to sce_comb
sce_comb$cluster_mnn &lt;- sce_mnn$cluster_mnn
reducedDim(sce_comb, &#39;corrected&#39;) &lt;- reducedDim(sce_mnn, &#39;corrected&#39;)
reducedDim(sce_comb, &#39;TSNE_corrected&#39;) &lt;- reducedDim(sce_mnn, &#39;TSNE&#39;)
reducedDim(sce_comb, &#39;UMAP_corrected&#39;) &lt;- reducedDim(sce_mnn, &#39;UMAP&#39;)
# plotUMAP(sce_mnn, color_by = &#39;cluster_mnn&#39;)

# MNN-related diagnostic examining the variance in the differences in expression between MNN pairs.
use_sce &lt;- sce_comb
rownames(use_sce) &lt;- make.names(rowData(use_sce)$gene_name, unique = TRUE)
metadata(sce_mnn)$mnnDeltaVariance &lt;- mnnDeltaVariance(
  use_sce,
  pairs=metadata(sce_mnn)$merge.info$pairs, 
  BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)
  )
rm(use_sce)

# Save SCE objects
saveRDS(sce_mnn, file.path(params$output_dir, &#39;sce_integrated_corrected.rds&#39;))
saveRDS(sce_comb, file.path(params$output_dir, &#39;sce_integrated.rds&#39;))</code></pre>
</div>
<div id="markers-detection" class="section level3" number="2.4">
<h3><span class="header-section-number">2.4</span> Markers
detection</h3>
<p>We identify the genes that drive separation between clusters and cell
types. These marker genes allow us to assign biological meaning to each
cluster based on their functional annotation. We added batch as a
blocking factor to the model. The block argument works for all effect
sizes shown above and is robust to differences in the log-fold changes
or variance between batches. However, it assumes that each pair of
clusters is present in at least one batch. In scenarios where cells from
two clusters never co-occur in the same batch, the associated pairwise
comparison will be impossible and is ignored during calculation of
summary statistics.</p>
<pre class="r"><code>marker_info &lt;- list()

if(!exists(&#39;sce_comb&#39;))
  sce_comb &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated.rds&#39;))

# TME-Normal markers for each cell type from ImmGen
ct_label &lt;- &#39;celltype_immgen_pruned.labels&#39;
## Remove low abundant cell types 
keep_labels &lt;- table(colData(sce_comb)[[ct_label]]) %&gt;% data.frame %&gt;% filter(Freq &gt;= 100) %&gt;% pull(Var1)
## Iterate by cell type
marker_info$origin_celltype_markers &lt;- foreach(use_label = keep_labels) %do% {
  use_cols &lt;- colData(sce_comb)[[ct_label]] %in%  use_label
  use_sce &lt;- sce_comb[,use_cols]
  scoreMarkers(
    use_sce, 
    groups = use_sce$origin, 
    lfc=0.5, 
    full.stats=TRUE,
    row.data=rowData(use_sce)[,c(&#39;gene_name&#39;, &#39;is.mito&#39;, &#39;is.ribo&#39;),drop=FALSE], 
    BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)
    )
}
names(marker_info$origin_celltype_markers) &lt;- keep_labels


# TME-Normal markers for each cell type and site from ImmGen
ct_label &lt;- &#39;celltype_immgen_pruned.labels&#39;
## Iterate by cell type and site
sce_comb$site_anatomy &lt;- gsub(&quot;_.*&quot;, &quot;&quot;, sce_comb$site)
marker_info$origin_site_celltype_markers &lt;-foreach(site = unique(sce_comb$site_anatomy)) %do% {
  use_sce &lt;- sce_comb[,sce_comb$site_anatomy == site]
  ## Remove low abundant cell types 
  keep_labels &lt;- table(colData(use_sce)[[ct_label]]) %&gt;% data.frame %&gt;% filter(Freq &gt;= 100) %&gt;% pull(Var1)
  ## Iterate by cell type
  res &lt;- foreach(use_label = keep_labels) %do% {
    use_cols &lt;- colData(use_sce)[[ct_label]] %in%  use_label
    use_sce_i&lt;- use_sce[,use_cols]
    scoreMarkers(
      use_sce_i, 
      groups = use_sce_i$origin, 
      lfc=0.5, 
      full.stats=TRUE,
      row.data=rowData(use_sce_i)[,c(&#39;gene_name&#39;, &#39;is.mito&#39;, &#39;is.ribo&#39;),drop=FALSE], 
      BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)
      )
  }
  names(res) &lt;- keep_labels
  res
}
names(marker_info$origin_site_celltype_markers) &lt;- unique(sce_comb$site)

saveRDS(marker_info, file.path(params$output_dir, &#39;scoreMarkers.rds&#39;))</code></pre>
</div>
<div id="macrophages-subtype-annotaion" class="section level3"
number="2.5">
<h3><span class="header-section-number">2.5</span> Macrophages subtype
annotaion</h3>
<pre class="r"><code>if(!exists(&#39;ref_mouse_imm&#39;))
  ref_mouse_imm &lt;- celldex::ImmGenData()

# Subset macrophages annotation from immgen
ref_mouse_imm_mf &lt;- ref_mouse_imm[,ref_mouse_imm$label.main == &#39;Macrophages&#39;]

# Run SingleR at cell level with fail label
use_cols &lt;- sce_comb$celltype_immgen_pruned.labels %in% &#39;Macrophages&#39;
pred_fine &lt;- SingleR(test=sce_comb[,use_cols], 
                ref=ref_mouse_imm_mf, 
                labels=ref_mouse_imm_mf$label.fine, 
                de.method=&quot;classic&quot;,
                BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))

sce_comb$mf_immgen_pruned.labels.fine &lt;- NA
sce_comb$mf_immgen_pruned.labels.fine[use_cols] &lt;- pred_fine$pruned.labels

saveRDS(sce_comb, file.path(params$output_dir, &#39;sce_integrated.rds&#39;))</code></pre>
</div>
<div id="load-data" class="section level3" number="2.6">
<h3><span class="header-section-number">2.6</span> Load data</h3>
<pre class="r"><code>sce_comb &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated.rds&#39;))
sce_mnn &lt;- readRDS(file.path(params$output_dir, &#39;sce_integrated_corrected.rds&#39;))
score_markers &lt;- readRDS(file.path(params$output_dir, &#39;scoreMarkers.rds&#39;))

# Coordinate colData between sce_comb and sce_mnn
colData(sce_mnn) &lt;- colData(sce_comb)
assay(sce_mnn, &#39;logcounts&#39;) &lt;- logcounts(sce_comb[rownames(sce_mnn),])

# Cell annotation
cell_annot &lt;- colData(sce_comb) %&gt;%
  data.frame %&gt;% 
  mutate(
    sample = paste(site, mouse_id, sep = &#39;_&#39;)
  )</code></pre>
</div>
</div>
<div id="umap-final" class="section level2" number="3">
<h2><span class="header-section-number">3</span> UMAP final</h2>
<div id="site" class="section level3" number="3.1">
<h3><span class="header-section-number">3.1</span> Site</h3>
<pre class="r"><code>plotUMAP(sce_mnn, colour_by=&quot;site_caps&quot;, point_alpha = 0.17, point_size = 0.05) + 
  scale_color_manual(values = site_type_palette) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 3))) +
  labs(
    color = &#39;&#39;,
    title = paste0(&#39;UMAP of all cells (&#39;,scales::number(ncol(sce_mnn), big.mark = &quot;&#39;&quot;),&#39; cells)&#39;)
  ) +
  theme(
    plot.title = element_text(size = 8, face = &#39;plain&#39;),
    axis.line = element_line(linewidth = rel(0.5)),
    axis.text=element_blank(),
    axis.ticks=element_blank()#,
    # axis.line =element_blank()
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/final-figures-umap-site-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="cell-type-individual-level" class="section level3"
number="3.2">
<h3><span class="header-section-number">3.2</span> Cell type individual
level</h3>
<p>Showing only cell types with &gt;= 50 cells</p>
<pre class="r"><code>celltype_label &lt;- &#39;celltype_immgen_pruned.labels&#39;
use_celltypes &lt;-  colData(sce_mnn)[,celltype_label] %&gt;% table %&gt;% data.frame %&gt;% filter(Freq &gt;= 50) %&gt;% collect %&gt;% .[[&#39;.&#39;]] %&gt;% as.character
use_sce &lt;- sce_mnn[,colData(sce_mnn)[,celltype_label] %in% use_celltypes]
use_sce[[celltype_label]] &lt;- factor(use_sce[[celltype_label]], levels = names(celltype_palette)[names(celltype_palette) %in% use_celltypes])

plotUMAP(use_sce, colour_by = celltype_label, point_alpha = 0.17, point_size = 0.05) + 
  scale_color_manual(values = celltype_palette) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 3))) +
  labs(
    color = &#39;&#39;,
    title = paste0(&#39;UMAP of selected cell types (&#39;,scales::number(ncol(use_sce), big.mark = &quot;&#39;&quot;),&#39; cells)&#39;)
  ) +
  theme(
    plot.title = element_text(size = 8, face = &#39;plain&#39;),
    axis.line = element_line(linewidth = rel(0.5)),
    axis.text=element_blank(),
    axis.ticks=element_blank()#,
    # axis.line =element_blank()
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/final-figures-umap-celltype-ind-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="lung-differential-abundance-of-cell-types"
class="section level2" number="4">
<h2><span class="header-section-number">4</span> Lung : Differential
abundance of cell types</h2>
<p>Differential abundance of cell types between normal lung tissue and
TME in lung metastasis</p>
<pre class="r"><code>ct_label &lt;- &#39;celltype_immgen_pruned.labels&#39;</code></pre>
<div id="statistical-analysis" class="section level3" number="4.1">
<h3><span class="header-section-number">4.1</span> Statistical
analysis</h3>
<p>Differential abundance using <code>speckle::propeller</code>
package</p>
<pre class="r"><code>use_data &lt;- cell_annot

# Remove low abundant cell types 
keep_labels &lt;- table(use_data[[ct_label]]) %&gt;% data.frame %&gt;% filter(Freq &gt;= 100) %&gt;% pull(Var1)
keep_rows &lt;- use_data[[ct_label]] %in% keep_labels
use_data &lt;- use_data[keep_rows,]

# Run propeller Global
propeller_global &lt;- propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$site) %&gt;% 
  rownames_to_column(&#39;cell_type&#39;)


# Run pairwise propeller to obtaib P-values
site_levels &lt;- use_data$site_caps %&gt;% unique
# site_combn &lt;- combn(site_levels, 2, simplify = FALSE)
site_combn &lt;- list(
  c(&#39;Lung Normal&#39;, &#39;Lung TME&#39;)
  # c(&#39;Liver Normal&#39;, &#39;Liver TME&#39;)
)
sc &lt;-  site_combn[[1]]
propeller_pairwise &lt;-foreach(sc = site_combn, .combine = rbind) %do% {
  use_data &lt;- cell_annot %&gt;%
    filter(site_caps %in% sc) %&gt;%
    mutate(
      group = ifelse(site_caps == sc[1], &#39;group1&#39;, &#39;group2&#39;)
    )
  keep_rows &lt;- use_data[[ct_label]] %in% keep_labels
  use_data &lt;- use_data[keep_rows,]
  propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$group) %&gt;%
    mutate(
      group1 = sc[1],
      group2 = sc[2],
      FDR.sign = symnum(FDR, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001,0.001, 0.01, 0.05, 1), symbols = c(&quot;****&quot;, &quot;***&quot;, &quot;**&quot;, &quot;*&quot;, &quot;&quot;)),
      p.sign = symnum(P.Value, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c(&quot;****&quot;, &quot;***&quot;, &quot;**&quot;, &quot;*&quot;, &quot;&quot;))
    ) %&gt;%
    dplyr::rename(cell_type = BaselineProp.clusters)
}</code></pre>
<p>Table of results with pairwise comparison between sites for cluster
abundance.</p>
<pre class="r"><code>propeller_pairwise %&gt;%
  arrange(FDR) %&gt;% 
  datatable(., 
            rownames = FALSE, 
            filter = &#39;top&#39;, 
            caption = &#39;Pairwise comparison between cell types for cluster abundance using speckle::propeller&#39;,
            extensions = &#39;Buttons&#39;, 
            options = list(
              dom = &#39;Blfrtip&#39;,
              buttons = c(&#39;csv&#39;, &#39;excel&#39;)
            )) %&gt;% 
  formatPercentage(names(propeller_pairwise)[2:4], digits = 3) %&gt;% 
  formatRound(names(propeller_pairwise)[5:6], digits = 3, interval = -1000) %&gt;% 
  formatSignif(c(&#39;P.Value&#39;, &#39;FDR&#39;), digits = 3)</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-0e41329c1963262c826a" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-0e41329c1963262c826a">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;DC&quot;,&quot;Endothelial cells&quot;,&quot;Fibroblasts&quot;,&quot;Macrophages&quot;,&quot;Monocytes&quot;,&quot;Neutrophils&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.013226909920182\" data-max=\"0.343215507411631\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.003094620394274\" data-max=\"0.464112467595832\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.002878996041525\" data-max=\"0.838224153278551\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.023229084680661\" data-max=\"161.206358363013\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-13.4375788615818\" data-max=\"12.7649856438602\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"1.882e-12\" data-max=\"0.040855897578028\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"1.1296e-11\" data-max=\"0.040855897578028\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Pairwise comparison between cell types for cluster abundance using speckle::propeller<\/caption>","data":[["Macrophages","Endothelial cells","Neutrophils","Fibroblasts","Monocytes","DC"],[0.3432155074116305,0.276054732041049,0.154618015963512,0.01322690992018244,0.1753705815279361,0.03751425313568985],[0.0194711798378831,0.464112467595832,0.2231761122415151,0.003094620394274169,0.2426393055298646,0.04750631440063126],[0.8382241532785503,0.002878996041525346,0.03538738955340506,0.02733678805416499,0.07301525286438433,0.02315742020797019],[0.02322908468066135,161.2063583630134,6.306656553592565,0.1132035112589856,3.323131756874599,2.051451067260109],[-13.43757886158176,12.76498564386018,4.827394664914782,-4.821532111493189,3.409209096018567,2.165343824382095],[1.882701035622922e-12,5.419774020842121e-12,6.961329360084986e-05,7.064152779430725e-05,0.002377263513055332,0.04085589757802752],[1.129620621373753e-11,1.625932206252636e-11,0.0001059622916914609,0.0001059622916914609,0.002852716215666399,0.04085589757802752],["Lung Normal","Lung Normal","Lung Normal","Lung Normal","Lung Normal","Lung Normal"],["Lung TME","Lung TME","Lung TME","Lung TME","Lung TME","Lung TME"],["****","****","***","***","**","*"],["****","****","****","****","**","*"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>cell_type<\/th>\n      <th>BaselineProp.Freq<\/th>\n      <th>PropMean.group1<\/th>\n      <th>PropMean.group2<\/th>\n      <th>PropRatio<\/th>\n      <th>Tstatistic<\/th>\n      <th>P.Value<\/th>\n      <th>FDR<\/th>\n      <th>group1<\/th>\n      <th>group2<\/th>\n      <th>FDR.sign<\/th>\n      <th>p.sign<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":7,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, -1000, \",\", \".\", null);\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, -1000, \",\", \".\", null);\n  }"},{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[1,2,3,4,5,6,7]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render","options.columnDefs.5.render","options.columnDefs.6.render"],"jsHooks":[]}</script>
</div>
<div id="distribution-by-site" class="section level3" number="4.2">
<h3><span class="header-section-number">4.2</span> Distribution by
site</h3>
<pre class="r"><code># Create data
use_data &lt;- cell_annot %&gt;% 
  filter(site_caps %in% unlist(site_combn)) %&gt;% 
  dplyr::rename(&#39;celltype_label&#39; = ct_label) %&gt;% 
  group_by(site_caps, celltype_label) %&gt;% 
  summarise(freq = n()) %&gt;% 
  dplyr::rename(cell_type = celltype_label) %&gt;% 
  na.omit()

# Remove low abundant cell types
cell_type_freq &lt;- use_data %&gt;% group_by(cell_type) %&gt;% summarise(totalfreq = sum(freq))
site_filtered &lt;- use_data %&gt;% group_by(site_caps) %&gt;% summarise(freq = sum(freq)) %&gt;% filter(freq &gt;= 500)
use_data &lt;- use_data %&gt;% 
  filter(site_caps %in% site_filtered$site_caps) %&gt;% 
  left_join(cell_type_freq) %&gt;% 
  filter(totalfreq &gt;= 50) %&gt;% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data &lt;- use_data %&gt;% 
  arrange(site_caps, cell_type)%&gt;% 
  group_by(site_caps) %&gt;%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion &lt; 0.03, &#39;&#39;, cell_type_label)
    
  )</code></pre>
<div id="barplot-stacked" class="section level4" number="4.2.1">
<h4><span class="header-section-number">4.2.1</span> Barplot
stacked</h4>
<pre class="r"><code>arranged_data %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, fill = cell_type)) +
  geom_col(alpha = 0.85, color = &#39;black&#39;, linewidth = 0.2) +
  scale_fill_manual(values = celltype_palette) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  # geom_text(aes(y = label_y, label = cell_type_label), colour = &quot;white&quot;, size = geom_text_size) +
  theme(axis.ticks.x = element_blank()) +
  labs(
    fill = NULL, 
    x = NULL,
    y = &#39;Proportion of cells&#39;
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/da_barplot_site_stacked_immgen-lung-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="barplot-facet" class="section level4" number="4.2.2">
<h4><span class="header-section-number">4.2.2</span> Barplot facet</h4>
<pre class="r"><code>arranged_data %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, fill = site_caps, group = cell_type)) +
  geom_col(alpha = 0.85, color = &#39;black&#39;, size = 0.2) +
  scale_fill_manual(values = site_palette) +
  scale_y_continuous(expand = c(0.02, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), ncol = 3, scales = &#39;free_y&#39;) +
  theme(
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x = element_blank(),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    panel.border = element_rect(colour = &quot;black&quot;, size=0.5),
    strip.background = element_blank()
  ) +
  labs(
    fill = NULL, 
    x = NULL,
    y = &#39;Proportion of cells&#39;
  ) #+</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/da_barplot_site_celltypefacet_immgen-lung-1.png" width="3120" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # guides(fill = &#39;none&#39;)</code></pre>
</div>
<div id="piechart" class="section level4" number="4.2.3">
<h4><span class="header-section-number">4.2.3</span> Piechart</h4>
<pre class="r"><code>arranged_data %&gt;% 
  ggplot(aes(x = &quot;&quot;, y = proportion, fill = cell_type)) +
  # geom_bar(stat=&quot;identity&quot;, width=1, color=&quot;white&quot;) +
  geom_bar(stat=&quot;identity&quot;, width=1) +
  coord_polar(&quot;y&quot;, start=0) +
  scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = &#39;&#39;, 
    x = NULL,
    y = NULL
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/da_piechart_site_celltypefacet_immgen-lung-1.png" width="2400" style="display: block; margin: auto;" /></p>
</div>
<div id="doughnut-chart" class="section level4" number="4.2.4">
<h4><span class="header-section-number">4.2.4</span> Doughnut chart</h4>
<pre class="r"><code>i &lt;- unique(arranged_data$site_caps)[1]
use_data &lt;- foreach(i = unique(arranged_data$site_caps), .combine = rbind) %do% {
  x &lt;- arranged_data %&gt;% 
    filter(site_caps == i) %&gt;% 
    arrange(cell_type)
  x$ymax &lt;- cumsum(x$proportion)
  x$ymin &lt;- c(0, head(x$ymax, n = -1))
  x
}

use_data %&gt;% 
  ggplot(aes(ymax=ymax, ymin=ymin, xmax=4, xmin=3, fill=cell_type)) +
  geom_rect() +
  coord_polar(theta=&quot;y&quot;) + # Try to remove that to understand how the chart is built initially
  xlim(c(2, 4)) + # Try to remove that to see how to make a pie chart
  scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = &#39;&#39;, 
    x = NULL,
    y = NULL
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/da_doughnut_site_celltypefacet_immgen-lung-1.png" width="2400" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="distribution-by-site-and-mouse" class="section level3"
number="4.3">
<h3><span class="header-section-number">4.3</span> Distribution by site
and mouse</h3>
<pre class="r"><code># Create data
use_data &lt;- cell_annot %&gt;% 
  filter(site_caps %in% unlist(site_combn)) %&gt;% 
  dplyr::rename(&#39;celltype_label&#39; = ct_label) %&gt;% 
  group_by(site_caps, mouse_id, celltype_label) %&gt;% 
  summarise(freq = n()) %&gt;% 
  dplyr::rename(cell_type = celltype_label) %&gt;% 
  na.omit()

# Remove low abundant cell types and sites
cell_type_freq &lt;- use_data %&gt;% group_by(cell_type) %&gt;% summarise(totalfreq = sum(freq))
site_filtered &lt;- use_data %&gt;% group_by(site_caps) %&gt;% summarise(freq = sum(freq)) %&gt;% filter(freq &gt;= 500)
use_data &lt;- use_data %&gt;% 
  filter(site_caps %in% site_filtered$site_caps) %&gt;% 
  left_join(cell_type_freq) %&gt;% 
  filter(totalfreq &gt;= 50) %&gt;% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data &lt;- use_data %&gt;% 
  arrange(site_caps, cell_type)%&gt;% 
  group_by(site_caps, mouse_id) %&gt;%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion &lt; 0.03, &#39;&#39;, cell_type_label)
  )</code></pre>
<div id="stripchart-boxplot" class="section level4" number="4.3.1">
<h4><span class="header-section-number">4.3.1</span>
Stripchart-Boxplot</h4>
<pre class="r"><code>max_prop_cell_type &lt;- arranged_data %&gt;% group_by(cell_type) %&gt;% summarise(max_proportion = max(proportion))
stat_test &lt;- propeller_pairwise %&gt;% 
  filter(group1 %in% arranged_data$site_caps &amp; group2 %in%  arranged_data$site_caps) %&gt;%
  left_join(max_prop_cell_type) %&gt;% 
  arrange(cell_type, group1, group2) %&gt;% 
  na.omit() %&gt;% 
  filter(FDR &lt; 0.05)
  
add_count_y &lt;- stat_test %&gt;% group_by(cell_type) %&gt;% reframe(group1 = group1, group2 = group2, n = 1:n())
stat_test &lt;- stat_test %&gt;% left_join(add_count_y) %&gt;% mutate(y.position = max_proportion + (n*max_proportion/10))


arranged_data %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, color = site_caps)) +
  geom_boxplot(alpha = 0.4, color = &#39;black&#39;, aes(fill = site_caps), 
               show.legend = FALSE, size = 0.25, fatten = 1) +
  # geom_point(alpha = 1, size = 2, shape = 21, color = &#39;black&#39;, aes(fill = site_caps)) +
  geom_quasirandom(method=&quot;smiley&quot;, width = 0.1, alpha = 1, size = 1, 
                   shape = 21, color = &#39;black&#39;, aes(fill = site_caps), stroke = 0.4) +
  # geom_quasirandom(method=&quot;smiley&quot;, width = 0.1, alpha = 1, size = 1, aes(color = site_caps)) +
  # scale_fill_OkabeIto() +
  # scale_color_OkabeIto() +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  # scale_y_continuous(expand = c(0, 0)) +
  scale_y_continuous(expand = c(0.02, 0, 0.08, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), scales = &#39;free_y&#39;, ncol = 2) +
  theme(
    # strip.background = element_blank(),
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    panel.border = element_rect(colour = &quot;black&quot;, size=0.5),
    strip.background = element_blank()
  ) +
  # guides( color = &#39;none&#39;) +
  guides(fill = guide_legend(override.aes = list(size = 3))) +
  labs(
    fill = NULL,
    x = NULL,
    y = &#39;Proportion of cells&#39;
  ) +
  stat_pvalue_manual(stat_test, label = &#39;FDR.sign&#39;, size = 2)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/da_stripchart_site_and_mouse_celltypefacet1_immgen-lung-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="liver-differential-abundance-of-cell-types"
class="section level2" number="5">
<h2><span class="header-section-number">5</span> Liver : Differential
abundance of cell types</h2>
<p>Differential abundance of cell types between normal lung tissue and
TME in lung metastasis</p>
<pre class="r"><code>ct_label &lt;- &#39;celltype_immgen_pruned.labels&#39;</code></pre>
<div id="statistical-analysis-1" class="section level3" number="5.1">
<h3><span class="header-section-number">5.1</span> Statistical
analysis</h3>
<p>Differential abundance using <code>speckle::propeller</code>
package</p>
<pre class="r"><code>use_data &lt;- cell_annot

# Remove low abundant cell types 
keep_labels &lt;- table(use_data[[ct_label]]) %&gt;% data.frame %&gt;% filter(Freq &gt;= 100) %&gt;% pull(Var1)
keep_rows &lt;- use_data[[ct_label]] %in% keep_labels
use_data &lt;- use_data[keep_rows,]

# Run propeller Global
propeller_global &lt;- propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$site) %&gt;% 
  rownames_to_column(&#39;cell_type&#39;)


# Run pairwise propeller to obtaib P-values
site_levels &lt;- use_data$site_caps %&gt;% unique
# site_combn &lt;- combn(site_levels, 2, simplify = FALSE)
site_combn &lt;- list(
  # c(&#39;Lung Normal&#39;, &#39;Lung TME&#39;)
  c(&#39;Liver Normal&#39;, &#39;Liver TME&#39;)
)
sc &lt;-  site_combn[[1]]
propeller_pairwise &lt;-foreach(sc = site_combn, .combine = rbind) %do% {
  use_data &lt;- cell_annot %&gt;%
    filter(site_caps %in% sc) %&gt;%
    mutate(
      group = ifelse(site_caps == sc[1], &#39;group1&#39;, &#39;group2&#39;)
    )
  keep_rows &lt;- use_data[[ct_label]] %in% keep_labels
  use_data &lt;- use_data[keep_rows,]
  propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$group) %&gt;%
    mutate(
      group1 = sc[1],
      group2 = sc[2],
      FDR.sign = symnum(FDR, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001,0.001, 0.01, 0.05, 1), symbols = c(&quot;****&quot;, &quot;***&quot;, &quot;**&quot;, &quot;*&quot;, &quot;&quot;)),
      p.sign = symnum(P.Value, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c(&quot;****&quot;, &quot;***&quot;, &quot;**&quot;, &quot;*&quot;, &quot;&quot;))
    ) %&gt;%
    dplyr::rename(cell_type = BaselineProp.clusters)
}</code></pre>
<p>Table of results with pairwise comparison between sites for cluster
abundance.</p>
<pre class="r"><code>propeller_pairwise %&gt;%
  arrange(FDR) %&gt;% 
  datatable(., 
            rownames = FALSE, 
            filter = &#39;top&#39;, 
            caption = &#39;Pairwise comparison between cell types for cluster abundance using speckle::propeller&#39;,
            extensions = &#39;Buttons&#39;, 
            options = list(
              dom = &#39;Blfrtip&#39;,
              buttons = c(&#39;csv&#39;, &#39;excel&#39;)
            )) %&gt;% 
  formatPercentage(names(propeller_pairwise)[2:4], digits = 3) %&gt;% 
  formatRound(names(propeller_pairwise)[5:6], digits = 3, interval = -1000) %&gt;% 
  formatSignif(c(&#39;P.Value&#39;, &#39;FDR&#39;), digits = 3)</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-cf74d9caecbce7389320" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-cf74d9caecbce7389320">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;DC&quot;,&quot;Endothelial cells&quot;,&quot;Fibroblasts&quot;,&quot;Macrophages&quot;,&quot;Monocytes&quot;,&quot;Neutrophils&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.00965895540186\" data-max=\"0.68602432625805\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.018122566374679\" data-max=\"0.748497145456025\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.001141584179223\" data-max=\"0.590183258726857\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.068306614344826\" data-max=\"15.8749277578495\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-5.06655578152341\" data-max=\"4.77438196303927\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"3.5120198581e-05\" data-max=\"0.341409587876926\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.000190578821397\" data-max=\"0.341409587876926\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Pairwise comparison between cell types for cluster abundance using speckle::propeller<\/caption>","data":[["Macrophages","DC","Fibroblasts","Endothelial cells","Monocytes","Neutrophils"],[0.1314094920104937,0.06403529692344384,0.009658955401860243,0.6860243262580491,0.07941807774862866,0.02945385165752444],[0.01818690343595203,0.1170606953181971,0.01812256637467967,0.7484971454560246,0.061268277866466,0.03686441154868057],[0.2662539142130593,0.009877897568459328,0.001141584179223672,0.5901832587268563,0.1113542651787403,0.02118908013366119],[0.06830661434482677,11.85077031897743,15.87492775784946,1.268245302434846,0.5502104276663423,1.739783478855101],[-5.066555781523409,4.774381963039269,4.673350789571644,1.183066349425227,-1.084680569550041,0.9706386078852517],[3.512019858161329e-05,7.37020640748081e-05,9.528941069862024e-05,0.2483701876423464,0.2888393480276321,0.3414095878769255],[0.0001905788213972405,0.0001905788213972405,0.0001905788213972405,0.3414095878769255,0.3414095878769255,0.3414095878769255],["Liver Normal","Liver Normal","Liver Normal","Liver Normal","Liver Normal","Liver Normal"],["Liver TME","Liver TME","Liver TME","Liver TME","Liver TME","Liver TME"],["***","***","***","","",""],["****","****","****","","",""]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>cell_type<\/th>\n      <th>BaselineProp.Freq<\/th>\n      <th>PropMean.group1<\/th>\n      <th>PropMean.group2<\/th>\n      <th>PropRatio<\/th>\n      <th>Tstatistic<\/th>\n      <th>P.Value<\/th>\n      <th>FDR<\/th>\n      <th>group1<\/th>\n      <th>group2<\/th>\n      <th>FDR.sign<\/th>\n      <th>p.sign<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":7,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, -1000, \",\", \".\", null);\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, -1000, \",\", \".\", null);\n  }"},{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[1,2,3,4,5,6,7]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render","options.columnDefs.5.render","options.columnDefs.6.render"],"jsHooks":[]}</script>
</div>
<div id="distribution-by-site-1" class="section level3" number="5.2">
<h3><span class="header-section-number">5.2</span> Distribution by
site</h3>
<pre class="r"><code># Create data
use_data &lt;- cell_annot %&gt;% 
  filter(site_caps %in% unlist(site_combn)) %&gt;% 
  dplyr::rename(&#39;celltype_label&#39; = ct_label) %&gt;% 
  group_by(site_caps, celltype_label) %&gt;% 
  summarise(freq = n()) %&gt;% 
  dplyr::rename(cell_type = celltype_label) %&gt;% 
  na.omit()

# Remove low abundant cell types
cell_type_freq &lt;- use_data %&gt;% group_by(cell_type) %&gt;% summarise(totalfreq = sum(freq))
site_filtered &lt;- use_data %&gt;% group_by(site_caps) %&gt;% summarise(freq = sum(freq)) %&gt;% filter(freq &gt;= 500)
use_data &lt;- use_data %&gt;% 
  filter(site_caps %in% site_filtered$site_caps) %&gt;% 
  left_join(cell_type_freq) %&gt;% 
  filter(totalfreq &gt;= 50) %&gt;% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data &lt;- use_data %&gt;% 
  arrange(site_caps, cell_type)%&gt;% 
  group_by(site_caps) %&gt;%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion &lt; 0.03, &#39;&#39;, cell_type_label)
    
  )</code></pre>
<div id="barplot-stacked-1" class="section level4" number="5.2.1">
<h4><span class="header-section-number">5.2.1</span> Barplot
stacked</h4>
<pre class="r"><code>arranged_data %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, fill = cell_type)) +
  geom_col(alpha = 0.85, color = &#39;black&#39;, size = 0.2) +
  scale_fill_manual(values = celltype_palette) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  # geom_text(aes(y = label_y, label = cell_type_label), colour = &quot;white&quot;, size = geom_text_size) +
  theme(axis.ticks.x = element_blank()) +
  labs(
    fill = NULL, 
    x = NULL,
    y = &#39;Proportion of cells&#39;
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/da_barplot_site_stacked_immgen-liver-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="barplot-facet-1" class="section level4" number="5.2.2">
<h4><span class="header-section-number">5.2.2</span> Barplot facet</h4>
<pre class="r"><code>arranged_data %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, fill = site_caps, group = cell_type)) +
  geom_col(alpha = 0.85, color = &#39;black&#39;, linewidth = 0.2) +
  scale_fill_manual(values = site_palette) +
  scale_y_continuous(expand = c(0.02, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), ncol = 3, scales = &#39;free_y&#39;) +
  theme(
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x = element_blank(),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    panel.border = element_rect(colour = &quot;black&quot;, size=0.5),
    strip.background = element_blank()
  ) +
  labs(
    fill = NULL, 
    x = NULL,
    y = &#39;Proportion of cells&#39;
  ) #+</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/da_barplot_site_celltypefacet_immgen-liver-1.png" width="3120" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # guides(fill = &#39;none&#39;)</code></pre>
</div>
<div id="piechart-1" class="section level4" number="5.2.3">
<h4><span class="header-section-number">5.2.3</span> Piechart</h4>
<pre class="r"><code>arranged_data %&gt;% 
  ggplot(aes(x = &quot;&quot;, y = proportion, fill = cell_type)) +
  # geom_bar(stat=&quot;identity&quot;, width=1, color=&quot;white&quot;) +
  geom_bar(stat=&quot;identity&quot;, width=1) +
  coord_polar(&quot;y&quot;, start=0) +
  scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = &#39;&#39;, 
    x = NULL,
    y = NULL
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/da_piechart_site_celltypefacet_immgen-liver-1.png" width="2400" style="display: block; margin: auto;" /></p>
</div>
<div id="doughnut-chart-1" class="section level4" number="5.2.4">
<h4><span class="header-section-number">5.2.4</span> Doughnut chart</h4>
<pre class="r"><code>i &lt;- unique(arranged_data$site_caps)[1]
use_data &lt;- foreach(i = unique(arranged_data$site_caps), .combine = rbind) %do% {
  x &lt;- arranged_data %&gt;% 
    filter(site_caps == i) %&gt;% 
    arrange(cell_type)
  x$ymax &lt;- cumsum(x$proportion)
  x$ymin &lt;- c(0, head(x$ymax, n = -1))
  x
}

use_data %&gt;% 
  ggplot(aes(ymax=ymax, ymin=ymin, xmax=4, xmin=3, fill=cell_type)) +
  geom_rect() +
  coord_polar(theta=&quot;y&quot;) + # Try to remove that to understand how the chart is built initially
  xlim(c(2, 4)) + # Try to remove that to see how to make a pie chart
  scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = &#39;&#39;, 
    x = NULL,
    y = NULL
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/da_doughnut_site_celltypefacet_immgen-liver-1.png" width="2400" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="distribution-by-site-and-mouse-1" class="section level3"
number="5.3">
<h3><span class="header-section-number">5.3</span> Distribution by site
and mouse</h3>
<pre class="r"><code># Create data
use_data &lt;- cell_annot %&gt;% 
  filter(site_caps %in% unlist(site_combn)) %&gt;% 
  dplyr::rename(&#39;celltype_label&#39; = ct_label) %&gt;% 
  group_by(site_caps, mouse_id, celltype_label) %&gt;% 
  summarise(freq = n()) %&gt;% 
  dplyr::rename(cell_type = celltype_label) %&gt;% 
  na.omit()

# Remove low abundant cell types and sites
cell_type_freq &lt;- use_data %&gt;% group_by(cell_type) %&gt;% summarise(totalfreq = sum(freq))
site_filtered &lt;- use_data %&gt;% group_by(site_caps) %&gt;% summarise(freq = sum(freq)) %&gt;% filter(freq &gt;= 500)
use_data &lt;- use_data %&gt;% 
  filter(site_caps %in% site_filtered$site_caps) %&gt;% 
  left_join(cell_type_freq) %&gt;% 
  filter(totalfreq &gt;= 50) %&gt;% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data &lt;- use_data %&gt;% 
  arrange(site_caps, cell_type)%&gt;% 
  group_by(site_caps, mouse_id) %&gt;%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion &lt; 0.03, &#39;&#39;, cell_type_label)
  )</code></pre>
<div id="stripchart-boxplot-1" class="section level4" number="5.3.1">
<h4><span class="header-section-number">5.3.1</span>
Stripchart-Boxplot</h4>
<pre class="r"><code>max_prop_cell_type &lt;- arranged_data %&gt;% group_by(cell_type) %&gt;% summarise(max_proportion = max(proportion))
stat_test &lt;- propeller_pairwise %&gt;% 
  filter(group1 %in% arranged_data$site_caps &amp; group2 %in%  arranged_data$site_caps) %&gt;%
  left_join(max_prop_cell_type) %&gt;% 
  arrange(cell_type, group1, group2) %&gt;% 
  na.omit() %&gt;% 
  filter(FDR &lt; 0.05)
  
add_count_y &lt;- stat_test %&gt;% group_by(cell_type) %&gt;% reframe(group1 = group1, group2 = group2, n = 1:n())
stat_test &lt;- stat_test %&gt;% left_join(add_count_y) %&gt;% mutate(y.position = max_proportion + (n*max_proportion/10))


arranged_data %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, color = site_caps)) +
  geom_boxplot(alpha = 0.4, color = &#39;black&#39;, aes(fill = site_caps), 
               show.legend = FALSE, size = 0.25, fatten = 1) +
  # geom_point(alpha = 1, size = 2, shape = 21, color = &#39;black&#39;, aes(fill = site_caps)) +
  geom_quasirandom(method=&quot;smiley&quot;, width = 0.1, alpha = 1, size = 1, 
                   shape = 21, color = &#39;black&#39;, aes(fill = site_caps), stroke = 0.4) +
  # geom_quasirandom(method=&quot;smiley&quot;, width = 0.1, alpha = 1, size = 1, aes(color = site_caps)) +
  # scale_fill_OkabeIto() +
  # scale_color_OkabeIto() +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  # scale_y_continuous(expand = c(0, 0)) +
  scale_y_continuous(expand = c(0.02, 0, 0.08, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), scales = &#39;free_y&#39;, ncol = 2) +
  theme(
    # strip.background = element_blank(),
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    panel.border = element_rect(colour = &quot;black&quot;, size=0.5),
    strip.background = element_blank()
  ) +
  # guides( color = &#39;none&#39;) +
  guides(fill = guide_legend(override.aes = list(size = 3))) +
  labs(
    fill = NULL,
    x = NULL,
    y = &#39;Proportion of cells&#39;
  ) +
  stat_pvalue_manual(stat_test, label = &#39;FDR.sign&#39;, size = 2)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/da_stripchart_site_and_mouse_celltypefacet1_immgen-liver-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="lung-tmenormal-markers-for-each-cell-type"
class="section level2" number="6">
<h2><span class="header-section-number">6</span> Lung: TME/Normal
markers for each cell type</h2>
<p>For each cell type we tried to identify potential markers to
distinguish between Normal and TME cells</p>
<pre class="r"><code>use_site &lt;- &#39;lung&#39;
n_top_mk &lt;- 20
celltype_label &lt;- &#39;celltype_immgen_pruned.labels&#39;
# names(score_markers$origin_site_celltype_markers[[use_site]])
j &lt;- score_markers$origin_site_celltype_markers[[use_site]][[1]][[1]]
mk_list &lt;- foreach(i = score_markers$origin_site_celltype_markers[[use_site]]) %do% {
  foreach(j = i, .combine = c) %do% {
    j %&gt;% data.frame %&gt;% arrange(desc(mean.AUC)) %&gt;% head(n_top_mk) %&gt;% rownames()
  } %&gt;% unique
}
names(mk_list) &lt;-names(score_markers$origin_site_celltype_markers[[use_site]])

# Save scoreMarkers to Excel
# File name summary
rmd_file &lt;- current_input()
if(is.null(rmd_file))
  rmd_file &lt;- &#39;tmp&#39;

# Generate file
file_xlsx &lt;- file.path(
  &#39;./docs/file&#39;, rmd_file, 
  paste0(&#39;scoreMarkers_&#39;, use_site, &#39;.xlsx&#39;)
  )
dir.create(dirname(file_xlsx), recursive = TRUE, showWarnings = FALSE)

# Generate workbook
wb &lt;- createWorkbook()

# Create DfF of results
i &lt;- score_markers$origin_site_celltype_markers[[use_site]][[1]]
use_mk_list &lt;- score_markers$origin_site_celltype_markers[[use_site]]
x &lt;- foreach(use_ct = names(use_mk_list)) %do% {
  foreach(use_condition = names(use_mk_list[[use_ct]])) %do% {
    sheet_name &lt;- paste(use_ct, use_condition)
    addWorksheet(wb, paste(use_ct, use_condition))
    res &lt;- use_mk_list[[use_ct]][[use_condition]] %&gt;% data.frame %&gt;% 
      arrange(desc(mean.AUC)) %&gt;% 
      dplyr::select(gene_name, is.mito, is.ribo, 
                    self.average, other.average, self.detected, other.detected,
                    mean.AUC,
                    everything())
    writeData(wb, sheet_name, res)
  }
}

saveWorkbook(wb, file_xlsx, TRUE)</code></pre>
<p>The tables of results can be downloaded using this <a
href="./file/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/scoreMarkers_lung.xlsx"><strong>link</strong></a></p>
<div id="dc" class="section level3" number="6.1">
<h3><span class="header-section-number">6.1</span> DC</h3>
<p>In the plot below, color represent the Z-score based on average
expression of the gene in each group. Z score is limited to -2 to 2
range. Point size represent the proportion of cells with detected
expression.</p>
<pre class="r"><code>use_celltype_label &lt;- &#39;DC&#39;
use_rows &lt;- mk_list[[use_celltype_label]]
use_cols &lt;- (sce_comb[[celltype_label]] == use_celltype_label &amp; !is.na(sce_comb[[celltype_label]]))
use_sce &lt;- sce_comb[use_rows, use_cols]
rownames(use_sce) &lt;- rowData(use_sce)$gene_name
res_plot &lt;- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group=&#39;origin&#39;,
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data &lt;- ggplot_build(res_plot)$plot$data %&gt;% 
  mutate(Group = factor(Group, levels = rev(c(&#39;normal&#39;, &#39;tme&#39;))))

# Create color scale
group &lt;- retrieveCellInfo(use_sce, &#39;origin&#39;, search=&quot;colData&quot;)$value
ids &lt;- DataFrame(group=group)
summarized &lt;- summarizeAssayByGroup(
        assay(use_sce, &quot;logcounts&quot;)[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c(&quot;mean&quot;, &quot;prop.detected&quot;),
        threshold=0)
    
ave &lt;- assay(summarized, &quot;mean&quot;)
use_heatmap_scale &lt;- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot &lt;- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = &#39;black&#39;) +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = &quot;black&quot;, fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = &#39;&#39;,
    y = &#39;&#39;,
    color = &#39;Z-score&#39;,
    size = &#39;Proportion\nof cells&#39;
  )

res_plot + theme( legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/lung-origin-celltype-markers-dc-dotplot-1.png" width="4872" style="display: block; margin: auto;" /></p>
</div>
<div id="endothelial-cells" class="section level3" number="6.2">
<h3><span class="header-section-number">6.2</span> Endothelial
cells</h3>
<p>In the plot below, color represent the Z-score based on average
expression of the gene in each group. Z score is limited to -2 to 2
range. Point size represent the proportion of cells with detected
expression.</p>
<pre class="r"><code>use_celltype_label &lt;- &#39;Endothelial cells&#39;
use_rows &lt;- mk_list[[use_celltype_label]]
use_cols &lt;- (sce_comb[[celltype_label]] == use_celltype_label &amp; !is.na(sce_comb[[celltype_label]]))
use_sce &lt;- sce_comb[use_rows, use_cols]
rownames(use_sce) &lt;- rowData(use_sce)$gene_name
res_plot &lt;- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group=&#39;origin&#39;,
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data &lt;- ggplot_build(res_plot)$plot$data %&gt;% 
  mutate(Group = factor(Group, levels = rev(c(&#39;normal&#39;, &#39;tme&#39;))))

# Create color scale
group &lt;- retrieveCellInfo(use_sce, &#39;origin&#39;, search=&quot;colData&quot;)$value
ids &lt;- DataFrame(group=group)
summarized &lt;- summarizeAssayByGroup(
        assay(use_sce, &quot;logcounts&quot;)[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c(&quot;mean&quot;, &quot;prop.detected&quot;),
        threshold=0)
    
ave &lt;- assay(summarized, &quot;mean&quot;)
use_heatmap_scale &lt;- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot &lt;- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = &#39;black&#39;) +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = &quot;black&quot;, fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = &#39;&#39;,
    y = &#39;&#39;,
    color = &#39;Z-score&#39;,
    size = &#39;Proportion\nof cells&#39;
  )

res_plot + theme( legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/lung-origin-celltype-markers-endothelial-dotplot-1.png" width="4872" style="display: block; margin: auto;" /></p>
</div>
<div id="fibroblasts" class="section level3" number="6.3">
<h3><span class="header-section-number">6.3</span> Fibroblasts</h3>
<p>In the plot below, color represent the Z-score based on average
expression of the gene in each group. Z score is limited to -2 to 2
range. Point size represent the proportion of cells with detected
expression.</p>
<pre class="r"><code>use_celltype_label &lt;- &#39;Fibroblasts&#39;
use_rows &lt;- mk_list[[use_celltype_label]]
use_cols &lt;- (sce_comb[[celltype_label]] == use_celltype_label &amp; !is.na(sce_comb[[celltype_label]]))
use_sce &lt;- sce_comb[use_rows, use_cols]
rownames(use_sce) &lt;- rowData(use_sce)$gene_name
res_plot &lt;- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group=&#39;origin&#39;,
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data &lt;- ggplot_build(res_plot)$plot$data %&gt;% 
  mutate(Group = factor(Group, levels = rev(c(&#39;normal&#39;, &#39;tme&#39;))))

# Create color scale
group &lt;- retrieveCellInfo(use_sce, &#39;origin&#39;, search=&quot;colData&quot;)$value
ids &lt;- DataFrame(group=group)
summarized &lt;- summarizeAssayByGroup(
        assay(use_sce, &quot;logcounts&quot;)[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c(&quot;mean&quot;, &quot;prop.detected&quot;),
        threshold=0)
    
ave &lt;- assay(summarized, &quot;mean&quot;)
use_heatmap_scale &lt;- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot &lt;- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = &#39;black&#39;) +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = &quot;black&quot;, fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = &#39;&#39;,
    y = &#39;&#39;,
    color = &#39;Z-score&#39;,
    size = &#39;Proportion\nof cells&#39;
  )

res_plot + theme( legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/lung-origin-celltype-markers-fibroblasts-dotplot-1.png" width="4872" style="display: block; margin: auto;" /></p>
</div>
<div id="macrophages" class="section level3" number="6.4">
<h3><span class="header-section-number">6.4</span> Macrophages</h3>
<p>In the plot below, color represent the Z-score based on average
expression of the gene in each group. Z score is limited to -2 to 2
range. Point size represent the proportion of cells with detected
expression.</p>
<pre class="r"><code>use_celltype_label &lt;- &#39;Macrophages&#39;
use_rows &lt;- mk_list[[use_celltype_label]]
use_cols &lt;- (sce_comb[[celltype_label]] == use_celltype_label &amp; !is.na(sce_comb[[celltype_label]]))
use_sce &lt;- sce_comb[use_rows, use_cols]
rownames(use_sce) &lt;- rowData(use_sce)$gene_name
res_plot &lt;- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group=&#39;origin&#39;,
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data &lt;- ggplot_build(res_plot)$plot$data %&gt;% 
  mutate(Group = factor(Group, levels = rev(c(&#39;normal&#39;, &#39;tme&#39;))))

# Create color scale
group &lt;- retrieveCellInfo(use_sce, &#39;origin&#39;, search=&quot;colData&quot;)$value
ids &lt;- DataFrame(group=group)
summarized &lt;- summarizeAssayByGroup(
        assay(use_sce, &quot;logcounts&quot;)[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c(&quot;mean&quot;, &quot;prop.detected&quot;),
        threshold=0)
    
ave &lt;- assay(summarized, &quot;mean&quot;)
use_heatmap_scale &lt;- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot &lt;- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = &#39;black&#39;) +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = &quot;black&quot;, fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = &#39;&#39;,
    y = &#39;&#39;,
    color = &#39;Z-score&#39;,
    size = &#39;Proportion\nof cells&#39;
  )

res_plot + theme( legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/lung-origin-celltype-markers-macrophages-dotplot-1.png" width="4872" style="display: block; margin: auto;" /></p>
</div>
<div id="monocytes" class="section level3" number="6.5">
<h3><span class="header-section-number">6.5</span> Monocytes</h3>
<p>In the plot below, color represent the Z-score based on average
expression of the gene in each group. Z score is limited to -2 to 2
range. Point size represent the proportion of cells with detected
expression.</p>
<pre class="r"><code>use_celltype_label &lt;- &#39;Monocytes&#39;
use_rows &lt;- mk_list[[use_celltype_label]]
use_cols &lt;- (sce_comb[[celltype_label]] == use_celltype_label &amp; !is.na(sce_comb[[celltype_label]]))
use_sce &lt;- sce_comb[use_rows, use_cols]
rownames(use_sce) &lt;- rowData(use_sce)$gene_name
res_plot &lt;- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group=&#39;origin&#39;,
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data &lt;- ggplot_build(res_plot)$plot$data %&gt;% 
  mutate(Group = factor(Group, levels = rev(c(&#39;normal&#39;, &#39;tme&#39;))))

# Create color scale
group &lt;- retrieveCellInfo(use_sce, &#39;origin&#39;, search=&quot;colData&quot;)$value
ids &lt;- DataFrame(group=group)
summarized &lt;- summarizeAssayByGroup(
        assay(use_sce, &quot;logcounts&quot;)[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c(&quot;mean&quot;, &quot;prop.detected&quot;),
        threshold=0)
    
ave &lt;- assay(summarized, &quot;mean&quot;)
use_heatmap_scale &lt;- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot &lt;- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = &#39;black&#39;) +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = &quot;black&quot;, fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = &#39;&#39;,
    y = &#39;&#39;,
    color = &#39;Z-score&#39;,
    size = &#39;Proportion\nof cells&#39;
  )

res_plot + theme( legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/lung-origin-celltype-markers-monocytes-dotplot-1.png" width="4872" style="display: block; margin: auto;" /></p>
</div>
<div id="neutrophils" class="section level3" number="6.6">
<h3><span class="header-section-number">6.6</span> Neutrophils</h3>
<p>In the plot below, color represent the Z-score based on average
expression of the gene in each group. Z score is limited to -2 to 2
range. Point size represent the proportion of cells with detected
expression.</p>
<pre class="r"><code>use_celltype_label &lt;- &#39;Neutrophils&#39;
use_rows &lt;- mk_list[[use_celltype_label]]
use_cols &lt;- (sce_comb[[celltype_label]] == use_celltype_label &amp; !is.na(sce_comb[[celltype_label]]))
use_sce &lt;- sce_comb[use_rows, use_cols]
rownames(use_sce) &lt;- rowData(use_sce)$gene_name
res_plot &lt;- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group=&#39;origin&#39;,
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data &lt;- ggplot_build(res_plot)$plot$data %&gt;% 
  mutate(Group = factor(Group, levels = rev(c(&#39;normal&#39;, &#39;tme&#39;))))

# Create color scale
group &lt;- retrieveCellInfo(use_sce, &#39;origin&#39;, search=&quot;colData&quot;)$value
ids &lt;- DataFrame(group=group)
summarized &lt;- summarizeAssayByGroup(
        assay(use_sce, &quot;logcounts&quot;)[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c(&quot;mean&quot;, &quot;prop.detected&quot;),
        threshold=0)
    
ave &lt;- assay(summarized, &quot;mean&quot;)
use_heatmap_scale &lt;- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot &lt;- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = &#39;black&#39;) +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = &quot;black&quot;, fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = &#39;&#39;,
    y = &#39;&#39;,
    color = &#39;Z-score&#39;,
    size = &#39;Proportion\nof cells&#39;
  )

res_plot + theme( legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/lung-origin-celltype-markers-neutrophils-dotplot-1.png" width="4872" style="display: block; margin: auto;" /></p>
</div>
<div id="legend" class="section level3" number="6.7">
<h3><span class="header-section-number">6.7</span> Legend</h3>
<pre class="r"><code>legend &lt;- cowplot::get_legend(res_plot)
grid.newpage()
grid.draw(legend)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/lung-origin-celltype-markers-legend-dotplot-1.png" width="1872" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="liver-tmenormal-markers-for-each-cell-type"
class="section level2" number="7">
<h2><span class="header-section-number">7</span> Liver: TME/Normal
markers for each cell type</h2>
<p>For each cell type we tried to identify potential markers to
distinguish between Normal and TME cells</p>
<pre class="r"><code>use_site &lt;- &#39;liver&#39;
n_top_mk &lt;- 20
celltype_label &lt;- &#39;celltype_immgen_pruned.labels&#39;
# names(score_markers$origin_site_celltype_markers[[use_site]])
j &lt;- score_markers$origin_site_celltype_markers[[use_site]][[1]][[1]]
mk_list &lt;- foreach(i = score_markers$origin_site_celltype_markers[[use_site]]) %do% {
  foreach(j = i, .combine = c) %do% {
    j %&gt;% data.frame %&gt;% arrange(desc(mean.AUC)) %&gt;% head(n_top_mk) %&gt;% rownames()
  } %&gt;% unique
}
names(mk_list) &lt;-names(score_markers$origin_site_celltype_markers[[use_site]])



# Save scoreMarkers to Excel
# File name summary
rmd_file &lt;- current_input()
if(is.null(rmd_file))
  rmd_file &lt;- &#39;tmp&#39;
# file_xlsx_list &lt;- list()

# Generate file
file_xlsx &lt;- file.path(
  &#39;./docs/file&#39;, rmd_file, 
  paste0(&#39;scoreMarkers_&#39;, use_site, &#39;.xlsx&#39;)
  )
dir.create(dirname(file_xlsx), recursive = TRUE, showWarnings = FALSE)

# Generate workbook
wb &lt;- createWorkbook()

# Create DfF of results
i &lt;- score_markers$origin_site_celltype_markers[[use_site]][[1]]
use_mk_list &lt;- score_markers$origin_site_celltype_markers[[use_site]]
x &lt;- foreach(use_ct = names(use_mk_list)) %do% {
  foreach(use_condition = names(use_mk_list[[use_ct]])) %do% {
    sheet_name &lt;- paste(use_ct, use_condition)
    addWorksheet(wb, paste(use_ct, use_condition))
    res &lt;- use_mk_list[[use_ct]][[use_condition]] %&gt;% data.frame %&gt;% 
      arrange(desc(mean.AUC)) %&gt;% 
      dplyr::select(gene_name, is.mito, is.ribo, 
                    self.average, other.average, self.detected, other.detected,
                    mean.AUC,
                    everything())
    writeData(wb, sheet_name, res)
  }
}

saveWorkbook(wb, file_xlsx, TRUE)</code></pre>
<p>The tables of results can be downloaded using this <a
href="./file/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/scoreMarkers_liver.xlsx"><strong>link</strong></a></p>
<div id="dc-1" class="section level3" number="7.1">
<h3><span class="header-section-number">7.1</span> DC</h3>
<p>In the plot below, color represent the Z-score based on average
expression of the gene in each group. Z score is limited to -2 to 2
range. Point size represent the proportion of cells with detected
expression.</p>
<pre class="r"><code>use_celltype_label &lt;- &#39;DC&#39;
use_rows &lt;- mk_list[[use_celltype_label]]
use_cols &lt;- (sce_comb[[celltype_label]] == use_celltype_label &amp; !is.na(sce_comb[[celltype_label]]))
use_sce &lt;- sce_comb[use_rows, use_cols]
rownames(use_sce) &lt;- rowData(use_sce)$gene_name
res_plot &lt;- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group=&#39;origin&#39;,
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data &lt;- ggplot_build(res_plot)$plot$data %&gt;% 
  mutate(Group = factor(Group, levels = rev(c(&#39;normal&#39;, &#39;tme&#39;))))

# Create color scale
group &lt;- retrieveCellInfo(use_sce, &#39;origin&#39;, search=&quot;colData&quot;)$value
ids &lt;- DataFrame(group=group)
summarized &lt;- summarizeAssayByGroup(
        assay(use_sce, &quot;logcounts&quot;)[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c(&quot;mean&quot;, &quot;prop.detected&quot;),
        threshold=0)
    
ave &lt;- assay(summarized, &quot;mean&quot;)
use_heatmap_scale &lt;- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot &lt;- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = &#39;black&#39;) +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = &quot;black&quot;, fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = &#39;&#39;,
    y = &#39;&#39;,
    color = &#39;Z-score&#39;,
    size = &#39;Proportion\nof cells&#39;
  )

res_plot + theme( legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/liver-origin-celltype-markers-dc-dotplot-1.png" width="4872" style="display: block; margin: auto;" /></p>
</div>
<div id="endothelial-cells-1" class="section level3" number="7.2">
<h3><span class="header-section-number">7.2</span> Endothelial
cells</h3>
<p>In the plot below, color represent the Z-score based on average
expression of the gene in each group. Z score is limited to -2 to 2
range. Point size represent the proportion of cells with detected
expression.</p>
<pre class="r"><code>use_celltype_label &lt;- &#39;Endothelial cells&#39;
use_rows &lt;- mk_list[[use_celltype_label]]
use_cols &lt;- (sce_comb[[celltype_label]] == use_celltype_label &amp; !is.na(sce_comb[[celltype_label]]))
use_sce &lt;- sce_comb[use_rows, use_cols]
rownames(use_sce) &lt;- rowData(use_sce)$gene_name
res_plot &lt;- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group=&#39;origin&#39;,
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data &lt;- ggplot_build(res_plot)$plot$data %&gt;% 
  mutate(Group = factor(Group, levels = rev(c(&#39;normal&#39;, &#39;tme&#39;))))

# Create color scale
group &lt;- retrieveCellInfo(use_sce, &#39;origin&#39;, search=&quot;colData&quot;)$value
ids &lt;- DataFrame(group=group)
summarized &lt;- summarizeAssayByGroup(
        assay(use_sce, &quot;logcounts&quot;)[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c(&quot;mean&quot;, &quot;prop.detected&quot;),
        threshold=0)
    
ave &lt;- assay(summarized, &quot;mean&quot;)
use_heatmap_scale &lt;- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot &lt;- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = &#39;black&#39;) +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = &quot;black&quot;, fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = &#39;&#39;,
    y = &#39;&#39;,
    color = &#39;Z-score&#39;,
    size = &#39;Proportion\nof cells&#39;
  )

res_plot + theme( legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/liver-origin-celltype-markers-endothelial-dotplot-1.png" width="4872" style="display: block; margin: auto;" /></p>
</div>
<div id="macrophages-1" class="section level3" number="7.3">
<h3><span class="header-section-number">7.3</span> Macrophages</h3>
<p>In the plot below, color represent the Z-score based on average
expression of the gene in each group. Z score is limited to -2 to 2
range. Point size represent the proportion of cells with detected
expression.</p>
<pre class="r"><code>use_celltype_label &lt;- &#39;Macrophages&#39;
use_rows &lt;- mk_list[[use_celltype_label]]
use_cols &lt;- (sce_comb[[celltype_label]] == use_celltype_label &amp; !is.na(sce_comb[[celltype_label]]))
use_sce &lt;- sce_comb[use_rows, use_cols]
rownames(use_sce) &lt;- rowData(use_sce)$gene_name
res_plot &lt;- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group=&#39;origin&#39;,
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data &lt;- ggplot_build(res_plot)$plot$data %&gt;% 
  mutate(Group = factor(Group, levels = rev(c(&#39;normal&#39;, &#39;tme&#39;))))

# Create color scale
group &lt;- retrieveCellInfo(use_sce, &#39;origin&#39;, search=&quot;colData&quot;)$value
ids &lt;- DataFrame(group=group)
summarized &lt;- summarizeAssayByGroup(
        assay(use_sce, &quot;logcounts&quot;)[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c(&quot;mean&quot;, &quot;prop.detected&quot;),
        threshold=0)
    
ave &lt;- assay(summarized, &quot;mean&quot;)
use_heatmap_scale &lt;- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot &lt;- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = &#39;black&#39;) +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = &quot;black&quot;, fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = &#39;&#39;,
    y = &#39;&#39;,
    color = &#39;Z-score&#39;,
    size = &#39;Proportion\nof cells&#39;
  )

res_plot + theme( legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/liver-origin-celltype-markers-macrophages-dotplot-1.png" width="4872" style="display: block; margin: auto;" /></p>
</div>
<div id="monocytes-1" class="section level3" number="7.4">
<h3><span class="header-section-number">7.4</span> Monocytes</h3>
<p>In the plot below, color represent the Z-score based on average
expression of the gene in each group. Z score is limited to -2 to 2
range. Point size represent the proportion of cells with detected
expression.</p>
<pre class="r"><code>use_celltype_label &lt;- &#39;Monocytes&#39;
use_rows &lt;- mk_list[[use_celltype_label]]
use_cols &lt;- (sce_comb[[celltype_label]] == use_celltype_label &amp; !is.na(sce_comb[[celltype_label]]))
use_sce &lt;- sce_comb[use_rows, use_cols]
rownames(use_sce) &lt;- rowData(use_sce)$gene_name
res_plot &lt;- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group=&#39;origin&#39;,
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data &lt;- ggplot_build(res_plot)$plot$data %&gt;% 
  mutate(Group = factor(Group, levels = rev(c(&#39;normal&#39;, &#39;tme&#39;))))

# Create color scale
group &lt;- retrieveCellInfo(use_sce, &#39;origin&#39;, search=&quot;colData&quot;)$value
ids &lt;- DataFrame(group=group)
summarized &lt;- summarizeAssayByGroup(
        assay(use_sce, &quot;logcounts&quot;)[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c(&quot;mean&quot;, &quot;prop.detected&quot;),
        threshold=0)
    
ave &lt;- assay(summarized, &quot;mean&quot;)
use_heatmap_scale &lt;- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot &lt;- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = &#39;black&#39;) +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = &quot;black&quot;, fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = &#39;&#39;,
    y = &#39;&#39;,
    color = &#39;Z-score&#39;,
    size = &#39;Proportion\nof cells&#39;
  )

res_plot + theme( legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/liver-origin-celltype-markers-monocytes-dotplot-1.png" width="4872" style="display: block; margin: auto;" /></p>
</div>
<div id="neutrophils-1" class="section level3" number="7.5">
<h3><span class="header-section-number">7.5</span> Neutrophils</h3>
<p>In the plot below, color represent the Z-score based on average
expression of the gene in each group. Z score is limited to -2 to 2
range. Point size represent the proportion of cells with detected
expression.</p>
<pre class="r"><code>use_celltype_label &lt;- &#39;Neutrophils&#39;
use_rows &lt;- mk_list[[use_celltype_label]]
use_cols &lt;- (sce_comb[[celltype_label]] == use_celltype_label &amp; !is.na(sce_comb[[celltype_label]]))
use_sce &lt;- sce_comb[use_rows, use_cols]
rownames(use_sce) &lt;- rowData(use_sce)$gene_name
res_plot &lt;- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group=&#39;origin&#39;,
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data &lt;- ggplot_build(res_plot)$plot$data %&gt;% 
  mutate(Group = factor(Group, levels = rev(c(&#39;normal&#39;, &#39;tme&#39;))))

# Create color scale
group &lt;- retrieveCellInfo(use_sce, &#39;origin&#39;, search=&quot;colData&quot;)$value
ids &lt;- DataFrame(group=group)
summarized &lt;- summarizeAssayByGroup(
        assay(use_sce, &quot;logcounts&quot;)[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c(&quot;mean&quot;, &quot;prop.detected&quot;),
        threshold=0)
    
ave &lt;- assay(summarized, &quot;mean&quot;)
use_heatmap_scale &lt;- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot &lt;- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = &#39;black&#39;) +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = &quot;black&quot;, fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = &#39;&#39;,
    y = &#39;&#39;,
    color = &#39;Z-score&#39;,
    size = &#39;Proportion\nof cells&#39;
  )

res_plot + theme( legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/liver-origin-celltype-markers-neutrophils-dotplot-1.png" width="4872" style="display: block; margin: auto;" /></p>
</div>
<div id="legend-1" class="section level3" number="7.6">
<h3><span class="header-section-number">7.6</span> Legend</h3>
<pre class="r"><code>legend &lt;- cowplot::get_legend(res_plot)
grid.newpage()
grid.draw(legend)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/liver-origin-celltype-markers-legend-dotplot-1.png" width="1872" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="bmp2-gene-expression-in-endothelial-liver-cells"
class="section level2" number="8">
<h2><span class="header-section-number">8</span> Bmp2 gene expression in
endothelial liver cells</h2>
<pre class="r"><code>use_data &lt;- cell_annot %&gt;% 
  mutate(Bmp2 =  logcounts(sce_comb)[&#39;Bmp2&#39;,]) %&gt;% 
  filter(grepl(&#39;Liver&#39;, site_caps)) %&gt;% 
  filter(celltype_immgen_pruned.labels == &#39;Endothelial cells&#39;) </code></pre>
<div id="violin-plot" class="section level3" number="8.1">
<h3><span class="header-section-number">8.1</span> Violin plot</h3>
<p>Black horizontal line represents the median expression.</p>
<pre class="r"><code>summary_stats &lt;- use_data %&gt;% 
  group_by(site_caps) %&gt;% 
  summarise(median = median(Bmp2)) %&gt;% 
  ungroup %&gt;% 
  mutate(site_caps = gsub(&#39;.* &#39;, &#39;&#39;, site_caps)) %&gt;% 
  column_to_rownames(&#39;site_caps&#39;)

use_data %&gt;% 
  ggplot(aes(x = site_caps, y = Bmp2, fill = site_caps, color = site_caps)) +
  geom_violin(color = &quot;gray60&quot;, alpha = 0.2, scale = &quot;width&quot;, width = 0.8, show.legend = FALSE) + 
  geom_quasirandom(size = 0.2, alpha = 0.6, width=0.4, 
                   pch = 16, 
                   groupOnX=TRUE, bandwidth=1, show.legend = FALSE) +
  stat_compare_means(show.legend = FALSE, size = geom_text_size) +
  stat_summary(fun = median, fun.min = median, fun.max = median, 
               geom = &quot;crossbar&quot;, width = 0.3, alpha = 1, color = &#39;black&#39;, 
               show.legend = FALSE) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  labs(
    x = NULL,
    y = &#39;Bmp2 expression&#39;,
    fill = NULL,
    color = NULL,
    caption = paste0(
      &#39;Median expression in Normal: &#39;, round(summary_stats[&#39;Normal&#39;, &#39;median&#39;], 3), &#39;\n&#39;,
      &#39;Median expression in TME: &#39;, round(summary_stats[&#39;TME&#39;, &#39;median&#39;], 3)
    )
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/bmp2-expr-violin-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="density-plot" class="section level3" number="8.2">
<h3><span class="header-section-number">8.2</span> Density plot</h3>
<p>Density plot that computes and draws kernel density estimate, which
is a smoothed version of the histogram.</p>
<pre class="r"><code>summary_stats &lt;- use_data %&gt;% 
  group_by(site_caps) %&gt;% 
  summarise(median = median(Bmp2)) %&gt;% 
  ungroup %&gt;% 
  mutate(site_caps = gsub(&#39;.* &#39;, &#39;&#39;, site_caps)) %&gt;% 
  column_to_rownames(&#39;site_caps&#39;)

use_data %&gt;% 
  ggplot(aes(Bmp2, fill = site_caps, color = site_caps)) +
  geom_density(alpha = 0.5) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  scale_x_continuous(expand = expansion(mult = c(0, 0))) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.01))) +
  labs(
    y = &#39;Density&#39;,
    x = &#39;Bmp2 expression&#39;,
    fill = NULL,
    color = NULL,
    caption = paste0(
      &#39;Median expression in Normal: &#39;, round(summary_stats[&#39;Normal&#39;, &#39;median&#39;], 3), &#39;\n&#39;,
      &#39;Median expression in TME: &#39;, round(summary_stats[&#39;TME&#39;, &#39;median&#39;], 3)
    )
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/bmp2-expr-density-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="violin-plot-by-mouse" class="section level3" number="8.3">
<h3><span class="header-section-number">8.3</span> Violin plot by
mouse</h3>
<p>Black horizontal line represents the median expression.</p>
<pre class="r"><code>summary_stats &lt;- use_data %&gt;% 
  group_by(site_caps) %&gt;% 
  summarise(median = median(Bmp2))

use_data %&gt;% 
  ggplot(aes(x = mouse_id, y = Bmp2, fill = site_caps, color = site_caps)) +
  geom_violin(color = &quot;gray60&quot;, alpha = 0.2, scale = &quot;width&quot;, width = 0.8, show.legend = FALSE) + 
  geom_quasirandom(size = 0.1, alpha = 0.6, width=0.4, 
                   pch = 16,
                   groupOnX=TRUE, bandwidth=1, show.legend = FALSE) +
  facet_grid(cols = vars(site_caps), scales = &#39;free_x&#39;) +
  stat_summary(fun = median, fun.min = median, fun.max = median, 
               geom = &quot;crossbar&quot;, width = 0.3, alpha = 1, color = &#39;black&#39;, 
               show.legend = FALSE) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  theme(
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank()
  ) +
  theme_facet +
  labs(
    x = NULL,
    y = &#39;Bmp2 expression&#39;,
    fill = NULL,
    color = NULL
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/bmp2-expr-violin-mouse-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="stripchart-boxplot-2" class="section level3" number="8.4">
<h3><span class="header-section-number">8.4</span>
Stripchart-Boxplot</h3>
<pre class="r"><code>use_data %&gt;% 
  group_by(site_caps, mouse_id) %&gt;%
  summarise(proportion = sum(Bmp2 &gt; 0)/n()) %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, color = site_caps)) +
  geom_boxplot(alpha = 0.4, color = &#39;black&#39;, aes(fill = site_caps), 
               show.legend = FALSE, size = 0.25, fatten = 1) +
  geom_quasirandom(method=&quot;smiley&quot;, width = 0.1, alpha = 1, size = 1, 
                   shape = 21, color = &#39;black&#39;, aes(fill = site_caps), stroke = 0.4) +
  stat_compare_means(size = geom_text_size) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  scale_y_continuous(expand = c(0.02, 0, 0.08, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  theme(
    # axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    panel.border = element_rect(colour = &quot;black&quot;, size=0.5),
    strip.background = element_blank()
  ) +
  guides( color = &#39;none&#39;, fill = &#39;none&#39;) +
  labs(
    fill = NULL,
    x = NULL,
    y = &#39;Proportion of cells with Bmp2 &gt; 0 expression&#39;
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/bmp2-expr-stripchart-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="granulin-gene-expression-in-lung-macrophages"
class="section level2" number="9">
<h2><span class="header-section-number">9</span> Granulin gene
expression in lung macrophages</h2>
<pre class="r"><code>use_data &lt;- cell_annot %&gt;% 
  mutate(Grn =  logcounts(sce_comb)[&#39;Grn&#39;,]) %&gt;% 
  filter(grepl(&#39;Lung&#39;, site_caps)) %&gt;% 
  filter(celltype_immgen_pruned.labels == &#39;Macrophages&#39;) </code></pre>
<div id="violin-plot-1" class="section level3" number="9.1">
<h3><span class="header-section-number">9.1</span> Violin plot</h3>
<p>Black horizontal line represents the median expression.</p>
<pre class="r"><code>summary_stats &lt;- use_data %&gt;% 
  group_by(site_caps) %&gt;% 
  summarise(median = median(Grn)) %&gt;% 
  ungroup %&gt;% 
  mutate(site_caps = gsub(&#39;.* &#39;, &#39;&#39;, site_caps)) %&gt;% 
  column_to_rownames(&#39;site_caps&#39;)

use_data %&gt;% 
  ggplot(aes(x = site_caps, y = Grn, fill = site_caps, color = site_caps)) +
  geom_violin(color = &quot;gray60&quot;, alpha = 0.2, scale = &quot;width&quot;, width = 0.8, show.legend = FALSE) + 
  geom_quasirandom(alpha = 0.6, width=0.4, 
                   pch = 16,
                   groupOnX=TRUE, bandwidth=1, show.legend = FALSE) +
  stat_compare_means(show.legend = FALSE, size = geom_text_size) +
  stat_summary(fun = median, fun.min = median, fun.max = median, 
               geom = &quot;crossbar&quot;, width = 0.3, alpha = 1, color = &#39;black&#39;, 
               show.legend = FALSE) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  labs(
    x = NULL,
    y = &#39;Grn expression&#39;,
    fill = NULL,
    color = NULL,
    caption = paste0(
      &#39;Median expression in Normal: &#39;, round(summary_stats[&#39;Normal&#39;, &#39;median&#39;], 3), &#39;\n&#39;,
      &#39;Median expression in TME: &#39;, round(summary_stats[&#39;TME&#39;, &#39;median&#39;], 3)
    )
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/grn-expr-violin-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="density-plot-1" class="section level3" number="9.2">
<h3><span class="header-section-number">9.2</span> Density plot</h3>
<p>Density plot that computes and draws kernel density estimate, which
is a smoothed version of the histogram.</p>
<pre class="r"><code>summary_stats &lt;- use_data %&gt;% 
  group_by(site_caps) %&gt;% 
  summarise(median = median(Grn)) %&gt;% 
  ungroup %&gt;% 
  mutate(site_caps = gsub(&#39;.* &#39;, &#39;&#39;, site_caps)) %&gt;% 
  column_to_rownames(&#39;site_caps&#39;)

use_data %&gt;% 
  ggplot(aes(Grn, fill = site_caps, color = site_caps)) +
  geom_density(alpha = 0.5) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  scale_x_continuous(expand = expansion(mult = c(0, 0))) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.01))) +
  labs(
    y = &#39;Density&#39;,
    x = &#39;Grn expression&#39;,
    fill = NULL,
    color = NULL,
    caption = paste0(
      &#39;Median expression in Normal: &#39;, round(summary_stats[&#39;Normal&#39;, &#39;median&#39;], 3), &#39;\n&#39;,
      &#39;Median expression in TME: &#39;, round(summary_stats[&#39;TME&#39;, &#39;median&#39;], 3)
    )
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/grn-expr-density-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="violin-plot-by-mouse-1" class="section level3" number="9.3">
<h3><span class="header-section-number">9.3</span> Violin plot by
mouse</h3>
<p>Black horizontal line represents the median expression.</p>
<pre class="r"><code>summary_stats &lt;- use_data %&gt;% 
  group_by(site_caps) %&gt;% 
  summarise(median = median(Grn))

use_data %&gt;% 
  ggplot(aes(x = mouse_id, y = Grn, fill = site_caps, color = site_caps)) +
  geom_violin(color = &quot;gray60&quot;, alpha = 0.2, scale = &quot;width&quot;, width = 0.8, show.legend = FALSE) + 
  geom_quasirandom(alpha = 0.6, width=0.4, 
                   pch = 16,
                   groupOnX=TRUE, bandwidth=1, 
                   show.legend = FALSE) +
  facet_grid(cols = vars(site_caps), scales = &#39;free_x&#39;) +
  stat_summary(fun = median, fun.min = median, fun.max = median, 
               geom = &quot;crossbar&quot;, width = 0.3, alpha = 1, color = &#39;black&#39;, 
               show.legend = FALSE) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  theme(
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank()
  ) +
  theme_facet +
  labs(
    x = NULL,
    y = &#39;Grn expression&#39;,
    fill = NULL,
    color = NULL
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/grn-expr-violin-mouse-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="stripchart-boxplot-3" class="section level3" number="9.4">
<h3><span class="header-section-number">9.4</span>
Stripchart-Boxplot</h3>
<pre class="r"><code>use_data %&gt;% 
  group_by(site_caps, mouse_id) %&gt;%
  summarise(proportion = sum(Grn &gt; 0)/n()) %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, color = site_caps)) +
  geom_boxplot(alpha = 0.4, color = &#39;black&#39;, aes(fill = site_caps), 
               show.legend = FALSE, size = 0.25, fatten = 1) +
  geom_quasirandom(method=&quot;smiley&quot;, width = 0.1, alpha = 1, size = 1, 
                   shape = 21, color = &#39;black&#39;, aes(fill = site_caps), stroke = 0.4) +
  stat_compare_means(size = geom_text_size) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  scale_y_continuous(expand = c(0.02, 0, 0.08, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  theme(
    # axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    panel.border = element_rect(colour = &quot;black&quot;, size=0.5),
    strip.background = element_blank()
  ) +
  guides( color = &#39;none&#39;, fill = &#39;none&#39;) +
  labs(
    fill = NULL,
    x = NULL,
    y = &#39;Proportion of cells with Grn &gt; 0 expression&#39;
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/grn-expr-stripchart-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="lung-mf-differential-abundance-of-macrophages-cell-types"
class="section level2" number="10">
<h2><span class="header-section-number">10</span> Lung MF: Differential
abundance of macrophages cell types</h2>
<p>Differential abundance of cell types between normal lung tissue and
TME in lung metastasis</p>
<pre class="r"><code>ct_label &lt;- &#39;mf_immgen_pruned.labels.fine&#39;</code></pre>
<div id="statistical-analysis-2" class="section level3" number="10.1">
<h3><span class="header-section-number">10.1</span> Statistical
analysis</h3>
<p>Differential abundance using <code>speckle::propeller</code>
package</p>
<pre class="r"><code>use_data &lt;- cell_annot %&gt;% 
  filter(celltype_immgen_pruned.labels == &#39;Macrophages&#39;)

# Remove low abundant cell types 
keep_labels &lt;- table(use_data[[ct_label]]) %&gt;% data.frame %&gt;% filter(Freq &gt;= 100) %&gt;% pull(Var1)
keep_rows &lt;- use_data[[ct_label]] %in% keep_labels
use_data &lt;- use_data[keep_rows,]

# Run propeller Global
propeller_global &lt;- propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$site) %&gt;% 
  rownames_to_column(&#39;cell_type&#39;)


# Run pairwise propeller to obtaib P-values
site_levels &lt;- use_data$site_caps %&gt;% unique
# site_combn &lt;- combn(site_levels, 2, simplify = FALSE)
site_combn &lt;- list(
  c(&#39;Lung Normal&#39;, &#39;Lung TME&#39;)
  # c(&#39;Liver Normal&#39;, &#39;Liver TME&#39;)
)
sc &lt;-  site_combn[[1]]
propeller_pairwise &lt;-foreach(sc = site_combn, .combine = rbind) %do% {
  use_data &lt;- cell_annot %&gt;%
    filter(site_caps %in% sc) %&gt;%
    mutate(
      group = ifelse(site_caps == sc[1], &#39;group1&#39;, &#39;group2&#39;)
    )
  keep_rows &lt;- use_data[[ct_label]] %in% keep_labels
  use_data &lt;- use_data[keep_rows,]
  propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$group) %&gt;%
    mutate(
      group1 = sc[1],
      group2 = sc[2],
      FDR.sign = symnum(FDR, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001,0.001, 0.01, 0.05, 1), symbols = c(&quot;****&quot;, &quot;***&quot;, &quot;**&quot;, &quot;*&quot;, &quot;&quot;)),
      p.sign = symnum(P.Value, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c(&quot;****&quot;, &quot;***&quot;, &quot;**&quot;, &quot;*&quot;, &quot;&quot;))
    ) %&gt;%
    dplyr::rename(cell_type = BaselineProp.clusters)
}</code></pre>
<p>Table of results with pairwise comparison between sites for cluster
abundance.</p>
<pre class="r"><code>propeller_pairwise %&gt;%
  arrange(FDR) %&gt;% 
  datatable(., 
            rownames = FALSE, 
            filter = &#39;top&#39;, 
            caption = &#39;Pairwise comparison between cell types for cluster abundance using speckle::propeller&#39;,
            extensions = &#39;Buttons&#39;, 
            options = list(
              dom = &#39;Blfrtip&#39;,
              buttons = c(&#39;csv&#39;, &#39;excel&#39;)
            )) %&gt;% 
  formatPercentage(names(propeller_pairwise)[2:4], digits = 3) %&gt;% 
  formatRound(names(propeller_pairwise)[5:6], digits = 3, interval = -1000) %&gt;% 
  formatSignif(c(&#39;P.Value&#39;, &#39;FDR&#39;), digits = 3)</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-684d012f49e01d513276" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-684d012f49e01d513276">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Macrophages (MF.103-11B+24-)&quot;,&quot;Macrophages (MF.11C-11B+)&quot;,&quot;Macrophages (MF.11CLOSER)&quot;,&quot;Macrophages (MF.II+480LO)&quot;,&quot;Macrophages (MF.MEDL)&quot;,&quot;Macrophages (MFIO5.II-480INT)&quot;,&quot;Macrophages (MFIO5.II+480INT)&quot;,&quot;Macrophages (MFIO5.II+480LO)&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.028642149929278\" data-max=\"0.430693069306931\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"0.56952380952381\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.02677589037098\" data-max=\"0.44573878142697\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"5.12187849751182\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-5.98212966714159\" data-max=\"4.06886537144778\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"1.143057296e-06\" data-max=\"0.631311593130626\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"9.144458373e-06\" data-max=\"0.631311593130626\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Pairwise comparison between cell types for cluster abundance using speckle::propeller<\/caption>","data":[["Macrophages (MFIO5.II+480INT)","Macrophages (MF.103-11B+24-)","Macrophages (MFIO5.II+480LO)","Macrophages (MFIO5.II-480INT)","Macrophages (MF.11C-11B+)","Macrophages (MF.MEDL)","Macrophages (MF.II+480LO)","Macrophages (MF.11CLOSER)"],[0.4306930693069307,0.1538189533239038,0.02864214992927864,0.2107496463932108,0.03606789250353607,0.04632248939179632,0.04950495049504951,0.0442008486562942],[0.03523809523809525,0.5695238095238097,0.1371428571428571,0.0419047619047619,0.1523809523809524,-0,0.02857142857142857,0.03523809523809525],[0.4457387814269697,0.1382195629842683,0.02677589037098017,0.209669917610825,0.03281855539834385,0.04605168447562865,0.05267515048864956,0.04805045724433493],[0.07905548430245506,4.120428376615769,5.12187849751182,0.199860630376851,4.643134060332287,0,0.5424081052712919,0.7333560856436961],[-5.982129667141585,4.068865371447778,3.779423585476012,-3.26835629888558,3.261084436990424,-2.394503458699571,-0.9849792058098847,0.4845275621816084],[1.143057296713446e-06,0.0002885044137993959,0.0006480734635482543,0.002587013909449388,0.002637144008519078,0.0226715482240486,0.3320235495891293,0.6313115931306257],[9.144458373707566e-06,0.001154017655197583,0.001728195902795345,0.004219430413630525,0.004219430413630525,0.03022873096539813,0.3794554852447192,0.6313115931306257],["Lung Normal","Lung Normal","Lung Normal","Lung Normal","Lung Normal","Lung Normal","Lung Normal","Lung Normal"],["Lung TME","Lung TME","Lung TME","Lung TME","Lung TME","Lung TME","Lung TME","Lung TME"],["****","**","**","**","**","*","",""],["****","***","***","**","**","*","",""]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>cell_type<\/th>\n      <th>BaselineProp.Freq<\/th>\n      <th>PropMean.group1<\/th>\n      <th>PropMean.group2<\/th>\n      <th>PropRatio<\/th>\n      <th>Tstatistic<\/th>\n      <th>P.Value<\/th>\n      <th>FDR<\/th>\n      <th>group1<\/th>\n      <th>group2<\/th>\n      <th>FDR.sign<\/th>\n      <th>p.sign<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":7,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, -1000, \",\", \".\", null);\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, -1000, \",\", \".\", null);\n  }"},{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[1,2,3,4,5,6,7]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render","options.columnDefs.5.render","options.columnDefs.6.render"],"jsHooks":[]}</script>
</div>
<div id="distribution-by-site-2" class="section level3" number="10.2">
<h3><span class="header-section-number">10.2</span> Distribution by
site</h3>
<pre class="r"><code># Create data
use_data &lt;- cell_annot %&gt;% 
  filter(site_caps %in% unlist(site_combn)) %&gt;% 
  dplyr::rename(&#39;celltype_label&#39; = ct_label) %&gt;% 
  group_by(site_caps, celltype_label) %&gt;% 
  summarise(freq = n()) %&gt;% 
  dplyr::rename(cell_type = celltype_label) %&gt;% 
  na.omit()

# Remove low abundant cell types
cell_type_freq &lt;- use_data %&gt;% group_by(cell_type) %&gt;% summarise(totalfreq = sum(freq))
# site_filtered &lt;- use_data %&gt;% group_by(site_caps) %&gt;% summarise(freq = sum(freq)) %&gt;% filter(freq &gt;= 500)
use_data &lt;- use_data %&gt;% 
  # filter(site_caps %in% site_filtered$site_caps) %&gt;% 
  left_join(cell_type_freq) %&gt;% 
  filter(totalfreq &gt;= 50) %&gt;% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data &lt;- use_data %&gt;% 
  arrange(site_caps, cell_type)%&gt;% 
  group_by(site_caps) %&gt;%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion &lt; 0.03, &#39;&#39;, cell_type_label)
    
  )</code></pre>
<div id="barplot-stacked-2" class="section level4" number="10.2.1">
<h4><span class="header-section-number">10.2.1</span> Barplot
stacked</h4>
<pre class="r"><code>arranged_data %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, fill = cell_type)) +
  geom_col(alpha = 0.85, color = &#39;black&#39;, linewidth = 0.2) +
  # scale_fill_manual(values = celltype_palette) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  # geom_text(aes(y = label_y, label = cell_type_label), colour = &quot;white&quot;, size = geom_text_size) +
  theme(axis.ticks.x = element_blank()) +
  labs(
    fill = NULL, 
    x = NULL,
    y = &#39;Proportion of cells&#39;
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/mf_da_barplot_site_stacked_immgen-lung-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="barplot-facet-2" class="section level4" number="10.2.2">
<h4><span class="header-section-number">10.2.2</span> Barplot facet</h4>
<pre class="r"><code>arranged_data %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, fill = site_caps, group = cell_type)) +
  geom_col(alpha = 0.85, color = &#39;black&#39;, size = 0.2) +
  scale_fill_manual(values = site_type_palette) +
  scale_y_continuous(expand = c(0.02, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), ncol = 3, scales = &#39;free_y&#39;) +
  theme(
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x = element_blank(),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    panel.border = element_rect(colour = &quot;black&quot;, size=0.5),
    strip.background = element_blank()
  ) +
  labs(
    fill = NULL, 
    x = NULL,
    y = &#39;Proportion of cells&#39;
  ) #+</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/mf_da_barplot_site_celltypefacet_immgen-lung-1.png" width="3120" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # guides(fill = &#39;none&#39;)</code></pre>
</div>
<div id="piechart-2" class="section level4" number="10.2.3">
<h4><span class="header-section-number">10.2.3</span> Piechart</h4>
<pre class="r"><code>arranged_data %&gt;% 
  ggplot(aes(x = &quot;&quot;, y = proportion, fill = cell_type)) +
  # geom_bar(stat=&quot;identity&quot;, width=1, color=&quot;white&quot;) +
  geom_bar(stat=&quot;identity&quot;, width=1) +
  coord_polar(&quot;y&quot;, start=0) +
  # scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = &#39;&#39;, 
    x = NULL,
    y = NULL
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/mf_da_piechart_site_celltypefacet_immgen-lung-1.png" width="2400" style="display: block; margin: auto;" /></p>
</div>
<div id="doughnut-chart-2" class="section level4" number="10.2.4">
<h4><span class="header-section-number">10.2.4</span> Doughnut
chart</h4>
<pre class="r"><code>i &lt;- unique(arranged_data$site_caps)[1]
use_data &lt;- foreach(i = unique(arranged_data$site_caps), .combine = rbind) %do% {
  x &lt;- arranged_data %&gt;% 
    filter(site_caps == i) %&gt;% 
    arrange(cell_type)
  x$ymax &lt;- cumsum(x$proportion)
  x$ymin &lt;- c(0, head(x$ymax, n = -1))
  x
}

use_data %&gt;% 
  ggplot(aes(ymax=ymax, ymin=ymin, xmax=4, xmin=3, fill=cell_type)) +
  geom_rect() +
  coord_polar(theta=&quot;y&quot;) + # Try to remove that to understand how the chart is built initially
  xlim(c(2, 4)) + # Try to remove that to see how to make a pie chart
  # scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = &#39;&#39;, 
    x = NULL,
    y = NULL
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/mf_da_doughnut_site_celltypefacet_immgen-lung-1.png" width="2400" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="distribution-by-site-and-mouse-2" class="section level3"
number="10.3">
<h3><span class="header-section-number">10.3</span> Distribution by site
and mouse</h3>
<pre class="r"><code># Create data
use_data &lt;- cell_annot %&gt;% 
  filter(site_caps %in% unlist(site_combn)) %&gt;% 
  dplyr::rename(&#39;celltype_label&#39; = ct_label) %&gt;% 
  group_by(site_caps, mouse_id, celltype_label) %&gt;% 
  summarise(freq = n()) %&gt;% 
  dplyr::rename(cell_type = celltype_label) %&gt;% 
  na.omit()

# Remove low abundant cell types and sites
cell_type_freq &lt;- use_data %&gt;% group_by(cell_type) %&gt;% summarise(totalfreq = sum(freq))
# site_filtered &lt;- use_data %&gt;% group_by(site_caps) %&gt;% summarise(freq = sum(freq)) %&gt;% filter(freq &gt;= 500)
use_data &lt;- use_data %&gt;% 
  # filter(site_caps %in% site_filtered$site_caps) %&gt;% 
  left_join(cell_type_freq) %&gt;% 
  filter(totalfreq &gt;= 50) %&gt;% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data &lt;- use_data %&gt;% 
  arrange(site_caps, cell_type)%&gt;% 
  group_by(site_caps, mouse_id) %&gt;%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion &lt; 0.03, &#39;&#39;, cell_type_label)
  )</code></pre>
<div id="stripchart-boxplot-4" class="section level4" number="10.3.1">
<h4><span class="header-section-number">10.3.1</span>
Stripchart-Boxplot</h4>
<pre class="r"><code>max_prop_cell_type &lt;- arranged_data %&gt;% group_by(cell_type) %&gt;% summarise(max_proportion = max(proportion))
stat_test &lt;- propeller_pairwise %&gt;% 
  filter(group1 %in% arranged_data$site_caps &amp; group2 %in%  arranged_data$site_caps) %&gt;%
  left_join(max_prop_cell_type) %&gt;% 
  arrange(cell_type, group1, group2) %&gt;% 
  na.omit() %&gt;% 
  filter(FDR &lt; 0.05)
  
add_count_y &lt;- stat_test %&gt;% group_by(cell_type) %&gt;% reframe(group1 = group1, group2 = group2, n = 1:n())
stat_test &lt;- stat_test %&gt;% left_join(add_count_y) %&gt;% mutate(y.position = max_proportion + (n*max_proportion/10))


arranged_data %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, color = site_caps)) +
  geom_boxplot(alpha = 0.4, color = &#39;black&#39;, aes(fill = site_caps), 
               show.legend = FALSE, size = 0.25, fatten = 1) +
  # geom_point(alpha = 1, size = 2, shape = 21, color = &#39;black&#39;, aes(fill = site_caps)) +
  geom_quasirandom(method=&quot;smiley&quot;, width = 0.1, alpha = 1, size = 1, 
                   shape = 21, color = &#39;black&#39;, aes(fill = site_caps), stroke = 0.4) +
  # geom_quasirandom(method=&quot;smiley&quot;, width = 0.1, alpha = 1, size = 1, aes(color = site_caps)) +
  # scale_fill_OkabeIto() +
  # scale_color_OkabeIto() +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  # scale_y_continuous(expand = c(0, 0)) +
  scale_y_continuous(expand = c(0.02, 0, 0.08, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), scales = &#39;free_y&#39;, ncol = 2) +
  theme(
    # strip.background = element_blank(),
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    panel.border = element_rect(colour = &quot;black&quot;, size=0.5),
    strip.text.x = element_text(size = 4),
    strip.background = element_blank()
  ) +
  # guides( color = &#39;none&#39;) +
  guides(fill = guide_legend(override.aes = list(size = 3))) +
  labs(
    fill = NULL,
    x = NULL,
    y = &#39;Proportion of cells&#39;
  ) +
  stat_pvalue_manual(stat_test, label = &#39;FDR.sign&#39;, size = 2)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/mf_da_stripchart_site_and_mouse_celltypefacet1_immgen-lung-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="liver-mf-differential-abundance-of-macrophages-cell-types"
class="section level2" number="11">
<h2><span class="header-section-number">11</span> Liver MF: Differential
abundance of macrophages cell types</h2>
<p>Differential abundance of cell types between normal lung tissue and
TME in lung metastasis</p>
<pre class="r"><code>ct_label &lt;- &#39;mf_immgen_pruned.labels.fine&#39;</code></pre>
<div id="statistical-analysis-3" class="section level3" number="11.1">
<h3><span class="header-section-number">11.1</span> Statistical
analysis</h3>
<p>Differential abundance using <code>speckle::propeller</code>
package</p>
<pre class="r"><code>use_data &lt;- cell_annot %&gt;% 
  filter(celltype_immgen_pruned.labels == &#39;Macrophages&#39;)

# Remove low abundant cell types 
keep_labels &lt;- table(use_data[[ct_label]]) %&gt;% data.frame %&gt;% filter(Freq &gt;= 100) %&gt;% pull(Var1)
keep_rows &lt;- use_data[[ct_label]] %in% keep_labels
use_data &lt;- use_data[keep_rows,]

# Run propeller Global
propeller_global &lt;- propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$site) %&gt;% 
  rownames_to_column(&#39;cell_type&#39;)


# Run pairwise propeller to obtaib P-values
site_levels &lt;- use_data$site_caps %&gt;% unique
# site_combn &lt;- combn(site_levels, 2, simplify = FALSE)
site_combn &lt;- list(
  # c(&#39;Lung Normal&#39;, &#39;Lung TME&#39;)
  c(&#39;Liver Normal&#39;, &#39;Liver TME&#39;)
)
sc &lt;-  site_combn[[1]]
propeller_pairwise &lt;-foreach(sc = site_combn, .combine = rbind) %do% {
  use_data &lt;- cell_annot %&gt;%
    filter(site_caps %in% sc) %&gt;%
    mutate(
      group = ifelse(site_caps == sc[1], &#39;group1&#39;, &#39;group2&#39;)
    )
  keep_rows &lt;- use_data[[ct_label]] %in% keep_labels
  use_data &lt;- use_data[keep_rows,]
  propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$group) %&gt;%
    mutate(
      group1 = sc[1],
      group2 = sc[2],
      FDR.sign = symnum(FDR, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001,0.001, 0.01, 0.05, 1), symbols = c(&quot;****&quot;, &quot;***&quot;, &quot;**&quot;, &quot;*&quot;, &quot;&quot;)),
      p.sign = symnum(P.Value, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c(&quot;****&quot;, &quot;***&quot;, &quot;**&quot;, &quot;*&quot;, &quot;&quot;))
    ) %&gt;%
    dplyr::rename(cell_type = BaselineProp.clusters)
}</code></pre>
<p>Table of results with pairwise comparison between sites for cluster
abundance.</p>
<pre class="r"><code>propeller_pairwise %&gt;%
  arrange(FDR) %&gt;% 
  datatable(., 
            rownames = FALSE, 
            filter = &#39;top&#39;, 
            caption = &#39;Pairwise comparison between cell types for cluster abundance using speckle::propeller&#39;,
            extensions = &#39;Buttons&#39;, 
            options = list(
              dom = &#39;Blfrtip&#39;,
              buttons = c(&#39;csv&#39;, &#39;excel&#39;)
            )) %&gt;% 
  formatPercentage(names(propeller_pairwise)[2:4], digits = 3) %&gt;% 
  formatRound(names(propeller_pairwise)[5:6], digits = 3, interval = -1000) %&gt;% 
  formatSignif(c(&#39;P.Value&#39;, &#39;FDR&#39;), digits = 3)</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-832cd5f680003d8dd78c" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-832cd5f680003d8dd78c">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Macrophages (MF.103-11B+24-)&quot;,&quot;Macrophages (MF.11C-11B+)&quot;,&quot;Macrophages (MF.11CLOSER)&quot;,&quot;Macrophages (MF.II+480LO)&quot;,&quot;Macrophages (MF.MEDL)&quot;,&quot;Macrophages (MFIO5.II-480INT)&quot;,&quot;Macrophages (MFIO5.II+480INT)&quot;,&quot;Macrophages (MFIO5.II+480LO)&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.009183673469387\" data-max=\"0.491836734693878\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"0.388888888888889\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.008001433856947\" data-max=\"0.495308115311825\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"41.6592\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-4.46965855766644\" data-max=\"5.0799664931681\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.00046709405527\" data-max=\"0.925483619570072\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0.00373675244216\" data-max=\"0.925483619570072\" data-scale=\"15\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","extensions":["Buttons"],"caption":"<caption>Pairwise comparison between cell types for cluster abundance using speckle::propeller<\/caption>","data":[["Macrophages (MF.11C-11B+)","Macrophages (MFIO5.II+480INT)","Macrophages (MFIO5.II-480INT)","Macrophages (MF.103-11B+24-)","Macrophages (MFIO5.II+480LO)","Macrophages (MF.11CLOSER)","Macrophages (MF.MEDL)","Macrophages (MF.II+480LO)"],[0.009183673469387756,0.4918367346938776,0.0173469387755102,0.07346938775510205,0.05714285714285714,0.1969387755102041,0.08061224489795918,0.07346938775510205],[0.3333333333333333,0.1111111111111111,-0,0.08333333333333333,0.08333333333333333,0.388888888888889,-0,-0],[0.008001433856947168,0.495308115311825,0.01615071859901685,0.05994799146055755,0.06292749124057667,0.189352019686806,0.07420162863916922,0.09411060120510183],[41.65919999999998,0.2243272574711607,0,1.390093834722754,1.324275474684721,2.053787910644539,0,0],[5.079966493168102,-4.469658557666438,3.393997627434778,1.303829474224736,1.136025485787558,0.9838985875478589,-0.2051058541597597,-0.09589672984223213],[0.0004670940552701008,0.001176693351140054,0.00677008215868314,0.2213000321927687,0.2822512025142145,0.3482072079734718,0.8415748276052872,0.9254836195700711],[0.003736752442160806,0.004706773404560214,0.01805355242315504,0.4426000643855373,0.4516019240227432,0.4642762772979624,0.9254836195700711,0.9254836195700711],["Liver Normal","Liver Normal","Liver Normal","Liver Normal","Liver Normal","Liver Normal","Liver Normal","Liver Normal"],["Liver TME","Liver TME","Liver TME","Liver TME","Liver TME","Liver TME","Liver TME","Liver TME"],["**","**","*","","","","",""],["***","**","**","","","","",""]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>cell_type<\/th>\n      <th>BaselineProp.Freq<\/th>\n      <th>PropMean.group1<\/th>\n      <th>PropMean.group2<\/th>\n      <th>PropRatio<\/th>\n      <th>Tstatistic<\/th>\n      <th>P.Value<\/th>\n      <th>FDR<\/th>\n      <th>group1<\/th>\n      <th>group2<\/th>\n      <th>FDR.sign<\/th>\n      <th>p.sign<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["csv","excel"],"columnDefs":[{"targets":6,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":7,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatSignif(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":4,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, -1000, \",\", \".\", null);\n  }"},{"targets":5,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 3, -1000, \",\", \".\", null);\n  }"},{"targets":1,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatPercentage(data, 3, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[1,2,3,4,5,6,7]}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.columnDefs.3.render","options.columnDefs.4.render","options.columnDefs.5.render","options.columnDefs.6.render"],"jsHooks":[]}</script>
</div>
<div id="distribution-by-site-3" class="section level3" number="11.2">
<h3><span class="header-section-number">11.2</span> Distribution by
site</h3>
<pre class="r"><code># Create data
use_data &lt;- cell_annot %&gt;% 
  filter(site_caps %in% unlist(site_combn)) %&gt;% 
  dplyr::rename(&#39;celltype_label&#39; = ct_label) %&gt;% 
  group_by(site_caps, celltype_label) %&gt;% 
  summarise(freq = n()) %&gt;% 
  dplyr::rename(cell_type = celltype_label) %&gt;% 
  na.omit()

# Remove low abundant cell types
cell_type_freq &lt;- use_data %&gt;% group_by(cell_type) %&gt;% summarise(totalfreq = sum(freq))
# site_filtered &lt;- use_data %&gt;% group_by(site_caps) %&gt;% summarise(freq = sum(freq)) %&gt;% filter(freq &gt;= 500)
use_data &lt;- use_data %&gt;% 
  # filter(site_caps %in% site_filtered$site_caps) %&gt;% 
  left_join(cell_type_freq) %&gt;% 
  filter(totalfreq &gt;= 50) %&gt;% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data &lt;- use_data %&gt;% 
  arrange(site_caps, cell_type)%&gt;% 
  group_by(site_caps) %&gt;%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion &lt; 0.03, &#39;&#39;, cell_type_label)
    
  )</code></pre>
<div id="barplot-stacked-3" class="section level4" number="11.2.1">
<h4><span class="header-section-number">11.2.1</span> Barplot
stacked</h4>
<pre class="r"><code>arranged_data %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, fill = cell_type)) +
  geom_col(alpha = 0.85, color = &#39;black&#39;, linewidth = 0.2) +
  # scale_fill_manual(values = celltype_palette) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  # geom_text(aes(y = label_y, label = cell_type_label), colour = &quot;white&quot;, size = geom_text_size) +
  theme(axis.ticks.x = element_blank()) +
  labs(
    fill = NULL, 
    x = NULL,
    y = &#39;Proportion of cells&#39;
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/mf_da_barplot_site_stacked_immgen-liver-1.png" width="2100" style="display: block; margin: auto;" /></p>
</div>
<div id="barplot-facet-3" class="section level4" number="11.2.2">
<h4><span class="header-section-number">11.2.2</span> Barplot facet</h4>
<pre class="r"><code>arranged_data %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, fill = site_caps, group = cell_type)) +
  geom_col(alpha = 0.85, color = &#39;black&#39;, size = 0.2) +
  scale_fill_manual(values = site_type_palette) +
  scale_y_continuous(expand = c(0.02, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), ncol = 3, scales = &#39;free_y&#39;) +
  theme(
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x = element_blank(),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    panel.border = element_rect(colour = &quot;black&quot;, size=0.5),
    strip.background = element_blank()
  ) +
  labs(
    fill = NULL, 
    x = NULL,
    y = &#39;Proportion of cells&#39;
  ) #+</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/mf_da_barplot_site_celltypefacet_immgen-liver-1.png" width="3120" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # guides(fill = &#39;none&#39;)</code></pre>
</div>
<div id="piechart-3" class="section level4" number="11.2.3">
<h4><span class="header-section-number">11.2.3</span> Piechart</h4>
<pre class="r"><code>arranged_data %&gt;% 
  ggplot(aes(x = &quot;&quot;, y = proportion, fill = cell_type)) +
  # geom_bar(stat=&quot;identity&quot;, width=1, color=&quot;white&quot;) +
  geom_bar(stat=&quot;identity&quot;, width=1) +
  coord_polar(&quot;y&quot;, start=0) +
  # scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = &#39;&#39;, 
    x = NULL,
    y = NULL
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/mf_da_piechart_site_celltypefacet_immgen-liver-1.png" width="2400" style="display: block; margin: auto;" /></p>
</div>
<div id="doughnut-chart-3" class="section level4" number="11.2.4">
<h4><span class="header-section-number">11.2.4</span> Doughnut
chart</h4>
<pre class="r"><code>i &lt;- unique(arranged_data$site_caps)[1]
use_data &lt;- foreach(i = unique(arranged_data$site_caps), .combine = rbind) %do% {
  x &lt;- arranged_data %&gt;% 
    filter(site_caps == i) %&gt;% 
    arrange(cell_type)
  x$ymax &lt;- cumsum(x$proportion)
  x$ymin &lt;- c(0, head(x$ymax, n = -1))
  x
}

use_data %&gt;% 
  ggplot(aes(ymax=ymax, ymin=ymin, xmax=4, xmin=3, fill=cell_type)) +
  geom_rect() +
  coord_polar(theta=&quot;y&quot;) + # Try to remove that to understand how the chart is built initially
  xlim(c(2, 4)) + # Try to remove that to see how to make a pie chart
  # scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = &#39;&#39;, 
    x = NULL,
    y = NULL
  )</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/mf_da_doughnut_site_celltypefacet_immgen-liver-1.png" width="2400" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="distribution-by-site-and-mouse-3" class="section level3"
number="11.3">
<h3><span class="header-section-number">11.3</span> Distribution by site
and mouse</h3>
<pre class="r"><code># Create data
use_data &lt;- cell_annot %&gt;% 
  filter(site_caps %in% unlist(site_combn)) %&gt;% 
  dplyr::rename(&#39;celltype_label&#39; = ct_label) %&gt;% 
  group_by(site_caps, mouse_id, celltype_label) %&gt;% 
  summarise(freq = n()) %&gt;% 
  dplyr::rename(cell_type = celltype_label) %&gt;% 
  na.omit()

# Remove low abundant cell types and sites
cell_type_freq &lt;- use_data %&gt;% group_by(cell_type) %&gt;% summarise(totalfreq = sum(freq))
# site_filtered &lt;- use_data %&gt;% group_by(site_caps) %&gt;% summarise(freq = sum(freq)) %&gt;% filter(freq &gt;= 500)
use_data &lt;- use_data %&gt;% 
  # filter(site_caps %in% site_filtered$site_caps) %&gt;% 
  left_join(cell_type_freq) %&gt;% 
  filter(totalfreq &gt;= 50) %&gt;% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data &lt;- use_data %&gt;% 
  arrange(site_caps, cell_type)%&gt;% 
  group_by(site_caps, mouse_id) %&gt;%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion &lt; 0.03, &#39;&#39;, cell_type_label)
  )</code></pre>
<div id="stripchart-boxplot-5" class="section level4" number="11.3.1">
<h4><span class="header-section-number">11.3.1</span>
Stripchart-Boxplot</h4>
<pre class="r"><code>max_prop_cell_type &lt;- arranged_data %&gt;% group_by(cell_type) %&gt;% summarise(max_proportion = max(proportion))
stat_test &lt;- propeller_pairwise %&gt;% 
  filter(group1 %in% arranged_data$site_caps &amp; group2 %in%  arranged_data$site_caps) %&gt;%
  left_join(max_prop_cell_type) %&gt;% 
  arrange(cell_type, group1, group2) %&gt;% 
  na.omit() %&gt;% 
  filter(FDR &lt; 0.05)
  
add_count_y &lt;- stat_test %&gt;% group_by(cell_type) %&gt;% reframe(group1 = group1, group2 = group2, n = 1:n())
stat_test &lt;- stat_test %&gt;% left_join(add_count_y) %&gt;% mutate(y.position = max_proportion + (n*max_proportion/10))


arranged_data %&gt;% 
  ggplot(aes(x = site_caps, y = proportion, color = site_caps)) +
  geom_boxplot(alpha = 0.4, color = &#39;black&#39;, aes(fill = site_caps), 
               show.legend = FALSE, size = 0.25, fatten = 1) +
  # geom_point(alpha = 1, size = 2, shape = 21, color = &#39;black&#39;, aes(fill = site_caps)) +
  geom_quasirandom(method=&quot;smiley&quot;, width = 0.1, alpha = 1, size = 1, 
                   shape = 21, color = &#39;black&#39;, aes(fill = site_caps), stroke = 0.4) +
  # geom_quasirandom(method=&quot;smiley&quot;, width = 0.1, alpha = 1, size = 1, aes(color = site_caps)) +
  # scale_fill_OkabeIto() +
  # scale_color_OkabeIto() +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  # scale_y_continuous(expand = c(0, 0)) +
  scale_y_continuous(expand = c(0.02, 0, 0.08, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), scales = &#39;free_y&#39;, ncol = 2) +
  theme(
    # strip.background = element_blank(),
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    panel.border = element_rect(colour = &quot;black&quot;, size=0.5),
    strip.text.x = element_text(size = 4),
    strip.background = element_blank()
  ) +
  # guides( color = &#39;none&#39;) +
  guides(fill = guide_legend(override.aes = list(size = 3))) +
  labs(
    fill = NULL,
    x = NULL,
    y = &#39;Proportion of cells&#39;
  ) +
  stat_pvalue_manual(stat_test, label = &#39;FDR.sign&#39;, size = 2)</code></pre>
<p><img src="figure/mvt1-tumor-nsg-normal-10x_rnaseq-comparison.Rmd/mf_da_stripchart_site_and_mouse_celltypefacet1_immgen-liver-1.png" width="2100" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<p>R version 4.2.2 (2022-10-31) Platform: x86_64-apple-darwin17.0
(64-bit) Running under: macOS Big Sur … 10.16</p>
<p>Matrix products: default BLAS:
/Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK:
/Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</p>
<p>locale: [1]
en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</p>
<p>attached base packages: [1] grid stats4 stats graphics grDevices
utils datasets [8] methods base</p>
<p>other attached packages: [1] circlize_0.4.15 GSVA_1.46.0<br />
[3] clusterProfiler_4.6.0 ComplexHeatmap_2.14.0<br />
[5] miloR_1.6.0 edgeR_3.40.2<br />
[7] limma_3.54.1 speckle_0.0.3<br />
[9] scDblFinder_1.13.8 SingleR_2.0.0<br />
[11] celldex_1.8.0 bluster_1.8.0<br />
[13] batchelor_1.14.1 BiocSingular_1.14.0<br />
[15] scran_1.26.2 SeuratObject_4.1.3<br />
[17] Seurat_4.3.0 DropletUtils_1.18.1<br />
[19] scater_1.26.1 scuttle_1.8.4<br />
[21] SingleCellExperiment_1.20.0 SummarizedExperiment_1.28.0 [23]
Biobase_2.58.0 GenomicRanges_1.50.2<br />
[25] GenomeInfoDb_1.34.9 IRanges_2.32.0<br />
[27] S4Vectors_0.36.1 BiocGenerics_0.44.0<br />
[29] MatrixGenerics_1.10.0 matrixStats_1.0.0<br />
[31] magrittr_2.0.3 openxlsx_4.2.5.2<br />
[33] patchwork_1.1.3 pheatmap_1.0.12<br />
[35] ggpubr_0.6.0 RColorBrewer_1.1-3<br />
[37] arsenal_3.6.3 ggbeeswarm_0.7.2<br />
[39] colorblindr_0.1.0 colorspace_2.1-1<br />
[41] cowplot_1.1.1 kableExtra_1.3.4<br />
[43] knitr_1.44 DT_0.30<br />
[45] foreach_1.5.2 showtext_0.9-6<br />
[47] showtextdb_3.0 sysfonts_0.8.8<br />
[49] lubridate_1.9.3 forcats_1.0.0<br />
[51] stringr_1.5.1 dplyr_1.1.4<br />
[53] purrr_1.0.2 readr_2.1.4<br />
[55] tidyr_1.3.1 tibble_3.2.1<br />
[57] ggplot2_3.5.0 tidyverse_2.0.0<br />
[59] workflowr_1.7.1</p>
<p>loaded via a namespace (and not attached): [1] rsvd_1.0.5
ica_1.0-3<br />
[3] svglite_2.1.2 ps_1.7.5<br />
[5] Rsamtools_2.14.0 lmtest_0.9-40<br />
[7] rprojroot_2.0.3 crayon_1.5.2<br />
[9] MASS_7.3-60 rhdf5filters_1.10.0<br />
[11] nlme_3.1-163 backports_1.4.1<br />
[13] GOSemSim_2.24.0 rlang_1.1.4<br />
[15] HDO.db_0.99.1 XVector_0.38.0<br />
[17] ROCR_1.0-11 irlba_2.3.5.1<br />
[19] callr_3.7.3 filelock_1.0.2<br />
[21] xgboost_1.7.5.1 BiocParallel_1.32.5<br />
[23] rjson_0.2.21 bit64_4.0.5<br />
[25] glue_1.8.0 sctransform_0.4.1<br />
[27] parallel_4.2.2 processx_3.8.2<br />
[29] vipor_0.4.5 spatstat.sparse_3.0-2<br />
[31] AnnotationDbi_1.60.0 DOSE_3.24.2<br />
[33] spatstat.geom_3.2-7 tidyselect_1.2.1<br />
[35] fitdistrplus_1.1-11 XML_3.99-0.14<br />
[37] zoo_1.8-12 GenomicAlignments_1.34.0<br />
[39] org.Mm.eg.db_3.16.0 xtable_1.8-4<br />
[41] evaluate_0.22 cli_3.6.3<br />
[43] zlibbioc_1.44.0 rstudioapi_0.15.0<br />
[45] miniUI_0.1.1.1 sp_2.1-1<br />
[47] whisker_0.4.1 bslib_0.5.1<br />
[49] fastmatch_1.1-4 treeio_1.22.0<br />
[51] shiny_1.7.5.1 xfun_0.40<br />
[53] clue_0.3-65 gson_0.1.0<br />
[55] cluster_2.1.4 tidygraph_1.2.3<br />
[57] KEGGREST_1.38.0 interactiveDisplayBase_1.36.0 [59] ggrepel_0.9.4
ape_5.7-1<br />
[61] listenv_0.9.0 Biostrings_2.66.0<br />
[63] png_0.1-8 future_1.33.0<br />
[65] withr_3.0.1 ggforce_0.4.1<br />
[67] bitops_1.0-7 plyr_1.8.9<br />
[69] GSEABase_1.60.0 dqrng_0.3.1<br />
[71] pillar_1.9.0 GlobalOptions_0.1.2<br />
[73] cachem_1.0.8 fs_1.6.4<br />
[75] GetoptLong_1.0.5 DelayedMatrixStats_1.20.0<br />
[77] vctrs_0.6.5 ellipsis_0.3.2<br />
[79] generics_0.1.3 tools_4.2.2<br />
[81] beeswarm_0.4.0 tweenr_2.0.2<br />
[83] munsell_0.5.1 fgsea_1.24.0<br />
[85] DelayedArray_0.24.0 fastmap_1.1.1<br />
[87] compiler_4.2.2 abind_1.4-5<br />
[89] httpuv_1.6.8 rtracklayer_1.58.0<br />
[91] ExperimentHub_2.6.0 plotly_4.10.3<br />
[93] GenomeInfoDbData_1.2.9 gridExtra_2.3<br />
[95] lattice_0.20-45 deldir_1.0-9<br />
[97] utf8_1.2.4 later_1.3.1<br />
[99] BiocFileCache_2.6.0 jsonlite_1.8.7<br />
[101] scales_1.3.0 graph_1.76.0<br />
[103] ScaledMatrix_1.6.0 tidytree_0.4.2<br />
[105] pbapply_1.7-2 carData_3.0-5<br />
[107] sparseMatrixStats_1.10.0 lazyeval_0.2.2<br />
[109] promises_1.2.1 doParallel_1.0.17<br />
[111] car_3.1-2 R.utils_2.12.2<br />
[113] goftest_1.2-3 spatstat.utils_3.0-3<br />
[115] reticulate_1.34.0 rmarkdown_2.25<br />
[117] statmod_1.5.0 webshot_0.5.5<br />
[119] Rtsne_0.16 downloader_0.4<br />
[121] uwot_0.1.16 igraph_1.5.1<br />
[123] HDF5Array_1.26.0 survival_3.5-7<br />
[125] ResidualMatrix_1.8.0 yaml_2.3.7<br />
[127] systemfonts_1.1.0 htmltools_0.5.6.1<br />
[129] memoise_2.0.1 BiocIO_1.8.0<br />
[131] locfit_1.5-9.8 graphlayouts_1.0.1<br />
[133] viridisLite_0.4.2 digest_0.6.37<br />
[135] mime_0.12 rappdirs_0.3.3<br />
[137] RSQLite_2.3.1 yulab.utils_0.1.7<br />
[139] future.apply_1.11.0 data.table_1.14.8<br />
[141] blob_1.2.4 R.oo_1.25.0<br />
[143] labeling_0.4.3 splines_4.2.2<br />
[145] Rhdf5lib_1.20.0 AnnotationHub_3.6.0<br />
[147] RCurl_1.98-1.12 broom_1.0.5<br />
[149] hms_1.1.3 rhdf5_2.42.0<br />
[151] BiocManager_1.30.22 shape_1.4.6<br />
[153] aplot_0.2.2 sass_0.4.7<br />
[155] Rcpp_1.0.13 RANN_2.6.1<br />
[157] enrichplot_1.18.3 fansi_1.0.6<br />
[159] tzdb_0.4.0 parallelly_1.36.0<br />
[161] R6_2.5.1 ggridges_0.5.4<br />
[163] lifecycle_1.0.4 zip_2.3.0<br />
[165] curl_5.1.0 ggsignif_0.6.4<br />
[167] leiden_0.4.3 jquerylib_0.1.4<br />
[169] Matrix_1.5-3 qvalue_2.30.0<br />
[171] RcppAnnoy_0.0.21 org.Hs.eg.db_3.16.0<br />
[173] iterators_1.0.14 spatstat.explore_3.2-3<br />
[175] htmlwidgets_1.6.2 beachmat_2.14.0<br />
[177] polyclip_1.10-6 crosstalk_1.2.0<br />
[179] shadowtext_0.1.2 gridGraphics_0.5-1<br />
[181] timechange_0.2.0 rvest_1.0.3<br />
[183] globals_0.16.2 spatstat.random_3.2-1<br />
[185] progressr_0.14.0 codetools_0.2-19<br />
[187] GO.db_3.16.0 metapod_1.6.0<br />
[189] gtools_3.9.4 getPass_0.2-2<br />
[191] dbplyr_2.3.4 R.methodsS3_1.8.2<br />
[193] gtable_0.3.5 DBI_1.1.3<br />
[195] git2r_0.32.0 ggfun_0.1.3<br />
[197] tensor_1.5 httr_1.4.7<br />
[199] KernSmooth_2.23-22 stringi_1.8.4<br />
[201] farver_2.1.2 reshape2_1.4.4<br />
[203] annotate_1.76.0 viridis_0.6.4<br />
[205] ggtree_3.4.4 xml2_1.3.5<br />
[207] BiocNeighbors_1.16.0 restfulr_0.0.15<br />
[209] ggplotify_0.1.2 scattermore_1.2<br />
[211] BiocVersion_3.16.0 bit_4.0.5<br />
[213] scatterpie_0.2.1 spatstat.data_3.0-3<br />
[215] ggraph_2.1.0 pkgconfig_2.0.3<br />
[217] rstatix_0.7.2</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
