---
title: "Comapre 10x RNA-seq analysis of microenvironment mets and normal tissue"
subtitle: "Analysis of TME and normal cells from lung and liver normal tissue and metastases"
author: "Francesc Castro-Giner"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
params:
  date: '`r format(Sys.Date(), "%B %d, %Y")`'
  output_dir_tme: ./output/p26532_o28268/p26532_o28268-10x_rnaseq-pll-tme-hq
  output_dir_normal: ./output/p26532_o34980/p26532_o34980-10x_rnaseq
  output_dir: ./output/p26532_o28268_o34980/p26532_o28268_o34980-10x_rnaseq-ll-tme-normal
  ncores: 6
---

## Load libraries, additional functions and data

Setup environment
```{r setup, include=TRUE, warning=FALSE, message=FALSE}
knitr::opts_chunk$set(results='asis', echo=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.align = 'center', fig.width = 3.5, fig.asp = 0.618, dpi = 600, dev = c("png", "pdf"))

options(stringsAsFactors = FALSE)

use_seed <- 1100101
set.seed(use_seed)

if(!dir.exists(params$output_dir))
  dir.create(params$output_dir, recursive = TRUE, showWarnings = FALSE)
```

Load packages
```{r load-libraries}
library(tidyverse)
library(showtext)
library(foreach)
library(DT)
library(knitr)
library(kableExtra)
library(cowplot)
library(colorblindr)
library(ggbeeswarm)
library(arsenal)
library(RColorBrewer)
library(ggpubr)
library(pheatmap)
library(patchwork)
library(openxlsx)
library(magrittr)

library(scater)
library(DropletUtils)
library(Seurat)
library(scran)
library(BiocSingular)
library(batchelor)
library(bluster)
library(celldex)
library(SingleR)
library(scDblFinder)
library(speckle)
library(miloR)
library(ComplexHeatmap)
library(clusterProfiler)
library(GSVA)
library(circlize)
```

Load ggplot theme
```{r ggplot-theme}
source("./configuration/rmarkdown/ggplot_theme.R")
source("./configuration/rmarkdown/color_palettes.R")
```

Load utils functions
```{r load-util-functions}
source('./code/R-functions/heatmap_scale.R')
```

Clean files generated in previous runs
```{r clean-files}
rmd_file <- current_input()
if(!is.null(rmd_file)) {
  figures_dir <- file.path('./docs/figure',rmd_file)
  if(dir.exists(figures_dir)) {
    unlink(file.path(figures_dir, "*"))
  }
}

```

## Data processing

### Original data
Data processing original TME sce data
```{r load-raw-data-tme, eval = FALSE}
sce_comb <- readRDS(file.path(params$output_dir_tme, 'sce_integrated-tme.rds'))

sr_immgen <- readRDS(file.path(params$output_dir_tme, 'SingleR_ImmGenData_annotation-tme.rds'))

# Add cell type annotation
sce_comb$celltype_immgen_pruned.labels <- sr_immgen$pred$pruned.labels
sce_comb$celltype_immgen.labels <- sr_immgen$pred$labels
sce_comb$celltype_immgen_pruned.labels.fine <- sr_immgen$pred_fine$pruned.labels
sce_comb$celltype_immgen.labels.fine <- sr_immgen$pred_fine$labels

# Add cluster level annotation using immgen database
sr_immgen$pred_cluster_simplified %<>%
  dplyr::rename(
    celltype_immgen_cluster_labels = celltype_cluster_labels,
    celltype_immgen_cluster_pruned.labels = celltype_cluster_pruned.labels
  )
colData(sce_comb) <- colData(sce_comb) %>% data.frame %>%
  mutate(cluster_mnn = as.character(cluster_mnn)) %>%
  left_join(sr_immgen$pred_cluster_simplified) %>%
  mutate(cluster_mnn = factor(cluster_mnn)) %>%
  DataFrame


# Select sites
sce_comb <- sce_comb[,sce_comb$site %in% c('liver', 'lung')]

# Remove doublets
sce_comb <- sce_comb[,sce_comb$scDblFinder.class != 'doublet']

# Site levels order
site_caps_ord <- c('Liver TME', 'Lung TME')
site_ord <- c('liver_tme', 'lung_tme')


# Add new colData vars
colData(sce_comb) <- colData(sce_comb) %>% data.frame %>% 
  mutate(
    site_caps = ifelse(site == 'liver', 'Liver', site),
    site_caps = ifelse(site == 'lung', 'Lung', site_caps),
    
    site = paste0(site, '_tme'),
    site_caps = paste(site_caps, 'TME'),
    
    site = factor(site, levels = site_ord),
    site_caps = factor(site_caps, levels = site_caps_ord),
    
    site_mouse = paste(site, mouse_id, sep = '_')
  ) %>% 
  DataFrame

sce_tme <- sce_comb
rm(sce_comb)
```

Data processing original Normal sce data
```{r load-raw-data-normal, eval = FALSE}
sce_comb <- readRDS(file.path(params$output_dir_normal, 'sce_integrated.rds'))
sr_immgen <- readRDS(file.path(params$output_dir_normal, 'SingleR_ImmGenData_annotation.rds'))

# Add cell type annotation
sce_comb$celltype_immgen_pruned.labels <- sr_immgen$pred$pruned.labels
sce_comb$celltype_immgen.labels <- sr_immgen$pred$labels
sce_comb$celltype_immgen_pruned.labels.fine <- sr_immgen$pred_fine$pruned.labels
sce_comb$celltype_immgen.labels.fine <- sr_immgen$pred_fine$labels

# Add cluster level annotation using immgen database
sr_immgen$pred_cluster_simplified %<>%
  dplyr::rename(
    celltype_immgen_cluster_labels = celltype_cluster_labels,
    celltype_immgen_cluster_pruned.labels = celltype_cluster_pruned.labels
  )
colData(sce_comb) <- colData(sce_comb) %>% data.frame %>%
  mutate(cluster = as.character(cluster)) %>%
  left_join(sr_immgen$pred_cluster_simplified) %>%
  mutate(cluster = factor(cluster)) %>%
  DataFrame

# Remove doublets
sce_comb <- sce_comb[,sce_comb$scDblFinder.class != 'doublet']


# Site levels order
site_caps_ord <- c('Liver Normal', 'Lung Normal')
site_ord <- c('liver_normal', 'lung_normal')


# Add new colData vars
colData(sce_comb) <- colData(sce_comb) %>% data.frame %>% 
  mutate(
    site_caps = ifelse(site == 'liver', 'Liver', site),
    site_caps = ifelse(site == 'lung', 'Lung', site_caps),
    
    site = paste0(site, '_normal'),
    site_caps = paste(site_caps, 'Normal'),
    
    site = factor(site, levels = site_ord),
    site_caps = factor(site_caps, levels = site_caps_ord),
    
    mouse_id = mouse_hto,
    site_mouse = paste(site, mouse_id, sep = '_')
  ) %>% 
  DataFrame

sce_normal<- sce_comb
rm(sce_comb)
```


### Batch integration
Here we prepare the data for batch correction (next section). First, we combine cells from all SCE objects into a single SCE and return pre-computed per-gene variance at batch level. Then we adjust for batch level library size. Returned object will contain log-normalized expression values after adjusting the size factors for systematic differences in coverage between

```{r combine-sce, eval = FALSE}
# Combine all SCE files into sce_comb and generate an combine gene  with pre-computed gene variance 
if(exists('sce_comb')) 
  rm(sce_comb)

# Combine count data
use_rows <- intersect(rownames(sce_tme), rownames(sce_normal))
new_counts <- cbind(counts(sce_tme[use_rows,]), counts(sce_normal[use_rows,]))
new_logcounts <- cbind(logcounts(sce_tme[use_rows,]), logcounts(sce_normal[use_rows,]))

# Combine colData
use_cols <- intersect(names(colData(sce_tme)), names(colData(sce_normal)))
use_cols <- use_cols[use_cols != 'cluster_id_batch']
new_colData <- rbind(colData(sce_tme)[,use_cols], colData(sce_normal)[,use_cols])

# Combined SCE
sce_comb <- SingleCellExperiment(
  assays = list(counts = new_counts, logcounts = new_logcounts),
  colData = new_colData,
  rowData = rowData(sce_normal)[use_rows,]
)

# Modify coldata and rownames
sce_comb$batch <-  gsub("_.*", "", sce_comb$Sample)
sce_comb$origin <- ifelse(sce_comb$batch == 'huNSG', 'normal', 'tme')
# sce_comb$site <- 
sce_comb$site<- factor(sce_comb$site, c('liver_normal', 'lung_normal', 'liver_tme', 'lung_tme'))
# sce_comb$site_caps<- recode(as.character(sce_comb$site), 
#                             liver_normal = "Liver Normal", 
#                             lung_normal = "Lung Normal", 
#                             liver_tme = "Liver TME", 
#                             lung_tme = "Lung TME")
sce_comb$site_caps<- factor(sce_comb$site_caps, c('Liver Normal', 'Lung Normal', 'Liver TME', 'Lung TME'))

# Recalculate per-gene variance at batch level
metadata(sce_comb)$precomputed_geneVar <- foreach(i = unique(sce_comb$Sample)) %do%{
  use_sce <- sce_comb[,sce_comb$Sample == i]
  modelGeneVar(use_sce, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
}


# Rescale each batch to adjust for differences in sequencing depth between batches. The multiBatchNorm() function recomputes log-normalized expression values after adjusting the size factors for systematic differences in coverage between SingleCellExperiment objects. (Size factors only remove biases between cells within a single batch.) This improves the quality of the correction by removing one aspect of the technical differences between batches.
sce_comb <- multiBatchNorm(sce_comb, batch = sce_comb$Sample, BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))

# Identifying a set of HVGs using stats from all batches.
combined_dec <- combineVar(metadata(sce_comb)$precomputed_geneVar)
chosen_hvgs <- combined_dec$bio > 0 & !rowData(sce_comb)$is.mito & !rowData(sce_comb)$is.ribo

# Dimensionality reduction
sce_comb <- runPCA(sce_comb, subset_row=chosen_hvgs, BSPARAM=BiocSingular::RandomParam(), BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
sce_comb$cluster_id  <- clusterCells(sce_comb, use.dimred="PCA", BLUSPARAM=SNNGraphParam(k=25, type="rank", cluster.fun="louvain", BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)))
tab <- table(Cluster=sce_comb$cluster_id, Batch=sce_comb$Sample)
sce_comb <- runTSNE(sce_comb, dimred="PCA", BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
sce_comb <- runUMAP(sce_comb, dimred="PCA", BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))

saveRDS(sce_comb, file.path(params$output_dir, 'sce_integrated.rds'))
```

### Batch correction using MNN
Here, we actually perform the correction itself. We also perform the clustering.
```{r batch-correction-keep-dpublets, eval = FALSE}
if(!exists('sce_comb'))
  sce_comb <- readRDS(file.path(params$output_dir, 'sce_integrated.rds'))

# Identifying a set of HVGs using stats from all batches.
combined_dec <- combineVar(metadata(sce_comb)$precomputed_geneVar)
# combined_hvg <- getTopHVGs(combined_dec, n=2000, row.names = FALSE)
chosen_hvgs <- combined_dec$bio > 0 & !rowData(sce_comb)$is.mito & !rowData(sce_comb)$is.ribo

# MNN correction
sce_mnn <- fastMNN(sce_comb,
                   batch = sce_comb$Sample, 
                   auto.merge = TRUE, 
                      subset.row=chosen_hvgs, 
                      BSPARAM=RandomParam(deferred=TRUE), 
                      BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed),
                      d=50, k=20)
# colData(sce_mnn) <- colData(sce_comb)
rowData(sce_mnn) <-  rowData(sce_comb)[chosen_hvgs,]

# Dimensionality reduction
sce_mnn <- runTSNE(sce_mnn, dimred="corrected", BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))
sce_mnn <- runUMAP(sce_mnn, dimred="corrected", BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))

# Clustering using corrected values
sce_mnn$cluster_mnn  <- clusterCells(sce_mnn, use.dimred="corrected", BLUSPARAM=SNNGraphParam(k=25, type="rank", cluster.fun="louvain", BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)))

# Add corrected info to sce_comb
sce_comb$cluster_mnn <- sce_mnn$cluster_mnn
reducedDim(sce_comb, 'corrected') <- reducedDim(sce_mnn, 'corrected')
reducedDim(sce_comb, 'TSNE_corrected') <- reducedDim(sce_mnn, 'TSNE')
reducedDim(sce_comb, 'UMAP_corrected') <- reducedDim(sce_mnn, 'UMAP')
# plotUMAP(sce_mnn, color_by = 'cluster_mnn')

# MNN-related diagnostic examining the variance in the differences in expression between MNN pairs.
use_sce <- sce_comb
rownames(use_sce) <- make.names(rowData(use_sce)$gene_name, unique = TRUE)
metadata(sce_mnn)$mnnDeltaVariance <- mnnDeltaVariance(
  use_sce,
  pairs=metadata(sce_mnn)$merge.info$pairs, 
  BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)
  )
rm(use_sce)

# Save SCE objects
saveRDS(sce_mnn, file.path(params$output_dir, 'sce_integrated_corrected.rds'))
saveRDS(sce_comb, file.path(params$output_dir, 'sce_integrated.rds'))
```

### Markers detection

We identify the genes that drive separation between clusters and cell types. These marker genes allow us to assign biological meaning to each cluster based on their functional annotation. We added batch as a blocking factor to the model. The block argument works for all effect sizes shown above and is robust to differences in the log-fold changes or variance between batches. However, it assumes that each pair of clusters is present in at least one batch. In scenarios where cells from two clusters never co-occur in the same batch, the associated pairwise comparison will be impossible and is ignored during calculation of summary statistics.

```{r score-marker-genes, eval = FALSE}
marker_info <- list()

if(!exists('sce_comb'))
  sce_comb <- readRDS(file.path(params$output_dir, 'sce_integrated.rds'))

# TME-Normal markers for each cell type from ImmGen
ct_label <- 'celltype_immgen_pruned.labels'
## Remove low abundant cell types 
keep_labels <- table(colData(sce_comb)[[ct_label]]) %>% data.frame %>% filter(Freq >= 100) %>% pull(Var1)
## Iterate by cell type
marker_info$origin_celltype_markers <- foreach(use_label = keep_labels) %do% {
  use_cols <- colData(sce_comb)[[ct_label]] %in%  use_label
  use_sce <- sce_comb[,use_cols]
  scoreMarkers(
    use_sce, 
    groups = use_sce$origin, 
    lfc=0.5, 
    full.stats=TRUE,
    row.data=rowData(use_sce)[,c('gene_name', 'is.mito', 'is.ribo'),drop=FALSE], 
    BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)
    )
}
names(marker_info$origin_celltype_markers) <- keep_labels


# TME-Normal markers for each cell type and site from ImmGen
ct_label <- 'celltype_immgen_pruned.labels'
## Iterate by cell type and site
sce_comb$site_anatomy <- gsub("_.*", "", sce_comb$site)
marker_info$origin_site_celltype_markers <-foreach(site = unique(sce_comb$site_anatomy)) %do% {
  use_sce <- sce_comb[,sce_comb$site_anatomy == site]
  ## Remove low abundant cell types 
  keep_labels <- table(colData(use_sce)[[ct_label]]) %>% data.frame %>% filter(Freq >= 100) %>% pull(Var1)
  ## Iterate by cell type
  res <- foreach(use_label = keep_labels) %do% {
    use_cols <- colData(use_sce)[[ct_label]] %in%  use_label
    use_sce_i<- use_sce[,use_cols]
    scoreMarkers(
      use_sce_i, 
      groups = use_sce_i$origin, 
      lfc=0.5, 
      full.stats=TRUE,
      row.data=rowData(use_sce_i)[,c('gene_name', 'is.mito', 'is.ribo'),drop=FALSE], 
      BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed)
      )
  }
  names(res) <- keep_labels
  res
}
names(marker_info$origin_site_celltype_markers) <- unique(sce_comb$site)

saveRDS(marker_info, file.path(params$output_dir, 'scoreMarkers.rds'))

```

### Macrophages subtype annotaion
```{r mfe-annotation-single-R-ImmGenData, eval = FALSE}
if(!exists('ref_mouse_imm'))
  ref_mouse_imm <- celldex::ImmGenData()

# Subset macrophages annotation from immgen
ref_mouse_imm_mf <- ref_mouse_imm[,ref_mouse_imm$label.main == 'Macrophages']

# Run SingleR at cell level with fail label
use_cols <- sce_comb$celltype_immgen_pruned.labels %in% 'Macrophages'
pred_fine <- SingleR(test=sce_comb[,use_cols], 
                ref=ref_mouse_imm_mf, 
                labels=ref_mouse_imm_mf$label.fine, 
                de.method="classic",
                BPPARAM=BiocParallel::MulticoreParam(params$ncores, RNGseed=use_seed))

sce_comb$mf_immgen_pruned.labels.fine <- NA
sce_comb$mf_immgen_pruned.labels.fine[use_cols] <- pred_fine$pruned.labels

saveRDS(sce_comb, file.path(params$output_dir, 'sce_integrated.rds'))
```

### Load data
```{r load-data, eval = TRUE}
sce_comb <- readRDS(file.path(params$output_dir, 'sce_integrated.rds'))
sce_mnn <- readRDS(file.path(params$output_dir, 'sce_integrated_corrected.rds'))
score_markers <- readRDS(file.path(params$output_dir, 'scoreMarkers.rds'))

# Coordinate colData between sce_comb and sce_mnn
colData(sce_mnn) <- colData(sce_comb)
assay(sce_mnn, 'logcounts') <- logcounts(sce_comb[rownames(sce_mnn),])

# Cell annotation
cell_annot <- colData(sce_comb) %>%
  data.frame %>% 
  mutate(
    sample = paste(site, mouse_id, sep = '_')
  )
```


## UMAP final
### Site
```{r final-figures-umap-site, fig.asp = 0.6}
plotUMAP(sce_mnn, colour_by="site_caps", point_alpha = 0.17, point_size = 0.05) + 
  scale_color_manual(values = site_type_palette) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 3))) +
  labs(
    color = '',
    title = paste0('UMAP of all cells (',scales::number(ncol(sce_mnn), big.mark = "'"),' cells)')
  ) +
  theme(
    plot.title = element_text(size = 8, face = 'plain'),
    axis.line = element_line(linewidth = rel(0.5)),
    axis.text=element_blank(),
    axis.ticks=element_blank()#,
    # axis.line =element_blank()
  )
```

### Cell type individual level
Showing only cell types with >= 50 cells
```{r final-figures-umap-celltype-ind, fig.asp = 0.6}
celltype_label <- 'celltype_immgen_pruned.labels'
use_celltypes <-  colData(sce_mnn)[,celltype_label] %>% table %>% data.frame %>% filter(Freq >= 50) %>% collect %>% .[['.']] %>% as.character
use_sce <- sce_mnn[,colData(sce_mnn)[,celltype_label] %in% use_celltypes]
use_sce[[celltype_label]] <- factor(use_sce[[celltype_label]], levels = names(celltype_palette)[names(celltype_palette) %in% use_celltypes])

plotUMAP(use_sce, colour_by = celltype_label, point_alpha = 0.17, point_size = 0.05) + 
  scale_color_manual(values = celltype_palette) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 3))) +
  labs(
    color = '',
    title = paste0('UMAP of selected cell types (',scales::number(ncol(use_sce), big.mark = "'"),' cells)')
  ) +
  theme(
    plot.title = element_text(size = 8, face = 'plain'),
    axis.line = element_line(linewidth = rel(0.5)),
    axis.text=element_blank(),
    axis.ticks=element_blank()#,
    # axis.line =element_blank()
  )
```



## Lung : Differential abundance of cell types 
Differential abundance of cell types between normal lung tissue and TME in lung metastasis

```{r da_dist_site_conf-lung}
ct_label <- 'celltype_immgen_pruned.labels'
```

### Statistical analysis
Differential abundance using `speckle::propeller` package
```{r da_propeller_immgen-lung}
use_data <- cell_annot

# Remove low abundant cell types 
keep_labels <- table(use_data[[ct_label]]) %>% data.frame %>% filter(Freq >= 100) %>% pull(Var1)
keep_rows <- use_data[[ct_label]] %in% keep_labels
use_data <- use_data[keep_rows,]

# Run propeller Global
propeller_global <- propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$site) %>% 
  rownames_to_column('cell_type')


# Run pairwise propeller to obtaib P-values
site_levels <- use_data$site_caps %>% unique
# site_combn <- combn(site_levels, 2, simplify = FALSE)
site_combn <- list(
  c('Lung Normal', 'Lung TME')
  # c('Liver Normal', 'Liver TME')
)
sc <-  site_combn[[1]]
propeller_pairwise <-foreach(sc = site_combn, .combine = rbind) %do% {
  use_data <- cell_annot %>%
    filter(site_caps %in% sc) %>%
    mutate(
      group = ifelse(site_caps == sc[1], 'group1', 'group2')
    )
  keep_rows <- use_data[[ct_label]] %in% keep_labels
  use_data <- use_data[keep_rows,]
  propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$group) %>%
    mutate(
      group1 = sc[1],
      group2 = sc[2],
      FDR.sign = symnum(FDR, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001,0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "")),
      p.sign = symnum(P.Value, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", ""))
    ) %>%
    dplyr::rename(cell_type = BaselineProp.clusters)
}

```

Table of results with pairwise comparison between sites for cluster abundance. 
```{r da_propeller_immgen_tab-lung}
propeller_pairwise %>%
  arrange(FDR) %>% 
  datatable(., 
            rownames = FALSE, 
            filter = 'top', 
            caption = 'Pairwise comparison between cell types for cluster abundance using speckle::propeller',
            extensions = 'Buttons', 
            options = list(
              dom = 'Blfrtip',
              buttons = c('csv', 'excel')
            )) %>% 
  formatPercentage(names(propeller_pairwise)[2:4], digits = 3) %>% 
  formatRound(names(propeller_pairwise)[5:6], digits = 3, interval = -1000) %>% 
  formatSignif(c('P.Value', 'FDR'), digits = 3)
```


### Distribution by site
```{r da_dist_site_data_immgen-lung}
# Create data
use_data <- cell_annot %>% 
  filter(site_caps %in% unlist(site_combn)) %>% 
  dplyr::rename('celltype_label' = ct_label) %>% 
  group_by(site_caps, celltype_label) %>% 
  summarise(freq = n()) %>% 
  dplyr::rename(cell_type = celltype_label) %>% 
  na.omit()

# Remove low abundant cell types
cell_type_freq <- use_data %>% group_by(cell_type) %>% summarise(totalfreq = sum(freq))
site_filtered <- use_data %>% group_by(site_caps) %>% summarise(freq = sum(freq)) %>% filter(freq >= 500)
use_data <- use_data %>% 
  filter(site_caps %in% site_filtered$site_caps) %>% 
  left_join(cell_type_freq) %>% 
  filter(totalfreq >= 50) %>% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data <- use_data %>% 
  arrange(site_caps, cell_type)%>% 
  group_by(site_caps) %>%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion < 0.03, '', cell_type_label)
    
  )

```

#### Barplot stacked
```{r da_barplot_site_stacked_immgen-lung}
arranged_data %>% 
  ggplot(aes(x = site_caps, y = proportion, fill = cell_type)) +
  geom_col(alpha = 0.85, color = 'black', linewidth = 0.2) +
  scale_fill_manual(values = celltype_palette) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  # geom_text(aes(y = label_y, label = cell_type_label), colour = "white", size = geom_text_size) +
  theme(axis.ticks.x = element_blank()) +
  labs(
    fill = NULL, 
    x = NULL,
    y = 'Proportion of cells'
  )
```

#### Barplot facet
```{r da_barplot_site_celltypefacet_immgen-lung, fig.width=5.2}
arranged_data %>% 
  ggplot(aes(x = site_caps, y = proportion, fill = site_caps, group = cell_type)) +
  geom_col(alpha = 0.85, color = 'black', size = 0.2) +
  scale_fill_manual(values = site_palette) +
  scale_y_continuous(expand = c(0.02, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), ncol = 3, scales = 'free_y') +
  theme(
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x = element_blank(),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    panel.border = element_rect(colour = "black", size=0.5),
    strip.background = element_blank()
  ) +
  labs(
    fill = NULL, 
    x = NULL,
    y = 'Proportion of cells'
  ) #+
  # guides(fill = 'none')
```

#### Piechart
```{r da_piechart_site_celltypefacet_immgen-lung, fig.width = 4, fig.asp = 0.3}
arranged_data %>% 
  ggplot(aes(x = "", y = proportion, fill = cell_type)) +
  # geom_bar(stat="identity", width=1, color="white") +
  geom_bar(stat="identity", width=1) +
  coord_polar("y", start=0) +
  scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = '', 
    x = NULL,
    y = NULL
  )

```

#### Doughnut chart
```{r da_doughnut_site_celltypefacet_immgen-lung, fig.width = 4, fig.asp = 0.3}
i <- unique(arranged_data$site_caps)[1]
use_data <- foreach(i = unique(arranged_data$site_caps), .combine = rbind) %do% {
  x <- arranged_data %>% 
    filter(site_caps == i) %>% 
    arrange(cell_type)
  x$ymax <- cumsum(x$proportion)
  x$ymin <- c(0, head(x$ymax, n = -1))
  x
}

use_data %>% 
  ggplot(aes(ymax=ymax, ymin=ymin, xmax=4, xmin=3, fill=cell_type)) +
  geom_rect() +
  coord_polar(theta="y") + # Try to remove that to understand how the chart is built initially
  xlim(c(2, 4)) + # Try to remove that to see how to make a pie chart
  scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = '', 
    x = NULL,
    y = NULL
  )
```

### Distribution by site and mouse
```{r da_dist_site_and_mouse_data_immgen-lung}
# Create data
use_data <- cell_annot %>% 
  filter(site_caps %in% unlist(site_combn)) %>% 
  dplyr::rename('celltype_label' = ct_label) %>% 
  group_by(site_caps, mouse_id, celltype_label) %>% 
  summarise(freq = n()) %>% 
  dplyr::rename(cell_type = celltype_label) %>% 
  na.omit()

# Remove low abundant cell types and sites
cell_type_freq <- use_data %>% group_by(cell_type) %>% summarise(totalfreq = sum(freq))
site_filtered <- use_data %>% group_by(site_caps) %>% summarise(freq = sum(freq)) %>% filter(freq >= 500)
use_data <- use_data %>% 
  filter(site_caps %in% site_filtered$site_caps) %>% 
  left_join(cell_type_freq) %>% 
  filter(totalfreq >= 50) %>% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data <- use_data %>% 
  arrange(site_caps, cell_type)%>% 
  group_by(site_caps, mouse_id) %>%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion < 0.03, '', cell_type_label)
  )
```



#### Stripchart-Boxplot
```{r da_stripchart_site_and_mouse_celltypefacet1_immgen-lung, fig.asp = 0.8}
max_prop_cell_type <- arranged_data %>% group_by(cell_type) %>% summarise(max_proportion = max(proportion))
stat_test <- propeller_pairwise %>% 
  filter(group1 %in% arranged_data$site_caps & group2 %in%  arranged_data$site_caps) %>%
  left_join(max_prop_cell_type) %>% 
  arrange(cell_type, group1, group2) %>% 
  na.omit() %>% 
  filter(FDR < 0.05)
  
add_count_y <- stat_test %>% group_by(cell_type) %>% reframe(group1 = group1, group2 = group2, n = 1:n())
stat_test <- stat_test %>% left_join(add_count_y) %>% mutate(y.position = max_proportion + (n*max_proportion/10))


arranged_data %>% 
  ggplot(aes(x = site_caps, y = proportion, color = site_caps)) +
  geom_boxplot(alpha = 0.4, color = 'black', aes(fill = site_caps), 
               show.legend = FALSE, size = 0.25, fatten = 1) +
  # geom_point(alpha = 1, size = 2, shape = 21, color = 'black', aes(fill = site_caps)) +
  geom_quasirandom(method="smiley", width = 0.1, alpha = 1, size = 1, 
                   shape = 21, color = 'black', aes(fill = site_caps), stroke = 0.4) +
  # geom_quasirandom(method="smiley", width = 0.1, alpha = 1, size = 1, aes(color = site_caps)) +
  # scale_fill_OkabeIto() +
  # scale_color_OkabeIto() +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  # scale_y_continuous(expand = c(0, 0)) +
  scale_y_continuous(expand = c(0.02, 0, 0.08, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), scales = 'free_y', ncol = 2) +
  theme(
    # strip.background = element_blank(),
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    panel.border = element_rect(colour = "black", size=0.5),
    strip.background = element_blank()
  ) +
  # guides( color = 'none') +
  guides(fill = guide_legend(override.aes = list(size = 3))) +
  labs(
    fill = NULL,
    x = NULL,
    y = 'Proportion of cells'
  ) +
  stat_pvalue_manual(stat_test, label = 'FDR.sign', size = 2)
```




## Liver : Differential abundance of cell types 
Differential abundance of cell types between normal lung tissue and TME in lung metastasis

```{r da_dist_site_conf-liver}
ct_label <- 'celltype_immgen_pruned.labels'
```

### Statistical analysis
Differential abundance using `speckle::propeller` package
```{r da_propeller_immgen-liver}
use_data <- cell_annot

# Remove low abundant cell types 
keep_labels <- table(use_data[[ct_label]]) %>% data.frame %>% filter(Freq >= 100) %>% pull(Var1)
keep_rows <- use_data[[ct_label]] %in% keep_labels
use_data <- use_data[keep_rows,]

# Run propeller Global
propeller_global <- propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$site) %>% 
  rownames_to_column('cell_type')


# Run pairwise propeller to obtaib P-values
site_levels <- use_data$site_caps %>% unique
# site_combn <- combn(site_levels, 2, simplify = FALSE)
site_combn <- list(
  # c('Lung Normal', 'Lung TME')
  c('Liver Normal', 'Liver TME')
)
sc <-  site_combn[[1]]
propeller_pairwise <-foreach(sc = site_combn, .combine = rbind) %do% {
  use_data <- cell_annot %>%
    filter(site_caps %in% sc) %>%
    mutate(
      group = ifelse(site_caps == sc[1], 'group1', 'group2')
    )
  keep_rows <- use_data[[ct_label]] %in% keep_labels
  use_data <- use_data[keep_rows,]
  propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$group) %>%
    mutate(
      group1 = sc[1],
      group2 = sc[2],
      FDR.sign = symnum(FDR, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001,0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "")),
      p.sign = symnum(P.Value, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", ""))
    ) %>%
    dplyr::rename(cell_type = BaselineProp.clusters)
}

```

Table of results with pairwise comparison between sites for cluster abundance. 
```{r da_propeller_immgen_tab-liver}
propeller_pairwise %>%
  arrange(FDR) %>% 
  datatable(., 
            rownames = FALSE, 
            filter = 'top', 
            caption = 'Pairwise comparison between cell types for cluster abundance using speckle::propeller',
            extensions = 'Buttons', 
            options = list(
              dom = 'Blfrtip',
              buttons = c('csv', 'excel')
            )) %>% 
  formatPercentage(names(propeller_pairwise)[2:4], digits = 3) %>% 
  formatRound(names(propeller_pairwise)[5:6], digits = 3, interval = -1000) %>% 
  formatSignif(c('P.Value', 'FDR'), digits = 3)
```


### Distribution by site
```{r da_dist_site_data_immgen-liver}
# Create data
use_data <- cell_annot %>% 
  filter(site_caps %in% unlist(site_combn)) %>% 
  dplyr::rename('celltype_label' = ct_label) %>% 
  group_by(site_caps, celltype_label) %>% 
  summarise(freq = n()) %>% 
  dplyr::rename(cell_type = celltype_label) %>% 
  na.omit()

# Remove low abundant cell types
cell_type_freq <- use_data %>% group_by(cell_type) %>% summarise(totalfreq = sum(freq))
site_filtered <- use_data %>% group_by(site_caps) %>% summarise(freq = sum(freq)) %>% filter(freq >= 500)
use_data <- use_data %>% 
  filter(site_caps %in% site_filtered$site_caps) %>% 
  left_join(cell_type_freq) %>% 
  filter(totalfreq >= 50) %>% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data <- use_data %>% 
  arrange(site_caps, cell_type)%>% 
  group_by(site_caps) %>%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion < 0.03, '', cell_type_label)
    
  )

```

#### Barplot stacked
```{r da_barplot_site_stacked_immgen-liver}
arranged_data %>% 
  ggplot(aes(x = site_caps, y = proportion, fill = cell_type)) +
  geom_col(alpha = 0.85, color = 'black', size = 0.2) +
  scale_fill_manual(values = celltype_palette) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  # geom_text(aes(y = label_y, label = cell_type_label), colour = "white", size = geom_text_size) +
  theme(axis.ticks.x = element_blank()) +
  labs(
    fill = NULL, 
    x = NULL,
    y = 'Proportion of cells'
  )
```

#### Barplot facet
```{r da_barplot_site_celltypefacet_immgen-liver, fig.width=5.2}
arranged_data %>% 
  ggplot(aes(x = site_caps, y = proportion, fill = site_caps, group = cell_type)) +
  geom_col(alpha = 0.85, color = 'black', linewidth = 0.2) +
  scale_fill_manual(values = site_palette) +
  scale_y_continuous(expand = c(0.02, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), ncol = 3, scales = 'free_y') +
  theme(
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x = element_blank(),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    panel.border = element_rect(colour = "black", size=0.5),
    strip.background = element_blank()
  ) +
  labs(
    fill = NULL, 
    x = NULL,
    y = 'Proportion of cells'
  ) #+
  # guides(fill = 'none')
```

#### Piechart
```{r da_piechart_site_celltypefacet_immgen-liver, fig.width = 4, fig.asp = 0.3}
arranged_data %>% 
  ggplot(aes(x = "", y = proportion, fill = cell_type)) +
  # geom_bar(stat="identity", width=1, color="white") +
  geom_bar(stat="identity", width=1) +
  coord_polar("y", start=0) +
  scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = '', 
    x = NULL,
    y = NULL
  )

```

#### Doughnut chart
```{r da_doughnut_site_celltypefacet_immgen-liver, fig.width = 4, fig.asp = 0.3}
i <- unique(arranged_data$site_caps)[1]
use_data <- foreach(i = unique(arranged_data$site_caps), .combine = rbind) %do% {
  x <- arranged_data %>% 
    filter(site_caps == i) %>% 
    arrange(cell_type)
  x$ymax <- cumsum(x$proportion)
  x$ymin <- c(0, head(x$ymax, n = -1))
  x
}

use_data %>% 
  ggplot(aes(ymax=ymax, ymin=ymin, xmax=4, xmin=3, fill=cell_type)) +
  geom_rect() +
  coord_polar(theta="y") + # Try to remove that to understand how the chart is built initially
  xlim(c(2, 4)) + # Try to remove that to see how to make a pie chart
  scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = '', 
    x = NULL,
    y = NULL
  )
```

### Distribution by site and mouse
```{r da_dist_site_and_mouse_data_immgen-liver}
# Create data
use_data <- cell_annot %>% 
  filter(site_caps %in% unlist(site_combn)) %>% 
  dplyr::rename('celltype_label' = ct_label) %>% 
  group_by(site_caps, mouse_id, celltype_label) %>% 
  summarise(freq = n()) %>% 
  dplyr::rename(cell_type = celltype_label) %>% 
  na.omit()

# Remove low abundant cell types and sites
cell_type_freq <- use_data %>% group_by(cell_type) %>% summarise(totalfreq = sum(freq))
site_filtered <- use_data %>% group_by(site_caps) %>% summarise(freq = sum(freq)) %>% filter(freq >= 500)
use_data <- use_data %>% 
  filter(site_caps %in% site_filtered$site_caps) %>% 
  left_join(cell_type_freq) %>% 
  filter(totalfreq >= 50) %>% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data <- use_data %>% 
  arrange(site_caps, cell_type)%>% 
  group_by(site_caps, mouse_id) %>%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion < 0.03, '', cell_type_label)
  )
```



#### Stripchart-Boxplot
```{r da_stripchart_site_and_mouse_celltypefacet1_immgen-liver, fig.asp = 0.8}
max_prop_cell_type <- arranged_data %>% group_by(cell_type) %>% summarise(max_proportion = max(proportion))
stat_test <- propeller_pairwise %>% 
  filter(group1 %in% arranged_data$site_caps & group2 %in%  arranged_data$site_caps) %>%
  left_join(max_prop_cell_type) %>% 
  arrange(cell_type, group1, group2) %>% 
  na.omit() %>% 
  filter(FDR < 0.05)
  
add_count_y <- stat_test %>% group_by(cell_type) %>% reframe(group1 = group1, group2 = group2, n = 1:n())
stat_test <- stat_test %>% left_join(add_count_y) %>% mutate(y.position = max_proportion + (n*max_proportion/10))


arranged_data %>% 
  ggplot(aes(x = site_caps, y = proportion, color = site_caps)) +
  geom_boxplot(alpha = 0.4, color = 'black', aes(fill = site_caps), 
               show.legend = FALSE, size = 0.25, fatten = 1) +
  # geom_point(alpha = 1, size = 2, shape = 21, color = 'black', aes(fill = site_caps)) +
  geom_quasirandom(method="smiley", width = 0.1, alpha = 1, size = 1, 
                   shape = 21, color = 'black', aes(fill = site_caps), stroke = 0.4) +
  # geom_quasirandom(method="smiley", width = 0.1, alpha = 1, size = 1, aes(color = site_caps)) +
  # scale_fill_OkabeIto() +
  # scale_color_OkabeIto() +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  # scale_y_continuous(expand = c(0, 0)) +
  scale_y_continuous(expand = c(0.02, 0, 0.08, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), scales = 'free_y', ncol = 2) +
  theme(
    # strip.background = element_blank(),
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    panel.border = element_rect(colour = "black", size=0.5),
    strip.background = element_blank()
  ) +
  # guides( color = 'none') +
  guides(fill = guide_legend(override.aes = list(size = 3))) +
  labs(
    fill = NULL,
    x = NULL,
    y = 'Proportion of cells'
  ) +
  stat_pvalue_manual(stat_test, label = 'FDR.sign', size = 2)
```




## Lung: TME/Normal markers for each cell type

For each cell type we tried to identify potential markers to distinguish between Normal and TME cells
```{r lung-origin-celltype-markers-conf}
use_site <- 'lung'
n_top_mk <- 20
celltype_label <- 'celltype_immgen_pruned.labels'
# names(score_markers$origin_site_celltype_markers[[use_site]])
j <- score_markers$origin_site_celltype_markers[[use_site]][[1]][[1]]
mk_list <- foreach(i = score_markers$origin_site_celltype_markers[[use_site]]) %do% {
  foreach(j = i, .combine = c) %do% {
    j %>% data.frame %>% arrange(desc(mean.AUC)) %>% head(n_top_mk) %>% rownames()
  } %>% unique
}
names(mk_list) <-names(score_markers$origin_site_celltype_markers[[use_site]])

# Save scoreMarkers to Excel
# File name summary
rmd_file <- current_input()
if(is.null(rmd_file))
  rmd_file <- 'tmp'

# Generate file
file_xlsx <- file.path(
  './docs/file', rmd_file, 
  paste0('scoreMarkers_', use_site, '.xlsx')
  )
dir.create(dirname(file_xlsx), recursive = TRUE, showWarnings = FALSE)

# Generate workbook
wb <- createWorkbook()

# Create DfF of results
i <- score_markers$origin_site_celltype_markers[[use_site]][[1]]
use_mk_list <- score_markers$origin_site_celltype_markers[[use_site]]
x <- foreach(use_ct = names(use_mk_list)) %do% {
  foreach(use_condition = names(use_mk_list[[use_ct]])) %do% {
    sheet_name <- paste(use_ct, use_condition)
    addWorksheet(wb, paste(use_ct, use_condition))
    res <- use_mk_list[[use_ct]][[use_condition]] %>% data.frame %>% 
      arrange(desc(mean.AUC)) %>% 
      dplyr::select(gene_name, is.mito, is.ribo, 
                    self.average, other.average, self.detected, other.detected,
                    mean.AUC,
                    everything())
    writeData(wb, sheet_name, res)
  }
}

saveWorkbook(wb, file_xlsx, TRUE)
```

The tables of results can be downloaded using this [**link**](`r gsub("docs/", "" , file_xlsx)`)


### DC
In the plot below, color represent the Z-score based on average expression of the gene in each group. Z score is limited to -2 to 2 range. Point size represent the proportion of cells with detected expression.
```{r lung-origin-celltype-markers-dc-dotplot, fig.dim = c(8.12, 2.096)}
use_celltype_label <- 'DC'
use_rows <- mk_list[[use_celltype_label]]
use_cols <- (sce_comb[[celltype_label]] == use_celltype_label & !is.na(sce_comb[[celltype_label]]))
use_sce <- sce_comb[use_rows, use_cols]
rownames(use_sce) <- rowData(use_sce)$gene_name
res_plot <- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group='origin',
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data <- ggplot_build(res_plot)$plot$data %>% 
  mutate(Group = factor(Group, levels = rev(c('normal', 'tme'))))

# Create color scale
group <- retrieveCellInfo(use_sce, 'origin', search="colData")$value
ids <- DataFrame(group=group)
summarized <- summarizeAssayByGroup(
        assay(use_sce, "logcounts")[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c("mean", "prop.detected"),
        threshold=0)
    
ave <- assay(summarized, "mean")
use_heatmap_scale <- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot <- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = 'black') +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = "black", fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = '',
    y = '',
    color = 'Z-score',
    size = 'Proportion\nof cells'
  )

res_plot + theme( legend.position = "none")
```

### Endothelial cells
In the plot below, color represent the Z-score based on average expression of the gene in each group. Z score is limited to -2 to 2 range. Point size represent the proportion of cells with detected expression.
```{r lung-origin-celltype-markers-endothelial-dotplot, fig.dim = c(8.12, 2.096)}
use_celltype_label <- 'Endothelial cells'
use_rows <- mk_list[[use_celltype_label]]
use_cols <- (sce_comb[[celltype_label]] == use_celltype_label & !is.na(sce_comb[[celltype_label]]))
use_sce <- sce_comb[use_rows, use_cols]
rownames(use_sce) <- rowData(use_sce)$gene_name
res_plot <- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group='origin',
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data <- ggplot_build(res_plot)$plot$data %>% 
  mutate(Group = factor(Group, levels = rev(c('normal', 'tme'))))

# Create color scale
group <- retrieveCellInfo(use_sce, 'origin', search="colData")$value
ids <- DataFrame(group=group)
summarized <- summarizeAssayByGroup(
        assay(use_sce, "logcounts")[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c("mean", "prop.detected"),
        threshold=0)
    
ave <- assay(summarized, "mean")
use_heatmap_scale <- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot <- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = 'black') +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = "black", fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = '',
    y = '',
    color = 'Z-score',
    size = 'Proportion\nof cells'
  )

res_plot + theme( legend.position = "none")
```

### Fibroblasts
In the plot below, color represent the Z-score based on average expression of the gene in each group. Z score is limited to -2 to 2 range. Point size represent the proportion of cells with detected expression.
```{r lung-origin-celltype-markers-fibroblasts-dotplot, fig.dim = c(8.12, 2.096)}
use_celltype_label <- 'Fibroblasts'
use_rows <- mk_list[[use_celltype_label]]
use_cols <- (sce_comb[[celltype_label]] == use_celltype_label & !is.na(sce_comb[[celltype_label]]))
use_sce <- sce_comb[use_rows, use_cols]
rownames(use_sce) <- rowData(use_sce)$gene_name
res_plot <- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group='origin',
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data <- ggplot_build(res_plot)$plot$data %>% 
  mutate(Group = factor(Group, levels = rev(c('normal', 'tme'))))

# Create color scale
group <- retrieveCellInfo(use_sce, 'origin', search="colData")$value
ids <- DataFrame(group=group)
summarized <- summarizeAssayByGroup(
        assay(use_sce, "logcounts")[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c("mean", "prop.detected"),
        threshold=0)
    
ave <- assay(summarized, "mean")
use_heatmap_scale <- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot <- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = 'black') +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = "black", fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = '',
    y = '',
    color = 'Z-score',
    size = 'Proportion\nof cells'
  )

res_plot + theme( legend.position = "none")
```

### Macrophages
In the plot below, color represent the Z-score based on average expression of the gene in each group. Z score is limited to -2 to 2 range. Point size represent the proportion of cells with detected expression.
```{r lung-origin-celltype-markers-macrophages-dotplot, fig.dim = c(8.12, 2.096)}
use_celltype_label <- 'Macrophages'
use_rows <- mk_list[[use_celltype_label]]
use_cols <- (sce_comb[[celltype_label]] == use_celltype_label & !is.na(sce_comb[[celltype_label]]))
use_sce <- sce_comb[use_rows, use_cols]
rownames(use_sce) <- rowData(use_sce)$gene_name
res_plot <- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group='origin',
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data <- ggplot_build(res_plot)$plot$data %>% 
  mutate(Group = factor(Group, levels = rev(c('normal', 'tme'))))

# Create color scale
group <- retrieveCellInfo(use_sce, 'origin', search="colData")$value
ids <- DataFrame(group=group)
summarized <- summarizeAssayByGroup(
        assay(use_sce, "logcounts")[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c("mean", "prop.detected"),
        threshold=0)
    
ave <- assay(summarized, "mean")
use_heatmap_scale <- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot <- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = 'black') +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = "black", fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = '',
    y = '',
    color = 'Z-score',
    size = 'Proportion\nof cells'
  )

res_plot + theme( legend.position = "none")
```

### Monocytes
In the plot below, color represent the Z-score based on average expression of the gene in each group. Z score is limited to -2 to 2 range. Point size represent the proportion of cells with detected expression.
```{r lung-origin-celltype-markers-monocytes-dotplot, fig.dim = c(8.12, 2.096)}
use_celltype_label <- 'Monocytes'
use_rows <- mk_list[[use_celltype_label]]
use_cols <- (sce_comb[[celltype_label]] == use_celltype_label & !is.na(sce_comb[[celltype_label]]))
use_sce <- sce_comb[use_rows, use_cols]
rownames(use_sce) <- rowData(use_sce)$gene_name
res_plot <- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group='origin',
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data <- ggplot_build(res_plot)$plot$data %>% 
  mutate(Group = factor(Group, levels = rev(c('normal', 'tme'))))

# Create color scale
group <- retrieveCellInfo(use_sce, 'origin', search="colData")$value
ids <- DataFrame(group=group)
summarized <- summarizeAssayByGroup(
        assay(use_sce, "logcounts")[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c("mean", "prop.detected"),
        threshold=0)
    
ave <- assay(summarized, "mean")
use_heatmap_scale <- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot <- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = 'black') +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = "black", fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = '',
    y = '',
    color = 'Z-score',
    size = 'Proportion\nof cells'
  )

res_plot + theme( legend.position = "none")
```

### Neutrophils
In the plot below, color represent the Z-score based on average expression of the gene in each group. Z score is limited to -2 to 2 range. Point size represent the proportion of cells with detected expression.
```{r lung-origin-celltype-markers-neutrophils-dotplot, fig.dim = c(8.12, 2.096)}
use_celltype_label <- 'Neutrophils'
use_rows <- mk_list[[use_celltype_label]]
use_cols <- (sce_comb[[celltype_label]] == use_celltype_label & !is.na(sce_comb[[celltype_label]]))
use_sce <- sce_comb[use_rows, use_cols]
rownames(use_sce) <- rowData(use_sce)$gene_name
res_plot <- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group='origin',
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data <- ggplot_build(res_plot)$plot$data %>% 
  mutate(Group = factor(Group, levels = rev(c('normal', 'tme'))))

# Create color scale
group <- retrieveCellInfo(use_sce, 'origin', search="colData")$value
ids <- DataFrame(group=group)
summarized <- summarizeAssayByGroup(
        assay(use_sce, "logcounts")[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c("mean", "prop.detected"),
        threshold=0)
    
ave <- assay(summarized, "mean")
use_heatmap_scale <- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot <- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = 'black') +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = "black", fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = '',
    y = '',
    color = 'Z-score',
    size = 'Proportion\nof cells'
  )

res_plot + theme( legend.position = "none")
```

### Legend
```{r lung-origin-celltype-markers-legend-dotplot, fig.dim = c(3.12, 3.096)}
legend <- cowplot::get_legend(res_plot)
grid.newpage()
grid.draw(legend)
```




## Liver: TME/Normal markers for each cell type

For each cell type we tried to identify potential markers to distinguish between Normal and TME cells
```{r liver-origin-celltype-markers-conf}
use_site <- 'liver'
n_top_mk <- 20
celltype_label <- 'celltype_immgen_pruned.labels'
# names(score_markers$origin_site_celltype_markers[[use_site]])
j <- score_markers$origin_site_celltype_markers[[use_site]][[1]][[1]]
mk_list <- foreach(i = score_markers$origin_site_celltype_markers[[use_site]]) %do% {
  foreach(j = i, .combine = c) %do% {
    j %>% data.frame %>% arrange(desc(mean.AUC)) %>% head(n_top_mk) %>% rownames()
  } %>% unique
}
names(mk_list) <-names(score_markers$origin_site_celltype_markers[[use_site]])



# Save scoreMarkers to Excel
# File name summary
rmd_file <- current_input()
if(is.null(rmd_file))
  rmd_file <- 'tmp'
# file_xlsx_list <- list()

# Generate file
file_xlsx <- file.path(
  './docs/file', rmd_file, 
  paste0('scoreMarkers_', use_site, '.xlsx')
  )
dir.create(dirname(file_xlsx), recursive = TRUE, showWarnings = FALSE)

# Generate workbook
wb <- createWorkbook()

# Create DfF of results
i <- score_markers$origin_site_celltype_markers[[use_site]][[1]]
use_mk_list <- score_markers$origin_site_celltype_markers[[use_site]]
x <- foreach(use_ct = names(use_mk_list)) %do% {
  foreach(use_condition = names(use_mk_list[[use_ct]])) %do% {
    sheet_name <- paste(use_ct, use_condition)
    addWorksheet(wb, paste(use_ct, use_condition))
    res <- use_mk_list[[use_ct]][[use_condition]] %>% data.frame %>% 
      arrange(desc(mean.AUC)) %>% 
      dplyr::select(gene_name, is.mito, is.ribo, 
                    self.average, other.average, self.detected, other.detected,
                    mean.AUC,
                    everything())
    writeData(wb, sheet_name, res)
  }
}

saveWorkbook(wb, file_xlsx, TRUE)
```

The tables of results can be downloaded using this [**link**](`r gsub("docs/", "" , file_xlsx)`)

### DC
In the plot below, color represent the Z-score based on average expression of the gene in each group. Z score is limited to -2 to 2 range. Point size represent the proportion of cells with detected expression.
```{r liver-origin-celltype-markers-dc-dotplot, fig.dim = c(8.12, 2.096)}
use_celltype_label <- 'DC'
use_rows <- mk_list[[use_celltype_label]]
use_cols <- (sce_comb[[celltype_label]] == use_celltype_label & !is.na(sce_comb[[celltype_label]]))
use_sce <- sce_comb[use_rows, use_cols]
rownames(use_sce) <- rowData(use_sce)$gene_name
res_plot <- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group='origin',
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data <- ggplot_build(res_plot)$plot$data %>% 
  mutate(Group = factor(Group, levels = rev(c('normal', 'tme'))))

# Create color scale
group <- retrieveCellInfo(use_sce, 'origin', search="colData")$value
ids <- DataFrame(group=group)
summarized <- summarizeAssayByGroup(
        assay(use_sce, "logcounts")[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c("mean", "prop.detected"),
        threshold=0)
    
ave <- assay(summarized, "mean")
use_heatmap_scale <- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot <- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = 'black') +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = "black", fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = '',
    y = '',
    color = 'Z-score',
    size = 'Proportion\nof cells'
  )

res_plot + theme( legend.position = "none")
```

### Endothelial cells
In the plot below, color represent the Z-score based on average expression of the gene in each group. Z score is limited to -2 to 2 range. Point size represent the proportion of cells with detected expression.
```{r liver-origin-celltype-markers-endothelial-dotplot, fig.dim = c(8.12, 2.096)}
use_celltype_label <- 'Endothelial cells'
use_rows <- mk_list[[use_celltype_label]]
use_cols <- (sce_comb[[celltype_label]] == use_celltype_label & !is.na(sce_comb[[celltype_label]]))
use_sce <- sce_comb[use_rows, use_cols]
rownames(use_sce) <- rowData(use_sce)$gene_name
res_plot <- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group='origin',
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data <- ggplot_build(res_plot)$plot$data %>% 
  mutate(Group = factor(Group, levels = rev(c('normal', 'tme'))))

# Create color scale
group <- retrieveCellInfo(use_sce, 'origin', search="colData")$value
ids <- DataFrame(group=group)
summarized <- summarizeAssayByGroup(
        assay(use_sce, "logcounts")[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c("mean", "prop.detected"),
        threshold=0)
    
ave <- assay(summarized, "mean")
use_heatmap_scale <- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot <- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = 'black') +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = "black", fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = '',
    y = '',
    color = 'Z-score',
    size = 'Proportion\nof cells'
  )

res_plot + theme( legend.position = "none")
```

### Macrophages
In the plot below, color represent the Z-score based on average expression of the gene in each group. Z score is limited to -2 to 2 range. Point size represent the proportion of cells with detected expression.
```{r liver-origin-celltype-markers-macrophages-dotplot, fig.dim = c(8.12, 2.096)}
use_celltype_label <- 'Macrophages'
use_rows <- mk_list[[use_celltype_label]]
use_cols <- (sce_comb[[celltype_label]] == use_celltype_label & !is.na(sce_comb[[celltype_label]]))
use_sce <- sce_comb[use_rows, use_cols]
rownames(use_sce) <- rowData(use_sce)$gene_name
res_plot <- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group='origin',
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data <- ggplot_build(res_plot)$plot$data %>% 
  mutate(Group = factor(Group, levels = rev(c('normal', 'tme'))))

# Create color scale
group <- retrieveCellInfo(use_sce, 'origin', search="colData")$value
ids <- DataFrame(group=group)
summarized <- summarizeAssayByGroup(
        assay(use_sce, "logcounts")[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c("mean", "prop.detected"),
        threshold=0)
    
ave <- assay(summarized, "mean")
use_heatmap_scale <- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot <- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = 'black') +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = "black", fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = '',
    y = '',
    color = 'Z-score',
    size = 'Proportion\nof cells'
  )

res_plot + theme( legend.position = "none")
```

### Monocytes
In the plot below, color represent the Z-score based on average expression of the gene in each group. Z score is limited to -2 to 2 range. Point size represent the proportion of cells with detected expression.
```{r liver-origin-celltype-markers-monocytes-dotplot, fig.dim = c(8.12, 2.096)}
use_celltype_label <- 'Monocytes'
use_rows <- mk_list[[use_celltype_label]]
use_cols <- (sce_comb[[celltype_label]] == use_celltype_label & !is.na(sce_comb[[celltype_label]]))
use_sce <- sce_comb[use_rows, use_cols]
rownames(use_sce) <- rowData(use_sce)$gene_name
res_plot <- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group='origin',
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data <- ggplot_build(res_plot)$plot$data %>% 
  mutate(Group = factor(Group, levels = rev(c('normal', 'tme'))))

# Create color scale
group <- retrieveCellInfo(use_sce, 'origin', search="colData")$value
ids <- DataFrame(group=group)
summarized <- summarizeAssayByGroup(
        assay(use_sce, "logcounts")[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c("mean", "prop.detected"),
        threshold=0)
    
ave <- assay(summarized, "mean")
use_heatmap_scale <- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot <- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = 'black') +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = "black", fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = '',
    y = '',
    color = 'Z-score',
    size = 'Proportion\nof cells'
  )

res_plot + theme( legend.position = "none")
```

### Neutrophils
In the plot below, color represent the Z-score based on average expression of the gene in each group. Z score is limited to -2 to 2 range. Point size represent the proportion of cells with detected expression.
```{r liver-origin-celltype-markers-neutrophils-dotplot, fig.dim = c(8.12, 2.096)}
use_celltype_label <- 'Neutrophils'
use_rows <- mk_list[[use_celltype_label]]
use_cols <- (sce_comb[[celltype_label]] == use_celltype_label & !is.na(sce_comb[[celltype_label]]))
use_sce <- sce_comb[use_rows, use_cols]
rownames(use_sce) <- rowData(use_sce)$gene_name
res_plot <- plotDots(
    use_sce,
    features=rownames(use_sce), 
    group='origin',
    center=TRUE, scale = TRUE,
    zlim = c(-2,2)
  )

# Get data
plot_data <- ggplot_build(res_plot)$plot$data %>% 
  mutate(Group = factor(Group, levels = rev(c('normal', 'tme'))))

# Create color scale
group <- retrieveCellInfo(use_sce, 'origin', search="colData")$value
ids <- DataFrame(group=group)
summarized <- summarizeAssayByGroup(
        assay(use_sce, "logcounts")[as.character(rownames(use_sce)), , drop = FALSE],
        ids=ids, 
        statistics=c("mean", "prop.detected"),
        threshold=0)
    
ave <- assay(summarized, "mean")
use_heatmap_scale <- heatmap_scale(ave, 
                                   center=TRUE, 
                                   scale=TRUE, 
                                   colour=NULL, 
                                   zlim=c(-2,2))

res_plot <- ggplot(plot_data) +
  geom_point(aes(x=.data$Group, 
                 y=.data$Feature, 
                 size=.data$NumDetected, 
                 # col=.data$Average,
                 fill=.data$Average),
             pch = 21,
             color = 'black') +
  scale_size(limits=c(0, max(plot_data$NumDetected))) +
  # use_heatmap_scale$colour_scale +
  use_heatmap_scale$fill_scale +
  # theme_minimal() +
  theme(
    panel.border = element_rect(colour = "black", fill=NA, linewidth = 2*one_pt)
    ) +
  coord_flip() +
  scale_y_discrete(guide = guide_axis(angle = 45)) +
  labs(
    x = '',
    y = '',
    color = 'Z-score',
    size = 'Proportion\nof cells'
  )

res_plot + theme( legend.position = "none")
```

### Legend
```{r liver-origin-celltype-markers-legend-dotplot, fig.dim = c(3.12, 3.096)}
legend <- cowplot::get_legend(res_plot)
grid.newpage()
grid.draw(legend)
```




## Bmp2 gene expression in endothelial liver cells
```{r bmp2-expr-conf}
use_data <- cell_annot %>% 
  mutate(Bmp2 =  logcounts(sce_comb)['Bmp2',]) %>% 
  filter(grepl('Liver', site_caps)) %>% 
  filter(celltype_immgen_pruned.labels == 'Endothelial cells') 
```

### Violin plot
Black horizontal line represents the median expression.
```{r bmp2-expr-violin}
summary_stats <- use_data %>% 
  group_by(site_caps) %>% 
  summarise(median = median(Bmp2)) %>% 
  ungroup %>% 
  mutate(site_caps = gsub('.* ', '', site_caps)) %>% 
  column_to_rownames('site_caps')

use_data %>% 
  ggplot(aes(x = site_caps, y = Bmp2, fill = site_caps, color = site_caps)) +
  geom_violin(color = "gray60", alpha = 0.2, scale = "width", width = 0.8, show.legend = FALSE) + 
  geom_quasirandom(size = 0.2, alpha = 0.6, width=0.4, 
                   pch = 16, 
                   groupOnX=TRUE, bandwidth=1, show.legend = FALSE) +
  stat_compare_means(show.legend = FALSE, size = geom_text_size) +
  stat_summary(fun = median, fun.min = median, fun.max = median, 
               geom = "crossbar", width = 0.3, alpha = 1, color = 'black', 
               show.legend = FALSE) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  labs(
    x = NULL,
    y = 'Bmp2 expression',
    fill = NULL,
    color = NULL,
    caption = paste0(
      'Median expression in Normal: ', round(summary_stats['Normal', 'median'], 3), '\n',
      'Median expression in TME: ', round(summary_stats['TME', 'median'], 3)
    )
  )
```


### Density plot
Density plot that computes and draws kernel density estimate, which is a smoothed version of the histogram.
```{r bmp2-expr-density}
summary_stats <- use_data %>% 
  group_by(site_caps) %>% 
  summarise(median = median(Bmp2)) %>% 
  ungroup %>% 
  mutate(site_caps = gsub('.* ', '', site_caps)) %>% 
  column_to_rownames('site_caps')

use_data %>% 
  ggplot(aes(Bmp2, fill = site_caps, color = site_caps)) +
  geom_density(alpha = 0.5) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  scale_x_continuous(expand = expansion(mult = c(0, 0))) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.01))) +
  labs(
    y = 'Density',
    x = 'Bmp2 expression',
    fill = NULL,
    color = NULL,
    caption = paste0(
      'Median expression in Normal: ', round(summary_stats['Normal', 'median'], 3), '\n',
      'Median expression in TME: ', round(summary_stats['TME', 'median'], 3)
    )
  )
```

### Violin plot by mouse
Black horizontal line represents the median expression.
```{r bmp2-expr-violin-mouse}
summary_stats <- use_data %>% 
  group_by(site_caps) %>% 
  summarise(median = median(Bmp2))

use_data %>% 
  ggplot(aes(x = mouse_id, y = Bmp2, fill = site_caps, color = site_caps)) +
  geom_violin(color = "gray60", alpha = 0.2, scale = "width", width = 0.8, show.legend = FALSE) + 
  geom_quasirandom(size = 0.1, alpha = 0.6, width=0.4, 
                   pch = 16,
                   groupOnX=TRUE, bandwidth=1, show.legend = FALSE) +
  facet_grid(cols = vars(site_caps), scales = 'free_x') +
  stat_summary(fun = median, fun.min = median, fun.max = median, 
               geom = "crossbar", width = 0.3, alpha = 1, color = 'black', 
               show.legend = FALSE) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  theme(
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank()
  ) +
  theme_facet +
  labs(
    x = NULL,
    y = 'Bmp2 expression',
    fill = NULL,
    color = NULL
  )
```

### Stripchart-Boxplot
```{r bmp2-expr-stripchart}
use_data %>% 
  group_by(site_caps, mouse_id) %>%
  summarise(proportion = sum(Bmp2 > 0)/n()) %>% 
  ggplot(aes(x = site_caps, y = proportion, color = site_caps)) +
  geom_boxplot(alpha = 0.4, color = 'black', aes(fill = site_caps), 
               show.legend = FALSE, size = 0.25, fatten = 1) +
  geom_quasirandom(method="smiley", width = 0.1, alpha = 1, size = 1, 
                   shape = 21, color = 'black', aes(fill = site_caps), stroke = 0.4) +
  stat_compare_means(size = geom_text_size) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  scale_y_continuous(expand = c(0.02, 0, 0.08, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  theme(
    # axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    panel.border = element_rect(colour = "black", size=0.5),
    strip.background = element_blank()
  ) +
  guides( color = 'none', fill = 'none') +
  labs(
    fill = NULL,
    x = NULL,
    y = 'Proportion of cells with Bmp2 > 0 expression'
  )
```




## Granulin gene expression in lung macrophages
```{r grn-expr-conf}
use_data <- cell_annot %>% 
  mutate(Grn =  logcounts(sce_comb)['Grn',]) %>% 
  filter(grepl('Lung', site_caps)) %>% 
  filter(celltype_immgen_pruned.labels == 'Macrophages') 
```

### Violin plot
Black horizontal line represents the median expression.
```{r grn-expr-violin}
summary_stats <- use_data %>% 
  group_by(site_caps) %>% 
  summarise(median = median(Grn)) %>% 
  ungroup %>% 
  mutate(site_caps = gsub('.* ', '', site_caps)) %>% 
  column_to_rownames('site_caps')

use_data %>% 
  ggplot(aes(x = site_caps, y = Grn, fill = site_caps, color = site_caps)) +
  geom_violin(color = "gray60", alpha = 0.2, scale = "width", width = 0.8, show.legend = FALSE) + 
  geom_quasirandom(alpha = 0.6, width=0.4, 
                   pch = 16,
                   groupOnX=TRUE, bandwidth=1, show.legend = FALSE) +
  stat_compare_means(show.legend = FALSE, size = geom_text_size) +
  stat_summary(fun = median, fun.min = median, fun.max = median, 
               geom = "crossbar", width = 0.3, alpha = 1, color = 'black', 
               show.legend = FALSE) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  labs(
    x = NULL,
    y = 'Grn expression',
    fill = NULL,
    color = NULL,
    caption = paste0(
      'Median expression in Normal: ', round(summary_stats['Normal', 'median'], 3), '\n',
      'Median expression in TME: ', round(summary_stats['TME', 'median'], 3)
    )
  )
```


### Density plot
Density plot that computes and draws kernel density estimate, which is a smoothed version of the histogram.
```{r grn-expr-density}
summary_stats <- use_data %>% 
  group_by(site_caps) %>% 
  summarise(median = median(Grn)) %>% 
  ungroup %>% 
  mutate(site_caps = gsub('.* ', '', site_caps)) %>% 
  column_to_rownames('site_caps')

use_data %>% 
  ggplot(aes(Grn, fill = site_caps, color = site_caps)) +
  geom_density(alpha = 0.5) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  scale_x_continuous(expand = expansion(mult = c(0, 0))) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.01))) +
  labs(
    y = 'Density',
    x = 'Grn expression',
    fill = NULL,
    color = NULL,
    caption = paste0(
      'Median expression in Normal: ', round(summary_stats['Normal', 'median'], 3), '\n',
      'Median expression in TME: ', round(summary_stats['TME', 'median'], 3)
    )
  )
```


### Violin plot by mouse
Black horizontal line represents the median expression.
```{r grn-expr-violin-mouse}
summary_stats <- use_data %>% 
  group_by(site_caps) %>% 
  summarise(median = median(Grn))

use_data %>% 
  ggplot(aes(x = mouse_id, y = Grn, fill = site_caps, color = site_caps)) +
  geom_violin(color = "gray60", alpha = 0.2, scale = "width", width = 0.8, show.legend = FALSE) + 
  geom_quasirandom(alpha = 0.6, width=0.4, 
                   pch = 16,
                   groupOnX=TRUE, bandwidth=1, 
                   show.legend = FALSE) +
  facet_grid(cols = vars(site_caps), scales = 'free_x') +
  stat_summary(fun = median, fun.min = median, fun.max = median, 
               geom = "crossbar", width = 0.3, alpha = 1, color = 'black', 
               show.legend = FALSE) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  theme(
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank()
  ) +
  theme_facet +
  labs(
    x = NULL,
    y = 'Grn expression',
    fill = NULL,
    color = NULL
  )
```

### Stripchart-Boxplot
```{r grn-expr-stripchart}
use_data %>% 
  group_by(site_caps, mouse_id) %>%
  summarise(proportion = sum(Grn > 0)/n()) %>% 
  ggplot(aes(x = site_caps, y = proportion, color = site_caps)) +
  geom_boxplot(alpha = 0.4, color = 'black', aes(fill = site_caps), 
               show.legend = FALSE, size = 0.25, fatten = 1) +
  geom_quasirandom(method="smiley", width = 0.1, alpha = 1, size = 1, 
                   shape = 21, color = 'black', aes(fill = site_caps), stroke = 0.4) +
  stat_compare_means(size = geom_text_size) +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  scale_y_continuous(expand = c(0.02, 0, 0.08, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  theme(
    # axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    panel.border = element_rect(colour = "black", size=0.5),
    strip.background = element_blank()
  ) +
  guides( color = 'none', fill = 'none') +
  labs(
    fill = NULL,
    x = NULL,
    y = 'Proportion of cells with Grn > 0 expression'
  )
```


## Lung MF: Differential abundance of macrophages cell types 
Differential abundance of cell types between normal lung tissue and TME in lung metastasis

```{r mf_da_dist_site_conf-lung}
ct_label <- 'mf_immgen_pruned.labels.fine'
```

### Statistical analysis
Differential abundance using `speckle::propeller` package
```{r mf_da_propeller_immgen-lung}
use_data <- cell_annot %>% 
  filter(celltype_immgen_pruned.labels == 'Macrophages')

# Remove low abundant cell types 
keep_labels <- table(use_data[[ct_label]]) %>% data.frame %>% filter(Freq >= 100) %>% pull(Var1)
keep_rows <- use_data[[ct_label]] %in% keep_labels
use_data <- use_data[keep_rows,]

# Run propeller Global
propeller_global <- propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$site) %>% 
  rownames_to_column('cell_type')


# Run pairwise propeller to obtaib P-values
site_levels <- use_data$site_caps %>% unique
# site_combn <- combn(site_levels, 2, simplify = FALSE)
site_combn <- list(
  c('Lung Normal', 'Lung TME')
  # c('Liver Normal', 'Liver TME')
)
sc <-  site_combn[[1]]
propeller_pairwise <-foreach(sc = site_combn, .combine = rbind) %do% {
  use_data <- cell_annot %>%
    filter(site_caps %in% sc) %>%
    mutate(
      group = ifelse(site_caps == sc[1], 'group1', 'group2')
    )
  keep_rows <- use_data[[ct_label]] %in% keep_labels
  use_data <- use_data[keep_rows,]
  propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$group) %>%
    mutate(
      group1 = sc[1],
      group2 = sc[2],
      FDR.sign = symnum(FDR, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001,0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "")),
      p.sign = symnum(P.Value, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", ""))
    ) %>%
    dplyr::rename(cell_type = BaselineProp.clusters)
}

```

Table of results with pairwise comparison between sites for cluster abundance. 
```{r mf_da_propeller_immgen_tab-lung}
propeller_pairwise %>%
  arrange(FDR) %>% 
  datatable(., 
            rownames = FALSE, 
            filter = 'top', 
            caption = 'Pairwise comparison between cell types for cluster abundance using speckle::propeller',
            extensions = 'Buttons', 
            options = list(
              dom = 'Blfrtip',
              buttons = c('csv', 'excel')
            )) %>% 
  formatPercentage(names(propeller_pairwise)[2:4], digits = 3) %>% 
  formatRound(names(propeller_pairwise)[5:6], digits = 3, interval = -1000) %>% 
  formatSignif(c('P.Value', 'FDR'), digits = 3)
```


### Distribution by site
```{r mf_da_dist_site_data_immgen-lung}
# Create data
use_data <- cell_annot %>% 
  filter(site_caps %in% unlist(site_combn)) %>% 
  dplyr::rename('celltype_label' = ct_label) %>% 
  group_by(site_caps, celltype_label) %>% 
  summarise(freq = n()) %>% 
  dplyr::rename(cell_type = celltype_label) %>% 
  na.omit()

# Remove low abundant cell types
cell_type_freq <- use_data %>% group_by(cell_type) %>% summarise(totalfreq = sum(freq))
# site_filtered <- use_data %>% group_by(site_caps) %>% summarise(freq = sum(freq)) %>% filter(freq >= 500)
use_data <- use_data %>% 
  # filter(site_caps %in% site_filtered$site_caps) %>% 
  left_join(cell_type_freq) %>% 
  filter(totalfreq >= 50) %>% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data <- use_data %>% 
  arrange(site_caps, cell_type)%>% 
  group_by(site_caps) %>%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion < 0.03, '', cell_type_label)
    
  )

```

#### Barplot stacked
```{r mf_da_barplot_site_stacked_immgen-lung}
arranged_data %>% 
  ggplot(aes(x = site_caps, y = proportion, fill = cell_type)) +
  geom_col(alpha = 0.85, color = 'black', linewidth = 0.2) +
  # scale_fill_manual(values = celltype_palette) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  # geom_text(aes(y = label_y, label = cell_type_label), colour = "white", size = geom_text_size) +
  theme(axis.ticks.x = element_blank()) +
  labs(
    fill = NULL, 
    x = NULL,
    y = 'Proportion of cells'
  )
```

#### Barplot facet
```{r mf_da_barplot_site_celltypefacet_immgen-lung, fig.width=5.2}
arranged_data %>% 
  ggplot(aes(x = site_caps, y = proportion, fill = site_caps, group = cell_type)) +
  geom_col(alpha = 0.85, color = 'black', size = 0.2) +
  scale_fill_manual(values = site_type_palette) +
  scale_y_continuous(expand = c(0.02, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), ncol = 3, scales = 'free_y') +
  theme(
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x = element_blank(),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    panel.border = element_rect(colour = "black", size=0.5),
    strip.background = element_blank()
  ) +
  labs(
    fill = NULL, 
    x = NULL,
    y = 'Proportion of cells'
  ) #+
  # guides(fill = 'none')
```

#### Piechart
```{r mf_da_piechart_site_celltypefacet_immgen-lung, fig.width = 4, fig.asp = 0.3}
arranged_data %>% 
  ggplot(aes(x = "", y = proportion, fill = cell_type)) +
  # geom_bar(stat="identity", width=1, color="white") +
  geom_bar(stat="identity", width=1) +
  coord_polar("y", start=0) +
  # scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = '', 
    x = NULL,
    y = NULL
  )

```

#### Doughnut chart
```{r mf_da_doughnut_site_celltypefacet_immgen-lung, fig.width = 4, fig.asp = 0.3}
i <- unique(arranged_data$site_caps)[1]
use_data <- foreach(i = unique(arranged_data$site_caps), .combine = rbind) %do% {
  x <- arranged_data %>% 
    filter(site_caps == i) %>% 
    arrange(cell_type)
  x$ymax <- cumsum(x$proportion)
  x$ymin <- c(0, head(x$ymax, n = -1))
  x
}

use_data %>% 
  ggplot(aes(ymax=ymax, ymin=ymin, xmax=4, xmin=3, fill=cell_type)) +
  geom_rect() +
  coord_polar(theta="y") + # Try to remove that to understand how the chart is built initially
  xlim(c(2, 4)) + # Try to remove that to see how to make a pie chart
  # scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = '', 
    x = NULL,
    y = NULL
  )
```

### Distribution by site and mouse
```{r mf_da_dist_site_and_mouse_data_immgen-lung}
# Create data
use_data <- cell_annot %>% 
  filter(site_caps %in% unlist(site_combn)) %>% 
  dplyr::rename('celltype_label' = ct_label) %>% 
  group_by(site_caps, mouse_id, celltype_label) %>% 
  summarise(freq = n()) %>% 
  dplyr::rename(cell_type = celltype_label) %>% 
  na.omit()

# Remove low abundant cell types and sites
cell_type_freq <- use_data %>% group_by(cell_type) %>% summarise(totalfreq = sum(freq))
# site_filtered <- use_data %>% group_by(site_caps) %>% summarise(freq = sum(freq)) %>% filter(freq >= 500)
use_data <- use_data %>% 
  # filter(site_caps %in% site_filtered$site_caps) %>% 
  left_join(cell_type_freq) %>% 
  filter(totalfreq >= 50) %>% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data <- use_data %>% 
  arrange(site_caps, cell_type)%>% 
  group_by(site_caps, mouse_id) %>%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion < 0.03, '', cell_type_label)
  )
```



#### Stripchart-Boxplot
```{r mf_da_stripchart_site_and_mouse_celltypefacet1_immgen-lung, fig.asp = 1.4}
max_prop_cell_type <- arranged_data %>% group_by(cell_type) %>% summarise(max_proportion = max(proportion))
stat_test <- propeller_pairwise %>% 
  filter(group1 %in% arranged_data$site_caps & group2 %in%  arranged_data$site_caps) %>%
  left_join(max_prop_cell_type) %>% 
  arrange(cell_type, group1, group2) %>% 
  na.omit() %>% 
  filter(FDR < 0.05)
  
add_count_y <- stat_test %>% group_by(cell_type) %>% reframe(group1 = group1, group2 = group2, n = 1:n())
stat_test <- stat_test %>% left_join(add_count_y) %>% mutate(y.position = max_proportion + (n*max_proportion/10))


arranged_data %>% 
  ggplot(aes(x = site_caps, y = proportion, color = site_caps)) +
  geom_boxplot(alpha = 0.4, color = 'black', aes(fill = site_caps), 
               show.legend = FALSE, size = 0.25, fatten = 1) +
  # geom_point(alpha = 1, size = 2, shape = 21, color = 'black', aes(fill = site_caps)) +
  geom_quasirandom(method="smiley", width = 0.1, alpha = 1, size = 1, 
                   shape = 21, color = 'black', aes(fill = site_caps), stroke = 0.4) +
  # geom_quasirandom(method="smiley", width = 0.1, alpha = 1, size = 1, aes(color = site_caps)) +
  # scale_fill_OkabeIto() +
  # scale_color_OkabeIto() +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  # scale_y_continuous(expand = c(0, 0)) +
  scale_y_continuous(expand = c(0.02, 0, 0.08, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), scales = 'free_y', ncol = 2) +
  theme(
    # strip.background = element_blank(),
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    panel.border = element_rect(colour = "black", size=0.5),
    strip.text.x = element_text(size = 4),
    strip.background = element_blank()
  ) +
  # guides( color = 'none') +
  guides(fill = guide_legend(override.aes = list(size = 3))) +
  labs(
    fill = NULL,
    x = NULL,
    y = 'Proportion of cells'
  ) +
  stat_pvalue_manual(stat_test, label = 'FDR.sign', size = 2)
```








## Liver MF: Differential abundance of macrophages cell types 
Differential abundance of cell types between normal lung tissue and TME in lung metastasis

```{r mf_da_dist_site_conf-liver}
ct_label <- 'mf_immgen_pruned.labels.fine'
```

### Statistical analysis
Differential abundance using `speckle::propeller` package
```{r mf_da_propeller_immgen-liver}
use_data <- cell_annot %>% 
  filter(celltype_immgen_pruned.labels == 'Macrophages')

# Remove low abundant cell types 
keep_labels <- table(use_data[[ct_label]]) %>% data.frame %>% filter(Freq >= 100) %>% pull(Var1)
keep_rows <- use_data[[ct_label]] %in% keep_labels
use_data <- use_data[keep_rows,]

# Run propeller Global
propeller_global <- propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$site) %>% 
  rownames_to_column('cell_type')


# Run pairwise propeller to obtaib P-values
site_levels <- use_data$site_caps %>% unique
# site_combn <- combn(site_levels, 2, simplify = FALSE)
site_combn <- list(
  # c('Lung Normal', 'Lung TME')
  c('Liver Normal', 'Liver TME')
)
sc <-  site_combn[[1]]
propeller_pairwise <-foreach(sc = site_combn, .combine = rbind) %do% {
  use_data <- cell_annot %>%
    filter(site_caps %in% sc) %>%
    mutate(
      group = ifelse(site_caps == sc[1], 'group1', 'group2')
    )
  keep_rows <- use_data[[ct_label]] %in% keep_labels
  use_data <- use_data[keep_rows,]
  propeller(clusters = use_data[[ct_label]], sample = use_data$sample, group = use_data$group) %>%
    mutate(
      group1 = sc[1],
      group2 = sc[2],
      FDR.sign = symnum(FDR, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001,0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "")),
      p.sign = symnum(P.Value, corr = FALSE, na = FALSE, cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", ""))
    ) %>%
    dplyr::rename(cell_type = BaselineProp.clusters)
}

```

Table of results with pairwise comparison between sites for cluster abundance. 
```{r mf_da_propeller_immgen_tab-liver}
propeller_pairwise %>%
  arrange(FDR) %>% 
  datatable(., 
            rownames = FALSE, 
            filter = 'top', 
            caption = 'Pairwise comparison between cell types for cluster abundance using speckle::propeller',
            extensions = 'Buttons', 
            options = list(
              dom = 'Blfrtip',
              buttons = c('csv', 'excel')
            )) %>% 
  formatPercentage(names(propeller_pairwise)[2:4], digits = 3) %>% 
  formatRound(names(propeller_pairwise)[5:6], digits = 3, interval = -1000) %>% 
  formatSignif(c('P.Value', 'FDR'), digits = 3)
```


### Distribution by site
```{r mf_da_dist_site_data_immgen-liver}
# Create data
use_data <- cell_annot %>% 
  filter(site_caps %in% unlist(site_combn)) %>% 
  dplyr::rename('celltype_label' = ct_label) %>% 
  group_by(site_caps, celltype_label) %>% 
  summarise(freq = n()) %>% 
  dplyr::rename(cell_type = celltype_label) %>% 
  na.omit()

# Remove low abundant cell types
cell_type_freq <- use_data %>% group_by(cell_type) %>% summarise(totalfreq = sum(freq))
# site_filtered <- use_data %>% group_by(site_caps) %>% summarise(freq = sum(freq)) %>% filter(freq >= 500)
use_data <- use_data %>% 
  # filter(site_caps %in% site_filtered$site_caps) %>% 
  left_join(cell_type_freq) %>% 
  filter(totalfreq >= 50) %>% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data <- use_data %>% 
  arrange(site_caps, cell_type)%>% 
  group_by(site_caps) %>%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion < 0.03, '', cell_type_label)
    
  )

```

#### Barplot stacked
```{r mf_da_barplot_site_stacked_immgen-liver}
arranged_data %>% 
  ggplot(aes(x = site_caps, y = proportion, fill = cell_type)) +
  geom_col(alpha = 0.85, color = 'black', linewidth = 0.2) +
  # scale_fill_manual(values = celltype_palette) +
  scale_y_continuous(expand = c(0, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  # geom_text(aes(y = label_y, label = cell_type_label), colour = "white", size = geom_text_size) +
  theme(axis.ticks.x = element_blank()) +
  labs(
    fill = NULL, 
    x = NULL,
    y = 'Proportion of cells'
  )
```

#### Barplot facet
```{r mf_da_barplot_site_celltypefacet_immgen-liver, fig.width=5.2}
arranged_data %>% 
  ggplot(aes(x = site_caps, y = proportion, fill = site_caps, group = cell_type)) +
  geom_col(alpha = 0.85, color = 'black', size = 0.2) +
  scale_fill_manual(values = site_type_palette) +
  scale_y_continuous(expand = c(0.02, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), ncol = 3, scales = 'free_y') +
  theme(
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x = element_blank(),
    axis.line.x = element_blank(),
    axis.ticks.x = element_blank(),
    panel.border = element_rect(colour = "black", size=0.5),
    strip.background = element_blank()
  ) +
  labs(
    fill = NULL, 
    x = NULL,
    y = 'Proportion of cells'
  ) #+
  # guides(fill = 'none')
```

#### Piechart
```{r mf_da_piechart_site_celltypefacet_immgen-liver, fig.width = 4, fig.asp = 0.3}
arranged_data %>% 
  ggplot(aes(x = "", y = proportion, fill = cell_type)) +
  # geom_bar(stat="identity", width=1, color="white") +
  geom_bar(stat="identity", width=1) +
  coord_polar("y", start=0) +
  # scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = '', 
    x = NULL,
    y = NULL
  )

```

#### Doughnut chart
```{r mf_da_doughnut_site_celltypefacet_immgen-liver, fig.width = 4, fig.asp = 0.3}
i <- unique(arranged_data$site_caps)[1]
use_data <- foreach(i = unique(arranged_data$site_caps), .combine = rbind) %do% {
  x <- arranged_data %>% 
    filter(site_caps == i) %>% 
    arrange(cell_type)
  x$ymax <- cumsum(x$proportion)
  x$ymin <- c(0, head(x$ymax, n = -1))
  x
}

use_data %>% 
  ggplot(aes(ymax=ymax, ymin=ymin, xmax=4, xmin=3, fill=cell_type)) +
  geom_rect() +
  coord_polar(theta="y") + # Try to remove that to understand how the chart is built initially
  xlim(c(2, 4)) + # Try to remove that to see how to make a pie chart
  # scale_fill_manual(values = celltype_palette) +
  facet_wrap(vars(site_caps), ncol = 3) +
  theme(
    strip.background = element_blank(),
    axis.text=element_blank(),
    axis.ticks=element_blank(),
    axis.line =element_blank()
  ) +
  labs(
    fill = '', 
    x = NULL,
    y = NULL
  )
```

### Distribution by site and mouse
```{r mf_da_dist_site_and_mouse_data_immgen-liver}
# Create data
use_data <- cell_annot %>% 
  filter(site_caps %in% unlist(site_combn)) %>% 
  dplyr::rename('celltype_label' = ct_label) %>% 
  group_by(site_caps, mouse_id, celltype_label) %>% 
  summarise(freq = n()) %>% 
  dplyr::rename(cell_type = celltype_label) %>% 
  na.omit()

# Remove low abundant cell types and sites
cell_type_freq <- use_data %>% group_by(cell_type) %>% summarise(totalfreq = sum(freq))
# site_filtered <- use_data %>% group_by(site_caps) %>% summarise(freq = sum(freq)) %>% filter(freq >= 500)
use_data <- use_data %>% 
  # filter(site_caps %in% site_filtered$site_caps) %>% 
  left_join(cell_type_freq) %>% 
  filter(totalfreq >= 50) %>% 
  mutate(cell_type = fct_reorder(cell_type, totalfreq))

# Arrange data and add labels
arranged_data <- use_data %>% 
  arrange(site_caps, cell_type)%>% 
  group_by(site_caps, mouse_id) %>%
  mutate(
    totalfreq_by_site = sum(freq),
    proportion = freq/ totalfreq_by_site,
    label_y = cumsum(freq)/ totalfreq_by_site,
    label_y = label_y - 0.5 * (freq / totalfreq_by_site),
    cell_type = fct_reorder(cell_type, totalfreq, .desc = TRUE),
    cell_type_label = as.character(cell_type),
    cell_type_label = ifelse(proportion < 0.03, '', cell_type_label)
  )
```



#### Stripchart-Boxplot
```{r mf_da_stripchart_site_and_mouse_celltypefacet1_immgen-liver, fig.asp = 0.8}
max_prop_cell_type <- arranged_data %>% group_by(cell_type) %>% summarise(max_proportion = max(proportion))
stat_test <- propeller_pairwise %>% 
  filter(group1 %in% arranged_data$site_caps & group2 %in%  arranged_data$site_caps) %>%
  left_join(max_prop_cell_type) %>% 
  arrange(cell_type, group1, group2) %>% 
  na.omit() %>% 
  filter(FDR < 0.05)
  
add_count_y <- stat_test %>% group_by(cell_type) %>% reframe(group1 = group1, group2 = group2, n = 1:n())
stat_test <- stat_test %>% left_join(add_count_y) %>% mutate(y.position = max_proportion + (n*max_proportion/10))


arranged_data %>% 
  ggplot(aes(x = site_caps, y = proportion, color = site_caps)) +
  geom_boxplot(alpha = 0.4, color = 'black', aes(fill = site_caps), 
               show.legend = FALSE, size = 0.25, fatten = 1) +
  # geom_point(alpha = 1, size = 2, shape = 21, color = 'black', aes(fill = site_caps)) +
  geom_quasirandom(method="smiley", width = 0.1, alpha = 1, size = 1, 
                   shape = 21, color = 'black', aes(fill = site_caps), stroke = 0.4) +
  # geom_quasirandom(method="smiley", width = 0.1, alpha = 1, size = 1, aes(color = site_caps)) +
  # scale_fill_OkabeIto() +
  # scale_color_OkabeIto() +
  scale_fill_manual(values = site_type_palette) +
  scale_color_manual(values = site_type_palette) +
  # scale_y_continuous(expand = c(0, 0)) +
  scale_y_continuous(expand = c(0.02, 0, 0.08, 0)) +
  scale_x_discrete(expand = c(0, 0.5)) +
  facet_wrap(vars(cell_type), scales = 'free_y', ncol = 2) +
  theme(
    # strip.background = element_blank(),
    # axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), 
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    panel.border = element_rect(colour = "black", size=0.5),
    strip.text.x = element_text(size = 4),
    strip.background = element_blank()
  ) +
  # guides( color = 'none') +
  guides(fill = guide_legend(override.aes = list(size = 3))) +
  labs(
    fill = NULL,
    x = NULL,
    y = 'Proportion of cells'
  ) +
  stat_pvalue_manual(stat_test, label = 'FDR.sign', size = 2)
```






