---
title: "Comapre 10x RNA-seq analysis of microenvironment mets and normal tissue"
subtitle: "Differential gene expression analysis of TME and normal cells from lung and liver normal tissue and metastases, using pseudobulk"
author: "Francesc Castro-Giner"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
params:
  date: '`r format(Sys.Date(), "%B %d, %Y")`'
  output_dir: ./output/p26532_o28268_o34980/p26532_o28268_o34980-10x_rnaseq-ll-tme-normal
  min_counts: 5
  min_present_prop: 0.40
  msigdb_dir: ./data/resources/MSigDB
  msigdb_vs: v2022.1.Mm
---

## Load libraries, additional functions and data

Setup environment
```{r setup, include=TRUE, warning=FALSE, message=FALSE}
knitr::opts_chunk$set(results='asis', echo=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.align = 'center', fig.width = 3.5, fig.asp = 0.618, dpi = 600, dev = c("png", "pdf"))

options(stringsAsFactors = FALSE)

use_seed <- 1100101
set.seed(use_seed)

if(!dir.exists(params$output_dir))
  dir.create(params$output_dir, recursive = TRUE, showWarnings = FALSE)
```

Load packages
```{r load-libraries}
library(tidyverse)
library(showtext)
library(foreach)
library(DT)
library(knitr)
library(kableExtra)
library(cowplot)
library(colorblindr)
library(ggbeeswarm)
library(arsenal)
library(RColorBrewer)
library(ggpubr)
library(pheatmap)
library(patchwork)
library(openxlsx)
library(magrittr)

library(scater)
library(DropletUtils)
library(Seurat)
library(scran)
library(BiocSingular)
library(batchelor)
library(bluster)
library(celldex)
library(SingleR)
library(scDblFinder)
library(speckle)
library(miloR)
library(ComplexHeatmap)
library(clusterProfiler)
library(GSVA)
library(circlize)
library(org.Mm.eg.db)

library(factoextra)
```

Load ggplot theme
```{r ggplot-theme}
source("./configuration/rmarkdown/ggplot_theme.R")
source("./configuration/rmarkdown/color_palettes.R")
```

Load utils functions
```{r load-util-functions}
source('./code/R-functions/heatmap_scale.R')
source('./code/R-functions/dge_wrappers.r')
source('./code/R-functions/dge_report.r')
source('./code/R-functions/gse_omnibus.r')
source('./code/R-functions/gse_report.r')
source('./code/R-functions/subchunkify.R')
source('./code/R-functions/pairwise_terms_jc.r')
```

Load MSigDB gene sets
```{r load-msigdb}
gmt_files_symbols <- list(
  msigdb.c2.cp = paste0(params$msigdb_dir, '/', params$msigdb_vs, '/m2.cp.', params$msigdb_vs, '.symbols.gmt'),
  msigdb.m2.cp.reactome= paste0(params$msigdb_dir, '/', params$msigdb_vs, '/m2.cp.reactome.', params$msigdb_vs, '.symbols.gmt')
)
```

Clean files generated in previous runs
```{r clean-files}
rmd_file <- current_input()
if(!is.null(rmd_file)) {
  figures_dir <- file.path('./docs/figure',rmd_file)
  if(dir.exists(figures_dir)) {
    unlink(file.path(figures_dir, "*"))
  }
}

```

## Configure analyses

### Data wrangling
Load original data
```{r load-original-data, eval = FALSE}
sce_comb <- readRDS(file.path(params$output_dir, 'sce_integrated.rds'))
```

Aggregate data
```{r aggregate-across-cell, eval = FALSE}
celltype_label <- 'celltype_immgen_pruned.labels'
sce_comb$anatomical_site <- gsub(" .*", "", sce_comb$site_caps)
sce_comb$condition <- gsub(".* ", "", sce_comb$site_caps)
se <- aggregateAcrossCells(sce_comb, ids = colData(sce_comb)[, c("anatomical_site", "condition", "site", "celltype_immgen_pruned.labels", "mouse_id")])
se <- as(se, "SummarizedExperiment")
rowData(se) <- rowData(sce_comb)
rownames(se) <- rowData(se)$ensembl_gene
se$sample_alias <- paste(se$anatomical_site, se$condition, se$celltype_immgen_pruned.labels, se$mouse_id, sep = '_')
colnames(se) <- se$sample_alias
```

TMM normalization
```{r dw-tmm, eval = FALSE}
norm_method <- "TMM"
exprs_mat_for_norm <- assay(se, 'counts')
norm_factors <- calcNormFactors(exprs_mat_for_norm, method = norm_method)
lib_size <- colSums(exprs_mat_for_norm)/10^6
se$size_factor <- (norm_factors * lib_size)
se <- logNormCounts(se, size.factors =se$size_factor)
se <- logNormCounts(se, size.factors =se$size_factor, log = FALSE)
```

CPM calculation
```{r dw-cpm, eval = FALSE}
assay(se, 'cpm') <- edgeR::cpm(assay(se, 'normcounts'))
assay(se, 'logcpm') <- edgeR::cpm(assay(se, 'normcounts'), log = TRUE)
```

Save aggregated data
```{r save-aggregated-data, eval = FALSE}
saveRDS(se, file = file.path(params$output_dir, 'sce_aggregated.rds'))
```

Load data
```{r load-data}
se <- readRDS(file.path(params$output_dir, 'sce_aggregated.rds'))
```


### Define of comparisons

#### Process data
```{r conf-comparisons}
x <- colData(se) %>% data.frame %>% 
  mutate(
    cell_type = celltype_immgen_pruned.labels,
    cell_type_site = paste(anatomical_site, cell_type)
  )


comp_list <- list(
  `Lung-DC-T-over-N` = list(
    TME = x %>% filter(anatomical_site == 'Lung' & cell_type == 'DC' & condition == 'TME') %>% 
      pull(sample_alias),
    Normal = x %>% filter(anatomical_site == 'Lung' & cell_type == 'DC' & condition == 'Normal') %>% 
      pull(sample_alias)
  ),
  `Lung-Endothelial-T-over-N` = list(
    TME = x %>% filter(anatomical_site == 'Lung' & cell_type == 'Endothelial cells' & condition == 'TME') %>% 
      pull(sample_alias),
    Normal = x %>% filter(anatomical_site == 'Lung' & cell_type == 'Endothelial cells' & condition == 'Normal') %>% 
      pull(sample_alias)
  ),
  `Lung-Fibroblasts-T-over-N` = list(
    TME = x %>% filter(anatomical_site == 'Lung' & cell_type == 'Fibroblasts' & condition == 'TME') %>% 
      pull(sample_alias),
    Normal = x %>% filter(anatomical_site == 'Lung' & cell_type == 'Fibroblasts' & condition == 'Normal') %>% 
      pull(sample_alias)
  ),
  `Lung-Macrophages-T-over-N` = list(
    TME = x %>% filter(anatomical_site == 'Lung' & cell_type == 'Macrophages' & condition == 'TME') %>% 
      pull(sample_alias),
    Normal = x %>% filter(anatomical_site == 'Lung' & cell_type == 'Macrophages' & condition == 'Normal') %>% 
      pull(sample_alias)
  ),
  `Lung-Monocytes-T-over-N` = list(
    TME = x %>% filter(anatomical_site == 'Lung' & cell_type == 'Monocytes' & condition == 'TME') %>% 
      pull(sample_alias),
    Normal = x %>% filter(anatomical_site == 'Lung' & cell_type == 'Monocytes' & condition == 'Normal') %>% 
      pull(sample_alias)
  ),
  `Lung-Neutrophils-T-over-N` = list(
    TME = x %>% filter(anatomical_site == 'Lung' & cell_type == 'Neutrophils' & condition == 'TME') %>% 
      pull(sample_alias),
    Normal = x %>% filter(anatomical_site == 'Lung' & cell_type == 'Neutrophils' & condition == 'Normal') %>% 
      pull(sample_alias)
  ),
  `Liver-DC-T-over-N` = list(
    TME = x %>% filter(anatomical_site == 'Liver' & cell_type == 'DC' & condition == 'TME') %>% 
      pull(sample_alias),
    Normal = x %>% filter(anatomical_site == 'Liver' & cell_type == 'DC' & condition == 'Normal') %>% 
      pull(sample_alias)
  ),
  `Liver-Endothelial-T-over-N` = list(
    TME = x %>% filter(anatomical_site == 'Liver' & cell_type == 'Endothelial cells' & condition == 'TME') %>% 
      pull(sample_alias),
    Normal = x %>% filter(anatomical_site == 'Liver' & cell_type == 'Endothelial cells' & condition == 'Normal') %>% 
      pull(sample_alias)
  ),
  `Liver-Macrophages-T-over-N` = list(
    TME = x %>% filter(anatomical_site == 'Liver' & cell_type == 'Macrophages' & condition == 'TME') %>% 
      pull(sample_alias),
    Normal = x %>% filter(anatomical_site == 'Liver' & cell_type == 'Macrophages' & condition == 'Normal') %>% 
      pull(sample_alias)
  ),
  `Liver-Monocytes-T-over-N` = list(
    TME = x %>% filter(anatomical_site == 'Liver' & cell_type == 'Monocytes' & condition == 'TME') %>% 
      pull(sample_alias),
    Normal = x %>% filter(anatomical_site == 'Liver' & cell_type == 'Monocytes' & condition == 'Normal') %>% 
      pull(sample_alias)
  ),
  `Liver-Neutrophils-T-over-N` = list(
    TME = x %>% filter(anatomical_site == 'Liver' & cell_type == 'Neutrophils' & condition == 'TME') %>% 
      pull(sample_alias),
    Normal = x %>% filter(anatomical_site == 'Liver' & cell_type == 'Neutrophils' & condition == 'Normal') %>% 
      pull(sample_alias)
  )
)

comp_list_description <- data.frame(
  comparison = names(comp_list),
  comparison_long = gsub("-T", "-TME", names(comp_list)) %>% 
  gsub("-N$", "-Normal", .) %>% 
  gsub("Lung-", "Lung: ", .) %>% 
  gsub("Liver-", "Liver: ", .) %>% 
  gsub("-", " ", .)
)

```


#### Table of comparisons

We filter comparisons with n < 3 samples in each group
```{r table-comparisons}
i = names(comp_list)[1]
comp_df <- foreach(i = names(comp_list), .combine = rbind) %do% {
  icomp <- comp_list[[i]]
  data.frame(
    comparison = i,
    `Sample Group (n)` =  length(icomp[[1]]),
    `Reference Group (n)` = length(icomp[[2]]),
    check.names = FALSE
  )
}

# filter comparisons were we have at least 3 samples in each group
comp_df %<>% 
  mutate(Evaluated = `Sample Group (n)` >= 3 & `Reference Group (n)` >= 3)
kepp_comp <- comp_df %>% filter(Evaluated) %>% pull(comparison)
comp_list <- comp_list[kepp_comp]

# Add long names
comp_df <- comp_df %>% left_join(comp_list_description) %>% 
  mutate(comparison = comparison_long) %>%
  dplyr::select(-comparison_long)

comp_df %>%
  datatable(., 
            rownames = FALSE, 
            filter = 'top', 
            caption = 'List of comparisons for differential expression. Comparisons with n < 3 samples in each group won\'t be evaluated',
            extensions = 'Buttons', 
            options = list(
              dom = 'Blfrtip',
              buttons = c('csv', 'excel')
            ))

```

## Data generation

### Run differential expression analysis
Get Ensembl gene info
```{r de-ensembl, eval = FALSE}
ensembl <-  biomaRt::useEnsembl(biomart="genes", dataset="mmusculus_gene_ensembl",
                                mirror = 'useast')
gene_desc <- biomaRt::getBM(attributes=c('external_gene_name','description'),
                              filters = 'external_gene_name',
                              values = rowData(se)$gene_name, mart =ensembl) %>%
    dplyr::rename('gene_name' = 'external_gene_name') %>%
    unique

gene_desc <- gene_desc[!duplicated(gene_desc$gene_name),]
saveRDS(gene_desc, file = file.path(dirname(params$output_dir), 'ensembl_gene_desc.rds'))
```

```{r de-ensembl-load, eval = FALSE}
gene_desc <- readRDS(file = file.path(dirname(params$output_dir), 'ensembl_gene_desc.rds'))
```

Run differential expression analysis using EdgeR QLF
```{r edgeR-QLF-run, eval = FALSE}
use_comp <-  comp_list[[1]]
dge_list <- foreach(use_comp = comp_list) %do%{
  se_cols <- se$sample_alias %in% unlist(use_comp)
  use_se <- se[,se_cols]
  use_se$group <- ifelse(
    use_se$sample_alias %in% use_comp[[1]],
    names(use_comp)[1],
    names(use_comp)[2]
  )
  
  if(is.null(use_comp$batch)) {
    dge <- edgeR_dge(
      use_se,
      # Design configuration for differential expression
      group_var =  'group',
      group_sample = names(use_comp)[1],
      group_ref = names(use_comp)[2],
      design_formula = "~ group",
      coef = 'last',
      # Conversion from SingleCellExperiment/Summarized Experiment to DGEList
      assay_to_DGEList = 'counts',
      assay_to_row_filter = "counts",
      # Feature filtering parameters
      use_filterByExpr = TRUE,
      min_counts = params$min_counts,
      min_present_prop = params$min_present_prop,
      # EdgeR workflow configuration
      run_calcNormFactors = 'TMM',
      estimateDisp_robust = TRUE,
      estimateDisp_trend.method = "locfit",
      glm_approach = "QLF",
      # Output configuration
      adjust_method = 'BH'
      )
    } else {
      design_formula = paste0("~ ", use_comp$batch," + group")
      dge <- edgeR_dge(
        use_se,
        # Design configuration for differential expression
        group_var =  'group',
        group_sample = names(use_comp)[1],
        group_ref = names(use_comp)[2],
        batch_vars = use_comp$batch,
        design_formula = design_formula,
        coef = 'last',
        # Conversion from SingleCellExperiment/Summarized Experiment to DGEList
        assay_to_DGEList = 'counts',
        assay_to_row_filter = "counts",
        # Feature filtering parameters
        use_filterByExpr = TRUE,
        min_counts = params$min_counts,
        min_present_prop = params$min_present_prop,
        # EdgeR workflow configuration
        run_calcNormFactors = 'TMM',
        estimateDisp_robust = TRUE,
        estimateDisp_trend.method = "locfit",
        glm_approach = "QLF",
        # Output configuration
        adjust_method = 'BH'
        )
    }
  
  # Add gene description
  use_res <- dge$results %>%  left_join(., gene_desc, relationship = "many-to-many")
  dge$results <- use_res %>% 
    filter(!duplicated(feature)) %>% 
    mutate(rownames = feature) %>% 
    column_to_rownames('rownames')
  
   return(dge)
}

names(dge_list) <- names(comp_list)

# Save objects
saveRDS(dge_list, file = file.path(params$output_dir, 'dge-edgeR_QLF.rds'))

# ###########
# object = use_se
# # Design configuration for differential expression
# group_var =  'group'
# group_sample = names(use_comp)[1]
# group_ref = names(use_comp)[2]
# design_formula = "~ group"
# coef = 'last',
# # Conversion from SingleCellExperiment/Summarized Experiment to DGEList
# assay_to_DGEList = 'counts'
# assay_to_row_filter = "counts"
# # Feature filtering parameters
# use_filterByExpr = TRUE
# min_counts = params$min_count
# min_present_prop = params$min_present_prop,
# # EdgeR workflow configuration
# run_calcNormFactors = 'TMM'
# estimateDisp_robust = TRUE
# estimateDisp_trend.method = "locfit"
# glm_approach = "QLF"
# # Output configuration
# adjust_method = 'BH'
# 
# use_samples = NULL,
# 
# numeric_covar = NULL,
# batch_vars = NULL,
# contrast = NULL,
# # Conversion from SingleCellExperiment to DGEList
# spike_normalization = FALSE,
# use_colData = NULL,
# use_rowData = NULL,
# # Feature filtering parameters
# use_filterByExpr = TRUE,
# min_counts = 1,
# min_present_prop = 0.40,
# # EdgeR workflow configuration
# run_calcNormFactors = 'TMM',
# estimateDisp_robust = FALSE,
# estimateDisp_trend.method = "locfit",
# glmQLFit_robust = FALSE,
# glm_approach = "QLF",
# # Output configuration
# adjust_method = 'BH',
# assays_from_SingleCellExperiment = NULL

# 
# for(use_comp in comp_list) {
#   
#   se_cols <- se$sample_alias %in% unlist(use_comp)
#   use_se <- se[,se_cols]
#   use_se$group <- ifelse(
#     use_se$sample_alias %in% use_comp[[1]],
#     names(use_comp)[1],
#     names(use_comp)[2]
#   )
#   
#   if(is.null(use_comp$batch)) {
#     dge <- edgeR_dge(
#       use_se,
#       # Design configuration for differential expression
#       group_var =  'group',
#       group_sample = names(use_comp)[1],
#       group_ref = names(use_comp)[2],
#       design_formula = "~ group",
#       coef = 'last',
#       # Conversion from SingleCellExperiment/Summarized Experiment to DGEList
#       assay_to_DGEList = 'counts',
#       assay_to_row_filter = "counts",
#       # Feature filtering parameters
#       use_filterByExpr = TRUE,
#       min_counts = params$min_counts,
#       min_present_prop = params$min_present_prop,
#       # EdgeR workflow configuration
#       run_calcNormFactors = 'TMM',
#       estimateDisp_robust = TRUE,
#       estimateDisp_trend.method = "locfit",
#       glm_approach = "QLF",
#       # Output configuration
#       adjust_method = 'BH'
#       )
#     } else {
#       design_formula = paste0("~ ", use_comp$batch," + group")
#       dge <- edgeR_dge(
#         use_se,
#         # Design configuration for differential expression
#         group_var =  'group',
#         group_sample = names(use_comp)[1],
#         group_ref = names(use_comp)[2],
#         batch_vars = use_comp$batch,
#         design_formula = design_formula,
#         coef = 'last',
#         # Conversion from SingleCellExperiment/Summarized Experiment to DGEList
#         assay_to_DGEList = 'counts',
#         assay_to_row_filter = "counts",
#         # Feature filtering parameters
#         use_filterByExpr = TRUE,
#         min_counts = params$min_counts,
#         min_present_prop = params$min_present_prop,
#         # EdgeR workflow configuration
#         run_calcNormFactors = 'TMM',
#         estimateDisp_robust = TRUE,
#         estimateDisp_trend.method = "locfit",
#         glm_approach = "QLF",
#         # Output configuration
#         adjust_method = 'BH'
#         )
#     }
#   
#   # Add gene description
#   use_res <- dge$results %>%  left_join(., gene_desc, relationship = "many-to-many")
#   dge$results <- use_res %>% 
#     filter(!duplicated(feature)) %>% 
#     mutate(rownames = feature) %>% 
#     column_to_rownames('rownames')
# }

```

### Run functional enrichment
```{r gse-run, eval = FALSE}
dge_list <- readRDS(file.path(params$output_dir, 'dge-edgeR_QLF.rds'))

gse_list <- foreach(dge = dge_list) %do%{
  gse_res <- gse_omnibus(
    feature_names = dge$results$gene_name,
    p = dge$results$FDR,
    fc = dge$results$logFC,
    fc_thrs = 0.5,
    gmt_files = gmt_files_symbols,
    
    run_enricher = FALSE,
    run_enrichGO = FALSE,
    run_enrichKEGG = FALSE,
    run_enrichReactome = FALSE,
    
    run_GSEA = TRUE,
    run_gseGO = TRUE,
    run_gseKEGG = FALSE,
    run_gseReactome = FALSE,
    
    go_ontologies = c('BP'),
    go_simplify_p = 0.05,
    args_gse = list(minGSSize = 10, maxGSSize = 250, 
                    pvalueCutoff = 1, 
                    BPPARAM = BiocParallel::MulticoreParam(1)),
    annot_db = 'org.Mm.eg.db',
    organism = 'mmusculus'
    )
  
   return(gse_res)

}
names(gse_list) <- names(dge_list)

# Save objects
saveRDS(gse_list, file = file.path(params$output_dir,'gse_omnibus-edgeR_QLF.rds'))


# 
# gse_list <- list()
# for(dge_names in names(dge_list)) {
# dge <- dge_list[[dge_names]]
# cat('######', dge_names, "\n\n\n")
# cat('######', dge$comparison, "\n\n\n")
# gse_list[[dge_names]] <- gse_omnibus(
#     feature_names = dge$results$gene_name,
#     p = dge$results$FDR,
#     fc = dge$results$logFC,
#     fc_thrs = 0.5,
#     gmt_files = gmt_files_symbols,
#     
#     run_enricher = FALSE,
#     run_enrichGO = FALSE,
#     run_enrichKEGG = FALSE,
#     run_enrichReactome = FALSE,
#     
#     run_GSEA = TRUE,
#     run_gseGO = TRUE,
#     run_gseKEGG = FALSE,
#     run_gseReactome = FALSE,
#     
#     go_ontologies = c('BP', 'MF'),
#     go_simplify_p = 0.05,
#     args_gse = list(minGSSize = 10, maxGSSize = 250, 
#                     pvalueCutoff = 1, 
#                     BPPARAM = BiocParallel::MulticoreParam(1))
#     )
# 
# }
# # Save objects
# saveRDS(gse_list, file = file.path(params$output_dir,'gse_omnibus-edgeR_QLF.rds'))

```

### Load data
```{r dge-gse-load}
dge_list <- readRDS(file = file.path(params$output_dir, 'dge-edgeR_QLF.rds'))
gse_list <- readRDS(file = file.path(params$output_dir, 'gse_omnibus-edgeR_QLF.rds'))
```








## MF: Gene set enrichment analysis

Gene set enrichment analysis
```{r mf-marker-gsea-c2reactome_gobp-conf}
use_gse_list <- gse_list[c("Lung-Macrophages-T-over-N", "Liver-Macrophages-T-over-N")]
names(use_gse_list) <- c('Lung', 'Liver')
test_type <- 'GSEA'
gset_collection <- 'msigdb.m2.cp.reactome'
p_thrs <- 0.01
topn <- 20

# Load all the results
gse_df <- foreach(i = names(use_gse_list), .combine = rbind) %do% {
  i_1 <- use_gse_list[[i]][[test_type]][[gset_collection]]@result %>% 
    mutate(group = i)
  i_2 <- use_gse_list[[i]][['gseGO']][['BP']]@result %>% 
    filter(simplify) %>% 
    dplyr::select(-simplify) %>% 
    mutate(group = i)
  rbind(i_1, i_2)
} %>% 
  mutate(
    group = factor(group, c('Lung', 'Liver')),
    Description = clean_msigdb_names(Description) %>% tolower() %>% str_to_title,
    Description = ifelse(grepl("^GO:", ID), paste0("GO:", Description), Description)
  ) 

# Select topn by group in selected gset
top_by_group <- foreach(i = names(use_gse_list), .combine = c) %do% {
  gse_df %>% 
    filter(group == i) %>% 
    filter(p.adjust < p_thrs) %>% 
    arrange(p.adjust) %>% pull(Description) %>%
    head(topn)
}

# Data frame to plot
res_top <- gse_df %>% 
  filter(Description %in% top_by_group) %>% 
  mutate(log.p.adjust = -log10(p.adjust))

# Rows and column orders by hclust
use_mat <- res_top %>% 
  dplyr::select(Description, log.p.adjust, group) %>% 
  pivot_wider(names_from = group, values_from = log.p.adjust) %>% 
  column_to_rownames('Description')
use_mat[is.na(use_mat)] <- 0

row_hclust <- hclust(dist(use_mat))
row_hclust_labels <- row_hclust$labels[row_hclust$order]

col_hclust <- hclust(dist(t(use_mat)))
col_hclust_labels <- col_hclust$labels[col_hclust$order]

use_mat <- use_mat[row_hclust_labels,col_hclust_labels]

# GSET clusters by kmeans using silhouette
use_kmax <- ifelse(nrow(use_mat) <= 20, nrow(use_mat)-2, 20)
nb_clust <- fviz_nbclust(use_mat, kmeans, method = "silhouette", k.max = use_kmax)
silhouette_best_k <- nb_clust$data %>% slice_max(y) %>% pull(clusters) %>% as.character %>% as.numeric
# silhouette_best_k <- 5
kmeans_mat <- kmeans(use_mat, silhouette_best_k, nstart = 25)
kmeans_mat_df <- kmeans_mat$cluster %>%
  data.frame %>%
  set_names('kcluster') %>%
  rownames_to_column('Description') %>%
  mutate(kcluster = as.character(kcluster))


# Order factors by hclust
res_top %<>% 
  left_join(kmeans_mat_df) %>% 
  mutate(
    Description = factor(Description, levels = row_hclust_labels),
    group = fct_relevel(group, col_hclust_labels)
  )

# Calculate jaccard similarity matrix
gset_collection <- foreach(i = levels(res_top$Description)) %do% {
  res_top %>% filter(Description == i) %>% pull(core_enrichment) %>% 
    paste(collapse = '/') %>% 
    strsplit('/') %>% unlist %>% unique
}
names(gset_collection) <- levels(res_top$Description)

use_terms <- res_top$Description %>% unique
jc_matrix <- pairwise_terms_jc(gset_collection[use_terms])
```

#### Results table
```{r mf-marker-gsea-c2reactome_gobp-table}
gse_df %>% 
  filter(p.adjust < p_thrs) %>% 
  arrange(desc(p.adjust)) %>% 
  dplyr::select(group, Description:p.adjust, Count:GeneRatio, core_enrichment)  %>%
  dplyr::rename(site = group) %>% 
  datatable(., 
            rownames = FALSE, 
            filter = 'top', 
            caption = 'Results from gene-set enrichment analysis',
            extensions = 'Buttons', 
            options = list(
              dom = 'Blfrtip',
              buttons = c('csv', 'excel')
            ))  %>% 
  formatRound(c('enrichmentScore', 'NES', 'GeneRatio'), digits = 2) %>%
  formatSignif(c('pvalue', 'p.adjust'), digits = 3)
```

#### Number of gene sets
```{r mf-marker-gsea-c2reactome_gobp-ngset}
gse_df %>% 
  filter(p.adjust < p_thrs) %>% 
  group_by(group) %>% 
  summarise(n = n()) %>% 
  ggplot(aes(x=group, y=n)) +
  geom_bar(stat="identity", color = 'black')+
  geom_text(aes(label=n), vjust=1.4, color="white", size=geom_text_size)+
  labs(
    x = 'Site',
    y = 'Number of gene sets',
    title = 'Number of significant gene sets by site',
    caption = paste("Adjusted P value <", p_thrs)
    
  )
```

#### Gene-set similarity matrix
Heatmap showing the pair-wise similarity matrix of top `r topn` gene sets with an adjusted P value < `r p_thrs` in each gene group. Heatmap colors represent the Jaccard similarity coefficient.

```{r mf-marker-gsea-c2reactome_gobp-jcmatrix, fig.width=7.2, fig.asp=1.9}
col_fun <- colorRamp2(seq(0, 1, length.out = 4), brewer.pal(4, "GnBu"))
n_split <- 9

use_mat <- jc_matrix
colnames(use_mat) <- colnames(use_mat) %>% clean_msigdb_names %>% tolower() %>% str_to_title %>% ep_str_wrap(70)
rownames(use_mat) <- rownames(use_mat) %>% clean_msigdb_names %>% tolower() %>% str_to_title %>% ep_str_wrap(70)

ha_top <- HeatmapAnnotation(
  foo = anno_block(
    # labels = c("Translation", "Cell division"),
    labels = 1:n_split,
    labels_gp = gpar(col = "black", fontsize = 8),
    gp = gpar(lwd = 0, lty = 0))
  )

ht <- Heatmap(
  use_mat,
  name = 'Jaccard index',
  column_split = n_split,
  row_split = n_split,
  column_title = NULL,
  row_title = NULL,
  col = col_fun,
  show_column_dend = FALSE,
  show_column_names = FALSE,
  border = TRUE,
  top_annotation = ha_top,
  heatmap_legend_param = list(title_gp = gpar(fontsize = 8),labels_gp = gpar(fontsize = 8)),
  width = unit(7, "cm"))

ht = draw(ht)

cat("\n\n")

# Extract term order according to hclust from similarity heatmap
row_order <- row_order(ht) %>% set_names(1:n_split)
term_similarity_clusters <- foreach(i=names(row_order), .combine = rbind) %do% {
  data.frame(
    jc_sim_cl = i, 
    row_order = row_order[[i]], 
    Description =  rownames(jc_matrix)[row_order[[i]]]
  )
}

# Add similarity order to data for dot plot
res_top %<>% left_join(term_similarity_clusters)
```



#### Dotplots
The plots below show the top `r topn` gene sets with an adjusted P value < `r p_thrs` in each gene group
```{r mf-marker-gsea-c2reactome_gobp-dotplot, fig.width = 4.2, fig.asp = 1.29}
use_linewidth <- 1/2.141959
x_max <- res_top$NES %>% abs %>% max()

res_top %>%
  # Combine similarity clusters
  mutate(
    # jc_sim_cl = ifelse(jc_sim_cl %in% c('1'), 'Translation', jc_sim_cl),
    # jc_sim_cl = ifelse(jc_sim_cl %in% c('3','5','6'), 'Energy\ngeneration', jc_sim_cl),
    # jc_sim_cl = ifelse(jc_sim_cl %in% c('4', '8'), 'Cell stress', jc_sim_cl),
    # jc_sim_cl = factor(
    #   jc_sim_cl,
    #   levels = c('Translation', 'Energy\ngeneration', 'Cell stress', '2', '7', '9')
    #   ),

    p.adjust = ifelse(p.adjust > 0.05, 1, p.adjust),
    p.adjust = -log10( p.adjust),
    p.adjust = ifelse(p.adjust >= 10, 10, p.adjust),
    group = recode(group, primary_tumor = "PT", lung = "Lung", liver = "Liver"),
    group = factor(group, levels = c('PT', 'Lung', 'Liver'))
  ) %>% 
  ggplot(aes(NES, Description, color = NES, size = p.adjust)) +
  geom_point() +
  facet_grid(rows = vars(jc_sim_cl), cols = vars(group), scale = 'free_y', space = 'free_y') +
  scale_x_continuous(limits = c(-x_max, x_max),
                        expand = expansion(mult = c(0.08, 0.08))) +
  scale_y_discrete(labels = default_labeller(50)) +
  geom_vline(xintercept = 0, lty = 3) +
  scale_size_continuous(range=c(0.3, 2.5)) +
  scale_color_gradient2(trans = "reverse") +
  labs(
    size = expression(paste("lo", g[10],"(Adjusted P value)")),
    color = "NES"
  ) +
  theme(
    panel.border = element_rect(color = "black", fill = NA, size = use_linewidth/2),
    
    axis.line = element_blank(),
    axis.ticks = element_line(color = "black", size = use_linewidth/2),
    axis.text.y = element_text(size = 4),
    axis.text.x = element_text(size = 4),
    
    legend.text = element_text(size = 4), 
    legend.title = element_text(size = 4),
    legend.key.width = unit(0.2, "cm"),
    
    strip.background = element_blank()
  )
  
```





## Endothelial: Gene set enrichment analysis

Gene set enrichment analysis
```{r endothelial-marker-gsea-c2reactome_gobp-conf}
use_gse_list <- gse_list[c("Lung-Endothelial-T-over-N", "Liver-Endothelial-T-over-N")]
names(use_gse_list) <- c('Lung', 'Liver')
test_type <- 'GSEA'
gset_collection <- 'msigdb.m2.cp.reactome'
p_thrs <- 0.01
topn <- 40

# Load all the results
gse_df <- foreach(i = names(use_gse_list), .combine = rbind) %do% {
  i_1 <- use_gse_list[[i]][[test_type]][[gset_collection]]@result %>% 
    mutate(group = i)
  i_2 <- use_gse_list[[i]][['gseGO']][['BP']]@result %>% 
    filter(simplify) %>% 
    dplyr::select(-simplify) %>% 
    mutate(group = i)
  rbind(i_1, i_2)
} %>% 
  mutate(
    group = factor(group, c('Lung', 'Liver')),
    Description = clean_msigdb_names(Description) %>% tolower() %>% str_to_title,
    Description = ifelse(grepl("^GO:", ID), paste0("GO:", Description), Description)
  ) 

# Select topn by group in selected gset
top_by_group <- foreach(i = names(use_gse_list), .combine = c) %do% {
  gse_df %>% 
    filter(group == i) %>% 
    filter(p.adjust < p_thrs) %>% 
    arrange(p.adjust) %>% pull(Description) %>%
    head(topn)
}

# Data frame to plot
res_top <- gse_df %>% 
  filter(Description %in% top_by_group) %>% 
  mutate(log.p.adjust = -log10(p.adjust))

# Rows and column orders by hclust
use_mat <- res_top %>% 
  dplyr::select(Description, log.p.adjust, group) %>% 
  pivot_wider(names_from = group, values_from = log.p.adjust) %>% 
  column_to_rownames('Description')
use_mat[is.na(use_mat)] <- 0

row_hclust <- hclust(dist(use_mat))
row_hclust_labels <- row_hclust$labels[row_hclust$order]

col_hclust <- hclust(dist(t(use_mat)))
col_hclust_labels <- col_hclust$labels[col_hclust$order]

use_mat <- use_mat[row_hclust_labels,col_hclust_labels]

# GSET clusters by kmeans using silhouette
use_kmax <- ifelse(nrow(use_mat) <= 20, nrow(use_mat)-5, 20)
nb_clust <- fviz_nbclust(use_mat, kmeans, method = "silhouette", k.max = 7)
silhouette_best_k <- nb_clust$data %>% slice_max(y) %>% pull(clusters) %>% as.character %>% as.numeric
# silhouette_best_k <- 5
kmeans_mat <- kmeans(use_mat, silhouette_best_k, nstart = 25)
kmeans_mat_df <- kmeans_mat$cluster %>%
  data.frame %>%
  set_names('kcluster') %>%
  rownames_to_column('Description') %>%
  mutate(kcluster = as.character(kcluster))


# Order factors by hclust
res_top %<>% 
  left_join(kmeans_mat_df) %>% 
  mutate(
    Description = factor(Description, levels = row_hclust_labels),
    group = fct_relevel(group, col_hclust_labels)
  )

# Calculate jaccard similarity matrix
gset_collection <- foreach(i = levels(res_top$Description)) %do% {
  res_top %>% filter(Description == i) %>% pull(core_enrichment) %>% 
    paste(collapse = '/') %>% 
    strsplit('/') %>% unlist %>% unique
}
names(gset_collection) <- levels(res_top$Description)

use_terms <- res_top$Description %>% unique
jc_matrix <- pairwise_terms_jc(gset_collection[use_terms])
```

#### Results table
```{r endothelial-marker-gsea-c2reactome_gobp-table}
gse_df %>% 
  filter(p.adjust < p_thrs) %>% 
  arrange(desc(p.adjust)) %>% 
  dplyr::select(group, Description:p.adjust, Count:GeneRatio, core_enrichment)  %>%
  dplyr::rename(site = group) %>% 
  datatable(., 
            rownames = FALSE, 
            filter = 'top', 
            caption = 'Results from gene-set enrichment analysis',
            extensions = 'Buttons', 
            options = list(
              dom = 'Blfrtip',
              buttons = c('csv', 'excel')
            ))  %>% 
  formatRound(c('enrichmentScore', 'NES', 'GeneRatio'), digits = 2) %>%
  formatSignif(c('pvalue', 'p.adjust'), digits = 3)
```

#### Number of gene sets
```{r endothelial-marker-gsea-c2reactome_gobp-ngset}
gse_df %>% 
  filter(p.adjust < p_thrs) %>% 
  group_by(group) %>% 
  summarise(n = n()) %>% 
  ggplot(aes(x=group, y=n)) +
  geom_bar(stat="identity", color = 'black')+
  geom_text(aes(label=n), vjust=1.4, color="white", size=geom_text_size)+
  labs(
    x = 'Site',
    y = 'Number of gene sets',
    title = 'Number of significant gene sets by site',
    caption = paste("Adjusted P value <", p_thrs)
    
  )
```

#### Gene-set similarity matrix
Heatmap showing the pair-wise similarity matrix of top `r topn` gene sets with an adjusted P value < `r p_thrs` in each gene group. Heatmap colors represent the Jaccard similarity coefficient.

```{r endothelial-marker-gsea-c2reactome_gobp-jcmatrix, fig.width=7.2, fig.asp=1.4}
col_fun <- colorRamp2(seq(0, 1, length.out = 4), brewer.pal(4, "GnBu"))
n_split <- 9

use_mat <- jc_matrix
colnames(use_mat) <- colnames(use_mat) %>% clean_msigdb_names %>% tolower() %>% str_to_title %>% ep_str_wrap(70)
rownames(use_mat) <- rownames(use_mat) %>% clean_msigdb_names %>% tolower() %>% str_to_title %>% ep_str_wrap(70)

ha_top <- HeatmapAnnotation(
  foo = anno_block(
    # labels = c("Translation", "Cell division"),
    labels = 1:n_split,
    labels_gp = gpar(col = "black", fontsize = 8),
    gp = gpar(lwd = 0, lty = 0))
  )

ht <- Heatmap(
  use_mat,
  name = 'Jaccard index',
  column_split = n_split,
  row_split = n_split,
  column_title = NULL,
  row_title = NULL,
  col = col_fun,
  show_column_dend = FALSE,
  show_column_names = FALSE,
  border = TRUE,
  top_annotation = ha_top,
  heatmap_legend_param = list(title_gp = gpar(fontsize = 8),labels_gp = gpar(fontsize = 8)),
  width = unit(7, "cm"))

ht = draw(ht)

cat("\n\n")

# Extract term order according to hclust from similarity heatmap
row_order <- row_order(ht) %>% set_names(1:n_split)
term_similarity_clusters <- foreach(i=names(row_order), .combine = rbind) %do% {
  data.frame(
    jc_sim_cl = i, 
    row_order = row_order[[i]], 
    Description =  rownames(jc_matrix)[row_order[[i]]]
  )
}

# Add similarity order to data for dot plot
res_top %<>% left_join(term_similarity_clusters)
```



#### Dotplots
The plots below show the top `r topn` gene sets with an adjusted P value < `r p_thrs` in each gene group
```{r endothelial-marker-gsea-c2reactome_gobp-dotplot, fig.width = 4.2, fig.asp = 1.29}
use_linewidth <- 1/2.141959
x_max <- res_top$NES %>% abs %>% max()

res_top %>%
  # Combine similarity clusters
  mutate(
    # jc_sim_cl = ifelse(jc_sim_cl %in% c('1'), 'Translation', jc_sim_cl),
    # jc_sim_cl = ifelse(jc_sim_cl %in% c('3','5','6'), 'Energy\ngeneration', jc_sim_cl),
    # jc_sim_cl = ifelse(jc_sim_cl %in% c('4', '8'), 'Cell stress', jc_sim_cl),
    # jc_sim_cl = factor(
    #   jc_sim_cl,
    #   levels = c('Translation', 'Energy\ngeneration', 'Cell stress', '2', '7', '9')
    #   ),

    p.adjust = ifelse(p.adjust > 0.05, 1, p.adjust),
    p.adjust = -log10( p.adjust),
    p.adjust = ifelse(p.adjust >= 10, 10, p.adjust),
    group = recode(group, primary_tumor = "PT", lung = "Lung", liver = "Liver"),
    group = factor(group, levels = c('PT', 'Lung', 'Liver'))
  ) %>% 
  filter(group == 'Liver') %>% 
  ggplot(aes(NES, Description, color = NES, size = p.adjust)) +
  geom_point() +
  facet_grid(rows = vars(jc_sim_cl), cols = vars(group), scale = 'free_y', space = 'free_y') +
  scale_x_continuous(limits = c(-x_max, x_max),
                        expand = expansion(mult = c(0.08, 0.08))) +
  scale_y_discrete(labels = default_labeller(50)) +
  geom_vline(xintercept = 0, lty = 3) +
  scale_size_continuous(range=c(0.3, 2.5)) +
  scale_color_gradient2(trans = "reverse") +
  labs(
    size = expression(paste("lo", g[10],"(Adjusted P value)")),
    color = "NES"
  ) +
  theme(
    panel.border = element_rect(color = "black", fill = NA, size = use_linewidth/2),
    
    axis.line = element_blank(),
    axis.ticks = element_line(color = "black", size = use_linewidth/2),
    axis.text.y = element_text(size = 4),
    axis.text.x = element_text(size = 4),
    
    legend.text = element_text(size = 4), 
    legend.title = element_text(size = 4),
    legend.key.width = unit(0.2, "cm"),
    
    strip.background = element_blank()
  )
  
```





## Knitr:template
```{r knitr-template-rnaseq-deg-edger, eval = TRUE}
# knit template: https://workflowr.github.io/workflowr/articles/wflow-07-common-code.html
src <- knit_expand(
  file = here::here("analysis/templates/rnaseq-deg-edger.Rmd")
  )
```
`r knit(text = src)`



