---
title: "10x RNA-seq analysis of huNSG cancer"
subtitle: "Demultiplexing with hashtag oligos (HTOs)"
author: "Francesc Castro-Giner"
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
params:
  date: '`r format(Sys.Date(), "%B %d, %Y")`'
  cellranger_count_dir: ./data/rnaseq/p26532_o34980/pipelines/cellranger_featurebarcodes
  hash_annot: ./data/rnaseq/p26532_o34980/rawdata/featurebarcodes.csv
  sample_annot: ./data/rnaseq/p26532_o34980/rawdata/samples.csv
  ncores: 8
---

## Load libraries, additional functions and data

Setup environment

```{r setup, include=TRUE, warning=FALSE, message=FALSE}
knitr::opts_chunk$set(results='asis', echo=TRUE, message=FALSE, warning=FALSE, error=FALSE, fig.align = 'center', fig.width = 3.5, fig.asp = 0.618, dpi = 600, dev = c("png", "pdf"))

options(stringsAsFactors = FALSE)

set.seed(1011110)
```

Load packages

```{r load-libraries}
library(tidyverse)
library(showtext)
library(foreach)
library(DT)
library(knitr)
library(kableExtra)
library(cowplot)
library(colorblindr)
library(ggbeeswarm)
library(arsenal)

library(scater)
library(DropletUtils)
library(Seurat)
library(scran)
library(scDblFinder)
library(BiocSingular)

# library(ComplexHeatmap)
# library(circlize)
# library(RColorBrewer)
# library(DT)
```

Load ggplot theme

```{r ggplot-theme}
source("./configuration/rmarkdown/ggplot_theme.R")
```

Clean files generated in previous runs
```{r clean-files}
rmd_file <- current_input()
if(!is.null(rmd_file)) {
  figures_dir <- file.path('./docs/figure',rmd_file)
  if(dir.exists(figures_dir)) {
    unlink(file.path(figures_dir, "*"))
  }
}

```

## Introduction

## Description

## Data wrangling

Load hashtag metadata
```{r load-hashtag-metadata}
hash_annot <- read_csv(params$hash_annot) %>% dplyr::select(sample_alias, fb_id, additional_fields)

# Split additional fields
x <- str_split(hash_annot$additional_fields, ";")
add_fields <- foreach(y = x, .combine = rbind) %do% {
  res <- str_split(y, "=", simplify = T)
  res[,2] %>% set_names(res[,1])
}
hash_annot <- cbind(hash_annot, add_fields) %>% 
  dplyr::select(-additional_fields) %>% 
  rename('sample_alias' = 'Sample')

# There is a problem in the field of sample
hash_annot <- hash_annot %>% 
  mutate(Sample = ifelse(Sample == 'NSG lungs and livers',
                         'huNSG_ms3_Plate_4120_1',
                         Sample))

```

Load sample metadata
```{r load-sample-metadata}
sample_annot <- read_csv(params$sample_annot) %>% 
  dplyr::select(sample_alias, mouse_id) %>% 
  rename('sample_alias' = 'Sample')
  
hash_annot <- hash_annot %>% 
  left_join(sample_annot)
```

Load SCE file information
```{r list-sce}
# Load files, using system instead of list.files for speed reasons
sce_files <- system(paste("find", params$cellranger_count_dir, "-maxdepth 2 -type f -regex '.*/sce[.]rds$'"), intern = TRUE) %>% sort
names(sce_files) <- sce_files %>% dirname %>% basename
```

Load perCellQCMetrics from SCE files
```{r load-perCellQCMetrics}
use_sce_file <- sce_files[1]
cell_qc_metrics <- foreach(use_sce_file = sce_files, .combine = rbind) %do% {
  sce <- readRDS(use_sce_file)
  colData(sce)
} %>% data.frame
rm(sce)
```


## Quality control (by batch)

### Selected QC metrics

We identify low-quality cells using the following QC metrics :

-   Library size : the total sum of UMI counts across all relevant features. Cells with small library sizes are of low quality as the RNA has been lost at some point during library preparation.
-   Number of detected features : number of genes with non-zero counts for that cell. Any cell with very few expressed genes is likely to be of poor quality as the diverse transcript population has not been successfully captured.
-   Proportion of mitochondrial genes : High proportions are indicative of poor-quality cells (Islam et al. 2014; Ilicic et al. 2016), presumably because of the loss of cytoplasmic RNA from perforated cells.

### Detect low quality cells

We use adaptive thresholds for each selected QC metrics. This assumes that most of the dataset consists of high quality cells and identify cells that are outliers for the selected QC metrics based on the median absolute deviation (MAD) from the median value of each metric across all cells. By default, we consider a value to be an outlier if it is more than 3 MADs from the median in the "problematic" direction. This is loosely motivated by the fact that such a filter will retain 99% of non-outlier values that follow a normal distribution.

In cases were multiple batches are involved, each with different sequencing coverage, we should consider to apply the adaptive threshold to each batch separately. That said, the use of batch= involves the stronger assumption that most cells in each batch are of high quality. If an entire batch failed, outlier detection will not be able to act as an appropriate QC filter for that batch. In such cases, it is better to compute a shared median and MAD from the other batches and use those estimates to obtain an appropriate filter threshold for cells in the problematic batches.

```{r qc-metrics-batch}
discard_reasons <-  perCellQCFilters(
  cell_qc_metrics,
  sum.field = "sum",
  detected.field = "detected",
  sub.fields = c("subsets_Mito_percent"),
  batch=cell_qc_metrics$Sample,
  nmads = 3
)
cell_qc_metrics <- cbind(dplyr::select(cell_qc_metrics, Sample:total), discard_reasons)

thresholds <- cbind(
  `Library size` = attr(discard_reasons$low_lib_size, "thresholds")['lower',],
  `Number of detected features size` = attr(discard_reasons$low_n_features, "thresholds")['lower',],
  `Mitochondrial proportion` = attr(discard_reasons$high_subsets_Mito_percent, "thresholds")['higher',]
) %>% data.frame(check.names = FALSE) %>% rownames_to_column('Sample')

res <- rbind(
  c(`Reason` = 'Low library size', `Number of cells` = sum(discard_reasons$low_lib_size)),
  c(`Reason` = 'Low number of detected features size', `Number of cells` = sum(discard_reasons$low_n_features)),
  c(`Reason` = 'High mitochondrial proportion', `Number of cells` = sum(discard_reasons$high_subsets_Mito_percent)),
  c(`Reason` = 'Total removed', `Number of cells` = sum(discard_reasons$discard)),
  c(`Reason` = 'Cells remaining', `Number of cells` = sum(!discard_reasons$discard))
) %>% data.frame(check.names = FALSE) %>% 
  mutate(
    `Number of cells` = as.numeric(`Number of cells`),
    `Percentage of cells` = round(100*(`Number of cells` / nrow(cell_qc_metrics)), 2)
  ) %>% 
  dplyr::select(Reason, `Number of cells`, `Percentage of cells`)


res %>%
  kbl(caption = 'Number of cells removed') %>%
  kable_paper(bootstrap_options = c("striped", "hover", "condensed"), full_width = F)

```

```{r qc-metrics-thresholds-per-batch}
tab1 <- tableby(Sample ~ low_lib_size + low_n_features + high_subsets_Mito_percent + discard, data=cell_qc_metrics)
mylabels <- list(low_lib_size = "Low UMI counts", low_n_features = "Low number of detected features size", high_subsets_Mito_percent = "High mitochondrial proportion", discard = "Total removed")
sum_tab <- summary(tab1, labelTranslations = mylabels, cat.simplify=TRUE) %>% as.data.frame
colnames(sum_tab)[1] <- 'var'
sum_tab <- sum_tab %>% column_to_rownames('var') %>% t()
sum_tab %>%
  kbl(caption = 'Number of cells removed per sample') %>%
  kable_paper(bootstrap_options = c("striped", "hover", "condensed"), full_width = F)
```

```{r qc-metrics-per-batch}
thresholds %>%
  kbl(caption = 'Thresholds applied by batch') %>%
  kable_paper(bootstrap_options = c("striped", "hover", "condensed"), full_width = F)
```

```{r qc-stats-per-batch}
tab_all <- cell_qc_metrics %>% 
  group_by(Sample) %>% 
  summarise(`Library size` = median(sum), `Number of detected features` = median(detected), `Mitochondrial proportion` = median(subsets_Mito_percent))
tab_pass <- cell_qc_metrics %>% 
  filter(!discard) %>% 
  group_by(Sample) %>% 
  summarise(`Library size` = median(sum), `Number of detected features` = median(detected), `Mitochondrial proportion` = median(subsets_Mito_percent))

tab_full <- full_join(tab_all, tab_pass, by = 'Sample')

colnames(tab_full) <- colnames(tab_full) %>% gsub("\\.[xy]", "", .)

tab_full %>%
  kbl(caption = 'Median values for each quality metric') %>%
  kable_paper(bootstrap_options = c("striped", "hover", "condensed"), full_width = F) %>%
  add_header_above(c(" ", "All cells" = 3, "Filtered" = 3))
```

### Tresholds

```{r qc-metrics-per-batch-2}
thresholds %>%
  kbl(caption = 'Thresholds applied by batch') %>%
  kable_paper(bootstrap_options = c("striped", "hover", "condensed"), full_width = F)
```

```{r qc-metrics-batch-fig-thrs, fig.width=7.2, fig.asp=1, eval = FALSE}
thresholds %>% 
  pivot_longer(-Sample, values_to = 'Threshold') %>% 
  ggplot(aes(Sample, Threshold)) +
  geom_col() +
  facet_wrap(vars(name), ncol = 1, scales = 'free_y') +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(
    x = '',
    title = 'Thresholds applied by batch'
  )
```



### Diagnostic plots

```{r qc-metrics-fig-sum, fig.width=7.2, fig.asp=1}
cell_qc_metrics %>% 
  arrange(discard) %>% 
  ggplot(aes(Sample, sum, color = discard)) + 
  geom_violin(colour = "gray60", alpha = 0.2, scale = "width", width = 0.8) + 
  geom_quasirandom(width=0.4, groupOnX=TRUE, bandwidth=1, size = 1, alpha = 0.4) +
  # geom_hline(yintercept = attr(discard_reasons$low_lib_size, "thresholds")[['lower']], lty = 3) + 
  scale_color_OkabeIto(order = c(2, 1)) +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(
    x = '',
    y = 'UMIs counts',
    title = 'Number of UMIs'
  )
```

```{r qc-metrics-fig-detected, fig.width=7.2, fig.asp=1}
cell_qc_metrics %>% 
  arrange(discard) %>% 
  ggplot(aes(Sample, detected, color = discard)) + 
  geom_violin(colour = "gray60", alpha = 0.2, scale = "width", width = 0.8) + 
  geom_quasirandom(width=0.4, groupOnX=TRUE, bandwidth=1, size = 1, alpha = 0.4) +
  # geom_hline(yintercept = attr(discard_reasons$low_n_features, "thresholds")[['lower']], lty = 3) + 
  scale_color_OkabeIto(order = c(2, 1)) +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(
    x = '',
    y = 'Detected features',
    title = 'Number of genes'
  )
```

```{r qc-metrics-fig-subsets_Mito_percent, fig.width=7.2, fig.asp=1}
cell_qc_metrics %>% 
  arrange(discard) %>% 
  ggplot(aes(Sample, subsets_Mito_percent, color = discard)) + 
  geom_violin(colour = "gray60", alpha = 0.2, scale = "width", width = 0.8) + 
  geom_quasirandom(width=0.4, groupOnX=TRUE, bandwidth=1, size = 1, alpha = 0.4) +
  # geom_hline(yintercept = attr(discard_reasons$high_subsets_Mito_percent, "thresholds")[['higher']], lty = 3) + 
  scale_color_OkabeIto(order = c(2, 1)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(
    x = '',
    y = 'Percentage of mitochondrial genes',
    title = 'Mitochondrial proportion'
  )
```

```{r qc-metrics-fig-subsets_Ribo_percent, fig.width=7.2, fig.asp=1}
cell_qc_metrics %>% 
  arrange(discard) %>% 
  ggplot(aes(Sample, subsets_Ribo_percent, color = discard)) + 
  geom_violin(colour = "gray60", alpha = 0.2, scale = "width", width = 0.8) + 
  geom_quasirandom(width=0.4, groupOnX=TRUE, bandwidth=1, size = 1, alpha = 0.4) +
  scale_color_OkabeIto(order = c(2, 1)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(
    x = '',
    y = 'Percentage of ribosomal genes',
    title = 'Ribosomal proportion'
  )
```

```{r qc-metrics-fig-sum_vs_mito}
cell_qc_metrics %>% 
  # arrange(discard) %>% 
  ggplot(aes(sum, subsets_Mito_percent, color = discard)) + 
  geom_point(alpha = 0.3, size = 1) +
  scale_color_OkabeIto(order = c(2, 1)) +
  labs(
    x = 'UMIs counts',
    y = 'Percentage of mitochondrial genes'
  )
```

```{r qc-metrics-fig-detected_vs_mito}
cell_qc_metrics %>% 
  # arrange(discard) %>% 
  ggplot(aes(detected, subsets_Mito_percent, color = discard)) + 
  geom_point(alpha = 0.3, size = 1) +
  scale_color_OkabeIto(order = c(2, 1)) +
  labs(
    x = 'Detected features',
    y = 'Percentage of mitochondrial genes'
  )
```

```{r qc-metrics-fig-sum_vs_detected}
cell_qc_metrics %>% 
  # arrange(discard) %>% 
  ggplot(aes(sum, detected, color = discard)) + 
  geom_point(alpha = 0.3, size = 1) +
  scale_color_OkabeIto(order = c(2, 1)) +
  labs(
    x = 'UMIs counts',
    y = 'Detected features'
  ) + 
  geom_smooth()

```




### Remove cells
```{r qc-metrics-remove-cells}
use_sce_file <- sce_files[1]
r.rm <- foreach(use_sce_file = sce_files) %do% {
  sce <- readRDS(use_sce_file)
  filtered_sce_file <- gsub("sce.rds", "sce_filtered.rds", use_sce_file)
  x <- colData(sce) %>% data.frame %>% left_join(cell_qc_metrics %>% dplyr::select(Sample, Barcode, discard))
  # Add GFP and mCherry expression to colData
  # colData(sce) <- cbind(data.frame(colData(sce)), counts(sce[c("GFP", "mCherry"),]) %>% as.matrix %>% t) %>% DataFrame()
  sce <- sce[,!x$discard]
  saveRDS(sce, filtered_sce_file)
}
rm(sce)
sce_filtered_files <- system(paste("find", params$cellranger_count_dir, "-maxdepth 2 -type f -regex '.*/sce_filtered[.]rds$'"), intern = TRUE) %>% sort
names(sce_filtered_files) <- names(sce_files)
```


## Demultiplexing with hashtag oligos (HTOs) and doublet detection

Demultiplexing results based on antibodies enrichment and filtering of the data based on HTO assignment. Here we filter out (1) cells with two HTO assigned (doublets) and (2) cells with HTO present that was not used in the experiment. 

One limitation of removing doublets based on HTO is that doublets marked with the same HTO are not detected. Here we are also marking cells who have a similar profile to doublets (guilt by association) using scDblFinder::recoverDoublets. While this might not have an effect in samples with cancer only (only one cell type), in an heterogeneous population like the niche, predicted doublets implying multiple cell types are still possible. 

```{r dmx-hto, fig.width=10.2, fig.asp=0.9, eval = TRUE}
sce_sample_name <- names(sce_filtered_files)[1]
dmx_stats <- foreach(sce_sample_name = names(sce_filtered_files), .combine = rbind) %do% {
  cat("###", sce_sample_name, "\n\n") 
  use_sce_file <- sce_filtered_files[[sce_sample_name]]
  sce <- readRDS(use_sce_file)
  sce$sample_id <- paste(sce$Sample, sce$Barcode, sep = '_')
  colnames(sce) <- sce$sample_id
  
  feat_mat <- counts(sce[rowData(sce)$type == 'Gene Expression',])
  hto_mat <- counts(sce[rowData(sce)$type == 'Antibody Capture',])
  colnames(feat_mat) <- sce$sample_id 
  colnames(hto_mat) <- sce$sample_id 
  
  # Generate Seurat object
  data_hashtag <- CreateSeuratObject(counts = feat_mat)
  # Normalize RNA data with log normalization
  data_hashtag <- NormalizeData(data_hashtag, verbose = FALSE)
  # Find and scale variable features
  data_hashtag <- FindVariableFeatures(data_hashtag, selection.method = "mean.var.plot", verbose = FALSE)
  data_hashtag <- ScaleData(data_hashtag, features = VariableFeatures(data_hashtag), verbose = FALSE)

  # Adding HTO data as an independent assay
  data_hashtag[["HTO"]] <- CreateAssayObject(counts = hto_mat)
  # Normalize HTO data, here we use centered log-ratio (CLR) transformation
  data_hashtag <- NormalizeData(data_hashtag, assay = "HTO", normalization.method = "CLR", verbose = FALSE)

  # Demultiplex cells based on HTO enrichment
  data_hashtag <- HTODemux(data_hashtag, assay = "HTO", positive.quantile = 0.99, kfunc = "kmeans", verbose = FALSE)
  
  # Group cells based on the max HTO signal
  # cat("#### HTO enrichment\n\n")
  # cat("Visualize enrichment for selected HTOs with ridge plots\n\n")
  # Idents(data_hashtag) <- "HTO_maxID"
  # RidgePlot(data_hashtag, assay = "HTO", features = rownames(data_hashtag[["HTO"]]), ncol = 2) %>% print
  # cat("\n\n")
  
  # Compare number of UMIs for singlets, doublets and negative cells
  # cat("#### UMIs comparison\n\n") 
  # cat("Compare number of UMIs for singlets, doublets and negative cells\n\n")
  # Idents(data_hashtag) <- "HTO_classification.global"
  # VlnPlot(data_hashtag, features = "nCount_RNA", pt.size = 0.1, log = TRUE) %>% print
  # cat("\n\n")
  
  # Compare number of UMIs for singlets, doublets and negative cells
  cat("#### HTO heatmap\n\nHeatmap with the expression of the antibodies across the cells. The Negative barcodes have a negligible background signal for any of the antobodies. Multiplets express more than one antibody.\n\n") 
  HTOHeatmap(data_hashtag, assay = "HTO", ncells = 5000) %>% print
  cat("\n\n")
  
  # cat("#### HTO clustering\n\n") 
  # cat("Generate a two dimensional tSNE embedding for HTOs.\n\n")
  # # Calculate a tSNE embedding of the HTO data
  # data_hashtag_HTO <- subset(data_hashtag, idents = "Negative", invert = TRUE)
  # DefaultAssay(data_hashtag_HTO) <- "HTO"
  # data_hashtag_HTO <- ScaleData(data_hashtag_HTO, features = rownames(data_hashtag_HTO), verbose = FALSE)
  # data_hashtag_HTO <- RunPCA(data_hashtag_HTO, features = rownames(data_hashtag_HTO), reduction.name = "pcahto", reduction.key = "pcahto_", approx = FALSE, verbose = FALSE)
  # data_hashtag_HTO <- RunTSNE(data_hashtag_HTO, reduction = "pcahto", reduction.key = "htoTSNE_", reduction.name = "tsne_hto", check_duplicates = FALSE)
  # DimPlot(data_hashtag_HTO, group.by = "hash.ID", cols = palette_OkabeIto[c(6, 1:5, 7, 8)]) %>% print
  # cat("\n\n")
  # rm(data_hashtag_HTO)
  
  # cat("#### RNA clustering,\n\n") 
  # cat("Cluster and visualize cells using the usual scRNA-seq workflow, and examine for the potential presence of batch effects. Using top-1000 features for clustering.\n\n")
  # data_hashtag_singlet <- subset(data_hashtag, idents = "Singlet")
  # # Run PCA on most variable features
  # data_hashtag_singlet <- FindVariableFeatures(data_hashtag_singlet, selection.method = "mean.var.plot", verbose = FALSE)
  # # Scaling RNA data, we only scale the variable features here for efficiency
  # data_hashtag_singlet <- ScaleData(data_hashtag_singlet, features = VariableFeatures(data_hashtag_singlet), verbose = FALSE)
  # # Run PCA
  # data_hashtag_singlet <- RunPCA(data_hashtag_singlet, features = VariableFeatures(data_hashtag_singlet), verbose = FALSE)
  # # We select the top 10 PCs for clustering and tSNE based on PCElbowPlot
  # # data_hashtag_singlet <- RunTSNE(data_hashtag_singlet, reduction = "pca", dims = 1:10, verbose = FALSE)
  # data_hashtag_singlet <- RunUMAP(data_hashtag_singlet, reduction = "pca", dims = 1:10, verbose = FALSE)
  # # Projecting singlet identities on TSNE visualization
  # DimPlot(data_hashtag_singlet, group.by = "HTO_classification", cols = palette_OkabeIto) %>% print
  # cat("\n\n")
  # rm(data_hashtag_singlet)
  
  # Data post-processing
  # Annotate sce with HTO classification
  colData(sce) <- colData(sce) %>% 
    data.frame %>% 
    left_join(data_hashtag@meta.data %>% rownames_to_column('sample_id') %>%  dplyr::select(sample_id, starts_with("HTO"))) %>% 
    mutate(HTO_doublet = HTO_classification.global == 'Doublet') %>% 
    dplyr::select(-sample_id) %>% 
    DataFrame
  
  # Detect doublets using guilt by association for unmarked doublets
  cat("#### Additional doublet detections\n\n")
  cat("One limitation of removing doublets based on HTO is that doublets marked with the same HTO are not detected. Here we are also marking cells who have a similar profile to doublets (guilt by association) using scDblFinder::recoverDoublets. While this might not have an effect in samples with cancer only (only one cell type), in an heterogeneous population like the niche, predicted doublets implying multiple cell types are still possible.\n\n")
  sce.hash <- logNormCounts(sce)
  dec.hash <- modelGeneVar(sce.hash)
  top.hash <- getTopHVGs(dec.hash, n=1000)
  sce.hash <- runPCA(sce.hash, subset_row=top.hash, ncomponents=20)
  hashed.doublets <- recoverDoublets(sce.hash, use.dimred="PCA", doublets=sce.hash$HTO_doublet, samples=table(sce.hash$HTO_maxID))
  sce.hash$proportion <- hashed.doublets$proportion
  sce.hash$predicted <- hashed.doublets$predicted
  sce.hash <- runTSNE(sce.hash, dimred="PCA")
  
  gridExtra::grid.arrange(
    plotTSNE(sce.hash, colour_by="proportion") + ggtitle("Doublet proportions"),
    plotTSNE(sce.hash, colour_by="HTO_doublet", point_alpha = 0.6) + ggtitle("Known doublets"),
    ggcells(sce.hash) +
        geom_point(aes(x=TSNE.1, y=TSNE.2), color="grey") +
        geom_point(aes(x=TSNE.1, y=TSNE.2), color="red", 
            data=function(x) x[x$predicted,]) +
        ggtitle("Predicted intra-sample doublets"),
    ncol=2        
  )
  cat("\n\n")
  
  rm(sce.hash)
  sce$recoverDoublets.proportion <- hashed.doublets$proportion
  sce$recoverDoublets.predicted <- hashed.doublets$predicted
  
  # Add HTO annotation
  colData(sce) <- colData(sce) %>% data.frame %>%
    left_join(hash_annot %>% rename("fb_id" = "HTO_classification")) %>%
    DataFrame
  
  # Filter sce, selecting only singlets
  sce_f <- sce[,sce$HTO_classification.global == 'Singlet' & !sce$recoverDoublets.predicted]

  # Filter sce, removing cells without cell type. It can be the case for hash detected and not used
  # sce_f <- sce_f[,!is.na(sce_f$cell_type)]
  
  # Save filtered SCE
  dmx_sce_file <- gsub("sce_filtered.rds", "sce_dmx.rds", use_sce_file)
  saveRDS(sce_f, dmx_sce_file)
  
  # Clean data
  rm(sce_f)
  
  # Return demultiplexing stats
  colData(sce) %>% data.frame
  

}
```

## HTO classification
```{r dmx-hto-classification-table}
dmx_stats_tab <- dmx_stats %>% 
  mutate(singlets_final = ifelse(HTO_classification.global == 'Singlet' & recoverDoublets.predicted == FALSE, TRUE, FALSE))

tab1 <- tableby(Sample ~ HTO_classification.global + recoverDoublets.predicted + singlets_final, data=dmx_stats_tab, cat.simplify=TRUE)
mylabels <- list(HTO_classification.global = "HTO classification", recoverDoublets.predicted = "Predicted doublets", singlets_final = "Final number of cells")
res_tab <- summary(tab1, labelTranslations = mylabels) %>% as.data.frame
colnames(res_tab)[1] <- 'var'
res_tab <- res_tab %>% column_to_rownames('var') %>% t()
colnames(res_tab) <- colnames(res_tab) %>% gsub(".*;", "", .) %>% gsub("\\*\\*", "", .)
res_tab %>%
  kbl(caption = 'Classification of cells based on HTO') %>%
  kable_paper(bootstrap_options = c("striped", "hover", "condensed"), full_width = F)

```


## Diagnostic plots after demultiplexing

Diagnostic plots combining all bat
```{r}
sce_dmxfiles <- system(paste("find", params$cellranger_count_dir, "-maxdepth 2 -type f -regex '.*/sce_dmx[.]rds$'"), intern = TRUE) %>% sort
cell_qc_metrics <- foreach(use_sce_file = sce_dmxfiles, .combine = rbind) %do% {
  sce <- readRDS(use_sce_file)
  colData(sce)
} %>% data.frame
```


#### By cell type
```{r dmx-qc-metrics-fig-sum, fig.width=7.2, fig.asp=1}
cell_qc_metrics %>% 
  ggplot(aes(Sample, sum, color = cell_type)) + 
  geom_violin(colour = "gray60", alpha = 0.2, scale = "width", width = 0.8) + 
  geom_quasirandom(width=0.4, groupOnX=TRUE, bandwidth=1, size = 1, alpha = 0.4) +
  scale_color_OkabeIto(order = c(3, 6)) +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(
    x = '',
    y = 'UMIs counts',
    title = 'Number of UMIs'
  )
```

```{r dmx-qc-metrics-fig-detected, fig.width=7.2, fig.asp=1}
cell_qc_metrics %>% 
  ggplot(aes(Sample, detected, color = cell_type)) + 
  geom_violin(colour = "gray60", alpha = 0.2, scale = "width", width = 0.8) + 
  geom_quasirandom(width=0.4, groupOnX=TRUE, bandwidth=1, size = 1, alpha = 0.4) +
  scale_color_OkabeIto(order = c(3, 6)) +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(
    x = '',
    y = 'Detected features',
    title = 'Number of genes'
  )
```

```{r dmx-qc-metrics-fig-subsets_Mito_percent, fig.width=7.2, fig.asp=1}
cell_qc_metrics %>% 
  ggplot(aes(Sample, subsets_Mito_percent, color = cell_type)) + 
  geom_violin(colour = "gray60", alpha = 0.2, scale = "width", width = 0.8) + 
  geom_quasirandom(width=0.4, groupOnX=TRUE, bandwidth=1, size = 1, alpha = 0.4) +
  scale_color_OkabeIto(order = c(3, 6)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(
    x = '',
    y = 'Percentage of mitochondrial genes',
    title = 'Mitochondrial proportion'
  )

```



#### By site and mouse ID
```{r dmx-qc-metrics-fig-sum-site_mouse, fig.width=7.2, fig.asp=1}
cell_qc_metrics %>% 
  ggplot(aes(site, sum, color = cell_type)) + 
  geom_violin(colour = "gray60", alpha = 0.2, scale = "width", width = 0.8) + 
  geom_quasirandom(width=0.4, groupOnX=TRUE, bandwidth=1, size = 1, alpha = 0.4) +
  scale_color_OkabeIto(order = c(3, 6)) +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(
    x = '',
    y = 'UMIs counts',
    title = 'Number of UMIs'
  ) +
  facet_wrap(facets = vars(mouse_id, Sample), scales = 'free_x')
```

```{r dmx-qc-metrics-fig-detected-site_mouse, fig.width=7.2, fig.asp=1}
cell_qc_metrics %>% 
  ggplot(aes(site, detected, color = cell_type)) + 
  geom_violin(colour = "gray60", alpha = 0.2, scale = "width", width = 0.8) + 
  geom_quasirandom(width=0.4, groupOnX=TRUE, bandwidth=1, size = 1, alpha = 0.4) +
  scale_color_OkabeIto(order = c(3, 6)) +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(
    x = '',
    y = 'Detected features',
    title = 'Number of genes'
  ) +
  facet_wrap(facets = vars(mouse_id, Sample), scales = 'free_x')

```

```{r dmx-qc-metrics-fig-subsets_Mito_percent-site_mouse, fig.width=7.2, fig.asp=1}
cell_qc_metrics %>% 
  ggplot(aes(site, subsets_Mito_percent, color = cell_type)) + 
  geom_violin(colour = "gray60", alpha = 0.2, scale = "width", width = 0.8) + 
  geom_quasirandom(width=0.4, groupOnX=TRUE, bandwidth=1, size = 1, alpha = 0.4) +
  scale_color_OkabeIto(order = c(3, 6)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(
    x = '',
    y = 'Percentage of mitochondrial genes',
    title = 'Mitochondrial proportion'
  ) +
  facet_wrap(facets = vars(mouse_id, Sample), scales = 'free_x')
```



## Normalization, variance modelling and cell cycle annotation

Here we perform normalization using scran::quickCluster and dimensionality reduction at batch level.

```{r normalization-dec, fig.width = 10.2, fig.asp=1}
sce_dmxfiles <- system(paste("find", params$cellranger_count_dir, "-maxdepth 2 -type f -regex '.*/sce_dmx[.]rds$'"), intern = TRUE) %>% sort
names(sce_dmxfiles) <-sce_dmxfiles %>% dirname %>% basename
mm.pairs <- readRDS(system.file("exdata", "mouse_cycle_markers.rds", package="scran"))
sce_sample_name <- names(sce_dmxfiles)[1]
for(sce_sample_name in names(sce_dmxfiles)) {
  cat("###", sce_sample_name, "\n\n") 
  use_sce <- readRDS(sce_dmxfiles[[sce_sample_name]])
  use_sce <- use_sce[rowData(use_sce)$type == 'Gene Expression',]
  # Normalization
  clust.sce <- quickCluster(use_sce, BPPARAM=BiocParallel::MulticoreParam(params$ncores)) 
  use_sce <- computeSumFactors(use_sce, cluster=clust.sce, BPPARAM=BiocParallel::MulticoreParam(params$ncores))
  use_sce <- logNormCounts(use_sce)
  # Cell cycle assignment
  rowData(use_sce)$ensembl_gene.version <- rownames(use_sce)
  rowData(use_sce)$ensembl_gene <-  rowData(use_sce)$ensembl_gene.version %>% gsub("\\..*", "", .)
  cc_assignments <- cyclone(use_sce, mm.pairs, gene.names=rowData(use_sce)$ensembl_gene, BPPARAM=BiocParallel::MulticoreParam(params$ncores))
  use_sce$cell_cycle_phase <- cc_assignments$phases
  colData(use_sce) <- cbind(colData(use_sce), cc_assignments$normalized.scores)
  # Variance modelling
  metadata(use_sce)$dec <- modelGeneVar(use_sce)
  metadata(use_sce)$hvgs <- getTopHVGs(metadata(use_sce)$dec, prop = 0.1)
  # Dimensionality reduction
  use_sce <- runPCA(use_sce, subset_row=metadata(use_sce)$hvgs, ncomponents=50, BSPARAM=RandomParam(), BPPARAM=BiocParallel::MulticoreParam(params$ncores))
  use_sce <- runTSNE(use_sce, dimred="PCA", BPPARAM=BiocParallel::MulticoreParam(params$ncores))
  use_sce <- runUMAP(use_sce, dimred="PCA", BPPARAM=BiocParallel::MulticoreParam(params$ncores))
  g <- buildSNNGraph(use_sce, k=10, use.dimred='PCA', BPPARAM=BiocParallel::MulticoreParam(params$ncores))
  use_sce$cluster_id_batch <- igraph::cluster_walktrap(g)$membership %>% factor
  
  cat("#### TSNE\n\n") 
  gridExtra::grid.arrange(
    plotTSNE(use_sce, colour_by="cluster_id_batch", text_by="cluster_id_batch", point_alpha = 0.1) + 
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) + 
      ggtitle("Cluster ID"),
    plotTSNE(use_sce, colour_by="site", point_alpha = 0.1) + 
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) + 
      ggtitle("Site"),
    plotTSNE(use_sce, colour_by="detected", point_alpha = 0.1) +
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) + 
      ggtitle("Number of detected features"),
    plotTSNE(use_sce, colour_by="subsets_Mito_detected", point_alpha = 0.1) + 
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) + 
      ggtitle("Mitochondrial percentage"),
    plotTSNE(use_sce, colour_by="cell_cycle_phase", point_alpha = 0.1) + 
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) + 
      ggtitle("Cell cycle phase"),
    ncol=2
  )
  cat("\n\n") 
  
  cat("#### UMAP\n\n") 
  gridExtra::grid.arrange(
    plotUMAP(use_sce, colour_by="cluster_id_batch", text_by="cluster_id_batch", point_alpha = 0.1) + 
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) + 
      ggtitle("Cluster ID"),
    plotUMAP(use_sce, colour_by="site", point_alpha = 0.1) + 
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) + 
      ggtitle("Site"),
    plotUMAP(use_sce, colour_by="detected", point_alpha = 0.1) + 
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) + 
      ggtitle("Number of detected features"),
    plotUMAP(use_sce, colour_by="subsets_Mito_detected", point_alpha = 0.1) + 
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) + 
      ggtitle("Mitochondrial percentage"),
    plotUMAP(use_sce, colour_by="cell_cycle_phase", point_alpha = 0.1) + 
      guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) + 
      ggtitle("Cell cycle phase"),
    ncol=2
  )
  cat("\n\n") 
  # Save filtered SCE
  norm_sce_file <- gsub("sce_dmx.rds", "sce_normalized.rds", sce_dmxfiles[[sce_sample_name]])
  saveRDS(use_sce, norm_sce_file)
}
```
